diff -r -U 3 minecraft\net\minecraft\world\BossInfoLerping.java minecraft_patched\net\minecraft\world\BossInfoLerping.java
--- minecraft\net\minecraft\world\BossInfoLerping.java
+++ minecraft_patched\net\minecraft\world\BossInfoLerping.java
@@ -34,6 +34,7 @@
       return this.field_186750_b + (this.field_186766_h - this.field_186750_b) * f;
    }
 
+   @SuppressWarnings("incomplete-switch")
    public void func_186765_a(SPacketUpdateBossInfo p_186765_1_) {
       switch(p_186765_1_.func_186902_b()) {
       case UPDATE_NAME:
