diff -r -U 3 minecraft\net\minecraft\util\IntIdentityHashBiMap.java minecraft_patched\net\minecraft\util\IntIdentityHashBiMap.java
--- minecraft\net\minecraft\util\IntIdentityHashBiMap.java
+++ minecraft_patched\net\minecraft\util\IntIdentityHashBiMap.java
@@ -20,9 +20,9 @@
 
    public IntIdentityHashBiMap(int p_i46830_1_) {
       p_i46830_1_ = (int)((float)p_i46830_1_ / 0.8F);
-      this.field_186818_b = (Object[])(new Object[p_i46830_1_]);
+      this.field_186818_b = (K[])(new Object[p_i46830_1_]);
       this.field_186819_c = new int[p_i46830_1_];
-      this.field_186820_d = (Object[])(new Object[p_i46830_1_]);
+      this.field_186820_d = (K[])(new Object[p_i46830_1_]);
    }
 
    public int func_186815_a(K p_186815_1_) {
@@ -55,9 +55,9 @@
    private void func_186807_d(int p_186807_1_) {
       K[] ak = this.field_186818_b;
       int[] aint = this.field_186819_c;
-      this.field_186818_b = (Object[])(new Object[p_186807_1_]);
+      this.field_186818_b = (K[])(new Object[p_186807_1_]);
       this.field_186819_c = new int[p_186807_1_];
-      this.field_186820_d = (Object[])(new Object[p_186807_1_]);
+      this.field_186820_d = (K[])(new Object[p_186807_1_]);
       this.field_186821_e = 0;
       this.field_186822_f = 0;
 
