diff -r -U 3 minecraft\net\minecraft\entity\item\EntityItem.java minecraft_patched\net\minecraft\entity\item\EntityItem.java
--- minecraft\net\minecraft\entity\item\EntityItem.java
+++ minecraft_patched\net\minecraft\entity\item\EntityItem.java
@@ -70,7 +70,7 @@
    }
 
    protected void func_70088_a() {
-      this.func_184212_Q().func_187214_a(field_184533_c, Optional.<T>absent());
+      this.func_184212_Q().func_187214_a(field_184533_c, Optional.<ItemStack>absent());
    }
 
    public void func_70071_h_() {
@@ -339,7 +339,7 @@
    }
 
    public void func_92058_a(@Nullable ItemStack p_92058_1_) {
-      this.func_184212_Q().func_187227_b(field_184533_c, Optional.<T>fromNullable(p_92058_1_));
+      this.func_184212_Q().func_187227_b(field_184533_c, Optional.fromNullable(p_92058_1_));
       this.func_184212_Q().func_187217_b(field_184533_c);
    }
 
