diff -r -U 3 minecraft\net\minecraft\command\server\CommandTestForBlock.java minecraft_patched\net\minecraft\command\server\CommandTestForBlock.java
--- minecraft\net\minecraft\command\server\CommandTestForBlock.java
+++ minecraft_patched\net\minecraft\command\server\CommandTestForBlock.java
@@ -98,6 +98,6 @@
    }
 
    public List<String> func_184883_a(MinecraftServer p_184883_1_, ICommandSender p_184883_2_, String[] p_184883_3_, @Nullable BlockPos p_184883_4_) {
-      return p_184883_3_.length > 0 && p_184883_3_.length <= 3?func_175771_a(p_184883_3_, 0, p_184883_4_):(p_184883_3_.length == 4?func_175762_a(p_184883_3_, Block.field_149771_c.func_148742_b()):Collections.emptyList());
+      return p_184883_3_.length > 0 && p_184883_3_.length <= 3?func_175771_a(p_184883_3_, 0, p_184883_4_):(p_184883_3_.length == 4?func_175762_a(p_184883_3_, Block.field_149771_c.func_148742_b()):Collections.<String>emptyList());
    }
 }
