diff -r -U 3 minecraft\net\minecraft\entity\monster\EntityShulker.java minecraft_patched\net\minecraft\entity\monster\EntityShulker.java
--- minecraft\net\minecraft\entity\monster\EntityShulker.java
+++ minecraft_patched\net\minecraft\entity\monster\EntityShulker.java
@@ -120,7 +120,7 @@
    protected void func_70088_a() {
       super.func_70088_a();
       this.field_70180_af.func_187214_a(field_184700_a, EnumFacing.DOWN);
-      this.field_70180_af.func_187214_a(field_184701_b, Optional.<T>absent());
+      this.field_70180_af.func_187214_a(field_184701_b, Optional.<BlockPos>absent());
       this.field_70180_af.func_187214_a(field_184702_c, Byte.valueOf((byte)0));
    }
 
@@ -145,9 +145,9 @@
          int i = p_70037_1_.func_74762_e("APX");
          int j = p_70037_1_.func_74762_e("APY");
          int k = p_70037_1_.func_74762_e("APZ");
-         this.field_70180_af.func_187227_b(field_184701_b, Optional.<T>of(new BlockPos(i, j, k)));
+         this.field_70180_af.func_187227_b(field_184701_b, Optional.of(new BlockPos(i, j, k)));
       } else {
-         this.field_70180_af.func_187227_b(field_184701_b, Optional.<T>absent());
+         this.field_70180_af.func_187227_b(field_184701_b, Optional.<BlockPos>absent());
       }
 
    }
@@ -170,7 +170,7 @@
       BlockPos blockpos = (BlockPos)((Optional)this.field_70180_af.func_187225_a(field_184701_b)).orNull();
       if(blockpos == null && !this.field_70170_p.field_72995_K) {
          blockpos = new BlockPos(this);
-         this.field_70180_af.func_187227_b(field_184701_b, Optional.<T>of(blockpos));
+         this.field_70180_af.func_187227_b(field_184701_b, Optional.of(blockpos));
       }
 
       if(this.func_184218_aH()) {
@@ -186,11 +186,11 @@
             if(iblockstate.func_177230_c() == Blocks.field_180384_M) {
                EnumFacing enumfacing = (EnumFacing)iblockstate.func_177229_b(BlockPistonBase.field_176387_N);
                blockpos = blockpos.func_177972_a(enumfacing);
-               this.field_70180_af.func_187227_b(field_184701_b, Optional.<T>of(blockpos));
+               this.field_70180_af.func_187227_b(field_184701_b, Optional.of(blockpos));
             } else if(iblockstate.func_177230_c() == Blocks.field_150332_K) {
                EnumFacing enumfacing3 = (EnumFacing)iblockstate.func_177229_b(BlockPistonExtension.field_176387_N);
                blockpos = blockpos.func_177972_a(enumfacing3);
-               this.field_70180_af.func_187227_b(field_184701_b, Optional.<T>of(blockpos));
+               this.field_70180_af.func_187227_b(field_184701_b, Optional.of(blockpos));
             } else {
                this.func_184689_o();
             }
@@ -326,7 +326,7 @@
 
                if(flag) {
                   this.func_184185_a(SoundEvents.field_187791_eX, 1.0F, 1.0F);
-                  this.field_70180_af.func_187227_b(field_184701_b, Optional.<T>of(blockpos1));
+                  this.field_70180_af.func_187227_b(field_184701_b, Optional.of(blockpos1));
                   this.field_70180_af.func_187227_b(field_184702_c, Byte.valueOf((byte)0));
                   this.func_70624_b((EntityLivingBase)null);
                   return true;
@@ -418,7 +418,7 @@
    }
 
    public void func_184694_g(@Nullable BlockPos p_184694_1_) {
-      this.field_70180_af.func_187227_b(field_184701_b, Optional.<T>fromNullable(p_184694_1_));
+      this.field_70180_af.func_187227_b(field_184701_b, Optional.fromNullable(p_184694_1_));
    }
 
    public int func_184684_db() {
