diff -r -U 3 minecraft\net\minecraft\client\renderer\tileentity\TileEntityRendererDispatcher.java minecraft_patched\net\minecraft\client\renderer\tileentity\TileEntityRendererDispatcher.java
--- minecraft\net\minecraft\client\renderer\tileentity\TileEntityRendererDispatcher.java
+++ minecraft_patched\net\minecraft\client\renderer\tileentity\TileEntityRendererDispatcher.java
@@ -85,16 +85,16 @@
    public <T extends TileEntity> TileEntitySpecialRenderer<T> func_147546_a(Class<? extends TileEntity> p_147546_1_) {
       TileEntitySpecialRenderer<? extends TileEntity> tileentityspecialrenderer = (TileEntitySpecialRenderer)this.field_147559_m.get(p_147546_1_);
       if(tileentityspecialrenderer == null && p_147546_1_ != TileEntity.class) {
-         tileentityspecialrenderer = this.<TileEntity>func_147546_a(p_147546_1_.getSuperclass());
+         tileentityspecialrenderer = this.<TileEntity>func_147546_a((Class<? extends TileEntity>)p_147546_1_.getSuperclass());
          this.field_147559_m.put(p_147546_1_, tileentityspecialrenderer);
       }
 
-      return tileentityspecialrenderer;
+      return (TileEntitySpecialRenderer<T>)tileentityspecialrenderer;
    }
 
    @Nullable
    public <T extends TileEntity> TileEntitySpecialRenderer<T> func_147547_b(@Nullable TileEntity p_147547_1_) {
-      return p_147547_1_ == null?null:this.func_147546_a(p_147547_1_.getClass());
+      return (TileEntitySpecialRenderer<T>)(p_147547_1_ == null?null:this.func_147546_a(p_147547_1_.getClass()));
    }
 
    public void func_190056_a(World p_190056_1_, TextureManager p_190056_2_, FontRenderer p_190056_3_, Entity p_190056_4_, RayTraceResult p_190056_5_, float p_190056_6_) {
