--- a/net/minecraft/block/state/pattern/BlockMaterialMatcher.java
+++ b/net/minecraft/block/state/pattern/BlockMaterialMatcher.java
@@ -21,6 +21,6 @@
 
     public boolean apply(@Nullable IBlockState p_apply_1_)
     {
-        return p_apply_1_ != null && p_apply_1_.func_185904_a() == this.field_189887_a;
+        return p_apply_1_ != null && p_apply_1_.getMaterial() == this.field_189887_a;
     }
 }
