--- a/net/minecraft/block/state/pattern/BlockStateMatcher.java
+++ b/net/minecraft/block/state/pattern/BlockStateMatcher.java
@@ -34,7 +34,7 @@
 
     public boolean apply(@Nullable IBlockState p_apply_1_)
     {
-        if (p_apply_1_ != null && p_apply_1_.func_177230_c().equals(this.field_177641_a.func_177622_c()))
+        if (p_apply_1_ != null && p_apply_1_.getBlock().equals(this.field_177641_a.func_177622_c()))
         {
             for (Entry < IProperty<?>, Predicate<? >> entry : this.field_177640_b.entrySet())
             {
