--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -359,7 +359,7 @@
                 }
 
                 ++i1;
-                worldserver.getChunkProvider().func_186025_d(blockpos.func_177958_n() + l1 >> 4, blockpos.func_177952_p() + i2 >> 4);
+                worldserver.getChunkProvider().func_186025_d(blockpos.getX() + l1 >> 4, blockpos.getZ() + i2 >> 4);
             }
         }
 
@@ -1323,7 +1323,7 @@
 
     public BlockPos func_180425_c()
     {
-        return BlockPos.field_177992_a;
+        return BlockPos.createDefault;
     }
 
     public Vec3d func_174791_d()
