--- a/net/minecraft/world/gen/structure/ComponentScatteredFeaturePieces.java
+++ b/net/minecraft/world/gen/structure/ComponentScatteredFeaturePieces.java
@@ -77,12 +77,12 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, -4, 0, this.scatteredFeatureSizeX - 1, 0, this.scatteredFeatureSizeZ - 1, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, -4, 0, this.scatteredFeatureSizeX - 1, 0, this.scatteredFeatureSizeZ - 1, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
 
             for (int i = 1; i <= 9; ++i)
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, i, i, i, this.scatteredFeatureSizeX - 1 - i, i, this.scatteredFeatureSizeZ - 1 - i, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, i + 1, i, i + 1, this.scatteredFeatureSizeX - 2 - i, i, this.scatteredFeatureSizeZ - 2 - i, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, i, i, i, this.scatteredFeatureSizeX - 1 - i, i, this.scatteredFeatureSizeZ - 1 - i, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, i + 1, i, i + 1, this.scatteredFeatureSizeX - 2 - i, i, this.scatteredFeatureSizeZ - 2 - i, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             }
 
             for (int i2 = 0; i2 < this.scatteredFeatureSizeX; ++i2)
@@ -90,30 +90,30 @@
                 for (int j = 0; j < this.scatteredFeatureSizeZ; ++j)
                 {
                     int k = -5;
-                    this.func_175808_b(p_74875_1_, Blocks.sandstone.func_176223_P(), i2, -5, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.sandstone.getBlockState(), i2, -5, j, p_74875_3_);
                 }
             }
 
-            IBlockState iblockstate1 = Blocks.sandstone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
-            IBlockState iblockstate2 = Blocks.sandstone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
-            IBlockState iblockstate3 = Blocks.sandstone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST);
-            IBlockState iblockstate = Blocks.sandstone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST);
+            IBlockState iblockstate1 = Blocks.sandstone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
+            IBlockState iblockstate2 = Blocks.sandstone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
+            IBlockState iblockstate3 = Blocks.sandstone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST);
+            IBlockState iblockstate = Blocks.sandstone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST);
             int l = ~EnumDyeColor.ORANGE.func_176767_b() & 15;
             int i1 = ~EnumDyeColor.BLUE.func_176767_b() & 15;
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 9, 4, Blocks.sandstone.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 10, 1, 3, 10, 3, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 9, 4, Blocks.sandstone.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 10, 1, 3, 10, 3, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
             this.func_175811_a(p_74875_1_, iblockstate1, 2, 10, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate2, 2, 10, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate3, 0, 10, 2, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate, 4, 10, 2, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 5, 0, 0, this.scatteredFeatureSizeX - 1, 9, 4, Blocks.sandstone.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 4, 10, 1, this.scatteredFeatureSizeX - 2, 10, 3, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 5, 0, 0, this.scatteredFeatureSizeX - 1, 9, 4, Blocks.sandstone.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 4, 10, 1, this.scatteredFeatureSizeX - 2, 10, 3, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
             this.func_175811_a(p_74875_1_, iblockstate1, this.scatteredFeatureSizeX - 3, 10, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate2, this.scatteredFeatureSizeX - 3, 10, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate3, this.scatteredFeatureSizeX - 5, 10, 2, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate, this.scatteredFeatureSizeX - 1, 10, 2, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 0, 0, 12, 4, 4, Blocks.sandstone.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 1, 0, 11, 3, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 0, 0, 12, 4, 4, Blocks.sandstone.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 1, 0, 11, 3, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 9, 1, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 9, 2, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 9, 3, 1, p_74875_3_);
@@ -121,46 +121,46 @@
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 11, 3, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 11, 2, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 11, 1, 1, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 8, 3, 3, Blocks.sandstone.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 2, 8, 2, 2, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, 1, 16, 3, 3, Blocks.sandstone.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, 2, 16, 2, 2, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 4, 5, this.scatteredFeatureSizeX - 6, 4, this.scatteredFeatureSizeZ - 6, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 4, 9, 11, 4, 11, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 8, 3, 3, Blocks.sandstone.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 2, 8, 2, 2, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, 1, 16, 3, 3, Blocks.sandstone.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, 2, 16, 2, 2, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 4, 5, this.scatteredFeatureSizeX - 6, 4, this.scatteredFeatureSizeZ - 6, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 4, 9, 11, 4, 11, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, 1, 8, 8, 3, 8, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, 8, 12, 3, 8, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, 1, 12, 8, 3, 12, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, 12, 12, 3, 12, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 5, 4, 4, 11, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 5, 1, 5, this.scatteredFeatureSizeX - 2, 4, 11, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 7, 9, 6, 7, 11, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 7, 7, 9, this.scatteredFeatureSizeX - 7, 7, 11, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 5, 4, 4, 11, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 5, 1, 5, this.scatteredFeatureSizeX - 2, 4, 11, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 7, 9, 6, 7, 11, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 7, 7, 9, this.scatteredFeatureSizeX - 7, 7, 11, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, 9, 5, 7, 11, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 6, 5, 9, this.scatteredFeatureSizeX - 6, 7, 11, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 5, 5, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 5, 6, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 6, 6, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), this.scatteredFeatureSizeX - 6, 5, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), this.scatteredFeatureSizeX - 6, 6, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), this.scatteredFeatureSizeX - 7, 6, 10, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 4, 4, 2, 6, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 3, 4, 4, this.scatteredFeatureSizeX - 3, 6, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 5, 5, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 5, 6, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 6, 6, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), this.scatteredFeatureSizeX - 6, 5, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), this.scatteredFeatureSizeX - 6, 6, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), this.scatteredFeatureSizeX - 7, 6, 10, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 4, 4, 2, 6, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 3, 4, 4, this.scatteredFeatureSizeX - 3, 6, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175811_a(p_74875_1_, iblockstate1, 2, 4, 5, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 2, 3, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, this.scatteredFeatureSizeX - 3, 4, 5, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, this.scatteredFeatureSizeX - 3, 3, 4, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 3, 2, 2, 3, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 3, 1, 3, this.scatteredFeatureSizeX - 2, 2, 3, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176223_P(), 1, 1, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176223_P(), this.scatteredFeatureSizeX - 2, 1, 2, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 3, 2, 2, 3, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 3, 1, 3, this.scatteredFeatureSizeX - 2, 2, 3, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.sandstone.getBlockState(), 1, 1, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.sandstone.getBlockState(), this.scatteredFeatureSizeX - 2, 1, 2, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.SAND.func_176624_a()), 1, 2, 2, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.SAND.func_176624_a()), this.scatteredFeatureSizeX - 2, 2, 2, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate, 2, 1, 2, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate3, this.scatteredFeatureSizeX - 3, 1, 2, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 5, 4, 3, 18, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 5, 3, 5, this.scatteredFeatureSizeX - 5, 3, 17, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, 5, 4, 2, 16, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 6, 1, 5, this.scatteredFeatureSizeX - 5, 2, 16, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 5, 4, 3, 18, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 5, 3, 5, this.scatteredFeatureSizeX - 5, 3, 17, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, 5, 4, 2, 16, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, this.scatteredFeatureSizeX - 6, 1, 5, this.scatteredFeatureSizeX - 5, 2, 16, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
 
             for (int j1 = 5; j1 <= 17; j1 += 2)
             {
@@ -235,32 +235,32 @@
             }
 
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, 4, 0, 12, 6, 0, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 8, 6, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 12, 6, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 8, 6, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 12, 6, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.stained_hardened_clay.func_176203_a(l), 9, 5, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.CHISELED.func_176675_a()), 10, 5, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.stained_hardened_clay.func_176203_a(l), 11, 5, 0, p_74875_3_);
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, -14, 8, 12, -11, 12, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, -10, 8, 12, -10, 12, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.CHISELED.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.CHISELED.func_176675_a()), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, -9, 8, 12, -9, 12, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, -8, 8, 12, -1, 12, Blocks.sandstone.func_176223_P(), Blocks.sandstone.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, -11, 9, 11, -1, 11, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.stone_pressure_plate.func_176223_P(), 10, -11, 10, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, -13, 9, 11, -13, 11, Blocks.tnt.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 8, -11, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 8, -10, 10, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, -8, 8, 12, -1, 12, Blocks.sandstone.getBlockState(), Blocks.sandstone.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, -11, 9, 11, -1, 11, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.stone_pressure_plate.getBlockState(), 10, -11, 10, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, -13, 9, 11, -13, 11, Blocks.tnt.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 8, -11, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 8, -10, 10, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.CHISELED.func_176675_a()), 7, -10, 10, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 7, -11, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 12, -11, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 12, -10, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 12, -11, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 12, -10, 10, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.CHISELED.func_176675_a()), 13, -10, 10, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 13, -11, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 10, -11, 8, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 10, -10, 8, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 10, -11, 8, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 10, -10, 8, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.CHISELED.func_176675_a()), 10, -10, 7, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 10, -11, 7, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 10, -11, 12, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 10, -10, 12, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 10, -11, 12, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 10, -10, 12, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.CHISELED.func_176675_a()), 10, -10, 13, p_74875_3_);
             this.func_175811_a(p_74875_1_, Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a()), 10, -11, 13, p_74875_3_);
 
@@ -343,7 +343,7 @@
 
                         if (p_74935_2_.func_175898_b(blockpos$mutableblockpos))
                         {
-                            i += Math.max(p_74935_1_.func_175672_r(blockpos$mutableblockpos).func_177956_o(), p_74935_1_.provider.getAverageGroundLevel());
+                            i += Math.max(p_74935_1_.func_175672_r(blockpos$mutableblockpos).getY(), p_74935_1_.provider.getAverageGroundLevel());
                             ++j;
                         }
                     }
@@ -416,8 +416,8 @@
                         if ("chest".equals(entry.getValue()))
                         {
                             BlockPos blockpos2 = (BlockPos)entry.getKey();
-                            p_74875_1_.func_180501_a(blockpos2, Blocks.air.func_176223_P(), 3);
-                            TileEntity tileentity = p_74875_1_.func_175625_s(blockpos2.func_177977_b());
+                            p_74875_1_.func_180501_a(blockpos2, Blocks.air.getBlockState(), 3);
+                            TileEntity tileentity = p_74875_1_.func_175625_s(blockpos2.getBlockPosBelow());
 
                             if (tileentity instanceof TileEntityChest)
                             {
@@ -429,7 +429,7 @@
                 else
                 {
                     BlockPos blockpos3 = Template.func_186266_a(placementsettings, new BlockPos(3, 0, 5));
-                    p_74875_1_.func_180501_a(blockpos.func_177971_a(blockpos3), Blocks.snow.func_176223_P(), 3);
+                    p_74875_1_.func_180501_a(blockpos.func_177971_a(blockpos3), Blocks.snow.getBlockState(), 3);
                 }
 
                 return true;
@@ -502,10 +502,10 @@
                 this.fillWithAir(p_74875_1_, p_74875_3_, 5, 2, 12, 6, 2, 12);
                 this.fillWithAir(p_74875_1_, p_74875_3_, 5, 5, 1, 6, 5, 1);
                 this.fillWithAir(p_74875_1_, p_74875_3_, 5, 5, 13, 6, 5, 13);
-                this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, 5, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 10, 5, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, 5, 9, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 10, 5, 9, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, 5, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 10, 5, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, 5, 9, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 10, 5, 9, p_74875_3_);
 
                 for (int i = 0; i <= 14; i += 14)
                 {
@@ -537,10 +537,10 @@
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 9, 10, 4, 9, 10, false, p_74875_2_, junglePyramidsRandomScatteredStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 7, 9, 10, 7, 9, 10, false, p_74875_2_, junglePyramidsRandomScatteredStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 5, 9, 7, 6, 9, 7, false, p_74875_2_, junglePyramidsRandomScatteredStones);
-                IBlockState iblockstate2 = Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST);
-                IBlockState iblockstate3 = Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST);
-                IBlockState iblockstate = Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
-                IBlockState iblockstate1 = Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
+                IBlockState iblockstate2 = Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST);
+                IBlockState iblockstate3 = Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST);
+                IBlockState iblockstate = Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
+                IBlockState iblockstate1 = Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
                 this.func_175811_a(p_74875_1_, iblockstate1, 5, 9, 6, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate1, 6, 9, 6, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate, 5, 9, 8, p_74875_3_);
@@ -587,78 +587,78 @@
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 7, -2, 1, 9, -2, 1, false, p_74875_2_, junglePyramidsRandomScatteredStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 6, -3, 1, 6, -3, 1, false, p_74875_2_, junglePyramidsRandomScatteredStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 6, -1, 1, 6, -1, 1, false, p_74875_2_, junglePyramidsRandomScatteredStones);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.func_176223_P().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.EAST).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 1, -3, 8, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.func_176223_P().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.WEST).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 4, -3, 8, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire.func_176223_P().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 2, -3, 8, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire.func_176223_P().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 3, -3, 8, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 5, -3, 7, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 5, -3, 6, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 5, -3, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 5, -3, 4, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 5, -3, 3, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 5, -3, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 5, -3, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 4, -3, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 3, -3, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.getBlockState().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.EAST).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 1, -3, 8, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.getBlockState().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.WEST).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 4, -3, 8, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire.getBlockState().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 2, -3, 8, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire.getBlockState().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 3, -3, 8, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 5, -3, 7, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 5, -3, 6, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 5, -3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 5, -3, 4, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 5, -3, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 5, -3, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 5, -3, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 4, -3, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 3, -3, 1, p_74875_3_);
 
                 if (!this.field_74945_j)
                 {
                     this.field_74945_j = this.func_189419_a(p_74875_1_, p_74875_3_, p_74875_2_, 3, -2, 1, EnumFacing.NORTH, LootTableList.field_189420_m);
                 }
 
-                this.func_175811_a(p_74875_1_, Blocks.vine.func_176223_P().func_177226_a(BlockVine.field_176279_N, Boolean.valueOf(true)), 3, -2, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.func_176223_P().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.NORTH).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 7, -3, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.func_176223_P().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.SOUTH).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 7, -3, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire.func_176223_P().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 7, -3, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire.func_176223_P().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 7, -3, 3, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.tripwire.func_176223_P().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 7, -3, 4, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 8, -3, 6, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 9, -3, 6, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 9, -3, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 9, -3, 4, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 9, -2, 4, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.vine.getBlockState().func_177226_a(BlockVine.field_176279_N, Boolean.valueOf(true)), 3, -2, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.getBlockState().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.NORTH).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 7, -3, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire_hook.getBlockState().func_177226_a(BlockTripWireHook.field_176264_a, EnumFacing.SOUTH).func_177226_a(BlockTripWireHook.field_176265_M, Boolean.valueOf(true)), 7, -3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire.getBlockState().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 7, -3, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire.getBlockState().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 7, -3, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.tripwire.getBlockState().func_177226_a(BlockTripWire.field_176294_M, Boolean.valueOf(true)), 7, -3, 4, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 8, -3, 6, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 9, -3, 6, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 9, -3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 9, -3, 4, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 9, -2, 4, p_74875_3_);
 
                 if (!this.field_74946_k)
                 {
                     this.field_74946_k = this.func_189419_a(p_74875_1_, p_74875_3_, p_74875_2_, 9, -2, 3, EnumFacing.WEST, LootTableList.field_189420_m);
                 }
 
-                this.func_175811_a(p_74875_1_, Blocks.vine.func_176223_P().func_177226_a(BlockVine.field_176278_M, Boolean.valueOf(true)), 8, -1, 3, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.vine.func_176223_P().func_177226_a(BlockVine.field_176278_M, Boolean.valueOf(true)), 8, -2, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.vine.getBlockState().func_177226_a(BlockVine.field_176278_M, Boolean.valueOf(true)), 8, -1, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.vine.getBlockState().func_177226_a(BlockVine.field_176278_M, Boolean.valueOf(true)), 8, -2, 3, p_74875_3_);
 
                 if (!this.field_74947_h)
                 {
                     this.field_74947_h = this.func_186167_a(p_74875_1_, p_74875_3_, p_74875_2_, 8, -3, 3, LootTableList.field_186430_l);
                 }
 
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 9, -3, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 8, -3, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 4, -3, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 5, -2, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 5, -1, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 6, -3, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 7, -2, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 7, -1, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 8, -3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 9, -3, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 8, -3, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 4, -3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 5, -2, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 5, -1, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 6, -3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 7, -2, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 7, -1, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 8, -3, 5, p_74875_3_);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 9, -1, 1, 9, -1, 5, false, p_74875_2_, junglePyramidsRandomScatteredStones);
                 this.fillWithAir(p_74875_1_, p_74875_3_, 8, -3, 8, 10, -1, 10);
                 this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176203_a(BlockStoneBrick.field_176252_O), 8, -2, 11, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176203_a(BlockStoneBrick.field_176252_O), 9, -2, 11, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176203_a(BlockStoneBrick.field_176252_O), 10, -2, 11, p_74875_3_);
-                IBlockState iblockstate4 = Blocks.lever.func_176223_P().func_177226_a(BlockLever.field_176360_a, BlockLever.EnumOrientation.NORTH);
+                IBlockState iblockstate4 = Blocks.lever.getBlockState().func_177226_a(BlockLever.field_176360_a, BlockLever.EnumOrientation.NORTH);
                 this.func_175811_a(p_74875_1_, iblockstate4, 8, -2, 12, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate4, 9, -2, 12, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate4, 10, -2, 12, p_74875_3_);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 8, -3, 8, 8, -3, 10, false, p_74875_2_, junglePyramidsRandomScatteredStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 10, -3, 8, 10, -3, 10, false, p_74875_2_, junglePyramidsRandomScatteredStones);
-                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.func_176223_P(), 10, -2, 9, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 8, -2, 9, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 8, -2, 10, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.func_176223_P(), 10, -1, 9, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.sticky_piston.func_176223_P().func_177226_a(BlockPistonBase.field_176387_N, EnumFacing.UP), 9, -2, 8, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.sticky_piston.func_176223_P().func_177226_a(BlockPistonBase.field_176387_N, EnumFacing.WEST), 10, -2, 8, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.sticky_piston.func_176223_P().func_177226_a(BlockPistonBase.field_176387_N, EnumFacing.WEST), 10, -1, 8, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.unpowered_repeater.func_176223_P().func_177226_a(BlockRedstoneRepeater.field_185512_D, EnumFacing.NORTH), 10, -2, 10, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.mossy_cobblestone.getBlockState(), 10, -2, 9, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 8, -2, 9, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 8, -2, 10, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.redstone_wire.getBlockState(), 10, -1, 9, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.sticky_piston.getBlockState().func_177226_a(BlockPistonBase.field_176387_N, EnumFacing.UP), 9, -2, 8, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.sticky_piston.getBlockState().func_177226_a(BlockPistonBase.field_176387_N, EnumFacing.WEST), 10, -2, 8, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.sticky_piston.getBlockState().func_177226_a(BlockPistonBase.field_176387_N, EnumFacing.WEST), 10, -1, 8, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.unpowered_repeater.getBlockState().func_177226_a(BlockRedstoneRepeater.field_185512_D, EnumFacing.NORTH), 10, -2, 10, p_74875_3_);
 
                 if (!this.field_74948_i)
                 {
@@ -679,11 +679,11 @@
             {
                 if (p_75062_1_.nextFloat() < 0.4F)
                 {
-                    this.field_151562_a = Blocks.cobblestone.func_176223_P();
+                    this.field_151562_a = Blocks.cobblestone.getBlockState();
                 }
                 else
                 {
-                    this.field_151562_a = Blocks.mossy_cobblestone.func_176223_P();
+                    this.field_151562_a = Blocks.mossy_cobblestone.getBlockState();
                 }
             }
         }
@@ -729,24 +729,24 @@
                 this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 3, 1, 3, 6, Blocks.planks.func_176203_a(BlockPlanks.EnumType.SPRUCE.func_176839_a()), Blocks.planks.func_176203_a(BlockPlanks.EnumType.SPRUCE.func_176839_a()), false);
                 this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 3, 5, 3, 6, Blocks.planks.func_176203_a(BlockPlanks.EnumType.SPRUCE.func_176839_a()), Blocks.planks.func_176203_a(BlockPlanks.EnumType.SPRUCE.func_176839_a()), false);
                 this.func_175804_a(p_74875_1_, p_74875_3_, 2, 2, 7, 4, 3, 7, Blocks.planks.func_176203_a(BlockPlanks.EnumType.SPRUCE.func_176839_a()), Blocks.planks.func_176203_a(BlockPlanks.EnumType.SPRUCE.func_176839_a()), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 2, 1, 3, 2, Blocks.log.func_176223_P(), Blocks.log.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 0, 2, 5, 3, 2, Blocks.log.func_176223_P(), Blocks.log.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 7, 1, 3, 7, Blocks.log.func_176223_P(), Blocks.log.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 0, 7, 5, 3, 7, Blocks.log.func_176223_P(), Blocks.log.func_176223_P(), false);
-                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 2, 3, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 3, 3, 7, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, 3, 4, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 5, 3, 4, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 5, 3, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.flower_pot.func_176223_P().func_177226_a(BlockFlowerPot.field_176443_b, BlockFlowerPot.EnumFlowerType.MUSHROOM_RED), 1, 3, 5, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.crafting_table.func_176223_P(), 3, 2, 6, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.cauldron.func_176223_P(), 4, 2, 6, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 1, 2, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 5, 2, 1, p_74875_3_);
-                IBlockState iblockstate = Blocks.spruce_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
-                IBlockState iblockstate1 = Blocks.spruce_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST);
-                IBlockState iblockstate2 = Blocks.spruce_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST);
-                IBlockState iblockstate3 = Blocks.spruce_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 2, 1, 3, 2, Blocks.log.getBlockState(), Blocks.log.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 0, 2, 5, 3, 2, Blocks.log.getBlockState(), Blocks.log.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 7, 1, 3, 7, Blocks.log.getBlockState(), Blocks.log.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 0, 7, 5, 3, 7, Blocks.log.getBlockState(), Blocks.log.getBlockState(), false);
+                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 2, 3, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 3, 3, 7, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, 3, 4, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 5, 3, 4, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 5, 3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.flower_pot.getBlockState().func_177226_a(BlockFlowerPot.field_176443_b, BlockFlowerPot.EnumFlowerType.MUSHROOM_RED), 1, 3, 5, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.crafting_table.getBlockState(), 3, 2, 6, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.cauldron.getBlockState(), 4, 2, 6, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 1, 2, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 5, 2, 1, p_74875_3_);
+                IBlockState iblockstate = Blocks.spruce_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
+                IBlockState iblockstate1 = Blocks.spruce_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST);
+                IBlockState iblockstate2 = Blocks.spruce_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST);
+                IBlockState iblockstate3 = Blocks.spruce_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
                 this.func_175804_a(p_74875_1_, p_74875_3_, 0, 4, 1, 6, 4, 1, iblockstate, iblockstate, false);
                 this.func_175804_a(p_74875_1_, p_74875_3_, 0, 4, 2, 0, 4, 7, iblockstate1, iblockstate1, false);
                 this.func_175804_a(p_74875_1_, p_74875_3_, 6, 4, 2, 6, 4, 7, iblockstate2, iblockstate2, false);
@@ -756,7 +756,7 @@
                 {
                     for (int j = 1; j <= 5; j += 4)
                     {
-                        this.func_175808_b(p_74875_1_, Blocks.log.func_176223_P(), j, -1, i, p_74875_3_);
+                        this.func_175808_b(p_74875_1_, Blocks.log.getBlockState(), j, -1, i, p_74875_3_);
                     }
                 }
 
