--- a/net/minecraft/world/gen/structure/MapGenStronghold.java
+++ b/net/minecraft/world/gen/structure/MapGenStronghold.java
@@ -147,8 +147,8 @@
 
                 if (blockpos != null)
                 {
-                    j1 = blockpos.func_177958_n() >> 4;
-                    k1 = blockpos.func_177952_p() >> 4;
+                    j1 = blockpos.getX() >> 4;
+                    k1 = blockpos.getZ() >> 4;
                 }
 
                 if (i1 >= l)
