--- a/net/minecraft/world/gen/structure/StructureOceanMonumentPieces.java
+++ b/net/minecraft/world/gen/structure/StructureOceanMonumentPieces.java
@@ -904,7 +904,7 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            int i = Math.max(p_74875_1_.func_181545_F(), 64) - this.boundingBox.minY;
+            int i = Math.max(p_74875_1_.getSpawnLevel(), 64) - this.boundingBox.minY;
             this.func_181655_a(p_74875_1_, p_74875_3_, 0, 0, 0, 58, i, 58, false);
             this.func_175840_a(false, 0, p_74875_1_, p_74875_2_, p_74875_3_);
             this.func_175840_a(true, 33, p_74875_1_, p_74875_2_, p_74875_3_);
@@ -1379,7 +1379,7 @@
             }
 
             this.func_175804_a(p_74875_1_, p_74875_3_, 6, 3, 6, 9, 6, 9, field_175827_c, field_175827_c, false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 4, 7, 8, 5, 8, Blocks.gold_block.func_176223_P(), Blocks.gold_block.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 4, 7, 8, 5, 8, Blocks.gold_block.getBlockState(), Blocks.gold_block.getBlockState(), false);
 
             for (i = 3; i <= 6; i += 3)
             {
@@ -1496,8 +1496,8 @@
         protected static final IBlockState field_175826_b = Blocks.field_180397_cI.func_176203_a(BlockPrismarine.field_176333_M);
         protected static final IBlockState field_175827_c = Blocks.field_180397_cI.func_176203_a(BlockPrismarine.field_176334_N);
         protected static final IBlockState field_175824_d = field_175826_b;
-        protected static final IBlockState field_175825_e = Blocks.field_180398_cJ.func_176223_P();
-        protected static final IBlockState field_175822_f = Blocks.water.func_176223_P();
+        protected static final IBlockState field_175825_e = Blocks.field_180398_cJ.getBlockState();
+        protected static final IBlockState field_175822_f = Blocks.water.getBlockState();
         protected static final int field_175823_g = func_175820_a(2, 0, 0);
         protected static final int field_175831_h = func_175820_a(2, 2, 0);
         protected static final int field_175832_i = func_175820_a(0, 1, 0);
@@ -1580,11 +1580,11 @@
                 {
                     for (int k = p_181655_5_; k <= p_181655_8_; ++k)
                     {
-                        if (!p_181655_9_ || this.func_175807_a(p_181655_1_, j, i, k, p_181655_2_).func_185904_a() != Material.air)
+                        if (!p_181655_9_ || this.func_175807_a(p_181655_1_, j, i, k, p_181655_2_).getMaterial() != Material.air)
                         {
-                            if (this.getYWithOffset(i) >= p_181655_1_.func_181545_F())
+                            if (this.getYWithOffset(i) >= p_181655_1_.getSpawnLevel())
                             {
-                                this.func_175811_a(p_181655_1_, Blocks.air.func_176223_P(), j, i, k, p_181655_2_);
+                                this.func_175811_a(p_181655_1_, Blocks.air.getBlockState(), j, i, k, p_181655_2_);
                             }
                             else
                             {
