--- a/net/minecraft/world/gen/structure/StructureComponentTemplate.java
+++ b/net/minecraft/world/gen/structure/StructureComponentTemplate.java
@@ -43,9 +43,9 @@
      */
     protected void writeStructureToNBT(NBTTagCompound p_143012_1_)
     {
-        p_143012_1_.setInteger("TPX", this.field_186178_c.func_177958_n());
-        p_143012_1_.setInteger("TPY", this.field_186178_c.func_177956_o());
-        p_143012_1_.setInteger("TPZ", this.field_186178_c.func_177952_p());
+        p_143012_1_.setInteger("TPX", this.field_186178_c.getX());
+        p_143012_1_.setInteger("TPY", this.field_186178_c.getY());
+        p_143012_1_.setInteger("TPZ", this.field_186178_c.getZ());
     }
 
     /**
@@ -81,7 +81,7 @@
     {
         Rotation rotation = this.field_186177_b.func_186215_c();
         BlockPos blockpos = this.field_186176_a.func_186257_a(rotation);
-        this.boundingBox = new StructureBoundingBox(0, 0, 0, blockpos.func_177958_n(), blockpos.func_177956_o() - 1, blockpos.func_177952_p());
+        this.boundingBox = new StructureBoundingBox(0, 0, 0, blockpos.getX(), blockpos.getY() - 1, blockpos.getZ());
 
         switch (rotation)
         {
@@ -90,18 +90,18 @@
                 break;
 
             case CLOCKWISE_90:
-                this.boundingBox.offset(-blockpos.func_177958_n(), 0, 0);
+                this.boundingBox.offset(-blockpos.getX(), 0, 0);
                 break;
 
             case COUNTERCLOCKWISE_90:
-                this.boundingBox.offset(0, 0, -blockpos.func_177952_p());
+                this.boundingBox.offset(0, 0, -blockpos.getZ());
                 break;
 
             case CLOCKWISE_180:
-                this.boundingBox.offset(-blockpos.func_177958_n(), 0, -blockpos.func_177952_p());
+                this.boundingBox.offset(-blockpos.getX(), 0, -blockpos.getZ());
         }
 
-        this.boundingBox.offset(this.field_186178_c.func_177958_n(), this.field_186178_c.func_177956_o(), this.field_186178_c.func_177952_p());
+        this.boundingBox.offset(this.field_186178_c.getX(), this.field_186178_c.getY(), this.field_186178_c.getZ());
     }
 
     public void func_181138_a(int p_181138_1_, int p_181138_2_, int p_181138_3_)
