--- a/net/minecraft/world/gen/structure/StructureVillagePieces.java
+++ b/net/minecraft/world/gen/structure/StructureVillagePieces.java
@@ -291,12 +291,12 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 12 - 1, 0);
             }
 
-            IBlockState iblockstate = Blocks.cobblestone.func_176223_P();
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST));
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 3, 3, 7, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 1, 3, 9, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            IBlockState iblockstate = Blocks.cobblestone.getBlockState();
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST));
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 3, 3, 7, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 1, 3, 9, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 0, 3, 0, 8, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 0, 3, 10, 0, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 10, 3, iblockstate, iblockstate, false);
@@ -322,43 +322,43 @@
             this.func_175811_a(p_74875_1_, iblockstate1, 3, 1, 5, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate2, 1, 2, 7, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate3, 3, 2, 7, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 3, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 3, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 6, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 7, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 6, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 7, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 6, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 7, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 6, 4, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 7, 4, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 3, 6, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 3, 6, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 3, 8, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 3, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 3, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 6, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 7, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 6, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 7, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 6, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 7, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 6, 4, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 7, 4, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 3, 6, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 3, 6, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 3, 8, p_74875_3_);
             this.func_189926_a(p_74875_1_, EnumFacing.SOUTH, 2, 4, 7, p_74875_3_);
             this.func_189926_a(p_74875_1_, EnumFacing.EAST, 1, 4, 6, p_74875_3_);
             this.func_189926_a(p_74875_1_, EnumFacing.WEST, 3, 4, 6, p_74875_3_);
             this.func_189926_a(p_74875_1_, EnumFacing.NORTH, 2, 4, 5, p_74875_3_);
-            IBlockState iblockstate4 = Blocks.ladder.func_176223_P().func_177226_a(BlockLadder.field_176382_a, EnumFacing.WEST);
+            IBlockState iblockstate4 = Blocks.ladder.getBlockState().func_177226_a(BlockLadder.field_176382_a, EnumFacing.WEST);
 
             for (int i = 1; i <= 9; ++i)
             {
                 this.func_175811_a(p_74875_1_, iblockstate4, 3, i, 3, p_74875_3_);
             }
 
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 1, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 1, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 2, 0, p_74875_3_);
             this.func_189927_a(p_74875_1_, p_74875_3_, p_74875_2_, 2, 1, 0, EnumFacing.NORTH);
 
-            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).func_185904_a() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_185904_a() != Material.air)
+            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).getMaterial() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getMaterial() != Material.air)
             {
                 this.func_175811_a(p_74875_1_, iblockstate1, 2, 0, -1, p_74875_3_);
 
-                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_177230_c() == Blocks.field_185774_da)
+                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getBlock() == Blocks.field_185774_da)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.grass.func_176223_P(), 2, -1, -1, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.grass.getBlockState(), 2, -1, -1, p_74875_3_);
                 }
             }
 
@@ -481,19 +481,19 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 4 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.log.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 0, 12, 4, 8, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 2, 0, 7, Blocks.farmland.func_176223_P(), Blocks.farmland.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 1, 5, 0, 7, Blocks.farmland.func_176223_P(), Blocks.farmland.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 0, 1, 8, 0, 7, Blocks.farmland.func_176223_P(), Blocks.farmland.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 10, 0, 1, 11, 0, 7, Blocks.farmland.func_176223_P(), Blocks.farmland.func_176223_P(), false);
+            IBlockState iblockstate = this.func_175847_a(Blocks.log.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 0, 12, 4, 8, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 2, 0, 7, Blocks.farmland.getBlockState(), Blocks.farmland.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 1, 5, 0, 7, Blocks.farmland.getBlockState(), Blocks.farmland.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 0, 1, 8, 0, 7, Blocks.farmland.getBlockState(), Blocks.farmland.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 10, 0, 1, 11, 0, 7, Blocks.farmland.getBlockState(), Blocks.farmland.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 0, 0, 8, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 6, 0, 0, 6, 0, 8, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 12, 0, 0, 12, 0, 8, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 0, 11, 0, 0, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 8, 11, 0, 8, iblockstate, iblockstate, false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 0, 1, 3, 0, 7, Blocks.water.func_176223_P(), Blocks.water.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 0, 1, 9, 0, 7, Blocks.water.func_176223_P(), Blocks.water.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 0, 1, 3, 0, 7, Blocks.water.getBlockState(), Blocks.water.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 0, 1, 9, 0, 7, Blocks.water.getBlockState(), Blocks.water.getBlockState(), false);
 
             for (int i = 1; i <= 7; ++i)
             {
@@ -520,7 +520,7 @@
                 for (int k2 = 0; k2 < 13; ++k2)
                 {
                     this.clearCurrentPositionBlocksUpwards(p_74875_1_, k2, 4, j2, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.dirt.func_176223_P(), k2, -1, j2, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.dirt.getBlockState(), k2, -1, j2, p_74875_3_);
                 }
             }
 
@@ -600,15 +600,15 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 4 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.log.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 0, 6, 4, 8, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 2, 0, 7, Blocks.farmland.func_176223_P(), Blocks.farmland.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 1, 5, 0, 7, Blocks.farmland.func_176223_P(), Blocks.farmland.func_176223_P(), false);
+            IBlockState iblockstate = this.func_175847_a(Blocks.log.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 0, 6, 4, 8, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 2, 0, 7, Blocks.farmland.getBlockState(), Blocks.farmland.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 1, 5, 0, 7, Blocks.farmland.getBlockState(), Blocks.farmland.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 0, 0, 8, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 6, 0, 0, 6, 0, 8, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 0, 5, 0, 0, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 8, 5, 0, 8, iblockstate, iblockstate, false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 0, 1, 3, 0, 7, Blocks.water.func_176223_P(), Blocks.water.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 0, 1, 3, 0, 7, Blocks.water.getBlockState(), Blocks.water.getBlockState(), false);
 
             for (int i = 1; i <= 7; ++i)
             {
@@ -627,7 +627,7 @@
                 for (int k1 = 0; k1 < 7; ++k1)
                 {
                     this.clearCurrentPositionBlocksUpwards(p_74875_1_, k1, 4, j1, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.dirt.func_176223_P(), k1, -1, j1, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.dirt.getBlockState(), k1, -1, j1, p_74875_3_);
                 }
             }
 
@@ -668,16 +668,16 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 7 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.func_176223_P());
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH));
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
-            IBlockState iblockstate4 = this.func_175847_a(Blocks.planks.func_176223_P());
-            IBlockState iblockstate5 = this.func_175847_a(Blocks.log.func_176223_P());
-            IBlockState iblockstate6 = this.func_175847_a(Blocks.field_180407_aO.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 7, 4, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 1, 6, 8, 4, 10, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 0, 6, 8, 0, 10, Blocks.dirt.func_176223_P(), Blocks.dirt.func_176223_P(), false);
+            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.getBlockState());
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH));
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
+            IBlockState iblockstate4 = this.func_175847_a(Blocks.planks.getBlockState());
+            IBlockState iblockstate5 = this.func_175847_a(Blocks.log.getBlockState());
+            IBlockState iblockstate6 = this.func_175847_a(Blocks.field_180407_aO.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 7, 4, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 1, 6, 8, 4, 10, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 0, 6, 8, 0, 10, Blocks.dirt.getBlockState(), Blocks.dirt.getBlockState(), false);
             this.func_175811_a(p_74875_1_, iblockstate, 6, 0, 6, p_74875_3_);
             this.func_175804_a(p_74875_1_, p_74875_3_, 2, 1, 6, 2, 1, 10, iblockstate6, iblockstate6, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, 1, 6, 8, 1, 10, iblockstate6, iblockstate6, false);
@@ -712,39 +712,39 @@
             this.func_175811_a(p_74875_1_, iblockstate5, 0, 2, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate5, 8, 2, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate5, 8, 2, 4, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 2, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 3, 2, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 5, 2, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 6, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 3, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 5, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 6, 2, 5, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 2, 1, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.func_176223_P(), 2, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.getBlockState(), 2, 2, 3, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate4, 1, 1, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate7, 2, 1, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate3, 1, 1, 3, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 0, 1, 7, 0, 3, Blocks.double_stone_slab.func_176223_P(), Blocks.double_stone_slab.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.double_stone_slab.func_176223_P(), 6, 1, 1, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.double_stone_slab.func_176223_P(), 6, 1, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 1, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 2, 0, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 0, 1, 7, 0, 3, Blocks.double_stone_slab.getBlockState(), Blocks.double_stone_slab.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.double_stone_slab.getBlockState(), 6, 1, 1, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.double_stone_slab.getBlockState(), 6, 1, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 1, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 2, 0, p_74875_3_);
             this.func_189926_a(p_74875_1_, EnumFacing.NORTH, 2, 3, 1, p_74875_3_);
             this.func_189927_a(p_74875_1_, p_74875_3_, p_74875_2_, 2, 1, 0, EnumFacing.NORTH);
 
-            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).func_185904_a() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_185904_a() != Material.air)
+            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).getMaterial() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getMaterial() != Material.air)
             {
                 this.func_175811_a(p_74875_1_, iblockstate7, 2, 0, -1, p_74875_3_);
 
-                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_177230_c() == Blocks.field_185774_da)
+                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getBlock() == Blocks.field_185774_da)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.grass.func_176223_P(), 2, -1, -1, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.grass.getBlockState(), 2, -1, -1, p_74875_3_);
                 }
             }
 
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 6, 1, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 6, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 6, 1, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 6, 2, 5, p_74875_3_);
             this.func_189926_a(p_74875_1_, EnumFacing.SOUTH, 6, 3, 4, p_74875_3_);
             this.func_189927_a(p_74875_1_, p_74875_3_, p_74875_2_, 6, 1, 5, EnumFacing.SOUTH);
 
@@ -800,14 +800,14 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 9 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.func_176223_P());
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH));
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST));
-            IBlockState iblockstate4 = this.func_175847_a(Blocks.planks.func_176223_P());
-            IBlockState iblockstate5 = this.func_175847_a(Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate6 = this.func_175847_a(Blocks.field_180407_aO.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 7, 5, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.getBlockState());
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH));
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST));
+            IBlockState iblockstate4 = this.func_175847_a(Blocks.planks.getBlockState());
+            IBlockState iblockstate5 = this.func_175847_a(Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate6 = this.func_175847_a(Blocks.field_180407_aO.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 7, 5, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 8, 0, 5, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 8, 5, 5, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 1, 8, 6, 4, iblockstate, iblockstate, false);
@@ -834,27 +834,27 @@
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 5, 7, 4, 5, iblockstate4, iblockstate4, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 8, 2, 1, 8, 4, 4, iblockstate4, iblockstate4, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 0, 7, 4, 0, iblockstate4, iblockstate4, false);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 2, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 5, 2, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 6, 2, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 3, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 5, 3, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 6, 3, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 3, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 3, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 3, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 3, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 2, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 3, 2, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 5, 2, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 6, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 5, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 6, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 3, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 5, 3, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 6, 3, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 3, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 3, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 3, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 3, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 3, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 5, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 6, 2, 5, p_74875_3_);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 1, 7, 4, 1, iblockstate4, iblockstate4, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 4, 7, 4, 4, iblockstate4, iblockstate4, false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 7, 3, 4, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 7, 3, 4, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
             this.func_175811_a(p_74875_1_, iblockstate4, 7, 1, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate3, 7, 1, 3, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 6, 1, 4, p_74875_3_);
@@ -862,21 +862,21 @@
             this.func_175811_a(p_74875_1_, iblockstate1, 4, 1, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 3, 1, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 6, 1, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.func_176223_P(), 6, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.getBlockState(), 6, 2, 3, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 4, 1, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.func_176223_P(), 4, 2, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.crafting_table.func_176223_P(), 7, 1, 1, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, 1, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.getBlockState(), 4, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.crafting_table.getBlockState(), 7, 1, 1, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, 1, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, 2, 0, p_74875_3_);
             this.func_189927_a(p_74875_1_, p_74875_3_, p_74875_2_, 1, 1, 0, EnumFacing.NORTH);
 
-            if (this.func_175807_a(p_74875_1_, 1, 0, -1, p_74875_3_).func_185904_a() == Material.air && this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).func_185904_a() != Material.air)
+            if (this.func_175807_a(p_74875_1_, 1, 0, -1, p_74875_3_).getMaterial() == Material.air && this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).getMaterial() != Material.air)
             {
                 this.func_175811_a(p_74875_1_, iblockstate5, 1, 0, -1, p_74875_3_);
 
-                if (this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).func_177230_c() == Blocks.field_185774_da)
+                if (this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).getBlock() == Blocks.field_185774_da)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.grass.func_176223_P(), 1, -1, -1, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.grass.getBlockState(), 1, -1, -1, p_74875_3_);
                 }
             }
 
@@ -946,18 +946,18 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 6 - 1, 0);
             }
 
-            IBlockState iblockstate = Blocks.cobblestone.func_176223_P();
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.planks.func_176223_P());
-            IBlockState iblockstate4 = this.func_175847_a(Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate5 = this.func_175847_a(Blocks.log.func_176223_P());
-            IBlockState iblockstate6 = this.func_175847_a(Blocks.field_180407_aO.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 0, 9, 4, 6, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            IBlockState iblockstate = Blocks.cobblestone.getBlockState();
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.planks.getBlockState());
+            IBlockState iblockstate4 = this.func_175847_a(Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate5 = this.func_175847_a(Blocks.log.getBlockState());
+            IBlockState iblockstate6 = this.func_175847_a(Blocks.field_180407_aO.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 0, 9, 4, 6, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 9, 0, 6, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 4, 0, 9, 4, 6, iblockstate, iblockstate, false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 9, 5, 6, Blocks.stone_slab.func_176223_P(), Blocks.stone_slab.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 1, 8, 5, 5, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 9, 5, 6, Blocks.stone_slab.getBlockState(), Blocks.stone_slab.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 1, 8, 5, 5, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 0, 2, 3, 0, iblockstate3, iblockstate3, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 0, 0, 4, 0, iblockstate5, iblockstate5, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, 0, 3, 4, 0, iblockstate5, iblockstate5, false);
@@ -970,21 +970,21 @@
             this.func_175804_a(p_74875_1_, p_74875_3_, 5, 1, 0, 5, 3, 0, iblockstate6, iblockstate6, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 9, 1, 0, 9, 3, 0, iblockstate6, iblockstate6, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 6, 1, 4, 9, 4, 6, iblockstate, iblockstate, false);
-            this.func_175811_a(p_74875_1_, Blocks.flowing_lava.func_176223_P(), 7, 1, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.flowing_lava.func_176223_P(), 8, 1, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.iron_bars.func_176223_P(), 9, 2, 5, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.iron_bars.func_176223_P(), 9, 2, 4, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 2, 4, 8, 2, 5, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175811_a(p_74875_1_, Blocks.flowing_lava.getBlockState(), 7, 1, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.flowing_lava.getBlockState(), 8, 1, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.iron_bars.getBlockState(), 9, 2, 5, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.iron_bars.getBlockState(), 9, 2, 4, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 2, 4, 8, 2, 5, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175811_a(p_74875_1_, iblockstate, 6, 1, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.furnace.func_176223_P(), 6, 2, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.furnace.func_176223_P(), 6, 3, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.double_stone_slab.func_176223_P(), 8, 1, 1, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 4, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 2, 6, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 2, 6, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.furnace.getBlockState(), 6, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.furnace.getBlockState(), 6, 3, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.double_stone_slab.getBlockState(), 8, 1, 1, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 4, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 2, 6, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 2, 6, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 2, 1, 4, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.func_176223_P(), 2, 2, 4, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.getBlockState(), 2, 2, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate3, 1, 1, 5, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 2, 1, 5, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate2, 1, 1, 4, p_74875_3_);
@@ -997,13 +997,13 @@
 
             for (int i = 6; i <= 8; ++i)
             {
-                if (this.func_175807_a(p_74875_1_, i, 0, -1, p_74875_3_).func_185904_a() == Material.air && this.func_175807_a(p_74875_1_, i, -1, -1, p_74875_3_).func_185904_a() != Material.air)
+                if (this.func_175807_a(p_74875_1_, i, 0, -1, p_74875_3_).getMaterial() == Material.air && this.func_175807_a(p_74875_1_, i, -1, -1, p_74875_3_).getMaterial() != Material.air)
                 {
                     this.func_175811_a(p_74875_1_, iblockstate4, i, 0, -1, p_74875_3_);
 
-                    if (this.func_175807_a(p_74875_1_, i, -1, -1, p_74875_3_).func_177230_c() == Blocks.field_185774_da)
+                    if (this.func_175807_a(p_74875_1_, i, -1, -1, p_74875_3_).getBlock() == Blocks.field_185774_da)
                     {
-                        this.func_175811_a(p_74875_1_, Blocks.grass.func_176223_P(), i, -1, -1, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.grass.getBlockState(), i, -1, -1, p_74875_3_);
                     }
                 }
             }
@@ -1060,15 +1060,15 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 7 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.func_176223_P());
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH));
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST));
-            IBlockState iblockstate4 = this.func_175847_a(Blocks.oak_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
-            IBlockState iblockstate5 = this.func_175847_a(Blocks.planks.func_176223_P());
-            IBlockState iblockstate6 = this.func_175847_a(Blocks.log.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 7, 4, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 1, 6, 8, 4, 10, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.getBlockState());
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH));
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST));
+            IBlockState iblockstate4 = this.func_175847_a(Blocks.oak_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST));
+            IBlockState iblockstate5 = this.func_175847_a(Blocks.planks.getBlockState());
+            IBlockState iblockstate6 = this.func_175847_a(Blocks.log.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 7, 4, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 1, 6, 8, 4, 10, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 2, 0, 5, 8, 0, 10, iblockstate5, iblockstate5, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 7, 0, 4, iblockstate5, iblockstate5, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 0, 3, 5, iblockstate, iblockstate, false);
@@ -1135,41 +1135,41 @@
 
             this.func_175811_a(p_74875_1_, iblockstate6, 0, 2, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 0, 2, 4, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 3, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 4, 2, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 5, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 5, 2, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 6, 2, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 8, 2, 1, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 3, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 8, 2, 4, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate5, 8, 2, 5, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 8, 2, 6, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 7, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 8, 2, 8, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 7, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 8, 2, 8, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 8, 2, 9, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 2, 2, 6, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 2, 7, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 2, 8, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 2, 7, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 2, 8, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 2, 2, 9, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 4, 4, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 5, 4, 10, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 5, 4, 10, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate6, 6, 4, 10, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate5, 5, 5, 10, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 1, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 1, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 2, 0, p_74875_3_);
             this.func_189926_a(p_74875_1_, EnumFacing.NORTH, 2, 3, 1, p_74875_3_);
             this.func_189927_a(p_74875_1_, p_74875_3_, p_74875_2_, 2, 1, 0, EnumFacing.NORTH);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, -1, 3, 2, -1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, -1, 3, 2, -1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
 
-            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).func_185904_a() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_185904_a() != Material.air)
+            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).getMaterial() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getMaterial() != Material.air)
             {
                 this.func_175811_a(p_74875_1_, iblockstate7, 2, 0, -1, p_74875_3_);
 
-                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_177230_c() == Blocks.field_185774_da)
+                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getBlock() == Blocks.field_185774_da)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.grass.func_176223_P(), 2, -1, -1, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.grass.getBlockState(), 2, -1, -1, p_74875_3_);
                 }
             }
 
@@ -1244,11 +1244,11 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 6 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.func_176223_P());
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.planks.func_176223_P());
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.log.func_176223_P());
-            IBlockState iblockstate4 = this.func_175847_a(Blocks.field_180407_aO.func_176223_P());
+            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.getBlockState());
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.planks.getBlockState());
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.log.getBlockState());
+            IBlockState iblockstate4 = this.func_175847_a(Blocks.field_180407_aO.getBlockState());
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 0, 4, iblockstate, iblockstate, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 4, 0, 4, 4, 4, iblockstate3, iblockstate3, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 1, 3, 4, 3, iblockstate1, iblockstate1, false);
@@ -1267,9 +1267,9 @@
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 3, 3, iblockstate1, iblockstate1, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 3, 3, iblockstate1, iblockstate1, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 4, 3, 3, 4, iblockstate1, iblockstate1, false);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 2, 2, 4, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 4, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 2, 2, 4, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 4, 2, 2, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 1, 1, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 1, 2, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 1, 3, 0, p_74875_3_);
@@ -1278,17 +1278,17 @@
             this.func_175811_a(p_74875_1_, iblockstate1, 3, 2, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 3, 1, 0, p_74875_3_);
 
-            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).func_185904_a() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_185904_a() != Material.air)
+            if (this.func_175807_a(p_74875_1_, 2, 0, -1, p_74875_3_).getMaterial() == Material.air && this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getMaterial() != Material.air)
             {
                 this.func_175811_a(p_74875_1_, iblockstate2, 2, 0, -1, p_74875_3_);
 
-                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).func_177230_c() == Blocks.field_185774_da)
+                if (this.func_175807_a(p_74875_1_, 2, -1, -1, p_74875_3_).getBlock() == Blocks.field_185774_da)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.grass.func_176223_P(), 2, -1, -1, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.grass.getBlockState(), 2, -1, -1, p_74875_3_);
                 }
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 3, 3, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 3, 3, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
 
             if (this.isRoofAccessible)
             {
@@ -1312,7 +1312,7 @@
 
             if (this.isRoofAccessible)
             {
-                IBlockState iblockstate5 = Blocks.ladder.func_176223_P().func_177226_a(BlockLadder.field_176382_a, EnumFacing.SOUTH);
+                IBlockState iblockstate5 = Blocks.ladder.getBlockState().func_177226_a(BlockLadder.field_176382_a, EnumFacing.SOUTH);
                 this.func_175811_a(p_74875_1_, iblockstate5, 3, 1, 3, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate5, 3, 2, 3, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate5, 3, 3, 3, p_74875_3_);
@@ -1453,10 +1453,10 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            IBlockState iblockstate = this.func_175847_a(Blocks.field_185774_da.func_176223_P());
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.planks.func_176223_P());
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.gravel.func_176223_P());
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.cobblestone.func_176223_P());
+            IBlockState iblockstate = this.func_175847_a(Blocks.field_185774_da.getBlockState());
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.planks.getBlockState());
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.gravel.getBlockState());
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.cobblestone.getBlockState());
 
             for (int i = this.boundingBox.minX; i <= this.boundingBox.maxX; ++i)
             {
@@ -1466,37 +1466,37 @@
 
                     if (p_74875_3_.func_175898_b(blockpos))
                     {
-                        blockpos = p_74875_1_.func_175672_r(blockpos).func_177977_b();
+                        blockpos = p_74875_1_.func_175672_r(blockpos).getBlockPosBelow();
 
-                        if (blockpos.func_177956_o() < p_74875_1_.func_181545_F())
+                        if (blockpos.getY() < p_74875_1_.getSpawnLevel())
                         {
-                            blockpos = new BlockPos(blockpos.func_177958_n(), p_74875_1_.func_181545_F() - 1, blockpos.func_177952_p());
+                            blockpos = new BlockPos(blockpos.getX(), p_74875_1_.getSpawnLevel() - 1, blockpos.getZ());
                         }
 
-                        while (blockpos.func_177956_o() >= p_74875_1_.func_181545_F() - 1)
+                        while (blockpos.getY() >= p_74875_1_.getSpawnLevel() - 1)
                         {
-                            IBlockState iblockstate4 = p_74875_1_.func_180495_p(blockpos);
+                            IBlockState iblockstate4 = p_74875_1_.getBlockState(blockpos);
 
-                            if (iblockstate4.func_177230_c() == Blocks.grass && p_74875_1_.func_175623_d(blockpos.func_177984_a()))
+                            if (iblockstate4.getBlock() == Blocks.grass && p_74875_1_.isEmpty(blockpos.getBlockPosAbove()))
                             {
                                 p_74875_1_.func_180501_a(blockpos, iblockstate, 2);
                                 break;
                             }
 
-                            if (iblockstate4.func_185904_a().isLiquid())
+                            if (iblockstate4.getMaterial().isLiquid())
                             {
                                 p_74875_1_.func_180501_a(blockpos, iblockstate1, 2);
                                 break;
                             }
 
-                            if (iblockstate4.func_177230_c() == Blocks.sand || iblockstate4.func_177230_c() == Blocks.sandstone || iblockstate4.func_177230_c() == Blocks.field_180395_cM)
+                            if (iblockstate4.getBlock() == Blocks.sand || iblockstate4.getBlock() == Blocks.sandstone || iblockstate4.getBlock() == Blocks.field_180395_cM)
                             {
                                 p_74875_1_.func_180501_a(blockpos, iblockstate2, 2);
-                                p_74875_1_.func_180501_a(blockpos.func_177977_b(), iblockstate3, 2);
+                                p_74875_1_.func_180501_a(blockpos.getBlockPosBelow(), iblockstate3, 2);
                                 break;
                             }
 
-                            blockpos = blockpos.func_177977_b();
+                            blockpos = blockpos.getBlockPosBelow();
                         }
                     }
                 }
@@ -1615,8 +1615,8 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 4 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.field_180407_aO.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 2, 3, 1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            IBlockState iblockstate = this.func_175847_a(Blocks.field_180407_aO.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 2, 3, 1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175811_a(p_74875_1_, iblockstate, 1, 0, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate, 1, 1, 0, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate, 1, 2, 0, p_74875_3_);
@@ -1743,7 +1743,7 @@
 
                     if (p_74889_2_.func_175898_b(blockpos$mutableblockpos))
                     {
-                        i += Math.max(p_74889_1_.func_175672_r(blockpos$mutableblockpos).func_177956_o(), p_74889_1_.provider.getAverageGroundLevel() - 1);
+                        i += Math.max(p_74889_1_.func_175672_r(blockpos$mutableblockpos).getY(), p_74889_1_.provider.getAverageGroundLevel() - 1);
                         ++j;
                     }
                 }
@@ -1811,83 +1811,83 @@
         {
             if (this.field_189928_h == 1)
             {
-                if (p_175847_1_.func_177230_c() == Blocks.log || p_175847_1_.func_177230_c() == Blocks.log2)
+                if (p_175847_1_.getBlock() == Blocks.log || p_175847_1_.getBlock() == Blocks.log2)
                 {
-                    return Blocks.sandstone.func_176223_P();
+                    return Blocks.sandstone.getBlockState();
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.cobblestone)
+                if (p_175847_1_.getBlock() == Blocks.cobblestone)
                 {
                     return Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.DEFAULT.func_176675_a());
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.planks)
+                if (p_175847_1_.getBlock() == Blocks.planks)
                 {
                     return Blocks.sandstone.func_176203_a(BlockSandStone.EnumType.SMOOTH.func_176675_a());
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.oak_stairs)
+                if (p_175847_1_.getBlock() == Blocks.oak_stairs)
                 {
-                    return Blocks.sandstone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
+                    return Blocks.sandstone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.stone_stairs)
+                if (p_175847_1_.getBlock() == Blocks.stone_stairs)
                 {
-                    return Blocks.sandstone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
+                    return Blocks.sandstone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.gravel)
+                if (p_175847_1_.getBlock() == Blocks.gravel)
                 {
-                    return Blocks.sandstone.func_176223_P();
+                    return Blocks.sandstone.getBlockState();
                 }
             }
             else if (this.field_189928_h == 3)
             {
-                if (p_175847_1_.func_177230_c() == Blocks.log || p_175847_1_.func_177230_c() == Blocks.log2)
+                if (p_175847_1_.getBlock() == Blocks.log || p_175847_1_.getBlock() == Blocks.log2)
                 {
-                    return Blocks.log.func_176223_P().func_177226_a(BlockOldLog.field_176301_b, BlockPlanks.EnumType.SPRUCE).func_177226_a(BlockLog.field_176299_a, p_175847_1_.func_177229_b(BlockLog.field_176299_a));
+                    return Blocks.log.getBlockState().func_177226_a(BlockOldLog.field_176301_b, BlockPlanks.EnumType.SPRUCE).func_177226_a(BlockLog.field_176299_a, p_175847_1_.func_177229_b(BlockLog.field_176299_a));
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.planks)
+                if (p_175847_1_.getBlock() == Blocks.planks)
                 {
-                    return Blocks.planks.func_176223_P().func_177226_a(BlockPlanks.field_176383_a, BlockPlanks.EnumType.SPRUCE);
+                    return Blocks.planks.getBlockState().func_177226_a(BlockPlanks.field_176383_a, BlockPlanks.EnumType.SPRUCE);
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.oak_stairs)
+                if (p_175847_1_.getBlock() == Blocks.oak_stairs)
                 {
-                    return Blocks.spruce_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
+                    return Blocks.spruce_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.field_180407_aO)
+                if (p_175847_1_.getBlock() == Blocks.field_180407_aO)
                 {
-                    return Blocks.field_180408_aP.func_176223_P();
+                    return Blocks.field_180408_aP.getBlockState();
                 }
             }
             else if (this.field_189928_h == 2)
             {
-                if (p_175847_1_.func_177230_c() == Blocks.log || p_175847_1_.func_177230_c() == Blocks.log2)
+                if (p_175847_1_.getBlock() == Blocks.log || p_175847_1_.getBlock() == Blocks.log2)
                 {
-                    return Blocks.log2.func_176223_P().func_177226_a(BlockNewLog.field_176300_b, BlockPlanks.EnumType.ACACIA).func_177226_a(BlockLog.field_176299_a, p_175847_1_.func_177229_b(BlockLog.field_176299_a));
+                    return Blocks.log2.getBlockState().func_177226_a(BlockNewLog.field_176300_b, BlockPlanks.EnumType.ACACIA).func_177226_a(BlockLog.field_176299_a, p_175847_1_.func_177229_b(BlockLog.field_176299_a));
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.planks)
+                if (p_175847_1_.getBlock() == Blocks.planks)
                 {
-                    return Blocks.planks.func_176223_P().func_177226_a(BlockPlanks.field_176383_a, BlockPlanks.EnumType.ACACIA);
+                    return Blocks.planks.getBlockState().func_177226_a(BlockPlanks.field_176383_a, BlockPlanks.EnumType.ACACIA);
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.oak_stairs)
+                if (p_175847_1_.getBlock() == Blocks.oak_stairs)
                 {
-                    return Blocks.acacia_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
+                    return Blocks.acacia_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, p_175847_1_.func_177229_b(BlockStairs.field_176309_a));
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.cobblestone)
+                if (p_175847_1_.getBlock() == Blocks.cobblestone)
                 {
-                    return Blocks.log2.func_176223_P().func_177226_a(BlockNewLog.field_176300_b, BlockPlanks.EnumType.ACACIA).func_177226_a(BlockLog.field_176299_a, BlockLog.EnumAxis.Y);
+                    return Blocks.log2.getBlockState().func_177226_a(BlockNewLog.field_176300_b, BlockPlanks.EnumType.ACACIA).func_177226_a(BlockLog.field_176299_a, BlockLog.EnumAxis.Y);
                 }
 
-                if (p_175847_1_.func_177230_c() == Blocks.field_180407_aO)
+                if (p_175847_1_.getBlock() == Blocks.field_180407_aO)
                 {
-                    return Blocks.field_180405_aT.func_176223_P();
+                    return Blocks.field_180405_aT.getBlockState();
                 }
             }
 
@@ -1921,7 +1921,7 @@
         {
             if (!this.field_189929_i)
             {
-                this.func_175811_a(p_189926_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, p_189926_2_), p_189926_3_, p_189926_4_, p_189926_5_, p_189926_6_);
+                this.func_175811_a(p_189926_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, p_189926_2_), p_189926_3_, p_189926_4_, p_189926_5_, p_189926_6_);
             }
         }
 
@@ -1980,13 +1980,13 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 3, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.func_176223_P());
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.field_180407_aO.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 4, 12, 4, iblockstate, Blocks.flowing_water.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 12, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 3, 12, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, 12, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 3, 12, 3, p_74875_3_);
+            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.getBlockState());
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.field_180407_aO.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 4, 12, 4, iblockstate, Blocks.flowing_water.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 12, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 3, 12, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, 12, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 3, 12, 3, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 1, 13, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 1, 14, 1, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 4, 13, 1, p_74875_3_);
@@ -2003,7 +2003,7 @@
                 {
                     if (j == 0 || j == 5 || i == 0 || i == 5)
                     {
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), j, 11, i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), j, 11, i, p_74875_3_);
                         this.clearCurrentPositionBlocksUpwards(p_74875_1_, j, 12, i, p_74875_3_);
                     }
                 }
@@ -2065,14 +2065,14 @@
                 this.boundingBox.offset(0, this.field_143015_k - this.boundingBox.maxY + 6 - 1, 0);
             }
 
-            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.func_176223_P());
-            IBlockState iblockstate1 = this.func_175847_a(Blocks.planks.func_176223_P());
-            IBlockState iblockstate2 = this.func_175847_a(Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
-            IBlockState iblockstate3 = this.func_175847_a(Blocks.log.func_176223_P());
-            IBlockState iblockstate4 = this.func_175847_a(Blocks.field_180407_aO.func_176223_P());
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 3, 5, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+            IBlockState iblockstate = this.func_175847_a(Blocks.cobblestone.getBlockState());
+            IBlockState iblockstate1 = this.func_175847_a(Blocks.planks.getBlockState());
+            IBlockState iblockstate2 = this.func_175847_a(Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH));
+            IBlockState iblockstate3 = this.func_175847_a(Blocks.log.getBlockState());
+            IBlockState iblockstate4 = this.func_175847_a(Blocks.field_180407_aO.getBlockState());
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 3, 5, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 3, 0, 4, iblockstate, iblockstate, false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 2, 0, 3, Blocks.dirt.func_176223_P(), Blocks.dirt.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 1, 2, 0, 3, Blocks.dirt.getBlockState(), Blocks.dirt.getBlockState(), false);
 
             if (this.isTallHouse)
             {
@@ -2101,26 +2101,26 @@
             this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, 1, 3, 3, 3, iblockstate1, iblockstate1, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 0, 2, 3, 0, iblockstate1, iblockstate1, false);
             this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 4, 2, 3, 4, iblockstate1, iblockstate1, false);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 0, 2, 2, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.glass_pane.func_176223_P(), 3, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 0, 2, 2, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.glass_pane.getBlockState(), 3, 2, 2, p_74875_3_);
 
             if (this.tablePosition > 0)
             {
                 this.func_175811_a(p_74875_1_, iblockstate4, this.tablePosition, 1, 3, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.func_176223_P(), this.tablePosition, 2, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.wooden_pressure_plate.getBlockState(), this.tablePosition, 2, 3, p_74875_3_);
             }
 
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, 1, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, 2, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, 1, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, 2, 0, p_74875_3_);
             this.func_189927_a(p_74875_1_, p_74875_3_, p_74875_2_, 1, 1, 0, EnumFacing.NORTH);
 
-            if (this.func_175807_a(p_74875_1_, 1, 0, -1, p_74875_3_).func_185904_a() == Material.air && this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).func_185904_a() != Material.air)
+            if (this.func_175807_a(p_74875_1_, 1, 0, -1, p_74875_3_).getMaterial() == Material.air && this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).getMaterial() != Material.air)
             {
                 this.func_175811_a(p_74875_1_, iblockstate2, 1, 0, -1, p_74875_3_);
 
-                if (this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).func_177230_c() == Blocks.field_185774_da)
+                if (this.func_175807_a(p_74875_1_, 1, -1, -1, p_74875_3_).getBlock() == Blocks.field_185774_da)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.grass.func_176223_P(), 1, -1, -1, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.grass.getBlockState(), 1, -1, -1, p_74875_3_);
                 }
             }
 
