--- a/net/minecraft/world/gen/structure/StructureMineshaftPieces.java
+++ b/net/minecraft/world/gen/structure/StructureMineshaftPieces.java
@@ -297,11 +297,11 @@
         {
             BlockPos blockpos = new BlockPos(this.getXWithOffset(p_186167_4_, p_186167_6_), this.getYWithOffset(p_186167_5_), this.getZWithOffset(p_186167_4_, p_186167_6_));
 
-            if (p_186167_2_.func_175898_b(blockpos) && p_186167_1_.func_180495_p(blockpos).func_185904_a() == Material.air && p_186167_1_.func_180495_p(blockpos.func_177977_b()).func_185904_a() != Material.air)
+            if (p_186167_2_.func_175898_b(blockpos) && p_186167_1_.getBlockState(blockpos).getMaterial() == Material.air && p_186167_1_.getBlockState(blockpos.getBlockPosBelow()).getMaterial() != Material.air)
             {
-                IBlockState iblockstate = Blocks.rail.func_176223_P().func_177226_a(BlockRail.field_176565_b, p_186167_3_.nextBoolean() ? BlockRailBase.EnumRailDirection.NORTH_SOUTH : BlockRailBase.EnumRailDirection.EAST_WEST);
+                IBlockState iblockstate = Blocks.rail.getBlockState().func_177226_a(BlockRail.field_176565_b, p_186167_3_.nextBoolean() ? BlockRailBase.EnumRailDirection.NORTH_SOUTH : BlockRailBase.EnumRailDirection.EAST_WEST);
                 this.func_175811_a(p_186167_1_, iblockstate, p_186167_4_, p_186167_5_, p_186167_6_, p_186167_2_);
-                EntityMinecartChest entityminecartchest = new EntityMinecartChest(p_186167_1_, (double)((float)blockpos.func_177958_n() + 0.5F), (double)((float)blockpos.func_177956_o() + 0.5F), (double)((float)blockpos.func_177952_p() + 0.5F));
+                EntityMinecartChest entityminecartchest = new EntityMinecartChest(p_186167_1_, (double)((float)blockpos.getX() + 0.5F), (double)((float)blockpos.getY() + 0.5F), (double)((float)blockpos.getZ() + 0.5F));
                 entityminecartchest.func_184289_a(p_186167_7_, p_186167_3_.nextLong());
                 p_186167_1_.spawnEntityInWorld(entityminecartchest);
                 return true;
@@ -326,12 +326,12 @@
                 int l = 2;
                 int i1 = this.sectionCount * 5 - 1;
                 IBlockState iblockstate = this.func_189917_F_();
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 2, 1, i1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                this.func_189914_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.8F, 0, 2, 0, 2, 2, i1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false, 0);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 2, 1, i1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                this.func_189914_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.8F, 0, 2, 0, 2, 2, i1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false, 0);
 
                 if (this.hasSpiders)
                 {
-                    this.func_189914_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.6F, 0, 0, 0, 2, 1, i1, Blocks.web.func_176223_P(), Blocks.air.func_176223_P(), false, 8);
+                    this.func_189914_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.6F, 0, 0, 0, 2, 1, i1, Blocks.web.getBlockState(), Blocks.air.getBlockState(), false, 8);
                 }
 
                 for (int j1 = 0; j1 < this.sectionCount; ++j1)
@@ -368,7 +368,7 @@
                         if (p_74875_3_.func_175898_b(blockpos) && this.func_189916_b(p_74875_1_, 1, 0, i2, p_74875_3_) < 8)
                         {
                             this.spawnerPlaced = true;
-                            p_74875_1_.func_180501_a(blockpos, Blocks.mob_spawner.func_176223_P(), 2);
+                            p_74875_1_.func_180501_a(blockpos, Blocks.mob_spawner.getBlockState(), 2);
                             TileEntity tileentity = p_74875_1_.func_175625_s(blockpos);
 
                             if (tileentity instanceof TileEntityMobSpawner)
@@ -386,7 +386,7 @@
                         int k3 = -1;
                         IBlockState iblockstate3 = this.func_175807_a(p_74875_1_, l2, -1, i3, p_74875_3_);
 
-                        if (iblockstate3.func_185904_a() == Material.air && this.func_189916_b(p_74875_1_, l2, -1, i3, p_74875_3_) < 8)
+                        if (iblockstate3.getMaterial() == Material.air && this.func_189916_b(p_74875_1_, l2, -1, i3, p_74875_3_) < 8)
                         {
                             int l3 = -1;
                             this.func_175811_a(p_74875_1_, iblockstate, l2, -1, i3, p_74875_3_);
@@ -396,13 +396,13 @@
 
                 if (this.hasRails)
                 {
-                    IBlockState iblockstate1 = Blocks.rail.func_176223_P().func_177226_a(BlockRail.field_176565_b, BlockRailBase.EnumRailDirection.NORTH_SOUTH);
+                    IBlockState iblockstate1 = Blocks.rail.getBlockState().func_177226_a(BlockRail.field_176565_b, BlockRailBase.EnumRailDirection.NORTH_SOUTH);
 
                     for (int j3 = 0; j3 <= i1; ++j3)
                     {
                         IBlockState iblockstate2 = this.func_175807_a(p_74875_1_, 1, -1, j3, p_74875_3_);
 
-                        if (iblockstate2.func_185904_a() != Material.air && iblockstate2.func_185913_b())
+                        if (iblockstate2.getMaterial() != Material.air && iblockstate2.func_185913_b())
                         {
                             float f = this.func_189916_b(p_74875_1_, 1, 0, j3, p_74875_3_) > 8 ? 0.9F : 0.7F;
                             this.func_175809_a(p_74875_1_, p_74875_3_, p_74875_2_, f, 1, 0, j3, iblockstate1);
@@ -420,7 +420,7 @@
             {
                 IBlockState iblockstate = this.func_189917_F_();
                 IBlockState iblockstate1 = this.func_189919_b();
-                IBlockState iblockstate2 = Blocks.air.func_176223_P();
+                IBlockState iblockstate2 = Blocks.air.getBlockState();
                 this.func_175804_a(p_189921_1_, p_189921_2_, p_189921_3_, p_189921_4_, p_189921_5_, p_189921_3_, p_189921_6_ - 1, p_189921_5_, iblockstate1, iblockstate2, false);
                 this.func_175804_a(p_189921_1_, p_189921_2_, p_189921_7_, p_189921_4_, p_189921_5_, p_189921_7_, p_189921_6_ - 1, p_189921_5_, iblockstate1, iblockstate2, false);
 
@@ -432,8 +432,8 @@
                 else
                 {
                     this.func_175804_a(p_189921_1_, p_189921_2_, p_189921_3_, p_189921_6_, p_189921_5_, p_189921_7_, p_189921_6_, p_189921_5_, iblockstate, iblockstate2, false);
-                    this.func_175809_a(p_189921_1_, p_189921_2_, p_189921_8_, 0.05F, p_189921_3_ + 1, p_189921_6_, p_189921_5_ - 1, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.NORTH));
-                    this.func_175809_a(p_189921_1_, p_189921_2_, p_189921_8_, 0.05F, p_189921_3_ + 1, p_189921_6_, p_189921_5_ + 1, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.SOUTH));
+                    this.func_175809_a(p_189921_1_, p_189921_2_, p_189921_8_, 0.05F, p_189921_3_ + 1, p_189921_6_, p_189921_5_ - 1, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.NORTH));
+                    this.func_175809_a(p_189921_1_, p_189921_2_, p_189921_8_, 0.05F, p_189921_3_ + 1, p_189921_6_, p_189921_5_ + 1, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.SOUTH));
                 }
             }
         }
@@ -442,7 +442,7 @@
         {
             if (this.func_189916_b(p_189922_1_, p_189922_5_, p_189922_6_, p_189922_7_, p_189922_2_) < 8)
             {
-                this.func_175809_a(p_189922_1_, p_189922_2_, p_189922_3_, p_189922_4_, p_189922_5_, p_189922_6_, p_189922_7_, Blocks.web.func_176223_P());
+                this.func_175809_a(p_189922_1_, p_189922_2_, p_189922_3_, p_189922_4_, p_189922_5_, p_189922_6_, p_189922_7_, Blocks.web.getBlockState());
             }
         }
     }
@@ -584,16 +584,16 @@
 
                 if (this.isMultipleFloors)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ, this.boundingBox.maxX - 1, this.boundingBox.minY + 3 - 1, this.boundingBox.maxZ, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY, this.boundingBox.minZ + 1, this.boundingBox.maxX, this.boundingBox.minY + 3 - 1, this.boundingBox.maxZ - 1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.maxY - 2, this.boundingBox.minZ, this.boundingBox.maxX - 1, this.boundingBox.maxY, this.boundingBox.maxZ, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.maxY - 2, this.boundingBox.minZ + 1, this.boundingBox.maxX, this.boundingBox.maxY, this.boundingBox.maxZ - 1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.minY + 3, this.boundingBox.minZ + 1, this.boundingBox.maxX - 1, this.boundingBox.minY + 3, this.boundingBox.maxZ - 1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ, this.boundingBox.maxX - 1, this.boundingBox.minY + 3 - 1, this.boundingBox.maxZ, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY, this.boundingBox.minZ + 1, this.boundingBox.maxX, this.boundingBox.minY + 3 - 1, this.boundingBox.maxZ - 1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.maxY - 2, this.boundingBox.minZ, this.boundingBox.maxX - 1, this.boundingBox.maxY, this.boundingBox.maxZ, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.maxY - 2, this.boundingBox.minZ + 1, this.boundingBox.maxX, this.boundingBox.maxY, this.boundingBox.maxZ - 1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.minY + 3, this.boundingBox.minZ + 1, this.boundingBox.maxX - 1, this.boundingBox.minY + 3, this.boundingBox.maxZ - 1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
                 else
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ, this.boundingBox.maxX - 1, this.boundingBox.maxY, this.boundingBox.maxZ, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY, this.boundingBox.minZ + 1, this.boundingBox.maxX, this.boundingBox.maxY, this.boundingBox.maxZ - 1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ, this.boundingBox.maxX - 1, this.boundingBox.maxY, this.boundingBox.maxZ, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY, this.boundingBox.minZ + 1, this.boundingBox.maxX, this.boundingBox.maxY, this.boundingBox.maxZ - 1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 this.func_189923_b(p_74875_1_, p_74875_3_, this.boundingBox.minX + 1, this.boundingBox.minY, this.boundingBox.minZ + 1, this.boundingBox.maxY);
@@ -605,7 +605,7 @@
                 {
                     for (int j = this.boundingBox.minZ; j <= this.boundingBox.maxZ; ++j)
                     {
-                        if (this.func_175807_a(p_74875_1_, i, this.boundingBox.minY - 1, j, p_74875_3_).func_185904_a() == Material.air && this.func_189916_b(p_74875_1_, i, this.boundingBox.minY - 1, j, p_74875_3_) < 8)
+                        if (this.func_175807_a(p_74875_1_, i, this.boundingBox.minY - 1, j, p_74875_3_).getMaterial() == Material.air && this.func_189916_b(p_74875_1_, i, this.boundingBox.minY - 1, j, p_74875_3_) < 8)
                         {
                             this.func_175811_a(p_74875_1_, iblockstate, i, this.boundingBox.minY - 1, j, p_74875_3_);
                         }
@@ -618,9 +618,9 @@
 
         private void func_189923_b(World p_189923_1_, StructureBoundingBox p_189923_2_, int p_189923_3_, int p_189923_4_, int p_189923_5_, int p_189923_6_)
         {
-            if (this.func_175807_a(p_189923_1_, p_189923_3_, p_189923_6_ + 1, p_189923_5_, p_189923_2_).func_185904_a() != Material.air)
+            if (this.func_175807_a(p_189923_1_, p_189923_3_, p_189923_6_ + 1, p_189923_5_, p_189923_2_).getMaterial() != Material.air)
             {
-                this.func_175804_a(p_189923_1_, p_189923_2_, p_189923_3_, p_189923_4_, p_189923_5_, p_189923_3_, p_189923_6_, p_189923_5_, this.func_189917_F_(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_189923_1_, p_189923_2_, p_189923_3_, p_189923_4_, p_189923_5_, p_189923_3_, p_189923_6_, p_189923_5_, this.func_189917_F_(), Blocks.air.getBlockState(), false);
             }
         }
     }
@@ -655,10 +655,10 @@
             {
                 case NORMAL:
                 default:
-                    return Blocks.planks.func_176223_P();
+                    return Blocks.planks.getBlockState();
 
                 case MESA:
-                    return Blocks.planks.func_176223_P().func_177226_a(BlockPlanks.field_176383_a, BlockPlanks.EnumType.DARK_OAK);
+                    return Blocks.planks.getBlockState().func_177226_a(BlockPlanks.field_176383_a, BlockPlanks.EnumType.DARK_OAK);
             }
         }
 
@@ -668,10 +668,10 @@
             {
                 case NORMAL:
                 default:
-                    return Blocks.field_180407_aO.func_176223_P();
+                    return Blocks.field_180407_aO.getBlockState();
 
                 case MESA:
-                    return Blocks.field_180406_aS.func_176223_P();
+                    return Blocks.field_180406_aS.getBlockState();
             }
         }
 
@@ -679,7 +679,7 @@
         {
             for (int i = p_189918_3_; i <= p_189918_4_; ++i)
             {
-                if (this.func_175807_a(p_189918_1_, i, p_189918_5_ + 1, p_189918_6_, p_189918_2_).func_185904_a() == Material.air)
+                if (this.func_175807_a(p_189918_1_, i, p_189918_5_ + 1, p_189918_6_, p_189918_2_).getMaterial() == Material.air)
                 {
                     return false;
                 }
@@ -797,15 +797,15 @@
             }
             else
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY, this.boundingBox.minZ, this.boundingBox.maxX, this.boundingBox.minY, this.boundingBox.maxZ, Blocks.dirt.func_176223_P(), Blocks.air.func_176223_P(), true);
-                this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY + 1, this.boundingBox.minZ, this.boundingBox.maxX, Math.min(this.boundingBox.minY + 3, this.boundingBox.maxY), this.boundingBox.maxZ, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY, this.boundingBox.minZ, this.boundingBox.maxX, this.boundingBox.minY, this.boundingBox.maxZ, Blocks.dirt.getBlockState(), Blocks.air.getBlockState(), true);
+                this.func_175804_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY + 1, this.boundingBox.minZ, this.boundingBox.maxX, Math.min(this.boundingBox.minY + 3, this.boundingBox.maxY), this.boundingBox.maxZ, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
 
                 for (StructureBoundingBox structureboundingbox : this.roomsLinkedToTheRoom)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, structureboundingbox.minX, structureboundingbox.maxY - 2, structureboundingbox.minZ, structureboundingbox.maxX, structureboundingbox.maxY, structureboundingbox.maxZ, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, structureboundingbox.minX, structureboundingbox.maxY - 2, structureboundingbox.minZ, structureboundingbox.maxX, structureboundingbox.maxY, structureboundingbox.maxZ, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
-                this.func_180777_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY + 4, this.boundingBox.minZ, this.boundingBox.maxX, this.boundingBox.maxY, this.boundingBox.maxZ, Blocks.air.func_176223_P(), false);
+                this.func_180777_a(p_74875_1_, p_74875_3_, this.boundingBox.minX, this.boundingBox.minY + 4, this.boundingBox.minZ, this.boundingBox.maxX, this.boundingBox.maxY, this.boundingBox.maxZ, Blocks.air.getBlockState(), false);
                 return true;
             }
         }
@@ -934,12 +934,12 @@
             }
             else
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 2, 7, 1, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 7, 2, 2, 8, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 2, 7, 1, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 7, 2, 2, 8, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
 
                 for (int i = 0; i < 5; ++i)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5 - i - (i < 4 ? 1 : 0), 2 + i, 2, 7 - i, 2 + i, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5 - i - (i < 4 ? 1 : 0), 2 + i, 2, 7 - i, 2 + i, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 return true;
