--- a/net/minecraft/world/gen/structure/StructureNetherBridgePieces.java
+++ b/net/minecraft/world/gen/structure/StructureNetherBridgePieces.java
@@ -142,15 +142,15 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 1, 4, 4, 1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 3, 4, 4, 3, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 3, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 1, 4, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 3, 4, 3, 4, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 1, 4, 4, 1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 3, 4, 4, 3, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 3, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 1, 4, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 3, 4, 3, 4, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             if (this.field_111021_b && p_74875_3_.func_175898_b(new BlockPos(this.getXWithOffset(3, 3), this.getYWithOffset(2), this.getZWithOffset(3, 3))))
             {
@@ -158,13 +158,13 @@
                 this.func_186167_a(p_74875_1_, p_74875_3_, p_74875_2_, 3, 2, 3, LootTableList.field_186425_g);
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int i = 0; i <= 4; ++i)
             {
                 for (int j = 0; j <= 4; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
                 }
             }
 
@@ -213,15 +213,15 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 1, 0, 4, 1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 3, 0, 4, 3, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 4, 4, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 1, 4, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 3, 4, 3, 4, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 1, 0, 4, 1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 3, 0, 4, 3, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 4, 4, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 1, 4, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 3, 4, 3, 4, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             if (this.field_111020_b && p_74875_3_.func_175898_b(new BlockPos(this.getXWithOffset(1, 3), this.getYWithOffset(2), this.getZWithOffset(1, 3))))
             {
@@ -229,13 +229,13 @@
                 this.func_186167_a(p_74875_1_, p_74875_3_, p_74875_2_, 1, 2, 3, LootTableList.field_186425_g);
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int i = 0; i <= 4; ++i)
             {
                 for (int j = 0; j <= 4; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
                 }
             }
 
@@ -269,15 +269,15 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            IBlockState iblockstate = Blocks.nether_brick_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
+            IBlockState iblockstate = Blocks.nether_brick_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
 
             for (int i = 0; i <= 9; ++i)
             {
                 int j = Math.max(1, 7 - i);
                 int k = Math.min(Math.max(j + 5, 14 - i), 13);
                 int l = i;
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, i, 4, j, i, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, j + 1, i, 3, k - 1, i, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, i, 4, j, i, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, j + 1, i, 3, k - 1, i, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
 
                 if (i <= 6)
                 {
@@ -286,19 +286,19 @@
                     this.func_175811_a(p_74875_1_, iblockstate, 3, j + 1, i, p_74875_3_);
                 }
 
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, k, i, 4, k, i, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, j + 1, i, 0, k - 1, i, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 4, j + 1, i, 4, k - 1, i, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, k, i, 4, k, i, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, j + 1, i, 0, k - 1, i, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 4, j + 1, i, 4, k - 1, i, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
                 if ((i & 1) == 0)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, j + 2, i, 0, j + 3, i, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, j + 2, i, 4, j + 3, i, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, j + 2, i, 0, j + 3, i, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, j + 2, i, 4, j + 3, i, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
                 }
 
                 for (int i1 = 0; i1 <= 4; ++i1)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i1, -1, l, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i1, -1, l, p_74875_3_);
                 }
             }
 
@@ -341,31 +341,31 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 8, 1, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 8, 5, 8, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 8, 6, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 2, 5, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 0, 8, 5, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 0, 1, 4, 0, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 3, 0, 7, 4, 0, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 8, 2, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 4, 2, 2, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 1, 4, 7, 2, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 8, 8, 3, 8, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 6, 0, 3, 7, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 3, 6, 8, 3, 7, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 4, 0, 5, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 3, 4, 8, 5, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 5, 2, 5, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 3, 5, 7, 5, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 5, 1, 5, 5, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 4, 5, 7, 5, 5, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 8, 1, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 8, 5, 8, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 8, 6, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 2, 5, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 0, 8, 5, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 0, 1, 4, 0, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 3, 0, 7, 4, 0, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 8, 2, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 4, 2, 2, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 1, 4, 7, 2, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 8, 8, 3, 8, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 6, 0, 3, 7, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 3, 6, 8, 3, 7, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 4, 0, 5, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 3, 4, 8, 5, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 5, 2, 5, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 3, 5, 7, 5, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 5, 1, 5, 5, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 4, 5, 7, 5, 5, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
 
             for (int i = 0; i <= 5; ++i)
             {
                 for (int j = 0; j <= 8; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), j, -1, i, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), j, -1, i, p_74875_3_);
                 }
             }
 
@@ -399,21 +399,21 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 1, 0, 4, 1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 3, 0, 4, 3, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 1, 4, 4, 1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 3, 4, 4, 3, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 1, 0, 4, 1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 3, 0, 4, 3, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 1, 4, 4, 1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 3, 4, 4, 3, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int i = 0; i <= 4; ++i)
             {
                 for (int j = 0; j <= 4; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
                 }
             }
 
@@ -449,30 +449,30 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 6, 1, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 6, 7, 6, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 1, 6, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 6, 1, 6, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 0, 6, 6, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 6, 6, 6, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 6, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 5, 0, 6, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 0, 6, 6, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 5, 6, 6, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 6, 0, 4, 6, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 5, 0, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 6, 6, 4, 6, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 6, 4, 5, 6, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 2, 0, 6, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 2, 0, 5, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 6, 2, 6, 6, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 5, 2, 6, 5, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 6, 1, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 6, 7, 6, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 1, 6, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 6, 1, 6, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 0, 6, 6, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 6, 6, 6, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 6, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 5, 0, 6, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 0, 6, 6, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 5, 6, 6, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 6, 0, 4, 6, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 5, 0, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 6, 6, 4, 6, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 6, 4, 5, 6, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 2, 0, 6, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 2, 0, 5, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 6, 2, 6, 6, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 5, 2, 6, 5, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
 
             for (int i = 0; i <= 6; ++i)
             {
                 for (int j = 0; j <= 6; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
                 }
             }
 
@@ -508,19 +508,19 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 0, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 4, 4, 5, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 5, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 0, 5, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 4, 5, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 0, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 4, 4, 5, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 0, 4, 6, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int i = 0; i <= 4; ++i)
             {
                 for (int j = 0; j <= 4; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
                 }
             }
 
@@ -571,43 +571,43 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 3, 0, 11, 4, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 7, 18, 4, 11, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 0, 10, 7, 18, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 8, 18, 7, 10, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 5, 0, 7, 5, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 5, 11, 7, 5, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 0, 11, 5, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 11, 11, 5, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 7, 7, 5, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 7, 18, 5, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 11, 7, 5, 11, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 11, 18, 5, 11, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 2, 0, 11, 2, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 2, 13, 11, 2, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 0, 0, 11, 1, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 0, 15, 11, 1, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 3, 0, 11, 4, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 7, 18, 4, 11, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 0, 10, 7, 18, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 8, 18, 7, 10, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 5, 0, 7, 5, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 5, 11, 7, 5, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 0, 11, 5, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 11, 11, 5, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 7, 7, 5, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 7, 18, 5, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 11, 7, 5, 11, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 11, 18, 5, 11, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 2, 0, 11, 2, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 2, 13, 11, 2, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 0, 0, 11, 1, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 0, 15, 11, 1, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int i = 7; i <= 11; ++i)
             {
                 for (int j = 0; j <= 2; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, 18 - j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, 18 - j, p_74875_3_);
                 }
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 7, 5, 2, 11, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 13, 2, 7, 18, 2, 11, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 7, 3, 1, 11, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 15, 0, 7, 18, 1, 11, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 7, 5, 2, 11, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 13, 2, 7, 18, 2, 11, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 7, 3, 1, 11, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 15, 0, 7, 18, 1, 11, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int k = 0; k <= 2; ++k)
             {
                 for (int l = 7; l <= 11; ++l)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), k, -1, l, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), 18 - k, -1, l, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), k, -1, l, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), 18 - k, -1, l, p_74875_3_);
                 }
             }
 
@@ -658,19 +658,19 @@
                 for (int j = 3; j <= 4; ++j)
                 {
                     int k = random.nextInt(8);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, i, j, 0, i, j, k, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, i, j, 0, i, j, k, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
                 }
             }
 
             int l = random.nextInt(8);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 0, 5, l, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 0, 5, l, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
             l = random.nextInt(8);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 0, 4, 5, l, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 0, 4, 5, l, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (l = 0; l <= 4; ++l)
             {
                 int i1 = random.nextInt(5);
-                this.func_175804_a(p_74875_1_, p_74875_3_, l, 2, 0, l, 2, i1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, l, 2, 0, l, 2, i1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
             }
 
             for (l = 0; l <= 4; ++l)
@@ -678,7 +678,7 @@
                 for (int j1 = 0; j1 <= 1; ++j1)
                 {
                     int k1 = random.nextInt(3);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, l, j1, 0, l, j1, k1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, l, j1, 0, l, j1, k1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
                 }
             }
 
@@ -712,59 +712,59 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 0, 12, 4, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 12, 13, 12, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 1, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 0, 12, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 11, 4, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 11, 10, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 11, 7, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 12, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 0, 10, 12, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 0, 7, 12, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 11, 2, 10, 12, 10, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 8, 0, 7, 8, 0, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 0, 12, 4, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 12, 13, 12, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 1, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 0, 12, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 11, 4, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 11, 10, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 11, 7, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 12, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 0, 10, 12, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 0, 7, 12, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 11, 2, 10, 12, 10, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 8, 0, 7, 8, 0, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
 
             for (int i = 1; i <= 11; i += 2)
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 0, i, 11, 0, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 12, i, 11, 12, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 10, i, 0, 11, i, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 12, 10, i, 12, 11, i, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, 13, 0, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, 13, 12, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), 0, 13, i, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), 12, 13, i, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), i + 1, 13, 0, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), i + 1, 13, 12, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, i + 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 12, 13, i + 1, p_74875_3_);
-            }
-
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, 12, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 12, 13, 0, p_74875_3_);
+                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 0, i, 11, 0, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 12, i, 11, 12, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 10, i, 0, 11, i, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 12, 10, i, 12, 11, i, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), i, 13, 0, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), i, 13, 12, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), 0, 13, i, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), 12, 13, i, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), i + 1, 13, 0, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), i + 1, 13, 12, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, i + 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 12, 13, i + 1, p_74875_3_);
+            }
+
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, 12, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 12, 13, 0, p_74875_3_);
 
             for (int k = 3; k <= 9; k += 2)
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 7, k, 1, 8, k, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 11, 7, k, 11, 8, k, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 7, k, 1, 8, k, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 11, 7, k, 11, 8, k, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 8, 2, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 12, 2, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 0, 8, 1, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 9, 8, 1, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 4, 3, 1, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 0, 4, 12, 1, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 8, 2, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 12, 2, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 0, 8, 1, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 9, 8, 1, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 4, 3, 1, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 0, 4, 12, 1, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int l = 4; l <= 8; ++l)
             {
                 for (int j = 0; j <= 2; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), l, -1, j, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), l, -1, 12 - j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), l, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), l, -1, 12 - j, p_74875_3_);
                 }
             }
 
@@ -772,15 +772,15 @@
             {
                 for (int j1 = 4; j1 <= 8; ++j1)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i1, -1, j1, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), 12 - i1, -1, j1, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i1, -1, j1, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), 12 - i1, -1, j1, p_74875_3_);
                 }
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, 5, 7, 5, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 1, 6, 6, 4, 6, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), 6, 0, 6, p_74875_3_);
-            IBlockState iblockstate = Blocks.flowing_lava.func_176223_P();
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, 5, 7, 5, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 1, 6, 6, 4, 6, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), 6, 0, 6, p_74875_3_);
+            IBlockState iblockstate = Blocks.flowing_lava.getBlockState();
             this.func_175811_a(p_74875_1_, iblockstate, 6, 5, 6, p_74875_3_);
             BlockPos blockpos = new BlockPos(this.getXWithOffset(6, 6), this.getYWithOffset(5), this.getZWithOffset(6, 6));
 
@@ -820,46 +820,46 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 0, 12, 4, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 12, 13, 12, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 1, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 0, 12, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 11, 4, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 11, 10, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 11, 7, 12, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 12, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 0, 10, 12, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 0, 7, 12, 1, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 11, 2, 10, 12, 10, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 0, 12, 4, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 12, 13, 12, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 1, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 11, 5, 0, 12, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 11, 4, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 11, 10, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 11, 7, 12, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 12, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 0, 10, 12, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 9, 0, 7, 12, 1, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 11, 2, 10, 12, 10, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int i = 1; i <= 11; i += 2)
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 0, i, 11, 0, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 12, i, 11, 12, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 10, i, 0, 11, i, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 12, 10, i, 12, 11, i, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, 13, 0, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, 13, 12, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), 0, 13, i, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), 12, 13, i, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), i + 1, 13, 0, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), i + 1, 13, 12, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, i + 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 12, 13, i + 1, p_74875_3_);
-            }
-
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, 12, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 0, 13, 0, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 12, 13, 0, p_74875_3_);
+                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 0, i, 11, 0, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, i, 10, 12, i, 11, 12, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 10, i, 0, 11, i, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 12, 10, i, 12, 11, i, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), i, 13, 0, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), i, 13, 12, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), 0, 13, i, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), 12, 13, i, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), i + 1, 13, 0, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), i + 1, 13, 12, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, i + 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 12, 13, i + 1, p_74875_3_);
+            }
+
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, 12, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 0, 13, 0, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 12, 13, 0, p_74875_3_);
 
             for (int j1 = 3; j1 <= 9; j1 += 2)
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 7, j1, 1, 8, j1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 11, 7, j1, 11, 8, j1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 7, j1, 1, 8, j1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 11, 7, j1, 11, 8, j1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
             }
 
-            IBlockState iblockstate = Blocks.nether_brick_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
+            IBlockState iblockstate = Blocks.nether_brick_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
 
             for (int j = 0; j <= 6; ++j)
             {
@@ -872,16 +872,16 @@
 
                 if (k >= 5 && k <= 8)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, k, 7, j + 4, k, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, k, 7, j + 4, k, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
                 }
                 else if (k >= 9 && k <= 10)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 5, 8, k, 7, j + 4, k, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 5, 8, k, 7, j + 4, k, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
                 }
 
                 if (j >= 1)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 5, 6 + j, k, 7, 9 + j, k, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 5, 6 + j, k, 7, 9 + j, k, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
             }
 
@@ -890,15 +890,15 @@
                 this.func_175811_a(p_74875_1_, iblockstate, k1, 12, 11, p_74875_3_);
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 6, 7, 5, 7, 7, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 6, 7, 7, 7, 7, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 13, 12, 7, 13, 12, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 2, 3, 5, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 9, 3, 5, 10, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 4, 2, 5, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 5, 2, 10, 5, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 5, 9, 10, 5, 10, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 10, 5, 4, 10, 5, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 6, 7, 5, 7, 7, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 7, 6, 7, 7, 7, 7, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 13, 12, 7, 13, 12, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 2, 3, 5, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 9, 3, 5, 10, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 4, 2, 5, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 5, 2, 10, 5, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 5, 9, 10, 5, 10, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 10, 5, 4, 10, 5, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
             IBlockState iblockstate1 = iblockstate.func_177226_a(BlockStairs.field_176309_a, EnumFacing.EAST);
             IBlockState iblockstate2 = iblockstate.func_177226_a(BlockStairs.field_176309_a, EnumFacing.WEST);
             this.func_175811_a(p_74875_1_, iblockstate2, 4, 5, 2, p_74875_3_);
@@ -909,23 +909,23 @@
             this.func_175811_a(p_74875_1_, iblockstate1, 8, 5, 3, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 8, 5, 9, p_74875_3_);
             this.func_175811_a(p_74875_1_, iblockstate1, 8, 5, 10, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 4, 4, 4, 4, 8, Blocks.soul_sand.func_176223_P(), Blocks.soul_sand.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 4, 4, 9, 4, 8, Blocks.soul_sand.func_176223_P(), Blocks.soul_sand.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 5, 4, 4, 5, 8, Blocks.nether_wart.func_176223_P(), Blocks.nether_wart.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 4, 9, 5, 8, Blocks.nether_wart.func_176223_P(), Blocks.nether_wart.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 8, 2, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 12, 2, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 0, 8, 1, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 9, 8, 1, 12, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 4, 3, 1, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 0, 4, 12, 1, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 4, 4, 4, 4, 8, Blocks.soul_sand.getBlockState(), Blocks.soul_sand.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 4, 4, 9, 4, 8, Blocks.soul_sand.getBlockState(), Blocks.soul_sand.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 5, 4, 4, 5, 8, Blocks.nether_wart.getBlockState(), Blocks.nether_wart.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 4, 9, 5, 8, Blocks.nether_wart.getBlockState(), Blocks.nether_wart.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 0, 8, 2, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 4, 12, 2, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 0, 8, 1, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 0, 9, 8, 1, 12, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 4, 3, 1, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 0, 4, 12, 1, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int l1 = 4; l1 <= 8; ++l1)
             {
                 for (int i1 = 0; i1 <= 2; ++i1)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), l1, -1, i1, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), l1, -1, 12 - i1, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), l1, -1, i1, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), l1, -1, 12 - i1, p_74875_3_);
                 }
             }
 
@@ -933,8 +933,8 @@
             {
                 for (int j2 = 4; j2 <= 8; ++j2)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i2, -1, j2, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), 12 - i2, -1, j2, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i2, -1, j2, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), 12 - i2, -1, j2, p_74875_3_);
                 }
             }
 
@@ -1188,31 +1188,31 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 6, 1, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 6, 10, 6, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 1, 8, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 0, 6, 8, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 1, 0, 8, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 1, 6, 8, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 6, 5, 8, 6, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 2, 0, 5, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 3, 2, 6, 5, 2, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 3, 4, 6, 5, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick.func_176223_P(), 5, 2, 5, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 5, 4, 3, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 2, 5, 3, 4, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 2, 5, 2, 5, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 5, 1, 6, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 7, 1, 5, 7, 4, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 8, 2, 6, 8, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 6, 0, 4, 8, 0, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 5, 0, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 6, 1, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 6, 10, 6, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 1, 8, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 0, 6, 8, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 1, 0, 8, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 2, 1, 6, 8, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 6, 5, 8, 6, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 2, 0, 5, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 3, 2, 6, 5, 2, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 3, 4, 6, 5, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick.getBlockState(), 5, 2, 5, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 2, 5, 4, 3, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 3, 2, 5, 3, 4, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 2, 5, 2, 5, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 5, 1, 6, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 7, 1, 5, 7, 4, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 8, 2, 6, 8, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 6, 0, 4, 8, 0, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 5, 0, 4, 5, 0, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
 
             for (int i = 0; i <= 6; ++i)
             {
                 for (int j = 0; j <= 6; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
                 }
             }
 
@@ -1288,32 +1288,32 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 0, 4, 4, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 0, 3, 7, 18, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 0, 5, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 0, 4, 5, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 2, 5, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 13, 4, 2, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 15, 4, 1, 18, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 0, 4, 4, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 0, 3, 7, 18, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 0, 0, 5, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 0, 4, 5, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 4, 2, 5, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 13, 4, 2, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 1, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 0, 15, 4, 1, 18, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
 
             for (int i = 0; i <= 4; ++i)
             {
                 for (int j = 0; j <= 2; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, 18 - j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, 18 - j, p_74875_3_);
                 }
             }
 
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 4, 1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 4, 0, 4, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 14, 0, 4, 14, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 17, 0, 4, 17, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 4, 1, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 4, 4, 4, 4, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 14, 4, 4, 14, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 17, 4, 4, 17, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 4, 1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 4, 0, 4, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 14, 0, 4, 14, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 17, 0, 4, 17, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 4, 1, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 4, 4, 4, 4, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 3, 14, 4, 4, 14, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 17, 4, 4, 17, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
             return true;
         }
     }
@@ -1353,24 +1353,24 @@
 
         public boolean addComponentParts(World p_74875_1_, Random p_74875_2_, StructureBoundingBox p_74875_3_)
         {
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 6, 7, 7, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 0, 5, 1, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 1, 5, 2, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 2, 5, 3, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 3, 5, 4, 7, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 0, 1, 4, 2, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 0, 5, 4, 2, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 2, 1, 5, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, 2, 5, 5, 3, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 3, 0, 5, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 5, 3, 6, 5, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 8, 5, 5, 8, Blocks.nether_brick.func_176223_P(), Blocks.nether_brick.func_176223_P(), false);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 1, 6, 3, p_74875_3_);
-            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.func_176223_P(), 5, 6, 3, p_74875_3_);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 3, 0, 6, 8, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 6, 3, 6, 6, 8, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 6, 8, 5, 7, 8, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 8, 8, 4, 8, 8, Blocks.nether_brick_fence.func_176223_P(), Blocks.nether_brick_fence.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 2, 0, 6, 7, 7, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 0, 0, 5, 1, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 1, 5, 2, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 2, 5, 3, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 3, 5, 4, 7, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 2, 0, 1, 4, 2, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 0, 5, 4, 2, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 2, 1, 5, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, 2, 5, 5, 3, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 3, 0, 5, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 5, 3, 6, 5, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 8, 5, 5, 8, Blocks.nether_brick.getBlockState(), Blocks.nether_brick.getBlockState(), false);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 1, 6, 3, p_74875_3_);
+            this.func_175811_a(p_74875_1_, Blocks.nether_brick_fence.getBlockState(), 5, 6, 3, p_74875_3_);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 0, 6, 3, 0, 6, 8, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 6, 6, 3, 6, 6, 8, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 6, 8, 5, 7, 8, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 2, 8, 8, 4, 8, 8, Blocks.nether_brick_fence.getBlockState(), Blocks.nether_brick_fence.getBlockState(), false);
 
             if (!this.hasSpawner)
             {
@@ -1379,7 +1379,7 @@
                 if (p_74875_3_.func_175898_b(blockpos))
                 {
                     this.hasSpawner = true;
-                    p_74875_1_.func_180501_a(blockpos, Blocks.mob_spawner.func_176223_P(), 2);
+                    p_74875_1_.func_180501_a(blockpos, Blocks.mob_spawner.getBlockState(), 2);
                     TileEntity tileentity = p_74875_1_.func_175625_s(blockpos);
 
                     if (tileentity instanceof TileEntityMobSpawner)
@@ -1393,7 +1393,7 @@
             {
                 for (int j = 0; j <= 6; ++j)
                 {
-                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.func_176223_P(), i, -1, j, p_74875_3_);
+                    this.func_175808_b(p_74875_1_, Blocks.nether_brick.getBlockState(), i, -1, j, p_74875_3_);
                 }
             }
 
