--- a/net/minecraft/world/gen/structure/StructureBoundingBox.java
+++ b/net/minecraft/world/gen/structure/StructureBoundingBox.java
@@ -99,12 +99,12 @@
 
     public StructureBoundingBox(Vec3i p_i45626_1_, Vec3i p_i45626_2_)
     {
-        this.minX = Math.min(p_i45626_1_.func_177958_n(), p_i45626_2_.func_177958_n());
-        this.minY = Math.min(p_i45626_1_.func_177956_o(), p_i45626_2_.func_177956_o());
-        this.minZ = Math.min(p_i45626_1_.func_177952_p(), p_i45626_2_.func_177952_p());
-        this.maxX = Math.max(p_i45626_1_.func_177958_n(), p_i45626_2_.func_177958_n());
-        this.maxY = Math.max(p_i45626_1_.func_177956_o(), p_i45626_2_.func_177956_o());
-        this.maxZ = Math.max(p_i45626_1_.func_177952_p(), p_i45626_2_.func_177952_p());
+        this.minX = Math.min(p_i45626_1_.getX(), p_i45626_2_.getX());
+        this.minY = Math.min(p_i45626_1_.getY(), p_i45626_2_.getY());
+        this.minZ = Math.min(p_i45626_1_.getZ(), p_i45626_2_.getZ());
+        this.maxX = Math.max(p_i45626_1_.getX(), p_i45626_2_.getX());
+        this.maxY = Math.max(p_i45626_1_.getY(), p_i45626_2_.getY());
+        this.maxZ = Math.max(p_i45626_1_.getZ(), p_i45626_2_.getZ());
     }
 
     public StructureBoundingBox(int p_i2033_1_, int p_i2033_2_, int p_i2033_3_, int p_i2033_4_)
@@ -161,7 +161,7 @@
 
     public boolean func_175898_b(Vec3i p_175898_1_)
     {
-        return p_175898_1_.func_177958_n() >= this.minX && p_175898_1_.func_177958_n() <= this.maxX && p_175898_1_.func_177952_p() >= this.minZ && p_175898_1_.func_177952_p() <= this.maxZ && p_175898_1_.func_177956_o() >= this.minY && p_175898_1_.func_177956_o() <= this.maxY;
+        return p_175898_1_.getX() >= this.minX && p_175898_1_.getX() <= this.maxX && p_175898_1_.getZ() >= this.minZ && p_175898_1_.getZ() <= this.maxZ && p_175898_1_.getY() >= this.minY && p_175898_1_.getY() <= this.maxY;
     }
 
     public Vec3i func_175896_b()
