--- a/net/minecraft/world/gen/structure/StructureStart.java
+++ b/net/minecraft/world/gen/structure/StructureStart.java
@@ -119,7 +119,7 @@
      */
     protected void markAvailableHeight(World p_75067_1_, Random p_75067_2_, int p_75067_3_)
     {
-        int i = p_75067_1_.func_181545_F() - p_75067_3_;
+        int i = p_75067_1_.getSpawnLevel() - p_75067_3_;
         int j = this.boundingBox.getYSize() + 1;
 
         if (j < i)
