--- a/net/minecraft/world/gen/structure/StructureStrongholdPieces.java
+++ b/net/minecraft/world/gen/structure/StructureStrongholdPieces.java
@@ -290,7 +290,7 @@
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 4, 6, true, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, this.field_143013_d, 1, 1, 0);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, StructureStrongholdPieces.Stronghold.Door.OPENING, 1, 1, 6);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, 2, 3, 1, 4, Blocks.stonebrick.func_176223_P(), Blocks.stonebrick.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, 2, 3, 1, 4, Blocks.stonebrick.getBlockState(), Blocks.stonebrick.getBlockState(), false);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.SMOOTHBRICK.func_176624_a()), 3, 1, 1, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.SMOOTHBRICK.func_176624_a()), 3, 1, 5, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.SMOOTHBRICK.func_176624_a()), 3, 2, 2, p_74875_3_);
@@ -379,26 +379,26 @@
             {
                 for (int i = 0; i < this.field_74993_a; ++i)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 0, 0, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 0, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 2, 0, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3, 0, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 4, 0, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 0, 0, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 0, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 2, 0, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3, 0, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 4, 0, i, p_74875_3_);
 
                     for (int j = 1; j <= 3; ++j)
                     {
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 0, j, i, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 1, j, i, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 2, j, i, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 3, j, i, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 4, j, i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 0, j, i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 1, j, i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 2, j, i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 3, j, i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 4, j, i, p_74875_3_);
                     }
 
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 0, 4, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 4, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 2, 4, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3, 4, i, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 4, 4, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 0, 4, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 4, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 2, 4, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3, 4, i, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 4, 4, i, p_74875_3_);
                 }
 
                 return true;
@@ -501,39 +501,39 @@
 
                 if (this.field_74996_b)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 1, 0, 5, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, 1, 0, 5, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 if (this.field_74995_d)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 9, 3, 1, 9, 5, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 9, 3, 1, 9, 5, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 if (this.field_74997_c)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 7, 0, 7, 9, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 5, 7, 0, 7, 9, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 if (this.field_74999_h)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 9, 5, 7, 9, 7, 9, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 9, 5, 7, 9, 7, 9, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
-                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 1, 10, 7, 3, 10, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 1, 10, 7, 3, 10, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 1, 2, 1, 8, 2, 6, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 1, 5, 4, 4, 9, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 8, 1, 5, 8, 4, 9, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 1, 4, 7, 3, 4, 9, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 1, 3, 5, 3, 3, 6, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 3, 3, 4, Blocks.stone_slab.func_176223_P(), Blocks.stone_slab.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 6, 3, 4, 6, Blocks.stone_slab.func_176223_P(), Blocks.stone_slab.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 3, 4, 3, 3, 4, Blocks.stone_slab.getBlockState(), Blocks.stone_slab.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 4, 6, 3, 4, 6, Blocks.stone_slab.getBlockState(), Blocks.stone_slab.getBlockState(), false);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 5, 1, 7, 7, 1, 8, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 1, 9, 7, 1, 9, Blocks.stone_slab.func_176223_P(), Blocks.stone_slab.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 7, 7, 2, 7, Blocks.stone_slab.func_176223_P(), Blocks.stone_slab.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 7, 4, 5, 9, Blocks.stone_slab.func_176223_P(), Blocks.stone_slab.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 7, 8, 5, 9, Blocks.stone_slab.func_176223_P(), Blocks.stone_slab.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, 7, 7, 5, 9, Blocks.double_stone_slab.func_176223_P(), Blocks.double_stone_slab.func_176223_P(), false);
-                this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.SOUTH), 6, 5, 6, p_74875_3_);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 1, 9, 7, 1, 9, Blocks.stone_slab.getBlockState(), Blocks.stone_slab.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 2, 7, 7, 2, 7, Blocks.stone_slab.getBlockState(), Blocks.stone_slab.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 7, 4, 5, 9, Blocks.stone_slab.getBlockState(), Blocks.stone_slab.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 8, 5, 7, 8, 5, 9, Blocks.stone_slab.getBlockState(), Blocks.stone_slab.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 5, 7, 7, 5, 9, Blocks.double_stone_slab.getBlockState(), Blocks.double_stone_slab.getBlockState(), false);
+                this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.SOUTH), 6, 5, 6, p_74875_3_);
                 return true;
             }
         }
@@ -587,11 +587,11 @@
 
                 if (enumfacing != EnumFacing.NORTH && enumfacing != EnumFacing.EAST)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 3, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 3, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
                 else
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 3, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 3, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 return true;
@@ -662,7 +662,7 @@
 
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 0, 0, 0, 13, i - 1, 14, true, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, this.field_143013_d, 4, 1, 0);
-                this.func_189914_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.07F, 2, 1, 1, 11, 4, 13, Blocks.web.func_176223_P(), Blocks.web.func_176223_P(), false, 0);
+                this.func_189914_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.07F, 2, 1, 1, 11, 4, 13, Blocks.web.getBlockState(), Blocks.web.getBlockState(), false, 0);
                 int j = 1;
                 int k = 12;
 
@@ -670,54 +670,54 @@
                 {
                     if ((l - 1) % 4 == 0)
                     {
-                        this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, l, 1, 4, l, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
-                        this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, l, 12, 4, l, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
-                        this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.EAST), 2, 3, l, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.WEST), 11, 3, l, p_74875_3_);
+                        this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, l, 1, 4, l, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
+                        this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, l, 12, 4, l, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
+                        this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.EAST), 2, 3, l, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.WEST), 11, 3, l, p_74875_3_);
 
                         if (this.isLargeRoom)
                         {
-                            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 6, l, 1, 9, l, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
-                            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 6, l, 12, 9, l, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
+                            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 6, l, 1, 9, l, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
+                            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 6, l, 12, 9, l, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
                         }
                     }
                     else
                     {
-                        this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, l, 1, 4, l, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
-                        this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, l, 12, 4, l, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
+                        this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, l, 1, 4, l, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
+                        this.func_175804_a(p_74875_1_, p_74875_3_, 12, 1, l, 12, 4, l, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
 
                         if (this.isLargeRoom)
                         {
-                            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 6, l, 1, 9, l, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
-                            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 6, l, 12, 9, l, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
+                            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 6, l, 1, 9, l, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
+                            this.func_175804_a(p_74875_1_, p_74875_3_, 12, 6, l, 12, 9, l, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
                         }
                     }
                 }
 
                 for (int k1 = 3; k1 < 12; k1 += 2)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, k1, 4, 3, k1, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 6, 1, k1, 7, 3, k1, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 9, 1, k1, 10, 3, k1, Blocks.bookshelf.func_176223_P(), Blocks.bookshelf.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 3, 1, k1, 4, 3, k1, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 6, 1, k1, 7, 3, k1, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 9, 1, k1, 10, 3, k1, Blocks.bookshelf.getBlockState(), Blocks.bookshelf.getBlockState(), false);
                 }
 
                 if (this.isLargeRoom)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 1, 3, 5, 13, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 10, 5, 1, 12, 5, 13, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 1, 9, 5, 2, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 12, 9, 5, 13, Blocks.planks.func_176223_P(), Blocks.planks.func_176223_P(), false);
-                    this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 9, 5, 11, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 8, 5, 11, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 9, 5, 10, p_74875_3_);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 3, 6, 2, 3, 6, 12, Blocks.field_180407_aO.func_176223_P(), Blocks.field_180407_aO.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 10, 6, 2, 10, 6, 10, Blocks.field_180407_aO.func_176223_P(), Blocks.field_180407_aO.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 6, 2, 9, 6, 2, Blocks.field_180407_aO.func_176223_P(), Blocks.field_180407_aO.func_176223_P(), false);
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 6, 12, 8, 6, 12, Blocks.field_180407_aO.func_176223_P(), Blocks.field_180407_aO.func_176223_P(), false);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 9, 6, 11, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 8, 6, 11, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 9, 6, 10, p_74875_3_);
-                    IBlockState iblockstate1 = Blocks.ladder.func_176223_P().func_177226_a(BlockLadder.field_176382_a, EnumFacing.SOUTH);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 1, 5, 1, 3, 5, 13, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 10, 5, 1, 12, 5, 13, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 1, 9, 5, 2, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 5, 12, 9, 5, 13, Blocks.planks.getBlockState(), Blocks.planks.getBlockState(), false);
+                    this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 9, 5, 11, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 8, 5, 11, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 9, 5, 10, p_74875_3_);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 3, 6, 2, 3, 6, 12, Blocks.field_180407_aO.getBlockState(), Blocks.field_180407_aO.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 10, 6, 2, 10, 6, 10, Blocks.field_180407_aO.getBlockState(), Blocks.field_180407_aO.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 6, 2, 9, 6, 2, Blocks.field_180407_aO.getBlockState(), Blocks.field_180407_aO.getBlockState(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 6, 12, 8, 6, 12, Blocks.field_180407_aO.getBlockState(), Blocks.field_180407_aO.getBlockState(), false);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 9, 6, 11, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 8, 6, 11, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 9, 6, 10, p_74875_3_);
+                    IBlockState iblockstate1 = Blocks.ladder.getBlockState().func_177226_a(BlockLadder.field_176382_a, EnumFacing.SOUTH);
                     this.func_175811_a(p_74875_1_, iblockstate1, 10, 1, 13, p_74875_3_);
                     this.func_175811_a(p_74875_1_, iblockstate1, 10, 2, 13, p_74875_3_);
                     this.func_175811_a(p_74875_1_, iblockstate1, 10, 3, 13, p_74875_3_);
@@ -727,19 +727,19 @@
                     this.func_175811_a(p_74875_1_, iblockstate1, 10, 7, 13, p_74875_3_);
                     int i1 = 7;
                     int j1 = 7;
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 6, 9, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 7, 9, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 6, 8, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 7, 8, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 6, 7, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 7, 7, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 5, 7, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 8, 7, 7, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 6, 7, 6, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 6, 7, 8, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 7, 7, 6, p_74875_3_);
-                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.func_176223_P(), 7, 7, 8, p_74875_3_);
-                    IBlockState iblockstate = Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.UP);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 6, 9, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 7, 9, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 6, 8, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 7, 8, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 6, 7, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 7, 7, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 5, 7, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 8, 7, 7, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 6, 7, 6, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 6, 7, 8, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 7, 7, 6, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.field_180407_aO.getBlockState(), 7, 7, 8, p_74875_3_);
+                    IBlockState iblockstate = Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.UP);
                     this.func_175811_a(p_74875_1_, iblockstate, 5, 8, 7, p_74875_3_);
                     this.func_175811_a(p_74875_1_, iblockstate, 8, 8, 7, p_74875_3_);
                     this.func_175811_a(p_74875_1_, iblockstate, 6, 8, 6, p_74875_3_);
@@ -752,7 +752,7 @@
 
                 if (this.isLargeRoom)
                 {
-                    this.func_175811_a(p_74875_1_, Blocks.air.func_176223_P(), 12, 9, 1, p_74875_3_);
+                    this.func_175811_a(p_74875_1_, Blocks.air.getBlockState(), 12, 9, 1, p_74875_3_);
                     this.func_186167_a(p_74875_1_, p_74875_3_, p_74875_2_, 12, 8, 1, LootTableList.field_186426_h);
                 }
 
@@ -838,23 +838,23 @@
             this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 2, i, 14, 8, i, 14, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
             this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 1, 1, 1, 2, 1, 4, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
             this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 8, 1, 1, 9, 1, 4, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 1, 1, 3, Blocks.flowing_lava.func_176223_P(), Blocks.flowing_lava.func_176223_P(), false);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 1, 1, 9, 1, 3, Blocks.flowing_lava.func_176223_P(), Blocks.flowing_lava.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 1, 1, 1, 3, Blocks.flowing_lava.getBlockState(), Blocks.flowing_lava.getBlockState(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 9, 1, 1, 9, 1, 3, Blocks.flowing_lava.getBlockState(), Blocks.flowing_lava.getBlockState(), false);
             this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 3, 1, 8, 7, 1, 12, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
-            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 9, 6, 1, 11, Blocks.flowing_lava.func_176223_P(), Blocks.flowing_lava.func_176223_P(), false);
+            this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 9, 6, 1, 11, Blocks.flowing_lava.getBlockState(), Blocks.flowing_lava.getBlockState(), false);
 
             for (int j = 3; j < 14; j += 2)
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, j, 0, 4, j, Blocks.iron_bars.func_176223_P(), Blocks.iron_bars.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 10, 3, j, 10, 4, j, Blocks.iron_bars.func_176223_P(), Blocks.iron_bars.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 3, j, 0, 4, j, Blocks.iron_bars.getBlockState(), Blocks.iron_bars.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 10, 3, j, 10, 4, j, Blocks.iron_bars.getBlockState(), Blocks.iron_bars.getBlockState(), false);
             }
 
             for (int i1 = 2; i1 < 9; i1 += 2)
             {
-                this.func_175804_a(p_74875_1_, p_74875_3_, i1, 3, 15, i1, 4, 15, Blocks.iron_bars.func_176223_P(), Blocks.iron_bars.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, i1, 3, 15, i1, 4, 15, Blocks.iron_bars.getBlockState(), Blocks.iron_bars.getBlockState(), false);
             }
 
-            IBlockState iblockstate3 = Blocks.stone_brick_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
+            IBlockState iblockstate3 = Blocks.stone_brick_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.NORTH);
             this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 1, 5, 6, 1, 7, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
             this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 2, 6, 6, 2, 7, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
             this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 3, 7, 6, 3, 7, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
@@ -866,10 +866,10 @@
                 this.func_175811_a(p_74875_1_, iblockstate3, k, 3, 6, p_74875_3_);
             }
 
-            IBlockState iblockstate4 = Blocks.end_portal_frame.func_176223_P().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.NORTH);
-            IBlockState iblockstate = Blocks.end_portal_frame.func_176223_P().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.SOUTH);
-            IBlockState iblockstate1 = Blocks.end_portal_frame.func_176223_P().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.EAST);
-            IBlockState iblockstate2 = Blocks.end_portal_frame.func_176223_P().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.WEST);
+            IBlockState iblockstate4 = Blocks.end_portal_frame.getBlockState().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.NORTH);
+            IBlockState iblockstate = Blocks.end_portal_frame.getBlockState().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.SOUTH);
+            IBlockState iblockstate1 = Blocks.end_portal_frame.getBlockState().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.EAST);
+            IBlockState iblockstate2 = Blocks.end_portal_frame.getBlockState().func_177226_a(BlockEndPortalFrame.field_176508_a, EnumFacing.WEST);
             boolean flag = true;
             boolean[] aboolean = new boolean[12];
 
@@ -894,7 +894,7 @@
 
             if (flag)
             {
-                IBlockState iblockstate5 = Blocks.end_portal.func_176223_P();
+                IBlockState iblockstate5 = Blocks.end_portal.getBlockState();
                 this.func_175811_a(p_74875_1_, iblockstate5, 4, 3, 9, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate5, 5, 3, 9, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate5, 6, 3, 9, p_74875_3_);
@@ -914,7 +914,7 @@
                 if (p_74875_3_.func_175898_b(blockpos))
                 {
                     this.hasSpawner = true;
-                    p_74875_1_.func_180501_a(blockpos, Blocks.mob_spawner.func_176223_P(), 2);
+                    p_74875_1_.func_180501_a(blockpos, Blocks.mob_spawner.getBlockState(), 2);
                     TileEntity tileentity = p_74875_1_.func_175625_s(blockpos);
 
                     if (tileentity instanceof TileEntityMobSpawner)
@@ -963,17 +963,17 @@
             {
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 0, 0, 0, 8, 4, 10, true, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, this.field_143013_d, 1, 1, 0);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 10, 3, 3, 10, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 1, 1, 10, 3, 3, 10, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 3, 1, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 1, 3, 4, 3, 3, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 1, 7, 4, 3, 7, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 4, 1, 9, 4, 3, 9, false, p_74875_2_, StructureStrongholdPieces.strongholdStones);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 4, 4, 3, 6, Blocks.iron_bars.func_176223_P(), Blocks.iron_bars.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 1, 5, 7, 3, 5, Blocks.iron_bars.func_176223_P(), Blocks.iron_bars.func_176223_P(), false);
-                this.func_175811_a(p_74875_1_, Blocks.iron_bars.func_176223_P(), 4, 3, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.iron_bars.func_176223_P(), 4, 3, 8, p_74875_3_);
-                IBlockState iblockstate = Blocks.iron_door.func_176223_P().func_177226_a(BlockDoor.field_176520_a, EnumFacing.WEST);
-                IBlockState iblockstate1 = Blocks.iron_door.func_176223_P().func_177226_a(BlockDoor.field_176520_a, EnumFacing.WEST).func_177226_a(BlockDoor.field_176523_O, BlockDoor.EnumDoorHalf.UPPER);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 4, 4, 3, 6, Blocks.iron_bars.getBlockState(), Blocks.iron_bars.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 5, 1, 5, 7, 3, 5, Blocks.iron_bars.getBlockState(), Blocks.iron_bars.getBlockState(), false);
+                this.func_175811_a(p_74875_1_, Blocks.iron_bars.getBlockState(), 4, 3, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.iron_bars.getBlockState(), 4, 3, 8, p_74875_3_);
+                IBlockState iblockstate = Blocks.iron_door.getBlockState().func_177226_a(BlockDoor.field_176520_a, EnumFacing.WEST);
+                IBlockState iblockstate1 = Blocks.iron_door.getBlockState().func_177226_a(BlockDoor.field_176520_a, EnumFacing.WEST).func_177226_a(BlockDoor.field_176523_O, BlockDoor.EnumDoorHalf.UPPER);
                 this.func_175811_a(p_74875_1_, iblockstate, 4, 1, 2, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate1, 4, 2, 2, p_74875_3_);
                 this.func_175811_a(p_74875_1_, iblockstate, 4, 1, 8, p_74875_3_);
@@ -1013,11 +1013,11 @@
 
                 if (enumfacing != EnumFacing.NORTH && enumfacing != EnumFacing.EAST)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 3, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 1, 0, 3, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
                 else
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 3, 3, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 1, 4, 3, 3, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 return true;
@@ -1077,94 +1077,94 @@
             {
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 0, 0, 0, 10, 6, 10, true, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, this.field_143013_d, 4, 1, 0);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 10, 6, 3, 10, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 4, 0, 3, 6, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
-                this.func_175804_a(p_74875_1_, p_74875_3_, 10, 1, 4, 10, 3, 6, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 10, 6, 3, 10, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 4, 0, 3, 6, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
+                this.func_175804_a(p_74875_1_, p_74875_3_, 10, 1, 4, 10, 3, 6, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
 
                 switch (this.roomType)
                 {
                     case 0:
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 5, 1, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 5, 2, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 5, 3, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.WEST), 4, 3, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.EAST), 6, 3, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.SOUTH), 5, 3, 4, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.NORTH), 5, 3, 6, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 4, 1, 4, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 4, 1, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 4, 1, 6, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 6, 1, 4, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 6, 1, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 6, 1, 6, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 5, 1, 4, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176223_P(), 5, 1, 6, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 5, 1, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 5, 2, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 5, 3, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.WEST), 4, 3, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.EAST), 6, 3, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.SOUTH), 5, 3, 4, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.NORTH), 5, 3, 6, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 4, 1, 4, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 4, 1, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 4, 1, 6, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 6, 1, 4, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 6, 1, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 6, 1, 6, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 5, 1, 4, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stone_slab.getBlockState(), 5, 1, 6, p_74875_3_);
                         break;
 
                     case 1:
                         for (int i1 = 0; i1 < 5; ++i1)
                         {
-                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3, 1, 3 + i1, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 7, 1, 3 + i1, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3 + i1, 1, 3, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3 + i1, 1, 7, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3, 1, 3 + i1, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 7, 1, 3 + i1, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3 + i1, 1, 3, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3 + i1, 1, 7, p_74875_3_);
                         }
 
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 5, 1, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 5, 2, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 5, 3, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.flowing_water.func_176223_P(), 5, 4, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 5, 1, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 5, 2, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 5, 3, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.flowing_water.getBlockState(), 5, 4, 5, p_74875_3_);
                         break;
 
                     case 2:
                         for (int i = 1; i <= 9; ++i)
                         {
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 1, 3, i, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 9, 3, i, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 1, 3, i, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 9, 3, i, p_74875_3_);
                         }
 
                         for (int j = 1; j <= 9; ++j)
                         {
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), j, 3, 1, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), j, 3, 9, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), j, 3, 1, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), j, 3, 9, p_74875_3_);
                         }
 
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 5, 1, 4, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 5, 1, 6, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 5, 3, 4, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 5, 3, 6, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 4, 1, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 6, 1, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 4, 3, 5, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 6, 3, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 5, 1, 4, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 5, 1, 6, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 5, 3, 4, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 5, 3, 6, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 4, 1, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 6, 1, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 4, 3, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 6, 3, 5, p_74875_3_);
 
                         for (int k = 1; k <= 3; ++k)
                         {
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 4, k, 4, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 6, k, 4, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 4, k, 6, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.func_176223_P(), 6, k, 6, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 4, k, 4, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 6, k, 4, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 4, k, 6, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.cobblestone.getBlockState(), 6, k, 6, p_74875_3_);
                         }
 
-                        this.func_175811_a(p_74875_1_, Blocks.torch.func_176223_P(), 5, 3, 5, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.torch.getBlockState(), 5, 3, 5, p_74875_3_);
 
                         for (int l = 2; l <= 8; ++l)
                         {
-                            this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 2, 3, l, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 3, 3, l, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 2, 3, l, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 3, 3, l, p_74875_3_);
 
                             if (l <= 3 || l >= 7)
                             {
-                                this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 4, 3, l, p_74875_3_);
-                                this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 5, 3, l, p_74875_3_);
-                                this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 6, 3, l, p_74875_3_);
+                                this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 4, 3, l, p_74875_3_);
+                                this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 5, 3, l, p_74875_3_);
+                                this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 6, 3, l, p_74875_3_);
                             }
 
-                            this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 7, 3, l, p_74875_3_);
-                            this.func_175811_a(p_74875_1_, Blocks.planks.func_176223_P(), 8, 3, l, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 7, 3, l, p_74875_3_);
+                            this.func_175811_a(p_74875_1_, Blocks.planks.getBlockState(), 8, 3, l, p_74875_3_);
                         }
 
-                        IBlockState iblockstate = Blocks.ladder.func_176223_P().func_177226_a(BlockLadder.field_176382_a, EnumFacing.WEST);
+                        IBlockState iblockstate = Blocks.ladder.getBlockState().func_177226_a(BlockLadder.field_176382_a, EnumFacing.WEST);
                         this.func_175811_a(p_74875_1_, iblockstate, 9, 1, 3, p_74875_3_);
                         this.func_175811_a(p_74875_1_, iblockstate, 9, 2, 3, p_74875_3_);
                         this.func_175811_a(p_74875_1_, iblockstate, 9, 3, 3, p_74875_3_);
@@ -1249,22 +1249,22 @@
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 10, 4, true, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, this.field_143013_d, 1, 7, 0);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, StructureStrongholdPieces.Stronghold.Door.OPENING, 1, 1, 4);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 2, 6, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 5, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 2, 6, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 5, 1, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.STONE.func_176624_a()), 1, 6, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 5, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 4, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 5, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 4, 3, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.STONE.func_176624_a()), 1, 5, 3, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 2, 4, 3, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3, 3, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 2, 4, 3, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3, 3, 3, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.STONE.func_176624_a()), 3, 4, 3, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3, 3, 2, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3, 2, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3, 3, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3, 2, 1, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.STONE.func_176624_a()), 3, 3, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 2, 2, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 1, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 2, 2, 1, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 1, 1, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.STONE.func_176624_a()), 1, 2, 1, p_74875_3_);
-                this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 1, 2, p_74875_3_);
+                this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 1, 2, p_74875_3_);
                 this.func_175811_a(p_74875_1_, Blocks.stone_slab.func_176203_a(BlockStoneSlab.EnumType.STONE.func_176624_a()), 1, 1, 3, p_74875_3_);
                 return true;
             }
@@ -1328,7 +1328,7 @@
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 10, 7, true, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, this.field_143013_d, 1, 7, 0);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, StructureStrongholdPieces.Stronghold.Door.OPENING, 1, 1, 7);
-                IBlockState iblockstate = Blocks.stone_stairs.func_176223_P().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
+                IBlockState iblockstate = Blocks.stone_stairs.getBlockState().func_177226_a(BlockStairs.field_176309_a, EnumFacing.SOUTH);
 
                 for (int i = 0; i < 6; ++i)
                 {
@@ -1338,9 +1338,9 @@
 
                     if (i < 5)
                     {
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 1, 5 - i, 1 + i, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 2, 5 - i, 1 + i, p_74875_3_);
-                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.func_176223_P(), 3, 5 - i, 1 + i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 1, 5 - i, 1 + i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 2, 5 - i, 1 + i, p_74875_3_);
+                        this.func_175811_a(p_74875_1_, Blocks.stonebrick.getBlockState(), 3, 5 - i, 1 + i, p_74875_3_);
                     }
                 }
 
@@ -1375,12 +1375,12 @@
                 }
                 else
                 {
-                    this.field_151562_a = Blocks.stonebrick.func_176223_P();
+                    this.field_151562_a = Blocks.stonebrick.getBlockState();
                 }
             }
             else
             {
-                this.field_151562_a = Blocks.air.func_176223_P();
+                this.field_151562_a = Blocks.air.getBlockState();
             }
         }
     }
@@ -1450,8 +1450,8 @@
                 this.fillWithRandomizedBlocks(p_74875_1_, p_74875_3_, 0, 0, 0, 4, 4, 6, true, p_74875_2_, StructureStrongholdPieces.strongholdStones);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, this.field_143013_d, 1, 1, 0);
                 this.placeDoor(p_74875_1_, p_74875_2_, p_74875_3_, StructureStrongholdPieces.Stronghold.Door.OPENING, 1, 1, 6);
-                IBlockState iblockstate = Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.EAST);
-                IBlockState iblockstate1 = Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, EnumFacing.WEST);
+                IBlockState iblockstate = Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.EAST);
+                IBlockState iblockstate1 = Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, EnumFacing.WEST);
                 this.func_175809_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.1F, 1, 2, 1, iblockstate);
                 this.func_175809_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.1F, 3, 2, 1, iblockstate1);
                 this.func_175809_a(p_74875_1_, p_74875_3_, p_74875_2_, 0.1F, 1, 2, 5, iblockstate);
@@ -1459,12 +1459,12 @@
 
                 if (this.expandsX)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 2, 0, 3, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 0, 1, 2, 0, 3, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 if (this.expandsZ)
                 {
-                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 2, 4, 3, 4, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74875_1_, p_74875_3_, 4, 1, 2, 4, 3, 4, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                 }
 
                 return true;
@@ -1500,45 +1500,45 @@
             switch (p_74990_4_)
             {
                 case OPENING:
-                    this.func_175804_a(p_74990_1_, p_74990_3_, p_74990_5_, p_74990_6_, p_74990_7_, p_74990_5_ + 3 - 1, p_74990_6_ + 3 - 1, p_74990_7_, Blocks.air.func_176223_P(), Blocks.air.func_176223_P(), false);
+                    this.func_175804_a(p_74990_1_, p_74990_3_, p_74990_5_, p_74990_6_, p_74990_7_, p_74990_5_ + 3 - 1, p_74990_6_ + 3 - 1, p_74990_7_, Blocks.air.getBlockState(), Blocks.air.getBlockState(), false);
                     break;
 
                 case WOOD_DOOR:
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 1, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 2, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 2, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.field_180413_ao.func_176223_P(), p_74990_5_ + 1, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.field_180413_ao.func_176223_P().func_177226_a(BlockDoor.field_176523_O, BlockDoor.EnumDoorHalf.UPPER), p_74990_5_ + 1, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 1, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 2, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 2, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.field_180413_ao.getBlockState(), p_74990_5_ + 1, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.field_180413_ao.getBlockState().func_177226_a(BlockDoor.field_176523_O, BlockDoor.EnumDoorHalf.UPPER), p_74990_5_ + 1, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
                     break;
 
                 case GRATES:
-                    this.func_175811_a(p_74990_1_, Blocks.air.func_176223_P(), p_74990_5_ + 1, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.air.func_176223_P(), p_74990_5_ + 1, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.func_176223_P(), p_74990_5_, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.func_176223_P(), p_74990_5_, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.func_176223_P(), p_74990_5_, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.func_176223_P(), p_74990_5_ + 1, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.func_176223_P(), p_74990_5_ + 2, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.func_176223_P(), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.func_176223_P(), p_74990_5_ + 2, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.air.getBlockState(), p_74990_5_ + 1, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.air.getBlockState(), p_74990_5_ + 1, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.getBlockState(), p_74990_5_, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.getBlockState(), p_74990_5_, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.getBlockState(), p_74990_5_, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.getBlockState(), p_74990_5_ + 1, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.getBlockState(), p_74990_5_ + 2, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.getBlockState(), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_bars.getBlockState(), p_74990_5_ + 2, p_74990_6_, p_74990_7_, p_74990_3_);
                     break;
 
                 case IRON_DOOR:
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 1, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 2, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.func_176223_P(), p_74990_5_ + 2, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_door.func_176223_P(), p_74990_5_ + 1, p_74990_6_, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.iron_door.func_176223_P().func_177226_a(BlockDoor.field_176523_O, BlockDoor.EnumDoorHalf.UPPER), p_74990_5_ + 1, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stone_button.func_176223_P().func_177226_a(BlockButton.field_176387_N, EnumFacing.NORTH), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_ + 1, p_74990_3_);
-                    this.func_175811_a(p_74990_1_, Blocks.stone_button.func_176223_P().func_177226_a(BlockButton.field_176387_N, EnumFacing.SOUTH), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_ - 1, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 1, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 2, p_74990_6_ + 2, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stonebrick.getBlockState(), p_74990_5_ + 2, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_door.getBlockState(), p_74990_5_ + 1, p_74990_6_, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.iron_door.getBlockState().func_177226_a(BlockDoor.field_176523_O, BlockDoor.EnumDoorHalf.UPPER), p_74990_5_ + 1, p_74990_6_ + 1, p_74990_7_, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stone_button.getBlockState().func_177226_a(BlockButton.field_176387_N, EnumFacing.NORTH), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_ + 1, p_74990_3_);
+                    this.func_175811_a(p_74990_1_, Blocks.stone_button.getBlockState().func_177226_a(BlockButton.field_176387_N, EnumFacing.SOUTH), p_74990_5_ + 2, p_74990_6_ + 1, p_74990_7_ - 1, p_74990_3_);
             }
         }
 
