--- a/net/minecraft/world/gen/structure/StructureMineshaftPieces.java
+++ b/net/minecraft/world/gen/structure/StructureMineshaftPieces.java
@@ -716,7 +716,7 @@
 
             int k;
 
-            for (lvt_5_1_ = 0; k < this.boundingBox.getXSize(); k = k + 4)
+            for (k = 0; k < this.boundingBox.getXSize(); k = k + 4)
             {
                 k = k + rand.nextInt(this.boundingBox.getXSize());
 
