--- a/net/minecraft/world/gen/structure/StructureMineshaftStart.java
+++ b/net/minecraft/world/gen/structure/StructureMineshaftStart.java
@@ -23,7 +23,7 @@
         if (p_i47149_5_ == MapGenMineshaft.Type.MESA)
         {
             int i = -5;
-            int j = p_i47149_1_.func_181545_F() - this.boundingBox.maxY + this.boundingBox.getYSize() / 2 - -5;
+            int j = p_i47149_1_.getSpawnLevel() - this.boundingBox.maxY + this.boundingBox.getYSize() / 2 - -5;
             this.boundingBox.offset(0, j, 0);
 
             for (StructureComponent structurecomponent : this.components)
