--- a/net/minecraft/entity/projectile/EntityPotion.java
+++ b/net/minecraft/entity/projectile/EntityPotion.java
@@ -106,12 +106,12 @@
 
             if (p_70184_1_.typeOfHit == RayTraceResult.Type.BLOCK && potiontype == PotionTypes.field_185230_b && list.isEmpty())
             {
-                BlockPos blockpos = p_70184_1_.func_178782_a().func_177972_a(p_70184_1_.field_178784_b);
+                BlockPos blockpos = p_70184_1_.getBlock().getBlockPosRelative(p_70184_1_.sideHit);
                 this.func_184542_a(blockpos);
 
                 for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
                 {
-                    this.func_184542_a(blockpos.func_177972_a(enumfacing));
+                    this.func_184542_a(blockpos.getBlockPosRelative(enumfacing));
                 }
 
                 this.worldObj.func_175718_b(2002, new BlockPos(this), PotionType.func_185171_a(potiontype));
@@ -198,9 +198,9 @@
 
     private void func_184542_a(BlockPos p_184542_1_)
     {
-        if (this.worldObj.func_180495_p(p_184542_1_).func_177230_c() == Blocks.fire)
+        if (this.worldObj.getBlockState(p_184542_1_).getBlock() == Blocks.fire)
         {
-            this.worldObj.func_180501_a(p_184542_1_, Blocks.air.func_176223_P(), 2);
+            this.worldObj.func_180501_a(p_184542_1_, Blocks.air.getBlockState(), 2);
         }
     }
 
