--- a/net/minecraft/entity/projectile/EntityWitherSkull.java
+++ b/net/minecraft/entity/projectile/EntityWitherSkull.java
@@ -63,7 +63,7 @@
     public float func_180428_a(Explosion p_180428_1_, World p_180428_2_, BlockPos p_180428_3_, IBlockState p_180428_4_)
     {
         float f = super.func_180428_a(p_180428_1_, p_180428_2_, p_180428_3_, p_180428_4_);
-        Block block = p_180428_4_.func_177230_c();
+        Block block = p_180428_4_.getBlock();
 
         if (this.isInvulnerable() && EntityWither.func_181033_a(block))
         {
