--- a/net/minecraft/entity/projectile/EntityTippedArrow.java
+++ b/net/minecraft/entity/projectile/EntityTippedArrow.java
@@ -43,7 +43,7 @@
 
     public void func_184555_a(ItemStack p_184555_1_)
     {
-        if (p_184555_1_.getItem() == Items.field_185167_i)
+        if (p_184555_1_.getItem() == Items.tippedArrow)
         {
             this.field_184560_g = PotionUtils.func_185187_c(p_184555_1_.getTagCompound());
             Collection<PotionEffect> collection = PotionUtils.func_185190_b(p_184555_1_);
@@ -120,7 +120,7 @@
 
             for (int j = 0; j < p_184556_1_; ++j)
             {
-                this.worldObj.func_175688_a(EnumParticleTypes.SPELL_MOB, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, d0, d1, d2, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.SPELL_MOB, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, d0, d1, d2, new int[0]);
             }
         }
     }
@@ -201,7 +201,7 @@
         }
     }
 
-    protected ItemStack func_184550_j()
+    protected ItemStack getItemStack()
     {
         if (this.field_184561_h.isEmpty() && this.field_184560_g == PotionTypes.field_185229_a)
         {
@@ -209,7 +209,7 @@
         }
         else
         {
-            ItemStack itemstack = new ItemStack(Items.field_185167_i);
+            ItemStack itemstack = new ItemStack(Items.tippedArrow);
             PotionUtils.func_185188_a(itemstack, this.field_184560_g);
             PotionUtils.func_185184_a(itemstack, this.field_184561_h);
             return itemstack;
@@ -230,7 +230,7 @@
 
                 for (int j = 0; j < 20; ++j)
                 {
-                    this.worldObj.func_175688_a(EnumParticleTypes.SPELL_MOB, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, d0, d1, d2, new int[0]);
+                    this.worldObj.spawnParticle(EnumParticleTypes.SPELL_MOB, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, d0, d1, d2, new int[0]);
                 }
             }
         }
