--- a/net/minecraft/entity/projectile/EntityShulkerBullet.java
+++ b/net/minecraft/entity/projectile/EntityShulkerBullet.java
@@ -67,9 +67,9 @@
         this(p_i46772_1_);
         this.field_184570_a = p_i46772_2_;
         BlockPos blockpos = new BlockPos(p_i46772_2_);
-        double d0 = (double)blockpos.func_177958_n() + 0.5D;
-        double d1 = (double)blockpos.func_177956_o() + 0.5D;
-        double d2 = (double)blockpos.func_177952_p() + 0.5D;
+        double d0 = (double)blockpos.getX() + 0.5D;
+        double d1 = (double)blockpos.getY() + 0.5D;
+        double d2 = (double)blockpos.getZ() + 0.5D;
         this.setLocationAndAngles(d0, d1, d2, this.rotationYaw, this.rotationPitch);
         this.field_184571_b = p_i46772_3_;
         this.field_184573_c = EnumFacing.UP;
@@ -85,9 +85,9 @@
         {
             BlockPos blockpos = new BlockPos(this.field_184570_a);
             NBTTagCompound nbttagcompound = NBTUtil.func_186862_a(this.field_184570_a.getUniqueID());
-            nbttagcompound.setInteger("X", blockpos.func_177958_n());
-            nbttagcompound.setInteger("Y", blockpos.func_177956_o());
-            nbttagcompound.setInteger("Z", blockpos.func_177952_p());
+            nbttagcompound.setInteger("X", blockpos.getX());
+            nbttagcompound.setInteger("Y", blockpos.getY());
+            nbttagcompound.setInteger("Z", blockpos.getZ());
             tagCompound.setTag("Owner", nbttagcompound);
         }
 
@@ -95,9 +95,9 @@
         {
             BlockPos blockpos1 = new BlockPos(this.field_184571_b);
             NBTTagCompound nbttagcompound1 = NBTUtil.func_186862_a(this.field_184571_b.getUniqueID());
-            nbttagcompound1.setInteger("X", blockpos1.func_177958_n());
-            nbttagcompound1.setInteger("Y", blockpos1.func_177956_o());
-            nbttagcompound1.setInteger("Z", blockpos1.func_177952_p());
+            nbttagcompound1.setInteger("X", blockpos1.getX());
+            nbttagcompound1.setInteger("Y", blockpos1.getY());
+            nbttagcompound1.setInteger("Z", blockpos1.getZ());
             tagCompound.setTag("Target", nbttagcompound1);
         }
 
@@ -158,7 +158,7 @@
 
         if (this.field_184571_b == null)
         {
-            blockpos = (new BlockPos(this)).func_177977_b();
+            blockpos = (new BlockPos(this)).getBlockPosBelow();
         }
         else
         {
@@ -166,9 +166,9 @@
             blockpos = new BlockPos(this.field_184571_b.posX, this.field_184571_b.posY + d0, this.field_184571_b.posZ);
         }
 
-        double d1 = (double)blockpos.func_177958_n() + 0.5D;
-        double d2 = (double)blockpos.func_177956_o() + d0;
-        double d3 = (double)blockpos.func_177952_p() + 0.5D;
+        double d1 = (double)blockpos.getX() + 0.5D;
+        double d2 = (double)blockpos.getY() + d0;
+        double d3 = (double)blockpos.getZ() + 0.5D;
         EnumFacing enumfacing = null;
 
         if (blockpos.func_177957_d(this.posX, this.posY, this.posZ) >= 4.0D)
@@ -178,11 +178,11 @@
 
             if (p_184569_1_ != EnumFacing.Axis.X)
             {
-                if (blockpos1.func_177958_n() < blockpos.func_177958_n() && this.worldObj.func_175623_d(blockpos1.func_177974_f()))
+                if (blockpos1.getX() < blockpos.getX() && this.worldObj.isEmpty(blockpos1.getBlockPosEast()))
                 {
                     list.add(EnumFacing.EAST);
                 }
-                else if (blockpos1.func_177958_n() > blockpos.func_177958_n() && this.worldObj.func_175623_d(blockpos1.func_177976_e()))
+                else if (blockpos1.getX() > blockpos.getX() && this.worldObj.isEmpty(blockpos1.getBlockPosWest()))
                 {
                     list.add(EnumFacing.WEST);
                 }
@@ -190,11 +190,11 @@
 
             if (p_184569_1_ != EnumFacing.Axis.Y)
             {
-                if (blockpos1.func_177956_o() < blockpos.func_177956_o() && this.worldObj.func_175623_d(blockpos1.func_177984_a()))
+                if (blockpos1.getY() < blockpos.getY() && this.worldObj.isEmpty(blockpos1.getBlockPosAbove()))
                 {
                     list.add(EnumFacing.UP);
                 }
-                else if (blockpos1.func_177956_o() > blockpos.func_177956_o() && this.worldObj.func_175623_d(blockpos1.func_177977_b()))
+                else if (blockpos1.getY() > blockpos.getY() && this.worldObj.isEmpty(blockpos1.getBlockPosBelow()))
                 {
                     list.add(EnumFacing.DOWN);
                 }
@@ -202,11 +202,11 @@
 
             if (p_184569_1_ != EnumFacing.Axis.Z)
             {
-                if (blockpos1.func_177952_p() < blockpos.func_177952_p() && this.worldObj.func_175623_d(blockpos1.func_177968_d()))
+                if (blockpos1.getZ() < blockpos.getZ() && this.worldObj.isEmpty(blockpos1.getBlockPosSouth()))
                 {
                     list.add(EnumFacing.SOUTH);
                 }
-                else if (blockpos1.func_177952_p() > blockpos.func_177952_p() && this.worldObj.func_175623_d(blockpos1.func_177978_c()))
+                else if (blockpos1.getZ() > blockpos.getZ() && this.worldObj.isEmpty(blockpos1.getBlockPosNorth()))
                 {
                     list.add(EnumFacing.NORTH);
                 }
@@ -216,7 +216,7 @@
 
             if (list.isEmpty())
             {
-                for (int i = 5; !this.worldObj.func_175623_d(blockpos1.func_177972_a(enumfacing)) && i > 0; --i)
+                for (int i = 5; !this.worldObj.isEmpty(blockpos1.getBlockPosRelative(enumfacing)) && i > 0; --i)
                 {
                     enumfacing = EnumFacing.func_176741_a(this.rand);
                 }
@@ -327,7 +327,7 @@
 
             if (this.worldObj.isRemote)
             {
-                this.worldObj.func_175688_a(EnumParticleTypes.END_ROD, this.posX - this.motionX, this.posY - this.motionY + 0.15D, this.posZ - this.motionZ, 0.0D, 0.0D, 0.0D, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.END_ROD, this.posX - this.motionX, this.posY - this.motionY + 0.15D, this.posZ - this.motionZ, 0.0D, 0.0D, 0.0D, new int[0]);
             }
             else if (this.field_184571_b != null && !this.field_184571_b.isDead)
             {
@@ -346,7 +346,7 @@
                     BlockPos blockpos = new BlockPos(this);
                     EnumFacing.Axis enumfacing$axis = this.field_184573_c.func_176740_k();
 
-                    if (this.worldObj.func_175677_d(blockpos.func_177972_a(this.field_184573_c), false))
+                    if (this.worldObj.func_175677_d(blockpos.getBlockPosRelative(this.field_184573_c), false))
                     {
                         this.func_184569_a(enumfacing$axis);
                     }
@@ -354,7 +354,7 @@
                     {
                         BlockPos blockpos1 = new BlockPos(this.field_184571_b);
 
-                        if (enumfacing$axis == EnumFacing.Axis.X && blockpos.func_177958_n() == blockpos1.func_177958_n() || enumfacing$axis == EnumFacing.Axis.Z && blockpos.func_177952_p() == blockpos1.func_177952_p() || enumfacing$axis == EnumFacing.Axis.Y && blockpos.func_177956_o() == blockpos1.func_177956_o())
+                        if (enumfacing$axis == EnumFacing.Axis.X && blockpos.getX() == blockpos1.getX() || enumfacing$axis == EnumFacing.Axis.Z && blockpos.getZ() == blockpos1.getZ() || enumfacing$axis == EnumFacing.Axis.Y && blockpos.getY() == blockpos1.getY())
                         {
                             this.func_184569_a(enumfacing$axis);
                         }
@@ -399,7 +399,7 @@
         if (p_184567_1_.entityHit == null)
         {
             ((WorldServer)this.worldObj).func_175739_a(EnumParticleTypes.EXPLOSION_LARGE, this.posX, this.posY, this.posZ, 2, 0.2D, 0.2D, 0.2D, 0.0D, new int[0]);
-            this.func_184185_a(SoundEvents.field_187775_eP, 1.0F, 1.0F);
+            this.playSound(SoundEvents.field_187775_eP, 1.0F, 1.0F);
         }
         else
         {
@@ -434,7 +434,7 @@
     {
         if (!this.worldObj.isRemote)
         {
-            this.func_184185_a(SoundEvents.field_187777_eQ, 1.0F, 1.0F);
+            this.playSound(SoundEvents.field_187777_eQ, 1.0F, 1.0F);
             ((WorldServer)this.worldObj).func_175739_a(EnumParticleTypes.CRIT, this.posX, this.posY, this.posZ, 15, 0.2D, 0.2D, 0.2D, 0.0D, new int[0]);
             this.setDead();
         }
