--- a/net/minecraft/entity/projectile/EntityArrow.java
+++ b/net/minecraft/entity/projectile/EntityArrow.java
@@ -202,14 +202,14 @@
         }
 
         BlockPos blockpos = new BlockPos(this.field_145791_d, this.field_145792_e, this.field_145789_f);
-        IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
-        Block block = iblockstate.func_177230_c();
+        IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
+        Block block = iblockstate.getBlock();
 
-        if (iblockstate.func_185904_a() != Material.air)
+        if (iblockstate.getMaterial() != Material.air)
         {
             AxisAlignedBB axisalignedbb = iblockstate.func_185890_d(this.worldObj, blockpos);
 
-            if (axisalignedbb != Block.field_185506_k && axisalignedbb.func_186670_a(blockpos).isVecInside(new Vec3d(this.posX, this.posY, this.posZ)))
+            if (axisalignedbb != Block.field_185506_k && axisalignedbb.offset(blockpos).isVecInside(new Vec3d(this.posX, this.posY, this.posZ)))
             {
                 this.inGround = true;
             }
@@ -286,7 +286,7 @@
             {
                 for (int k = 0; k < 4; ++k)
                 {
-                    this.worldObj.func_175688_a(EnumParticleTypes.CRIT, this.posX + this.motionX * (double)k / 4.0D, this.posY + this.motionY * (double)k / 4.0D, this.posZ + this.motionZ * (double)k / 4.0D, -this.motionX, -this.motionY + 0.2D, -this.motionZ, new int[0]);
+                    this.worldObj.spawnParticle(EnumParticleTypes.CRIT, this.posX + this.motionX * (double)k / 4.0D, this.posY + this.motionY * (double)k / 4.0D, this.posZ + this.motionZ * (double)k / 4.0D, -this.motionX, -this.motionY + 0.2D, -this.motionZ, new int[0]);
                 }
             }
 
@@ -326,7 +326,7 @@
                 for (int i = 0; i < 4; ++i)
                 {
                     float f3 = 0.25F;
-                    this.worldObj.func_175688_a(EnumParticleTypes.WATER_BUBBLE, this.posX - this.motionX * 0.25D, this.posY - this.motionY * 0.25D, this.posZ - this.motionZ * 0.25D, this.motionX, this.motionY, this.motionZ, new int[0]);
+                    this.worldObj.spawnParticle(EnumParticleTypes.WATER_BUBBLE, this.posX - this.motionX * 0.25D, this.posY - this.motionY * 0.25D, this.posZ - this.motionZ * 0.25D, this.motionX, this.motionY, this.motionZ, new int[0]);
                 }
 
                 f1 = 0.6F;
@@ -416,7 +416,7 @@
                     }
                 }
 
-                this.func_184185_a(SoundEvents.field_187731_t, 1.0F, 1.2F / (this.rand.nextFloat() * 0.2F + 0.9F));
+                this.playSound(SoundEvents.field_187731_t, 1.0F, 1.2F / (this.rand.nextFloat() * 0.2F + 0.9F));
 
                 if (!(entity instanceof EntityEnderman))
                 {
@@ -436,7 +436,7 @@
                 {
                     if (this.canBePickedUp == EntityArrow.PickupStatus.ALLOWED)
                     {
-                        this.entityDropItem(this.func_184550_j(), 0.1F);
+                        this.entityDropItem(this.getItemStack(), 0.1F);
                     }
 
                     this.setDead();
@@ -445,12 +445,12 @@
         }
         else
         {
-            BlockPos blockpos = p_184549_1_.func_178782_a();
-            this.field_145791_d = blockpos.func_177958_n();
-            this.field_145792_e = blockpos.func_177956_o();
-            this.field_145789_f = blockpos.func_177952_p();
-            IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
-            this.field_145790_g = iblockstate.func_177230_c();
+            BlockPos blockpos = p_184549_1_.getBlock();
+            this.field_145791_d = blockpos.getX();
+            this.field_145792_e = blockpos.getY();
+            this.field_145789_f = blockpos.getZ();
+            IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
+            this.field_145790_g = iblockstate.getBlock();
             this.inData = this.field_145790_g.func_176201_c(iblockstate);
             this.motionX = (double)((float)(p_184549_1_.hitVec.xCoord - this.posX));
             this.motionY = (double)((float)(p_184549_1_.hitVec.yCoord - this.posY));
@@ -459,12 +459,12 @@
             this.posX -= this.motionX / (double)f2 * 0.05000000074505806D;
             this.posY -= this.motionY / (double)f2 * 0.05000000074505806D;
             this.posZ -= this.motionZ / (double)f2 * 0.05000000074505806D;
-            this.func_184185_a(SoundEvents.field_187731_t, 1.0F, 1.2F / (this.rand.nextFloat() * 0.2F + 0.9F));
+            this.playSound(SoundEvents.field_187731_t, 1.0F, 1.2F / (this.rand.nextFloat() * 0.2F + 0.9F));
             this.inGround = true;
             this.arrowShake = 7;
             this.setIsCritical(false);
 
-            if (iblockstate.func_185904_a() != Material.air)
+            if (iblockstate.getMaterial() != Material.air)
             {
                 this.field_145790_g.func_180634_a(this.worldObj, blockpos, iblockstate, this);
             }
@@ -479,7 +479,7 @@
     protected Entity func_184551_a(Vec3d p_184551_1_, Vec3d p_184551_2_)
     {
         Entity entity = null;
-        List<Entity> list = this.worldObj.func_175674_a(this, this.func_174813_aQ().addCoord(this.motionX, this.motionY, this.motionZ).func_186662_g(1.0D), field_184553_f);
+        List<Entity> list = this.worldObj.func_175674_a(this, this.func_174813_aQ().addCoord(this.motionX, this.motionY, this.motionZ).expand(1.0D), field_184553_f);
         double d0 = 0.0D;
 
         for (int i = 0; i < list.size(); ++i)
@@ -488,7 +488,7 @@
 
             if (entity1 != this.shootingEntity || this.ticksInAir >= 5)
             {
-                AxisAlignedBB axisalignedbb = entity1.func_174813_aQ().func_186662_g(0.30000001192092896D);
+                AxisAlignedBB axisalignedbb = entity1.func_174813_aQ().expand(0.30000001192092896D);
                 RayTraceResult raytraceresult = axisalignedbb.calculateIntercept(p_184551_1_, p_184551_2_);
 
                 if (raytraceresult != null)
@@ -581,21 +581,21 @@
         {
             boolean flag = this.canBePickedUp == EntityArrow.PickupStatus.ALLOWED || this.canBePickedUp == EntityArrow.PickupStatus.CREATIVE_ONLY && entityIn.capabilities.isCreativeMode;
 
-            if (this.canBePickedUp == EntityArrow.PickupStatus.ALLOWED && !entityIn.inventory.addItemStackToInventory(this.func_184550_j()))
+            if (this.canBePickedUp == EntityArrow.PickupStatus.ALLOWED && !entityIn.inventory.addItemStackToInventory(this.getItemStack()))
             {
                 flag = false;
             }
 
             if (flag)
             {
-                this.func_184185_a(SoundEvents.field_187638_cR, 0.2F, ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.7F + 1.0F) * 2.0F);
+                this.playSound(SoundEvents.field_187638_cR, 0.2F, ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.7F + 1.0F) * 2.0F);
                 entityIn.onItemPickup(this, 1);
                 this.setDead();
             }
         }
     }
 
-    protected abstract ItemStack func_184550_j();
+    protected abstract ItemStack getItemStack();
 
     /**
      * returns if this entity triggers Block.onEntityWalking on the blocks they walk on. used for spiders and wolves to
