--- a/net/minecraft/entity/projectile/EntityArrow.java
+++ b/net/minecraft/entity/projectile/EntityArrow.java
@@ -7,6 +7,7 @@
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
@@ -91,6 +92,12 @@
 
         if (shooter instanceof EntityPlayer)
         {
+        	//VIVE SET SPAWNPOS
+            Vec3d aimSource = Minecraft.getMinecraft().roomScale.getControllerPos_World(1);
+            Vec3d controllerDirection = Minecraft.getMinecraft().bowTracker.getAimVector();
+            this.setPosition(aimSource.xCoord + controllerDirection.xCoord * 0.1f,aimSource.yCoord + controllerDirection.yCoord * 0.1f,aimSource.zCoord + controllerDirection.zCoord * 0.1f);
+            ///
+            
             this.pickupStatus = EntityArrow.PickupStatus.ALLOWED;
         }
     }
