--- a/net/minecraft/entity/projectile/EntitySpectralArrow.java
+++ b/net/minecraft/entity/projectile/EntitySpectralArrow.java
@@ -38,13 +38,13 @@
 
         if (this.worldObj.isRemote && !this.inGround)
         {
-            this.worldObj.func_175688_a(EnumParticleTypes.SPELL_INSTANT, this.posX, this.posY, this.posZ, 0.0D, 0.0D, 0.0D, new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.SPELL_INSTANT, this.posX, this.posY, this.posZ, 0.0D, 0.0D, 0.0D, new int[0]);
         }
     }
 
-    protected ItemStack func_184550_j()
+    protected ItemStack getItemStack()
     {
-        return new ItemStack(Items.field_185166_h);
+        return new ItemStack(Items.spectralArrow);
     }
 
     protected void func_184548_a(EntityLivingBase p_184548_1_)
