--- a/net/minecraft/entity/projectile/EntityFireball.java
+++ b/net/minecraft/entity/projectile/EntityFireball.java
@@ -103,7 +103,7 @@
 
             if (this.inGround)
             {
-                if (this.worldObj.func_180495_p(new BlockPos(this.field_145795_e, this.field_145793_f, this.field_145794_g)).func_177230_c() == this.field_145796_h)
+                if (this.worldObj.getBlockState(new BlockPos(this.field_145795_e, this.field_145793_f, this.field_145794_g)).getBlock() == this.field_145796_h)
                 {
                     ++this.ticksAlive;
 
@@ -145,7 +145,7 @@
                 for (int i = 0; i < 4; ++i)
                 {
                     float f1 = 0.25F;
-                    this.worldObj.func_175688_a(EnumParticleTypes.WATER_BUBBLE, this.posX - this.motionX * 0.25D, this.posY - this.motionY * 0.25D, this.posZ - this.motionZ * 0.25D, this.motionX, this.motionY, this.motionZ, new int[0]);
+                    this.worldObj.spawnParticle(EnumParticleTypes.WATER_BUBBLE, this.posX - this.motionX * 0.25D, this.posY - this.motionY * 0.25D, this.posZ - this.motionZ * 0.25D, this.motionX, this.motionY, this.motionZ, new int[0]);
                 }
 
                 f = 0.8F;
@@ -157,7 +157,7 @@
             this.motionX *= (double)f;
             this.motionY *= (double)f;
             this.motionZ *= (double)f;
-            this.worldObj.func_175688_a(this.func_184563_j(), this.posX, this.posY + 0.5D, this.posZ, 0.0D, 0.0D, 0.0D, new int[0]);
+            this.worldObj.spawnParticle(this.func_184563_j(), this.posX, this.posY + 0.5D, this.posZ, 0.0D, 0.0D, 0.0D, new int[0]);
             this.setPosition(this.posX, this.posY, this.posZ);
         }
         else
