--- a/net/minecraft/world/chunk/EmptyChunk.java
+++ b/net/minecraft/world/chunk/EmptyChunk.java
@@ -50,9 +50,9 @@
     {
     }
 
-    public IBlockState func_177435_g(BlockPos p_177435_1_)
+    public IBlockState getBlockState(BlockPos p_177435_1_)
     {
-        return Blocks.air.func_176223_P();
+        return Blocks.air.getBlockState();
     }
 
     public int func_177437_b(BlockPos p_177437_1_)
