--- a/net/minecraft/crash/CrashReportCategory.java
+++ b/net/minecraft/crash/CrashReportCategory.java
@@ -26,7 +26,7 @@
 
     public static String func_180522_a(BlockPos p_180522_0_)
     {
-        return func_184876_a(p_180522_0_.func_177958_n(), p_180522_0_.func_177956_o(), p_180522_0_.func_177952_p());
+        return func_184876_a(p_180522_0_.getX(), p_180522_0_.getY(), p_180522_0_.getZ());
     }
 
     public static String func_184876_a(int p_184876_0_, int p_184876_1_, int p_184876_2_)
