--- a/net/minecraft/crash/CrashReport.java
+++ b/net/minecraft/crash/CrashReport.java
@@ -245,8 +245,10 @@
 
         StringBuilder stringbuilder = new StringBuilder();
         stringbuilder.append("---- Minecraft Crash Report ----\n");
-        Reflector.call(Reflector.BlamingTransformer_onCrash, new Object[] {stringbuilder});
-        Reflector.call(Reflector.CoreModManager_onCrash, new Object[] {stringbuilder});
+        if(Reflector.BlamingTransformer_onCrash.exists())
+        	Reflector.call(Reflector.BlamingTransformer_onCrash, new Object[] {stringbuilder});
+        if(Reflector.CoreModManager_onCrash.exists())
+        	Reflector.call(Reflector.CoreModManager_onCrash, new Object[] {stringbuilder});
         stringbuilder.append("// ");
         stringbuilder.append(getWittyComment());
         stringbuilder.append("\n\n");
