--- a/net/minecraft/entity/projectile/EntityArrow.java
+++ b/net/minecraft/entity/projectile/EntityArrow.java
@@ -7,6 +7,7 @@
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
@@ -91,6 +92,12 @@
 
         if (shooter instanceof EntityPlayer)
         {
+        	//VIVE SET SPAWNPOS
+            Vec3d aimSource = Minecraft.getMinecraft().roomScale.getControllerPos_World(1);
+            Vec3d controllerDirection = Minecraft.getMinecraft().bowTracker.getAimVector();
+            this.setPosition(aimSource.xCoord + controllerDirection.xCoord * 0.1f,aimSource.yCoord + controllerDirection.yCoord * 0.1f,aimSource.zCoord + controllerDirection.zCoord * 0.1f);
+            ///
+            
             this.pickupStatus = EntityArrow.PickupStatus.ALLOWED;
         }
     }
@@ -116,12 +123,20 @@
         this.dataManager.register(CRITICAL, Byte.valueOf((byte)0));
     }
 
-    public void setAim(Entity p_184547_1_, float p_184547_2_, float p_184547_3_, float p_184547_4_, float p_184547_5_, float p_184547_6_)
+    public void setAim(Entity p_184547_1_, float p_184547_2_, float p_184547_3_, float p_184547_4_, float velocity, float inaccuracy)
     {
         float f = -MathHelper.sin(p_184547_3_ * 0.017453292F) * MathHelper.cos(p_184547_2_ * 0.017453292F);
         float f1 = -MathHelper.sin(p_184547_2_ * 0.017453292F);
         float f2 = MathHelper.cos(p_184547_3_ * 0.017453292F) * MathHelper.cos(p_184547_2_ * 0.017453292F);
-        this.setThrowableHeading((double)f, (double)f1, (double)f2, p_184547_5_, p_184547_6_);
+        
+        //VIVE SET VELOCITY
+        if(this.shootingEntity instanceof EntityPlayer && !Minecraft.getMinecraft().vrSettings.seated ){
+        	velocity = (float) (Minecraft.getMinecraft().bowTracker.getDrawPercent() * 3f);
+        	this.damage *= 2;
+        }
+        ///
+        
+        this.setThrowableHeading((double)f, (double)f1, (double)f2, velocity, inaccuracy);
         this.motionX += p_184547_1_.motionX;
         this.motionZ += p_184547_1_.motionZ;
 
