--- a/net/minecraft/entity/projectile/EntityFishHook.java
+++ b/net/minecraft/entity/projectile/EntityFishHook.java
@@ -3,6 +3,7 @@
 import java.util.List;
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
+import net.minecraft.client.Minecraft;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.item.EntityItem;
@@ -77,7 +78,14 @@
         this.angler = fishingPlayer;
         this.angler.fishEntity = this;
         this.setSize(0.25F, 0.25F);
-        this.setLocationAndAngles(fishingPlayer.posX, fishingPlayer.posY + (double)fishingPlayer.getEyeHeight(), fishingPlayer.posZ, fishingPlayer.rotationYaw, fishingPlayer.rotationPitch);
+        
+    	//VIVE USE CONTROLLER
+        float yaw = Minecraft.getMinecraft().roomScale.getControllerYaw_World(0);
+        float pitch = -Minecraft.getMinecraft().roomScale.getControllerPitch_World(0);
+        Vec3d aimSource = Minecraft.getMinecraft().roomScale.getControllerPos_World(0);
+        this.setLocationAndAngles(aimSource.xCoord, aimSource.yCoord, aimSource.zCoord, yaw, pitch);
+        //END VIVE
+        
         this.posX -= (double)(MathHelper.cos(this.rotationYaw * 0.017453292F) * 0.16F);
         this.posY -= 0.10000000149011612D;
         this.posZ -= (double)(MathHelper.sin(this.rotationYaw * 0.017453292F) * 0.16F);
