--- a/net/minecraft/init/Items.java
+++ b/net/minecraft/init/Items.java
@@ -219,6 +219,8 @@
     public static final Item BEETROOT_SEEDS;
     public static final Item BEETROOT;
     public static final Item BEETROOT_SOUP;
+    public static final Item EATME_CAKE;
+    public static final Item DRIINKME_BOTTLE;
 
     private static Item getRegisteredItem(String name)
     {
@@ -447,6 +449,8 @@
             BEETROOT_SEEDS = getRegisteredItem("beetroot_seeds");
             BEETROOT = getRegisteredItem("beetroot");
             BEETROOT_SOUP = getRegisteredItem("beetroot_soup");
+            EATME_CAKE = getRegisteredItem("eatme_cake");
+            DRIINKME_BOTTLE = getRegisteredItem("drinkme_bottle");
         }
     }
 }
