diff -r -U 3 minecraft_server\net\minecraft\command\CommandCompare.java minecraft_server_patched\net\minecraft\command\CommandCompare.java
--- minecraft_server\net\minecraft\command\CommandCompare.java
+++ minecraft_server_patched\net\minecraft\command\CommandCompare.java
@@ -107,6 +107,6 @@
    }
 
    public List<String> func_184883_a(MinecraftServer p_184883_1_, ICommandSender p_184883_2_, String[] p_184883_3_, @Nullable BlockPos p_184883_4_) {
-      return p_184883_3_.length > 0 && p_184883_3_.length <= 3?func_175771_a(p_184883_3_, 0, p_184883_4_):(p_184883_3_.length > 3 && p_184883_3_.length <= 6?func_175771_a(p_184883_3_, 3, p_184883_4_):(p_184883_3_.length > 6 && p_184883_3_.length <= 9?func_175771_a(p_184883_3_, 6, p_184883_4_):(p_184883_3_.length == 10?func_71530_a(p_184883_3_, new String[]{"masked", "all"}):Collections.emptyList())));
+      return p_184883_3_.length > 0 && p_184883_3_.length <= 3?func_175771_a(p_184883_3_, 0, p_184883_4_):(p_184883_3_.length > 3 && p_184883_3_.length <= 6?func_175771_a(p_184883_3_, 3, p_184883_4_):(p_184883_3_.length > 6 && p_184883_3_.length <= 9?func_175771_a(p_184883_3_, 6, p_184883_4_):(p_184883_3_.length == 10?func_71530_a(p_184883_3_, new String[]{"masked", "all"}):Collections.<String>emptyList())));
    }
 }
