diff -r -U 3 minecraft_server\net\minecraft\command\server\CommandSetDefaultSpawnpoint.java minecraft_server_patched\net\minecraft\command\server\CommandSetDefaultSpawnpoint.java
--- minecraft_server\net\minecraft\command\server\CommandSetDefaultSpawnpoint.java
+++ minecraft_server_patched\net\minecraft\command\server\CommandSetDefaultSpawnpoint.java
@@ -42,6 +42,6 @@
    }
 
    public List<String> func_184883_a(MinecraftServer p_184883_1_, ICommandSender p_184883_2_, String[] p_184883_3_, @Nullable BlockPos p_184883_4_) {
-      return p_184883_3_.length > 0 && p_184883_3_.length <= 3?func_175771_a(p_184883_3_, 0, p_184883_4_):Collections.emptyList();
+      return p_184883_3_.length > 0 && p_184883_3_.length <= 3?func_175771_a(p_184883_3_, 0, p_184883_4_):Collections.<String>emptyList();
    }
 }
