diff -r -U 3 minecraft_server\net\minecraft\entity\Entity.java minecraft_server_patched\net\minecraft\entity\Entity.java
--- minecraft_server\net\minecraft\entity\Entity.java
+++ minecraft_server_patched\net\minecraft\entity\Entity.java
@@ -486,7 +486,7 @@
          }
 
          this.func_174826_a(this.func_174813_aQ().func_72317_d(0.0D, p_70091_3_, 0.0D));
-         i = this.field_70122_E || d4 != p_70091_3_ && d4 < 0.0D;
+         boolean i_ = this.field_70122_E || d4 != p_70091_3_ && d4 < 0.0D;
          int j4 = 0;
 
          for(int k = list1.size(); j4 < k; ++j4) {
@@ -501,7 +501,7 @@
          }
 
          this.func_174826_a(this.func_174813_aQ().func_72317_d(0.0D, 0.0D, p_70091_5_));
-         if(this.field_70138_W > 0.0F && i && (d3 != p_70091_1_ || d5 != p_70091_5_)) {
+         if(this.field_70138_W > 0.0F && i_ && (d3 != p_70091_1_ || d5 != p_70091_5_)) {
             double d11 = p_70091_1_;
             double d7 = p_70091_3_;
             double d8 = p_70091_5_;
@@ -2089,7 +2089,7 @@
    private <T extends Entity> void func_184175_a(Class<T> p_184175_1_, Set<T> p_184175_2_) {
       for(Entity entity : this.func_184188_bt()) {
          if(p_184175_1_.isAssignableFrom(entity.getClass())) {
-            p_184175_2_.add(entity);
+            p_184175_2_.add((T)entity);
          }
 
          entity.func_184175_a(p_184175_1_, p_184175_2_);
