diff -r -U 3 minecraft_server\net\minecraft\entity\player\EntityPlayer.java minecraft_server_patched\net\minecraft\entity\player\EntityPlayer.java
--- minecraft_server\net\minecraft\entity\player\EntityPlayer.java
+++ minecraft_server_patched\net\minecraft\entity\player\EntityPlayer.java
@@ -1167,6 +1167,7 @@
       return EntityPlayer.SleepResult.OK;
    }
 
+   @SuppressWarnings("incomplete-switch")
    private void func_175139_a(EnumFacing p_175139_1_) {
       this.field_71079_bU = 0.0F;
       this.field_71089_bV = 0.0F;
