--- a/net/minecraft/client/renderer/color/BlockColors.java
+++ b/net/minecraft/client/renderer/color/BlockColors.java
@@ -51,7 +51,7 @@
 
                         if (item instanceof ItemBlock)
                         {
-                            IBlockState iblockstate = Block.getBlockFromItem(item).func_176223_P();
+                            IBlockState iblockstate = Block.getBlockFromItem(item).getBlockState();
                             return blockcolors.func_186724_a(iblockstate, p_186720_2_, p_186720_3_, p_186720_4_);
                         }
                     }
@@ -144,7 +144,7 @@
 
     public int func_189991_a(IBlockState p_189991_1_)
     {
-        IBlockColor iblockcolor = (IBlockColor)this.field_186725_a.func_148745_a(Block.getIdFromBlock(p_189991_1_.func_177230_c()));
+        IBlockColor iblockcolor = (IBlockColor)this.field_186725_a.func_148745_a(Block.getIdFromBlock(p_189991_1_.getBlock()));
 
         if (iblockcolor != null)
         {
@@ -159,7 +159,7 @@
 
     public int func_186724_a(IBlockState p_186724_1_, @Nullable IBlockAccess p_186724_2_, @Nullable BlockPos p_186724_3_, int p_186724_4_)
     {
-        IBlockColor iblockcolor = (IBlockColor)this.field_186725_a.func_148745_a(Block.getIdFromBlock(p_186724_1_.func_177230_c()));
+        IBlockColor iblockcolor = (IBlockColor)this.field_186725_a.func_148745_a(Block.getIdFromBlock(p_186724_1_.getBlock()));
         return iblockcolor == null ? -1 : iblockcolor.func_186720_a(p_186724_1_, p_186724_2_, p_186724_3_, p_186724_4_);
     }
 
