--- a/net/minecraft/client/gui/achievement/GuiAchievements.java
+++ b/net/minecraft/client/gui/achievement/GuiAchievements.java
@@ -272,9 +272,9 @@
         int j1 = l + 17;
         this.zLevel = 0.0F;
         GlStateManager.func_179143_c(518);
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         GlStateManager.func_179109_b((float)i1, (float)j1, -200.0F);
-        GlStateManager.func_179152_a(1.0F / this.field_146570_r, 1.0F / this.field_146570_r, 0.0F);
+        GlStateManager.scale(1.0F / this.field_146570_r, 1.0F / this.field_146570_r, 0.0F);
         GlStateManager.func_179098_w();
         GlStateManager.func_179140_f();
         GlStateManager.func_179091_B();
@@ -487,7 +487,7 @@
 
         GlStateManager.func_179097_i();
         GlStateManager.func_179147_l();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
         this.mc.getTextureManager().bindTexture(field_146561_C);
         this.drawTexturedModalRect(k, l, 0, 0, this.field_146555_f, this.field_146557_g);
@@ -558,7 +558,7 @@
 
     private TextureAtlasSprite func_175371_a(Block p_175371_1_)
     {
-        return Minecraft.getMinecraft().func_175602_ab().func_175023_a().func_178122_a(p_175371_1_.func_176223_P());
+        return Minecraft.getMinecraft().func_175602_ab().func_175023_a().func_178122_a(p_175371_1_.getBlockState());
     }
 
     /**
