--- a/net/minecraft/client/gui/achievement/GuiAchievement.java
+++ b/net/minecraft/client/gui/achievement/GuiAchievement.java
@@ -50,21 +50,21 @@
     private void updateAchievementWindowScale()
     {
         GlStateManager.func_179083_b(0, 0, this.mc.displayWidth, this.mc.displayHeight);
-        GlStateManager.func_179128_n(5889);
-        GlStateManager.func_179096_D();
-        GlStateManager.func_179128_n(5888);
-        GlStateManager.func_179096_D();
+        GlStateManager.matrixMode(5889);
+        GlStateManager.loadIdentity();
+        GlStateManager.matrixMode(5888);
+        GlStateManager.loadIdentity();
         this.width = this.mc.displayWidth;
         this.height = this.mc.displayHeight;
         ScaledResolution scaledresolution = new ScaledResolution(this.mc);
         this.width = scaledresolution.getScaledWidth();
         this.height = scaledresolution.getScaledHeight();
         GlStateManager.func_179086_m(256);
-        GlStateManager.func_179128_n(5889);
-        GlStateManager.func_179096_D();
+        GlStateManager.matrixMode(5889);
+        GlStateManager.loadIdentity();
         GlStateManager.func_179130_a(0.0D, (double)this.width, (double)this.height, 0.0D, 1000.0D, 3000.0D);
-        GlStateManager.func_179128_n(5888);
-        GlStateManager.func_179096_D();
+        GlStateManager.matrixMode(5888);
+        GlStateManager.loadIdentity();
         GlStateManager.func_179109_b(0.0F, 0.0F, -2000.0F);
     }
 
