--- a/net/minecraft/src/Config.java
+++ b/net/minecraft/src/Config.java
@@ -125,18 +125,21 @@
 
     public static void initGameSettings(GameSettings p_initGameSettings_0_)
     {
-        if (gameSettings == null)
-        {
+       // if (gameSettings == null)
+       // {
             gameSettings = p_initGameSettings_0_;
             desktopDisplayMode = Display.getDesktopDisplayMode();
             updateAvailableProcessors();
             ReflectorForge.putLaunchBlackboard("optifine.ForgeSplashCompatible", Boolean.TRUE);
-        }
+        //}
     }
 
     public static void initDisplay()
     {
         checkInitialized();
+        //VIVE
+        gameSettings.ofAaLevel = 0;
+        //
         antialiasingLevel = gameSettings.ofAaLevel;
         checkDisplaySettings();
         checkDisplayMode();
@@ -1016,7 +1019,8 @@
 
     public static boolean isVignetteEnabled()
     {
-        return isShaders() && !Shaders.isVignette() ? false : (gameSettings.ofVignette == 0 ? gameSettings.fancyGraphics : gameSettings.ofVignette == 2);
+    	return true;
+       // return isShaders() && !Shaders.isVignette() ? false : (gameSettings.ofVignette == 0 ? gameSettings.fancyGraphics : gameSettings.ofVignette == 2);
     }
 
     public static boolean isStarsEnabled()
@@ -1937,12 +1941,7 @@
 
     public static void updateFramebufferSize()
     {
-        minecraft.getFramebuffer().createBindFramebuffer(minecraft.displayWidth, minecraft.displayHeight);
-
-        if (minecraft.entityRenderer != null)
-        {
-            minecraft.entityRenderer.updateShaderGroupSize(minecraft.displayWidth, minecraft.displayHeight);
-        }
+    	Minecraft.getMinecraft().reinitFramebuffers = true; 
     }
 
     public static Object[] addObjectToArray(Object[] p_addObjectToArray_0_, Object p_addObjectToArray_1_)
@@ -2142,6 +2141,7 @@
 
     public static boolean isDynamicLights()
     {
+    	boolean wat = Minecraft.getMinecraft().gameSettings == gameSettings;
         return gameSettings.ofDynamicLights != 3;
     }
 
