--- a/net/minecraft/src/Config.java
+++ b/net/minecraft/src/Config.java
@@ -135,6 +135,9 @@
     public static void initDisplay()
     {
         checkInitialized();
+        //VIVE
+        gameSettings.ofAaLevel = 0;
+        //
         antialiasingLevel = gameSettings.ofAaLevel;
         checkDisplaySettings();
         checkDisplayMode();
@@ -920,7 +923,8 @@
 
     public static boolean isVignetteEnabled()
     {
-        return gameSettings.ofVignette == 0 ? gameSettings.fancyGraphics : gameSettings.ofVignette == 2;
+    	return true;
+       // return gameSettings.ofVignette == 0 ? gameSettings.fancyGraphics : gameSettings.ofVignette == 2;
     }
 
     public static boolean isStarsEnabled()
@@ -1812,12 +1816,7 @@
 
     public static void updateFramebufferSize()
     {
-        minecraft.getFramebuffer().createBindFramebuffer(minecraft.displayWidth, minecraft.displayHeight);
-
-        if (minecraft.entityRenderer != null)
-        {
-            minecraft.entityRenderer.updateShaderGroupSize(minecraft.displayWidth, minecraft.displayHeight);
-        }
+    	Minecraft.getMinecraft().reinitFramebuffers = true; 
     }
 
     public static Object[] addObjectToArray(Object[] p_addObjectToArray_0_, Object p_addObjectToArray_1_)
