--- a/net/minecraft/src/CustomColors.java
+++ b/net/minecraft/src/CustomColors.java
@@ -369,8 +369,9 @@
         String s = "palette.block.";
         Map map = new HashMap();
 
-        for (String s1 : p_readCustomColormaps_0_.keySet())
+        for (Object s3 : p_readCustomColormaps_0_.keySet())
         {
+        	String s1 = (String) s3;
             String s2 = p_readCustomColormaps_0_.getProperty(s1);
 
             if (s1.startsWith(s))
@@ -1352,8 +1353,9 @@
         Set set = p_readSpawnEggColors_0_.keySet();
         int i = 0;
 
-        for (String s : set)
+        for (Object s3 : set)
         {
+        	String s = (String) s3;
             String s1 = p_readSpawnEggColors_0_.getProperty(s);
 
             if (s.startsWith(p_readSpawnEggColors_2_))
@@ -1463,8 +1465,9 @@
         float[][] afloat1 = new float[aenumdyecolor.length][];
         int k = 0;
 
-        for (String s : p_readDyeColors_0_.keySet())
+        for (Object s3 : p_readDyeColors_0_.keySet())
         {
+        	String s = (String) s3;
             String s1 = p_readDyeColors_0_.getProperty(s);
 
             if (s.startsWith(p_readDyeColors_2_))
@@ -1536,8 +1539,9 @@
         Arrays.fill((int[])aint, (int) - 1);
         int i = 0;
 
-        for (String s : p_readTextColors_0_.keySet())
+        for (Object s3 : p_readTextColors_0_.keySet())
         {
+        	String s = (String) s3;
             String s1 = p_readTextColors_0_.getProperty(s);
 
             if (s.startsWith(p_readTextColors_2_))
@@ -1592,8 +1596,9 @@
         Arrays.fill((int[])aint, (int) - 1);
         int i = 0;
 
-        for (String s : p_readMapColors_0_.keySet())
+        for (Object s3 : p_readMapColors_0_.keySet())
         {
+        	String s = (String) s3;
             String s1 = p_readMapColors_0_.getProperty(s);
 
             if (s.startsWith(p_readMapColors_2_))
@@ -1631,8 +1636,9 @@
         Arrays.fill((int[])aint, (int) - 1);
         int i = 0;
 
-        for (String s : p_readPotionColors_0_.keySet())
+        for (Object s3 : p_readPotionColors_0_.keySet())
         {
+        	String s = (String) s3;
             String s1 = p_readPotionColors_0_.getProperty(s);
 
             if (s.startsWith(p_readPotionColors_2_))
