--- a/net/minecraft/world/chunk/storage/ExtendedBlockStorage.java
+++ b/net/minecraft/world/chunk/storage/ExtendedBlockStorage.java
@@ -51,8 +51,8 @@
     public void func_177484_a(int p_177484_1_, int p_177484_2_, int p_177484_3_, IBlockState p_177484_4_)
     {
         IBlockState iblockstate = this.func_177485_a(p_177484_1_, p_177484_2_, p_177484_3_);
-        Block block = iblockstate.func_177230_c();
-        Block block1 = p_177484_4_.func_177230_c();
+        Block block = iblockstate.getBlock();
+        Block block1 = p_177484_4_.getBlock();
 
         if (block != Blocks.air)
         {
@@ -145,7 +145,7 @@
             {
                 for (int k = 0; k < 16; ++k)
                 {
-                    Block block = this.func_177485_a(i, j, k).func_177230_c();
+                    Block block = this.func_177485_a(i, j, k).getBlock();
 
                     if (block != Blocks.air)
                     {
