--- a/net/minecraft/entity/player/EntityPlayerMP.java
+++ b/net/minecraft/entity/player/EntityPlayerMP.java
@@ -180,7 +180,7 @@
         if (!p_i45285_2_.provider.func_177495_o() && p_i45285_2_.getWorldInfo().getGameType() != GameType.ADVENTURE)
         {
             int i = Math.max(0, p_i45285_1_.func_184108_a(p_i45285_2_));
-            int j = MathHelper.floor_double(p_i45285_2_.func_175723_af().func_177729_b((double)blockpos.func_177958_n(), (double)blockpos.func_177952_p()));
+            int j = MathHelper.floor_double(p_i45285_2_.func_175723_af().func_177729_b((double)blockpos.getX(), (double)blockpos.getZ()));
 
             if (j < i)
             {
@@ -200,7 +200,7 @@
         this.stepHeight = 0.0F;
         this.func_174828_a(blockpos, 0.0F, 0.0F);
 
-        while (!p_i45285_2_.func_184144_a(this, this.func_174813_aQ()).isEmpty() && this.posY < 255.0D)
+        while (!p_i45285_2_.getCollisionBoxes(this, this.func_174813_aQ()).isEmpty() && this.posY < 255.0D)
         {
             this.setPosition(this.posX, this.posY + 1.0D, this.posZ);
         }
@@ -233,14 +233,14 @@
     {
         super.writeEntityToNBT(tagCompound);
         tagCompound.setInteger("playerGameType", this.theItemInWorldManager.getGameType().getID());
-        Entity entity = this.func_184208_bv();
+        Entity entity = this.getMovementEntity();
 
-        if (this.func_184187_bx() != null && entity != this & entity.func_184180_b(EntityPlayerMP.class).size() == 1)
+        if (this.getRidingEntity() != null && entity != this & entity.func_184180_b(EntityPlayerMP.class).size() == 1)
         {
             NBTTagCompound nbttagcompound = new NBTTagCompound();
             NBTTagCompound nbttagcompound1 = new NBTTagCompound();
             entity.writeToNBTOptional(nbttagcompound1);
-            nbttagcompound.func_186854_a("Attach", this.func_184187_bx().getUniqueID());
+            nbttagcompound.func_186854_a("Attach", this.getRidingEntity().getUniqueID());
             nbttagcompound.setTag("Entity", nbttagcompound1);
             tagCompound.setTag("RootVehicle", nbttagcompound);
         }
@@ -631,7 +631,7 @@
             }
 
             this.mcServer.func_184103_al().func_187242_a(this, p_184204_1_);
-            this.playerNetServerHandler.sendPacket(new SPacketEffect(1032, BlockPos.field_177992_a, 0, false));
+            this.playerNetServerHandler.sendPacket(new SPacketEffect(1032, BlockPos.createDefault, 0, false));
             this.lastExperience = -1;
             this.lastHealth = -1.0F;
             this.lastFoodLevel = -1;
@@ -705,7 +705,7 @@
 
     public boolean func_184205_a(Entity p_184205_1_, boolean p_184205_2_)
     {
-        Entity entity = this.func_184187_bx();
+        Entity entity = this.getRidingEntity();
 
         if (!super.func_184205_a(p_184205_1_, p_184205_2_))
         {
@@ -713,7 +713,7 @@
         }
         else
         {
-            Entity entity1 = this.func_184187_bx();
+            Entity entity1 = this.getRidingEntity();
 
             if (entity1 != entity && this.playerNetServerHandler != null)
             {
@@ -724,11 +724,11 @@
         }
     }
 
-    public void func_184210_p()
+    public void dismount()
     {
-        Entity entity = this.func_184187_bx();
-        super.func_184210_p();
-        Entity entity1 = this.func_184187_bx();
+        Entity entity = this.getRidingEntity();
+        super.dismount();
+        Entity entity1 = this.getRidingEntity();
 
         if (entity1 != entity && this.playerNetServerHandler != null)
         {
@@ -762,13 +762,13 @@
         int j = MathHelper.floor_double(this.posY - 0.20000000298023224D);
         int k = MathHelper.floor_double(this.posZ);
         BlockPos blockpos = new BlockPos(i, j, k);
-        IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
+        IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
 
-        if (iblockstate.func_185904_a() == Material.air)
+        if (iblockstate.getMaterial() == Material.air)
         {
-            BlockPos blockpos1 = blockpos.func_177977_b();
-            IBlockState iblockstate1 = this.worldObj.func_180495_p(blockpos1);
-            Block block = iblockstate1.func_177230_c();
+            BlockPos blockpos1 = blockpos.getBlockPosBelow();
+            IBlockState iblockstate1 = this.worldObj.getBlockState(blockpos1);
+            Block block = iblockstate1.getBlock();
 
             if (block instanceof BlockFence || block instanceof BlockWall || block instanceof BlockFenceGate)
             {
@@ -986,7 +986,7 @@
 
     public void setEntityActionState(float p_110430_1_, float p_110430_2_, boolean p_110430_3_, boolean p_110430_4_)
     {
-        if (this.func_184218_aH())
+        if (this.isRiding())
         {
             if (p_110430_1_ >= -1.0F && p_110430_1_ <= 1.0F)
             {
@@ -1162,7 +1162,7 @@
 
         if (gameType == GameType.SPECTATOR)
         {
-            this.func_184210_p();
+            this.dismount();
         }
         else
         {
