--- a/net/minecraft/world/biome/BiomeDesert.java
+++ b/net/minecraft/world/biome/BiomeDesert.java
@@ -14,8 +14,8 @@
     {
         super(p_i46711_1_);
         this.spawnableCreatureList.clear();
-        this.topBlock = Blocks.sand.func_176223_P();
-        this.fillerBlock = Blocks.sand.func_176223_P();
+        this.topBlock = Blocks.sand.getBlockState();
+        this.fillerBlock = Blocks.sand.getBlockState();
         this.theBiomeDecorator.treesPerChunk = -999;
         this.theBiomeDecorator.deadBushPerChunk = 2;
         this.theBiomeDecorator.reedsPerChunk = 50;
@@ -32,7 +32,7 @@
         {
             int i = p_180624_2_.nextInt(16) + 8;
             int j = p_180624_2_.nextInt(16) + 8;
-            BlockPos blockpos = p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(i, 0, j)).func_177984_a();
+            BlockPos blockpos = p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(i, 0, j)).getBlockPosAbove();
             (new WorldGenDesertWells()).func_180709_b(p_180624_1_, p_180624_2_, blockpos);
         }
 
