--- a/net/minecraft/world/biome/BiomeMushroomIsland.java
+++ b/net/minecraft/world/biome/BiomeMushroomIsland.java
@@ -13,7 +13,7 @@
         this.theBiomeDecorator.grassPerChunk = -100;
         this.theBiomeDecorator.mushroomsPerChunk = 1;
         this.theBiomeDecorator.bigMushroomsPerChunk = 1;
-        this.topBlock = Blocks.mycelium.func_176223_P();
+        this.topBlock = Blocks.mycelium.getBlockState();
         this.spawnableMonsterList.clear();
         this.spawnableCreatureList.clear();
         this.spawnableWaterCreatureList.clear();
