--- a/net/minecraft/world/biome/BiomeTaiga.java
+++ b/net/minecraft/world/biome/BiomeTaiga.java
@@ -82,7 +82,7 @@
         {
             int j1 = p_180624_2_.nextInt(16) + 8;
             int k1 = p_180624_2_.nextInt(16) + 8;
-            int l1 = p_180624_2_.nextInt(p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(j1, 0, k1)).func_177956_o() + 32);
+            int l1 = p_180624_2_.nextInt(p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(j1, 0, k1)).getY() + 32);
             field_180280_ag.func_180709_b(p_180624_1_, p_180624_2_, p_180624_3_.func_177982_a(j1, l1, k1));
         }
 
@@ -93,16 +93,16 @@
     {
         if (this.field_150644_aH == BiomeTaiga.Type.MEGA || this.field_150644_aH == BiomeTaiga.Type.MEGA_SPRUCE)
         {
-            this.topBlock = Blocks.grass.func_176223_P();
-            this.fillerBlock = Blocks.dirt.func_176223_P();
+            this.topBlock = Blocks.grass.getBlockState();
+            this.fillerBlock = Blocks.dirt.getBlockState();
 
             if (p_180622_6_ > 1.75D)
             {
-                this.topBlock = Blocks.dirt.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.COARSE_DIRT);
+                this.topBlock = Blocks.dirt.getBlockState().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.COARSE_DIRT);
             }
             else if (p_180622_6_ > -0.95D)
             {
-                this.topBlock = Blocks.dirt.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.PODZOL);
+                this.topBlock = Blocks.dirt.getBlockState().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.PODZOL);
             }
         }
 
