--- a/net/minecraft/world/biome/BiomeStoneBeach.java
+++ b/net/minecraft/world/biome/BiomeStoneBeach.java
@@ -8,8 +8,8 @@
     {
         super(p_i46696_1_);
         this.spawnableCreatureList.clear();
-        this.topBlock = Blocks.stone.func_176223_P();
-        this.fillerBlock = Blocks.stone.func_176223_P();
+        this.topBlock = Blocks.stone.getBlockState();
+        this.fillerBlock = Blocks.stone.getBlockState();
         this.theBiomeDecorator.treesPerChunk = -999;
         this.theBiomeDecorator.deadBushPerChunk = 0;
         this.theBiomeDecorator.reedsPerChunk = 0;
