--- a/net/minecraft/world/biome/Biome.java
+++ b/net/minecraft/world/biome/Biome.java
@@ -53,14 +53,14 @@
 public abstract class Biome
 {
     private static final Logger logger = LogManager.getLogger();
-    protected static final IBlockState field_185365_a = Blocks.stone.func_176223_P();
-    protected static final IBlockState field_185366_b = Blocks.air.func_176223_P();
-    protected static final IBlockState field_185367_c = Blocks.bedrock.func_176223_P();
-    protected static final IBlockState field_185368_d = Blocks.gravel.func_176223_P();
-    protected static final IBlockState field_185369_e = Blocks.field_180395_cM.func_176223_P();
-    protected static final IBlockState field_185370_f = Blocks.sandstone.func_176223_P();
-    protected static final IBlockState field_185371_g = Blocks.ice.func_176223_P();
-    protected static final IBlockState field_185372_h = Blocks.water.func_176223_P();
+    protected static final IBlockState field_185365_a = Blocks.stone.getBlockState();
+    protected static final IBlockState field_185366_b = Blocks.air.getBlockState();
+    protected static final IBlockState field_185367_c = Blocks.bedrock.getBlockState();
+    protected static final IBlockState field_185368_d = Blocks.gravel.getBlockState();
+    protected static final IBlockState field_185369_e = Blocks.field_180395_cM.getBlockState();
+    protected static final IBlockState field_185370_f = Blocks.sandstone.getBlockState();
+    protected static final IBlockState field_185371_g = Blocks.ice.getBlockState();
+    protected static final IBlockState field_185372_h = Blocks.water.getBlockState();
     public static final Set<Biome> explorationBiomesList = Sets.<Biome>newHashSet();
     public static final ObjectIntIdentityMap<Biome> field_185373_j = new ObjectIntIdentityMap();
     protected static final NoiseGeneratorPerlin temperatureNoise = new NoiseGeneratorPerlin(new Random(1234L), 1);
@@ -104,10 +104,10 @@
     private final String field_185364_H;
 
     /** The block expected to be on the top of this biome */
-    public IBlockState topBlock = Blocks.grass.func_176223_P();
+    public IBlockState topBlock = Blocks.grass.getBlockState();
 
     /** The block to fill spots in when not on the top */
-    public IBlockState fillerBlock = Blocks.dirt.func_176223_P();
+    public IBlockState fillerBlock = Blocks.dirt.getBlockState();
 
     /** The biome decorator. */
     public BiomeDecorator theBiomeDecorator;
@@ -256,10 +256,10 @@
 
     public final float func_180626_a(BlockPos p_180626_1_)
     {
-        if (p_180626_1_.func_177956_o() > 64)
+        if (p_180626_1_.getY() > 64)
         {
-            float f = (float)(temperatureNoise.func_151601_a((double)((float)p_180626_1_.func_177958_n() / 8.0F), (double)((float)p_180626_1_.func_177952_p() / 8.0F)) * 4.0D);
-            return this.func_185353_n() - (f + (float)p_180626_1_.func_177956_o() - 64.0F) * 0.05F / 30.0F;
+            float f = (float)(temperatureNoise.func_151601_a((double)((float)p_180626_1_.getX() / 8.0F), (double)((float)p_180626_1_.getZ() / 8.0F)) * 4.0D);
+            return this.func_185353_n() - (f + (float)p_180626_1_.getY() - 64.0F) * 0.05F / 30.0F;
         }
         else
         {
@@ -293,7 +293,7 @@
 
     public final void func_180628_b(World p_180628_1_, Random p_180628_2_, ChunkPrimer p_180628_3_, int p_180628_4_, int p_180628_5_, double p_180628_6_)
     {
-        int i = p_180628_1_.func_181545_F();
+        int i = p_180628_1_.getSpawnLevel();
         IBlockState iblockstate = this.topBlock;
         IBlockState iblockstate1 = this.fillerBlock;
         int j = -1;
@@ -312,11 +312,11 @@
             {
                 IBlockState iblockstate2 = p_180628_3_.func_177856_a(i1, j1, l);
 
-                if (iblockstate2.func_185904_a() == Material.air)
+                if (iblockstate2.getMaterial() == Material.air)
                 {
                     j = -1;
                 }
-                else if (iblockstate2.func_177230_c() == Blocks.stone)
+                else if (iblockstate2.getBlock() == Blocks.stone)
                 {
                     if (j == -1)
                     {
@@ -331,7 +331,7 @@
                             iblockstate1 = this.fillerBlock;
                         }
 
-                        if (j1 < i && (iblockstate == null || iblockstate.func_185904_a() == Material.air))
+                        if (j1 < i && (iblockstate == null || iblockstate.getMaterial() == Material.air))
                         {
                             if (this.func_180626_a(blockpos$mutableblockpos.func_181079_c(p_180628_4_, j1, p_180628_5_)) < 0.15F)
                             {
@@ -365,7 +365,7 @@
                         --j;
                         p_180628_3_.func_177855_a(i1, j1, l, iblockstate1);
 
-                        if (j == 0 && iblockstate1.func_177230_c() == Blocks.sand && k > 1)
+                        if (j == 0 && iblockstate1.getBlock() == Blocks.sand && k > 1)
                         {
                             j = p_180628_2_.nextInt(4) + Math.max(0, j1 - 63);
                             iblockstate1 = iblockstate1.func_177229_b(BlockSand.field_176504_a) == BlockSand.EnumType.RED_SAND ? field_185369_e : field_185370_f;
