--- a/net/minecraft/world/biome/BiomePlains.java
+++ b/net/minecraft/world/biome/BiomePlains.java
@@ -25,7 +25,7 @@
 
     public BlockFlower.EnumFlowerType func_180623_a(Random p_180623_1_, BlockPos p_180623_2_)
     {
-        double d0 = field_180281_af.func_151601_a((double)p_180623_2_.func_177958_n() / 200.0D, (double)p_180623_2_.func_177952_p() / 200.0D);
+        double d0 = field_180281_af.func_151601_a((double)p_180623_2_.getX() / 200.0D, (double)p_180623_2_.getZ() / 200.0D);
 
         if (d0 < -0.8D)
         {
@@ -60,7 +60,7 @@
 
     public void func_180624_a(World p_180624_1_, Random p_180624_2_, BlockPos p_180624_3_)
     {
-        double d0 = field_180281_af.func_151601_a((double)(p_180624_3_.func_177958_n() + 8) / 200.0D, (double)(p_180624_3_.func_177952_p() + 8) / 200.0D);
+        double d0 = field_180281_af.func_151601_a((double)(p_180624_3_.getX() + 8) / 200.0D, (double)(p_180624_3_.getZ() + 8) / 200.0D);
 
         if (d0 < -0.8D)
         {
@@ -77,7 +77,7 @@
             {
                 int j = p_180624_2_.nextInt(16) + 8;
                 int k = p_180624_2_.nextInt(16) + 8;
-                int l = p_180624_2_.nextInt(p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(j, 0, k)).func_177956_o() + 32);
+                int l = p_180624_2_.nextInt(p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(j, 0, k)).getY() + 32);
                 field_180280_ag.func_180709_b(p_180624_1_, p_180624_2_, p_180624_3_.func_177982_a(j, l, k));
             }
         }
@@ -90,7 +90,7 @@
             {
                 int j1 = p_180624_2_.nextInt(16) + 8;
                 int k1 = p_180624_2_.nextInt(16) + 8;
-                int l1 = p_180624_2_.nextInt(p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(j1, 0, k1)).func_177956_o() + 32);
+                int l1 = p_180624_2_.nextInt(p_180624_1_.func_175645_m(p_180624_3_.func_177982_a(j1, 0, k1)).getY() + 32);
                 field_180280_ag.func_180709_b(p_180624_1_, p_180624_2_, p_180624_3_.func_177982_a(j1, l1, k1));
             }
         }
