--- a/net/minecraft/entity/passive/EntitySquid.java
+++ b/net/minecraft/entity/passive/EntitySquid.java
@@ -225,7 +225,7 @@
      */
     public boolean getCanSpawnHere()
     {
-        return this.posY > 45.0D && this.posY < (double)this.worldObj.func_181545_F() && super.getCanSpawnHere();
+        return this.posY > 45.0D && this.posY < (double)this.worldObj.getSpawnLevel() && super.getCanSpawnHere();
     }
 
     public void handleHealthUpdate(byte p_70103_1_)
