--- a/net/minecraft/entity/passive/EntityRabbit.java
+++ b/net/minecraft/entity/passive/EntityRabbit.java
@@ -150,7 +150,7 @@
 
         if (p_70637_1_)
         {
-            this.func_184185_a(this.func_184771_da(), this.getSoundVolume(), ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F) * 0.8F);
+            this.playSound(this.func_184771_da(), this.getSoundVolume(), ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F) * 0.8F);
         }
     }
 
@@ -345,7 +345,7 @@
     {
         if (this.func_175531_cl() == 99)
         {
-            this.func_184185_a(SoundEvents.field_187818_ek, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
+            this.playSound(SoundEvents.field_187818_ek, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
             return p_70652_1_.attackEntityFrom(DamageSource.causeMobDamage(this), 8.0F);
         }
         else
@@ -475,7 +475,7 @@
     {
         BlockCarrot blockcarrot = (BlockCarrot)Blocks.carrots;
         IBlockState iblockstate = blockcarrot.func_185528_e(blockcarrot.func_185526_g());
-        this.worldObj.func_175688_a(EnumParticleTypes.BLOCK_DUST, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, 0.0D, 0.0D, 0.0D, new int[] {Block.func_176210_f(iblockstate)});
+        this.worldObj.spawnParticle(EnumParticleTypes.BLOCK_DUST, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, 0.0D, 0.0D, 0.0D, new int[] {Block.func_176210_f(iblockstate)});
         this.field_175541_bs = 40;
     }
 
@@ -591,14 +591,14 @@
         public void updateTask()
         {
             super.updateTask();
-            this.field_179500_c.getLookHelper().setLookPosition((double)this.field_179494_b.func_177958_n() + 0.5D, (double)(this.field_179494_b.func_177956_o() + 1), (double)this.field_179494_b.func_177952_p() + 0.5D, 10.0F, (float)this.field_179500_c.getVerticalFaceSpeed());
+            this.field_179500_c.getLookHelper().setLookPosition((double)this.field_179494_b.getX() + 0.5D, (double)(this.field_179494_b.getY() + 1), (double)this.field_179494_b.getZ() + 0.5D, 10.0F, (float)this.field_179500_c.getVerticalFaceSpeed());
 
             if (this.func_179487_f())
             {
                 World world = this.field_179500_c.worldObj;
-                BlockPos blockpos = this.field_179494_b.func_177984_a();
-                IBlockState iblockstate = world.func_180495_p(blockpos);
-                Block block = iblockstate.func_177230_c();
+                BlockPos blockpos = this.field_179494_b.getBlockPosAbove();
+                IBlockState iblockstate = world.getBlockState(blockpos);
+                Block block = iblockstate.getBlock();
 
                 if (this.field_179499_e && block instanceof BlockCarrot)
                 {
@@ -606,7 +606,7 @@
 
                     if (integer.intValue() == 0)
                     {
-                        world.func_180501_a(blockpos, Blocks.air.func_176223_P(), 2);
+                        world.func_180501_a(blockpos, Blocks.air.getBlockState(), 2);
                         world.func_175655_b(blockpos, true);
                     }
                     else
@@ -625,13 +625,13 @@
 
         protected boolean func_179488_a(World p_179488_1_, BlockPos p_179488_2_)
         {
-            Block block = p_179488_1_.func_180495_p(p_179488_2_).func_177230_c();
+            Block block = p_179488_1_.getBlockState(p_179488_2_).getBlock();
 
             if (block == Blocks.farmland && this.field_179498_d && !this.field_179499_e)
             {
-                p_179488_2_ = p_179488_2_.func_177984_a();
-                IBlockState iblockstate = p_179488_1_.func_180495_p(p_179488_2_);
-                block = iblockstate.func_177230_c();
+                p_179488_2_ = p_179488_2_.getBlockPosAbove();
+                IBlockState iblockstate = p_179488_1_.getBlockState(p_179488_2_);
+                block = iblockstate.getBlock();
 
                 if (block instanceof BlockCarrot && ((BlockCarrot)block).func_185525_y(iblockstate))
                 {
