--- a/net/minecraft/entity/passive/EntityHorse.java
+++ b/net/minecraft/entity/passive/EntityHorse.java
@@ -1717,8 +1717,6 @@
 
     public void updatePassenger(Entity passenger)
     {
-        super.updatePassenger(passenger);
-
         if (passenger instanceof EntityLiving)
         {
             EntityLiving entityliving = (EntityLiving)passenger;
@@ -1737,7 +1735,9 @@
             {
                 ((EntityLivingBase)passenger).renderYawOffset = this.renderYawOffset;
             }
-        }
+        } else 
+            super.updatePassenger(passenger); //VIVE move this down here, do not call setPosition twice per tick.
+        	
     }
 
     /**
