--- a/net/minecraft/entity/passive/EntityAnimal.java
+++ b/net/minecraft/entity/passive/EntityAnimal.java
@@ -58,7 +58,7 @@
                 double d0 = this.rand.nextGaussian() * 0.02D;
                 double d1 = this.rand.nextGaussian() * 0.02D;
                 double d2 = this.rand.nextGaussian() * 0.02D;
-                this.worldObj.func_175688_a(EnumParticleTypes.HEART, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, d0, d1, d2, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.HEART, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, d0, d1, d2, new int[0]);
             }
         }
     }
@@ -81,7 +81,7 @@
 
     public float func_180484_a(BlockPos p_180484_1_)
     {
-        return this.worldObj.func_180495_p(p_180484_1_.func_177977_b()).func_177230_c() == Blocks.grass ? 10.0F : this.worldObj.func_175724_o(p_180484_1_) - 0.5F;
+        return this.worldObj.getBlockState(p_180484_1_.getBlockPosBelow()).getBlock() == Blocks.grass ? 10.0F : this.worldObj.func_175724_o(p_180484_1_) - 0.5F;
     }
 
     /**
@@ -119,7 +119,7 @@
         int j = MathHelper.floor_double(this.func_174813_aQ().minY);
         int k = MathHelper.floor_double(this.posZ);
         BlockPos blockpos = new BlockPos(i, j, k);
-        return this.worldObj.func_180495_p(blockpos.func_177977_b()).func_177230_c() == this.field_175506_bl && this.worldObj.func_175699_k(blockpos) > 8 && super.getCanSpawnHere();
+        return this.worldObj.getBlockState(blockpos.getBlockPosBelow()).getBlock() == this.field_175506_bl && this.worldObj.func_175699_k(blockpos) > 8 && super.getCanSpawnHere();
     }
 
     /**
@@ -227,7 +227,7 @@
                 double d0 = this.rand.nextGaussian() * 0.02D;
                 double d1 = this.rand.nextGaussian() * 0.02D;
                 double d2 = this.rand.nextGaussian() * 0.02D;
-                this.worldObj.func_175688_a(EnumParticleTypes.HEART, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, d0, d1, d2, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.HEART, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, d0, d1, d2, new int[0]);
             }
         }
         else
