--- a/net/minecraft/entity/passive/EntityWolf.java
+++ b/net/minecraft/entity/passive/EntityWolf.java
@@ -153,7 +153,7 @@
 
     protected void func_180429_a(BlockPos p_180429_1_, Block p_180429_2_)
     {
-        this.func_184185_a(SoundEvents.field_187869_gK, 0.15F, 1.0F);
+        this.playSound(SoundEvents.field_187869_gK, 0.15F, 1.0F);
     }
 
     public static void func_189788_b(DataFixer p_189788_0_)
@@ -264,7 +264,7 @@
         {
             if (this.timeWolfIsShaking == 0.0F)
             {
-                this.func_184185_a(SoundEvents.field_187867_gJ, this.getSoundVolume(), (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
+                this.playSound(SoundEvents.field_187867_gJ, this.getSoundVolume(), (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
             }
 
             this.prevTimeWolfIsShaking = this.timeWolfIsShaking;
@@ -287,7 +287,7 @@
                 {
                     float f1 = (this.rand.nextFloat() * 2.0F - 1.0F) * this.width * 0.5F;
                     float f2 = (this.rand.nextFloat() * 2.0F - 1.0F) * this.width * 0.5F;
-                    this.worldObj.func_175688_a(EnumParticleTypes.WATER_SPLASH, this.posX + (double)f1, (double)(f + 0.8F), this.posZ + (double)f2, this.motionX, this.motionY, this.motionZ, new int[0]);
+                    this.worldObj.spawnParticle(EnumParticleTypes.WATER_SPLASH, this.posX + (double)f1, (double)(f + 0.8F), this.posZ + (double)f2, this.motionX, this.motionY, this.motionZ, new int[0]);
                 }
             }
         }
