--- a/net/minecraft/entity/passive/EntityPig.java
+++ b/net/minecraft/entity/passive/EntityPig.java
@@ -93,7 +93,7 @@
         else
         {
             EntityPlayer entityplayer = (EntityPlayer)entity;
-            ItemStack itemstack = entityplayer.func_184614_ca();
+            ItemStack itemstack = entityplayer.getMainHandItem();
 
             if (itemstack != null && itemstack.getItem() == Items.carrot_on_a_stick)
             {
@@ -101,7 +101,7 @@
             }
             else
             {
-                itemstack = entityplayer.func_184592_cb();
+                itemstack = entityplayer.getOffhandItem();
                 return itemstack != null && itemstack.getItem() == Items.carrot_on_a_stick;
             }
         }
@@ -153,7 +153,7 @@
 
     protected void func_180429_a(BlockPos p_180429_1_, Block p_180429_2_)
     {
-        this.func_184185_a(SoundEvents.field_187709_dP, 0.15F, 1.0F);
+        this.playSound(SoundEvents.field_187709_dP, 0.15F, 1.0F);
     }
 
     public boolean func_184645_a(EntityPlayer p_184645_1_, EnumHand p_184645_2_, @Nullable ItemStack p_184645_3_)
@@ -226,7 +226,7 @@
         if (!this.worldObj.isRemote && !this.isDead)
         {
             EntityPigZombie entitypigzombie = new EntityPigZombie(this.worldObj);
-            entitypigzombie.func_184201_a(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.golden_sword));
+            entitypigzombie.equipItemInSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.golden_sword));
             entitypigzombie.setLocationAndAngles(this.posX, this.posY, this.posZ, this.rotationYaw, this.rotationPitch);
             entitypigzombie.setAlwaysRenderNameTag(this.func_175446_cd());
 
