--- a/net/minecraft/entity/passive/EntityHorse.java
+++ b/net/minecraft/entity/passive/EntityHorse.java
@@ -393,7 +393,7 @@
 
         if (!this.func_174814_R())
         {
-            this.worldObj.func_184148_a((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187711_cp, this.func_184176_by(), 1.0F, 1.0F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F);
+            this.worldObj.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187711_cp, this.func_184176_by(), 1.0F, 1.0F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F);
         }
     }
 
@@ -401,7 +401,7 @@
     {
         if (p_180430_1_ > 1.0F)
         {
-            this.func_184185_a(SoundEvents.field_187723_ct, 0.4F, 1.0F);
+            this.playSound(SoundEvents.field_187723_ct, 0.4F, 1.0F);
         }
 
         int i = MathHelper.ceiling_float_int((p_180430_1_ * 0.5F - 3.0F) * p_180430_2_);
@@ -418,13 +418,13 @@
                 }
             }
 
-            IBlockState iblockstate = this.worldObj.func_180495_p(new BlockPos(this.posX, this.posY - 0.2D - (double)this.prevRotationYaw, this.posZ));
-            Block block = iblockstate.func_177230_c();
+            IBlockState iblockstate = this.worldObj.getBlockState(new BlockPos(this.posX, this.posY - 0.2D - (double)this.prevRotationYaw, this.posZ));
+            Block block = iblockstate.getBlock();
 
-            if (iblockstate.func_185904_a() != Material.air && !this.func_174814_R())
+            if (iblockstate.getMaterial() != Material.air && !this.func_174814_R())
             {
                 SoundType soundtype = block.func_185467_w();
-                this.worldObj.func_184148_a((EntityPlayer)null, this.posX, this.posY, this.posZ, soundtype.func_185844_d(), this.func_184176_by(), soundtype.func_185843_a() * 0.5F, soundtype.func_185847_b() * 0.75F);
+                this.worldObj.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, soundtype.func_185844_d(), this.func_184176_by(), soundtype.func_185843_a() * 0.5F, soundtype.func_185847_b() * 0.75F);
             }
         }
     }
@@ -487,16 +487,16 @@
         {
             if (horsearmortype == HorseArmorType.NONE && horsearmortype != this.func_184783_dl())
             {
-                this.func_184185_a(SoundEvents.field_187702_cm, 0.5F, 1.0F);
+                this.playSound(SoundEvents.field_187702_cm, 0.5F, 1.0F);
             }
             else if (horsearmortype != this.func_184783_dl())
             {
-                this.func_184185_a(SoundEvents.field_187702_cm, 0.5F, 1.0F);
+                this.playSound(SoundEvents.field_187702_cm, 0.5F, 1.0F);
             }
 
             if (!flag && this.isHorseSaddled())
             {
-                this.func_184185_a(SoundEvents.field_187726_cu, 0.5F, 1.0F);
+                this.playSound(SoundEvents.field_187726_cu, 0.5F, 1.0F);
             }
         }
     }
@@ -582,12 +582,12 @@
     {
         SoundType soundtype = p_180429_2_.func_185467_w();
 
-        if (this.worldObj.func_180495_p(p_180429_1_.func_177984_a()).func_177230_c() == Blocks.snow_layer)
+        if (this.worldObj.getBlockState(p_180429_1_.getBlockPosAbove()).getBlock() == Blocks.snow_layer)
         {
             soundtype = Blocks.snow_layer.func_185467_w();
         }
 
-        if (!p_180429_2_.func_176223_P().func_185904_a().isLiquid())
+        if (!p_180429_2_.getBlockState().getMaterial().isLiquid())
         {
             HorseType horsetype = this.func_184781_cZ();
 
@@ -597,25 +597,25 @@
 
                 if (this.gallopTime > 5 && this.gallopTime % 3 == 0)
                 {
-                    this.func_184185_a(SoundEvents.field_187714_cq, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
+                    this.playSound(SoundEvents.field_187714_cq, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
 
                     if (horsetype == HorseType.HORSE && this.rand.nextInt(10) == 0)
                     {
-                        this.func_184185_a(SoundEvents.field_187705_cn, soundtype.func_185843_a() * 0.6F, soundtype.func_185847_b());
+                        this.playSound(SoundEvents.field_187705_cn, soundtype.func_185843_a() * 0.6F, soundtype.func_185847_b());
                     }
                 }
                 else if (this.gallopTime <= 5)
                 {
-                    this.func_184185_a(SoundEvents.field_187732_cw, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
+                    this.playSound(SoundEvents.field_187732_cw, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
                 }
             }
             else if (soundtype == SoundType.field_185848_a)
             {
-                this.func_184185_a(SoundEvents.field_187732_cw, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
+                this.playSound(SoundEvents.field_187732_cw, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
             }
             else
             {
-                this.func_184185_a(SoundEvents.field_187729_cv, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
+                this.playSound(SoundEvents.field_187729_cv, soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
             }
         }
     }
@@ -887,7 +887,7 @@
                 if (!flag && this.func_184781_cZ().func_188600_f() && !this.isChested() && p_184645_3_.getItem() == Item.getItemFromBlock(Blocks.chest))
                 {
                     this.setChested(true);
-                    this.func_184185_a(SoundEvents.field_187584_ax, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
+                    this.playSound(SoundEvents.field_187584_ax, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
                     flag = true;
                     this.func_110226_cD();
                 }
@@ -996,7 +996,7 @@
                 this.heal(1.0F);
             }
 
-            if (!this.isEatingHaystack() && !this.func_184207_aI() && this.rand.nextInt(300) == 0 && this.worldObj.func_180495_p(new BlockPos(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY) - 1, MathHelper.floor_double(this.posZ))).func_177230_c() == Blocks.grass)
+            if (!this.isEatingHaystack() && !this.func_184207_aI() && this.rand.nextInt(300) == 0 && this.worldObj.getBlockState(new BlockPos(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY) - 1, MathHelper.floor_double(this.posZ))).getBlock() == Blocks.grass)
             {
                 this.setEatingHaystack(true);
             }
@@ -1145,7 +1145,7 @@
      */
     private boolean canMate()
     {
-        return !this.func_184207_aI() && !this.func_184218_aH() && this.isTame() && this.isAdultHorse() && this.func_184781_cZ().func_188590_i() && this.getHealth() >= this.getMaxHealth() && this.isInLove();
+        return !this.func_184207_aI() && !this.isRiding() && this.isTame() && this.isAdultHorse() && this.func_184781_cZ().func_188590_i() && this.getHealth() >= this.getMaxHealth() && this.isInLove();
     }
 
     public void setEatingHaystack(boolean p_110227_1_)
@@ -1179,7 +1179,7 @@
 
         if (soundevent != null)
         {
-            this.func_184185_a(soundevent, this.getSoundVolume(), this.getSoundPitch());
+            this.playSound(soundevent, this.getSoundVolume(), this.getSoundPitch());
         }
     }
 
@@ -1259,7 +1259,7 @@
                     float f1 = MathHelper.cos(this.rotationYaw * 0.017453292F);
                     this.motionX += (double)(-0.4F * f * this.jumpPower);
                     this.motionZ += (double)(0.4F * f1 * this.jumpPower);
-                    this.func_184185_a(SoundEvents.field_187720_cs, 0.4F, 1.0F);
+                    this.playSound(SoundEvents.field_187720_cs, 0.4F, 1.0F);
                 }
 
                 this.jumpPower = 0.0F;
@@ -1684,7 +1684,7 @@
             double d0 = this.rand.nextGaussian() * 0.02D;
             double d1 = this.rand.nextGaussian() * 0.02D;
             double d2 = this.rand.nextGaussian() * 0.02D;
-            this.worldObj.func_175688_a(enumparticletypes, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, d0, d1, d2, new int[0]);
+            this.worldObj.spawnParticle(enumparticletypes, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, d0, d1, d2, new int[0]);
         }
     }
 
