--- a/net/minecraft/entity/passive/EntityChicken.java
+++ b/net/minecraft/entity/passive/EntityChicken.java
@@ -105,7 +105,7 @@
 
         if (!this.worldObj.isRemote && !this.isChild() && !this.func_152116_bZ() && --this.timeUntilNextEgg <= 0)
         {
-            this.func_184185_a(SoundEvents.field_187665_Y, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
+            this.playSound(SoundEvents.field_187665_Y, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
             this.dropItem(Items.egg, 1);
             this.timeUntilNextEgg = this.rand.nextInt(6000) + 6000;
         }
@@ -132,7 +132,7 @@
 
     protected void func_180429_a(BlockPos p_180429_1_, Block p_180429_2_)
     {
-        this.func_184185_a(SoundEvents.field_187538_aa, 0.15F, 1.0F);
+        this.playSound(SoundEvents.field_187538_aa, 0.15F, 1.0F);
     }
 
     @Nullable
