--- a/net/minecraft/client/renderer/tileentity/TileEntitySpecialRenderer.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntitySpecialRenderer.java
@@ -20,7 +20,7 @@
     {
         ITextComponent itextcomponent = p_180535_1_.getFormattedCommandSenderName();
 
-        if (itextcomponent != null && this.field_147501_a.field_190057_j != null && p_180535_1_.func_174877_v().equals(this.field_147501_a.field_190057_j.func_178782_a()))
+        if (itextcomponent != null && this.field_147501_a.field_190057_j != null && p_180535_1_.func_174877_v().equals(this.field_147501_a.field_190057_j.getBlock()))
         {
             this.func_190053_a(true);
             this.func_190052_a(p_180535_1_, itextcomponent.getFormattedText(), p_180535_2_, p_180535_4_, p_180535_6_, 12);
