--- a/net/minecraft/client/renderer/tileentity/TileEntityBannerRenderer.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntityBannerRenderer.java
@@ -20,14 +20,14 @@
         boolean flag1 = !flag || p_180535_1_.getBlockType() == Blocks.field_180393_cK;
         int i = flag ? p_180535_1_.getBlockMetadata() : 0;
         long j = flag ? p_180535_1_.getWorld().getTotalWorldTime() : 0L;
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         float f = 0.6666667F;
 
         if (flag1)
         {
             GlStateManager.func_179109_b((float)p_180535_2_ + 0.5F, (float)p_180535_4_ + 0.5F, (float)p_180535_6_ + 0.5F);
             float f1 = (float)(i * 360) / 16.0F;
-            GlStateManager.func_179114_b(-f1, 0.0F, 1.0F, 0.0F);
+            GlStateManager.rotate(-f1, 0.0F, 1.0F, 0.0F);
             this.field_178465_e.field_178688_b.showModel = true;
         }
         else
@@ -50,13 +50,13 @@
             }
 
             GlStateManager.func_179109_b((float)p_180535_2_ + 0.5F, (float)p_180535_4_ - 0.16666667F, (float)p_180535_6_ + 0.5F);
-            GlStateManager.func_179114_b(-f2, 0.0F, 1.0F, 0.0F);
+            GlStateManager.rotate(-f2, 0.0F, 1.0F, 0.0F);
             GlStateManager.func_179109_b(0.0F, -0.3125F, -0.4375F);
             this.field_178465_e.field_178688_b.showModel = false;
         }
 
         BlockPos blockpos = p_180535_1_.func_174877_v();
-        float f3 = (float)(blockpos.func_177958_n() * 7 + blockpos.func_177956_o() * 9 + blockpos.func_177952_p() * 13) + (float)j + p_180535_8_;
+        float f3 = (float)(blockpos.getX() * 7 + blockpos.getY() * 9 + blockpos.getZ() * 13) + (float)j + p_180535_8_;
         this.field_178465_e.field_178690_a.rotateAngleX = (-0.0125F + 0.01F * MathHelper.cos(f3 * (float)Math.PI * 0.02F)) * (float)Math.PI;
         GlStateManager.func_179091_B();
         ResourceLocation resourcelocation = this.func_178463_a(p_180535_1_);
@@ -64,14 +64,14 @@
         if (resourcelocation != null)
         {
             this.bindTexture(resourcelocation);
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179152_a(0.6666667F, -0.6666667F, -0.6666667F);
+            GlStateManager.push();
+            GlStateManager.scale(0.6666667F, -0.6666667F, -0.6666667F);
             this.field_178465_e.func_178687_a();
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
 
         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 
     @Nullable
