--- a/net/minecraft/client/renderer/tileentity/TileEntitySkullRenderer.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntitySkullRenderer.java
@@ -51,11 +51,11 @@
         if (p_188190_8_ >= 0)
         {
             this.bindTexture(field_178460_a[p_188190_8_]);
-            GlStateManager.func_179128_n(5890);
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179152_a(4.0F, 2.0F, 1.0F);
+            GlStateManager.matrixMode(5890);
+            GlStateManager.push();
+            GlStateManager.scale(4.0F, 2.0F, 1.0F);
             GlStateManager.func_179109_b(0.0625F, 0.0625F, 0.0625F);
-            GlStateManager.func_179128_n(5888);
+            GlStateManager.matrixMode(5888);
         }
         else
         {
@@ -108,7 +108,7 @@
             }
         }
 
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         GlStateManager.func_179129_p();
 
         if (p_188190_4_ == EnumFacing.UP)
@@ -142,7 +142,7 @@
 
         float f = 0.0625F;
         GlStateManager.func_179091_B();
-        GlStateManager.func_179152_a(-1.0F, -1.0F, 1.0F);
+        GlStateManager.scale(-1.0F, -1.0F, 1.0F);
         GlStateManager.func_179141_d();
 
         if (p_188190_6_ == 3)
@@ -151,13 +151,13 @@
         }
 
         modelbase.render((Entity)null, p_188190_9_, 0.0F, 0.0F, p_188190_5_, 0.0F, 0.0625F);
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
 
         if (p_188190_8_ >= 0)
         {
-            GlStateManager.func_179128_n(5890);
-            GlStateManager.func_179121_F();
-            GlStateManager.func_179128_n(5888);
+            GlStateManager.matrixMode(5890);
+            GlStateManager.pop();
+            GlStateManager.matrixMode(5888);
         }
     }
 }
