--- a/net/minecraft/client/renderer/tileentity/TileEntitySpecialRenderer.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntitySpecialRenderer.java
@@ -84,7 +84,7 @@
             float f = this.rendererDispatcher.entityYaw;
             float f1 = this.rendererDispatcher.entityPitch;
             boolean flag = false;
-            EntityRenderer.func_189692_a(this.getFontRenderer(), p_190052_2_, (float)p_190052_3_ + 0.5F, (float)p_190052_5_ + 1.5F, (float)p_190052_7_ + 0.5F, 0, f, f1, false, false);
+            EntityRenderer.drawNameplate(this.getFontRenderer(), p_190052_2_, (float)p_190052_3_ + 0.5F, (float)p_190052_5_ + 1.5F, (float)p_190052_7_ + 0.5F, 0, f, f1, false, false);
         }
     }
 }
