--- a/net/minecraft/client/renderer/tileentity/TileEntityRendererDispatcher.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntityRendererDispatcher.java
@@ -122,7 +122,7 @@
             OpenGlHelper.setLightmapTextureCoords(OpenGlHelper.lightmapTexUnit, (float)j, (float)k);
             GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
             BlockPos blockpos = p_180546_1_.func_174877_v();
-            this.func_178469_a(p_180546_1_, (double)blockpos.func_177958_n() - staticPlayerX, (double)blockpos.func_177956_o() - staticPlayerY, (double)blockpos.func_177952_p() - staticPlayerZ, p_180546_2_, p_180546_3_);
+            this.func_178469_a(p_180546_1_, (double)blockpos.getX() - staticPlayerX, (double)blockpos.getY() - staticPlayerY, (double)blockpos.getZ() - staticPlayerZ, p_180546_2_, p_180546_3_);
         }
     }
 
