--- a/net/minecraft/world/gen/structure/template/PlacementSettings.java
+++ b/net/minecraft/world/gen/structure/template/PlacementSettings.java
@@ -129,8 +129,8 @@
         }
         else
         {
-            int i = p_189947_1_.func_177958_n();
-            int j = p_189947_1_.func_177952_p();
+            int i = p_189947_1_.getX();
+            int j = p_189947_1_.getZ();
             return new Random((long)(i * i * 4987142 + i * 5947611) + (long)(j * j) * 4392871L + (long)(j * 389711) ^ 987234911L);
         }
     }
