--- a/net/minecraft/world/gen/structure/template/Template.java
+++ b/net/minecraft/world/gen/structure/template/Template.java
@@ -36,7 +36,7 @@
 {
     private final List<Template.BlockInfo> field_186270_a = Lists.<Template.BlockInfo>newArrayList();
     private final List<Template.EntityInfo> field_186271_b = Lists.<Template.EntityInfo>newArrayList();
-    private BlockPos field_186272_c = BlockPos.field_177992_a;
+    private BlockPos field_186272_c = BlockPos.createDefault;
     private String field_186273_d = "?";
 
     public BlockPos func_186259_a()
@@ -56,22 +56,22 @@
 
     public void func_186254_a(World p_186254_1_, BlockPos p_186254_2_, BlockPos p_186254_3_, boolean p_186254_4_, @Nullable Block p_186254_5_)
     {
-        if (p_186254_3_.func_177958_n() >= 1 && p_186254_3_.func_177956_o() >= 1 && p_186254_3_.func_177952_p() >= 1)
+        if (p_186254_3_.getX() >= 1 && p_186254_3_.getY() >= 1 && p_186254_3_.getZ() >= 1)
         {
             BlockPos blockpos = p_186254_2_.func_177971_a(p_186254_3_).func_177982_a(-1, -1, -1);
             List<Template.BlockInfo> list = Lists.<Template.BlockInfo>newArrayList();
             List<Template.BlockInfo> list1 = Lists.<Template.BlockInfo>newArrayList();
             List<Template.BlockInfo> list2 = Lists.<Template.BlockInfo>newArrayList();
-            BlockPos blockpos1 = new BlockPos(Math.min(p_186254_2_.func_177958_n(), blockpos.func_177958_n()), Math.min(p_186254_2_.func_177956_o(), blockpos.func_177956_o()), Math.min(p_186254_2_.func_177952_p(), blockpos.func_177952_p()));
-            BlockPos blockpos2 = new BlockPos(Math.max(p_186254_2_.func_177958_n(), blockpos.func_177958_n()), Math.max(p_186254_2_.func_177956_o(), blockpos.func_177956_o()), Math.max(p_186254_2_.func_177952_p(), blockpos.func_177952_p()));
+            BlockPos blockpos1 = new BlockPos(Math.min(p_186254_2_.getX(), blockpos.getX()), Math.min(p_186254_2_.getY(), blockpos.getY()), Math.min(p_186254_2_.getZ(), blockpos.getZ()));
+            BlockPos blockpos2 = new BlockPos(Math.max(p_186254_2_.getX(), blockpos.getX()), Math.max(p_186254_2_.getY(), blockpos.getY()), Math.max(p_186254_2_.getZ(), blockpos.getZ()));
             this.field_186272_c = p_186254_3_;
 
             for (BlockPos.MutableBlockPos blockpos$mutableblockpos : BlockPos.func_177975_b(blockpos1, blockpos2))
             {
                 BlockPos blockpos3 = blockpos$mutableblockpos.func_177973_b(blockpos1);
-                IBlockState iblockstate = p_186254_1_.func_180495_p(blockpos$mutableblockpos);
+                IBlockState iblockstate = p_186254_1_.getBlockState(blockpos$mutableblockpos);
 
-                if (p_186254_5_ == null || p_186254_5_ != iblockstate.func_177230_c())
+                if (p_186254_5_ == null || p_186254_5_ != iblockstate.getBlock())
                 {
                     TileEntity tileentity = p_186254_1_.func_175625_s(blockpos$mutableblockpos);
 
@@ -123,7 +123,7 @@
 
         for (Entity entity : list)
         {
-            Vec3d vec3d = new Vec3d(entity.posX - (double)p_186255_2_.func_177958_n(), entity.posY - (double)p_186255_2_.func_177956_o(), entity.posZ - (double)p_186255_2_.func_177952_p());
+            Vec3d vec3d = new Vec3d(entity.posX - (double)p_186255_2_.getX(), entity.posY - (double)p_186255_2_.getY(), entity.posZ - (double)p_186255_2_.getZ());
             NBTTagCompound nbttagcompound = new NBTTagCompound();
             entity.writeToNBTOptional(nbttagcompound);
             BlockPos blockpos;
@@ -154,7 +154,7 @@
             {
                 IBlockState iblockstate = template$blockinfo.field_186243_b;
 
-                if (iblockstate.func_177230_c() == Blocks.field_185779_df && template$blockinfo.field_186244_c != null)
+                if (iblockstate.getBlock() == Blocks.field_185779_df && template$blockinfo.field_186244_c != null)
                 {
                     TileEntityStructure.Mode tileentitystructure$mode = TileEntityStructure.Mode.valueOf(template$blockinfo.field_186244_c.getString("mode"));
 
@@ -199,7 +199,7 @@
 
     public void func_189960_a(World p_189960_1_, BlockPos p_189960_2_, @Nullable ITemplateProcessor p_189960_3_, PlacementSettings p_189960_4_, int p_189960_5_)
     {
-        if (!this.field_186270_a.isEmpty() && this.field_186272_c.func_177958_n() >= 1 && this.field_186272_c.func_177956_o() >= 1 && this.field_186272_c.func_177952_p() >= 1)
+        if (!this.field_186270_a.isEmpty() && this.field_186272_c.getX() >= 1 && this.field_186272_c.getY() >= 1 && this.field_186272_c.getZ() >= 1)
         {
             Block block = p_189960_4_.func_186219_f();
             StructureBoundingBox structureboundingbox = p_189960_4_.func_186213_g();
@@ -211,7 +211,7 @@
 
                 if (template$blockinfo1 != null)
                 {
-                    Block block1 = template$blockinfo1.field_186243_b.func_177230_c();
+                    Block block1 = template$blockinfo1.field_186243_b.getBlock();
 
                     if ((block == null || block != block1) && (!p_189960_4_.func_186227_h() || block1 != Blocks.field_185779_df) && (structureboundingbox == null || structureboundingbox.func_175898_b(blockpos)))
                     {
@@ -229,7 +229,7 @@
                                     ((IInventory)tileentity).func_174888_l();
                                 }
 
-                                p_189960_1_.func_180501_a(blockpos, Blocks.field_180401_cv.func_176223_P(), 4);
+                                p_189960_1_.func_180501_a(blockpos, Blocks.field_180401_cv.getBlockState(), 4);
                             }
                         }
 
@@ -239,9 +239,9 @@
 
                             if (tileentity2 != null)
                             {
-                                template$blockinfo1.field_186244_c.setInteger("x", blockpos.func_177958_n());
-                                template$blockinfo1.field_186244_c.setInteger("y", blockpos.func_177956_o());
-                                template$blockinfo1.field_186244_c.setInteger("z", blockpos.func_177952_p());
+                                template$blockinfo1.field_186244_c.setInteger("x", blockpos.getX());
+                                template$blockinfo1.field_186244_c.setInteger("y", blockpos.getY());
+                                template$blockinfo1.field_186244_c.setInteger("z", blockpos.getZ());
                                 tileentity2.readFromNBT(template$blockinfo1.field_186244_c);
                                 tileentity2.func_189668_a(p_189960_4_.func_186212_b());
                                 tileentity2.func_189667_a(p_189960_4_.func_186215_c());
@@ -253,13 +253,13 @@
 
             for (Template.BlockInfo template$blockinfo2 : this.field_186270_a)
             {
-                if (block == null || block != template$blockinfo2.field_186243_b.func_177230_c())
+                if (block == null || block != template$blockinfo2.field_186243_b.getBlock())
                 {
                     BlockPos blockpos1 = func_186266_a(p_189960_4_, template$blockinfo2.field_186242_a).func_177971_a(p_189960_2_);
 
                     if (structureboundingbox == null || structureboundingbox.func_175898_b(blockpos1))
                     {
-                        p_189960_1_.func_175722_b(blockpos1, template$blockinfo2.field_186243_b.func_177230_c());
+                        p_189960_1_.func_175722_b(blockpos1, template$blockinfo2.field_186243_b.getBlock());
 
                         if (template$blockinfo2.field_186244_c != null)
                         {
@@ -291,7 +291,7 @@
             {
                 NBTTagCompound nbttagcompound = template$entityinfo.field_186249_c;
                 Vec3d vec3d = func_186269_a(template$entityinfo.field_186247_a, p_186263_3_, p_186263_4_);
-                Vec3d vec3d1 = vec3d.addVector((double)p_186263_2_.func_177958_n(), (double)p_186263_2_.func_177956_o(), (double)p_186263_2_.func_177952_p());
+                Vec3d vec3d1 = vec3d.addVector((double)p_186263_2_.getX(), (double)p_186263_2_.getY(), (double)p_186263_2_.getZ());
                 NBTTagList nbttaglist = new NBTTagList();
                 nbttaglist.appendTag(new NBTTagDouble(vec3d1.xCoord));
                 nbttaglist.appendTag(new NBTTagDouble(vec3d1.yCoord));
@@ -326,7 +326,7 @@
         {
             case COUNTERCLOCKWISE_90:
             case CLOCKWISE_90:
-                return new BlockPos(this.field_186272_c.func_177952_p(), this.field_186272_c.func_177956_o(), this.field_186272_c.func_177958_n());
+                return new BlockPos(this.field_186272_c.getZ(), this.field_186272_c.getY(), this.field_186272_c.getX());
 
             default:
                 return this.field_186272_c;
@@ -335,9 +335,9 @@
 
     private static BlockPos func_186268_a(BlockPos p_186268_0_, Mirror p_186268_1_, Rotation p_186268_2_)
     {
-        int i = p_186268_0_.func_177958_n();
-        int j = p_186268_0_.func_177956_o();
-        int k = p_186268_0_.func_177952_p();
+        int i = p_186268_0_.getX();
+        int j = p_186268_0_.getY();
+        int k = p_186268_0_.getZ();
         boolean flag = true;
 
         switch (p_186268_1_)
@@ -409,8 +409,8 @@
 
     public BlockPos func_189961_a(BlockPos p_189961_1_, Mirror p_189961_2_, Rotation p_189961_3_)
     {
-        int i = this.func_186259_a().func_177958_n() - 1;
-        int j = this.func_186259_a().func_177952_p() - 1;
+        int i = this.func_186259_a().getX() - 1;
+        int j = this.func_186259_a().getZ() - 1;
         int k = p_189961_2_ == Mirror.FRONT_BACK ? i : 0;
         int l = p_189961_2_ == Mirror.LEFT_RIGHT ? j : 0;
         BlockPos blockpos = p_189961_1_;
@@ -444,7 +444,7 @@
         for (Template.BlockInfo template$blockinfo : this.field_186270_a)
         {
             NBTTagCompound nbttagcompound = new NBTTagCompound();
-            nbttagcompound.setTag("pos", this.func_186267_a(new int[] {template$blockinfo.field_186242_a.func_177958_n(), template$blockinfo.field_186242_a.func_177956_o(), template$blockinfo.field_186242_a.func_177952_p()}));
+            nbttagcompound.setTag("pos", this.func_186267_a(new int[] {template$blockinfo.field_186242_a.getX(), template$blockinfo.field_186242_a.getY(), template$blockinfo.field_186242_a.getZ()}));
             nbttagcompound.setInteger("state", template$basicpalette.func_189954_a(template$blockinfo.field_186243_b));
 
             if (template$blockinfo.field_186244_c != null)
@@ -461,7 +461,7 @@
         {
             NBTTagCompound nbttagcompound1 = new NBTTagCompound();
             nbttagcompound1.setTag("pos", this.func_186264_a(new double[] {template$entityinfo.field_186247_a.xCoord, template$entityinfo.field_186247_a.yCoord, template$entityinfo.field_186247_a.zCoord}));
-            nbttagcompound1.setTag("blockPos", this.func_186267_a(new int[] {template$entityinfo.field_186248_b.func_177958_n(), template$entityinfo.field_186248_b.func_177956_o(), template$entityinfo.field_186248_b.func_177952_p()}));
+            nbttagcompound1.setTag("blockPos", this.func_186267_a(new int[] {template$entityinfo.field_186248_b.getX(), template$entityinfo.field_186248_b.getY(), template$entityinfo.field_186248_b.getZ()}));
 
             if (template$entityinfo.field_186249_c != null)
             {
@@ -481,7 +481,7 @@
         p_189552_1_.setTag("palette", nbttaglist2);
         p_189552_1_.setTag("blocks", nbttaglist);
         p_189552_1_.setTag("entities", nbttaglist1);
-        p_189552_1_.setTag("size", this.func_186267_a(new int[] {this.field_186272_c.func_177958_n(), this.field_186272_c.func_177956_o(), this.field_186272_c.func_177952_p()}));
+        p_189552_1_.setTag("size", this.func_186267_a(new int[] {this.field_186272_c.getX(), this.field_186272_c.getY(), this.field_186272_c.getZ()}));
         p_189552_1_.setInteger("version", 1);
         p_189552_1_.setString("author", this.field_186273_d);
         return p_189552_1_;
@@ -568,7 +568,7 @@
 
     static class BasicPalette implements Iterable<IBlockState>
     {
-        public static final IBlockState field_189957_a = Blocks.air.func_176223_P();
+        public static final IBlockState field_189957_a = Blocks.air.getBlockState();
         final ObjectIntIdentityMap<IBlockState> field_189958_b;
         private int field_189959_c;
 
