--- a/net/minecraft/tileentity/TileEntityHopper.java
+++ b/net/minecraft/tileentity/TileEntityHopper.java
@@ -192,7 +192,7 @@
      */
     public boolean isUseableByPlayer(EntityPlayer player)
     {
-        return this.worldObj.func_175625_s(this.field_174879_c) != this ? false : player.getDistanceSq((double)this.field_174879_c.func_177958_n() + 0.5D, (double)this.field_174879_c.func_177956_o() + 0.5D, (double)this.field_174879_c.func_177952_p() + 0.5D) <= 64.0D;
+        return this.worldObj.func_175625_s(this.field_174879_c) != this ? false : player.getDistanceSq((double)this.field_174879_c.getX() + 0.5D, (double)this.field_174879_c.getY() + 0.5D, (double)this.field_174879_c.getZ() + 0.5D) <= 64.0D;
     }
 
     public void func_174889_b(EntityPlayer p_174889_1_)
@@ -602,7 +602,7 @@
         int j = MathHelper.floor_double(p_145893_3_);
         int k = MathHelper.floor_double(p_145893_5_);
         BlockPos blockpos = new BlockPos(i, j, k);
-        Block block = p_145893_0_.func_180495_p(blockpos).func_177230_c();
+        Block block = p_145893_0_.getBlockState(blockpos).getBlock();
 
         if (block.hasTileEntity())
         {
@@ -642,7 +642,7 @@
      */
     public double getXPos()
     {
-        return (double)this.field_174879_c.func_177958_n() + 0.5D;
+        return (double)this.field_174879_c.getX() + 0.5D;
     }
 
     /**
@@ -650,7 +650,7 @@
      */
     public double getYPos()
     {
-        return (double)this.field_174879_c.func_177956_o() + 0.5D;
+        return (double)this.field_174879_c.getY() + 0.5D;
     }
 
     /**
@@ -658,7 +658,7 @@
      */
     public double getZPos()
     {
-        return (double)this.field_174879_c.func_177952_p() + 0.5D;
+        return (double)this.field_174879_c.getZ() + 0.5D;
     }
 
     public void setTransferCooldown(int p_145896_1_)
