--- a/net/minecraft/tileentity/TileEntityEnchantmentTable.java
+++ b/net/minecraft/tileentity/TileEntityEnchantmentTable.java
@@ -57,12 +57,12 @@
     {
         this.bookSpreadPrev = this.bookSpread;
         this.bookRotationPrev = this.bookRotation;
-        EntityPlayer entityplayer = this.worldObj.func_184137_a((double)((float)this.field_174879_c.func_177958_n() + 0.5F), (double)((float)this.field_174879_c.func_177956_o() + 0.5F), (double)((float)this.field_174879_c.func_177952_p() + 0.5F), 3.0D, false);
+        EntityPlayer entityplayer = this.worldObj.func_184137_a((double)((float)this.field_174879_c.getX() + 0.5F), (double)((float)this.field_174879_c.getY() + 0.5F), (double)((float)this.field_174879_c.getZ() + 0.5F), 3.0D, false);
 
         if (entityplayer != null)
         {
-            double d0 = entityplayer.posX - (double)((float)this.field_174879_c.func_177958_n() + 0.5F);
-            double d1 = entityplayer.posZ - (double)((float)this.field_174879_c.func_177952_p() + 0.5F);
+            double d0 = entityplayer.posX - (double)((float)this.field_174879_c.getX() + 0.5F);
+            double d1 = entityplayer.posZ - (double)((float)this.field_174879_c.getZ() + 0.5F);
             this.field_145924_q = (float)MathHelper.func_181159_b(d1, d0);
             this.bookSpread += 0.1F;
 
