--- a/net/minecraft/tileentity/TileEntityBrewingStand.java
+++ b/net/minecraft/tileentity/TileEntityBrewingStand.java
@@ -135,9 +135,9 @@
             if (!Arrays.equals(aboolean, this.filledSlots))
             {
                 this.filledSlots = aboolean;
-                IBlockState iblockstate = this.worldObj.func_180495_p(this.func_174877_v());
+                IBlockState iblockstate = this.worldObj.getBlockState(this.func_174877_v());
 
-                if (!(iblockstate.func_177230_c() instanceof BlockBrewingStand))
+                if (!(iblockstate.getBlock() instanceof BlockBrewingStand))
                 {
                     return;
                 }
@@ -220,7 +220,7 @@
             }
             else
             {
-                InventoryHelper.func_180173_a(this.worldObj, (double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p(), itemstack1);
+                InventoryHelper.func_180173_a(this.worldObj, (double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), itemstack1);
             }
         }
 
@@ -350,7 +350,7 @@
      */
     public boolean isUseableByPlayer(EntityPlayer player)
     {
-        return this.worldObj.func_175625_s(this.field_174879_c) != this ? false : player.getDistanceSq((double)this.field_174879_c.func_177958_n() + 0.5D, (double)this.field_174879_c.func_177956_o() + 0.5D, (double)this.field_174879_c.func_177952_p() + 0.5D) <= 64.0D;
+        return this.worldObj.func_175625_s(this.field_174879_c) != this ? false : player.getDistanceSq((double)this.field_174879_c.getX() + 0.5D, (double)this.field_174879_c.getY() + 0.5D, (double)this.field_174879_c.getZ() + 0.5D) <= 64.0D;
     }
 
     public void func_174889_b(EntityPlayer p_174889_1_)
