--- a/net/minecraft/tileentity/TileEntityCommandBlock.java
+++ b/net/minecraft/tileentity/TileEntityCommandBlock.java
@@ -29,7 +29,7 @@
         }
         public Vec3d func_174791_d()
         {
-            return new Vec3d((double)TileEntityCommandBlock.this.field_174879_c.func_177958_n() + 0.5D, (double)TileEntityCommandBlock.this.field_174879_c.func_177956_o() + 0.5D, (double)TileEntityCommandBlock.this.field_174879_c.func_177952_p() + 0.5D);
+            return new Vec3d((double)TileEntityCommandBlock.this.field_174879_c.getX() + 0.5D, (double)TileEntityCommandBlock.this.field_174879_c.getY() + 0.5D, (double)TileEntityCommandBlock.this.field_174879_c.getZ() + 0.5D);
         }
         public World getEntityWorld()
         {
@@ -42,7 +42,7 @@
         }
         public void func_145756_e()
         {
-            IBlockState iblockstate = TileEntityCommandBlock.this.worldObj.func_180495_p(TileEntityCommandBlock.this.field_174879_c);
+            IBlockState iblockstate = TileEntityCommandBlock.this.worldObj.getBlockState(TileEntityCommandBlock.this.field_174879_c);
             TileEntityCommandBlock.this.getWorld().func_184138_a(TileEntityCommandBlock.this.field_174879_c, iblockstate, iblockstate, 3);
         }
         public int func_145751_f()
@@ -51,9 +51,9 @@
         }
         public void func_145757_a(ByteBuf p_145757_1_)
         {
-            p_145757_1_.writeInt(TileEntityCommandBlock.this.field_174879_c.func_177958_n());
-            p_145757_1_.writeInt(TileEntityCommandBlock.this.field_174879_c.func_177956_o());
-            p_145757_1_.writeInt(TileEntityCommandBlock.this.field_174879_c.func_177952_p());
+            p_145757_1_.writeInt(TileEntityCommandBlock.this.field_174879_c.getX());
+            p_145757_1_.writeInt(TileEntityCommandBlock.this.field_174879_c.getY());
+            p_145757_1_.writeInt(TileEntityCommandBlock.this.field_174879_c.getZ());
         }
         public Entity func_174793_f()
         {
@@ -142,7 +142,7 @@
             {
                 BlockPos blockpos = this.func_174877_v();
                 BlockCommandBlock blockcommandblock = (BlockCommandBlock)block;
-                this.field_184261_g = !this.func_184258_j() || blockcommandblock.func_185562_e(this.worldObj, blockpos, this.worldObj.func_180495_p(blockpos));
+                this.field_184261_g = !this.func_184258_j() || blockcommandblock.func_185562_e(this.worldObj, blockpos, this.worldObj.getBlockState(blockpos));
                 this.worldObj.func_175684_a(blockpos, block, block.tickRate(this.worldObj));
 
                 if (this.field_184261_g)
@@ -181,8 +181,8 @@
 
     public boolean func_184258_j()
     {
-        IBlockState iblockstate = this.worldObj.func_180495_p(this.func_174877_v());
-        return iblockstate.func_177230_c() instanceof BlockCommandBlock ? ((Boolean)iblockstate.func_177229_b(BlockCommandBlock.field_185565_b)).booleanValue() : false;
+        IBlockState iblockstate = this.worldObj.getBlockState(this.func_174877_v());
+        return iblockstate.getBlock() instanceof BlockCommandBlock ? ((Boolean)iblockstate.func_177229_b(BlockCommandBlock.field_185565_b)).booleanValue() : false;
     }
 
     /**
