--- a/net/minecraft/tileentity/TileEntityEndGateway.java
+++ b/net/minecraft/tileentity/TileEntityEndGateway.java
@@ -164,7 +164,7 @@
             if (this.field_184317_h != null)
             {
                 BlockPos blockpos = this.field_184318_i ? this.field_184317_h : this.func_184303_j();
-                p_184306_1_.setPositionAndUpdate((double)blockpos.func_177958_n() + 0.5D, (double)blockpos.func_177956_o() + 0.5D, (double)blockpos.func_177952_p() + 0.5D);
+                p_184306_1_.setPositionAndUpdate((double)blockpos.getX() + 0.5D, (double)blockpos.getY() + 0.5D, (double)blockpos.getZ() + 0.5D);
             }
 
             this.func_184300_h();
@@ -175,20 +175,20 @@
     {
         BlockPos blockpos = func_184308_a(this.worldObj, this.field_184317_h, 5, false);
         field_184314_a.debug("Best exit position for portal at {} is {}", new Object[] {this.field_184317_h, blockpos});
-        return blockpos.func_177984_a();
+        return blockpos.getBlockPosAbove();
     }
 
     private void func_184311_k()
     {
-        Vec3d vec3d = (new Vec3d((double)this.func_174877_v().func_177958_n(), 0.0D, (double)this.func_174877_v().func_177952_p())).normalize();
-        Vec3d vec3d1 = vec3d.func_186678_a(1024.0D);
+        Vec3d vec3d = (new Vec3d((double)this.func_174877_v().getX(), 0.0D, (double)this.func_174877_v().getZ())).normalize();
+        Vec3d vec3d1 = vec3d.multiply(1024.0D);
 
-        for (int i = 16; func_184301_a(this.worldObj, vec3d1).getTopFilledSegment() > 0 && i-- > 0; vec3d1 = vec3d1.func_178787_e(vec3d.func_186678_a(-16.0D)))
+        for (int i = 16; func_184301_a(this.worldObj, vec3d1).getTopFilledSegment() > 0 && i-- > 0; vec3d1 = vec3d1.addVector(vec3d.multiply(-16.0D)))
         {
             field_184314_a.debug("Skipping backwards past nonempty chunk at {}", new Object[] {vec3d1});
         }
 
-        for (int j = 16; func_184301_a(this.worldObj, vec3d1).getTopFilledSegment() == 0 && j-- > 0; vec3d1 = vec3d1.func_178787_e(vec3d.func_186678_a(16.0D)))
+        for (int j = 16; func_184301_a(this.worldObj, vec3d1).getTopFilledSegment() == 0 && j-- > 0; vec3d1 = vec3d1.addVector(vec3d.multiply(16.0D)))
         {
             field_184314_a.debug("Skipping forward past empty chunk at {}", new Object[] {vec3d1});
         }
@@ -210,7 +210,7 @@
 
         this.field_184317_h = func_184308_a(this.worldObj, this.field_184317_h, 16, true);
         field_184314_a.debug("Creating portal at {}", new Object[] {this.field_184317_h});
-        this.field_184317_h = this.field_184317_h.func_177981_b(10);
+        this.field_184317_h = this.field_184317_h.getBlockPosAbove(10);
         this.func_184312_b(this.field_184317_h);
         this.markDirty();
     }
@@ -225,12 +225,12 @@
             {
                 if (i != 0 || j != 0 || p_184308_3_)
                 {
-                    for (int k = 255; k > (blockpos == null ? 0 : blockpos.func_177956_o()); --k)
+                    for (int k = 255; k > (blockpos == null ? 0 : blockpos.getY()); --k)
                     {
-                        BlockPos blockpos1 = new BlockPos(p_184308_1_.func_177958_n() + i, k, p_184308_1_.func_177952_p() + j);
-                        IBlockState iblockstate = p_184308_0_.func_180495_p(blockpos1);
+                        BlockPos blockpos1 = new BlockPos(p_184308_1_.getX() + i, k, p_184308_1_.getZ() + j);
+                        IBlockState iblockstate = p_184308_0_.getBlockState(blockpos1);
 
-                        if (iblockstate.func_185898_k() && (p_184308_3_ || iblockstate.func_177230_c() != Blocks.bedrock))
+                        if (iblockstate.func_185898_k() && (p_184308_3_ || iblockstate.getBlock() != Blocks.bedrock))
                         {
                             blockpos = blockpos1;
                             break;
@@ -257,11 +257,11 @@
         BlockPos blockpos2 = null;
         double d0 = 0.0D;
 
-        for (BlockPos blockpos3 : BlockPos.func_177980_a(blockpos, blockpos1))
+        for (BlockPos blockpos3 : BlockPos.getBlocksBetween(blockpos, blockpos1))
         {
-            IBlockState iblockstate = p_184307_0_.func_177435_g(blockpos3);
+            IBlockState iblockstate = p_184307_0_.getBlockState(blockpos3);
 
-            if (iblockstate.func_177230_c() == Blocks.end_stone && !p_184307_0_.func_177435_g(blockpos3.func_177981_b(1)).func_185898_k() && !p_184307_0_.func_177435_g(blockpos3.func_177981_b(2)).func_185898_k())
+            if (iblockstate.getBlock() == Blocks.end_stone && !p_184307_0_.getBlockState(blockpos3.getBlockPosAbove(1)).func_185898_k() && !p_184307_0_.getBlockState(blockpos3.getBlockPosAbove(2)).func_185898_k())
             {
                 double d1 = blockpos3.func_177957_d(0.0D, 0.0D, 0.0D);
 
@@ -295,7 +295,7 @@
 
     public boolean func_184313_a(EnumFacing p_184313_1_)
     {
-        return this.getBlockType().func_176223_P().func_185894_c(this.worldObj, this.func_174877_v(), p_184313_1_);
+        return this.getBlockType().getBlockState().func_185894_c(this.worldObj, this.func_174877_v(), p_184313_1_);
     }
 
     public int func_184304_i()
