--- a/net/minecraft/tileentity/TileEntityMobSpawner.java
+++ b/net/minecraft/tileentity/TileEntityMobSpawner.java
@@ -37,7 +37,7 @@
 
             if (this.getSpawnerWorld() != null)
             {
-                IBlockState iblockstate = this.getSpawnerWorld().func_180495_p(this.func_177221_b());
+                IBlockState iblockstate = this.getSpawnerWorld().getBlockState(this.func_177221_b());
                 this.getSpawnerWorld().func_184138_a(TileEntityMobSpawner.this.field_174879_c, iblockstate, iblockstate, 4);
             }
         }
