--- a/net/minecraft/tileentity/TileEntityNote.java
+++ b/net/minecraft/tileentity/TileEntityNote.java
@@ -42,9 +42,9 @@
 
     public void func_175108_a(World p_175108_1_, BlockPos p_175108_2_)
     {
-        if (p_175108_1_.func_180495_p(p_175108_2_.func_177984_a()).func_185904_a() == Material.air)
+        if (p_175108_1_.getBlockState(p_175108_2_.getBlockPosAbove()).getMaterial() == Material.air)
         {
-            Material material = p_175108_1_.func_180495_p(p_175108_2_.func_177977_b()).func_185904_a();
+            Material material = p_175108_1_.getBlockState(p_175108_2_.getBlockPosBelow()).getMaterial();
             int i = 0;
 
             if (material == Material.rock)
