--- a/net/minecraft/tileentity/TileEntity.java
+++ b/net/minecraft/tileentity/TileEntity.java
@@ -27,7 +27,7 @@
 
     /** the instance of the world the tile entity is in. */
     protected World worldObj;
-    protected BlockPos field_174879_c = BlockPos.field_177992_a;
+    protected BlockPos field_174879_c = BlockPos.createDefault;
     protected boolean tileEntityInvalid;
     private int blockMetadata = -1;
 
@@ -95,9 +95,9 @@
         else
         {
             p_189516_1_.setString("id", s);
-            p_189516_1_.setInteger("x", this.field_174879_c.func_177958_n());
-            p_189516_1_.setInteger("y", this.field_174879_c.func_177956_o());
-            p_189516_1_.setInteger("z", this.field_174879_c.func_177952_p());
+            p_189516_1_.setInteger("x", this.field_174879_c.getX());
+            p_189516_1_.setInteger("y", this.field_174879_c.getY());
+            p_189516_1_.setInteger("z", this.field_174879_c.getZ());
             return p_189516_1_;
         }
     }
@@ -150,8 +150,8 @@
     {
         if (this.blockMetadata == -1)
         {
-            IBlockState iblockstate = this.worldObj.func_180495_p(this.field_174879_c);
-            this.blockMetadata = iblockstate.func_177230_c().func_176201_c(iblockstate);
+            IBlockState iblockstate = this.worldObj.getBlockState(this.field_174879_c);
+            this.blockMetadata = iblockstate.getBlock().func_176201_c(iblockstate);
         }
 
         return this.blockMetadata;
@@ -165,8 +165,8 @@
     {
         if (this.worldObj != null)
         {
-            IBlockState iblockstate = this.worldObj.func_180495_p(this.field_174879_c);
-            this.blockMetadata = iblockstate.func_177230_c().func_176201_c(iblockstate);
+            IBlockState iblockstate = this.worldObj.getBlockState(this.field_174879_c);
+            this.blockMetadata = iblockstate.getBlock().func_176201_c(iblockstate);
             this.worldObj.func_175646_b(this.field_174879_c, this);
 
             if (this.getBlockType() != Blocks.air)
@@ -181,9 +181,9 @@
      */
     public double getDistanceSq(double p_145835_1_, double p_145835_3_, double p_145835_5_)
     {
-        double d0 = (double)this.field_174879_c.func_177958_n() + 0.5D - p_145835_1_;
-        double d1 = (double)this.field_174879_c.func_177956_o() + 0.5D - p_145835_3_;
-        double d2 = (double)this.field_174879_c.func_177952_p() + 0.5D - p_145835_5_;
+        double d0 = (double)this.field_174879_c.getX() + 0.5D - p_145835_1_;
+        double d1 = (double)this.field_174879_c.getY() + 0.5D - p_145835_3_;
+        double d2 = (double)this.field_174879_c.getZ() + 0.5D - p_145835_5_;
         return d0 * d0 + d1 * d1 + d2 * d2;
     }
 
@@ -204,7 +204,7 @@
     {
         if (this.blockType == null && this.worldObj != null)
         {
-            this.blockType = this.worldObj.func_180495_p(this.field_174879_c).func_177230_c();
+            this.blockType = this.worldObj.getBlockState(this.field_174879_c).getBlock();
         }
 
         return this.blockType;
@@ -270,7 +270,7 @@
             {
                 public String call() throws Exception
                 {
-                    int i = Block.getIdFromBlock(TileEntity.this.worldObj.func_180495_p(TileEntity.this.field_174879_c).func_177230_c());
+                    int i = Block.getIdFromBlock(TileEntity.this.worldObj.getBlockState(TileEntity.this.field_174879_c).getBlock());
 
                     try
                     {
@@ -286,8 +286,8 @@
             {
                 public String call() throws Exception
                 {
-                    IBlockState iblockstate = TileEntity.this.worldObj.func_180495_p(TileEntity.this.field_174879_c);
-                    int i = iblockstate.func_177230_c().func_176201_c(iblockstate);
+                    IBlockState iblockstate = TileEntity.this.worldObj.getBlockState(TileEntity.this.field_174879_c);
+                    int i = iblockstate.getBlock().func_176201_c(iblockstate);
 
                     if (i < 0)
                     {
