--- a/net/minecraft/tileentity/TileEntityPiston.java
+++ b/net/minecraft/tileentity/TileEntityPiston.java
@@ -110,7 +110,7 @@
 
     private void func_184322_i()
     {
-        AxisAlignedBB axisalignedbb = this.func_184321_a(this.worldObj, this.field_174879_c).func_186670_a(this.field_174879_c);
+        AxisAlignedBB axisalignedbb = this.func_184321_a(this.worldObj, this.field_174879_c).offset(this.field_174879_c);
         List<Entity> list = this.worldObj.getEntitiesWithinAABBExcludingEntity((Entity)null, axisalignedbb);
 
         if (!list.isEmpty())
@@ -123,7 +123,7 @@
 
                 if (entity.func_184192_z() != EnumPushReaction.IGNORE)
                 {
-                    if (this.field_174932_a.func_177230_c() == Blocks.field_180399_cE)
+                    if (this.field_174932_a.getBlock() == Blocks.field_180399_cE)
                     {
                         switch (enumfacing.func_176740_k())
                         {
@@ -204,10 +204,10 @@
             this.worldObj.func_175713_t(this.field_174879_c);
             this.invalidate();
 
-            if (this.worldObj.func_180495_p(this.field_174879_c).func_177230_c() == Blocks.field_180384_M)
+            if (this.worldObj.getBlockState(this.field_174879_c).getBlock() == Blocks.field_180384_M)
             {
                 this.worldObj.func_180501_a(this.field_174879_c, this.field_174932_a, 3);
-                this.worldObj.func_180496_d(this.field_174879_c, this.field_174932_a.func_177230_c());
+                this.worldObj.func_180496_d(this.field_174879_c, this.field_174932_a.getBlock());
             }
         }
     }
@@ -225,10 +225,10 @@
             this.worldObj.func_175713_t(this.field_174879_c);
             this.invalidate();
 
-            if (this.worldObj.func_180495_p(this.field_174879_c).func_177230_c() == Blocks.field_180384_M)
+            if (this.worldObj.getBlockState(this.field_174879_c).getBlock() == Blocks.field_180384_M)
             {
                 this.worldObj.func_180501_a(this.field_174879_c, this.field_174932_a, 3);
-                this.worldObj.func_180496_d(this.field_174879_c, this.field_174932_a.func_177230_c());
+                this.worldObj.func_180496_d(this.field_174879_c, this.field_174932_a.getBlock());
             }
         }
         else
@@ -261,8 +261,8 @@
     public NBTTagCompound func_189515_b(NBTTagCompound p_189515_1_)
     {
         super.func_189515_b(p_189515_1_);
-        p_189515_1_.setInteger("blockId", Block.getIdFromBlock(this.field_174932_a.func_177230_c()));
-        p_189515_1_.setInteger("blockData", this.field_174932_a.func_177230_c().func_176201_c(this.field_174932_a));
+        p_189515_1_.setInteger("blockId", Block.getIdFromBlock(this.field_174932_a.getBlock()));
+        p_189515_1_.setInteger("blockData", this.field_174932_a.getBlock().func_176201_c(this.field_174932_a));
         p_189515_1_.setInteger("facing", this.field_174931_f.func_176745_a());
         p_189515_1_.setFloat("progress", this.lastProgress);
         p_189515_1_.setBoolean("extending", this.extending);
