--- a/net/minecraft/src/ConnectedParser.java
+++ b/net/minecraft/src/ConnectedParser.java
@@ -73,7 +73,7 @@
 
                 if (amatchblock != null)
                 {
-                    list.addAll(Arrays.<E>asList(amatchblock));
+                    list.addAll(Arrays.asList(amatchblock));
                 }
             }
 
@@ -403,7 +403,7 @@
         return comparable;
     }
 
-    public static Comparable getPropertyValue(String p_getPropertyValue_0_, Collection p_getPropertyValue_1_)
+    public static Comparable getPropertyValue(String p_getPropertyValue_0_, Collection<Comparable> p_getPropertyValue_1_)
     {
         for (Comparable comparable : p_getPropertyValue_1_)
         {
