--- a/net/minecraft/src/ConnectedParser.java
+++ b/net/minecraft/src/ConnectedParser.java
@@ -74,7 +74,7 @@
                     return NO_MATCH_BLOCKS;
                 }
 
-                list.addAll(Arrays.<E>asList(amatchblock));
+                list.addAll(Arrays.asList(amatchblock));
             }
 
             MatchBlock[] amatchblock1 = (MatchBlock[])((MatchBlock[])list.toArray(new MatchBlock[list.size()]));
@@ -382,7 +382,7 @@
         return comparable;
     }
 
-    public static Comparable getPropertyValue(String p_getPropertyValue_0_, Collection p_getPropertyValue_1_)
+    public static Comparable getPropertyValue(String p_getPropertyValue_0_, Collection<Comparable> p_getPropertyValue_1_)
     {
         for (Comparable comparable : p_getPropertyValue_1_)
         {
