--- a/net/minecraft/client/renderer/debug/DebugRendererWater.java
+++ b/net/minecraft/client/renderer/debug/DebugRendererWater.java
@@ -39,27 +39,27 @@
         GlStateManager.func_179090_x();
         GlStateManager.func_187441_d(6.0F);
 
-        for (BlockPos blockpos1 : BlockPos.PooledMutableBlockPos.func_177980_a(blockpos.func_177982_a(-10, -10, -10), blockpos.func_177982_a(10, 10, 10)))
+        for (BlockPos blockpos1 : BlockPos.PooledMutableBlockPos.getBlocksBetween(blockpos.func_177982_a(-10, -10, -10), blockpos.func_177982_a(10, 10, 10)))
         {
-            IBlockState iblockstate = world.func_180495_p(blockpos1);
+            IBlockState iblockstate = world.getBlockState(blockpos1);
 
-            if (iblockstate.func_177230_c() == Blocks.water || iblockstate.func_177230_c() == Blocks.flowing_water)
+            if (iblockstate.getBlock() == Blocks.water || iblockstate.getBlock() == Blocks.flowing_water)
             {
                 double d0 = (double)EntityBoat.func_184452_b(iblockstate, world, blockpos1);
-                RenderGlobal.func_189696_b((new AxisAlignedBB((double)((float)blockpos1.func_177958_n() + 0.01F), (double)((float)blockpos1.func_177956_o() + 0.01F), (double)((float)blockpos1.func_177952_p() + 0.01F), (double)((float)blockpos1.func_177958_n() + 0.99F), d0, (double)((float)blockpos1.func_177952_p() + 0.99F))).offset(-this.field_190063_c, -this.field_190064_d, -this.field_190065_e), 1.0F, 1.0F, 1.0F, 0.2F);
+                RenderGlobal.func_189696_b((new AxisAlignedBB((double)((float)blockpos1.getX() + 0.01F), (double)((float)blockpos1.getY() + 0.01F), (double)((float)blockpos1.getZ() + 0.01F), (double)((float)blockpos1.getX() + 0.99F), d0, (double)((float)blockpos1.getZ() + 0.99F))).offset(-this.field_190063_c, -this.field_190064_d, -this.field_190065_e), 1.0F, 1.0F, 1.0F, 0.2F);
             }
         }
 
-        for (BlockPos blockpos2 : BlockPos.PooledMutableBlockPos.func_177980_a(blockpos.func_177982_a(-10, -10, -10), blockpos.func_177982_a(10, 10, 10)))
+        for (BlockPos blockpos2 : BlockPos.PooledMutableBlockPos.getBlocksBetween(blockpos.func_177982_a(-10, -10, -10), blockpos.func_177982_a(10, 10, 10)))
         {
-            IBlockState iblockstate1 = world.func_180495_p(blockpos2);
+            IBlockState iblockstate1 = world.getBlockState(blockpos2);
 
-            if (iblockstate1.func_177230_c() == Blocks.water || iblockstate1.func_177230_c() == Blocks.flowing_water)
+            if (iblockstate1.getBlock() == Blocks.water || iblockstate1.getBlock() == Blocks.flowing_water)
             {
                 Integer integer = (Integer)iblockstate1.func_177229_b(BlockLiquid.field_176367_b);
                 double d1 = integer.intValue() > 7 ? 0.9D : 1.0D - 0.11D * (double)integer.intValue();
-                String s = iblockstate1.func_177230_c() == Blocks.flowing_water ? "f" : "s";
-                DebugRenderer.func_190076_a(s + " " + integer, (double)blockpos2.func_177958_n() + 0.5D, (double)blockpos2.func_177956_o() + d1, (double)blockpos2.func_177952_p() + 0.5D, p_190060_1_, -16777216);
+                String s = iblockstate1.getBlock() == Blocks.flowing_water ? "f" : "s";
+                DebugRenderer.func_190076_a(s + " " + integer, (double)blockpos2.getX() + 0.5D, (double)blockpos2.getY() + d1, (double)blockpos2.getZ() + 0.5D, p_190060_1_, -16777216);
             }
         }
 
