--- a/net/minecraft/entity/EntityLiving.java
+++ b/net/minecraft/entity/EntityLiving.java
@@ -237,7 +237,7 @@
 
         if (soundevent != null)
         {
-            this.func_184185_a(soundevent, this.getSoundVolume(), this.getSoundPitch());
+            this.playSound(soundevent, this.getSoundVolume(), this.getSoundPitch());
         }
     }
 
@@ -315,7 +315,7 @@
                 double d1 = this.rand.nextGaussian() * 0.02D;
                 double d2 = this.rand.nextGaussian() * 0.02D;
                 double d3 = 10.0D;
-                this.worldObj.func_175688_a(EnumParticleTypes.EXPLOSION_NORMAL, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width - d0 * 10.0D, this.posY + (double)(this.rand.nextFloat() * this.height) - d1 * 10.0D, this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width - d2 * 10.0D, d0, d1, d2, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.EXPLOSION_NORMAL, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width - d0 * 10.0D, this.posY + (double)(this.rand.nextFloat() * this.height) - d1 * 10.0D, this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width - d2 * 10.0D, d0, d1, d2, new int[0]);
             }
         }
         else
@@ -350,7 +350,7 @@
             if (this.ticksExisted % 5 == 0)
             {
                 boolean flag = !(this.func_184179_bs() instanceof EntityLiving);
-                boolean flag1 = !(this.func_184187_bx() instanceof EntityBoat);
+                boolean flag1 = !(this.getRidingEntity() instanceof EntityBoat);
                 this.tasks.func_188527_a(1, flag);
                 this.tasks.func_188527_a(4, flag && flag1);
                 this.tasks.func_188527_a(2, flag);
@@ -477,9 +477,9 @@
             else if (this.leashedToEntity instanceof EntityHanging)
             {
                 BlockPos blockpos = ((EntityHanging)this.leashedToEntity).func_174857_n();
-                nbttagcompound2.setInteger("X", blockpos.func_177958_n());
-                nbttagcompound2.setInteger("Y", blockpos.func_177956_o());
-                nbttagcompound2.setInteger("Z", blockpos.func_177952_p());
+                nbttagcompound2.setInteger("X", blockpos.getX());
+                nbttagcompound2.setInteger("Y", blockpos.getY());
+                nbttagcompound2.setInteger("Z", blockpos.getZ());
             }
 
             tagCompound.setTag("Leash", nbttagcompound2);
@@ -661,7 +661,7 @@
         ItemStack itemstack = p_175445_1_.getEntityItem();
         EntityEquipmentSlot entityequipmentslot = func_184640_d(itemstack);
         boolean flag = true;
-        ItemStack itemstack1 = this.func_184582_a(entityequipmentslot);
+        ItemStack itemstack1 = this.getEquippedItemStack(entityequipmentslot);
 
         if (itemstack1 != null)
         {
@@ -751,7 +751,7 @@
                 }
             }
 
-            this.func_184201_a(entityequipmentslot, itemstack);
+            this.equipItemInSlot(entityequipmentslot, itemstack);
 
             switch (entityequipmentslot.func_188453_a())
             {
@@ -841,9 +841,9 @@
         this.updateAITasks();
         this.worldObj.theProfiler.endSection();
 
-        if (this.func_184218_aH() && this.func_184187_bx() instanceof EntityLiving)
+        if (this.isRiding() && this.getRidingEntity() instanceof EntityLiving)
         {
-            EntityLiving entityliving = (EntityLiving)this.func_184187_bx();
+            EntityLiving entityliving = (EntityLiving)this.getRidingEntity();
             entityliving.getNavigator().setPath(this.getNavigator().getPath(), 1.5D);
             entityliving.getMoveHelper().func_188487_a(this.getMoveHelper());
         }
@@ -928,7 +928,7 @@
      */
     public boolean getCanSpawnHere()
     {
-        IBlockState iblockstate = this.worldObj.func_180495_p((new BlockPos(this)).func_177977_b());
+        IBlockState iblockstate = this.worldObj.getBlockState((new BlockPos(this)).getBlockPosBelow());
         return iblockstate.func_189884_a(this);
     }
 
@@ -937,7 +937,7 @@
      */
     public boolean handleLavaMovement()
     {
-        return !this.worldObj.isAnyLiquid(this.func_174813_aQ()) && this.worldObj.func_184144_a(this, this.func_174813_aQ()).isEmpty() && this.worldObj.checkNoEntityCollision(this.func_174813_aQ(), this);
+        return !this.worldObj.isAnyLiquid(this.func_174813_aQ()) && this.worldObj.getCollisionBoxes(this, this.func_174813_aQ()).isEmpty() && this.worldObj.checkNoEntityCollision(this.func_174813_aQ(), this);
     }
 
     /**
@@ -990,7 +990,7 @@
     }
 
     @Nullable
-    public ItemStack func_184582_a(EntityEquipmentSlot p_184582_1_)
+    public ItemStack getEquippedItemStack(EntityEquipmentSlot p_184582_1_)
     {
         ItemStack itemstack = null;
 
@@ -1007,7 +1007,7 @@
         return itemstack;
     }
 
-    public void func_184201_a(EntityEquipmentSlot p_184201_1_, @Nullable ItemStack p_184201_2_)
+    public void equipItemInSlot(EntityEquipmentSlot p_184201_1_, @Nullable ItemStack p_184201_2_)
     {
         switch (p_184201_1_.func_188453_a())
         {
@@ -1027,7 +1027,7 @@
     {
         for (EntityEquipmentSlot entityequipmentslot : EntityEquipmentSlot.values())
         {
-            ItemStack itemstack = this.func_184582_a(entityequipmentslot);
+            ItemStack itemstack = this.getEquippedItemStack(entityequipmentslot);
             double d0;
 
             switch (entityequipmentslot.func_188453_a())
@@ -1099,7 +1099,7 @@
             {
                 if (entityequipmentslot.func_188453_a() == EntityEquipmentSlot.Type.ARMOR)
                 {
-                    ItemStack itemstack = this.func_184582_a(entityequipmentslot);
+                    ItemStack itemstack = this.getEquippedItemStack(entityequipmentslot);
 
                     if (!flag && this.rand.nextFloat() < f)
                     {
@@ -1114,7 +1114,7 @@
 
                         if (item != null)
                         {
-                            this.func_184201_a(entityequipmentslot, new ItemStack(item));
+                            this.equipItemInSlot(entityequipmentslot, new ItemStack(item));
                         }
                     }
                 }
@@ -1228,16 +1228,16 @@
     {
         float f = p_180483_1_.func_180170_c();
 
-        if (this.func_184614_ca() != null && this.rand.nextFloat() < 0.25F * f)
+        if (this.getMainHandItem() != null && this.rand.nextFloat() < 0.25F * f)
         {
-            EnchantmentHelper.addRandomEnchantment(this.rand, this.func_184614_ca(), (int)(5.0F + f * (float)this.rand.nextInt(18)), false);
+            EnchantmentHelper.addRandomEnchantment(this.rand, this.getMainHandItem(), (int)(5.0F + f * (float)this.rand.nextInt(18)), false);
         }
 
         for (EntityEquipmentSlot entityequipmentslot : EntityEquipmentSlot.values())
         {
             if (entityequipmentslot.func_188453_a() == EntityEquipmentSlot.Type.ARMOR)
             {
-                ItemStack itemstack = this.func_184582_a(entityequipmentslot);
+                ItemStack itemstack = this.getEquippedItemStack(entityequipmentslot);
 
                 if (itemstack != null && this.rand.nextFloat() < 0.5F * f)
                 {
@@ -1410,9 +1410,9 @@
             ((WorldServer)this.worldObj).getEntityTracker().sendToAllTrackingEntity(this, new SPacketEntityAttach(this, this.leashedToEntity));
         }
 
-        if (this.func_184218_aH())
+        if (this.isRiding())
         {
-            this.func_184210_p();
+            this.dismount();
         }
     }
 
@@ -1436,7 +1436,7 @@
             {
                 UUID uuid = this.leashNBTTag.func_186857_a("UUID");
 
-                for (EntityLivingBase entitylivingbase : this.worldObj.getEntitiesWithinAABB(EntityLivingBase.class, this.func_174813_aQ().func_186662_g(10.0D)))
+                for (EntityLivingBase entitylivingbase : this.worldObj.getEntitiesWithinAABB(EntityLivingBase.class, this.func_174813_aQ().expand(10.0D)))
                 {
                     if (entitylivingbase.getUniqueID().equals(uuid))
                     {
@@ -1506,7 +1506,7 @@
         }
         else
         {
-            this.func_184201_a(entityequipmentslot, p_174820_2_);
+            this.equipItemInSlot(entityequipmentslot, p_174820_2_);
             return true;
         }
     }
