--- a/net/minecraft/entity/Entity.java
+++ b/net/minecraft/entity/Entity.java
@@ -91,7 +91,7 @@
     public boolean preventEntitySpawning;
     private final List<Entity> field_184244_h;
     protected int field_184245_j;
-    private Entity field_184239_as;
+    private Entity ridingEntity;
     public boolean forceSpawn;
 
     /** Reference to the World object. */
@@ -127,7 +127,9 @@
     public float prevRotationPitch;
 
     /** Axis aligned bounding box. */
-    private AxisAlignedBB boundingBox;
+    //VIVE make public
+    public AxisAlignedBB boundingBox;
+    //
     public boolean onGround;
 
     /**
@@ -260,7 +262,7 @@
     protected EnumFacing field_181018_ap;
     private boolean invulnerable;
     protected UUID entityUniqueID;
-    protected String field_189513_ar;
+    protected String uuid;
     private final CommandResultStats field_174837_as;
     private final List<ItemStack> field_184235_aE;
     protected boolean field_184238_ar;
@@ -279,7 +281,7 @@
         this.fireResistance = 1;
         this.firstUpdate = true;
         this.entityUniqueID = MathHelper.func_180182_a(this.rand);
-        this.field_189513_ar = this.entityUniqueID.toString();
+        this.uuid = this.entityUniqueID.toString();
         this.field_174837_as = new CommandResultStats();
         this.field_184235_aE = Lists.<ItemStack>newArrayList();
         this.field_184236_aF = Sets.<String>newHashSet();
@@ -368,7 +370,7 @@
             {
                 this.setPosition(this.posX, this.posY, this.posZ);
 
-                if (this.worldObj.func_184144_a(this, this.func_174813_aQ()).isEmpty())
+                if (this.worldObj.getCollisionBoxes(this, this.func_174813_aQ()).isEmpty())
                 {
                     break;
                 }
@@ -406,7 +408,7 @@
             this.width = width;
             this.height = height;
             AxisAlignedBB axisalignedbb = this.func_174813_aQ();
-            this.func_174826_a(new AxisAlignedBB(axisalignedbb.minX, axisalignedbb.minY, axisalignedbb.minZ, axisalignedbb.minX + (double)this.width, axisalignedbb.minY + (double)this.height, axisalignedbb.minZ + (double)this.width));
+            this.setBoundingBox(new AxisAlignedBB(axisalignedbb.minX, axisalignedbb.minY, axisalignedbb.minZ, axisalignedbb.minX + (double)this.width, axisalignedbb.minY + (double)this.height, axisalignedbb.minZ + (double)this.width));
 
             if (this.width > f && !this.firstUpdate && !this.worldObj.isRemote)
             {
@@ -434,7 +436,7 @@
         this.posZ = z;
         float f = this.width / 2.0F;
         float f1 = this.height;
-        this.func_174826_a(new AxisAlignedBB(x - (double)f, y, z - (double)f, x + (double)f, y + (double)f1, z + (double)f));
+        this.setBoundingBox(new AxisAlignedBB(x - (double)f, y, z - (double)f, x + (double)f, y + (double)f1, z + (double)f));
     }
 
     /**
@@ -451,9 +453,9 @@
         this.prevRotationPitch += this.rotationPitch - f;
         this.prevRotationYaw += this.rotationYaw - f1;
 
-        if (this.field_184239_as != null)
+        if (this.ridingEntity != null)
         {
-            this.field_184239_as.func_184190_l(this);
+            this.ridingEntity.func_184190_l(this);
         }
     }
 
@@ -477,9 +479,9 @@
     {
         this.worldObj.theProfiler.startSection("entityBaseTick");
 
-        if (this.func_184218_aH() && this.func_184187_bx().isDead)
+        if (this.isRiding() && this.getRidingEntity().isDead)
         {
-            this.func_184210_p();
+            this.dismount();
         }
 
         if (this.field_184245_j > 0)
@@ -504,7 +506,7 @@
 
                 if (minecraftserver.getAllowNether())
                 {
-                    if (!this.func_184218_aH())
+                    if (!this.isRiding())
                     {
                         int i = this.getMaxInPortalTime();
 
@@ -669,7 +671,7 @@
 
     private boolean func_174809_b(AxisAlignedBB p_174809_1_)
     {
-        return this.worldObj.func_184144_a(this, p_174809_1_).isEmpty() && !this.worldObj.isAnyLiquid(p_174809_1_);
+        return this.worldObj.getCollisionBoxes(this, p_174809_1_).isEmpty() && !this.worldObj.isAnyLiquid(p_174809_1_);
     }
 
     /**
@@ -679,7 +681,7 @@
     {
         if (this.noClip)
         {
-            this.func_174826_a(this.func_174813_aQ().offset(x, y, z));
+            this.setBoundingBox(this.func_174813_aQ().offset(x, y, z));
             this.func_174829_m();
         }
         else
@@ -707,7 +709,7 @@
 
             if (flag)
             {
-                for (double d6 = 0.05D; x != 0.0D && this.worldObj.func_184144_a(this, this.func_174813_aQ().offset(x, -1.0D, 0.0D)).isEmpty(); d3 = x)
+                for (double d6 = 0.05D; x != 0.0D && this.worldObj.getCollisionBoxes(this, this.func_174813_aQ().offset(x, -1.0D, 0.0D)).isEmpty(); d3 = x)
                 {
                     if (x < 0.05D && x >= -0.05D)
                     {
@@ -723,7 +725,7 @@
                     }
                 }
 
-                for (; z != 0.0D && this.worldObj.func_184144_a(this, this.func_174813_aQ().offset(0.0D, -1.0D, z)).isEmpty(); d5 = z)
+                for (; z != 0.0D && this.worldObj.getCollisionBoxes(this, this.func_174813_aQ().offset(0.0D, -1.0D, z)).isEmpty(); d5 = z)
                 {
                     if (z < 0.05D && z >= -0.05D)
                     {
@@ -739,7 +741,7 @@
                     }
                 }
 
-                for (; x != 0.0D && z != 0.0D && this.worldObj.func_184144_a(this, this.func_174813_aQ().offset(x, -1.0D, z)).isEmpty(); d5 = z)
+                for (; x != 0.0D && z != 0.0D && this.worldObj.getCollisionBoxes(this, this.func_174813_aQ().offset(x, -1.0D, z)).isEmpty(); d5 = z)
                 {
                     if (x < 0.05D && x >= -0.05D)
                     {
@@ -771,7 +773,7 @@
                 }
             }
 
-            List<AxisAlignedBB> list1 = this.worldObj.func_184144_a(this, this.func_174813_aQ().addCoord(x, y, z));
+            List<AxisAlignedBB> list1 = this.worldObj.getCollisionBoxes(this, this.func_174813_aQ().addCoord(x, y, z));
             AxisAlignedBB axisalignedbb = this.func_174813_aQ();
             int i = 0;
 
@@ -780,7 +782,7 @@
                 y = ((AxisAlignedBB)list1.get(i)).calculateYOffset(this.func_174813_aQ(), y);
             }
 
-            this.func_174826_a(this.func_174813_aQ().offset(0.0D, y, 0.0D));
+            this.setBoundingBox(this.func_174813_aQ().offset(0.0D, y, 0.0D));
             boolean i_ = this.onGround || d4 != y && d4 < 0.0D;
             int j4 = 0;
 
@@ -789,7 +791,7 @@
                 x = ((AxisAlignedBB)list1.get(j4)).calculateXOffset(this.func_174813_aQ(), x);
             }
 
-            this.func_174826_a(this.func_174813_aQ().offset(x, 0.0D, 0.0D));
+            this.setBoundingBox(this.func_174813_aQ().offset(x, 0.0D, 0.0D));
             j4 = 0;
 
             for (int k4 = list1.size(); j4 < k4; ++j4)
@@ -797,7 +799,7 @@
                 z = ((AxisAlignedBB)list1.get(j4)).calculateZOffset(this.func_174813_aQ(), z);
             }
 
-            this.func_174826_a(this.func_174813_aQ().offset(0.0D, 0.0D, z));
+            this.setBoundingBox(this.func_174813_aQ().offset(0.0D, 0.0D, z));
 
             if (this.stepHeight > 0.0F && i_ && (d3 != x || d5 != z))
             {
@@ -805,9 +807,9 @@
                 double d7 = y;
                 double d8 = z;
                 AxisAlignedBB axisalignedbb1 = this.func_174813_aQ();
-                this.func_174826_a(axisalignedbb);
+                this.setBoundingBox(axisalignedbb);
                 y = (double)this.stepHeight;
-                List<AxisAlignedBB> list = this.worldObj.func_184144_a(this, this.func_174813_aQ().addCoord(d3, y, d5));
+                List<AxisAlignedBB> list = this.worldObj.getCollisionBoxes(this, this.func_174813_aQ().addCoord(d3, y, d5));
                 AxisAlignedBB axisalignedbb2 = this.func_174813_aQ();
                 AxisAlignedBB axisalignedbb3 = axisalignedbb2.addCoord(d3, 0.0D, d5);
                 double d9 = y;
@@ -873,14 +875,14 @@
                     x = d15;
                     z = d16;
                     y = -d9;
-                    this.func_174826_a(axisalignedbb2);
+                    this.setBoundingBox(axisalignedbb2);
                 }
                 else
                 {
                     x = d18;
                     z = d19;
                     y = -d17;
-                    this.func_174826_a(axisalignedbb4);
+                    this.setBoundingBox(axisalignedbb4);
                 }
 
                 int l3 = 0;
@@ -890,14 +892,14 @@
                     y = ((AxisAlignedBB)list.get(l3)).calculateYOffset(this.func_174813_aQ(), y);
                 }
 
-                this.func_174826_a(this.func_174813_aQ().offset(0.0D, y, 0.0D));
+                this.setBoundingBox(this.func_174813_aQ().offset(0.0D, y, 0.0D));
 
                 if (d11 * d11 + d8 * d8 >= x * x + z * z)
                 {
                     x = d11;
                     y = d7;
                     z = d8;
-                    this.func_174826_a(axisalignedbb1);
+                    this.setBoundingBox(axisalignedbb1);
                 }
             }
 
@@ -912,13 +914,13 @@
             int l4 = MathHelper.floor_double(this.posY - 0.20000000298023224D);
             int i5 = MathHelper.floor_double(this.posZ);
             BlockPos blockpos = new BlockPos(j4, l4, i5);
-            IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
+            IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
 
-            if (iblockstate.func_185904_a() == Material.air)
+            if (iblockstate.getMaterial() == Material.air)
             {
-                BlockPos blockpos1 = blockpos.func_177977_b();
-                IBlockState iblockstate1 = this.worldObj.func_180495_p(blockpos1);
-                Block block1 = iblockstate1.func_177230_c();
+                BlockPos blockpos1 = blockpos.getBlockPosBelow();
+                IBlockState iblockstate1 = this.worldObj.getBlockState(blockpos1);
+                Block block1 = iblockstate1.getBlock();
 
                 if (block1 instanceof BlockFence || block1 instanceof BlockWall || block1 instanceof BlockFenceGate)
                 {
@@ -939,14 +941,14 @@
                 this.motionZ = 0.0D;
             }
 
-            Block block = iblockstate.func_177230_c();
+            Block block = iblockstate.getBlock();
 
             if (d4 != y)
             {
                 block.func_176216_a(this.worldObj, this);
             }
 
-            if (this.canTriggerWalking() && !flag && !this.func_184218_aH())
+            if (this.canTriggerWalking() && !flag && !this.isRiding())
             {
                 double d12 = this.posX - d0;
                 double d13 = this.posY - d1;
@@ -965,7 +967,7 @@
                 this.distanceWalkedModified = (float)((double)this.distanceWalkedModified + (double)MathHelper.sqrt_double(d12 * d12 + d14 * d14) * 0.6D);
                 this.distanceWalkedOnStepModified = (float)((double)this.distanceWalkedOnStepModified + (double)MathHelper.sqrt_double(d12 * d12 + d13 * d13 + d14 * d14) * 0.6D);
 
-                if (this.distanceWalkedOnStepModified > (float)this.nextStepDistance && iblockstate.func_185904_a() != Material.air)
+                if (this.distanceWalkedOnStepModified > (float)this.nextStepDistance && iblockstate.getMaterial() != Material.air)
                 {
                     this.nextStepDistance = (int)this.distanceWalkedOnStepModified + 1;
 
@@ -978,7 +980,7 @@
                             f = 1.0F;
                         }
 
-                        this.func_184185_a(this.func_184184_Z(), f, 1.0F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.4F);
+                        this.playSound(this.getStepSound(), f, 1.0F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.4F);
                     }
 
                     this.func_180429_a(blockpos, block);
@@ -999,7 +1001,7 @@
 
             boolean flag1 = this.isWet();
 
-            if (this.worldObj.func_147470_e(this.func_174813_aQ().func_186664_h(0.001D)))
+            if (this.worldObj.func_147470_e(this.func_174813_aQ().contract(0.001D)))
             {
                 this.dealFireDamage(1);
 
@@ -1020,7 +1022,7 @@
 
             if (flag1 && this.fire > 0)
             {
-                this.func_184185_a(SoundEvents.field_187541_bC, 0.7F, 1.6F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.4F);
+                this.playSound(SoundEvents.field_187541_bC, 0.7F, 1.6F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.4F);
                 this.fire = -this.fireResistance;
             }
 
@@ -1036,14 +1038,14 @@
         this.posZ = (axisalignedbb.minZ + axisalignedbb.maxZ) / 2.0D;
     }
 
-    protected SoundEvent func_184184_Z()
+    protected SoundEvent getStepSound()
     {
-        return SoundEvents.field_187549_bG;
+        return SoundEvents.swimSound;
     }
 
-    protected SoundEvent func_184181_aa()
+    protected SoundEvent getSplashSound()
     {
-        return SoundEvents.field_187547_bF;
+        return SoundEvents.splashSound;
     }
 
     protected void doBlockCollisions()
@@ -1055,18 +1057,18 @@
 
         if (this.worldObj.func_175707_a(blockpos$pooledmutableblockpos, blockpos$pooledmutableblockpos1))
         {
-            for (int i = blockpos$pooledmutableblockpos.func_177958_n(); i <= blockpos$pooledmutableblockpos1.func_177958_n(); ++i)
+            for (int i = blockpos$pooledmutableblockpos.getX(); i <= blockpos$pooledmutableblockpos1.getX(); ++i)
             {
-                for (int j = blockpos$pooledmutableblockpos.func_177956_o(); j <= blockpos$pooledmutableblockpos1.func_177956_o(); ++j)
+                for (int j = blockpos$pooledmutableblockpos.getY(); j <= blockpos$pooledmutableblockpos1.getY(); ++j)
                 {
-                    for (int k = blockpos$pooledmutableblockpos.func_177952_p(); k <= blockpos$pooledmutableblockpos1.func_177952_p(); ++k)
+                    for (int k = blockpos$pooledmutableblockpos.getZ(); k <= blockpos$pooledmutableblockpos1.getZ(); ++k)
                     {
                         blockpos$pooledmutableblockpos2.func_181079_c(i, j, k);
-                        IBlockState iblockstate = this.worldObj.func_180495_p(blockpos$pooledmutableblockpos2);
+                        IBlockState iblockstate = this.worldObj.getBlockState(blockpos$pooledmutableblockpos2);
 
                         try
                         {
-                            iblockstate.func_177230_c().func_180634_a(this.worldObj, blockpos$pooledmutableblockpos2, iblockstate, this);
+                            iblockstate.getBlock().func_180634_a(this.worldObj, blockpos$pooledmutableblockpos2, iblockstate, this);
                         }
                         catch (Throwable throwable)
                         {
@@ -1089,22 +1091,22 @@
     {
         SoundType soundtype = p_180429_2_.func_185467_w();
 
-        if (this.worldObj.func_180495_p(p_180429_1_.func_177984_a()).func_177230_c() == Blocks.snow_layer)
+        if (this.worldObj.getBlockState(p_180429_1_.getBlockPosAbove()).getBlock() == Blocks.snow_layer)
         {
             soundtype = Blocks.snow_layer.func_185467_w();
-            this.func_184185_a(soundtype.func_185844_d(), soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
+            this.playSound(soundtype.func_185844_d(), soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
         }
-        else if (!p_180429_2_.func_176223_P().func_185904_a().isLiquid())
+        else if (!p_180429_2_.getBlockState().getMaterial().isLiquid())
         {
-            this.func_184185_a(soundtype.func_185844_d(), soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
+            this.playSound(soundtype.func_185844_d(), soundtype.func_185843_a() * 0.15F, soundtype.func_185847_b());
         }
     }
 
-    public void func_184185_a(SoundEvent p_184185_1_, float p_184185_2_, float p_184185_3_)
+    public void playSound(SoundEvent p_184185_1_, float p_184185_2_, float p_184185_3_)
     {
         if (!this.func_174814_R())
         {
-            this.worldObj.func_184148_a((EntityPlayer)null, this.posX, this.posY, this.posZ, p_184185_1_, this.func_184176_by(), p_184185_2_, p_184185_3_);
+            this.worldObj.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, p_184185_1_, this.func_184176_by(), p_184185_2_, p_184185_3_);
         }
     }
 
@@ -1143,7 +1145,7 @@
         {
             if (this.fallDistance > 0.0F)
             {
-                p_184231_4_.func_177230_c().func_180658_a(this.worldObj, p_184231_5_, this, this.fallDistance);
+                p_184231_4_.getBlock().func_180658_a(this.worldObj, p_184231_5_, this, this.fallDistance);
             }
 
             this.fallDistance = 0.0F;
@@ -1232,11 +1234,11 @@
      */
     public boolean handleWaterMovement()
     {
-        if (this.func_184187_bx() instanceof EntityBoat)
+        if (this.getRidingEntity() instanceof EntityBoat)
         {
             this.inWater = false;
         }
-        else if (this.worldObj.handleMaterialAcceleration(this.func_174813_aQ().expand(0.0D, -0.4000000059604645D, 0.0D).func_186664_h(0.001D), Material.water, this))
+        else if (this.worldObj.handleMaterialAcceleration(this.func_174813_aQ().expand(0.0D, -0.4000000059604645D, 0.0D).contract(0.001D), Material.water, this))
         {
             if (!this.inWater && !this.firstUpdate)
             {
@@ -1267,21 +1269,21 @@
             f = 1.0F;
         }
 
-        this.func_184185_a(this.func_184181_aa(), f, 1.0F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.4F);
+        this.playSound(this.getSplashSound(), f, 1.0F + (this.rand.nextFloat() - this.rand.nextFloat()) * 0.4F);
         float f1 = (float)MathHelper.floor_double(this.func_174813_aQ().minY);
 
         for (int i = 0; (float)i < 1.0F + this.width * 20.0F; ++i)
         {
             float f2 = (this.rand.nextFloat() * 2.0F - 1.0F) * this.width;
             float f3 = (this.rand.nextFloat() * 2.0F - 1.0F) * this.width;
-            this.worldObj.func_175688_a(EnumParticleTypes.WATER_BUBBLE, this.posX + (double)f2, (double)(f1 + 1.0F), this.posZ + (double)f3, this.motionX, this.motionY - (double)(this.rand.nextFloat() * 0.2F), this.motionZ, new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.WATER_BUBBLE, this.posX + (double)f2, (double)(f1 + 1.0F), this.posZ + (double)f3, this.motionX, this.motionY - (double)(this.rand.nextFloat() * 0.2F), this.motionZ, new int[0]);
         }
 
         for (int j = 0; (float)j < 1.0F + this.width * 20.0F; ++j)
         {
             float f4 = (this.rand.nextFloat() * 2.0F - 1.0F) * this.width;
             float f5 = (this.rand.nextFloat() * 2.0F - 1.0F) * this.width;
-            this.worldObj.func_175688_a(EnumParticleTypes.WATER_SPLASH, this.posX + (double)f4, (double)(f1 + 1.0F), this.posZ + (double)f5, this.motionX, this.motionY, this.motionZ, new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.WATER_SPLASH, this.posX + (double)f4, (double)(f1 + 1.0F), this.posZ + (double)f5, this.motionX, this.motionY, this.motionZ, new int[0]);
         }
     }
 
@@ -1299,11 +1301,11 @@
         int j = MathHelper.floor_double(this.posY - 0.20000000298023224D);
         int k = MathHelper.floor_double(this.posZ);
         BlockPos blockpos = new BlockPos(i, j, k);
-        IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
+        IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
 
         if (iblockstate.func_185901_i() != EnumBlockRenderType.INVISIBLE)
         {
-            this.worldObj.func_175688_a(EnumParticleTypes.BLOCK_CRACK, this.posX + ((double)this.rand.nextFloat() - 0.5D) * (double)this.width, this.func_174813_aQ().minY + 0.1D, this.posZ + ((double)this.rand.nextFloat() - 0.5D) * (double)this.width, -this.motionX * 4.0D, 1.5D, -this.motionZ * 4.0D, new int[] {Block.func_176210_f(iblockstate)});
+            this.worldObj.spawnParticle(EnumParticleTypes.BLOCK_CRACK, this.posX + ((double)this.rand.nextFloat() - 0.5D) * (double)this.width, this.func_174813_aQ().minY + 0.1D, this.posZ + ((double)this.rand.nextFloat() - 0.5D) * (double)this.width, -this.motionX * 4.0D, 1.5D, -this.motionZ * 4.0D, new int[] {Block.func_176210_f(iblockstate)});
         }
     }
 
@@ -1312,7 +1314,7 @@
      */
     public boolean isInsideOfMaterial(Material materialIn)
     {
-        if (this.func_184187_bx() instanceof EntityBoat)
+        if (this.getRidingEntity() instanceof EntityBoat)
         {
             return false;
         }
@@ -1320,12 +1322,12 @@
         {
             double d0 = this.posY + (double)this.getEyeHeight();
             BlockPos blockpos = new BlockPos(this.posX, d0, this.posZ);
-            IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
+            IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
 
-            if (iblockstate.func_185904_a() == materialIn)
+            if (iblockstate.getMaterial() == materialIn)
             {
-                float f = BlockLiquid.getLiquidHeightPercent(iblockstate.func_177230_c().func_176201_c(iblockstate)) - 0.11111111F;
-                float f1 = (float)(blockpos.func_177956_o() + 1) - f;
+                float f = BlockLiquid.getLiquidHeightPercent(iblockstate.getBlock().func_176201_c(iblockstate)) - 0.11111111F;
+                float f1 = (float)(blockpos.getY() + 1) - f;
                 boolean flag = d0 < (double)f1;
                 return !flag && this instanceof EntityPlayer ? false : flag;
             }
@@ -1442,7 +1444,7 @@
 
     public void func_174828_a(BlockPos p_174828_1_, float p_174828_2_, float p_174828_3_)
     {
-        this.setLocationAndAngles((double)p_174828_1_.func_177958_n() + 0.5D, (double)p_174828_1_.func_177956_o(), (double)p_174828_1_.func_177952_p() + 0.5D, p_174828_2_, p_174828_3_);
+        this.setLocationAndAngles((double)p_174828_1_.getX() + 0.5D, (double)p_174828_1_.getY(), (double)p_174828_1_.getZ() + 0.5D, p_174828_2_, p_174828_3_);
     }
 
     /**
@@ -1731,7 +1733,7 @@
     {
         String s = this.getEntityString();
 
-        if (!this.isDead && s != null && !this.func_184218_aH())
+        if (!this.isDead && s != null && !this.isRiding())
         {
             tagCompund.setString("id", s);
             this.func_189511_e(tagCompund);
@@ -1891,7 +1893,7 @@
             if (tagCompund.func_186855_b("UUID"))
             {
                 this.entityUniqueID = tagCompund.func_186857_a("UUID");
-                this.field_189513_ar = this.entityUniqueID.toString();
+                this.uuid = this.entityUniqueID.toString();
             }
 
             this.setPosition(this.posX, this.posY, this.posZ);
@@ -2044,11 +2046,11 @@
                 int k = MathHelper.floor_double(this.posX + (double)(((float)((i >> 1) % 2) - 0.5F) * this.width * 0.8F));
                 int l = MathHelper.floor_double(this.posZ + (double)(((float)((i >> 2) % 2) - 0.5F) * this.width * 0.8F));
 
-                if (blockpos$pooledmutableblockpos.func_177958_n() != k || blockpos$pooledmutableblockpos.func_177956_o() != j || blockpos$pooledmutableblockpos.func_177952_p() != l)
+                if (blockpos$pooledmutableblockpos.getX() != k || blockpos$pooledmutableblockpos.getY() != j || blockpos$pooledmutableblockpos.getZ() != l)
                 {
                     blockpos$pooledmutableblockpos.func_181079_c(k, j, l);
 
-                    if (this.worldObj.func_180495_p(blockpos$pooledmutableblockpos).func_177230_c().func_176214_u())
+                    if (this.worldObj.getBlockState(blockpos$pooledmutableblockpos).getBlock().func_176214_u())
                     {
                         blockpos$pooledmutableblockpos.func_185344_t();
                         return true;
@@ -2082,11 +2084,11 @@
      */
     public void updateRidden()
     {
-        Entity entity = this.func_184187_bx();
+        Entity entity = this.getRidingEntity();
 
-        if (this.func_184218_aH() && entity.isDead)
+        if (this.isRiding() && entity.isDead)
         {
-            this.func_184210_p();
+            this.dismount();
         }
         else
         {
@@ -2095,7 +2097,7 @@
             this.motionZ = 0.0D;
             this.onUpdate();
 
-            if (this.func_184218_aH())
+            if (this.isRiding())
             {
                 entity.func_184232_k(this);
             }
@@ -2139,13 +2141,13 @@
     {
         if (p_184205_2_ || this.func_184228_n(p_184205_1_) && p_184205_1_.func_184219_q(this))
         {
-            if (this.func_184218_aH())
+            if (this.isRiding())
             {
-                this.func_184210_p();
+                this.dismount();
             }
 
-            this.field_184239_as = p_184205_1_;
-            this.field_184239_as.func_184200_o(this);
+            this.ridingEntity = p_184205_1_;
+            this.ridingEntity.addPassenger(this);
             return true;
         }
         else
@@ -2163,23 +2165,23 @@
     {
         for (int i = this.field_184244_h.size() - 1; i >= 0; --i)
         {
-            ((Entity)this.field_184244_h.get(i)).func_184210_p();
+            ((Entity)this.field_184244_h.get(i)).dismount();
         }
     }
 
-    public void func_184210_p()
+    public void dismount()
     {
-        if (this.field_184239_as != null)
+        if (this.ridingEntity != null)
         {
-            Entity entity = this.field_184239_as;
-            this.field_184239_as = null;
-            entity.func_184225_p(this);
+            Entity entity = this.ridingEntity;
+            this.ridingEntity = null;
+            entity.removePassenger(this);
         }
     }
 
-    protected void func_184200_o(Entity p_184200_1_)
+    protected void addPassenger(Entity p_184200_1_)
     {
-        if (p_184200_1_.func_184187_bx() != this)
+        if (p_184200_1_.getRidingEntity() != this)
         {
             throw new IllegalStateException("Use x.startRiding(y), not y.addPassenger(x)");
         }
@@ -2196,9 +2198,9 @@
         }
     }
 
-    protected void func_184225_p(Entity p_184225_1_)
+    protected void removePassenger(Entity p_184225_1_)
     {
-        if (p_184225_1_.func_184187_bx() == this)
+        if (p_184225_1_.getRidingEntity() == this)
         {
             throw new IllegalStateException("Use x.stopRiding(y), not y.removePassenger(x)");
         }
@@ -2241,7 +2243,7 @@
 
     public Vec3d func_189651_aD()
     {
-        return Vec3d.func_189984_a(this.func_189653_aC());
+        return Vec3d.fromPitchYawVector(this.func_189653_aC());
     }
 
     public void func_181015_d(BlockPos p_181015_1_)
@@ -2256,10 +2258,10 @@
             {
                 this.field_181016_an = new BlockPos(p_181015_1_);
                 BlockPattern.PatternHelper blockpattern$patternhelper = Blocks.portal.func_181089_f(this.worldObj, this.field_181016_an);
-                double d0 = blockpattern$patternhelper.func_177669_b().func_176740_k() == EnumFacing.Axis.X ? (double)blockpattern$patternhelper.func_181117_a().func_177952_p() : (double)blockpattern$patternhelper.func_181117_a().func_177958_n();
+                double d0 = blockpattern$patternhelper.func_177669_b().func_176740_k() == EnumFacing.Axis.X ? (double)blockpattern$patternhelper.func_181117_a().getZ() : (double)blockpattern$patternhelper.func_181117_a().getX();
                 double d1 = blockpattern$patternhelper.func_177669_b().func_176740_k() == EnumFacing.Axis.X ? this.posZ : this.posX;
                 d1 = Math.abs(MathHelper.func_181160_c(d1 - (double)(blockpattern$patternhelper.func_177669_b().func_176746_e().func_176743_c() == EnumFacing.AxisDirection.NEGATIVE ? 1 : 0), d0, d0 - (double)blockpattern$patternhelper.func_181118_d()));
-                double d2 = MathHelper.func_181160_c(this.posY - 1.0D, (double)blockpattern$patternhelper.func_181117_a().func_177956_o(), (double)(blockpattern$patternhelper.func_181117_a().func_177956_o() - blockpattern$patternhelper.func_181119_e()));
+                double d2 = MathHelper.func_181160_c(this.posY - 1.0D, (double)blockpattern$patternhelper.func_181117_a().getY(), (double)(blockpattern$patternhelper.func_181117_a().getY() - blockpattern$patternhelper.func_181119_e()));
                 this.field_181017_ao = new Vec3d(d1, d2, 0.0D);
                 this.field_181018_ap = blockpattern$patternhelper.func_177669_b();
             }
@@ -2312,7 +2314,7 @@
         return Iterables.<ItemStack>concat(this.func_184214_aD(), this.func_184193_aE());
     }
 
-    public void func_184201_a(EntityEquipmentSlot p_184201_1_, @Nullable ItemStack p_184201_2_)
+    public void equipItemInSlot(EntityEquipmentSlot p_184201_1_, @Nullable ItemStack p_184201_2_)
     {
     }
 
@@ -2325,9 +2327,9 @@
         return !this.isImmuneToFire && (this.fire > 0 || flag && this.getFlag(0));
     }
 
-    public boolean func_184218_aH()
+    public boolean isRiding()
     {
-        return this.func_184187_bx() != null;
+        return this.getRidingEntity() != null;
     }
 
     public boolean func_184207_aI()
@@ -2488,9 +2490,9 @@
     protected boolean pushOutOfBlocks(double x, double y, double z)
     {
         BlockPos blockpos = new BlockPos(x, y, z);
-        double d0 = x - (double)blockpos.func_177958_n();
-        double d1 = y - (double)blockpos.func_177956_o();
-        double d2 = z - (double)blockpos.func_177952_p();
+        double d0 = x - (double)blockpos.getX();
+        double d1 = y - (double)blockpos.getY();
+        double d2 = z - (double)blockpos.getZ();
         List<AxisAlignedBB> list = this.worldObj.func_147461_a(this.func_174813_aQ());
 
         if (list.isEmpty())
@@ -2502,31 +2504,31 @@
             EnumFacing enumfacing = EnumFacing.UP;
             double d3 = Double.MAX_VALUE;
 
-            if (!this.worldObj.func_175665_u(blockpos.func_177976_e()) && d0 < d3)
+            if (!this.worldObj.func_175665_u(blockpos.getBlockPosWest()) && d0 < d3)
             {
                 d3 = d0;
                 enumfacing = EnumFacing.WEST;
             }
 
-            if (!this.worldObj.func_175665_u(blockpos.func_177974_f()) && 1.0D - d0 < d3)
+            if (!this.worldObj.func_175665_u(blockpos.getBlockPosEast()) && 1.0D - d0 < d3)
             {
                 d3 = 1.0D - d0;
                 enumfacing = EnumFacing.EAST;
             }
 
-            if (!this.worldObj.func_175665_u(blockpos.func_177978_c()) && d2 < d3)
+            if (!this.worldObj.func_175665_u(blockpos.getBlockPosNorth()) && d2 < d3)
             {
                 d3 = d2;
                 enumfacing = EnumFacing.NORTH;
             }
 
-            if (!this.worldObj.func_175665_u(blockpos.func_177968_d()) && 1.0D - d2 < d3)
+            if (!this.worldObj.func_175665_u(blockpos.getBlockPosSouth()) && 1.0D - d2 < d3)
             {
                 d3 = 1.0D - d2;
                 enumfacing = EnumFacing.SOUTH;
             }
 
-            if (!this.worldObj.func_175665_u(blockpos.func_177984_a()) && 1.0D - d1 < d3)
+            if (!this.worldObj.func_175665_u(blockpos.getBlockPosAbove()) && 1.0D - d1 < d3)
             {
                 d3 = 1.0D - d1;
                 enumfacing = EnumFacing.UP;
@@ -2763,7 +2765,7 @@
 
     public float func_180428_a(Explosion p_180428_1_, World p_180428_2_, BlockPos p_180428_3_, IBlockState p_180428_4_)
     {
-        return p_180428_4_.func_177230_c().getExplosionResistance(this);
+        return p_180428_4_.getBlock().getExplosionResistance(this);
     }
 
     public boolean func_174816_a(Explosion p_174816_1_, World p_174816_2_, BlockPos p_174816_3_, IBlockState p_174816_4_, float p_174816_5_)
@@ -2828,7 +2830,7 @@
         {
             public String call() throws Exception
             {
-                return Entity.this.func_184187_bx().toString();
+                return Entity.this.getRidingEntity().toString();
             }
         });
     }
@@ -2844,7 +2846,7 @@
     public void func_184221_a(UUID p_184221_1_)
     {
         this.entityUniqueID = p_184221_1_;
-        this.field_189513_ar = this.entityUniqueID.toString();
+        this.uuid = this.entityUniqueID.toString();
     }
 
     public UUID getUniqueID()
@@ -2854,7 +2856,7 @@
 
     public String func_189512_bd()
     {
-        return this.field_189513_ar;
+        return this.uuid;
     }
 
     public boolean isPushedByWater()
@@ -2970,7 +2972,7 @@
         return this.func_174813_aQ();
     }
 
-    public void func_174826_a(AxisAlignedBB p_174826_1_)
+    public void setBoundingBox(AxisAlignedBB p_174826_1_)
     {
         this.boundingBox = p_174826_1_;
     }
@@ -3190,11 +3192,14 @@
         }
     }
 
-    public Entity func_184208_bv()
+    /**
+     * returns either this entity or this entity's ridingEntity if riding.
+     */
+    public Entity getMovementEntity()
     {
         Entity entity;
 
-        for (entity = this; entity.func_184218_aH(); entity = entity.func_184187_bx())
+        for (entity = this; entity.isRiding(); entity = entity.getRidingEntity())
         {
             ;
         }
@@ -3204,7 +3209,7 @@
 
     public boolean func_184223_x(Entity p_184223_1_)
     {
-        return this.func_184208_bv() == p_184223_1_.func_184208_bv();
+        return this.getMovementEntity() == p_184223_1_.getMovementEntity();
     }
 
     public boolean func_184215_y(Entity p_184215_1_)
@@ -3232,9 +3237,9 @@
     }
 
     @Nullable
-    public Entity func_184187_bx()
+    public Entity getRidingEntity()
     {
-        return this.field_184239_as;
+        return this.ridingEntity;
     }
 
     public EnumPushReaction func_184192_z()
