--- a/net/minecraft/entity/EntityTrackerEntry.java
+++ b/net/minecraft/entity/EntityTrackerEntry.java
@@ -188,7 +188,7 @@
 
         if (this.updateCounter % this.updateFrequency == 0 || this.trackedEntity.isAirBorne || this.trackedEntity.func_184212_Q().func_187223_a())
         {
-            if (this.trackedEntity.func_184218_aH())
+            if (this.trackedEntity.isRiding())
             {
                 int j1 = MathHelper.floor_float(this.trackedEntity.rotationYaw * 256.0F / 360.0F);
                 int l1 = MathHelper.floor_float(this.trackedEntity.rotationPitch * 256.0F / 360.0F);
@@ -430,7 +430,7 @@
                     {
                         for (EntityEquipmentSlot entityequipmentslot : EntityEquipmentSlot.values())
                         {
-                            ItemStack itemstack = ((EntityLivingBase)this.trackedEntity).func_184582_a(entityequipmentslot);
+                            ItemStack itemstack = ((EntityLivingBase)this.trackedEntity).getEquippedItemStack(entityequipmentslot);
 
                             if (itemstack != null)
                             {
