--- a/net/optifine/entity/model/CustomEntityModels.java
+++ b/net/optifine/entity/model/CustomEntityModels.java
@@ -14,7 +14,9 @@
 import net.minecraft.client.renderer.entity.RenderManager;
 import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher;
 import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
+import net.minecraft.entity.Entity;
 import net.minecraft.src.Config;
+import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.ResourceLocation;
 import net.optifine.entity.model.anim.ModelResolver;
 import net.optifine.entity.model.anim.ModelUpdater;
@@ -23,12 +25,12 @@
 {
     private static boolean active = false;
     private static Map<Class, Render> originalEntityRenderMap = null;
-    private static Map<Class, TileEntitySpecialRenderer> originalTileEntityRenderMap = null;
+    private static Map<Class<? extends TileEntity>, TileEntitySpecialRenderer<? extends TileEntity>> originalTileEntityRenderMap = null;
 
     public static void update()
     {
         Map<Class, Render> map = getEntityRenderMap();
-        Map<Class, TileEntitySpecialRenderer> map1 = getTileEntityRenderMap();
+        Map<Class<? extends TileEntity>, TileEntitySpecialRenderer<? extends TileEntity>> map1 = getTileEntityRenderMap();
 
         if (map == null)
         {
@@ -44,7 +46,7 @@
             map.clear();
             map1.clear();
             map.putAll(originalEntityRenderMap);
-            map1.putAll(originalTileEntityRenderMap);
+            map1.putAll((Map<? extends Class<? extends TileEntity>, ? extends TileEntitySpecialRenderer<? extends TileEntity>>) originalTileEntityRenderMap);
 
             if (Config.isCustomEntityModels())
             {
@@ -103,9 +105,9 @@
         }
     }
 
-    private static Map<Class, TileEntitySpecialRenderer> getTileEntityRenderMap()
+    private static Map<Class<? extends TileEntity>, TileEntitySpecialRenderer<? extends TileEntity>> getTileEntityRenderMap()
     {
-        Map<Class, TileEntitySpecialRenderer> map = TileEntityRendererDispatcher.instance.mapSpecialRenderers;
+        Map<Class<? extends TileEntity>, TileEntitySpecialRenderer<? extends TileEntity>> map = TileEntityRendererDispatcher.instance.mapSpecialRenderers;
 
         if (originalTileEntityRenderMap == null)
         {
