--- a/net/minecraft/server/management/PlayerChunkMap.java
+++ b/net/minecraft/server/management/PlayerChunkMap.java
@@ -271,13 +271,13 @@
 
     public void func_180244_a(BlockPos p_180244_1_)
     {
-        int i = p_180244_1_.func_177958_n() >> 4;
-        int j = p_180244_1_.func_177952_p() >> 4;
+        int i = p_180244_1_.getX() >> 4;
+        int j = p_180244_1_.getZ() >> 4;
         PlayerChunkMapEntry playerchunkmapentry = this.func_187301_b(i, j);
 
         if (playerchunkmapentry != null)
         {
-            playerchunkmapentry.func_187265_a(p_180244_1_.func_177958_n() & 15, p_180244_1_.func_177956_o(), p_180244_1_.func_177952_p() & 15);
+            playerchunkmapentry.func_187265_a(p_180244_1_.getX() & 15, p_180244_1_.getY(), p_180244_1_.getZ() & 15);
         }
     }
 
