--- a/net/minecraft/server/management/PlayerList.java
+++ b/net/minecraft/server/management/PlayerList.java
@@ -208,7 +208,7 @@
                         }
                     }
 
-                    if (!player.func_184218_aH())
+                    if (!player.isRiding())
                     {
                         logger.warn("Couldn\'t reattach entity to player");
                         worldserver.removePlayerEntityDangerously(entity2);
@@ -386,14 +386,14 @@
         player.triggerAchievement(StatList.leaveGameStat);
         this.writePlayerData(player);
 
-        if (player.func_184218_aH())
+        if (player.isRiding())
         {
-            Entity entity = player.func_184208_bv();
+            Entity entity = player.getMovementEntity();
 
             if (entity.func_184180_b(EntityPlayerMP.class).size() == 1)
             {
                 logger.debug("Removing player mount");
-                player.func_184210_p();
+                player.dismount();
                 worldserver.removePlayerEntityDangerously(entity);
 
                 for (Entity entity1 : entity.func_184182_bu())
@@ -548,7 +548,7 @@
 
             if (blockpos1 != null)
             {
-                entityplayermp.setLocationAndAngles((double)((float)blockpos1.func_177958_n() + 0.5F), (double)((float)blockpos1.func_177956_o() + 0.1F), (double)((float)blockpos1.func_177952_p() + 0.5F), 0.0F, 0.0F);
+                entityplayermp.setLocationAndAngles((double)((float)blockpos1.getX() + 0.5F), (double)((float)blockpos1.getY() + 0.1F), (double)((float)blockpos1.getZ() + 0.5F), 0.0F, 0.0F);
                 entityplayermp.func_180473_a(blockpos, flag);
             }
             else
@@ -559,7 +559,7 @@
 
         worldserver.getChunkProvider().func_186025_d((int)entityplayermp.posX >> 4, (int)entityplayermp.posZ >> 4);
 
-        while (!worldserver.func_184144_a(entityplayermp, entityplayermp.func_174813_aQ()).isEmpty() && entityplayermp.posY < 256.0D)
+        while (!worldserver.getCollisionBoxes(entityplayermp, entityplayermp.func_174813_aQ()).isEmpty() && entityplayermp.posY < 256.0D)
         {
             entityplayermp.setPosition(entityplayermp.posX, entityplayermp.posY + 1.0D, entityplayermp.posZ);
         }
@@ -659,9 +659,9 @@
                 blockpos = p_82448_4_.func_180504_m();
             }
 
-            d0 = (double)blockpos.func_177958_n();
-            entityIn.posY = (double)blockpos.func_177956_o();
-            d1 = (double)blockpos.func_177952_p();
+            d0 = (double)blockpos.getX();
+            entityIn.posY = (double)blockpos.getY();
+            d1 = (double)blockpos.getZ();
             entityIn.setLocationAndAngles(d0, entityIn.posY, d1, 90.0F, 0.0F);
 
             if (entityIn.isEntityAlive())
