--- a/net/minecraft/inventory/ContainerWorkbench.java
+++ b/net/minecraft/inventory/ContainerWorkbench.java
@@ -78,7 +78,7 @@
 
     public boolean canInteractWith(EntityPlayer player)
     {
-        return this.worldObj.func_180495_p(this.field_178145_h).func_177230_c() != Blocks.crafting_table ? false : player.getDistanceSq((double)this.field_178145_h.func_177958_n() + 0.5D, (double)this.field_178145_h.func_177956_o() + 0.5D, (double)this.field_178145_h.func_177952_p() + 0.5D) <= 64.0D;
+        return this.worldObj.getBlockState(this.field_178145_h).getBlock() != Blocks.crafting_table ? false : player.getDistanceSq((double)this.field_178145_h.getX() + 0.5D, (double)this.field_178145_h.getY() + 0.5D, (double)this.field_178145_h.getZ() + 0.5D) <= 64.0D;
     }
 
     @Nullable
