--- a/net/minecraft/inventory/ContainerRepair.java
+++ b/net/minecraft/inventory/ContainerRepair.java
@@ -43,7 +43,7 @@
 
     public ContainerRepair(InventoryPlayer p_i45806_1_, World p_i45806_2_, EntityPlayer p_i45806_3_)
     {
-        this(p_i45806_1_, p_i45806_2_, BlockPos.field_177992_a, p_i45806_3_);
+        this(p_i45806_1_, p_i45806_2_, BlockPos.createDefault, p_i45806_3_);
     }
 
     public ContainerRepair(InventoryPlayer p_i45807_1_, final World p_i45807_2_, final BlockPos p_i45807_3_, EntityPlayer p_i45807_4_)
@@ -101,9 +101,9 @@
                 }
 
                 ContainerRepair.this.maximumCost = 0;
-                IBlockState iblockstate = p_i45807_2_.func_180495_p(p_i45807_3_);
+                IBlockState iblockstate = p_i45807_2_.getBlockState(p_i45807_3_);
 
-                if (!p_82870_1_.capabilities.isCreativeMode && !p_i45807_2_.isRemote && iblockstate.func_177230_c() == Blocks.anvil && p_82870_1_.getRNG().nextFloat() < 0.12F)
+                if (!p_82870_1_.capabilities.isCreativeMode && !p_i45807_2_.isRemote && iblockstate.getBlock() == Blocks.anvil && p_82870_1_.getRNG().nextFloat() < 0.12F)
                 {
                     int l = ((Integer)iblockstate.func_177229_b(BlockAnvil.field_176505_b)).intValue();
                     ++l;
@@ -391,7 +391,7 @@
 
     public boolean canInteractWith(EntityPlayer player)
     {
-        return this.theWorld.func_180495_p(this.field_178156_j).func_177230_c() != Blocks.anvil ? false : player.getDistanceSq((double)this.field_178156_j.func_177958_n() + 0.5D, (double)this.field_178156_j.func_177956_o() + 0.5D, (double)this.field_178156_j.func_177952_p() + 0.5D) <= 64.0D;
+        return this.theWorld.getBlockState(this.field_178156_j).getBlock() != Blocks.anvil ? false : player.getDistanceSq((double)this.field_178156_j.getX() + 0.5D, (double)this.field_178156_j.getY() + 0.5D, (double)this.field_178156_j.getZ() + 0.5D) <= 64.0D;
     }
 
     @Nullable
