--- a/net/minecraft/inventory/ContainerEnchantment.java
+++ b/net/minecraft/inventory/ContainerEnchantment.java
@@ -36,7 +36,7 @@
 
     public ContainerEnchantment(InventoryPlayer p_i45797_1_, World p_i45797_2_)
     {
-        this(p_i45797_1_, p_i45797_2_, BlockPos.field_177992_a);
+        this(p_i45797_1_, p_i45797_2_, BlockPos.createDefault);
     }
 
     public ContainerEnchantment(InventoryPlayer p_i45798_1_, World p_i45798_2_, BlockPos p_i45798_3_)
@@ -170,36 +170,36 @@
                     {
                         for (int k = -1; k <= 1; ++k)
                         {
-                            if ((j != 0 || k != 0) && this.worldPointer.func_175623_d(this.field_178150_j.func_177982_a(k, 0, j)) && this.worldPointer.func_175623_d(this.field_178150_j.func_177982_a(k, 1, j)))
+                            if ((j != 0 || k != 0) && this.worldPointer.isEmpty(this.field_178150_j.func_177982_a(k, 0, j)) && this.worldPointer.isEmpty(this.field_178150_j.func_177982_a(k, 1, j)))
                             {
-                                if (this.worldPointer.func_180495_p(this.field_178150_j.func_177982_a(k * 2, 0, j * 2)).func_177230_c() == Blocks.bookshelf)
+                                if (this.worldPointer.getBlockState(this.field_178150_j.func_177982_a(k * 2, 0, j * 2)).getBlock() == Blocks.bookshelf)
                                 {
                                     ++l;
                                 }
 
-                                if (this.worldPointer.func_180495_p(this.field_178150_j.func_177982_a(k * 2, 1, j * 2)).func_177230_c() == Blocks.bookshelf)
+                                if (this.worldPointer.getBlockState(this.field_178150_j.func_177982_a(k * 2, 1, j * 2)).getBlock() == Blocks.bookshelf)
                                 {
                                     ++l;
                                 }
 
                                 if (k != 0 && j != 0)
                                 {
-                                    if (this.worldPointer.func_180495_p(this.field_178150_j.func_177982_a(k * 2, 0, j)).func_177230_c() == Blocks.bookshelf)
+                                    if (this.worldPointer.getBlockState(this.field_178150_j.func_177982_a(k * 2, 0, j)).getBlock() == Blocks.bookshelf)
                                     {
                                         ++l;
                                     }
 
-                                    if (this.worldPointer.func_180495_p(this.field_178150_j.func_177982_a(k * 2, 1, j)).func_177230_c() == Blocks.bookshelf)
+                                    if (this.worldPointer.getBlockState(this.field_178150_j.func_177982_a(k * 2, 1, j)).getBlock() == Blocks.bookshelf)
                                     {
                                         ++l;
                                     }
 
-                                    if (this.worldPointer.func_180495_p(this.field_178150_j.func_177982_a(k, 0, j * 2)).func_177230_c() == Blocks.bookshelf)
+                                    if (this.worldPointer.getBlockState(this.field_178150_j.func_177982_a(k, 0, j * 2)).getBlock() == Blocks.bookshelf)
                                     {
                                         ++l;
                                     }
 
-                                    if (this.worldPointer.func_180495_p(this.field_178150_j.func_177982_a(k, 1, j * 2)).func_177230_c() == Blocks.bookshelf)
+                                    if (this.worldPointer.getBlockState(this.field_178150_j.func_177982_a(k, 1, j * 2)).getBlock() == Blocks.bookshelf)
                                     {
                                         ++l;
                                     }
@@ -363,7 +363,7 @@
 
     public boolean canInteractWith(EntityPlayer player)
     {
-        return this.worldPointer.func_180495_p(this.field_178150_j).func_177230_c() != Blocks.enchanting_table ? false : player.getDistanceSq((double)this.field_178150_j.func_177958_n() + 0.5D, (double)this.field_178150_j.func_177956_o() + 0.5D, (double)this.field_178150_j.func_177952_p() + 0.5D) <= 64.0D;
+        return this.worldPointer.getBlockState(this.field_178150_j).getBlock() != Blocks.enchanting_table ? false : player.getDistanceSq((double)this.field_178150_j.getX() + 0.5D, (double)this.field_178150_j.getY() + 0.5D, (double)this.field_178150_j.getZ() + 0.5D) <= 64.0D;
     }
 
     @Nullable
