--- a/net/minecraft/village/Village.java
+++ b/net/minecraft/village/Village.java
@@ -33,10 +33,10 @@
      * This is the sum of all door coordinates and used to calculate the actual village center by dividing by the number
      * of doors.
      */
-    private BlockPos centerHelper = BlockPos.field_177992_a;
+    private BlockPos centerHelper = BlockPos.createDefault;
 
     /** This is the actual village center. */
-    private BlockPos center = BlockPos.field_177992_a;
+    private BlockPos center = BlockPos.createDefault;
     private int villageRadius;
     private int lastAddDoorTimestamp;
     private int tickCounter;
@@ -105,7 +105,7 @@
 
             if (this.func_179866_a(blockpos) && this.func_179861_a(new BlockPos(p_179862_2_, p_179862_3_, p_179862_4_), blockpos))
             {
-                return new Vec3d((double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p());
+                return new Vec3d((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ());
             }
         }
 
@@ -114,22 +114,22 @@
 
     private boolean func_179861_a(BlockPos p_179861_1_, BlockPos p_179861_2_)
     {
-        if (!this.worldObj.func_180495_p(p_179861_2_.func_177977_b()).func_185896_q())
+        if (!this.worldObj.getBlockState(p_179861_2_.getBlockPosBelow()).func_185896_q())
         {
             return false;
         }
         else
         {
-            int i = p_179861_2_.func_177958_n() - p_179861_1_.func_177958_n() / 2;
-            int j = p_179861_2_.func_177952_p() - p_179861_1_.func_177952_p() / 2;
+            int i = p_179861_2_.getX() - p_179861_1_.getX() / 2;
+            int j = p_179861_2_.getZ() - p_179861_1_.getZ() / 2;
 
-            for (int k = i; k < i + p_179861_1_.func_177958_n(); ++k)
+            for (int k = i; k < i + p_179861_1_.getX(); ++k)
             {
-                for (int l = p_179861_2_.func_177956_o(); l < p_179861_2_.func_177956_o() + p_179861_1_.func_177956_o(); ++l)
+                for (int l = p_179861_2_.getY(); l < p_179861_2_.getY() + p_179861_1_.getY(); ++l)
                 {
-                    for (int i1 = j; i1 < j + p_179861_1_.func_177952_p(); ++i1)
+                    for (int i1 = j; i1 < j + p_179861_1_.getZ(); ++i1)
                     {
-                        if (this.worldObj.func_180495_p(new BlockPos(k, l, i1)).func_185915_l())
+                        if (this.worldObj.getBlockState(new BlockPos(k, l, i1)).func_185915_l())
                         {
                             return false;
                         }
@@ -143,13 +143,13 @@
 
     private void updateNumIronGolems()
     {
-        List<EntityIronGolem> list = this.worldObj.<EntityIronGolem>getEntitiesWithinAABB(EntityIronGolem.class, new AxisAlignedBB((double)(this.center.func_177958_n() - this.villageRadius), (double)(this.center.func_177956_o() - 4), (double)(this.center.func_177952_p() - this.villageRadius), (double)(this.center.func_177958_n() + this.villageRadius), (double)(this.center.func_177956_o() + 4), (double)(this.center.func_177952_p() + this.villageRadius)));
+        List<EntityIronGolem> list = this.worldObj.<EntityIronGolem>getEntitiesWithinAABB(EntityIronGolem.class, new AxisAlignedBB((double)(this.center.getX() - this.villageRadius), (double)(this.center.getY() - 4), (double)(this.center.getZ() - this.villageRadius), (double)(this.center.getX() + this.villageRadius), (double)(this.center.getY() + 4), (double)(this.center.getZ() + this.villageRadius)));
         this.numIronGolems = list.size();
     }
 
     private void updateNumVillagers()
     {
-        List<EntityVillager> list = this.worldObj.<EntityVillager>getEntitiesWithinAABB(EntityVillager.class, new AxisAlignedBB((double)(this.center.func_177958_n() - this.villageRadius), (double)(this.center.func_177956_o() - 4), (double)(this.center.func_177952_p() - this.villageRadius), (double)(this.center.func_177958_n() + this.villageRadius), (double)(this.center.func_177956_o() + 4), (double)(this.center.func_177952_p() + this.villageRadius)));
+        List<EntityVillager> list = this.worldObj.<EntityVillager>getEntitiesWithinAABB(EntityVillager.class, new AxisAlignedBB((double)(this.center.getX() - this.villageRadius), (double)(this.center.getY() - 4), (double)(this.center.getZ() - this.villageRadius), (double)(this.center.getX() + this.villageRadius), (double)(this.center.getY() + 4), (double)(this.center.getZ() + this.villageRadius)));
         this.numVillagers = list.size();
 
         if (this.numVillagers == 0)
@@ -239,7 +239,7 @@
                 BlockPos blockpos = villagedoorinfo1.func_179852_d();
                 EnumFacing enumfacing = villagedoorinfo1.func_188567_j();
 
-                if (this.worldObj.func_180495_p(blockpos.func_177967_a(enumfacing, 1)).func_177230_c().func_176205_b(this.worldObj, blockpos.func_177967_a(enumfacing, 1)) && this.worldObj.func_180495_p(blockpos.func_177967_a(enumfacing, -1)).func_177230_c().func_176205_b(this.worldObj, blockpos.func_177967_a(enumfacing, -1)) && this.worldObj.func_180495_p(blockpos.func_177984_a().func_177967_a(enumfacing, 1)).func_177230_c().func_176205_b(this.worldObj, blockpos.func_177984_a().func_177967_a(enumfacing, 1)) && this.worldObj.func_180495_p(blockpos.func_177984_a().func_177967_a(enumfacing, -1)).func_177230_c().func_176205_b(this.worldObj, blockpos.func_177984_a().func_177967_a(enumfacing, -1)))
+                if (this.worldObj.getBlockState(blockpos.getBlockPosRelative(enumfacing, 1)).getBlock().func_176205_b(this.worldObj, blockpos.getBlockPosRelative(enumfacing, 1)) && this.worldObj.getBlockState(blockpos.getBlockPosRelative(enumfacing, -1)).getBlock().func_176205_b(this.worldObj, blockpos.getBlockPosRelative(enumfacing, -1)) && this.worldObj.getBlockState(blockpos.getBlockPosAbove().getBlockPosRelative(enumfacing, 1)).getBlock().func_176205_b(this.worldObj, blockpos.getBlockPosAbove().getBlockPosRelative(enumfacing, 1)) && this.worldObj.getBlockState(blockpos.getBlockPosAbove().getBlockPosRelative(enumfacing, -1)).getBlock().func_176205_b(this.worldObj, blockpos.getBlockPosAbove().getBlockPosRelative(enumfacing, -1)))
                 {
                     villagedoorinfo = villagedoorinfo1;
                     i = j;
@@ -260,7 +260,7 @@
         {
             for (VillageDoorInfo villagedoorinfo : this.villageDoorInfoList)
             {
-                if (villagedoorinfo.func_179852_d().func_177958_n() == p_179864_1_.func_177958_n() && villagedoorinfo.func_179852_d().func_177952_p() == p_179864_1_.func_177952_p() && Math.abs(villagedoorinfo.func_179852_d().func_177956_o() - p_179864_1_.func_177956_o()) <= 1)
+                if (villagedoorinfo.func_179852_d().getX() == p_179864_1_.getX() && villagedoorinfo.func_179852_d().getZ() == p_179864_1_.getZ() && Math.abs(villagedoorinfo.func_179852_d().getY() - p_179864_1_.getY()) <= 1)
                 {
                     return villagedoorinfo;
                 }
@@ -394,9 +394,9 @@
 
     private boolean func_179860_f(BlockPos p_179860_1_)
     {
-        IBlockState iblockstate = this.worldObj.func_180495_p(p_179860_1_);
-        Block block = iblockstate.func_177230_c();
-        return block instanceof BlockDoor ? iblockstate.func_185904_a() == Material.wood : false;
+        IBlockState iblockstate = this.worldObj.getBlockState(p_179860_1_);
+        Block block = iblockstate.getBlock();
+        return block instanceof BlockDoor ? iblockstate.getMaterial() == Material.wood : false;
     }
 
     private void updateVillageRadiusAndCenter()
@@ -405,12 +405,12 @@
 
         if (i == 0)
         {
-            this.center = BlockPos.field_177992_a;
+            this.center = BlockPos.createDefault;
             this.villageRadius = 0;
         }
         else
         {
-            this.center = new BlockPos(this.centerHelper.func_177958_n() / i, this.centerHelper.func_177956_o() / i, this.centerHelper.func_177952_p() / i);
+            this.center = new BlockPos(this.centerHelper.getX() / i, this.centerHelper.getY() / i, this.centerHelper.getZ() / i);
             int j = 0;
 
             for (VillageDoorInfo villagedoorinfo : this.villageDoorInfoList)
@@ -506,20 +506,20 @@
         p_82689_1_.setInteger("Stable", this.lastAddDoorTimestamp);
         p_82689_1_.setInteger("Tick", this.tickCounter);
         p_82689_1_.setInteger("MTick", this.noBreedTicks);
-        p_82689_1_.setInteger("CX", this.center.func_177958_n());
-        p_82689_1_.setInteger("CY", this.center.func_177956_o());
-        p_82689_1_.setInteger("CZ", this.center.func_177952_p());
-        p_82689_1_.setInteger("ACX", this.centerHelper.func_177958_n());
-        p_82689_1_.setInteger("ACY", this.centerHelper.func_177956_o());
-        p_82689_1_.setInteger("ACZ", this.centerHelper.func_177952_p());
+        p_82689_1_.setInteger("CX", this.center.getX());
+        p_82689_1_.setInteger("CY", this.center.getY());
+        p_82689_1_.setInteger("CZ", this.center.getZ());
+        p_82689_1_.setInteger("ACX", this.centerHelper.getX());
+        p_82689_1_.setInteger("ACY", this.centerHelper.getY());
+        p_82689_1_.setInteger("ACZ", this.centerHelper.getZ());
         NBTTagList nbttaglist = new NBTTagList();
 
         for (VillageDoorInfo villagedoorinfo : this.villageDoorInfoList)
         {
             NBTTagCompound nbttagcompound = new NBTTagCompound();
-            nbttagcompound.setInteger("X", villagedoorinfo.func_179852_d().func_177958_n());
-            nbttagcompound.setInteger("Y", villagedoorinfo.func_179852_d().func_177956_o());
-            nbttagcompound.setInteger("Z", villagedoorinfo.func_179852_d().func_177952_p());
+            nbttagcompound.setInteger("X", villagedoorinfo.func_179852_d().getX());
+            nbttagcompound.setInteger("Y", villagedoorinfo.func_179852_d().getY());
+            nbttagcompound.setInteger("Z", villagedoorinfo.func_179852_d().getZ());
             nbttagcompound.setInteger("IDX", villagedoorinfo.func_179847_f());
             nbttagcompound.setInteger("IDZ", villagedoorinfo.func_179855_g());
             nbttagcompound.setInteger("TS", villagedoorinfo.getInsidePosY());
