--- a/net/minecraft/pathfinding/PathNavigateGround.java
+++ b/net/minecraft/pathfinding/PathNavigateGround.java
@@ -32,7 +32,7 @@
      */
     protected boolean canNavigate()
     {
-        return this.theEntity.onGround || this.func_179684_h() && this.isInLiquid() || this.theEntity.func_184218_aH();
+        return this.theEntity.onGround || this.func_179684_h() && this.isInLiquid() || this.theEntity.isRiding();
     }
 
     protected Vec3d getEntityPosition()
@@ -42,29 +42,29 @@
 
     public Path func_179680_a(BlockPos p_179680_1_)
     {
-        if (this.worldObj.func_180495_p(p_179680_1_).func_185904_a() == Material.air)
+        if (this.worldObj.getBlockState(p_179680_1_).getMaterial() == Material.air)
         {
             BlockPos blockpos;
 
-            for (blockpos = p_179680_1_.func_177977_b(); blockpos.func_177956_o() > 0 && this.worldObj.func_180495_p(blockpos).func_185904_a() == Material.air; blockpos = blockpos.func_177977_b())
+            for (blockpos = p_179680_1_.getBlockPosBelow(); blockpos.getY() > 0 && this.worldObj.getBlockState(blockpos).getMaterial() == Material.air; blockpos = blockpos.getBlockPosBelow())
             {
                 ;
             }
 
-            if (blockpos.func_177956_o() > 0)
+            if (blockpos.getY() > 0)
             {
-                return super.func_179680_a(blockpos.func_177984_a());
+                return super.func_179680_a(blockpos.getBlockPosAbove());
             }
 
-            while (blockpos.func_177956_o() < this.worldObj.getHeight() && this.worldObj.func_180495_p(blockpos).func_185904_a() == Material.air)
+            while (blockpos.getY() < this.worldObj.getHeight() && this.worldObj.getBlockState(blockpos).getMaterial() == Material.air)
             {
-                blockpos = blockpos.func_177984_a();
+                blockpos = blockpos.getBlockPosAbove();
             }
 
             p_179680_1_ = blockpos;
         }
 
-        if (!this.worldObj.func_180495_p(p_179680_1_).func_185904_a().isSolid())
+        if (!this.worldObj.getBlockState(p_179680_1_).getMaterial().isSolid())
         {
             return super.func_179680_a(p_179680_1_);
         }
@@ -72,7 +72,7 @@
         {
             BlockPos blockpos1;
 
-            for (blockpos1 = p_179680_1_.func_177984_a(); blockpos1.func_177956_o() < this.worldObj.getHeight() && this.worldObj.func_180495_p(blockpos1).func_185904_a().isSolid(); blockpos1 = blockpos1.func_177984_a())
+            for (blockpos1 = p_179680_1_.getBlockPosAbove(); blockpos1.getY() < this.worldObj.getHeight() && this.worldObj.getBlockState(blockpos1).getMaterial().isSolid(); blockpos1 = blockpos1.getBlockPosAbove())
             {
                 ;
             }
@@ -95,13 +95,13 @@
         if (this.theEntity.isInWater() && this.func_179684_h())
         {
             int i = (int)this.theEntity.func_174813_aQ().minY;
-            Block block = this.worldObj.func_180495_p(new BlockPos(MathHelper.floor_double(this.theEntity.posX), i, MathHelper.floor_double(this.theEntity.posZ))).func_177230_c();
+            Block block = this.worldObj.getBlockState(new BlockPos(MathHelper.floor_double(this.theEntity.posX), i, MathHelper.floor_double(this.theEntity.posZ))).getBlock();
             int j = 0;
 
             while (block == Blocks.flowing_water || block == Blocks.water)
             {
                 ++i;
-                block = this.worldObj.func_180495_p(new BlockPos(MathHelper.floor_double(this.theEntity.posX), i, MathHelper.floor_double(this.theEntity.posZ))).func_177230_c();
+                block = this.worldObj.getBlockState(new BlockPos(MathHelper.floor_double(this.theEntity.posX), i, MathHelper.floor_double(this.theEntity.posZ))).getBlock();
                 ++j;
 
                 if (j > 16)
@@ -129,8 +129,8 @@
         {
             PathPoint pathpoint = this.currentPath.getPathPointFromIndex(i);
             PathPoint pathpoint1 = i + 1 < this.currentPath.getCurrentPathLength() ? this.currentPath.getPathPointFromIndex(i + 1) : null;
-            IBlockState iblockstate = this.worldObj.func_180495_p(new BlockPos(pathpoint.xCoord, pathpoint.yCoord, pathpoint.zCoord));
-            Block block = iblockstate.func_177230_c();
+            IBlockState iblockstate = this.worldObj.getBlockState(new BlockPos(pathpoint.xCoord, pathpoint.yCoord, pathpoint.zCoord));
+            Block block = iblockstate.getBlock();
 
             if (block == Blocks.cauldron)
             {
@@ -304,14 +304,14 @@
 
     private boolean func_179692_b(int p_179692_1_, int p_179692_2_, int p_179692_3_, int p_179692_4_, int p_179692_5_, int p_179692_6_, Vec3d p_179692_7_, double p_179692_8_, double p_179692_10_)
     {
-        for (BlockPos blockpos : BlockPos.func_177980_a(new BlockPos(p_179692_1_, p_179692_2_, p_179692_3_), new BlockPos(p_179692_1_ + p_179692_4_ - 1, p_179692_2_ + p_179692_5_ - 1, p_179692_3_ + p_179692_6_ - 1)))
+        for (BlockPos blockpos : BlockPos.getBlocksBetween(new BlockPos(p_179692_1_, p_179692_2_, p_179692_3_), new BlockPos(p_179692_1_ + p_179692_4_ - 1, p_179692_2_ + p_179692_5_ - 1, p_179692_3_ + p_179692_6_ - 1)))
         {
-            double d0 = (double)blockpos.func_177958_n() + 0.5D - p_179692_7_.xCoord;
-            double d1 = (double)blockpos.func_177952_p() + 0.5D - p_179692_7_.zCoord;
+            double d0 = (double)blockpos.getX() + 0.5D - p_179692_7_.xCoord;
+            double d1 = (double)blockpos.getZ() + 0.5D - p_179692_7_.zCoord;
 
             if (d0 * p_179692_8_ + d1 * p_179692_10_ >= 0.0D)
             {
-                Block block = this.worldObj.func_180495_p(blockpos).func_177230_c();
+                Block block = this.worldObj.getBlockState(blockpos).getBlock();
 
                 if (!block.func_176205_b(this.worldObj, blockpos))
                 {
