--- a/net/minecraft/pathfinding/PathWorldListener.java
+++ b/net/minecraft/pathfinding/PathWorldListener.java
@@ -67,7 +67,7 @@
     {
     }
 
-    public void func_184375_a(@Nullable EntityPlayer p_184375_1_, SoundEvent p_184375_2_, SoundCategory p_184375_3_, double p_184375_4_, double p_184375_6_, double p_184375_8_, float p_184375_10_, float p_184375_11_)
+    public void playSound(@Nullable EntityPlayer p_184375_1_, SoundEvent p_184375_2_, SoundCategory p_184375_3_, double p_184375_4_, double p_184375_6_, double p_184375_8_, float p_184375_10_, float p_184375_11_)
     {
     }
 
