--- a/net/minecraft/pathfinding/PathFinder.java
+++ b/net/minecraft/pathfinding/PathFinder.java
@@ -32,7 +32,7 @@
     @Nullable
     public Path func_186336_a(IBlockAccess p_186336_1_, EntityLiving p_186336_2_, BlockPos p_186336_3_, float p_186336_4_)
     {
-        return this.func_186334_a(p_186336_1_, p_186336_2_, (double)((float)p_186336_3_.func_177958_n() + 0.5F), (double)((float)p_186336_3_.func_177956_o() + 0.5F), (double)((float)p_186336_3_.func_177952_p() + 0.5F), p_186336_4_);
+        return this.func_186334_a(p_186336_1_, p_186336_2_, (double)((float)p_186336_3_.getX() + 0.5F), (double)((float)p_186336_3_.getY() + 0.5F), (double)((float)p_186336_3_.getZ() + 0.5F), p_186336_4_);
     }
 
     @Nullable
