diff -r -U 3 minecraft\net\minecraft\client\renderer\block\model\ModelBakery.java minecraft_patched\net\minecraft\client\renderer\block\model\ModelBakery.java
--- minecraft\net\minecraft\client\renderer\block\model\ModelBakery.java
+++ minecraft_patched\net\minecraft\client\renderer\block\model\ModelBakery.java
@@ -259,7 +259,7 @@
                reader = new InputStreamReader(iresource.func_110527_b(), Charsets.UTF_8);
             }
 
-            lvt_5_2_ = ModelBlock.func_178307_a(reader);
+            lvt_5_1_ = ModelBlock.func_178307_a(reader);
             lvt_5_1_.field_178317_b = p_177594_1_.toString();
             ModelBlock modelblock1 = lvt_5_1_;
             return modelblock1;
@@ -395,7 +395,7 @@
 
          IBakedModel ibakedmodel2 = multipartbakedmodel$builder.func_188647_a();
 
-         for(ModelResourceLocation modelresourcelocation1 : (Collection)entry.getValue()) {
+         for(ModelResourceLocation modelresourcelocation1 : entry.getValue()) {
             if(!modelblockdefinition.func_188000_b(modelresourcelocation1.func_177518_c())) {
                this.field_177605_n.func_82595_a(modelresourcelocation1, ibakedmodel2);
             }
