--- a/net/minecraft/world/storage/MapData.java
+++ b/net/minecraft/world/storage/MapData.java
@@ -152,7 +152,7 @@
         {
             EntityItemFrame entityitemframe = p_76191_2_.getItemFrame();
             BlockPos blockpos = entityitemframe.func_174857_n();
-            this.func_82567_a(1, p_76191_1_.worldObj, "frame-" + entityitemframe.getEntityId(), (double)blockpos.func_177958_n(), (double)blockpos.func_177952_p(), (double)(entityitemframe.field_174860_b.func_176736_b() * 90));
+            this.func_82567_a(1, p_76191_1_.worldObj, "frame-" + entityitemframe.getEntityId(), (double)blockpos.getX(), (double)blockpos.getZ(), (double)(entityitemframe.field_174860_b.func_176736_b() * 90));
         }
 
         if (p_76191_2_.hasTagCompound() && p_76191_2_.getTagCompound().hasKey("Decorations", 9))
