--- a/net/minecraft/world/storage/WorldInfo.java
+++ b/net/minecraft/world/storage/WorldInfo.java
@@ -524,9 +524,9 @@
 
     public void func_176143_a(BlockPos p_176143_1_)
     {
-        this.spawnX = p_176143_1_.func_177958_n();
-        this.spawnY = p_176143_1_.func_177956_o();
-        this.spawnZ = p_176143_1_.func_177952_p();
+        this.spawnX = p_176143_1_.getX();
+        this.spawnY = p_176143_1_.getY();
+        this.spawnZ = p_176143_1_.getZ();
     }
 
     /**
