--- a/net/minecraft/entity/ai/EntityAIFleeSun.java
+++ b/net/minecraft/entity/ai/EntityAIFleeSun.java
@@ -42,7 +42,7 @@
         {
             return false;
         }
-        else if (this.theCreature.func_184582_a(EntityEquipmentSlot.HEAD) != null)
+        else if (this.theCreature.getEquippedItemStack(EntityEquipmentSlot.HEAD) != null)
         {
             return false;
         }
@@ -92,7 +92,7 @@
 
             if (!this.theWorld.func_175678_i(blockpos1) && this.theCreature.func_180484_a(blockpos1) < 0.0F)
             {
-                return new Vec3d((double)blockpos1.func_177958_n(), (double)blockpos1.func_177956_o(), (double)blockpos1.func_177952_p());
+                return new Vec3d((double)blockpos1.getX(), (double)blockpos1.getY(), (double)blockpos1.getZ());
             }
         }
 
