--- a/net/minecraft/entity/ai/EntityAITempt.java
+++ b/net/minecraft/entity/ai/EntityAITempt.java
@@ -76,7 +76,7 @@
         else
         {
             this.temptingPlayer = this.temptedEntity.worldObj.getClosestPlayerToEntity(this.temptedEntity, 10.0D);
-            return this.temptingPlayer == null ? false : this.func_188508_a(this.temptingPlayer.func_184614_ca()) || this.func_188508_a(this.temptingPlayer.func_184592_cb());
+            return this.temptingPlayer == null ? false : this.func_188508_a(this.temptingPlayer.getMainHandItem()) || this.func_188508_a(this.temptingPlayer.getOffhandItem());
         }
     }
 
