--- a/net/minecraft/entity/ai/EntityAIRestrictSun.java
+++ b/net/minecraft/entity/ai/EntityAIRestrictSun.java
@@ -18,7 +18,7 @@
      */
     public boolean shouldExecute()
     {
-        return this.theEntity.worldObj.isDaytime() && this.theEntity.func_184582_a(EntityEquipmentSlot.HEAD) == null;
+        return this.theEntity.worldObj.isDaytime() && this.theEntity.getEquippedItemStack(EntityEquipmentSlot.HEAD) == null;
     }
 
     /**
