--- a/net/minecraft/entity/ai/EntityAIMoveIndoors.java
+++ b/net/minecraft/entity/ai/EntityAIMoveIndoors.java
@@ -72,9 +72,9 @@
     {
         this.insidePosX = -1;
         BlockPos blockpos = this.doorInfo.func_179856_e();
-        int i = blockpos.func_177958_n();
-        int j = blockpos.func_177956_o();
-        int k = blockpos.func_177952_p();
+        int i = blockpos.getX();
+        int j = blockpos.getY();
+        int k = blockpos.getZ();
 
         if (this.entityObj.func_174818_b(blockpos) > 256.0D)
         {
@@ -96,8 +96,8 @@
      */
     public void resetTask()
     {
-        this.insidePosX = this.doorInfo.func_179856_e().func_177958_n();
-        this.insidePosZ = this.doorInfo.func_179856_e().func_177952_p();
+        this.insidePosX = this.doorInfo.func_179856_e().getX();
+        this.insidePosZ = this.doorInfo.func_179856_e().getZ();
         this.doorInfo = null;
     }
 }
