--- a/net/minecraft/entity/ai/EntityAIEatGrass.java
+++ b/net/minecraft/entity/ai/EntityAIEatGrass.java
@@ -43,7 +43,7 @@
         else
         {
             BlockPos blockpos = new BlockPos(this.grassEaterEntity.posX, this.grassEaterEntity.posY, this.grassEaterEntity.posZ);
-            return field_179505_b.apply(this.entityWorld.func_180495_p(blockpos)) ? true : this.entityWorld.func_180495_p(blockpos.func_177977_b()).func_177230_c() == Blocks.grass;
+            return field_179505_b.apply(this.entityWorld.getBlockState(blockpos)) ? true : this.entityWorld.getBlockState(blockpos.getBlockPosBelow()).getBlock() == Blocks.grass;
         }
     }
 
@@ -92,7 +92,7 @@
         {
             BlockPos blockpos = new BlockPos(this.grassEaterEntity.posX, this.grassEaterEntity.posY, this.grassEaterEntity.posZ);
 
-            if (field_179505_b.apply(this.entityWorld.func_180495_p(blockpos)))
+            if (field_179505_b.apply(this.entityWorld.getBlockState(blockpos)))
             {
                 if (this.entityWorld.getGameRules().getGameRuleBooleanValue("mobGriefing"))
                 {
@@ -103,14 +103,14 @@
             }
             else
             {
-                BlockPos blockpos1 = blockpos.func_177977_b();
+                BlockPos blockpos1 = blockpos.getBlockPosBelow();
 
-                if (this.entityWorld.func_180495_p(blockpos1).func_177230_c() == Blocks.grass)
+                if (this.entityWorld.getBlockState(blockpos1).getBlock() == Blocks.grass)
                 {
                     if (this.entityWorld.getGameRules().getGameRuleBooleanValue("mobGriefing"))
                     {
                         this.entityWorld.func_175718_b(2001, blockpos1, Block.getIdFromBlock(Blocks.grass));
-                        this.entityWorld.func_180501_a(blockpos1, Blocks.dirt.func_176223_P(), 2);
+                        this.entityWorld.func_180501_a(blockpos1, Blocks.dirt.getBlockState(), 2);
                     }
 
                     this.grassEaterEntity.eatGrassBonus();
