--- a/net/minecraft/entity/ai/EntityAIMoveToBlock.java
+++ b/net/minecraft/entity/ai/EntityAIMoveToBlock.java
@@ -11,7 +11,7 @@
     protected int field_179496_a;
     private int field_179493_e;
     private int field_179490_f;
-    protected BlockPos field_179494_b = BlockPos.field_177992_a;
+    protected BlockPos field_179494_b = BlockPos.createDefault;
     private boolean field_179491_g;
     private final int field_179497_h;
 
@@ -53,7 +53,7 @@
      */
     public void startExecuting()
     {
-        this.field_179495_c.getNavigator().tryMoveToXYZ((double)((float)this.field_179494_b.func_177958_n()) + 0.5D, (double)(this.field_179494_b.func_177956_o() + 1), (double)((float)this.field_179494_b.func_177952_p()) + 0.5D, this.field_179492_d);
+        this.field_179495_c.getNavigator().tryMoveToXYZ((double)((float)this.field_179494_b.getX()) + 0.5D, (double)(this.field_179494_b.getY() + 1), (double)((float)this.field_179494_b.getZ()) + 0.5D, this.field_179492_d);
         this.field_179493_e = 0;
         this.field_179490_f = this.field_179495_c.getRNG().nextInt(this.field_179495_c.getRNG().nextInt(1200) + 1200) + 1200;
     }
@@ -70,14 +70,14 @@
      */
     public void updateTask()
     {
-        if (this.field_179495_c.func_174831_c(this.field_179494_b.func_177984_a()) > 1.0D)
+        if (this.field_179495_c.func_174831_c(this.field_179494_b.getBlockPosAbove()) > 1.0D)
         {
             this.field_179491_g = false;
             ++this.field_179493_e;
 
             if (this.field_179493_e % 40 == 0)
             {
-                this.field_179495_c.getNavigator().tryMoveToXYZ((double)((float)this.field_179494_b.func_177958_n()) + 0.5D, (double)(this.field_179494_b.func_177956_o() + 1), (double)((float)this.field_179494_b.func_177952_p()) + 0.5D, this.field_179492_d);
+                this.field_179495_c.getNavigator().tryMoveToXYZ((double)((float)this.field_179494_b.getX()) + 0.5D, (double)(this.field_179494_b.getY() + 1), (double)((float)this.field_179494_b.getZ()) + 0.5D, this.field_179492_d);
             }
         }
         else
