--- a/net/minecraft/entity/ai/EntityAIFollowOwner.java
+++ b/net/minecraft/entity/ai/EntityAIFollowOwner.java
@@ -100,8 +100,8 @@
 
     private boolean func_181065_a(BlockPos p_181065_1_)
     {
-        IBlockState iblockstate = this.theWorld.func_180495_p(p_181065_1_);
-        return iblockstate.func_185904_a() == Material.air ? true : !iblockstate.func_185917_h();
+        IBlockState iblockstate = this.theWorld.getBlockState(p_181065_1_);
+        return iblockstate.getMaterial() == Material.air ? true : !iblockstate.func_185917_h();
     }
 
     /**
@@ -131,7 +131,7 @@
                             {
                                 for (int i1 = 0; i1 <= 4; ++i1)
                                 {
-                                    if ((l < 1 || i1 < 1 || l > 3 || i1 > 3) && this.theWorld.func_180495_p(new BlockPos(i + l, k - 1, j + i1)).func_185896_q() && this.func_181065_a(new BlockPos(i + l, k, j + i1)) && this.func_181065_a(new BlockPos(i + l, k + 1, j + i1)))
+                                    if ((l < 1 || i1 < 1 || l > 3 || i1 > 3) && this.theWorld.getBlockState(new BlockPos(i + l, k - 1, j + i1)).func_185896_q() && this.func_181065_a(new BlockPos(i + l, k, j + i1)) && this.func_181065_a(new BlockPos(i + l, k + 1, j + i1)))
                                     {
                                         this.thePet.setLocationAndAngles((double)((float)(i + l) + 0.5F), (double)k, (double)((float)(j + i1) + 0.5F), this.thePet.rotationYaw, this.thePet.rotationPitch);
                                         this.petPathfinder.clearPathEntity();
