--- a/net/minecraft/client/particle/ParticleManager.java
+++ b/net/minecraft/client/particle/ParticleManager.java
@@ -386,7 +386,7 @@
 
     public void func_180533_a(BlockPos p_180533_1_, IBlockState p_180533_2_)
     {
-        if (p_180533_2_.func_185904_a() != Material.air)
+        if (p_180533_2_.getMaterial() != Material.air)
         {
             p_180533_2_ = p_180533_2_.func_185899_b(this.worldObj, p_180533_1_);
             int i = 4;
@@ -397,10 +397,10 @@
                 {
                     for (int l = 0; l < 4; ++l)
                     {
-                        double d0 = (double)p_180533_1_.func_177958_n() + ((double)j + 0.5D) / 4.0D;
-                        double d1 = (double)p_180533_1_.func_177956_o() + ((double)k + 0.5D) / 4.0D;
-                        double d2 = (double)p_180533_1_.func_177952_p() + ((double)l + 0.5D) / 4.0D;
-                        this.addEffect((new ParticleDigging(this.worldObj, d0, d1, d2, d0 - (double)p_180533_1_.func_177958_n() - 0.5D, d1 - (double)p_180533_1_.func_177956_o() - 0.5D, d2 - (double)p_180533_1_.func_177952_p() - 0.5D, p_180533_2_)).func_174846_a(p_180533_1_));
+                        double d0 = (double)p_180533_1_.getX() + ((double)j + 0.5D) / 4.0D;
+                        double d1 = (double)p_180533_1_.getY() + ((double)k + 0.5D) / 4.0D;
+                        double d2 = (double)p_180533_1_.getZ() + ((double)l + 0.5D) / 4.0D;
+                        this.addEffect((new ParticleDigging(this.worldObj, d0, d1, d2, d0 - (double)p_180533_1_.getX() - 0.5D, d1 - (double)p_180533_1_.getY() - 0.5D, d2 - (double)p_180533_1_.getZ() - 0.5D, p_180533_2_)).func_174846_a(p_180533_1_));
                     }
                 }
             }
@@ -409,13 +409,13 @@
 
     public void func_180532_a(BlockPos p_180532_1_, EnumFacing p_180532_2_)
     {
-        IBlockState iblockstate = this.worldObj.func_180495_p(p_180532_1_);
+        IBlockState iblockstate = this.worldObj.getBlockState(p_180532_1_);
 
         if (iblockstate.func_185901_i() != EnumBlockRenderType.INVISIBLE)
         {
-            int i = p_180532_1_.func_177958_n();
-            int j = p_180532_1_.func_177956_o();
-            int k = p_180532_1_.func_177952_p();
+            int i = p_180532_1_.getX();
+            int j = p_180532_1_.getY();
+            int k = p_180532_1_.getZ();
             float f = 0.1F;
             AxisAlignedBB axisalignedbb = iblockstate.func_185900_c(this.worldObj, p_180532_1_);
             double d0 = (double)i + this.rand.nextDouble() * (axisalignedbb.maxX - axisalignedbb.minX - 0.20000000298023224D) + 0.10000000149011612D + axisalignedbb.minX;
