--- a/net/minecraft/client/particle/ParticleFallingDust.java
+++ b/net/minecraft/client/particle/ParticleFallingDust.java
@@ -19,9 +19,9 @@
     protected ParticleFallingDust(World p_i47135_1_, double p_i47135_2_, double p_i47135_4_, double p_i47135_6_, float p_i47135_8_, float p_i47135_9_, float p_i47135_10_)
     {
         super(p_i47135_1_, p_i47135_2_, p_i47135_4_, p_i47135_6_, 0.0D, 0.0D, 0.0D);
-        this.field_187129_i = 0.0D;
-        this.field_187130_j = 0.0D;
-        this.field_187131_k = 0.0D;
+        this.motionX = 0.0D;
+        this.motionY = 0.0D;
+        this.motionZ = 0.0D;
         this.particleRed = p_i47135_8_;
         this.particleGreen = p_i47135_9_;
         this.particleBlue = p_i47135_10_;
@@ -45,9 +45,9 @@
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
 
         if (this.particleAge++ >= this.particleMaxAge)
         {
@@ -63,9 +63,9 @@
         }
 
         this.setParticleTextureIndex(7 - this.particleAge * 8 / this.particleMaxAge);
-        this.func_187110_a(this.field_187129_i, this.field_187130_j, this.field_187131_k);
-        this.field_187130_j -= 0.003000000026077032D;
-        this.field_187130_j = Math.max(this.field_187130_j, -0.14000000059604645D);
+        this.func_187110_a(this.motionX, this.motionY, this.motionZ);
+        this.motionY -= 0.003000000026077032D;
+        this.motionY = Math.max(this.motionY, -0.14000000059604645D);
     }
 
     public static class Factory implements IParticleFactory
@@ -74,7 +74,7 @@
         {
             IBlockState iblockstate = Block.func_176220_d(p_178902_15_[0]);
 
-            if (iblockstate.func_177230_c() != Blocks.air && iblockstate.func_185901_i() == EnumBlockRenderType.INVISIBLE)
+            if (iblockstate.getBlock() != Blocks.air && iblockstate.func_185901_i() == EnumBlockRenderType.INVISIBLE)
             {
                 return null;
             }
@@ -82,9 +82,9 @@
             {
                 int i = Minecraft.getMinecraft().func_184125_al().func_189991_a(iblockstate);
 
-                if (iblockstate.func_177230_c() instanceof BlockFalling)
+                if (iblockstate.getBlock() instanceof BlockFalling)
                 {
-                    i = ((BlockFalling)iblockstate.func_177230_c()).func_189876_x(iblockstate);
+                    i = ((BlockFalling)iblockstate.getBlock()).func_189876_x(iblockstate);
                 }
 
                 float f = (float)(i >> 16 & 255) / 255.0F;
