--- a/net/minecraft/client/particle/ParticleWaterWake.java
+++ b/net/minecraft/client/particle/ParticleWaterWake.java
@@ -7,9 +7,9 @@
     protected ParticleWaterWake(World p_i45073_1_, double p_i45073_2_, double p_i45073_4_, double p_i45073_6_, double p_i45073_8_, double p_i45073_10_, double p_i45073_12_)
     {
         super(p_i45073_1_, p_i45073_2_, p_i45073_4_, p_i45073_6_, 0.0D, 0.0D, 0.0D);
-        this.field_187129_i *= 0.30000001192092896D;
-        this.field_187130_j = Math.random() * 0.20000000298023224D + 0.10000000149011612D;
-        this.field_187131_k *= 0.30000001192092896D;
+        this.motionX *= 0.30000001192092896D;
+        this.motionY = Math.random() * 0.20000000298023224D + 0.10000000149011612D;
+        this.motionZ *= 0.30000001192092896D;
         this.particleRed = 1.0F;
         this.particleGreen = 1.0F;
         this.particleBlue = 1.0F;
@@ -17,21 +17,21 @@
         this.func_187115_a(0.01F, 0.01F);
         this.particleMaxAge = (int)(8.0D / (Math.random() * 0.8D + 0.2D));
         this.particleGravity = 0.0F;
-        this.field_187129_i = p_i45073_8_;
-        this.field_187130_j = p_i45073_10_;
-        this.field_187131_k = p_i45073_12_;
+        this.motionX = p_i45073_8_;
+        this.motionY = p_i45073_10_;
+        this.motionZ = p_i45073_12_;
     }
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
-        this.field_187130_j -= (double)this.particleGravity;
-        this.func_187110_a(this.field_187129_i, this.field_187130_j, this.field_187131_k);
-        this.field_187129_i *= 0.9800000190734863D;
-        this.field_187130_j *= 0.9800000190734863D;
-        this.field_187131_k *= 0.9800000190734863D;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
+        this.motionY -= (double)this.particleGravity;
+        this.func_187110_a(this.motionX, this.motionY, this.motionZ);
+        this.motionX *= 0.9800000190734863D;
+        this.motionY *= 0.9800000190734863D;
+        this.motionZ *= 0.9800000190734863D;
         int i = 60 - this.particleMaxAge;
         float f = (float)i * 0.001F;
         this.func_187115_a(f, f);
