--- a/net/minecraft/client/particle/ParticleSmokeNormal.java
+++ b/net/minecraft/client/particle/ParticleSmokeNormal.java
@@ -17,12 +17,12 @@
     protected ParticleSmokeNormal(World p_i46348_1_, double p_i46348_2_, double p_i46348_4_, double p_i46348_6_, double p_i46348_8_, double p_i46348_10_, double p_i46348_12_, float p_i46348_14_)
     {
         super(p_i46348_1_, p_i46348_2_, p_i46348_4_, p_i46348_6_, 0.0D, 0.0D, 0.0D);
-        this.field_187129_i *= 0.10000000149011612D;
-        this.field_187130_j *= 0.10000000149011612D;
-        this.field_187131_k *= 0.10000000149011612D;
-        this.field_187129_i += p_i46348_8_;
-        this.field_187130_j += p_i46348_10_;
-        this.field_187131_k += p_i46348_12_;
+        this.motionX *= 0.10000000149011612D;
+        this.motionY *= 0.10000000149011612D;
+        this.motionZ *= 0.10000000149011612D;
+        this.motionX += p_i46348_8_;
+        this.motionY += p_i46348_10_;
+        this.motionZ += p_i46348_12_;
         float f = (float)(Math.random() * 0.30000001192092896D);
         this.particleRed = f;
         this.particleGreen = f;
@@ -44,9 +44,9 @@
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
 
         if (this.particleAge++ >= this.particleMaxAge)
         {
@@ -54,23 +54,23 @@
         }
 
         this.setParticleTextureIndex(7 - this.particleAge * 8 / this.particleMaxAge);
-        this.field_187130_j += 0.004D;
-        this.func_187110_a(this.field_187129_i, this.field_187130_j, this.field_187131_k);
+        this.motionY += 0.004D;
+        this.func_187110_a(this.motionX, this.motionY, this.motionZ);
 
-        if (this.field_187127_g == this.field_187124_d)
+        if (this.prevPosY == this.posY)
         {
-            this.field_187129_i *= 1.1D;
-            this.field_187131_k *= 1.1D;
+            this.motionX *= 1.1D;
+            this.motionZ *= 1.1D;
         }
 
-        this.field_187129_i *= 0.9599999785423279D;
-        this.field_187130_j *= 0.9599999785423279D;
-        this.field_187131_k *= 0.9599999785423279D;
+        this.motionX *= 0.9599999785423279D;
+        this.motionY *= 0.9599999785423279D;
+        this.motionZ *= 0.9599999785423279D;
 
         if (this.field_187132_l)
         {
-            this.field_187129_i *= 0.699999988079071D;
-            this.field_187131_k *= 0.699999988079071D;
+            this.motionX *= 0.699999988079071D;
+            this.motionZ *= 0.699999988079071D;
         }
     }
 
