--- a/net/minecraft/client/particle/ParticleFootStep.java
+++ b/net/minecraft/client/particle/ParticleFootStep.java
@@ -22,9 +22,9 @@
     {
         super(p_i1210_2_, p_i1210_3_, p_i1210_5_, p_i1210_7_, 0.0D, 0.0D, 0.0D);
         this.currentFootSteps = p_i1210_1_;
-        this.field_187129_i = 0.0D;
-        this.field_187130_j = 0.0D;
-        this.field_187131_k = 0.0D;
+        this.motionX = 0.0D;
+        this.motionY = 0.0D;
+        this.motionZ = 0.0D;
         this.footstepMaxAge = 200;
     }
 
@@ -42,10 +42,10 @@
         f1 = f1 * 0.2F;
         GlStateManager.func_179140_f();
         float f2 = 0.125F;
-        float f3 = (float)(this.field_187126_f - interpPosX);
-        float f4 = (float)(this.field_187127_g - interpPosY);
-        float f5 = (float)(this.field_187128_h - interpPosZ);
-        float f6 = this.field_187122_b.func_175724_o(new BlockPos(this.field_187126_f, this.field_187127_g, this.field_187128_h));
+        float f3 = (float)(this.prevPosX - interpPosX);
+        float f4 = (float)(this.prevPosY - interpPosY);
+        float f5 = (float)(this.prevPosZ - interpPosZ);
+        float f6 = this.field_187122_b.func_175724_o(new BlockPos(this.prevPosX, this.prevPosY, this.prevPosZ));
         this.currentFootSteps.bindTexture(field_110126_a);
         GlStateManager.func_179147_l();
         GlStateManager.func_187401_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA);
