--- a/net/minecraft/client/particle/ParticleSnowShovel.java
+++ b/net/minecraft/client/particle/ParticleSnowShovel.java
@@ -17,12 +17,12 @@
     protected ParticleSnowShovel(World p_i1228_1_, double p_i1228_2_, double p_i1228_4_, double p_i1228_6_, double p_i1228_8_, double p_i1228_10_, double p_i1228_12_, float p_i1228_14_)
     {
         super(p_i1228_1_, p_i1228_2_, p_i1228_4_, p_i1228_6_, p_i1228_8_, p_i1228_10_, p_i1228_12_);
-        this.field_187129_i *= 0.10000000149011612D;
-        this.field_187130_j *= 0.10000000149011612D;
-        this.field_187131_k *= 0.10000000149011612D;
-        this.field_187129_i += p_i1228_8_;
-        this.field_187130_j += p_i1228_10_;
-        this.field_187131_k += p_i1228_12_;
+        this.motionX *= 0.10000000149011612D;
+        this.motionY *= 0.10000000149011612D;
+        this.motionZ *= 0.10000000149011612D;
+        this.motionX += p_i1228_8_;
+        this.motionY += p_i1228_10_;
+        this.motionZ += p_i1228_12_;
         float f = 1.0F - (float)(Math.random() * 0.30000001192092896D);
         this.particleRed = f;
         this.particleGreen = f;
@@ -44,9 +44,9 @@
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
 
         if (this.particleAge++ >= this.particleMaxAge)
         {
@@ -54,16 +54,16 @@
         }
 
         this.setParticleTextureIndex(7 - this.particleAge * 8 / this.particleMaxAge);
-        this.field_187130_j -= 0.03D;
-        this.func_187110_a(this.field_187129_i, this.field_187130_j, this.field_187131_k);
-        this.field_187129_i *= 0.9900000095367432D;
-        this.field_187130_j *= 0.9900000095367432D;
-        this.field_187131_k *= 0.9900000095367432D;
+        this.motionY -= 0.03D;
+        this.func_187110_a(this.motionX, this.motionY, this.motionZ);
+        this.motionX *= 0.9900000095367432D;
+        this.motionY *= 0.9900000095367432D;
+        this.motionZ *= 0.9900000095367432D;
 
         if (this.field_187132_l)
         {
-            this.field_187129_i *= 0.699999988079071D;
-            this.field_187131_k *= 0.699999988079071D;
+            this.motionX *= 0.699999988079071D;
+            this.motionZ *= 0.699999988079071D;
         }
     }
 
