--- a/net/minecraft/client/particle/ParticleBreaking.java
+++ b/net/minecraft/client/particle/ParticleBreaking.java
@@ -18,12 +18,12 @@
     protected ParticleBreaking(World p_i1197_1_, double p_i1197_2_, double p_i1197_4_, double p_i1197_6_, double p_i1197_8_, double p_i1197_10_, double p_i1197_12_, Item p_i1197_14_, int p_i1197_15_)
     {
         this(p_i1197_1_, p_i1197_2_, p_i1197_4_, p_i1197_6_, p_i1197_14_, p_i1197_15_);
-        this.field_187129_i *= 0.10000000149011612D;
-        this.field_187130_j *= 0.10000000149011612D;
-        this.field_187131_k *= 0.10000000149011612D;
-        this.field_187129_i += p_i1197_8_;
-        this.field_187130_j += p_i1197_10_;
-        this.field_187131_k += p_i1197_12_;
+        this.motionX *= 0.10000000149011612D;
+        this.motionY *= 0.10000000149011612D;
+        this.motionZ *= 0.10000000149011612D;
+        this.motionX += p_i1197_8_;
+        this.motionY += p_i1197_10_;
+        this.motionZ += p_i1197_12_;
     }
 
     protected ParticleBreaking(World p_i1196_1_, double p_i1196_2_, double p_i1196_4_, double p_i1196_6_, Item p_i1196_8_, int p_i1196_9_)
@@ -58,9 +58,9 @@
             f3 = this.field_187119_C.getInterpolatedV((double)((this.particleTextureJitterY + 1.0F) / 4.0F * 16.0F));
         }
 
-        float f5 = (float)(this.field_187123_c + (this.field_187126_f - this.field_187123_c) * (double)p_180434_3_ - interpPosX);
-        float f6 = (float)(this.field_187124_d + (this.field_187127_g - this.field_187124_d) * (double)p_180434_3_ - interpPosY);
-        float f7 = (float)(this.field_187125_e + (this.field_187128_h - this.field_187125_e) * (double)p_180434_3_ - interpPosZ);
+        float f5 = (float)(this.posX + (this.prevPosX - this.posX) * (double)p_180434_3_ - interpPosX);
+        float f6 = (float)(this.posY + (this.prevPosY - this.posY) * (double)p_180434_3_ - interpPosY);
+        float f7 = (float)(this.posZ + (this.prevPosZ - this.posZ) * (double)p_180434_3_ - interpPosZ);
         int i = this.func_189214_a(p_180434_3_);
         int j = i >> 16 & 65535;
         int k = i & 65535;
