--- a/net/minecraft/client/particle/ParticlePortal.java
+++ b/net/minecraft/client/particle/ParticlePortal.java
@@ -14,15 +14,15 @@
     protected ParticlePortal(World p_i46351_1_, double p_i46351_2_, double p_i46351_4_, double p_i46351_6_, double p_i46351_8_, double p_i46351_10_, double p_i46351_12_)
     {
         super(p_i46351_1_, p_i46351_2_, p_i46351_4_, p_i46351_6_, p_i46351_8_, p_i46351_10_, p_i46351_12_);
-        this.field_187129_i = p_i46351_8_;
-        this.field_187130_j = p_i46351_10_;
-        this.field_187131_k = p_i46351_12_;
-        this.field_187126_f = p_i46351_2_;
-        this.field_187127_g = p_i46351_4_;
-        this.field_187128_h = p_i46351_6_;
-        this.portalPosX = this.field_187126_f;
-        this.portalPosY = this.field_187127_g;
-        this.portalPosZ = this.field_187128_h;
+        this.motionX = p_i46351_8_;
+        this.motionY = p_i46351_10_;
+        this.motionZ = p_i46351_12_;
+        this.prevPosX = p_i46351_2_;
+        this.prevPosY = p_i46351_4_;
+        this.prevPosZ = p_i46351_6_;
+        this.portalPosX = this.prevPosX;
+        this.portalPosY = this.prevPosY;
+        this.portalPosZ = this.prevPosZ;
         float f = this.field_187136_p.nextFloat() * 0.6F + 0.4F;
         this.particleScale = this.field_187136_p.nextFloat() * 0.2F + 0.5F;
         this.portalParticleScale = this.particleScale;
@@ -69,15 +69,15 @@
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
         float f = (float)this.particleAge / (float)this.particleMaxAge;
         float f1 = -f + f * f * 2.0F;
         float f2 = 1.0F - f1;
-        this.field_187126_f = this.portalPosX + this.field_187129_i * (double)f2;
-        this.field_187127_g = this.portalPosY + this.field_187130_j * (double)f2 + (double)(1.0F - f);
-        this.field_187128_h = this.portalPosZ + this.field_187131_k * (double)f2;
+        this.prevPosX = this.portalPosX + this.motionX * (double)f2;
+        this.prevPosY = this.portalPosY + this.motionY * (double)f2 + (double)(1.0F - f);
+        this.prevPosZ = this.portalPosZ + this.motionZ * (double)f2;
 
         if (this.particleAge++ >= this.particleMaxAge)
         {
