--- a/net/minecraft/client/particle/ParticleSimpleAnimated.java
+++ b/net/minecraft/client/particle/ParticleSimpleAnimated.java
@@ -44,9 +44,9 @@
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
 
         if (this.particleAge++ >= this.particleMaxAge)
         {
@@ -66,16 +66,16 @@
         }
 
         this.setParticleTextureIndex(this.field_187147_a + (this.field_187148_G - 1 - this.particleAge * this.field_187148_G / this.particleMaxAge));
-        this.field_187130_j += (double)this.field_187149_H;
-        this.func_187110_a(this.field_187129_i, this.field_187130_j, this.field_187131_k);
-        this.field_187129_i *= 0.9100000262260437D;
-        this.field_187130_j *= 0.9100000262260437D;
-        this.field_187131_k *= 0.9100000262260437D;
+        this.motionY += (double)this.field_187149_H;
+        this.func_187110_a(this.motionX, this.motionY, this.motionZ);
+        this.motionX *= 0.9100000262260437D;
+        this.motionY *= 0.9100000262260437D;
+        this.motionZ *= 0.9100000262260437D;
 
         if (this.field_187132_l)
         {
-            this.field_187129_i *= 0.699999988079071D;
-            this.field_187131_k *= 0.699999988079071D;
+            this.motionX *= 0.699999988079071D;
+            this.motionZ *= 0.699999988079071D;
         }
     }
 
