--- a/net/minecraft/client/particle/ParticleEmitter.java
+++ b/net/minecraft/client/particle/ParticleEmitter.java
@@ -38,7 +38,7 @@
                 double d3 = this.field_174851_a.posX + d0 * (double)this.field_174851_a.width / 4.0D;
                 double d4 = this.field_174851_a.func_174813_aQ().minY + (double)(this.field_174851_a.height / 2.0F) + d1 * (double)this.field_174851_a.height / 4.0D;
                 double d5 = this.field_174851_a.posZ + d2 * (double)this.field_174851_a.width / 4.0D;
-                this.field_187122_b.func_175682_a(this.field_174849_az, false, d3, d4, d5, d0, d1 + 0.2D, d2, new int[0]);
+                this.field_187122_b.spawnParticle(this.field_174849_az, false, d3, d4, d5, d0, d1 + 0.2D, d2, new int[0]);
             }
         }
 
