--- a/net/minecraft/client/particle/ParticleRain.java
+++ b/net/minecraft/client/particle/ParticleRain.java
@@ -12,9 +12,9 @@
     protected ParticleRain(World p_i1235_1_, double p_i1235_2_, double p_i1235_4_, double p_i1235_6_)
     {
         super(p_i1235_1_, p_i1235_2_, p_i1235_4_, p_i1235_6_, 0.0D, 0.0D, 0.0D);
-        this.field_187129_i *= 0.30000001192092896D;
-        this.field_187130_j = Math.random() * 0.20000000298023224D + 0.10000000149011612D;
-        this.field_187131_k *= 0.30000001192092896D;
+        this.motionX *= 0.30000001192092896D;
+        this.motionY = Math.random() * 0.20000000298023224D + 0.10000000149011612D;
+        this.motionZ *= 0.30000001192092896D;
         this.particleRed = 1.0F;
         this.particleGreen = 1.0F;
         this.particleBlue = 1.0F;
@@ -26,14 +26,14 @@
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
-        this.field_187130_j -= (double)this.particleGravity;
-        this.func_187110_a(this.field_187129_i, this.field_187130_j, this.field_187131_k);
-        this.field_187129_i *= 0.9800000190734863D;
-        this.field_187130_j *= 0.9800000190734863D;
-        this.field_187131_k *= 0.9800000190734863D;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
+        this.motionY -= (double)this.particleGravity;
+        this.func_187110_a(this.motionX, this.motionY, this.motionZ);
+        this.motionX *= 0.9800000190734863D;
+        this.motionY *= 0.9800000190734863D;
+        this.motionZ *= 0.9800000190734863D;
 
         if (this.particleMaxAge-- <= 0)
         {
@@ -47,19 +47,19 @@
                 this.func_187112_i();
             }
 
-            this.field_187129_i *= 0.699999988079071D;
-            this.field_187131_k *= 0.699999988079071D;
+            this.motionX *= 0.699999988079071D;
+            this.motionZ *= 0.699999988079071D;
         }
 
-        BlockPos blockpos = new BlockPos(this.field_187126_f, this.field_187127_g, this.field_187128_h);
-        IBlockState iblockstate = this.field_187122_b.func_180495_p(blockpos);
-        Material material = iblockstate.func_185904_a();
+        BlockPos blockpos = new BlockPos(this.prevPosX, this.prevPosY, this.prevPosZ);
+        IBlockState iblockstate = this.field_187122_b.getBlockState(blockpos);
+        Material material = iblockstate.getMaterial();
 
         if (material.isLiquid() || material.isSolid())
         {
             double d0;
 
-            if (iblockstate.func_177230_c() instanceof BlockLiquid)
+            if (iblockstate.getBlock() instanceof BlockLiquid)
             {
                 d0 = (double)(1.0F - BlockLiquid.getLiquidHeightPercent(((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue()));
             }
@@ -68,9 +68,9 @@
                 d0 = iblockstate.func_185900_c(this.field_187122_b, blockpos).maxY;
             }
 
-            double d1 = (double)MathHelper.floor_double(this.field_187127_g) + d0;
+            double d1 = (double)MathHelper.floor_double(this.prevPosY) + d0;
 
-            if (this.field_187127_g < d1)
+            if (this.prevPosY < d1)
             {
                 this.func_187112_i();
             }
