--- a/net/minecraft/client/particle/ParticleCrit.java
+++ b/net/minecraft/client/particle/ParticleCrit.java
@@ -17,12 +17,12 @@
     protected ParticleCrit(World p_i46285_1_, double p_i46285_2_, double p_i46285_4_, double p_i46285_6_, double p_i46285_8_, double p_i46285_10_, double p_i46285_12_, float p_i46285_14_)
     {
         super(p_i46285_1_, p_i46285_2_, p_i46285_4_, p_i46285_6_, 0.0D, 0.0D, 0.0D);
-        this.field_187129_i *= 0.10000000149011612D;
-        this.field_187130_j *= 0.10000000149011612D;
-        this.field_187131_k *= 0.10000000149011612D;
-        this.field_187129_i += p_i46285_8_ * 0.4D;
-        this.field_187130_j += p_i46285_10_ * 0.4D;
-        this.field_187131_k += p_i46285_12_ * 0.4D;
+        this.motionX *= 0.10000000149011612D;
+        this.motionY *= 0.10000000149011612D;
+        this.motionZ *= 0.10000000149011612D;
+        this.motionX += p_i46285_8_ * 0.4D;
+        this.motionY += p_i46285_10_ * 0.4D;
+        this.motionZ += p_i46285_12_ * 0.4D;
         float f = (float)(Math.random() * 0.30000001192092896D + 0.6000000238418579D);
         this.particleRed = f;
         this.particleGreen = f;
@@ -46,27 +46,27 @@
 
     public void func_189213_a()
     {
-        this.field_187123_c = this.field_187126_f;
-        this.field_187124_d = this.field_187127_g;
-        this.field_187125_e = this.field_187128_h;
+        this.posX = this.prevPosX;
+        this.posY = this.prevPosY;
+        this.posZ = this.prevPosZ;
 
         if (this.particleAge++ >= this.particleMaxAge)
         {
             this.func_187112_i();
         }
 
-        this.func_187110_a(this.field_187129_i, this.field_187130_j, this.field_187131_k);
+        this.func_187110_a(this.motionX, this.motionY, this.motionZ);
         this.particleGreen = (float)((double)this.particleGreen * 0.96D);
         this.particleBlue = (float)((double)this.particleBlue * 0.9D);
-        this.field_187129_i *= 0.699999988079071D;
-        this.field_187130_j *= 0.699999988079071D;
-        this.field_187131_k *= 0.699999988079071D;
-        this.field_187130_j -= 0.019999999552965164D;
+        this.motionX *= 0.699999988079071D;
+        this.motionY *= 0.699999988079071D;
+        this.motionZ *= 0.699999988079071D;
+        this.motionY -= 0.019999999552965164D;
 
         if (this.field_187132_l)
         {
-            this.field_187129_i *= 0.699999988079071D;
-            this.field_187131_k *= 0.699999988079071D;
+            this.motionX *= 0.699999988079071D;
+            this.motionZ *= 0.699999988079071D;
         }
     }
 
