--- a/net/minecraft/client/particle/ParticleFirework.java
+++ b/net/minecraft/client/particle/ParticleFirework.java
@@ -40,9 +40,9 @@
             float f3 = 0.375F;
             float f4 = 7.1F * MathHelper.sin(((float)this.particleAge + p_180434_3_ - 1.0F) * 0.25F * (float)Math.PI);
             this.setAlphaF(0.6F - ((float)this.particleAge + p_180434_3_ - 1.0F) * 0.25F * 0.5F);
-            float f5 = (float)(this.field_187123_c + (this.field_187126_f - this.field_187123_c) * (double)p_180434_3_ - interpPosX);
-            float f6 = (float)(this.field_187124_d + (this.field_187127_g - this.field_187124_d) * (double)p_180434_3_ - interpPosY);
-            float f7 = (float)(this.field_187125_e + (this.field_187128_h - this.field_187125_e) * (double)p_180434_3_ - interpPosZ);
+            float f5 = (float)(this.posX + (this.prevPosX - this.posX) * (double)p_180434_3_ - interpPosX);
+            float f6 = (float)(this.posY + (this.prevPosY - this.posY) * (double)p_180434_3_ - interpPosY);
+            float f7 = (float)(this.posZ + (this.prevPosZ - this.posZ) * (double)p_180434_3_ - interpPosZ);
             int i = this.func_189214_a(p_180434_3_);
             int j = i >> 16 & 65535;
             int k = i & 65535;
@@ -66,9 +66,9 @@
         public Spark(World p_i46465_1_, double p_i46465_2_, double p_i46465_4_, double p_i46465_6_, double p_i46465_8_, double p_i46465_10_, double p_i46465_12_, ParticleManager p_i46465_14_)
         {
             super(p_i46465_1_, p_i46465_2_, p_i46465_4_, p_i46465_6_, 160, 8, -0.004F);
-            this.field_187129_i = p_i46465_8_;
-            this.field_187130_j = p_i46465_10_;
-            this.field_187131_k = p_i46465_12_;
+            this.motionX = p_i46465_8_;
+            this.motionY = p_i46465_10_;
+            this.motionZ = p_i46465_12_;
             this.field_92047_az = p_i46465_14_;
             this.particleScale *= 0.75F;
             this.particleMaxAge = 48 + this.field_187136_p.nextInt(12);
@@ -103,7 +103,7 @@
 
             if (this.field_92054_ax && this.particleAge < this.particleMaxAge / 2 && (this.particleAge + this.particleMaxAge) % 2 == 0)
             {
-                ParticleFirework.Spark particlefirework$spark = new ParticleFirework.Spark(this.field_187122_b, this.field_187126_f, this.field_187127_g, this.field_187128_h, 0.0D, 0.0D, 0.0D, this.field_92047_az);
+                ParticleFirework.Spark particlefirework$spark = new ParticleFirework.Spark(this.field_187122_b, this.prevPosX, this.prevPosY, this.prevPosZ, 0.0D, 0.0D, 0.0D, this.field_92047_az);
                 particlefirework$spark.setAlphaF(0.99F);
                 particlefirework$spark.setRBGColorF(this.particleRed, this.particleGreen, this.particleBlue);
                 particlefirework$spark.particleAge = particlefirework$spark.particleMaxAge / 2;
@@ -132,9 +132,9 @@
         public Starter(World p_i46464_1_, double p_i46464_2_, double p_i46464_4_, double p_i46464_6_, double p_i46464_8_, double p_i46464_10_, double p_i46464_12_, ParticleManager p_i46464_14_, NBTTagCompound p_i46464_15_)
         {
             super(p_i46464_1_, p_i46464_2_, p_i46464_4_, p_i46464_6_, 0.0D, 0.0D, 0.0D);
-            this.field_187129_i = p_i46464_8_;
-            this.field_187130_j = p_i46464_10_;
-            this.field_187131_k = p_i46464_12_;
+            this.motionX = p_i46464_8_;
+            this.motionY = p_i46464_10_;
+            this.motionZ = p_i46464_12_;
             this.theEffectRenderer = p_i46464_14_;
             this.particleMaxAge = 8;
 
@@ -205,7 +205,7 @@
                     soundevent1 = flag ? SoundEvents.field_187622_bl : SoundEvents.field_187619_bk;
                 }
 
-                this.field_187122_b.func_184134_a(this.field_187126_f, this.field_187127_g, this.field_187128_h, soundevent1, SoundCategory.AMBIENT, 20.0F, 0.95F + this.field_187136_p.nextFloat() * 0.1F, true);
+                this.field_187122_b.func_184134_a(this.prevPosX, this.prevPosY, this.prevPosZ, soundevent1, SoundCategory.AMBIENT, 20.0F, 0.95F + this.field_187136_p.nextFloat() * 0.1F, true);
             }
 
             if (this.fireworkAge % 2 == 0 && this.fireworkExplosions != null && this.fireworkAge / 2 < this.fireworkExplosions.tagCount())
@@ -248,7 +248,7 @@
                 float f = (float)((j & 16711680) >> 16) / 255.0F;
                 float f1 = (float)((j & 65280) >> 8) / 255.0F;
                 float f2 = (float)((j & 255) >> 0) / 255.0F;
-                ParticleFirework.Overlay particlefirework$overlay = new ParticleFirework.Overlay(this.field_187122_b, this.field_187126_f, this.field_187127_g, this.field_187128_h);
+                ParticleFirework.Overlay particlefirework$overlay = new ParticleFirework.Overlay(this.field_187122_b, this.prevPosX, this.prevPosY, this.prevPosZ);
                 particlefirework$overlay.setRBGColorF(f, f1, f2);
                 this.theEffectRenderer.addEffect(particlefirework$overlay);
             }
@@ -261,7 +261,7 @@
                 {
                     boolean flag3 = this.func_92037_i();
                     SoundEvent soundevent = flag3 ? SoundEvents.field_187640_br : SoundEvents.field_187637_bq;
-                    this.field_187122_b.func_184134_a(this.field_187126_f, this.field_187127_g, this.field_187128_h, soundevent, SoundCategory.AMBIENT, 20.0F, 0.9F + this.field_187136_p.nextFloat() * 0.15F, true);
+                    this.field_187122_b.func_184134_a(this.prevPosX, this.prevPosY, this.prevPosZ, soundevent, SoundCategory.AMBIENT, 20.0F, 0.9F + this.field_187136_p.nextFloat() * 0.15F, true);
                 }
 
                 this.func_187112_i();
@@ -271,7 +271,7 @@
         private boolean func_92037_i()
         {
             Minecraft minecraft = Minecraft.getMinecraft();
-            return minecraft == null || minecraft.func_175606_aa() == null || minecraft.func_175606_aa().getDistanceSq(this.field_187126_f, this.field_187127_g, this.field_187128_h) >= 256.0D;
+            return minecraft == null || minecraft.getRenderViewEntity() == null || minecraft.getRenderViewEntity().getDistanceSq(this.prevPosX, this.prevPosY, this.prevPosZ) >= 256.0D;
         }
 
         private void createParticle(double p_92034_1_, double p_92034_3_, double p_92034_5_, double p_92034_7_, double p_92034_9_, double p_92034_11_, int[] p_92034_13_, int[] p_92034_14_, boolean p_92034_15_, boolean p_92034_16_)
@@ -293,9 +293,9 @@
 
         private void createBall(double p_92035_1_, int p_92035_3_, int[] p_92035_4_, int[] p_92035_5_, boolean p_92035_6_, boolean p_92035_7_)
         {
-            double d0 = this.field_187126_f;
-            double d1 = this.field_187127_g;
-            double d2 = this.field_187128_h;
+            double d0 = this.prevPosX;
+            double d1 = this.prevPosY;
+            double d2 = this.prevPosZ;
 
             for (int i = -p_92035_3_; i <= p_92035_3_; ++i)
             {
@@ -322,7 +322,7 @@
         {
             double d0 = p_92038_3_[0][0];
             double d1 = p_92038_3_[0][1];
-            this.createParticle(this.field_187126_f, this.field_187127_g, this.field_187128_h, d0 * p_92038_1_, d1 * p_92038_1_, 0.0D, p_92038_4_, p_92038_5_, p_92038_6_, p_92038_7_);
+            this.createParticle(this.prevPosX, this.prevPosY, this.prevPosZ, d0 * p_92038_1_, d1 * p_92038_1_, 0.0D, p_92038_4_, p_92038_5_, p_92038_6_, p_92038_7_);
             float f = this.field_187136_p.nextFloat() * (float)Math.PI;
             double d2 = p_92038_8_ ? 0.034D : 0.34D;
 
@@ -346,7 +346,7 @@
 
                         for (double d12 = -1.0D; d12 <= 1.0D; d12 += 2.0D)
                         {
-                            this.createParticle(this.field_187126_f, this.field_187127_g, this.field_187128_h, d9 * d12, d10, d11 * d12, p_92038_4_, p_92038_5_, p_92038_6_, p_92038_7_);
+                            this.createParticle(this.prevPosX, this.prevPosY, this.prevPosZ, d9 * d12, d10, d11 * d12, p_92038_4_, p_92038_5_, p_92038_6_, p_92038_7_);
                         }
                     }
 
@@ -363,10 +363,10 @@
 
             for (int i = 0; i < 70; ++i)
             {
-                double d2 = this.field_187129_i * 0.5D + this.field_187136_p.nextGaussian() * 0.15D + d0;
-                double d3 = this.field_187131_k * 0.5D + this.field_187136_p.nextGaussian() * 0.15D + d1;
-                double d4 = this.field_187130_j * 0.5D + this.field_187136_p.nextDouble() * 0.5D;
-                this.createParticle(this.field_187126_f, this.field_187127_g, this.field_187128_h, d2, d4, d3, p_92036_1_, p_92036_2_, p_92036_3_, p_92036_4_);
+                double d2 = this.motionX * 0.5D + this.field_187136_p.nextGaussian() * 0.15D + d0;
+                double d3 = this.motionZ * 0.5D + this.field_187136_p.nextGaussian() * 0.15D + d1;
+                double d4 = this.motionY * 0.5D + this.field_187136_p.nextDouble() * 0.5D;
+                this.createParticle(this.prevPosX, this.prevPosY, this.prevPosZ, d2, d4, d3, p_92036_1_, p_92036_2_, p_92036_3_, p_92036_4_);
             }
         }
 
