--- a/net/minecraft/client/multiplayer/WorldClient.java
+++ b/net/minecraft/client/multiplayer/WorldClient.java
@@ -307,9 +307,9 @@
     @Deprecated
     public boolean func_180503_b(BlockPos p_180503_1_, IBlockState p_180503_2_)
     {
-        int i = p_180503_1_.func_177958_n();
-        int j = p_180503_1_.func_177956_o();
-        int k = p_180503_1_.func_177952_p();
+        int i = p_180503_1_.getX();
+        int j = p_180503_1_.getY();
+        int k = p_180503_1_.getZ();
         this.invalidateBlockReceiveRegion(i, j, k, i, j, k);
         return super.func_180501_a(p_180503_1_, p_180503_2_, 3);
     }
@@ -341,11 +341,11 @@
             int k = i >> 8 & 15;
             int l = i >> 16 & 255;
             BlockPos blockpos = new BlockPos(j + p_147467_1_, l, k + p_147467_2_);
-            IBlockState iblockstate = p_147467_3_.func_177435_g(blockpos);
+            IBlockState iblockstate = p_147467_3_.getBlockState(blockpos);
             j = j + p_147467_1_;
             k = k + p_147467_2_;
 
-            if (iblockstate.func_185904_a() == Material.air && this.func_175699_k(blockpos) <= this.rand.nextInt(8) && this.func_175642_b(EnumSkyBlock.SKY, blockpos) <= 0 && this.mc.thePlayer != null && this.mc.thePlayer.getDistanceSq((double)j + 0.5D, (double)l + 0.5D, (double)k + 0.5D) > 4.0D)
+            if (iblockstate.getMaterial() == Material.air && this.func_175699_k(blockpos) <= this.rand.nextInt(8) && this.func_175642_b(EnumSkyBlock.SKY, blockpos) <= 0 && this.mc.thePlayer != null && this.mc.thePlayer.getDistanceSq((double)j + 0.5D, (double)l + 0.5D, (double)k + 0.5D) > 4.0D)
             {
                 this.func_184134_a((double)j + 0.5D, (double)l + 0.5D, (double)k + 0.5D, SoundEvents.field_187674_a, SoundCategory.AMBIENT, 0.7F, 0.8F + this.rand.nextFloat() * 0.2F, false);
                 this.field_184158_M = this.rand.nextInt(12000) + 6000;
@@ -357,7 +357,7 @@
     {
         int i = 32;
         Random random = new Random();
-        ItemStack itemstack = this.mc.thePlayer.func_184614_ca();
+        ItemStack itemstack = this.mc.thePlayer.getMainHandItem();
         boolean flag = this.mc.playerController.func_178889_l() == GameType.CREATIVE && itemstack != null && Block.getBlockFromItem(itemstack.getItem()) == Blocks.field_180401_cv;
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
@@ -374,12 +374,12 @@
         int j = p_184153_2_ + this.rand.nextInt(p_184153_4_) - this.rand.nextInt(p_184153_4_);
         int k = p_184153_3_ + this.rand.nextInt(p_184153_4_) - this.rand.nextInt(p_184153_4_);
         p_184153_7_.func_181079_c(i, j, k);
-        IBlockState iblockstate = this.func_180495_p(p_184153_7_);
-        iblockstate.func_177230_c().func_180655_c(iblockstate, this, p_184153_7_, p_184153_5_);
+        IBlockState iblockstate = this.getBlockState(p_184153_7_);
+        iblockstate.getBlock().func_180655_c(iblockstate, this, p_184153_7_, p_184153_5_);
 
-        if (p_184153_6_ && iblockstate.func_177230_c() == Blocks.field_180401_cv)
+        if (p_184153_6_ && iblockstate.getBlock() == Blocks.field_180401_cv)
         {
-            this.func_175688_a(EnumParticleTypes.BARRIER, (double)((float)i + 0.5F), (double)((float)j + 0.5F), (double)((float)k + 0.5F), 0.0D, 0.0D, 0.0D, new int[0]);
+            this.spawnParticle(EnumParticleTypes.BARRIER, (double)((float)i + 0.5F), (double)((float)j + 0.5F), (double)((float)k + 0.5F), 0.0D, 0.0D, 0.0D, new int[0]);
         }
     }
 
@@ -412,7 +412,7 @@
         for (int j1 = 0; j1 < this.loadedEntityList.size(); ++j1)
         {
             Entity entity1 = (Entity)this.loadedEntityList.get(j1);
-            Entity entity2 = entity1.func_184187_bx();
+            Entity entity2 = entity1.getRidingEntity();
 
             if (entity2 != null)
             {
@@ -421,7 +421,7 @@
                     continue;
                 }
 
-                entity1.func_184210_p();
+                entity1.dismount();
             }
 
             if (entity1.isDead)
@@ -477,7 +477,7 @@
         return crashreportcategory;
     }
 
-    public void func_184148_a(@Nullable EntityPlayer p_184148_1_, double p_184148_2_, double p_184148_4_, double p_184148_6_, SoundEvent p_184148_8_, SoundCategory p_184148_9_, float p_184148_10_, float p_184148_11_)
+    public void playSound(@Nullable EntityPlayer p_184148_1_, double p_184148_2_, double p_184148_4_, double p_184148_6_, SoundEvent p_184148_8_, SoundCategory p_184148_9_, float p_184148_10_, float p_184148_11_)
     {
         if (p_184148_1_ == this.mc.thePlayer)
         {
@@ -487,12 +487,12 @@
 
     public void func_184156_a(BlockPos p_184156_1_, SoundEvent p_184156_2_, SoundCategory p_184156_3_, float p_184156_4_, float p_184156_5_, boolean p_184156_6_)
     {
-        this.func_184134_a((double)p_184156_1_.func_177958_n() + 0.5D, (double)p_184156_1_.func_177956_o() + 0.5D, (double)p_184156_1_.func_177952_p() + 0.5D, p_184156_2_, p_184156_3_, p_184156_4_, p_184156_5_, p_184156_6_);
+        this.func_184134_a((double)p_184156_1_.getX() + 0.5D, (double)p_184156_1_.getY() + 0.5D, (double)p_184156_1_.getZ() + 0.5D, p_184156_2_, p_184156_3_, p_184156_4_, p_184156_5_, p_184156_6_);
     }
 
     public void func_184134_a(double p_184134_1_, double p_184134_3_, double p_184134_5_, SoundEvent p_184134_7_, SoundCategory p_184134_8_, float p_184134_9_, float p_184134_10_, boolean p_184134_11_)
     {
-        double d0 = this.mc.func_175606_aa().getDistanceSq(p_184134_1_, p_184134_3_, p_184134_5_);
+        double d0 = this.mc.getRenderViewEntity().getDistanceSq(p_184134_1_, p_184134_3_, p_184134_5_);
         PositionedSoundRecord positionedsoundrecord = new PositionedSoundRecord(p_184134_7_, p_184134_8_, p_184134_9_, p_184134_10_, (float)p_184134_1_, (float)p_184134_3_, (float)p_184134_5_);
 
         if (p_184134_11_ && d0 > 100.0D)
