--- a/net/minecraft/client/gui/inventory/GuiContainer.java
+++ b/net/minecraft/client/gui/inventory/GuiContainer.java
@@ -138,7 +138,7 @@
                 GlStateManager.enableDepth();
             }
         }
-
+        
         RenderHelper.disableStandardItemLighting();
         this.drawGuiContainerForegroundLayer(mouseX, mouseY);
         RenderHelper.enableGUIStandardItemLighting();
@@ -166,7 +166,6 @@
                     s = "" + TextFormatting.YELLOW + "0";
                 }
             }
-
             this.drawItemStack(itemstack, mouseX - i - 8, mouseY - j - k2, s);
         }
 
@@ -188,6 +187,7 @@
         }
 
         GlStateManager.popMatrix();
+        GlStateManager.colorMask(true, true, true, true);
 
         if (inventoryplayer.getItemStack() == null && this.theSlot != null && this.theSlot.getHasStack())
         {
@@ -419,7 +419,7 @@
                         }
                         else
                         {
-                            boolean flag2 = l != -999 && (Keyboard.isKeyDown(42) || Keyboard.isKeyDown(54));
+                            boolean flag2 = l != -999 && (Keyboard.isKeyDown(42) || Keyboard.isKeyDown(54) || pressShiftFake);
                             ClickType clicktype = ClickType.PICKUP;
 
                             if (flag2)
