--- a/net/minecraft/client/gui/inventory/GuiCrafting.java
+++ b/net/minecraft/client/gui/inventory/GuiCrafting.java
@@ -14,7 +14,7 @@
 
     public GuiCrafting(InventoryPlayer p_i45504_1_, World p_i45504_2_)
     {
-        this(p_i45504_1_, p_i45504_2_, BlockPos.field_177992_a);
+        this(p_i45504_1_, p_i45504_2_, BlockPos.createDefault);
     }
 
     public GuiCrafting(InventoryPlayer p_i45505_1_, World p_i45505_2_, BlockPos p_i45505_3_)
