--- a/net/minecraft/client/gui/inventory/GuiEditStructure.java
+++ b/net/minecraft/client/gui/inventory/GuiEditStructure.java
@@ -110,28 +110,28 @@
         BlockPos blockpos = this.field_189846_f.func_189711_e();
         this.field_189854_v = new GuiTextField(3, this.fontRendererObj, this.width / 2 - 152, 80, 80, 20);
         this.field_189854_v.setMaxStringLength(15);
-        this.field_189854_v.setText(Integer.toString(blockpos.func_177958_n()));
+        this.field_189854_v.setText(Integer.toString(blockpos.getX()));
         this.field_189843_S.add(this.field_189854_v);
         this.field_189855_w = new GuiTextField(4, this.fontRendererObj, this.width / 2 - 72, 80, 80, 20);
         this.field_189855_w.setMaxStringLength(15);
-        this.field_189855_w.setText(Integer.toString(blockpos.func_177956_o()));
+        this.field_189855_w.setText(Integer.toString(blockpos.getY()));
         this.field_189843_S.add(this.field_189855_w);
         this.field_189856_x = new GuiTextField(5, this.fontRendererObj, this.width / 2 + 8, 80, 80, 20);
         this.field_189856_x.setMaxStringLength(15);
-        this.field_189856_x.setText(Integer.toString(blockpos.func_177952_p()));
+        this.field_189856_x.setText(Integer.toString(blockpos.getZ()));
         this.field_189843_S.add(this.field_189856_x);
         BlockPos blockpos1 = this.field_189846_f.func_189717_g();
         this.field_189857_y = new GuiTextField(6, this.fontRendererObj, this.width / 2 - 152, 120, 80, 20);
         this.field_189857_y.setMaxStringLength(15);
-        this.field_189857_y.setText(Integer.toString(blockpos1.func_177958_n()));
+        this.field_189857_y.setText(Integer.toString(blockpos1.getX()));
         this.field_189843_S.add(this.field_189857_y);
         this.field_189858_z = new GuiTextField(7, this.fontRendererObj, this.width / 2 - 72, 120, 80, 20);
         this.field_189858_z.setMaxStringLength(15);
-        this.field_189858_z.setText(Integer.toString(blockpos1.func_177956_o()));
+        this.field_189858_z.setText(Integer.toString(blockpos1.getY()));
         this.field_189843_S.add(this.field_189858_z);
         this.field_189825_A = new GuiTextField(8, this.fontRendererObj, this.width / 2 + 8, 120, 80, 20);
         this.field_189825_A.setMaxStringLength(15);
-        this.field_189825_A.setText(Integer.toString(blockpos1.func_177952_p()));
+        this.field_189825_A.setText(Integer.toString(blockpos1.getZ()));
         this.field_189843_S.add(this.field_189825_A);
         this.field_189826_B = new GuiTextField(15, this.fontRendererObj, this.width / 2 - 152, 120, 80, 20);
         this.field_189826_B.setMaxStringLength(15);
