--- a/net/minecraft/client/gui/inventory/GuiInventory.java
+++ b/net/minecraft/client/gui/inventory/GuiInventory.java
@@ -93,19 +93,19 @@
     public static void drawEntityOnScreen(int p_147046_0_, int p_147046_1_, int p_147046_2_, float p_147046_3_, float p_147046_4_, EntityLivingBase p_147046_5_)
     {
         GlStateManager.func_179142_g();
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         GlStateManager.func_179109_b((float)p_147046_0_, (float)p_147046_1_, 50.0F);
-        GlStateManager.func_179152_a((float)(-p_147046_2_), (float)p_147046_2_, (float)p_147046_2_);
-        GlStateManager.func_179114_b(180.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.scale((float)(-p_147046_2_), (float)p_147046_2_, (float)p_147046_2_);
+        GlStateManager.rotate(180.0F, 0.0F, 0.0F, 1.0F);
         float f = p_147046_5_.renderYawOffset;
         float f1 = p_147046_5_.rotationYaw;
         float f2 = p_147046_5_.rotationPitch;
         float f3 = p_147046_5_.prevRotationYawHead;
         float f4 = p_147046_5_.rotationYawHead;
-        GlStateManager.func_179114_b(135.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(135.0F, 0.0F, 1.0F, 0.0F);
         RenderHelper.enableStandardItemLighting();
-        GlStateManager.func_179114_b(-135.0F, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b(-((float)Math.atan((double)(p_147046_4_ / 40.0F))) * 20.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(-135.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(-((float)Math.atan((double)(p_147046_4_ / 40.0F))) * 20.0F, 1.0F, 0.0F, 0.0F);
         p_147046_5_.renderYawOffset = (float)Math.atan((double)(p_147046_3_ / 40.0F)) * 20.0F;
         p_147046_5_.rotationYaw = (float)Math.atan((double)(p_147046_3_ / 40.0F)) * 40.0F;
         p_147046_5_.rotationPitch = -((float)Math.atan((double)(p_147046_4_ / 40.0F))) * 20.0F;
@@ -122,7 +122,7 @@
         p_147046_5_.rotationPitch = f2;
         p_147046_5_.prevRotationYawHead = f3;
         p_147046_5_.rotationYawHead = f4;
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         RenderHelper.disableStandardItemLighting();
         GlStateManager.func_179101_C();
         GlStateManager.func_179138_g(OpenGlHelper.lightmapTexUnit);
