--- a/net/minecraft/client/renderer/entity/layers/LayerHeldBlock.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerHeldBlock.java
@@ -26,13 +26,13 @@
         {
             BlockRendererDispatcher blockrendererdispatcher = Minecraft.getMinecraft().func_175602_ab();
             GlStateManager.func_179091_B();
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             GlStateManager.func_179109_b(0.0F, 0.6875F, -0.75F);
-            GlStateManager.func_179114_b(20.0F, 1.0F, 0.0F, 0.0F);
-            GlStateManager.func_179114_b(45.0F, 0.0F, 1.0F, 0.0F);
+            GlStateManager.rotate(20.0F, 1.0F, 0.0F, 0.0F);
+            GlStateManager.rotate(45.0F, 0.0F, 1.0F, 0.0F);
             GlStateManager.func_179109_b(0.25F, 0.1875F, 0.25F);
             float f = 0.5F;
-            GlStateManager.func_179152_a(-0.5F, -0.5F, 0.5F);
+            GlStateManager.scale(-0.5F, -0.5F, 0.5F);
             int i = p_177141_1_.getBrightnessForRender(p_177141_4_);
             int j = i % 65536;
             int k = i / 65536;
@@ -40,7 +40,7 @@
             GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
             this.field_177174_a.bindTexture(TextureMap.locationBlocksTexture);
             blockrendererdispatcher.func_175016_a(iblockstate, 1.0F);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
             GlStateManager.func_179101_C();
         }
     }
