--- a/net/minecraft/client/renderer/entity/layers/LayerDeadmau5Head.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerDeadmau5Head.java
@@ -23,17 +23,17 @@
             {
                 float f = p_177141_1_.prevRotationYaw + (p_177141_1_.rotationYaw - p_177141_1_.prevRotationYaw) * p_177141_4_ - (p_177141_1_.prevRenderYawOffset + (p_177141_1_.renderYawOffset - p_177141_1_.prevRenderYawOffset) * p_177141_4_);
                 float f1 = p_177141_1_.prevRotationPitch + (p_177141_1_.rotationPitch - p_177141_1_.prevRotationPitch) * p_177141_4_;
-                GlStateManager.func_179094_E();
-                GlStateManager.func_179114_b(f, 0.0F, 1.0F, 0.0F);
-                GlStateManager.func_179114_b(f1, 1.0F, 0.0F, 0.0F);
+                GlStateManager.push();
+                GlStateManager.rotate(f, 0.0F, 1.0F, 0.0F);
+                GlStateManager.rotate(f1, 1.0F, 0.0F, 0.0F);
                 GlStateManager.func_179109_b(0.375F * (float)(i * 2 - 1), 0.0F, 0.0F);
                 GlStateManager.func_179109_b(0.0F, -0.375F, 0.0F);
-                GlStateManager.func_179114_b(-f1, 1.0F, 0.0F, 0.0F);
-                GlStateManager.func_179114_b(-f, 0.0F, 1.0F, 0.0F);
+                GlStateManager.rotate(-f1, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(-f, 0.0F, 1.0F, 0.0F);
                 float f2 = 1.3333334F;
-                GlStateManager.func_179152_a(1.3333334F, 1.3333334F, 1.3333334F);
+                GlStateManager.scale(1.3333334F, 1.3333334F, 1.3333334F);
                 this.field_177208_a.func_177087_b().func_178727_b(0.0625F);
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
         }
     }
