--- a/net/minecraft/client/renderer/entity/layers/LayerEnderDragonDeath.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerEnderDragonDeath.java
@@ -33,17 +33,17 @@
             GlStateManager.func_179118_c();
             GlStateManager.func_179089_o();
             GlStateManager.func_179132_a(false);
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             GlStateManager.func_179109_b(0.0F, -1.0F, -2.0F);
 
             for (int i = 0; (float)i < (f + f * f) / 2.0F * 60.0F; ++i)
             {
-                GlStateManager.func_179114_b(random.nextFloat() * 360.0F, 1.0F, 0.0F, 0.0F);
-                GlStateManager.func_179114_b(random.nextFloat() * 360.0F, 0.0F, 1.0F, 0.0F);
-                GlStateManager.func_179114_b(random.nextFloat() * 360.0F, 0.0F, 0.0F, 1.0F);
-                GlStateManager.func_179114_b(random.nextFloat() * 360.0F, 1.0F, 0.0F, 0.0F);
-                GlStateManager.func_179114_b(random.nextFloat() * 360.0F, 0.0F, 1.0F, 0.0F);
-                GlStateManager.func_179114_b(random.nextFloat() * 360.0F + f * 90.0F, 0.0F, 0.0F, 1.0F);
+                GlStateManager.rotate(random.nextFloat() * 360.0F, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(random.nextFloat() * 360.0F, 0.0F, 1.0F, 0.0F);
+                GlStateManager.rotate(random.nextFloat() * 360.0F, 0.0F, 0.0F, 1.0F);
+                GlStateManager.rotate(random.nextFloat() * 360.0F, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(random.nextFloat() * 360.0F, 0.0F, 1.0F, 0.0F);
+                GlStateManager.rotate(random.nextFloat() * 360.0F + f * 90.0F, 0.0F, 0.0F, 1.0F);
                 float f2 = random.nextFloat() * 20.0F + 5.0F + f1 * 10.0F;
                 float f3 = random.nextFloat() * 2.0F + 1.0F + f1 * 2.0F;
                 vertexbuffer.func_181668_a(6, DefaultVertexFormats.field_181706_f);
@@ -55,7 +55,7 @@
                 tessellator.draw();
             }
 
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
             GlStateManager.func_179132_a(true);
             GlStateManager.func_179129_p();
             GlStateManager.func_179084_k();
