--- a/net/minecraft/client/renderer/entity/layers/LayerCustomHead.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerCustomHead.java
@@ -32,13 +32,13 @@
 
     public void func_177141_a(EntityLivingBase p_177141_1_, float p_177141_2_, float p_177141_3_, float p_177141_4_, float p_177141_5_, float p_177141_6_, float p_177141_7_, float p_177141_8_)
     {
-        ItemStack itemstack = p_177141_1_.func_184582_a(EntityEquipmentSlot.HEAD);
+        ItemStack itemstack = p_177141_1_.getEquippedItemStack(EntityEquipmentSlot.HEAD);
 
         if (itemstack != null && itemstack.getItem() != null)
         {
             Item item = itemstack.getItem();
             Minecraft minecraft = Minecraft.getMinecraft();
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
 
             if (p_177141_1_.isSneaking())
             {
@@ -52,7 +52,7 @@
                 float f = 2.0F;
                 float f1 = 1.4F;
                 GlStateManager.func_179109_b(0.0F, 0.5F * p_177141_8_, 0.0F);
-                GlStateManager.func_179152_a(0.7F, 0.7F, 0.7F);
+                GlStateManager.scale(0.7F, 0.7F, 0.7F);
                 GlStateManager.func_179109_b(0.0F, 16.0F * p_177141_8_, 0.0F);
             }
 
@@ -62,7 +62,7 @@
             if (item == Items.skull)
             {
                 float f2 = 1.1875F;
-                GlStateManager.func_179152_a(1.1875F, -1.1875F, -1.1875F);
+                GlStateManager.scale(1.1875F, -1.1875F, -1.1875F);
 
                 if (flag)
                 {
@@ -97,8 +97,8 @@
             {
                 float f3 = 0.625F;
                 GlStateManager.func_179109_b(0.0F, -0.25F, 0.0F);
-                GlStateManager.func_179114_b(180.0F, 0.0F, 1.0F, 0.0F);
-                GlStateManager.func_179152_a(0.625F, -0.625F, -0.625F);
+                GlStateManager.rotate(180.0F, 0.0F, 1.0F, 0.0F);
+                GlStateManager.scale(0.625F, -0.625F, -0.625F);
 
                 if (flag)
                 {
@@ -108,7 +108,7 @@
                 minecraft.func_175597_ag().func_178099_a(p_177141_1_, itemstack, ItemCameraTransforms.TransformType.HEAD);
             }
 
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
     }
 
