--- a/net/minecraft/client/renderer/entity/layers/LayerElytra.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerElytra.java
@@ -23,7 +23,7 @@
 
     public void func_177141_a(AbstractClientPlayer p_177141_1_, float p_177141_2_, float p_177141_3_, float p_177141_4_, float p_177141_5_, float p_177141_6_, float p_177141_7_, float p_177141_8_)
     {
-        ItemStack itemstack = p_177141_1_.func_184582_a(EntityEquipmentSlot.CHEST);
+        ItemStack itemstack = p_177141_1_.getEquippedItemStack(EntityEquipmentSlot.CHEST);
 
         if (itemstack != null && itemstack.getItem() == Items.field_185160_cR)
         {
@@ -43,7 +43,7 @@
                 this.field_188356_b.bindTexture(field_188355_a);
             }
 
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             GlStateManager.func_179109_b(0.0F, 0.0F, 0.125F);
             this.field_188357_c.setRotationAngles(p_177141_2_, p_177141_3_, p_177141_5_, p_177141_6_, p_177141_7_, p_177141_8_, p_177141_1_);
             this.field_188357_c.render(p_177141_1_, p_177141_2_, p_177141_3_, p_177141_5_, p_177141_6_, p_177141_7_, p_177141_8_);
@@ -53,7 +53,7 @@
                 LayerArmorBase.func_188364_a(this.field_188356_b, p_177141_1_, this.field_188357_c, p_177141_2_, p_177141_3_, p_177141_4_, p_177141_5_, p_177141_6_, p_177141_7_, p_177141_8_);
             }
 
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
     }
 
