--- a/net/minecraft/client/renderer/entity/layers/LayerElytra.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerElytra.java
@@ -3,6 +3,7 @@
 import net.minecraft.client.entity.AbstractClientPlayer;
 import net.minecraft.client.model.ModelElytra;
 import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.entity.RenderLivingBase;
 import net.minecraft.client.renderer.entity.RenderPlayer;
 import net.minecraft.entity.player.EnumPlayerModelParts;
 import net.minecraft.init.Items;
@@ -18,14 +19,19 @@
     private static final ResourceLocation TEXTURE_ELYTRA = new ResourceLocation("textures/entity/elytra.png");
 
     /** Instance of the player renderer. */
-    private final RenderPlayer renderPlayer;
+    private final RenderLivingBase<?>  renderPlayer;
 
     /** The model used by the Elytra. */
     private final ModelElytra modelElytra = new ModelElytra();
 
-    public LayerElytra(RenderPlayer renderPlayerIn)
+    public LayerElytra(RenderLivingBase<?> p_i47185_1_)
     {
-        this.renderPlayer = renderPlayerIn;
+        this.renderPlayer = p_i47185_1_;
+    }
+
+    public LayerElytra(RenderPlayer p_i47185_1_)
+    {
+        this.renderPlayer = p_i47185_1_;
     }
 
     public void doRenderLayer(AbstractClientPlayer entitylivingbaseIn, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch, float scale)
