--- a/net/minecraft/client/renderer/entity/layers/LayerIronGolemFlower.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerIronGolemFlower.java
@@ -25,20 +25,20 @@
         {
             BlockRendererDispatcher blockrendererdispatcher = Minecraft.getMinecraft().func_175602_ab();
             GlStateManager.func_179091_B();
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179114_b(5.0F + 180.0F * ((ModelIronGolem)this.field_177154_a.func_177087_b()).ironGolemRightArm.rotateAngleX / (float)Math.PI, 1.0F, 0.0F, 0.0F);
-            GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
+            GlStateManager.push();
+            GlStateManager.rotate(5.0F + 180.0F * ((ModelIronGolem)this.field_177154_a.func_177087_b()).ironGolemRightArm.rotateAngleX / (float)Math.PI, 1.0F, 0.0F, 0.0F);
+            GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
             GlStateManager.func_179109_b(-0.9375F, -0.625F, -0.9375F);
             float f = 0.5F;
-            GlStateManager.func_179152_a(0.5F, -0.5F, 0.5F);
+            GlStateManager.scale(0.5F, -0.5F, 0.5F);
             int i = p_177141_1_.getBrightnessForRender(p_177141_4_);
             int j = i % 65536;
             int k = i / 65536;
             OpenGlHelper.setLightmapTextureCoords(OpenGlHelper.lightmapTexUnit, (float)j, (float)k);
             GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
             this.field_177154_a.bindTexture(TextureMap.locationBlocksTexture);
-            blockrendererdispatcher.func_175016_a(Blocks.red_flower.func_176223_P(), 1.0F);
-            GlStateManager.func_179121_F();
+            blockrendererdispatcher.func_175016_a(Blocks.red_flower.getBlockState(), 1.0F);
+            GlStateManager.pop();
             GlStateManager.func_179101_C();
         }
     }
