--- a/net/minecraft/client/renderer/entity/layers/LayerArmorBase.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerArmorBase.java
@@ -92,7 +92,7 @@
     @Nullable
     public ItemStack func_188362_a(EntityLivingBase p_188362_1_, EntityEquipmentSlot p_188362_2_)
     {
-        return p_188362_1_.func_184582_a(p_188362_2_);
+        return p_188362_1_.getEquippedItemStack(p_188362_2_);
     }
 
     public T func_188360_a(EntityEquipmentSlot p_188360_1_)
@@ -121,19 +121,19 @@
             GlStateManager.func_187401_a(GlStateManager.SourceFactor.SRC_COLOR, GlStateManager.DestFactor.ONE);
             float f2 = 0.76F;
             GlStateManager.func_179131_c(0.38F, 0.19F, 0.608F, 1.0F);
-            GlStateManager.func_179128_n(5890);
-            GlStateManager.func_179096_D();
+            GlStateManager.matrixMode(5890);
+            GlStateManager.loadIdentity();
             float f3 = 0.33333334F;
-            GlStateManager.func_179152_a(0.33333334F, 0.33333334F, 0.33333334F);
-            GlStateManager.func_179114_b(30.0F - (float)i * 60.0F, 0.0F, 0.0F, 1.0F);
+            GlStateManager.scale(0.33333334F, 0.33333334F, 0.33333334F);
+            GlStateManager.rotate(30.0F - (float)i * 60.0F, 0.0F, 0.0F, 1.0F);
             GlStateManager.func_179109_b(0.0F, f * (0.001F + (float)i * 0.003F) * 20.0F, 0.0F);
-            GlStateManager.func_179128_n(5888);
+            GlStateManager.matrixMode(5888);
             p_188364_2_.render(p_188364_1_, p_188364_3_, p_188364_4_, p_188364_6_, p_188364_7_, p_188364_8_, p_188364_9_);
         }
 
-        GlStateManager.func_179128_n(5890);
-        GlStateManager.func_179096_D();
-        GlStateManager.func_179128_n(5888);
+        GlStateManager.matrixMode(5890);
+        GlStateManager.loadIdentity();
+        GlStateManager.matrixMode(5888);
         GlStateManager.func_179145_e();
         GlStateManager.func_179132_a(true);
         GlStateManager.func_179143_c(515);
