--- a/net/minecraft/client/renderer/entity/layers/LayerMooshroomMushroom.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerMooshroomMushroom.java
@@ -26,29 +26,29 @@
             this.field_177205_a.bindTexture(TextureMap.locationBlocksTexture);
             GlStateManager.func_179089_o();
             GlStateManager.func_187407_a(GlStateManager.CullFace.FRONT);
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179152_a(1.0F, -1.0F, 1.0F);
+            GlStateManager.push();
+            GlStateManager.scale(1.0F, -1.0F, 1.0F);
             GlStateManager.func_179109_b(0.2F, 0.35F, 0.5F);
-            GlStateManager.func_179114_b(42.0F, 0.0F, 1.0F, 0.0F);
-            GlStateManager.func_179094_E();
+            GlStateManager.rotate(42.0F, 0.0F, 1.0F, 0.0F);
+            GlStateManager.push();
             GlStateManager.func_179109_b(-0.5F, -0.5F, 0.5F);
-            blockrendererdispatcher.func_175016_a(Blocks.red_mushroom.func_176223_P(), 1.0F);
-            GlStateManager.func_179121_F();
-            GlStateManager.func_179094_E();
+            blockrendererdispatcher.func_175016_a(Blocks.red_mushroom.getBlockState(), 1.0F);
+            GlStateManager.pop();
+            GlStateManager.push();
             GlStateManager.func_179109_b(0.1F, 0.0F, -0.6F);
-            GlStateManager.func_179114_b(42.0F, 0.0F, 1.0F, 0.0F);
+            GlStateManager.rotate(42.0F, 0.0F, 1.0F, 0.0F);
             GlStateManager.func_179109_b(-0.5F, -0.5F, 0.5F);
-            blockrendererdispatcher.func_175016_a(Blocks.red_mushroom.func_176223_P(), 1.0F);
-            GlStateManager.func_179121_F();
-            GlStateManager.func_179121_F();
-            GlStateManager.func_179094_E();
+            blockrendererdispatcher.func_175016_a(Blocks.red_mushroom.getBlockState(), 1.0F);
+            GlStateManager.pop();
+            GlStateManager.pop();
+            GlStateManager.push();
             ((ModelQuadruped)this.field_177205_a.func_177087_b()).head.postRender(0.0625F);
-            GlStateManager.func_179152_a(1.0F, -1.0F, 1.0F);
+            GlStateManager.scale(1.0F, -1.0F, 1.0F);
             GlStateManager.func_179109_b(0.0F, 0.7F, -0.2F);
-            GlStateManager.func_179114_b(12.0F, 0.0F, 1.0F, 0.0F);
+            GlStateManager.rotate(12.0F, 0.0F, 1.0F, 0.0F);
             GlStateManager.func_179109_b(-0.5F, -0.5F, 0.5F);
-            blockrendererdispatcher.func_175016_a(Blocks.red_mushroom.func_176223_P(), 1.0F);
-            GlStateManager.func_179121_F();
+            blockrendererdispatcher.func_175016_a(Blocks.red_mushroom.getBlockState(), 1.0F);
+            GlStateManager.pop();
             GlStateManager.func_187407_a(GlStateManager.CullFace.BACK);
             GlStateManager.func_179129_p();
         }
