--- a/net/minecraft/client/renderer/entity/layers/LayerWitherAura.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerWitherAura.java
@@ -24,13 +24,13 @@
         {
             GlStateManager.func_179132_a(!p_177141_1_.isInvisible());
             this.field_177215_b.bindTexture(field_177217_a);
-            GlStateManager.func_179128_n(5890);
-            GlStateManager.func_179096_D();
+            GlStateManager.matrixMode(5890);
+            GlStateManager.loadIdentity();
             float f = (float)p_177141_1_.ticksExisted + p_177141_4_;
             float f1 = MathHelper.cos(f * 0.02F) * 3.0F;
             float f2 = f * 0.01F;
             GlStateManager.func_179109_b(f1, f2, 0.0F);
-            GlStateManager.func_179128_n(5888);
+            GlStateManager.matrixMode(5888);
             GlStateManager.func_179147_l();
             float f3 = 0.5F;
             GlStateManager.func_179131_c(0.5F, 0.5F, 0.5F, 1.0F);
@@ -39,9 +39,9 @@
             this.field_177216_c.setLivingAnimations(p_177141_1_, p_177141_2_, p_177141_3_, p_177141_4_);
             this.field_177216_c.func_178686_a(this.field_177215_b.func_177087_b());
             this.field_177216_c.render(p_177141_1_, p_177141_2_, p_177141_3_, p_177141_5_, p_177141_6_, p_177141_7_, p_177141_8_);
-            GlStateManager.func_179128_n(5890);
-            GlStateManager.func_179096_D();
-            GlStateManager.func_179128_n(5888);
+            GlStateManager.matrixMode(5890);
+            GlStateManager.loadIdentity();
+            GlStateManager.matrixMode(5888);
             GlStateManager.func_179145_e();
             GlStateManager.func_179084_k();
         }
