--- a/net/minecraft/block/state/BlockPistonStructureHelper.java
+++ b/net/minecraft/block/state/BlockPistonStructureHelper.java
@@ -28,12 +28,12 @@
         if (p_i45664_4_)
         {
             this.field_177257_d = p_i45664_3_;
-            this.field_177260_c = p_i45664_2_.func_177972_a(p_i45664_3_);
+            this.field_177260_c = p_i45664_2_.getBlockPosRelative(p_i45664_3_);
         }
         else
         {
             this.field_177257_d = p_i45664_3_.func_176734_d();
-            this.field_177260_c = p_i45664_2_.func_177967_a(p_i45664_3_, 2);
+            this.field_177260_c = p_i45664_2_.getBlockPosRelative(p_i45664_3_, 2);
         }
     }
 
@@ -41,7 +41,7 @@
     {
         this.field_177258_e.clear();
         this.field_177256_f.clear();
-        IBlockState iblockstate = this.field_177261_a.func_180495_p(this.field_177260_c);
+        IBlockState iblockstate = this.field_177261_a.getBlockState(this.field_177260_c);
 
         if (!BlockPistonBase.func_185646_a(iblockstate, this.field_177261_a, this.field_177260_c, this.field_177257_d, false))
         {
@@ -65,7 +65,7 @@
             {
                 BlockPos blockpos = (BlockPos)this.field_177258_e.get(i);
 
-                if (this.field_177261_a.func_180495_p(blockpos).func_177230_c() == Blocks.field_180399_cE && !this.func_177250_b(blockpos))
+                if (this.field_177261_a.getBlockState(blockpos).getBlock() == Blocks.field_180399_cE && !this.func_177250_b(blockpos))
                 {
                     return false;
                 }
@@ -77,10 +77,10 @@
 
     private boolean func_177251_a(BlockPos p_177251_1_)
     {
-        IBlockState iblockstate = this.field_177261_a.func_180495_p(p_177251_1_);
-        Block block = iblockstate.func_177230_c();
+        IBlockState iblockstate = this.field_177261_a.getBlockState(p_177251_1_);
+        Block block = iblockstate.getBlock();
 
-        if (iblockstate.func_185904_a() == Material.air)
+        if (iblockstate.getMaterial() == Material.air)
         {
             return true;
         }
@@ -108,11 +108,11 @@
             {
                 while (block == Blocks.field_180399_cE)
                 {
-                    BlockPos blockpos = p_177251_1_.func_177967_a(this.field_177257_d.func_176734_d(), i);
-                    iblockstate = this.field_177261_a.func_180495_p(blockpos);
-                    block = iblockstate.func_177230_c();
+                    BlockPos blockpos = p_177251_1_.getBlockPosRelative(this.field_177257_d.func_176734_d(), i);
+                    iblockstate = this.field_177261_a.getBlockState(blockpos);
+                    block = iblockstate.getBlock();
 
-                    if (iblockstate.func_185904_a() == Material.air || !BlockPistonBase.func_185646_a(iblockstate, this.field_177261_a, blockpos, this.field_177257_d, false) || blockpos.equals(this.field_177259_b))
+                    if (iblockstate.getMaterial() == Material.air || !BlockPistonBase.func_185646_a(iblockstate, this.field_177261_a, blockpos, this.field_177257_d, false) || blockpos.equals(this.field_177259_b))
                     {
                         break;
                     }
@@ -129,7 +129,7 @@
 
                 for (int j = i - 1; j >= 0; --j)
                 {
-                    this.field_177258_e.add(p_177251_1_.func_177967_a(this.field_177257_d.func_176734_d(), j));
+                    this.field_177258_e.add(p_177251_1_.getBlockPosRelative(this.field_177257_d.func_176734_d(), j));
                     ++i1;
                 }
 
@@ -137,7 +137,7 @@
 
                 while (true)
                 {
-                    BlockPos blockpos1 = p_177251_1_.func_177967_a(this.field_177257_d, j1);
+                    BlockPos blockpos1 = p_177251_1_.getBlockPosRelative(this.field_177257_d, j1);
                     int k = this.field_177258_e.indexOf(blockpos1);
 
                     if (k > -1)
@@ -148,7 +148,7 @@
                         {
                             BlockPos blockpos2 = (BlockPos)this.field_177258_e.get(l);
 
-                            if (this.field_177261_a.func_180495_p(blockpos2).func_177230_c() == Blocks.field_180399_cE && !this.func_177250_b(blockpos2))
+                            if (this.field_177261_a.getBlockState(blockpos2).getBlock() == Blocks.field_180399_cE && !this.func_177250_b(blockpos2))
                             {
                                 return false;
                             }
@@ -157,9 +157,9 @@
                         return true;
                     }
 
-                    iblockstate = this.field_177261_a.func_180495_p(blockpos1);
+                    iblockstate = this.field_177261_a.getBlockState(blockpos1);
 
-                    if (iblockstate.func_185904_a() == Material.air)
+                    if (iblockstate.getMaterial() == Material.air)
                     {
                         return true;
                     }
@@ -206,7 +206,7 @@
     {
         for (EnumFacing enumfacing : EnumFacing.values())
         {
-            if (enumfacing.func_176740_k() != this.field_177257_d.func_176740_k() && !this.func_177251_a(p_177250_1_.func_177972_a(enumfacing)))
+            if (enumfacing.func_176740_k() != this.field_177257_d.func_176740_k() && !this.func_177251_a(p_177250_1_.getBlockPosRelative(enumfacing)))
             {
                 return false;
             }
