--- a/net/minecraft/block/state/BlockStateBase.java
+++ b/net/minecraft/block/state/BlockStateBase.java
@@ -62,7 +62,7 @@
     public String toString()
     {
         StringBuilder stringbuilder = new StringBuilder();
-        stringbuilder.append(Block.blockRegistry.func_177774_c(this.func_177230_c()));
+        stringbuilder.append(Block.blockRegistry.func_177774_c(this.getBlock()));
 
         if (!this.func_177228_b().isEmpty())
         {
