--- a/net/minecraft/block/state/BlockWorldState.java
+++ b/net/minecraft/block/state/BlockWorldState.java
@@ -26,7 +26,7 @@
     {
         if (this.field_177514_c == null && (this.field_181628_c || this.field_177515_a.func_175667_e(this.field_177513_b)))
         {
-            this.field_177514_c = this.field_177515_a.func_180495_p(this.field_177513_b);
+            this.field_177514_c = this.field_177515_a.getBlockState(this.field_177513_b);
         }
 
         return this.field_177514_c;
