--- a/net/minecraft/block/state/BlockStateContainer.java
+++ b/net/minecraft/block/state/BlockStateContainer.java
@@ -214,7 +214,7 @@
             return this.field_177237_b;
         }
 
-        public Block func_177230_c()
+        public Block getBlock()
         {
             return this.field_177239_a;
         }
@@ -263,7 +263,7 @@
             return map;
         }
 
-        public Material func_185904_a()
+        public Material getMaterial()
         {
             return this.field_177239_a.getMaterial(this);
         }
