--- a/net/minecraft/entity/item/EntityBoat.java
+++ b/net/minecraft/entity/item/EntityBoat.java
@@ -344,7 +344,7 @@
 
                 if (!entity.func_184196_w(this))
                 {
-                    if (flag && this.func_184188_bt().size() < 2 && !entity.func_184218_aH() && entity.width < this.width && entity instanceof EntityLivingBase && !(entity instanceof EntityWaterMob) && !(entity instanceof EntityPlayer))
+                    if (flag && this.func_184188_bt().size() < 2 && !entity.isRiding() && entity.width < this.width && entity instanceof EntityLivingBase && !(entity instanceof EntityWaterMob) && !(entity instanceof EntityPlayer))
                     {
                         entity.func_184220_m(this);
                     }
@@ -439,7 +439,7 @@
                     {
                         if (f < 1.0F)
                         {
-                            float f2 = (float)blockpos$pooledmutableblockpos.func_177956_o() + f;
+                            float f2 = (float)blockpos$pooledmutableblockpos.getY() + f;
                             return f2;
                         }
 
@@ -449,9 +449,9 @@
                     for (int i2 = i1; i2 < j1; ++i2)
                     {
                         blockpos$pooledmutableblockpos.func_181079_c(l1, k1, i2);
-                        IBlockState iblockstate = this.worldObj.func_180495_p(blockpos$pooledmutableblockpos);
+                        IBlockState iblockstate = this.worldObj.getBlockState(blockpos$pooledmutableblockpos);
 
-                        if (iblockstate.func_185904_a() == Material.water)
+                        if (iblockstate.getMaterial() == Material.water)
                         {
                             f = Math.max(f, func_184456_a(iblockstate, this.worldObj, blockpos$pooledmutableblockpos));
                         }
@@ -505,12 +505,12 @@
                             if (j2 <= 0 || k2 != k && k2 != l - 1)
                             {
                                 blockpos$pooledmutableblockpos.func_181079_c(l1, k2, i2);
-                                IBlockState iblockstate = this.worldObj.func_180495_p(blockpos$pooledmutableblockpos);
+                                IBlockState iblockstate = this.worldObj.getBlockState(blockpos$pooledmutableblockpos);
                                 iblockstate.func_185908_a(this.worldObj, blockpos$pooledmutableblockpos, axisalignedbb1, list, this);
 
                                 if (!list.isEmpty())
                                 {
-                                    f += iblockstate.func_177230_c().slipperiness;
+                                    f += iblockstate.getBlock().slipperiness;
                                     ++k1;
                                 }
 
@@ -551,9 +551,9 @@
                     for (int i2 = i1; i2 < j1; ++i2)
                     {
                         blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2);
-                        IBlockState iblockstate = this.worldObj.func_180495_p(blockpos$pooledmutableblockpos);
+                        IBlockState iblockstate = this.worldObj.getBlockState(blockpos$pooledmutableblockpos);
 
-                        if (iblockstate.func_185904_a() == Material.water)
+                        if (iblockstate.getMaterial() == Material.water)
                         {
                             float f = func_184452_b(iblockstate, this.worldObj, blockpos$pooledmutableblockpos);
                             this.field_184465_aD = Math.max((double)f, this.field_184465_aD);
@@ -594,9 +594,9 @@
                     for (int i2 = i1; i2 < j1; ++i2)
                     {
                         blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2);
-                        IBlockState iblockstate = this.worldObj.func_180495_p(blockpos$pooledmutableblockpos);
+                        IBlockState iblockstate = this.worldObj.getBlockState(blockpos$pooledmutableblockpos);
 
-                        if (iblockstate.func_185904_a() == Material.water && d0 < (double)func_184452_b(iblockstate, this.worldObj, blockpos$pooledmutableblockpos))
+                        if (iblockstate.getMaterial() == Material.water && d0 < (double)func_184452_b(iblockstate, this.worldObj, blockpos$pooledmutableblockpos))
                         {
                             if (((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue() != 0)
                             {
@@ -621,12 +621,12 @@
     public static float func_184456_a(IBlockState p_184456_0_, IBlockAccess p_184456_1_, BlockPos p_184456_2_)
     {
         int i = ((Integer)p_184456_0_.func_177229_b(BlockLiquid.field_176367_b)).intValue();
-        return (i & 7) == 0 && p_184456_1_.func_180495_p(p_184456_2_.func_177984_a()).func_185904_a() == Material.water ? 1.0F : 1.0F - BlockLiquid.getLiquidHeightPercent(i);
+        return (i & 7) == 0 && p_184456_1_.getBlockState(p_184456_2_.getBlockPosAbove()).getMaterial() == Material.water ? 1.0F : 1.0F - BlockLiquid.getLiquidHeightPercent(i);
     }
 
     public static float func_184452_b(IBlockState p_184452_0_, IBlockAccess p_184452_1_, BlockPos p_184452_2_)
     {
-        return (float)p_184452_2_.func_177956_o() + func_184456_a(p_184452_0_, p_184452_1_, p_184452_2_);
+        return (float)p_184452_2_.getY() + func_184456_a(p_184452_0_, p_184452_1_, p_184452_2_);
     }
 
     private void func_184450_w()
@@ -755,7 +755,7 @@
                 }
             }
 
-            Vec3d vec3d = (new Vec3d((double)f, 0.0D, 0.0D)).func_178785_b(-this.rotationYaw * 0.017453292F - ((float)Math.PI / 2F));
+            Vec3d vec3d = (new Vec3d((double)f, 0.0D, 0.0D)).rotateY(-this.rotationYaw * 0.017453292F - ((float)Math.PI / 2F));
             p_184232_1_.setPosition(this.posX + vec3d.xCoord, this.posY + (double)f1, this.posZ + vec3d.zCoord);
             p_184232_1_.rotationYaw += this.field_184475_as;
             p_184232_1_.setRotationYawHead(p_184232_1_.getRotationYawHead() + this.field_184475_as);
@@ -818,7 +818,7 @@
     {
         this.field_184473_aH = this.motionY;
 
-        if (!this.func_184218_aH())
+        if (!this.isRiding())
         {
             if (p_184231_3_)
             {
@@ -853,7 +853,7 @@
 
                 this.fallDistance = 0.0F;
             }
-            else if (this.worldObj.func_180495_p((new BlockPos(this)).func_177977_b()).func_185904_a() != Material.water && p_184231_1_ < 0.0D)
+            else if (this.worldObj.getBlockState((new BlockPos(this)).getBlockPosBelow()).getMaterial() != Material.water && p_184231_1_ < 0.0D)
             {
                 this.fallDistance = (float)((double)this.fallDistance - p_184231_1_);
             }
