--- a/net/minecraft/entity/item/EntityFireworkRocket.java
+++ b/net/minecraft/entity/item/EntityFireworkRocket.java
@@ -132,14 +132,14 @@
 
         if (this.fireworkAge == 0 && !this.func_174814_R())
         {
-            this.worldObj.func_184148_a((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187631_bo, SoundCategory.AMBIENT, 3.0F, 1.0F);
+            this.worldObj.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187631_bo, SoundCategory.AMBIENT, 3.0F, 1.0F);
         }
 
         ++this.fireworkAge;
 
         if (this.worldObj.isRemote && this.fireworkAge % 2 < 2)
         {
-            this.worldObj.func_175688_a(EnumParticleTypes.FIREWORKS_SPARK, this.posX, this.posY - 0.3D, this.posZ, this.rand.nextGaussian() * 0.05D, -this.motionY * 0.5D, this.rand.nextGaussian() * 0.05D, new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.FIREWORKS_SPARK, this.posX, this.posY - 0.3D, this.posZ, this.rand.nextGaussian() * 0.05D, -this.motionY * 0.5D, this.rand.nextGaussian() * 0.05D, new int[0]);
         }
 
         if (!this.worldObj.isRemote && this.fireworkAge > this.lifetime)
