--- a/net/minecraft/entity/item/EntityFallingBlock.java
+++ b/net/minecraft/entity/item/EntityFallingBlock.java
@@ -81,7 +81,7 @@
 
     protected void entityInit()
     {
-        this.dataWatcher.func_187214_a(field_184532_d, BlockPos.field_177992_a);
+        this.dataWatcher.func_187214_a(field_184532_d, BlockPos.createDefault);
     }
 
     /**
@@ -97,9 +97,9 @@
      */
     public void onUpdate()
     {
-        Block block = this.field_175132_d.func_177230_c();
+        Block block = this.field_175132_d.getBlock();
 
-        if (this.field_175132_d.func_185904_a() == Material.air)
+        if (this.field_175132_d.getMaterial() == Material.air)
         {
             this.setDead();
         }
@@ -113,7 +113,7 @@
             {
                 BlockPos blockpos = new BlockPos(this);
 
-                if (this.worldObj.func_180495_p(blockpos).func_177230_c() == block)
+                if (this.worldObj.getBlockState(blockpos).getBlock() == block)
                 {
                     this.worldObj.func_175698_g(blockpos);
                 }
@@ -140,9 +140,9 @@
 
                 if (this.onGround)
                 {
-                    IBlockState iblockstate = this.worldObj.func_180495_p(blockpos1);
+                    IBlockState iblockstate = this.worldObj.getBlockState(blockpos1);
 
-                    if (BlockFalling.func_185759_i(this.worldObj.func_180495_p(new BlockPos(this.posX, this.posY - 0.009999999776482582D, this.posZ))))
+                    if (BlockFalling.func_185759_i(this.worldObj.getBlockState(new BlockPos(this.posX, this.posY - 0.009999999776482582D, this.posZ))))
                     {
                         this.onGround = false;
                         return;
@@ -152,13 +152,13 @@
                     this.motionZ *= 0.699999988079071D;
                     this.motionY *= -0.5D;
 
-                    if (iblockstate.func_177230_c() != Blocks.field_180384_M)
+                    if (iblockstate.getBlock() != Blocks.field_180384_M)
                     {
                         this.setDead();
 
                         if (!this.field_145808_f)
                         {
-                            if (this.worldObj.func_175716_a(block, blockpos1, true, EnumFacing.UP, (Entity)null, (ItemStack)null) && !BlockFalling.func_185759_i(this.worldObj.func_180495_p(blockpos1.func_177977_b())) && this.worldObj.func_180501_a(blockpos1, this.field_175132_d, 3))
+                            if (this.worldObj.func_175716_a(block, blockpos1, true, EnumFacing.UP, (Entity)null, (ItemStack)null) && !BlockFalling.func_185759_i(this.worldObj.getBlockState(blockpos1.getBlockPosBelow())) && this.worldObj.func_180501_a(blockpos1, this.field_175132_d, 3))
                             {
                                 if (block instanceof BlockFalling)
                                 {
@@ -195,7 +195,7 @@
                         }
                     }
                 }
-                else if (this.fallTime > 100 && !this.worldObj.isRemote && (blockpos1.func_177956_o() < 1 || blockpos1.func_177956_o() > 256) || this.fallTime > 600)
+                else if (this.fallTime > 100 && !this.worldObj.isRemote && (blockpos1.getY() < 1 || blockpos1.getY() > 256) || this.fallTime > 600)
                 {
                     if (this.shouldDropItem && this.worldObj.getGameRules().getGameRuleBooleanValue("doEntityDrops"))
                     {
@@ -210,7 +210,7 @@
 
     public void func_180430_e(float p_180430_1_, float p_180430_2_)
     {
-        Block block = this.field_175132_d.func_177230_c();
+        Block block = this.field_175132_d.getBlock();
 
         if (this.hurtEntities)
         {
@@ -254,7 +254,7 @@
      */
     protected void writeEntityToNBT(NBTTagCompound tagCompound)
     {
-        Block block = this.field_175132_d != null ? this.field_175132_d.func_177230_c() : Blocks.air;
+        Block block = this.field_175132_d != null ? this.field_175132_d.getBlock() : Blocks.air;
         ResourceLocation resourcelocation = (ResourceLocation)Block.blockRegistry.func_177774_c(block);
         tagCompound.setString("Block", resourcelocation == null ? "" : resourcelocation.toString());
         tagCompound.setByte("Data", (byte)block.func_176201_c(this.field_175132_d));
@@ -291,7 +291,7 @@
         }
 
         this.fallTime = tagCompund.getInteger("Time");
-        Block block = this.field_175132_d.func_177230_c();
+        Block block = this.field_175132_d.getBlock();
 
         if (tagCompund.hasKey("HurtEntities", 99))
         {
@@ -314,9 +314,9 @@
             this.tileEntityData = tagCompund.getCompoundTag("TileEntityData");
         }
 
-        if (block == null || block.func_176223_P().func_185904_a() == Material.air)
+        if (block == null || block.getBlockState().getMaterial() == Material.air)
         {
-            this.field_175132_d = Blocks.sand.func_176223_P();
+            this.field_175132_d = Blocks.sand.getBlockState();
         }
     }
 
@@ -344,7 +344,7 @@
 
         if (this.field_175132_d != null)
         {
-            Block block = this.field_175132_d.func_177230_c();
+            Block block = this.field_175132_d.getBlock();
             category.addCrashSection("Immitating block ID", Integer.valueOf(Block.getIdFromBlock(block)));
             category.addCrashSection("Immitating block data", Integer.valueOf(block.func_176201_c(this.field_175132_d)));
         }
