--- a/net/minecraft/entity/item/EntityEnderCrystal.java
+++ b/net/minecraft/entity/item/EntityEnderCrystal.java
@@ -67,9 +67,9 @@
         {
             BlockPos blockpos = new BlockPos(this);
 
-            if (this.worldObj.provider instanceof WorldProviderEnd && this.worldObj.func_180495_p(blockpos).func_177230_c() != Blocks.fire)
+            if (this.worldObj.provider instanceof WorldProviderEnd && this.worldObj.getBlockState(blockpos).getBlock() != Blocks.fire)
             {
-                this.worldObj.func_175656_a(blockpos, Blocks.fire.func_176223_P());
+                this.worldObj.func_175656_a(blockpos, Blocks.fire.getBlockState());
             }
         }
     }
