--- a/net/minecraft/entity/item/EntityItemFrame.java
+++ b/net/minecraft/entity/item/EntityItemFrame.java
@@ -68,7 +68,7 @@
             if (!this.worldObj.isRemote)
             {
                 this.func_146065_b(source.getEntity(), false);
-                this.func_184185_a(SoundEvents.field_187629_cO, 1.0F, 1.0F);
+                this.playSound(SoundEvents.field_187629_cO, 1.0F, 1.0F);
                 this.setDisplayedItem((ItemStack)null);
             }
 
@@ -106,13 +106,13 @@
      */
     public void onBroken(@Nullable Entity p_110128_1_)
     {
-        this.func_184185_a(SoundEvents.field_187623_cM, 1.0F, 1.0F);
+        this.playSound(SoundEvents.field_187623_cM, 1.0F, 1.0F);
         this.func_146065_b(p_110128_1_, true);
     }
 
     public void func_184523_o()
     {
-        this.func_184185_a(SoundEvents.field_187626_cN, 1.0F, 1.0F);
+        this.playSound(SoundEvents.field_187626_cN, 1.0F, 1.0F);
     }
 
     public void func_146065_b(@Nullable Entity p_146065_1_, boolean p_146065_2_)
@@ -188,7 +188,7 @@
 
         if (p_174864_1_ != null)
         {
-            this.func_184185_a(SoundEvents.field_187620_cL, 1.0F, 1.0F);
+            this.playSound(SoundEvents.field_187620_cL, 1.0F, 1.0F);
         }
 
         if (p_174864_2_ && this.field_174861_a != null)
@@ -290,7 +290,7 @@
         }
         else if (!this.worldObj.isRemote)
         {
-            this.func_184185_a(SoundEvents.field_187632_cP, 1.0F, 1.0F);
+            this.playSound(SoundEvents.field_187632_cP, 1.0F, 1.0F);
             this.setItemRotation(this.getRotation() + 1);
         }
 
