--- a/net/minecraft/entity/item/EntityArmorStand.java
+++ b/net/minecraft/entity/item/EntityArmorStand.java
@@ -125,7 +125,7 @@
     }
 
     @Nullable
-    public ItemStack func_184582_a(EntityEquipmentSlot p_184582_1_)
+    public ItemStack getEquippedItemStack(EntityEquipmentSlot p_184582_1_)
     {
         ItemStack itemstack = null;
 
@@ -142,7 +142,7 @@
         return itemstack;
     }
 
-    public void func_184201_a(EntityEquipmentSlot p_184201_1_, @Nullable ItemStack p_184201_2_)
+    public void equipItemInSlot(EntityEquipmentSlot p_184201_1_, @Nullable ItemStack p_184201_2_)
     {
         switch (p_184201_1_.func_188453_a())
         {
@@ -197,7 +197,7 @@
         }
         else
         {
-            this.func_184201_a(entityequipmentslot, p_174820_2_);
+            this.equipItemInSlot(entityequipmentslot, p_174820_2_);
             return true;
         }
     }
@@ -412,24 +412,24 @@
             boolean flag1 = this.func_175410_n();
             double d4 = flag1 ? p_184199_2_.yCoord * 2.0D : p_184199_2_.yCoord;
 
-            if (d4 >= 0.1D && d4 < 0.1D + (flag1 ? 0.8D : 0.45D) && this.func_184582_a(EntityEquipmentSlot.FEET) != null)
+            if (d4 >= 0.1D && d4 < 0.1D + (flag1 ? 0.8D : 0.45D) && this.getEquippedItemStack(EntityEquipmentSlot.FEET) != null)
             {
                 entityequipmentslot1 = EntityEquipmentSlot.FEET;
             }
-            else if (d4 >= 0.9D + (flag1 ? 0.3D : 0.0D) && d4 < 0.9D + (flag1 ? 1.0D : 0.7D) && this.func_184582_a(EntityEquipmentSlot.CHEST) != null)
+            else if (d4 >= 0.9D + (flag1 ? 0.3D : 0.0D) && d4 < 0.9D + (flag1 ? 1.0D : 0.7D) && this.getEquippedItemStack(EntityEquipmentSlot.CHEST) != null)
             {
                 entityequipmentslot1 = EntityEquipmentSlot.CHEST;
             }
-            else if (d4 >= 0.4D && d4 < 0.4D + (flag1 ? 1.0D : 0.8D) && this.func_184582_a(EntityEquipmentSlot.LEGS) != null)
+            else if (d4 >= 0.4D && d4 < 0.4D + (flag1 ? 1.0D : 0.8D) && this.getEquippedItemStack(EntityEquipmentSlot.LEGS) != null)
             {
                 entityequipmentslot1 = EntityEquipmentSlot.LEGS;
             }
-            else if (d4 >= 1.6D && this.func_184582_a(EntityEquipmentSlot.HEAD) != null)
+            else if (d4 >= 1.6D && this.getEquippedItemStack(EntityEquipmentSlot.HEAD) != null)
             {
                 entityequipmentslot1 = EntityEquipmentSlot.HEAD;
             }
 
-            boolean flag2 = this.func_184582_a(entityequipmentslot1) != null;
+            boolean flag2 = this.getEquippedItemStack(entityequipmentslot1) != null;
 
             if (this.func_184796_b(entityequipmentslot1) || this.func_184796_b(entityequipmentslot))
             {
@@ -472,7 +472,7 @@
 
     private void func_184795_a(EntityPlayer p_184795_1_, EntityEquipmentSlot p_184795_2_, @Nullable ItemStack p_184795_3_, EnumHand p_184795_4_)
     {
-        ItemStack itemstack = this.func_184582_a(p_184795_2_);
+        ItemStack itemstack = this.getEquippedItemStack(p_184795_2_);
 
         if (itemstack == null || (this.field_175442_bg & 1 << p_184795_2_.func_188452_c() + 8) == 0)
         {
@@ -482,7 +482,7 @@
                 {
                     ItemStack itemstack2 = p_184795_3_.copy();
                     itemstack2.stackSize = 1;
-                    this.func_184201_a(p_184795_2_, itemstack2);
+                    this.equipItemInSlot(p_184795_2_, itemstack2);
                 }
                 else if (p_184795_3_ != null && p_184795_3_.stackSize > 1)
                 {
@@ -490,14 +490,14 @@
                     {
                         ItemStack itemstack1 = p_184795_3_.copy();
                         itemstack1.stackSize = 1;
-                        this.func_184201_a(p_184795_2_, itemstack1);
+                        this.equipItemInSlot(p_184795_2_, itemstack1);
                         --p_184795_3_.stackSize;
                     }
                 }
                 else
                 {
-                    this.func_184201_a(p_184795_2_, p_184795_3_);
-                    p_184795_1_.func_184611_a(p_184795_4_, itemstack);
+                    this.equipItemInSlot(p_184795_2_, p_184795_3_);
+                    p_184795_1_.equipItemInHand(p_184795_4_, itemstack);
                 }
             }
         }
@@ -636,7 +636,7 @@
     {
         if (this.worldObj instanceof WorldServer)
         {
-            ((WorldServer)this.worldObj).func_175739_a(EnumParticleTypes.BLOCK_DUST, this.posX, this.posY + (double)this.height / 1.5D, this.posZ, 10, (double)(this.width / 4.0F), (double)(this.height / 4.0F), (double)(this.width / 4.0F), 0.05D, new int[] {Block.func_176210_f(Blocks.planks.func_176223_P())});
+            ((WorldServer)this.worldObj).func_175739_a(EnumParticleTypes.BLOCK_DUST, this.posX, this.posY + (double)this.height / 1.5D, this.posZ, 10, (double)(this.width / 4.0F), (double)(this.height / 4.0F), (double)(this.width / 4.0F), 0.05D, new int[] {Block.func_176210_f(Blocks.planks.getBlockState())});
         }
     }
 
@@ -664,7 +664,7 @@
 
     private void func_175409_C()
     {
-        this.worldObj.func_184148_a((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187701_j, this.func_184176_by(), 1.0F, 1.0F);
+        this.worldObj.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187701_j, this.func_184176_by(), 1.0F, 1.0F);
 
         for (int i = 0; i < this.field_184799_bw.length; ++i)
         {
@@ -672,7 +672,7 @@
             {
                 if (this.field_184799_bw[i] != null)
                 {
-                    Block.func_180635_a(this.worldObj, (new BlockPos(this)).func_177984_a(), this.field_184799_bw[i]);
+                    Block.func_180635_a(this.worldObj, (new BlockPos(this)).getBlockPosAbove(), this.field_184799_bw[i]);
                 }
 
                 this.field_184799_bw[i] = null;
@@ -685,7 +685,7 @@
             {
                 if (this.field_184800_bx[j] != null)
                 {
-                    Block.func_180635_a(this.worldObj, (new BlockPos(this)).func_177984_a(), this.field_184800_bx[j]);
+                    Block.func_180635_a(this.worldObj, (new BlockPos(this)).getBlockPosAbove(), this.field_184800_bx[j]);
                 }
 
                 this.field_184800_bx[j] = null;
