--- a/net/minecraft/entity/item/EntityMinecartHopper.java
+++ b/net/minecraft/entity/item/EntityMinecartHopper.java
@@ -25,7 +25,7 @@
     /** Whether this hopper minecart is being blocked by an activator rail. */
     private boolean isBlocked = true;
     private int transferTicker = -1;
-    private final BlockPos field_174900_c = BlockPos.field_177992_a;
+    private final BlockPos field_174900_c = BlockPos.createDefault;
 
     public EntityMinecartHopper(World p_i1720_1_)
     {
@@ -44,7 +44,7 @@
 
     public IBlockState func_180457_u()
     {
-        return Blocks.hopper.func_176223_P();
+        return Blocks.hopper.getBlockState();
     }
 
     public int getDefaultDisplayTileOffset()
