--- a/net/minecraft/entity/item/EntityEnderPearl.java
+++ b/net/minecraft/entity/item/EntityEnderPearl.java
@@ -58,7 +58,7 @@
 
         if (p_70184_1_.typeOfHit == RayTraceResult.Type.BLOCK)
         {
-            BlockPos blockpos = p_70184_1_.func_178782_a();
+            BlockPos blockpos = p_70184_1_.getBlock();
             TileEntity tileentity = this.worldObj.func_175625_s(blockpos);
 
             if (tileentity instanceof TileEntityEndGateway)
@@ -79,7 +79,7 @@
 
         for (int i = 0; i < 32; ++i)
         {
-            this.worldObj.func_175688_a(EnumParticleTypes.PORTAL, this.posX, this.posY + this.rand.nextDouble() * 2.0D, this.posZ, this.rand.nextGaussian(), 0.0D, this.rand.nextGaussian(), new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.PORTAL, this.posX, this.posY + this.rand.nextDouble() * 2.0D, this.posZ, this.rand.nextGaussian(), 0.0D, this.rand.nextGaussian(), new int[0]);
         }
 
         if (!this.worldObj.isRemote)
@@ -98,9 +98,9 @@
                         this.worldObj.spawnEntityInWorld(entityendermite);
                     }
 
-                    if (entitylivingbase.func_184218_aH())
+                    if (entitylivingbase.isRiding())
                     {
-                        entitylivingbase.func_184210_p();
+                        entitylivingbase.dismount();
                     }
 
                     entitylivingbase.setPositionAndUpdate(this.posX, this.posY, this.posZ);
