--- a/net/minecraft/world/border/WorldBorder.java
+++ b/net/minecraft/world/border/WorldBorder.java
@@ -34,7 +34,7 @@
 
     public boolean func_177746_a(BlockPos p_177746_1_)
     {
-        return (double)(p_177746_1_.func_177958_n() + 1) > this.func_177726_b() && (double)p_177746_1_.func_177958_n() < this.func_177728_d() && (double)(p_177746_1_.func_177952_p() + 1) > this.func_177736_c() && (double)p_177746_1_.func_177952_p() < this.func_177733_e();
+        return (double)(p_177746_1_.getX() + 1) > this.func_177726_b() && (double)p_177746_1_.getX() < this.func_177728_d() && (double)(p_177746_1_.getZ() + 1) > this.func_177736_c() && (double)p_177746_1_.getZ() < this.func_177733_e();
     }
 
     public boolean func_177730_a(ChunkPos p_177730_1_)
