--- a/net/minecraft/world/gen/ChunkProviderFlat.java
+++ b/net/minecraft/world/gen/ChunkProviderFlat.java
@@ -101,14 +101,14 @@
             {
                 IBlockState iblockstate = flatlayerinfo.func_175900_c();
 
-                if (iblockstate.func_177230_c() != Blocks.air)
+                if (iblockstate.getBlock() != Blocks.air)
                 {
                     flag = false;
                     this.cachedBlockIDs[i] = iblockstate;
                 }
             }
 
-            if (flatlayerinfo.func_175900_c().func_177230_c() == Blocks.air)
+            if (flatlayerinfo.func_175900_c().getBlock() == Blocks.air)
             {
                 k += flatlayerinfo.getLayerCount();
             }
@@ -119,7 +119,7 @@
             }
         }
 
-        p_i2004_1_.func_181544_b(j);
+        p_i2004_1_.setSpawnLevel(j);
         this.hasDecoration = flag && this.flatWorldGenInfo.getBiome() != Biome.func_185362_a(Biomes.field_185440_P) ? false : this.flatWorldGenInfo.getWorldFeatures().containsKey("decoration");
     }
 
@@ -193,7 +193,7 @@
         {
             BlockPos blockpos1 = blockpos.func_177982_a(this.random.nextInt(16) + 8, this.random.nextInt(this.random.nextInt(248) + 8), this.random.nextInt(16) + 8);
 
-            if (blockpos1.func_177956_o() < this.worldObj.func_181545_F() || this.random.nextInt(10) == 0)
+            if (blockpos1.getY() < this.worldObj.getSpawnLevel() || this.random.nextInt(10) == 0)
             {
                 this.lavaLakeGenerator.func_180709_b(this.worldObj, this.random, blockpos1);
             }
