--- a/net/minecraft/world/gen/MapGenCaves.java
+++ b/net/minecraft/world/gen/MapGenCaves.java
@@ -12,10 +12,10 @@
 
 public class MapGenCaves extends MapGenBase
 {
-    protected static final IBlockState field_186126_a = Blocks.lava.func_176223_P();
-    protected static final IBlockState field_186127_b = Blocks.air.func_176223_P();
-    protected static final IBlockState field_186128_c = Blocks.sandstone.func_176223_P();
-    protected static final IBlockState field_186129_d = Blocks.field_180395_cM.func_176223_P();
+    protected static final IBlockState field_186126_a = Blocks.lava.getBlockState();
+    protected static final IBlockState field_186127_b = Blocks.air.getBlockState();
+    protected static final IBlockState field_186128_c = Blocks.sandstone.getBlockState();
+    protected static final IBlockState field_186129_d = Blocks.field_180395_cM.getBlockState();
 
     protected void func_180703_a(long p_180703_1_, int p_180703_3_, int p_180703_4_, ChunkPrimer p_180703_5_, double p_180703_6_, double p_180703_8_, double p_180703_10_)
     {
@@ -142,7 +142,7 @@
                                 {
                                     IBlockState iblockstate = p_180702_5_.func_177856_a(j1, l1, k1);
 
-                                    if (iblockstate.func_177230_c() == Blocks.flowing_water || iblockstate.func_177230_c() == Blocks.water)
+                                    if (iblockstate.getBlock() == Blocks.flowing_water || iblockstate.getBlock() == Blocks.water)
                                     {
                                         flag3 = true;
                                     }
@@ -180,7 +180,7 @@
                                             IBlockState iblockstate1 = p_180702_5_.func_177856_a(j3, j2, i2);
                                             IBlockState iblockstate2 = (IBlockState)Objects.firstNonNull(p_180702_5_.func_177856_a(j3, j2 + 1, i2), field_186127_b);
 
-                                            if (iblockstate1.func_177230_c() == Blocks.grass || iblockstate1.func_177230_c() == Blocks.mycelium)
+                                            if (iblockstate1.getBlock() == Blocks.grass || iblockstate1.getBlock() == Blocks.mycelium)
                                             {
                                                 flag1 = true;
                                             }
@@ -195,10 +195,10 @@
                                                 {
                                                     p_180702_5_.func_177855_a(j3, j2, i2, field_186127_b);
 
-                                                    if (flag1 && p_180702_5_.func_177856_a(j3, j2 - 1, i2).func_177230_c() == Blocks.dirt)
+                                                    if (flag1 && p_180702_5_.func_177856_a(j3, j2 - 1, i2).getBlock() == Blocks.dirt)
                                                     {
                                                         blockpos$mutableblockpos.func_181079_c(j3 + p_180702_3_ * 16, 0, i2 + p_180702_4_ * 16);
-                                                        p_180702_5_.func_177855_a(j3, j2 - 1, i2, this.worldObj.func_180494_b(blockpos$mutableblockpos).topBlock.func_177230_c().func_176223_P());
+                                                        p_180702_5_.func_177855_a(j3, j2 - 1, i2, this.worldObj.func_180494_b(blockpos$mutableblockpos).topBlock.getBlock().getBlockState());
                                                     }
                                                 }
                                             }
@@ -220,7 +220,7 @@
 
     protected boolean func_175793_a(IBlockState p_175793_1_, IBlockState p_175793_2_)
     {
-        return p_175793_1_.func_177230_c() == Blocks.stone ? true : (p_175793_1_.func_177230_c() == Blocks.dirt ? true : (p_175793_1_.func_177230_c() == Blocks.grass ? true : (p_175793_1_.func_177230_c() == Blocks.hardened_clay ? true : (p_175793_1_.func_177230_c() == Blocks.stained_hardened_clay ? true : (p_175793_1_.func_177230_c() == Blocks.sandstone ? true : (p_175793_1_.func_177230_c() == Blocks.field_180395_cM ? true : (p_175793_1_.func_177230_c() == Blocks.mycelium ? true : (p_175793_1_.func_177230_c() == Blocks.snow_layer ? true : (p_175793_1_.func_177230_c() == Blocks.sand || p_175793_1_.func_177230_c() == Blocks.gravel) && p_175793_2_.func_185904_a() != Material.water))))))));
+        return p_175793_1_.getBlock() == Blocks.stone ? true : (p_175793_1_.getBlock() == Blocks.dirt ? true : (p_175793_1_.getBlock() == Blocks.grass ? true : (p_175793_1_.getBlock() == Blocks.hardened_clay ? true : (p_175793_1_.getBlock() == Blocks.stained_hardened_clay ? true : (p_175793_1_.getBlock() == Blocks.sandstone ? true : (p_175793_1_.getBlock() == Blocks.field_180395_cM ? true : (p_175793_1_.getBlock() == Blocks.mycelium ? true : (p_175793_1_.getBlock() == Blocks.snow_layer ? true : (p_175793_1_.getBlock() == Blocks.sand || p_175793_1_.getBlock() == Blocks.gravel) && p_175793_2_.getMaterial() != Material.water))))))));
     }
 
     protected void func_180701_a(World p_180701_1_, int p_180701_2_, int p_180701_3_, int p_180701_4_, int p_180701_5_, ChunkPrimer p_180701_6_)
