--- a/net/minecraft/world/gen/MapGenRavine.java
+++ b/net/minecraft/world/gen/MapGenRavine.java
@@ -10,8 +10,8 @@
 
 public class MapGenRavine extends MapGenBase
 {
-    protected static final IBlockState field_186135_a = Blocks.flowing_lava.func_176223_P();
-    protected static final IBlockState field_186136_b = Blocks.air.func_176223_P();
+    protected static final IBlockState field_186135_a = Blocks.flowing_lava.getBlockState();
+    protected static final IBlockState field_186136_b = Blocks.air.getBlockState();
     private final float[] field_75046_d = new float[1024];
 
     protected void func_180707_a(long p_180707_1_, int p_180707_3_, int p_180707_4_, ChunkPrimer p_180707_5_, double p_180707_6_, double p_180707_8_, double p_180707_10_, float p_180707_12_, float p_180707_13_, float p_180707_14_, int p_180707_15_, int p_180707_16_, double p_180707_17_)
@@ -130,7 +130,7 @@
                                 {
                                     IBlockState iblockstate = p_180707_5_.func_177856_a(j1, l1, k1);
 
-                                    if (iblockstate.func_177230_c() == Blocks.flowing_water || iblockstate.func_177230_c() == Blocks.water)
+                                    if (iblockstate.getBlock() == Blocks.flowing_water || iblockstate.getBlock() == Blocks.water)
                                     {
                                         flag2 = true;
                                     }
@@ -167,12 +167,12 @@
                                         {
                                             IBlockState iblockstate1 = p_180707_5_.func_177856_a(j3, j2, i2);
 
-                                            if (iblockstate1.func_177230_c() == Blocks.grass)
+                                            if (iblockstate1.getBlock() == Blocks.grass)
                                             {
                                                 flag = true;
                                             }
 
-                                            if (iblockstate1.func_177230_c() == Blocks.stone || iblockstate1.func_177230_c() == Blocks.dirt || iblockstate1.func_177230_c() == Blocks.grass)
+                                            if (iblockstate1.getBlock() == Blocks.stone || iblockstate1.getBlock() == Blocks.dirt || iblockstate1.getBlock() == Blocks.grass)
                                             {
                                                 if (j2 - 1 < 10)
                                                 {
@@ -182,7 +182,7 @@
                                                 {
                                                     p_180707_5_.func_177855_a(j3, j2, i2, field_186136_b);
 
-                                                    if (flag && p_180707_5_.func_177856_a(j3, j2 - 1, i2).func_177230_c() == Blocks.dirt)
+                                                    if (flag && p_180707_5_.func_177856_a(j3, j2 - 1, i2).getBlock() == Blocks.dirt)
                                                     {
                                                         blockpos$mutableblockpos.func_181079_c(j3 + p_180707_3_ * 16, 0, i2 + p_180707_4_ * 16);
                                                         p_180707_5_.func_177855_a(j3, j2 - 1, i2, this.worldObj.func_180494_b(blockpos$mutableblockpos).topBlock);
