--- a/net/minecraft/client/renderer/ActiveRenderInfo.java
+++ b/net/minecraft/client/renderer/ActiveRenderInfo.java
@@ -76,22 +76,22 @@
     {
         Vec3d vec3d = func_178806_a(p_186703_1_, (double)p_186703_2_);
         BlockPos blockpos = new BlockPos(vec3d);
-        IBlockState iblockstate = p_186703_0_.func_180495_p(blockpos);
+        IBlockState iblockstate = p_186703_0_.getBlockState(blockpos);
 
-        if (iblockstate.func_185904_a().isLiquid())
+        if (iblockstate.getMaterial().isLiquid())
         {
             float f = 0.0F;
 
-            if (iblockstate.func_177230_c() instanceof BlockLiquid)
+            if (iblockstate.getBlock() instanceof BlockLiquid)
             {
                 f = BlockLiquid.getLiquidHeightPercent(((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue()) - 0.11111111F;
             }
 
-            float f1 = (float)(blockpos.func_177956_o() + 1) - f;
+            float f1 = (float)(blockpos.getY() + 1) - f;
 
             if (vec3d.yCoord >= (double)f1)
             {
-                iblockstate = p_186703_0_.func_180495_p(blockpos.func_177984_a());
+                iblockstate = p_186703_0_.getBlockState(blockpos.getBlockPosAbove());
             }
         }
 
