--- a/net/minecraft/client/renderer/BlockFluidRenderer.java
+++ b/net/minecraft/client/renderer/BlockFluidRenderer.java
@@ -39,8 +39,8 @@
 
     public boolean func_178270_a(IBlockAccess p_178270_1_, IBlockState p_178270_2_, BlockPos p_178270_3_, VertexBuffer p_178270_4_)
     {
-        BlockLiquid blockliquid = (BlockLiquid)p_178270_2_.func_177230_c();
-        boolean flag = p_178270_2_.func_185904_a() == Material.lava;
+        BlockLiquid blockliquid = (BlockLiquid)p_178270_2_.getBlock();
+        boolean flag = p_178270_2_.getMaterial() == Material.lava;
         TextureAtlasSprite[] atextureatlassprite = flag ? this.field_178272_a : this.field_178271_b;
         int i = this.field_187500_a.func_186724_a(p_178270_2_, p_178270_1_, p_178270_3_, 0);
         float f = (float)(i >> 16 & 255) / 255.0F;
@@ -61,14 +61,14 @@
             float f4 = 1.0F;
             float f5 = 0.8F;
             float f6 = 0.6F;
-            Material material = p_178270_2_.func_185904_a();
+            Material material = p_178270_2_.getMaterial();
             float f7 = this.func_178269_a(p_178270_1_, p_178270_3_, material);
-            float f8 = this.func_178269_a(p_178270_1_, p_178270_3_.func_177968_d(), material);
-            float f9 = this.func_178269_a(p_178270_1_, p_178270_3_.func_177974_f().func_177968_d(), material);
-            float f10 = this.func_178269_a(p_178270_1_, p_178270_3_.func_177974_f(), material);
-            double d0 = (double)p_178270_3_.func_177958_n();
-            double d1 = (double)p_178270_3_.func_177956_o();
-            double d2 = (double)p_178270_3_.func_177952_p();
+            float f8 = this.func_178269_a(p_178270_1_, p_178270_3_.getBlockPosSouth(), material);
+            float f9 = this.func_178269_a(p_178270_1_, p_178270_3_.getBlockPosEast().getBlockPosSouth(), material);
+            float f10 = this.func_178269_a(p_178270_1_, p_178270_3_.getBlockPosEast(), material);
+            double d0 = (double)p_178270_3_.getX();
+            double d1 = (double)p_178270_3_.getY();
+            double d2 = (double)p_178270_3_.getZ();
             float f11 = 0.001F;
 
             if (flag1)
@@ -126,7 +126,7 @@
                 p_178270_4_.func_181662_b(d0 + 1.0D, d1 + (double)f9, d2 + 1.0D).func_181666_a(f24, f25, f26, 1.0F).func_187315_a((double)f15, (double)f19).func_187314_a(l2, i3).func_181675_d();
                 p_178270_4_.func_181662_b(d0 + 1.0D, d1 + (double)f10, d2 + 0.0D).func_181666_a(f24, f25, f26, 1.0F).func_187315_a((double)f16, (double)f20).func_187314_a(l2, i3).func_181675_d();
 
-                if (blockliquid.func_176364_g(p_178270_1_, p_178270_3_.func_177984_a()))
+                if (blockliquid.func_176364_g(p_178270_1_, p_178270_3_.getBlockPosAbove()))
                 {
                     p_178270_4_.func_181662_b(d0 + 0.0D, d1 + (double)f7, d2 + 0.0D).func_181666_a(f24, f25, f26, 1.0F).func_187315_a((double)f13, (double)f17).func_187314_a(l2, i3).func_181675_d();
                     p_178270_4_.func_181662_b(d0 + 1.0D, d1 + (double)f10, d2 + 0.0D).func_181666_a(f24, f25, f26, 1.0F).func_187315_a((double)f16, (double)f20).func_187314_a(l2, i3).func_181675_d();
@@ -141,7 +141,7 @@
                 float f36 = atextureatlassprite[0].getMaxU();
                 float f37 = atextureatlassprite[0].getMinV();
                 float f38 = atextureatlassprite[0].getMaxV();
-                int l1 = p_178270_2_.func_185889_a(p_178270_1_, p_178270_3_.func_177977_b());
+                int l1 = p_178270_2_.func_185889_a(p_178270_1_, p_178270_3_.getBlockPosBelow());
                 int i2 = l1 >> 16 & 65535;
                 int j2 = l1 & 65535;
                 p_178270_4_.func_181662_b(d0, d1, d2 + 1.0D).func_181666_a(0.5F, 0.5F, 0.5F, 1.0F).func_187315_a((double)f35, (double)f38).func_187314_a(i2, j2).func_181675_d();
@@ -181,7 +181,7 @@
 
                 if (!flag)
                 {
-                    Block block = p_178270_1_.func_180495_p(blockpos).func_177230_c();
+                    Block block = p_178270_1_.getBlockState(blockpos).getBlock();
 
                     if (block == Blocks.glass || block == Blocks.stained_glass)
                     {
@@ -276,13 +276,13 @@
         {
             BlockPos blockpos = p_178269_2_.func_177982_a(-(j & 1), 0, -(j >> 1 & 1));
 
-            if (p_178269_1_.func_180495_p(blockpos.func_177984_a()).func_185904_a() == p_178269_3_)
+            if (p_178269_1_.getBlockState(blockpos.getBlockPosAbove()).getMaterial() == p_178269_3_)
             {
                 return 1.0F;
             }
 
-            IBlockState iblockstate = p_178269_1_.func_180495_p(blockpos);
-            Material material = iblockstate.func_185904_a();
+            IBlockState iblockstate = p_178269_1_.getBlockState(blockpos);
+            Material material = iblockstate.getMaterial();
 
             if (material != p_178269_3_)
             {
