--- a/net/minecraft/client/renderer/ItemRenderer.java
+++ b/net/minecraft/client/renderer/ItemRenderer.java
@@ -1,5 +1,11 @@
 package net.minecraft.client.renderer;
 
+
+import com.mtbs3d.minecrift.render.OpenGLdebugging;
+
+import de.fruitfly.ovr.structs.Matrix4f;
+import de.fruitfly.ovr.structs.Vector3f;
+
 import com.google.common.base.Objects;
 import javax.annotation.Nullable;
 import net.minecraft.block.Block;
@@ -50,6 +56,11 @@
     private final RenderManager renderManager;
     private final RenderItem itemRenderer;
 
+    //VIVECRAFT
+    ItemStack itemToRender;
+    int controller = 0; 
+    ////
+    
     public ItemRenderer(Minecraft mcIn)
     {
         this.mc = mcIn;
@@ -402,7 +413,7 @@
             {
                 int j = flag1 ? 1 : -1;
 
-                switch (null.$SwitchMap$net$minecraft$item$EnumAction[p_187457_6_.getItemUseAction().ordinal()])
+                switch (p_187457_6_.getItemUseAction().ordinal())
                 {
                     case 1:
                         this.transformSideFirstPerson(enumhandside, p_187457_7_);
