--- a/net/minecraft/client/renderer/ItemRenderer.java
+++ b/net/minecraft/client/renderer/ItemRenderer.java
@@ -63,7 +63,7 @@
         {
             Item item = p_187462_2_.getItem();
             Block block = Block.getBlockFromItem(item);
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             boolean flag = this.field_178112_h.func_175050_a(p_187462_2_) && this.func_178107_a(block);
 
             if (flag)
@@ -78,7 +78,7 @@
                 GlStateManager.func_179132_a(true);
             }
 
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
     }
 
@@ -89,11 +89,11 @@
 
     private void func_178101_a(float p_178101_1_, float p_178101_2_)
     {
-        GlStateManager.func_179094_E();
-        GlStateManager.func_179114_b(p_178101_1_, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b(p_178101_2_, 0.0F, 1.0F, 0.0F);
+        GlStateManager.push();
+        GlStateManager.rotate(p_178101_1_, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(p_178101_2_, 0.0F, 1.0F, 0.0F);
         RenderHelper.enableStandardItemLighting();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 
     private void func_187464_b()
@@ -110,8 +110,8 @@
         EntityPlayerSP entityplayersp = this.mc.thePlayer;
         float f = entityplayersp.prevRenderArmPitch + (entityplayersp.renderArmPitch - entityplayersp.prevRenderArmPitch) * p_187458_1_;
         float f1 = entityplayersp.prevRenderArmYaw + (entityplayersp.renderArmYaw - entityplayersp.prevRenderArmYaw) * p_187458_1_;
-        GlStateManager.func_179114_b((entityplayersp.rotationPitch - f) * 0.1F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b((entityplayersp.rotationYaw - f1) * 0.1F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate((entityplayersp.rotationPitch - f) * 0.1F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate((entityplayersp.rotationYaw - f1) * 0.1F, 0.0F, 1.0F, 0.0F);
     }
 
     private float func_178100_c(float p_178100_1_)
@@ -127,11 +127,11 @@
         if (!this.mc.thePlayer.isInvisible())
         {
             GlStateManager.func_179129_p();
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179114_b(90.0F, 0.0F, 1.0F, 0.0F);
+            GlStateManager.push();
+            GlStateManager.rotate(90.0F, 0.0F, 1.0F, 0.0F);
             this.func_187455_a(EnumHandSide.RIGHT);
             this.func_187455_a(EnumHandSide.LEFT);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
             GlStateManager.func_179089_o();
         }
     }
@@ -141,11 +141,11 @@
         this.mc.getTextureManager().bindTexture(this.mc.thePlayer.getLocationSkin());
         Render<AbstractClientPlayer> render = this.field_178111_g.<AbstractClientPlayer>getEntityRenderObject(this.mc.thePlayer);
         RenderPlayer renderplayer = (RenderPlayer)render;
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         float f = p_187455_1_ == EnumHandSide.RIGHT ? 1.0F : -1.0F;
-        GlStateManager.func_179114_b(92.0F, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b(45.0F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b(f * -41.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.rotate(92.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(45.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(f * -41.0F, 0.0F, 0.0F, 1.0F);
         GlStateManager.func_179109_b(f * 0.3F, -1.1F, 0.45F);
 
         if (p_187455_1_ == EnumHandSide.RIGHT)
@@ -157,7 +157,7 @@
             renderplayer.func_177139_c(this.mc.thePlayer);
         }
 
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 
     private void func_187465_a(float p_187465_1_, EnumHandSide p_187465_2_, float p_187465_3_, ItemStack p_187465_4_)
@@ -167,13 +167,13 @@
 
         if (!this.mc.thePlayer.isInvisible())
         {
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179114_b(f * 10.0F, 0.0F, 0.0F, 1.0F);
+            GlStateManager.push();
+            GlStateManager.rotate(f * 10.0F, 0.0F, 0.0F, 1.0F);
             this.func_187456_a(p_187465_1_, p_187465_3_, p_187465_2_);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
 
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         GlStateManager.func_179109_b(f * 0.51F, -0.08F + p_187465_1_ * -1.2F, -0.75F);
         float f1 = MathHelper.sqrt_float(p_187465_3_);
         float f2 = MathHelper.sin(f1 * (float)Math.PI);
@@ -181,10 +181,10 @@
         float f4 = 0.4F * MathHelper.sin(f1 * ((float)Math.PI * 2F));
         float f5 = -0.3F * MathHelper.sin(p_187465_3_ * (float)Math.PI);
         GlStateManager.func_179109_b(f * f3, f4 - 0.3F * f2, f5);
-        GlStateManager.func_179114_b(f2 * -45.0F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b(f * f2 * -30.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(f2 * -45.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(f * f2 * -30.0F, 0.0F, 1.0F, 0.0F);
         this.func_187461_a(p_187465_4_);
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 
     private void func_187463_a(float p_187463_1_, float p_187463_2_, float p_187463_3_)
@@ -195,25 +195,25 @@
         GlStateManager.func_179109_b(0.0F, -f1 / 2.0F, f2);
         float f3 = this.func_178100_c(p_187463_1_);
         GlStateManager.func_179109_b(0.0F, 0.04F + p_187463_2_ * -1.2F + f3 * -0.5F, -0.72F);
-        GlStateManager.func_179114_b(f3 * -85.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(f3 * -85.0F, 1.0F, 0.0F, 0.0F);
         this.func_187466_c();
         float f4 = MathHelper.sin(f * (float)Math.PI);
-        GlStateManager.func_179114_b(f4 * 20.0F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179152_a(2.0F, 2.0F, 2.0F);
+        GlStateManager.rotate(f4 * 20.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.scale(2.0F, 2.0F, 2.0F);
         this.func_187461_a(this.field_187467_d);
     }
 
     private void func_187461_a(ItemStack p_187461_1_)
     {
-        GlStateManager.func_179114_b(180.0F, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b(180.0F, 0.0F, 0.0F, 1.0F);
-        GlStateManager.func_179152_a(0.38F, 0.38F, 0.38F);
+        GlStateManager.rotate(180.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(180.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.scale(0.38F, 0.38F, 0.38F);
         GlStateManager.func_179140_f();
         this.mc.getTextureManager().bindTexture(RES_MAP_BACKGROUND);
         Tessellator tessellator = Tessellator.func_178181_a();
         VertexBuffer vertexbuffer = tessellator.func_178180_c();
         GlStateManager.func_179109_b(-0.5F, -0.5F, 0.0F);
-        GlStateManager.func_179152_a(0.0078125F, 0.0078125F, 0.0078125F);
+        GlStateManager.scale(0.0078125F, 0.0078125F, 0.0078125F);
         vertexbuffer.func_181668_a(7, DefaultVertexFormats.field_181707_g);
         vertexbuffer.func_181662_b(-7.0D, 135.0D, 0.0D).func_187315_a(0.0D, 1.0D).func_181675_d();
         vertexbuffer.func_181662_b(135.0D, 135.0D, 0.0D).func_187315_a(1.0D, 1.0D).func_181675_d();
@@ -239,17 +239,17 @@
         float f3 = 0.4F * MathHelper.sin(f1 * ((float)Math.PI * 2F));
         float f4 = -0.4F * MathHelper.sin(p_187456_2_ * (float)Math.PI);
         GlStateManager.func_179109_b(f * (f2 + 0.64000005F), f3 + -0.6F + p_187456_1_ * -0.6F, f4 + -0.71999997F);
-        GlStateManager.func_179114_b(f * 45.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(f * 45.0F, 0.0F, 1.0F, 0.0F);
         float f5 = MathHelper.sin(p_187456_2_ * p_187456_2_ * (float)Math.PI);
         float f6 = MathHelper.sin(f1 * (float)Math.PI);
-        GlStateManager.func_179114_b(f * f6 * 70.0F, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b(f * f5 * -20.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.rotate(f * f6 * 70.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(f * f5 * -20.0F, 0.0F, 0.0F, 1.0F);
         AbstractClientPlayer abstractclientplayer = this.mc.thePlayer;
         this.mc.getTextureManager().bindTexture(abstractclientplayer.getLocationSkin());
         GlStateManager.func_179109_b(f * -1.0F, 3.6F, 3.5F);
-        GlStateManager.func_179114_b(f * 120.0F, 0.0F, 0.0F, 1.0F);
-        GlStateManager.func_179114_b(200.0F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b(f * -135.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(f * 120.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.rotate(200.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(f * -135.0F, 0.0F, 1.0F, 0.0F);
         GlStateManager.func_179109_b(f * 5.6F, 0.0F, 0.0F);
         RenderPlayer renderplayer = (RenderPlayer)this.field_178111_g.getEntityRenderObject(abstractclientplayer);
         GlStateManager.func_179129_p();
@@ -280,20 +280,20 @@
         float f3 = 1.0F - (float)Math.pow((double)f1, 27.0D);
         int i = p_187454_2_ == EnumHandSide.RIGHT ? 1 : -1;
         GlStateManager.func_179109_b(f3 * 0.6F * (float)i, f3 * -0.5F, f3 * 0.0F);
-        GlStateManager.func_179114_b((float)i * f3 * 90.0F, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b(f3 * 10.0F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b((float)i * f3 * 30.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.rotate((float)i * f3 * 90.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(f3 * 10.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate((float)i * f3 * 30.0F, 0.0F, 0.0F, 1.0F);
     }
 
     private void func_187453_a(EnumHandSide p_187453_1_, float p_187453_2_)
     {
         int i = p_187453_1_ == EnumHandSide.RIGHT ? 1 : -1;
         float f = MathHelper.sin(p_187453_2_ * p_187453_2_ * (float)Math.PI);
-        GlStateManager.func_179114_b((float)i * (45.0F + f * -20.0F), 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate((float)i * (45.0F + f * -20.0F), 0.0F, 1.0F, 0.0F);
         float f1 = MathHelper.sin(MathHelper.sqrt_float(p_187453_2_) * (float)Math.PI);
-        GlStateManager.func_179114_b((float)i * f1 * -20.0F, 0.0F, 0.0F, 1.0F);
-        GlStateManager.func_179114_b(f1 * -80.0F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b((float)i * -45.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate((float)i * f1 * -20.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.rotate(f1 * -80.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate((float)i * -45.0F, 0.0F, 1.0F, 0.0F);
     }
 
     private void func_187459_b(EnumHandSide p_187459_1_, float p_187459_2_)
@@ -354,7 +354,7 @@
     {
         boolean flag = p_187457_4_ == EnumHand.MAIN_HAND;
         EnumHandSide enumhandside = flag ? p_187457_1_.func_184591_cq() : p_187457_1_.func_184591_cq().func_188468_a();
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
 
         if (p_187457_6_ == null)
         {
@@ -401,9 +401,9 @@
                     case BOW:
                         this.func_187459_b(enumhandside, p_187457_7_);
                         GlStateManager.func_179109_b((float)j * -0.2785682F, 0.18344387F, 0.15731531F);
-                        GlStateManager.func_179114_b(-13.935F, 1.0F, 0.0F, 0.0F);
-                        GlStateManager.func_179114_b((float)j * 35.3F, 0.0F, 1.0F, 0.0F);
-                        GlStateManager.func_179114_b((float)j * -9.785F, 0.0F, 0.0F, 1.0F);
+                        GlStateManager.rotate(-13.935F, 1.0F, 0.0F, 0.0F);
+                        GlStateManager.rotate((float)j * 35.3F, 0.0F, 1.0F, 0.0F);
+                        GlStateManager.rotate((float)j * -9.785F, 0.0F, 0.0F, 1.0F);
                         float f5 = (float)p_187457_6_.getMaxItemUseDuration() - ((float)this.mc.thePlayer.func_184605_cv() - p_187457_2_ + 1.0F);
                         float f6 = f5 / 20.0F;
                         f6 = (f6 * f6 + f6 * 2.0F) / 3.0F;
@@ -422,8 +422,8 @@
                         }
 
                         GlStateManager.func_179109_b(f6 * 0.0F, f6 * 0.0F, f6 * 0.04F);
-                        GlStateManager.func_179152_a(1.0F, 1.0F, 1.0F + f6 * 0.2F);
-                        GlStateManager.func_179114_b((float)j * 45.0F, 0.0F, -1.0F, 0.0F);
+                        GlStateManager.scale(1.0F, 1.0F, 1.0F + f6 * 0.2F);
+                        GlStateManager.rotate((float)j * 45.0F, 0.0F, -1.0F, 0.0F);
                 }
             }
             else
@@ -440,7 +440,7 @@
             this.func_187462_a(p_187457_1_, p_187457_6_, flag1 ? ItemCameraTransforms.TransformType.FIRST_PERSON_RIGHT_HAND : ItemCameraTransforms.TransformType.FIRST_PERSON_LEFT_HAND, !flag1);
         }
 
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 
     /**
@@ -452,7 +452,7 @@
 
         if (this.mc.thePlayer.isEntityInsideOpaqueBlock())
         {
-            IBlockState iblockstate = this.mc.theWorld.func_180495_p(new BlockPos(this.mc.thePlayer));
+            IBlockState iblockstate = this.mc.theWorld.getBlockState(new BlockPos(this.mc.thePlayer));
             EntityPlayer entityplayer = this.mc.thePlayer;
 
             for (int i = 0; i < 8; ++i)
@@ -461,9 +461,9 @@
                 double d1 = entityplayer.posY + (double)(((float)((i >> 1) % 2) - 0.5F) * 0.1F);
                 double d2 = entityplayer.posZ + (double)(((float)((i >> 2) % 2) - 0.5F) * entityplayer.width * 0.8F);
                 BlockPos blockpos = new BlockPos(d0, d1 + (double)entityplayer.getEyeHeight(), d2);
-                IBlockState iblockstate1 = this.mc.theWorld.func_180495_p(blockpos);
+                IBlockState iblockstate1 = this.mc.theWorld.getBlockState(blockpos);
 
-                if (iblockstate1.func_177230_c().func_176214_u())
+                if (iblockstate1.getBlock().func_176214_u())
                 {
                     iblockstate = iblockstate1;
                 }
@@ -498,7 +498,7 @@
         VertexBuffer vertexbuffer = tessellator.func_178180_c();
         float f = 0.1F;
         GlStateManager.func_179131_c(0.1F, 0.1F, 0.1F, 0.5F);
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         float f1 = -1.0F;
         float f2 = 1.0F;
         float f3 = -1.0F;
@@ -514,7 +514,7 @@
         vertexbuffer.func_181662_b(1.0D, 1.0D, -0.5D).func_187315_a((double)f6, (double)f8).func_181675_d();
         vertexbuffer.func_181662_b(-1.0D, 1.0D, -0.5D).func_187315_a((double)f7, (double)f8).func_181675_d();
         tessellator.draw();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
     }
 
@@ -531,7 +531,7 @@
         GlStateManager.func_179131_c(f, f, f, 0.5F);
         GlStateManager.func_179147_l();
         GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         float f1 = 4.0F;
         float f2 = -1.0F;
         float f3 = 1.0F;
@@ -546,7 +546,7 @@
         vertexbuffer.func_181662_b(1.0D, 1.0D, -0.5D).func_187315_a((double)(0.0F + f7), (double)(0.0F + f8)).func_181675_d();
         vertexbuffer.func_181662_b(-1.0D, 1.0D, -0.5D).func_187315_a((double)(4.0F + f7), (double)(0.0F + f8)).func_181675_d();
         tessellator.draw();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
         GlStateManager.func_179084_k();
     }
@@ -567,7 +567,7 @@
 
         for (int i = 0; i < 2; ++i)
         {
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             TextureAtlasSprite textureatlassprite = this.mc.getTextureMapBlocks().getAtlasSprite("minecraft:blocks/fire_layer_1");
             this.mc.getTextureManager().bindTexture(TextureMap.locationBlocksTexture);
             float f1 = textureatlassprite.getMinU();
@@ -580,14 +580,14 @@
             float f8 = 0.5F;
             float f9 = -0.5F;
             GlStateManager.func_179109_b((float)(-(i * 2 - 1)) * 0.24F, -0.3F, 0.0F);
-            GlStateManager.func_179114_b((float)(i * 2 - 1) * 10.0F, 0.0F, 1.0F, 0.0F);
+            GlStateManager.rotate((float)(i * 2 - 1) * 10.0F, 0.0F, 1.0F, 0.0F);
             vertexbuffer.func_181668_a(7, DefaultVertexFormats.field_181707_g);
             vertexbuffer.func_181662_b(-0.5D, -0.5D, -0.5D).func_187315_a((double)f2, (double)f4).func_181675_d();
             vertexbuffer.func_181662_b(0.5D, -0.5D, -0.5D).func_187315_a((double)f1, (double)f4).func_181675_d();
             vertexbuffer.func_181662_b(0.5D, 0.5D, -0.5D).func_187315_a((double)f1, (double)f3).func_181675_d();
             vertexbuffer.func_181662_b(-0.5D, 0.5D, -0.5D).func_187315_a((double)f2, (double)f3).func_181675_d();
             tessellator.draw();
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
 
         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
@@ -601,8 +601,8 @@
         this.field_187470_g = this.field_187469_f;
         this.field_187472_i = this.field_187471_h;
         EntityPlayerSP entityplayersp = this.mc.thePlayer;
-        ItemStack itemstack = entityplayersp.func_184614_ca();
-        ItemStack itemstack1 = entityplayersp.func_184592_cb();
+        ItemStack itemstack = entityplayersp.getMainHandItem();
+        ItemStack itemstack1 = entityplayersp.getOffhandItem();
 
         if (entityplayersp.func_184838_M())
         {
