--- a/net/minecraft/client/renderer/ViewFrustum.java
+++ b/net/minecraft/client/renderer/ViewFrustum.java
@@ -144,9 +144,9 @@
     @Nullable
     protected RenderChunk func_178161_a(BlockPos p_178161_1_)
     {
-        int i = MathHelper.bucketInt(p_178161_1_.func_177958_n(), 16);
-        int j = MathHelper.bucketInt(p_178161_1_.func_177956_o(), 16);
-        int k = MathHelper.bucketInt(p_178161_1_.func_177952_p(), 16);
+        int i = MathHelper.bucketInt(p_178161_1_.getX(), 16);
+        int j = MathHelper.bucketInt(p_178161_1_.getY(), 16);
+        int k = MathHelper.bucketInt(p_178161_1_.getZ(), 16);
 
         if (j >= 0 && j < this.field_178168_c)
         {
