--- a/net/minecraft/world/IBlockAccess.java
+++ b/net/minecraft/world/IBlockAccess.java
@@ -14,9 +14,9 @@
 
     int func_175626_b(BlockPos p_175626_1_, int p_175626_2_);
 
-    IBlockState func_180495_p(BlockPos p_180495_1_);
+    IBlockState getBlockState(BlockPos p_180495_1_);
 
-    boolean func_175623_d(BlockPos p_175623_1_);
+    boolean isEmpty(BlockPos p_175623_1_);
 
     Biome func_180494_b(BlockPos p_180494_1_);
 
