--- a/net/minecraft/world/World.java
+++ b/net/minecraft/world/World.java
@@ -59,7 +59,7 @@
 
 public abstract class World implements IBlockAccess
 {
-    private int field_181546_a = 63;
+    private int spawnLevel = 63;
 
     /**
      * boolean; if true updates scheduled by scheduleBlockUpdate happen immediately
@@ -174,7 +174,7 @@
     {
         if (this.func_175667_e(p_180494_1_))
         {
-            Chunk chunk = this.func_175726_f(p_180494_1_);
+            Chunk chunk = this.getChunk(p_180494_1_);
 
             try
             {
@@ -229,31 +229,31 @@
         this.func_175652_B(new BlockPos(8, 64, 8));
     }
 
-    public IBlockState func_184141_c(BlockPos p_184141_1_)
+    public IBlockState getBlockSomething(BlockPos p_184141_1_)
     {
         BlockPos blockpos;
 
-        for (blockpos = new BlockPos(p_184141_1_.func_177958_n(), this.func_181545_F(), p_184141_1_.func_177952_p()); !this.func_175623_d(blockpos.func_177984_a()); blockpos = blockpos.func_177984_a())
+        for (blockpos = new BlockPos(p_184141_1_.getX(), this.getSpawnLevel(), p_184141_1_.getZ()); !this.isEmpty(blockpos.getBlockPosAbove()); blockpos = blockpos.getBlockPosAbove())
         {
             ;
         }
 
-        return this.func_180495_p(blockpos);
+        return this.getBlockState(blockpos);
     }
 
     private boolean func_175701_a(BlockPos p_175701_1_)
     {
-        return !this.func_189509_E(p_175701_1_) && p_175701_1_.func_177958_n() >= -30000000 && p_175701_1_.func_177952_p() >= -30000000 && p_175701_1_.func_177958_n() < 30000000 && p_175701_1_.func_177952_p() < 30000000;
+        return !this.isBlockYPosInvalid(p_175701_1_) && p_175701_1_.getX() >= -30000000 && p_175701_1_.getZ() >= -30000000 && p_175701_1_.getX() < 30000000 && p_175701_1_.getZ() < 30000000;
     }
 
-    private boolean func_189509_E(BlockPos p_189509_1_)
+    private boolean isBlockYPosInvalid(BlockPos p_189509_1_)
     {
-        return p_189509_1_.func_177956_o() < 0 || p_189509_1_.func_177956_o() >= 256;
+        return p_189509_1_.getY() < 0 || p_189509_1_.getY() >= 256;
     }
 
-    public boolean func_175623_d(BlockPos p_175623_1_)
+    public boolean isEmpty(BlockPos p_175623_1_)
     {
-        return this.func_180495_p(p_175623_1_).func_185904_a() == Material.air;
+        return this.getBlockState(p_175623_1_).getMaterial() == Material.air;
     }
 
     public boolean func_175667_e(BlockPos p_175667_1_)
@@ -263,7 +263,7 @@
 
     public boolean func_175668_a(BlockPos p_175668_1_, boolean p_175668_2_)
     {
-        return this.func_175680_a(p_175668_1_.func_177958_n() >> 4, p_175668_1_.func_177952_p() >> 4, p_175668_2_);
+        return this.func_175680_a(p_175668_1_.getX() >> 4, p_175668_1_.getZ() >> 4, p_175668_2_);
     }
 
     public boolean func_175697_a(BlockPos p_175697_1_, int p_175697_2_)
@@ -273,7 +273,7 @@
 
     public boolean func_175648_a(BlockPos p_175648_1_, int p_175648_2_, boolean p_175648_3_)
     {
-        return this.func_175663_a(p_175648_1_.func_177958_n() - p_175648_2_, p_175648_1_.func_177956_o() - p_175648_2_, p_175648_1_.func_177952_p() - p_175648_2_, p_175648_1_.func_177958_n() + p_175648_2_, p_175648_1_.func_177956_o() + p_175648_2_, p_175648_1_.func_177952_p() + p_175648_2_, p_175648_3_);
+        return this.func_175663_a(p_175648_1_.getX() - p_175648_2_, p_175648_1_.getY() - p_175648_2_, p_175648_1_.getZ() - p_175648_2_, p_175648_1_.getX() + p_175648_2_, p_175648_1_.getY() + p_175648_2_, p_175648_1_.getZ() + p_175648_2_, p_175648_3_);
     }
 
     public boolean func_175707_a(BlockPos p_175707_1_, BlockPos p_175707_2_)
@@ -283,7 +283,7 @@
 
     public boolean func_175706_a(BlockPos p_175706_1_, BlockPos p_175706_2_, boolean p_175706_3_)
     {
-        return this.func_175663_a(p_175706_1_.func_177958_n(), p_175706_1_.func_177956_o(), p_175706_1_.func_177952_p(), p_175706_2_.func_177958_n(), p_175706_2_.func_177956_o(), p_175706_2_.func_177952_p(), p_175706_3_);
+        return this.func_175663_a(p_175706_1_.getX(), p_175706_1_.getY(), p_175706_1_.getZ(), p_175706_2_.getX(), p_175706_2_.getY(), p_175706_2_.getZ(), p_175706_3_);
     }
 
     public boolean func_175711_a(StructureBoundingBox p_175711_1_)
@@ -326,9 +326,9 @@
 
     protected abstract boolean func_175680_a(int p_175680_1_, int p_175680_2_, boolean p_175680_3_);
 
-    public Chunk func_175726_f(BlockPos p_175726_1_)
+    public Chunk getChunk(BlockPos p_175726_1_)
     {
-        return this.getChunkFromChunkCoords(p_175726_1_.func_177958_n() >> 4, p_175726_1_.func_177952_p() >> 4);
+        return this.getChunkFromChunkCoords(p_175726_1_.getX() >> 4, p_175726_1_.getZ() >> 4);
     }
 
     /**
@@ -341,7 +341,7 @@
 
     public boolean func_180501_a(BlockPos p_180501_1_, IBlockState p_180501_2_, int p_180501_3_)
     {
-        if (this.func_189509_E(p_180501_1_))
+        if (this.isBlockYPosInvalid(p_180501_1_))
         {
             return false;
         }
@@ -351,8 +351,8 @@
         }
         else
         {
-            Chunk chunk = this.func_175726_f(p_180501_1_);
-            Block block = p_180501_2_.func_177230_c();
+            Chunk chunk = this.getChunk(p_180501_1_);
+            Block block = p_180501_2_.getBlock();
             IBlockState iblockstate = chunk.func_177436_a(p_180501_1_, p_180501_2_);
 
             if (iblockstate == null)
@@ -375,7 +375,7 @@
 
                 if (!this.isRemote && (p_180501_3_ & 1) != 0)
                 {
-                    this.func_175722_b(p_180501_1_, iblockstate.func_177230_c());
+                    this.func_175722_b(p_180501_1_, iblockstate.getBlock());
 
                     if (p_180501_2_.func_185912_n())
                     {
@@ -390,15 +390,15 @@
 
     public boolean func_175698_g(BlockPos p_175698_1_)
     {
-        return this.func_180501_a(p_175698_1_, Blocks.air.func_176223_P(), 3);
+        return this.func_180501_a(p_175698_1_, Blocks.air.getBlockState(), 3);
     }
 
     public boolean func_175655_b(BlockPos p_175655_1_, boolean p_175655_2_)
     {
-        IBlockState iblockstate = this.func_180495_p(p_175655_1_);
-        Block block = iblockstate.func_177230_c();
+        IBlockState iblockstate = this.getBlockState(p_175655_1_);
+        Block block = iblockstate.getBlock();
 
-        if (iblockstate.func_185904_a() == Material.air)
+        if (iblockstate.getMaterial() == Material.air)
         {
             return false;
         }
@@ -411,7 +411,7 @@
                 block.func_176226_b(this, p_175655_1_, iblockstate, 0);
             }
 
-            return this.func_180501_a(p_175655_1_, Blocks.air.func_176223_P(), 3);
+            return this.func_180501_a(p_175655_1_, Blocks.air.getBlockState(), 3);
         }
     }
 
@@ -461,7 +461,7 @@
 
     public void func_175704_b(BlockPos p_175704_1_, BlockPos p_175704_2_)
     {
-        this.markBlockRangeForRenderUpdate(p_175704_1_.func_177958_n(), p_175704_1_.func_177956_o(), p_175704_1_.func_177952_p(), p_175704_2_.func_177958_n(), p_175704_2_.func_177956_o(), p_175704_2_.func_177952_p());
+        this.markBlockRangeForRenderUpdate(p_175704_1_.getX(), p_175704_1_.getY(), p_175704_1_.getZ(), p_175704_2_.getX(), p_175704_2_.getY(), p_175704_2_.getZ());
     }
 
     public void markBlockRangeForRenderUpdate(int p_147458_1_, int p_147458_2_, int p_147458_3_, int p_147458_4_, int p_147458_5_, int p_147458_6_)
@@ -474,44 +474,44 @@
 
     public void func_175685_c(BlockPos p_175685_1_, Block p_175685_2_)
     {
-        this.func_180496_d(p_175685_1_.func_177976_e(), p_175685_2_);
-        this.func_180496_d(p_175685_1_.func_177974_f(), p_175685_2_);
-        this.func_180496_d(p_175685_1_.func_177977_b(), p_175685_2_);
-        this.func_180496_d(p_175685_1_.func_177984_a(), p_175685_2_);
-        this.func_180496_d(p_175685_1_.func_177978_c(), p_175685_2_);
-        this.func_180496_d(p_175685_1_.func_177968_d(), p_175685_2_);
+        this.func_180496_d(p_175685_1_.getBlockPosWest(), p_175685_2_);
+        this.func_180496_d(p_175685_1_.getBlockPosEast(), p_175685_2_);
+        this.func_180496_d(p_175685_1_.getBlockPosBelow(), p_175685_2_);
+        this.func_180496_d(p_175685_1_.getBlockPosAbove(), p_175685_2_);
+        this.func_180496_d(p_175685_1_.getBlockPosNorth(), p_175685_2_);
+        this.func_180496_d(p_175685_1_.getBlockPosSouth(), p_175685_2_);
     }
 
     public void func_175695_a(BlockPos p_175695_1_, Block p_175695_2_, EnumFacing p_175695_3_)
     {
         if (p_175695_3_ != EnumFacing.WEST)
         {
-            this.func_180496_d(p_175695_1_.func_177976_e(), p_175695_2_);
+            this.func_180496_d(p_175695_1_.getBlockPosWest(), p_175695_2_);
         }
 
         if (p_175695_3_ != EnumFacing.EAST)
         {
-            this.func_180496_d(p_175695_1_.func_177974_f(), p_175695_2_);
+            this.func_180496_d(p_175695_1_.getBlockPosEast(), p_175695_2_);
         }
 
         if (p_175695_3_ != EnumFacing.DOWN)
         {
-            this.func_180496_d(p_175695_1_.func_177977_b(), p_175695_2_);
+            this.func_180496_d(p_175695_1_.getBlockPosBelow(), p_175695_2_);
         }
 
         if (p_175695_3_ != EnumFacing.UP)
         {
-            this.func_180496_d(p_175695_1_.func_177984_a(), p_175695_2_);
+            this.func_180496_d(p_175695_1_.getBlockPosAbove(), p_175695_2_);
         }
 
         if (p_175695_3_ != EnumFacing.NORTH)
         {
-            this.func_180496_d(p_175695_1_.func_177978_c(), p_175695_2_);
+            this.func_180496_d(p_175695_1_.getBlockPosNorth(), p_175695_2_);
         }
 
         if (p_175695_3_ != EnumFacing.SOUTH)
         {
-            this.func_180496_d(p_175695_1_.func_177968_d(), p_175695_2_);
+            this.func_180496_d(p_175695_1_.getBlockPosSouth(), p_175695_2_);
         }
     }
 
@@ -519,7 +519,7 @@
     {
         if (!this.isRemote)
         {
-            IBlockState iblockstate = this.func_180495_p(p_180496_1_);
+            IBlockState iblockstate = this.getBlockState(p_180496_1_);
 
             try
             {
@@ -556,18 +556,18 @@
 
     public boolean func_175678_i(BlockPos p_175678_1_)
     {
-        return this.func_175726_f(p_175678_1_).func_177444_d(p_175678_1_);
+        return this.getChunk(p_175678_1_).func_177444_d(p_175678_1_);
     }
 
     public boolean func_175710_j(BlockPos p_175710_1_)
     {
-        if (p_175710_1_.func_177956_o() >= this.func_181545_F())
+        if (p_175710_1_.getY() >= this.getSpawnLevel())
         {
             return this.func_175678_i(p_175710_1_);
         }
         else
         {
-            BlockPos blockpos = new BlockPos(p_175710_1_.func_177958_n(), this.func_181545_F(), p_175710_1_.func_177952_p());
+            BlockPos blockpos = new BlockPos(p_175710_1_.getX(), this.getSpawnLevel(), p_175710_1_.getZ());
 
             if (!this.func_175678_i(blockpos))
             {
@@ -575,11 +575,11 @@
             }
             else
             {
-                for (blockpos = blockpos.func_177977_b(); blockpos.func_177956_o() > p_175710_1_.func_177956_o(); blockpos = blockpos.func_177977_b())
+                for (blockpos = blockpos.getBlockPosBelow(); blockpos.getY() > p_175710_1_.getY(); blockpos = blockpos.getBlockPosBelow())
                 {
-                    IBlockState iblockstate = this.func_180495_p(blockpos);
+                    IBlockState iblockstate = this.getBlockState(blockpos);
 
-                    if (iblockstate.func_185891_c() > 0 && !iblockstate.func_185904_a().isLiquid())
+                    if (iblockstate.func_185891_c() > 0 && !iblockstate.getMaterial().isLiquid())
                     {
                         return false;
                     }
@@ -592,18 +592,18 @@
 
     public int func_175699_k(BlockPos p_175699_1_)
     {
-        if (p_175699_1_.func_177956_o() < 0)
+        if (p_175699_1_.getY() < 0)
         {
             return 0;
         }
         else
         {
-            if (p_175699_1_.func_177956_o() >= 256)
+            if (p_175699_1_.getY() >= 256)
             {
-                p_175699_1_ = new BlockPos(p_175699_1_.func_177958_n(), 255, p_175699_1_.func_177952_p());
+                p_175699_1_ = new BlockPos(p_175699_1_.getX(), 255, p_175699_1_.getZ());
             }
 
-            return this.func_175726_f(p_175699_1_).func_177443_a(p_175699_1_, 0);
+            return this.getChunk(p_175699_1_).func_177443_a(p_175699_1_, 0);
         }
     }
 
@@ -614,15 +614,15 @@
 
     public int func_175721_c(BlockPos p_175721_1_, boolean p_175721_2_)
     {
-        if (p_175721_1_.func_177958_n() >= -30000000 && p_175721_1_.func_177952_p() >= -30000000 && p_175721_1_.func_177958_n() < 30000000 && p_175721_1_.func_177952_p() < 30000000)
+        if (p_175721_1_.getX() >= -30000000 && p_175721_1_.getZ() >= -30000000 && p_175721_1_.getX() < 30000000 && p_175721_1_.getZ() < 30000000)
         {
-            if (p_175721_2_ && this.func_180495_p(p_175721_1_).func_185916_f())
+            if (p_175721_2_ && this.getBlockState(p_175721_1_).func_185916_f())
             {
-                int i1 = this.func_175721_c(p_175721_1_.func_177984_a(), false);
-                int i = this.func_175721_c(p_175721_1_.func_177974_f(), false);
-                int j = this.func_175721_c(p_175721_1_.func_177976_e(), false);
-                int k = this.func_175721_c(p_175721_1_.func_177968_d(), false);
-                int l = this.func_175721_c(p_175721_1_.func_177978_c(), false);
+                int i1 = this.func_175721_c(p_175721_1_.getBlockPosAbove(), false);
+                int i = this.func_175721_c(p_175721_1_.getBlockPosEast(), false);
+                int j = this.func_175721_c(p_175721_1_.getBlockPosWest(), false);
+                int k = this.func_175721_c(p_175721_1_.getBlockPosSouth(), false);
+                int l = this.func_175721_c(p_175721_1_.getBlockPosNorth(), false);
 
                 if (i > i1)
                 {
@@ -646,18 +646,18 @@
 
                 return i1;
             }
-            else if (p_175721_1_.func_177956_o() < 0)
+            else if (p_175721_1_.getY() < 0)
             {
                 return 0;
             }
             else
             {
-                if (p_175721_1_.func_177956_o() >= 256)
+                if (p_175721_1_.getY() >= 256)
                 {
-                    p_175721_1_ = new BlockPos(p_175721_1_.func_177958_n(), 255, p_175721_1_.func_177952_p());
+                    p_175721_1_ = new BlockPos(p_175721_1_.getX(), 255, p_175721_1_.getZ());
                 }
 
-                Chunk chunk = this.func_175726_f(p_175721_1_);
+                Chunk chunk = this.getChunk(p_175721_1_);
                 return chunk.func_177443_a(p_175721_1_, this.skylightSubtracted);
             }
         }
@@ -669,7 +669,7 @@
 
     public BlockPos func_175645_m(BlockPos p_175645_1_)
     {
-        return new BlockPos(p_175645_1_.func_177958_n(), this.func_189649_b(p_175645_1_.func_177958_n(), p_175645_1_.func_177952_p()), p_175645_1_.func_177952_p());
+        return new BlockPos(p_175645_1_.getX(), this.func_189649_b(p_175645_1_.getX(), p_175645_1_.getZ()), p_175645_1_.getZ());
     }
 
     public int func_189649_b(int p_189649_1_, int p_189649_2_)
@@ -689,7 +689,7 @@
         }
         else
         {
-            i = this.func_181545_F() + 1;
+            i = this.getSpawnLevel() + 1;
         }
 
         return i;
@@ -717,7 +717,7 @@
         }
         else
         {
-            return this.func_181545_F() + 1;
+            return this.getSpawnLevel() + 1;
         }
     }
 
@@ -729,9 +729,9 @@
         }
         else
         {
-            if (p_175705_2_.func_177956_o() < 0)
+            if (p_175705_2_.getY() < 0)
             {
-                p_175705_2_ = new BlockPos(p_175705_2_.func_177958_n(), 0, p_175705_2_.func_177952_p());
+                p_175705_2_ = new BlockPos(p_175705_2_.getX(), 0, p_175705_2_.getZ());
             }
 
             if (!this.func_175701_a(p_175705_2_))
@@ -742,13 +742,13 @@
             {
                 return p_175705_1_.defaultLightValue;
             }
-            else if (this.func_180495_p(p_175705_2_).func_185916_f())
+            else if (this.getBlockState(p_175705_2_).func_185916_f())
             {
-                int i1 = this.func_175642_b(p_175705_1_, p_175705_2_.func_177984_a());
-                int i = this.func_175642_b(p_175705_1_, p_175705_2_.func_177974_f());
-                int j = this.func_175642_b(p_175705_1_, p_175705_2_.func_177976_e());
-                int k = this.func_175642_b(p_175705_1_, p_175705_2_.func_177968_d());
-                int l = this.func_175642_b(p_175705_1_, p_175705_2_.func_177978_c());
+                int i1 = this.func_175642_b(p_175705_1_, p_175705_2_.getBlockPosAbove());
+                int i = this.func_175642_b(p_175705_1_, p_175705_2_.getBlockPosEast());
+                int j = this.func_175642_b(p_175705_1_, p_175705_2_.getBlockPosWest());
+                int k = this.func_175642_b(p_175705_1_, p_175705_2_.getBlockPosSouth());
+                int l = this.func_175642_b(p_175705_1_, p_175705_2_.getBlockPosNorth());
 
                 if (i > i1)
                 {
@@ -774,7 +774,7 @@
             }
             else
             {
-                Chunk chunk = this.func_175726_f(p_175705_2_);
+                Chunk chunk = this.getChunk(p_175705_2_);
                 return chunk.func_177413_a(p_175705_1_, p_175705_2_);
             }
         }
@@ -782,9 +782,9 @@
 
     public int func_175642_b(EnumSkyBlock p_175642_1_, BlockPos p_175642_2_)
     {
-        if (p_175642_2_.func_177956_o() < 0)
+        if (p_175642_2_.getY() < 0)
         {
-            p_175642_2_ = new BlockPos(p_175642_2_.func_177958_n(), 0, p_175642_2_.func_177952_p());
+            p_175642_2_ = new BlockPos(p_175642_2_.getX(), 0, p_175642_2_.getZ());
         }
 
         if (!this.func_175701_a(p_175642_2_))
@@ -797,7 +797,7 @@
         }
         else
         {
-            Chunk chunk = this.func_175726_f(p_175642_2_);
+            Chunk chunk = this.getChunk(p_175642_2_);
             return chunk.func_177413_a(p_175642_1_, p_175642_2_);
         }
     }
@@ -808,7 +808,7 @@
         {
             if (this.func_175667_e(p_175653_2_))
             {
-                Chunk chunk = this.func_175726_f(p_175653_2_);
+                Chunk chunk = this.getChunk(p_175653_2_);
                 chunk.func_177431_a(p_175653_1_, p_175653_2_, p_175653_3_);
                 this.func_175679_n(p_175653_2_);
             }
@@ -841,16 +841,16 @@
         return this.provider.func_177497_p()[this.func_175671_l(p_175724_1_)];
     }
 
-    public IBlockState func_180495_p(BlockPos p_180495_1_)
+    public IBlockState getBlockState(BlockPos p_180495_1_)
     {
-        if (this.func_189509_E(p_180495_1_))
+        if (this.isBlockYPosInvalid(p_180495_1_))
         {
-            return Blocks.air.func_176223_P();
+            return Blocks.air.getBlockState();
         }
         else
         {
-            Chunk chunk = this.func_175726_f(p_180495_1_);
-            return chunk.func_177435_g(p_180495_1_);
+            Chunk chunk = this.getChunk(p_180495_1_);
+            return chunk.getBlockState(p_180495_1_);
         }
     }
 
@@ -897,8 +897,8 @@
                 int i1 = MathHelper.floor_double(p_147447_1_.yCoord);
                 int j1 = MathHelper.floor_double(p_147447_1_.zCoord);
                 BlockPos blockpos = new BlockPos(l, i1, j1);
-                IBlockState iblockstate = this.func_180495_p(blockpos);
-                Block block = iblockstate.func_177230_c();
+                IBlockState iblockstate = this.getBlockState(blockpos);
+                Block block = iblockstate.getBlock();
 
                 if ((!p_147447_4_ || iblockstate.func_185890_d(this, blockpos) != Block.field_185506_k) && block.func_176209_a(iblockstate, p_147447_3_))
                 {
@@ -1030,10 +1030,10 @@
                     i1 = MathHelper.floor_double(p_147447_1_.yCoord) - (enumfacing == EnumFacing.UP ? 1 : 0);
                     j1 = MathHelper.floor_double(p_147447_1_.zCoord) - (enumfacing == EnumFacing.SOUTH ? 1 : 0);
                     blockpos = new BlockPos(l, i1, j1);
-                    IBlockState iblockstate1 = this.func_180495_p(blockpos);
-                    Block block1 = iblockstate1.func_177230_c();
+                    IBlockState iblockstate1 = this.getBlockState(blockpos);
+                    Block block1 = iblockstate1.getBlock();
 
-                    if (!p_147447_4_ || iblockstate1.func_185904_a() == Material.portal || iblockstate1.func_185890_d(this, blockpos) != Block.field_185506_k)
+                    if (!p_147447_4_ || iblockstate1.getMaterial() == Material.portal || iblockstate1.func_185890_d(this, blockpos) != Block.field_185506_k)
                     {
                         if (block1.func_176209_a(iblockstate1, p_147447_3_))
                         {
@@ -1066,14 +1066,14 @@
 
     public void func_184133_a(@Nullable EntityPlayer p_184133_1_, BlockPos p_184133_2_, SoundEvent p_184133_3_, SoundCategory p_184133_4_, float p_184133_5_, float p_184133_6_)
     {
-        this.func_184148_a(p_184133_1_, (double)p_184133_2_.func_177958_n() + 0.5D, (double)p_184133_2_.func_177956_o() + 0.5D, (double)p_184133_2_.func_177952_p() + 0.5D, p_184133_3_, p_184133_4_, p_184133_5_, p_184133_6_);
+        this.playSound(p_184133_1_, (double)p_184133_2_.getX() + 0.5D, (double)p_184133_2_.getY() + 0.5D, (double)p_184133_2_.getZ() + 0.5D, p_184133_3_, p_184133_4_, p_184133_5_, p_184133_6_);
     }
 
-    public void func_184148_a(@Nullable EntityPlayer p_184148_1_, double p_184148_2_, double p_184148_4_, double p_184148_6_, SoundEvent p_184148_8_, SoundCategory p_184148_9_, float p_184148_10_, float p_184148_11_)
+    public void playSound(@Nullable EntityPlayer p_184148_1_, double p_184148_2_, double p_184148_4_, double p_184148_6_, SoundEvent p_184148_8_, SoundCategory p_184148_9_, float p_184148_10_, float p_184148_11_)
     {
         for (int i = 0; i < this.worldAccesses.size(); ++i)
         {
-            ((IWorldEventListener)this.worldAccesses.get(i)).func_184375_a(p_184148_1_, p_184148_8_, p_184148_9_, p_184148_2_, p_184148_4_, p_184148_6_, p_184148_10_, p_184148_11_);
+            ((IWorldEventListener)this.worldAccesses.get(i)).playSound(p_184148_1_, p_184148_8_, p_184148_9_, p_184148_2_, p_184148_4_, p_184148_6_, p_184148_10_, p_184148_11_);
         }
     }
 
@@ -1089,17 +1089,17 @@
         }
     }
 
-    public void func_175688_a(EnumParticleTypes p_175688_1_, double p_175688_2_, double p_175688_4_, double p_175688_6_, double p_175688_8_, double p_175688_10_, double p_175688_12_, int... p_175688_14_)
+    public void spawnParticle(EnumParticleTypes p_175688_1_, double p_175688_2_, double p_175688_4_, double p_175688_6_, double p_175688_8_, double p_175688_10_, double p_175688_12_, int... p_175688_14_)
     {
-        this.func_175720_a(p_175688_1_.func_179348_c(), p_175688_1_.func_179344_e(), p_175688_2_, p_175688_4_, p_175688_6_, p_175688_8_, p_175688_10_, p_175688_12_, p_175688_14_);
+        this.spawnParticle(p_175688_1_.func_179348_c(), p_175688_1_.func_179344_e(), p_175688_2_, p_175688_4_, p_175688_6_, p_175688_8_, p_175688_10_, p_175688_12_, p_175688_14_);
     }
 
-    public void func_175682_a(EnumParticleTypes p_175682_1_, boolean p_175682_2_, double p_175682_3_, double p_175682_5_, double p_175682_7_, double p_175682_9_, double p_175682_11_, double p_175682_13_, int... p_175682_15_)
+    public void spawnParticle(EnumParticleTypes p_175682_1_, boolean p_175682_2_, double p_175682_3_, double p_175682_5_, double p_175682_7_, double p_175682_9_, double p_175682_11_, double p_175682_13_, int... p_175682_15_)
     {
-        this.func_175720_a(p_175682_1_.func_179348_c(), p_175682_1_.func_179344_e() | p_175682_2_, p_175682_3_, p_175682_5_, p_175682_7_, p_175682_9_, p_175682_11_, p_175682_13_, p_175682_15_);
+        this.spawnParticle(p_175682_1_.func_179348_c(), p_175682_1_.func_179344_e() | p_175682_2_, p_175682_3_, p_175682_5_, p_175682_7_, p_175682_9_, p_175682_11_, p_175682_13_, p_175682_15_);
     }
 
-    private void func_175720_a(int p_175720_1_, boolean p_175720_2_, double p_175720_3_, double p_175720_5_, double p_175720_7_, double p_175720_9_, double p_175720_11_, double p_175720_13_, int... p_175720_15_)
+    private void spawnParticle(int p_175720_1_, boolean p_175720_2_, double p_175720_3_, double p_175720_5_, double p_175720_7_, double p_175720_9_, double p_175720_11_, double p_175720_13_, int... p_175720_15_)
     {
         for (int i = 0; i < this.worldAccesses.size(); ++i)
         {
@@ -1176,9 +1176,9 @@
             p_72900_1_.func_184226_ay();
         }
 
-        if (p_72900_1_.func_184218_aH())
+        if (p_72900_1_.isRiding())
         {
-            p_72900_1_.func_184210_p();
+            p_72900_1_.dismount();
         }
 
         p_72900_1_.setDead();
@@ -1233,7 +1233,7 @@
         this.worldAccesses.remove(p_72848_1_);
     }
 
-    public List<AxisAlignedBB> func_184144_a(@Nullable Entity p_184144_1_, AxisAlignedBB p_184144_2_)
+    public List<AxisAlignedBB> getCollisionBoxes(@Nullable Entity p_184144_1_, AxisAlignedBB p_184144_2_)
     {
         List<AxisAlignedBB> list = Lists.<AxisAlignedBB>newArrayList();
         int i = MathHelper.floor_double(p_184144_2_.minX) - 1;
@@ -1245,7 +1245,7 @@
         WorldBorder worldborder = this.func_175723_af();
         boolean flag = p_184144_1_ != null && p_184144_1_.func_174832_aS();
         boolean flag1 = p_184144_1_ != null && this.func_175673_a(worldborder, p_184144_1_);
-        IBlockState iblockstate = Blocks.stone.func_176223_P();
+        IBlockState iblockstate = Blocks.stone.getBlockState();
         BlockPos.PooledMutableBlockPos blockpos$pooledmutableblockpos = BlockPos.PooledMutableBlockPos.func_185346_s();
 
         for (int k1 = i; k1 < j; ++k1)
@@ -1278,7 +1278,7 @@
 
                             if (worldborder.func_177746_a(blockpos$pooledmutableblockpos) || !flag1)
                             {
-                                iblockstate1 = this.func_180495_p(blockpos$pooledmutableblockpos);
+                                iblockstate1 = this.getBlockState(blockpos$pooledmutableblockpos);
                             }
 
                             iblockstate1.func_185908_a(this, blockpos$pooledmutableblockpos, p_184144_2_, list, p_184144_1_);
@@ -1292,7 +1292,7 @@
 
         if (p_184144_1_ != null)
         {
-            List<Entity> list1 = this.getEntitiesWithinAABBExcludingEntity(p_184144_1_, p_184144_2_.func_186662_g(0.25D));
+            List<Entity> list1 = this.getEntitiesWithinAABBExcludingEntity(p_184144_1_, p_184144_2_.expand(0.25D));
 
             for (int k2 = 0; k2 < list1.size(); ++k2)
             {
@@ -1373,11 +1373,11 @@
 
                             if (k1 >= -30000000 && k1 < 30000000 && l1 >= -30000000 && l1 < 30000000)
                             {
-                                iblockstate = this.func_180495_p(blockpos$pooledmutableblockpos);
+                                iblockstate = this.getBlockState(blockpos$pooledmutableblockpos);
                             }
                             else
                             {
-                                iblockstate = Blocks.bedrock.func_176223_P();
+                                iblockstate = Blocks.bedrock.getBlockState();
                             }
 
                             iblockstate.func_185908_a(this, blockpos$pooledmutableblockpos, p_147461_1_, list, (Entity)null);
@@ -1424,7 +1424,7 @@
                                     return flag1;
                                 }
 
-                                IBlockState iblockstate = this.func_180495_p(blockpos$pooledmutableblockpos);
+                                IBlockState iblockstate = this.getBlockState(blockpos$pooledmutableblockpos);
                                 iblockstate.func_185908_a(this, blockpos$pooledmutableblockpos, p_184143_1_, list, (Entity)null);
 
                                 if (!list.isEmpty())
@@ -1613,19 +1613,19 @@
 
     public BlockPos func_175725_q(BlockPos p_175725_1_)
     {
-        return this.func_175726_f(p_175725_1_).func_177440_h(p_175725_1_);
+        return this.getChunk(p_175725_1_).func_177440_h(p_175725_1_);
     }
 
     public BlockPos func_175672_r(BlockPos p_175672_1_)
     {
-        Chunk chunk = this.func_175726_f(p_175672_1_);
+        Chunk chunk = this.getChunk(p_175672_1_);
         BlockPos blockpos;
         BlockPos blockpos1;
 
-        for (blockpos = new BlockPos(p_175672_1_.func_177958_n(), chunk.getTopFilledSegment() + 16, p_175672_1_.func_177952_p()); blockpos.func_177956_o() >= 0; blockpos = blockpos1)
+        for (blockpos = new BlockPos(p_175672_1_.getX(), chunk.getTopFilledSegment() + 16, p_175672_1_.getZ()); blockpos.getY() >= 0; blockpos = blockpos1)
         {
-            blockpos1 = blockpos.func_177977_b();
-            Material material = chunk.func_177435_g(blockpos1).func_185904_a();
+            blockpos1 = blockpos.getBlockPosBelow();
+            Material material = chunk.getBlockState(blockpos1).getMaterial();
 
             if (material.blocksMovement() && material != Material.leaves)
             {
@@ -1731,7 +1731,7 @@
         for (int i1 = 0; i1 < this.loadedEntityList.size(); ++i1)
         {
             Entity entity2 = (Entity)this.loadedEntityList.get(i1);
-            Entity entity3 = entity2.func_184187_bx();
+            Entity entity3 = entity2.getRidingEntity();
 
             if (entity3 != null)
             {
@@ -1740,7 +1740,7 @@
                     continue;
                 }
 
-                entity2.func_184210_p();
+                entity2.dismount();
             }
 
             this.theProfiler.startSection("tick");
@@ -1817,7 +1817,7 @@
 
                 if (this.func_175667_e(tileentity.func_174877_v()))
                 {
-                    this.func_175726_f(tileentity.func_174877_v()).func_177425_e(tileentity.func_174877_v());
+                    this.getChunk(tileentity.func_174877_v()).func_177425_e(tileentity.func_174877_v());
                 }
             }
         }
@@ -1848,8 +1848,8 @@
 
                     if (this.func_175667_e(tileentity1.func_174877_v()))
                     {
-                        Chunk chunk = this.func_175726_f(tileentity1.func_174877_v());
-                        IBlockState iblockstate = chunk.func_177435_g(tileentity1.func_174877_v());
+                        Chunk chunk = this.getChunk(tileentity1.func_174877_v());
+                        IBlockState iblockstate = chunk.getBlockState(tileentity1.func_174877_v());
                         chunk.func_177426_a(tileentity1.func_174877_v(), tileentity1);
                         this.func_184138_a(tileentity1.func_174877_v(), iblockstate, iblockstate, 3);
                     }
@@ -1924,7 +1924,7 @@
             {
                 ++p_72866_1_.ticksExisted;
 
-                if (p_72866_1_.func_184218_aH())
+                if (p_72866_1_.isRiding())
                 {
                     p_72866_1_.updateRidden();
                 }
@@ -1988,13 +1988,13 @@
             {
                 for (Entity entity : p_72866_1_.func_184188_bt())
                 {
-                    if (!entity.isDead && entity.func_184187_bx() == p_72866_1_)
+                    if (!entity.isDead && entity.getRidingEntity() == p_72866_1_)
                     {
                         this.updateEntity(entity);
                     }
                     else
                     {
-                        entity.func_184210_p();
+                        entity.dismount();
                     }
                 }
             }
@@ -2048,9 +2048,9 @@
             {
                 for (int i2 = i1; i2 < j1; ++i2)
                 {
-                    IBlockState iblockstate = this.func_180495_p(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2));
+                    IBlockState iblockstate = this.getBlockState(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2));
 
-                    if (iblockstate.func_185904_a() != Material.air)
+                    if (iblockstate.getMaterial() != Material.air)
                     {
                         blockpos$pooledmutableblockpos.func_185344_t();
                         return true;
@@ -2082,9 +2082,9 @@
             {
                 for (int i2 = i1; i2 < j1; ++i2)
                 {
-                    IBlockState iblockstate = this.func_180495_p(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2));
+                    IBlockState iblockstate = this.getBlockState(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2));
 
-                    if (iblockstate.func_185904_a().isLiquid())
+                    if (iblockstate.getMaterial().isLiquid())
                     {
                         blockpos$pooledmutableblockpos.func_185344_t();
                         return true;
@@ -2116,7 +2116,7 @@
                 {
                     for (int i2 = i1; i2 < j1; ++i2)
                     {
-                        Block block = this.func_180495_p(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2)).func_177230_c();
+                        Block block = this.getBlockState(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2)).getBlock();
 
                         if (block == Blocks.fire || block == Blocks.flowing_lava || block == Blocks.lava)
                         {
@@ -2162,10 +2162,10 @@
                     for (int i2 = i1; i2 < j1; ++i2)
                     {
                         blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2);
-                        IBlockState iblockstate = this.func_180495_p(blockpos$pooledmutableblockpos);
-                        Block block = iblockstate.func_177230_c();
+                        IBlockState iblockstate = this.getBlockState(blockpos$pooledmutableblockpos);
+                        Block block = iblockstate.getBlock();
 
-                        if (iblockstate.func_185904_a() == p_72918_2_)
+                        if (iblockstate.getMaterial() == p_72918_2_)
                         {
                             double d0 = (double)((float)(l1 + 1) - BlockLiquid.getLiquidHeightPercent(((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue()));
 
@@ -2213,7 +2213,7 @@
             {
                 for (int i2 = i1; i2 < j1; ++i2)
                 {
-                    if (this.func_180495_p(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2)).func_185904_a() == p_72875_2_)
+                    if (this.getBlockState(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2)).getMaterial() == p_72875_2_)
                     {
                         blockpos$pooledmutableblockpos.func_185344_t();
                         return true;
@@ -2245,9 +2245,9 @@
             {
                 for (int i2 = i1; i2 < j1; ++i2)
                 {
-                    IBlockState iblockstate = this.func_180495_p(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2));
+                    IBlockState iblockstate = this.getBlockState(blockpos$pooledmutableblockpos.func_181079_c(k1, l1, i2));
 
-                    if (iblockstate.func_185904_a() == p_72830_2_)
+                    if (iblockstate.getMaterial() == p_72830_2_)
                     {
                         int j2 = ((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue();
                         double d0 = (double)(l1 + 1);
@@ -2336,9 +2336,9 @@
 
     public boolean func_175719_a(@Nullable EntityPlayer p_175719_1_, BlockPos p_175719_2_, EnumFacing p_175719_3_)
     {
-        p_175719_2_ = p_175719_2_.func_177972_a(p_175719_3_);
+        p_175719_2_ = p_175719_2_.getBlockPosRelative(p_175719_3_);
 
-        if (this.func_180495_p(p_175719_2_).func_177230_c() == Blocks.fire)
+        if (this.getBlockState(p_175719_2_).getBlock() == Blocks.fire)
         {
             this.func_180498_a(p_175719_1_, 1009, p_175719_2_, 0);
             this.func_175698_g(p_175719_2_);
@@ -2369,7 +2369,7 @@
     @Nullable
     public TileEntity func_175625_s(BlockPos p_175625_1_)
     {
-        if (this.func_189509_E(p_175625_1_))
+        if (this.isBlockYPosInvalid(p_175625_1_))
         {
             return null;
         }
@@ -2384,7 +2384,7 @@
 
             if (tileentity == null)
             {
-                tileentity = this.func_175726_f(p_175625_1_).func_177424_a(p_175625_1_, Chunk.EnumCreateEntityType.IMMEDIATE);
+                tileentity = this.getChunk(p_175625_1_).func_177424_a(p_175625_1_, Chunk.EnumCreateEntityType.IMMEDIATE);
             }
 
             if (tileentity == null)
@@ -2414,7 +2414,7 @@
 
     public void func_175690_a(BlockPos p_175690_1_, @Nullable TileEntity p_175690_2_)
     {
-        if (!this.func_189509_E(p_175690_1_))
+        if (!this.isBlockYPosInvalid(p_175690_1_))
         {
             if (p_175690_2_ != null && !p_175690_2_.isInvalid())
             {
@@ -2439,7 +2439,7 @@
                 else
                 {
                     this.func_175700_a(p_175690_2_);
-                    this.func_175726_f(p_175690_1_).func_177426_a(p_175690_1_, p_175690_2_);
+                    this.getChunk(p_175690_1_).func_177426_a(p_175690_1_, p_175690_2_);
                 }
             }
         }
@@ -2463,7 +2463,7 @@
                 this.field_175730_i.remove(tileentity);
             }
 
-            this.func_175726_f(p_175713_1_).func_177425_e(p_175713_1_);
+            this.getChunk(p_175713_1_).func_177425_e(p_175713_1_);
         }
     }
 
@@ -2477,24 +2477,24 @@
 
     public boolean func_175665_u(BlockPos p_175665_1_)
     {
-        AxisAlignedBB axisalignedbb = this.func_180495_p(p_175665_1_).func_185890_d(this, p_175665_1_);
+        AxisAlignedBB axisalignedbb = this.getBlockState(p_175665_1_).func_185890_d(this, p_175665_1_);
         return axisalignedbb != Block.field_185506_k && axisalignedbb.getAverageEdgeLength() >= 1.0D;
     }
 
     public boolean func_175677_d(BlockPos p_175677_1_, boolean p_175677_2_)
     {
-        if (this.func_189509_E(p_175677_1_))
+        if (this.isBlockYPosInvalid(p_175677_1_))
         {
             return false;
         }
         else
         {
-            Chunk chunk = this.chunkProvider.func_186026_b(p_175677_1_.func_177958_n() >> 4, p_175677_1_.func_177952_p() >> 4);
+            Chunk chunk = this.chunkProvider.func_186026_b(p_175677_1_.getX() >> 4, p_175677_1_.getZ() >> 4);
 
             if (chunk != null && !chunk.isEmpty())
             {
-                IBlockState iblockstate = this.func_180495_p(p_175677_1_);
-                return iblockstate.func_185904_a().isOpaque() && iblockstate.func_185917_h();
+                IBlockState iblockstate = this.getBlockState(p_175677_1_);
+                return iblockstate.getMaterial().isOpaque() && iblockstate.func_185917_h();
             }
             else
             {
@@ -2656,7 +2656,7 @@
     public void func_189507_a(BlockPos p_189507_1_, IBlockState p_189507_2_, Random p_189507_3_)
     {
         this.scheduledUpdatesAreImmediate = true;
-        p_189507_2_.func_177230_c().func_180650_b(this, p_189507_1_, p_189507_2_, p_189507_3_);
+        p_189507_2_.getBlock().func_180650_b(this, p_189507_1_, p_189507_2_, p_189507_3_);
         this.scheduledUpdatesAreImmediate = false;
     }
 
@@ -2681,10 +2681,10 @@
         }
         else
         {
-            if (p_175670_1_.func_177956_o() >= 0 && p_175670_1_.func_177956_o() < 256 && this.func_175642_b(EnumSkyBlock.BLOCK, p_175670_1_) < 10)
+            if (p_175670_1_.getY() >= 0 && p_175670_1_.getY() < 256 && this.func_175642_b(EnumSkyBlock.BLOCK, p_175670_1_) < 10)
             {
-                IBlockState iblockstate = this.func_180495_p(p_175670_1_);
-                Block block = iblockstate.func_177230_c();
+                IBlockState iblockstate = this.getBlockState(p_175670_1_);
+                Block block = iblockstate.getBlock();
 
                 if ((block == Blocks.water || block == Blocks.flowing_water) && ((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue() == 0)
                 {
@@ -2693,7 +2693,7 @@
                         return true;
                     }
 
-                    boolean flag = this.func_175696_F(p_175670_1_.func_177976_e()) && this.func_175696_F(p_175670_1_.func_177974_f()) && this.func_175696_F(p_175670_1_.func_177978_c()) && this.func_175696_F(p_175670_1_.func_177968_d());
+                    boolean flag = this.func_175696_F(p_175670_1_.getBlockPosWest()) && this.func_175696_F(p_175670_1_.getBlockPosEast()) && this.func_175696_F(p_175670_1_.getBlockPosNorth()) && this.func_175696_F(p_175670_1_.getBlockPosSouth());
 
                     if (!flag)
                     {
@@ -2708,7 +2708,7 @@
 
     private boolean func_175696_F(BlockPos p_175696_1_)
     {
-        return this.func_180495_p(p_175696_1_).func_185904_a() == Material.water;
+        return this.getBlockState(p_175696_1_).getMaterial() == Material.water;
     }
 
     public boolean func_175708_f(BlockPos p_175708_1_, boolean p_175708_2_)
@@ -2726,11 +2726,11 @@
         }
         else
         {
-            if (p_175708_1_.func_177956_o() >= 0 && p_175708_1_.func_177956_o() < 256 && this.func_175642_b(EnumSkyBlock.BLOCK, p_175708_1_) < 10)
+            if (p_175708_1_.getY() >= 0 && p_175708_1_.getY() < 256 && this.func_175642_b(EnumSkyBlock.BLOCK, p_175708_1_) < 10)
             {
-                IBlockState iblockstate = this.func_180495_p(p_175708_1_);
+                IBlockState iblockstate = this.getBlockState(p_175708_1_);
 
-                if (iblockstate.func_185904_a() == Material.air && Blocks.snow_layer.func_176196_c(this, p_175708_1_))
+                if (iblockstate.getMaterial() == Material.air && Blocks.snow_layer.func_176196_c(this, p_175708_1_))
                 {
                     return true;
                 }
@@ -2761,7 +2761,7 @@
         }
         else
         {
-            IBlockState iblockstate = this.func_180495_p(p_175638_1_);
+            IBlockState iblockstate = this.getBlockState(p_175638_1_);
             int i = p_175638_2_ == EnumSkyBlock.SKY ? 0 : iblockstate.func_185906_d();
             int j = iblockstate.func_185891_c();
 
@@ -2822,9 +2822,9 @@
             this.theProfiler.startSection("getBrightness");
             int k = this.func_175642_b(p_180500_1_, p_180500_2_);
             int l = this.func_175638_a(p_180500_2_, p_180500_1_);
-            int i1 = p_180500_2_.func_177958_n();
-            int j1 = p_180500_2_.func_177956_o();
-            int k1 = p_180500_2_.func_177952_p();
+            int i1 = p_180500_2_.getX();
+            int j1 = p_180500_2_.getY();
+            int k1 = p_180500_2_.getZ();
 
             if (l > k)
             {
@@ -2864,7 +2864,7 @@
                                     int j4 = j2 + enumfacing.getFrontOffsetY();
                                     int k4 = k2 + enumfacing.getFrontOffsetZ();
                                     blockpos$pooledmutableblockpos.func_181079_c(i4, j4, k4);
-                                    int l4 = Math.max(1, this.func_180495_p(blockpos$pooledmutableblockpos).func_185891_c());
+                                    int l4 = Math.max(1, this.getBlockState(blockpos$pooledmutableblockpos).func_185891_c());
                                     i3 = this.func_175642_b(p_180500_1_, blockpos$pooledmutableblockpos);
 
                                     if (i3 == l2 - l4 && j < this.lightUpdateBlockList.length)
@@ -2908,32 +2908,32 @@
 
                         if (k6 + l6 + i7 < 17 && flag)
                         {
-                            if (this.func_175642_b(p_180500_1_, blockpos1.func_177976_e()) < j6)
+                            if (this.func_175642_b(p_180500_1_, blockpos1.getBlockPosWest()) < j6)
                             {
                                 this.lightUpdateBlockList[j++] = j5 - 1 - i1 + 32 + (k5 - j1 + 32 << 6) + (l5 - k1 + 32 << 12);
                             }
 
-                            if (this.func_175642_b(p_180500_1_, blockpos1.func_177974_f()) < j6)
+                            if (this.func_175642_b(p_180500_1_, blockpos1.getBlockPosEast()) < j6)
                             {
                                 this.lightUpdateBlockList[j++] = j5 + 1 - i1 + 32 + (k5 - j1 + 32 << 6) + (l5 - k1 + 32 << 12);
                             }
 
-                            if (this.func_175642_b(p_180500_1_, blockpos1.func_177977_b()) < j6)
+                            if (this.func_175642_b(p_180500_1_, blockpos1.getBlockPosBelow()) < j6)
                             {
                                 this.lightUpdateBlockList[j++] = j5 - i1 + 32 + (k5 - 1 - j1 + 32 << 6) + (l5 - k1 + 32 << 12);
                             }
 
-                            if (this.func_175642_b(p_180500_1_, blockpos1.func_177984_a()) < j6)
+                            if (this.func_175642_b(p_180500_1_, blockpos1.getBlockPosAbove()) < j6)
                             {
                                 this.lightUpdateBlockList[j++] = j5 - i1 + 32 + (k5 + 1 - j1 + 32 << 6) + (l5 - k1 + 32 << 12);
                             }
 
-                            if (this.func_175642_b(p_180500_1_, blockpos1.func_177978_c()) < j6)
+                            if (this.func_175642_b(p_180500_1_, blockpos1.getBlockPosNorth()) < j6)
                             {
                                 this.lightUpdateBlockList[j++] = j5 - i1 + 32 + (k5 - j1 + 32 << 6) + (l5 - 1 - k1 + 32 << 12);
                             }
 
-                            if (this.func_175642_b(p_180500_1_, blockpos1.func_177968_d()) < j6)
+                            if (this.func_175642_b(p_180500_1_, blockpos1.getBlockPosSouth()) < j6)
                             {
                                 this.lightUpdateBlockList[j++] = j5 - i1 + 32 + (k5 - j1 + 32 << 6) + (l5 + 1 - k1 + 32 << 12);
                             }
@@ -3096,7 +3096,7 @@
     {
         if (this.func_175667_e(p_175646_1_))
         {
-            this.func_175726_f(p_175646_1_).setChunkModified();
+            this.getChunk(p_175646_1_).setChunkModified();
         }
     }
 
@@ -3135,24 +3135,24 @@
 
     public boolean func_175716_a(Block p_175716_1_, BlockPos p_175716_2_, boolean p_175716_3_, EnumFacing p_175716_4_, @Nullable Entity p_175716_5_, @Nullable ItemStack p_175716_6_)
     {
-        IBlockState iblockstate = this.func_180495_p(p_175716_2_);
-        AxisAlignedBB axisalignedbb = p_175716_3_ ? null : p_175716_1_.func_176223_P().func_185890_d(this, p_175716_2_);
-        return axisalignedbb != Block.field_185506_k && !this.checkNoEntityCollision(axisalignedbb.func_186670_a(p_175716_2_), p_175716_5_) ? false : (iblockstate.func_185904_a() == Material.circuits && p_175716_1_ == Blocks.anvil ? true : iblockstate.func_185904_a().isReplaceable() && p_175716_1_.func_176193_a(this, p_175716_2_, p_175716_4_, p_175716_6_));
+        IBlockState iblockstate = this.getBlockState(p_175716_2_);
+        AxisAlignedBB axisalignedbb = p_175716_3_ ? null : p_175716_1_.getBlockState().func_185890_d(this, p_175716_2_);
+        return axisalignedbb != Block.field_185506_k && !this.checkNoEntityCollision(axisalignedbb.offset(p_175716_2_), p_175716_5_) ? false : (iblockstate.getMaterial() == Material.circuits && p_175716_1_ == Blocks.anvil ? true : iblockstate.getMaterial().isReplaceable() && p_175716_1_.func_176193_a(this, p_175716_2_, p_175716_4_, p_175716_6_));
     }
 
-    public int func_181545_F()
+    public int getSpawnLevel()
     {
-        return this.field_181546_a;
+        return this.spawnLevel;
     }
 
-    public void func_181544_b(int p_181544_1_)
+    public void setSpawnLevel(int p_181544_1_)
     {
-        this.field_181546_a = p_181544_1_;
+        this.spawnLevel = p_181544_1_;
     }
 
     public int func_175627_a(BlockPos p_175627_1_, EnumFacing p_175627_2_)
     {
-        return this.func_180495_p(p_175627_1_).func_185893_b(this, p_175627_1_, p_175627_2_);
+        return this.getBlockState(p_175627_1_).func_185893_b(this, p_175627_1_, p_175627_2_);
     }
 
     public WorldType func_175624_G()
@@ -3163,7 +3163,7 @@
     public int func_175676_y(BlockPos p_175676_1_)
     {
         int i = 0;
-        i = Math.max(i, this.func_175627_a(p_175676_1_.func_177977_b(), EnumFacing.DOWN));
+        i = Math.max(i, this.func_175627_a(p_175676_1_.getBlockPosBelow(), EnumFacing.DOWN));
 
         if (i >= 15)
         {
@@ -3171,7 +3171,7 @@
         }
         else
         {
-            i = Math.max(i, this.func_175627_a(p_175676_1_.func_177984_a(), EnumFacing.UP));
+            i = Math.max(i, this.func_175627_a(p_175676_1_.getBlockPosAbove(), EnumFacing.UP));
 
             if (i >= 15)
             {
@@ -3179,7 +3179,7 @@
             }
             else
             {
-                i = Math.max(i, this.func_175627_a(p_175676_1_.func_177978_c(), EnumFacing.NORTH));
+                i = Math.max(i, this.func_175627_a(p_175676_1_.getBlockPosNorth(), EnumFacing.NORTH));
 
                 if (i >= 15)
                 {
@@ -3187,7 +3187,7 @@
                 }
                 else
                 {
-                    i = Math.max(i, this.func_175627_a(p_175676_1_.func_177968_d(), EnumFacing.SOUTH));
+                    i = Math.max(i, this.func_175627_a(p_175676_1_.getBlockPosSouth(), EnumFacing.SOUTH));
 
                     if (i >= 15)
                     {
@@ -3195,7 +3195,7 @@
                     }
                     else
                     {
-                        i = Math.max(i, this.func_175627_a(p_175676_1_.func_177976_e(), EnumFacing.WEST));
+                        i = Math.max(i, this.func_175627_a(p_175676_1_.getBlockPosWest(), EnumFacing.WEST));
 
                         if (i >= 15)
                         {
@@ -3203,7 +3203,7 @@
                         }
                         else
                         {
-                            i = Math.max(i, this.func_175627_a(p_175676_1_.func_177974_f(), EnumFacing.EAST));
+                            i = Math.max(i, this.func_175627_a(p_175676_1_.getBlockPosEast(), EnumFacing.EAST));
                             return i >= 15 ? i : i;
                         }
                     }
@@ -3219,13 +3219,13 @@
 
     public int func_175651_c(BlockPos p_175651_1_, EnumFacing p_175651_2_)
     {
-        IBlockState iblockstate = this.func_180495_p(p_175651_1_);
+        IBlockState iblockstate = this.getBlockState(p_175651_1_);
         return iblockstate.func_185915_l() ? this.func_175676_y(p_175651_1_) : iblockstate.func_185911_a(this, p_175651_1_, p_175651_2_);
     }
 
     public boolean func_175640_z(BlockPos p_175640_1_)
     {
-        return this.func_175651_c(p_175640_1_.func_177977_b(), EnumFacing.DOWN) > 0 ? true : (this.func_175651_c(p_175640_1_.func_177984_a(), EnumFacing.UP) > 0 ? true : (this.func_175651_c(p_175640_1_.func_177978_c(), EnumFacing.NORTH) > 0 ? true : (this.func_175651_c(p_175640_1_.func_177968_d(), EnumFacing.SOUTH) > 0 ? true : (this.func_175651_c(p_175640_1_.func_177976_e(), EnumFacing.WEST) > 0 ? true : this.func_175651_c(p_175640_1_.func_177974_f(), EnumFacing.EAST) > 0))));
+        return this.func_175651_c(p_175640_1_.getBlockPosBelow(), EnumFacing.DOWN) > 0 ? true : (this.func_175651_c(p_175640_1_.getBlockPosAbove(), EnumFacing.UP) > 0 ? true : (this.func_175651_c(p_175640_1_.getBlockPosNorth(), EnumFacing.NORTH) > 0 ? true : (this.func_175651_c(p_175640_1_.getBlockPosSouth(), EnumFacing.SOUTH) > 0 ? true : (this.func_175651_c(p_175640_1_.getBlockPosWest(), EnumFacing.WEST) > 0 ? true : this.func_175651_c(p_175640_1_.getBlockPosEast(), EnumFacing.EAST) > 0))));
     }
 
     public int func_175687_A(BlockPos p_175687_1_)
@@ -3234,7 +3234,7 @@
 
         for (EnumFacing enumfacing : EnumFacing.values())
         {
-            int j = this.func_175651_c(p_175687_1_.func_177972_a(enumfacing), enumfacing);
+            int j = this.func_175651_c(p_175687_1_.getBlockPosRelative(enumfacing), enumfacing);
 
             if (j >= 15)
             {
@@ -3321,7 +3321,7 @@
     @Nullable
     public EntityPlayer func_184139_a(BlockPos p_184139_1_, double p_184139_2_, double p_184139_4_)
     {
-        return this.func_184150_a((double)((float)p_184139_1_.func_177958_n() + 0.5F), (double)((float)p_184139_1_.func_177956_o() + 0.5F), (double)((float)p_184139_1_.func_177952_p() + 0.5F), p_184139_2_, p_184139_4_, (Function<EntityPlayer, Double>)null, (Predicate<EntityPlayer>)null);
+        return this.func_184150_a((double)((float)p_184139_1_.getX() + 0.5F), (double)((float)p_184139_1_.getY() + 0.5F), (double)((float)p_184139_1_.getZ() + 0.5F), p_184139_2_, p_184139_4_, (Function<EntityPlayer, Double>)null, (Predicate<EntityPlayer>)null);
     }
 
     @Nullable
@@ -3516,7 +3516,7 @@
 
     public void func_175641_c(BlockPos p_175641_1_, Block p_175641_2_, int p_175641_3_, int p_175641_4_)
     {
-        this.func_180495_p(p_175641_1_).func_189547_a(this, p_175641_1_, p_175641_3_, p_175641_4_);
+        this.getBlockState(p_175641_1_).func_189547_a(this, p_175641_1_, p_175641_3_, p_175641_4_);
     }
 
     /**
@@ -3607,7 +3607,7 @@
         {
             return false;
         }
-        else if (this.func_175725_q(p_175727_1_).func_177956_o() > p_175727_1_.func_177956_o())
+        else if (this.func_175725_q(p_175727_1_).getY() > p_175727_1_.getY())
         {
             return false;
         }
@@ -3804,11 +3804,11 @@
     {
         for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
         {
-            BlockPos blockpos = p_175666_1_.func_177972_a(enumfacing);
+            BlockPos blockpos = p_175666_1_.getBlockPosRelative(enumfacing);
 
             if (this.func_175667_e(blockpos))
             {
-                IBlockState iblockstate = this.func_180495_p(blockpos);
+                IBlockState iblockstate = this.getBlockState(blockpos);
 
                 if (Blocks.unpowered_comparator.func_185547_C(iblockstate))
                 {
@@ -3816,8 +3816,8 @@
                 }
                 else if (iblockstate.func_185915_l())
                 {
-                    blockpos = blockpos.func_177972_a(enumfacing);
-                    iblockstate = this.func_180495_p(blockpos);
+                    blockpos = blockpos.getBlockPosRelative(enumfacing);
+                    iblockstate = this.getBlockState(blockpos);
 
                     if (Blocks.unpowered_comparator.func_185547_C(iblockstate))
                     {
@@ -3836,7 +3836,7 @@
         if (this.func_175667_e(p_175649_1_))
         {
             f = this.getCurrentMoonPhaseFactor();
-            i = this.func_175726_f(p_175649_1_).func_177416_w();
+            i = this.getChunk(p_175649_1_).func_177416_w();
         }
 
         return new DifficultyInstance(this.func_175659_aa(), this.getWorldTime(), i, f);
@@ -3883,8 +3883,8 @@
     public boolean chunkExists(int p_72916_1_, int p_72916_2_)
     {
         BlockPos blockpos = this.func_175694_M();
-        int i = p_72916_1_ * 16 + 8 - blockpos.func_177958_n();
-        int j = p_72916_2_ * 16 + 8 - blockpos.func_177952_p();
+        int i = p_72916_1_ * 16 + 8 - blockpos.getX();
+        int j = p_72916_2_ * 16 + 8 - blockpos.getZ();
         int k = 128;
         return i >= -128 && i <= 128 && j >= -128 && j <= 128;
     }
