--- a/net/minecraft/world/WorldProviderEnd.java
+++ b/net/minecraft/world/WorldProviderEnd.java
@@ -98,7 +98,7 @@
      */
     public boolean canCoordinateBeSpawn(int p_76566_1_, int p_76566_2_)
     {
-        return this.worldObj.func_184141_c(new BlockPos(p_76566_1_, 0, p_76566_2_)).func_185904_a().blocksMovement();
+        return this.worldObj.getBlockSomething(new BlockPos(p_76566_1_, 0, p_76566_2_)).getMaterial().blocksMovement();
     }
 
     public BlockPos func_177496_h()
