--- a/net/minecraft/world/WorldServer.java
+++ b/net/minecraft/world/WorldServer.java
@@ -319,14 +319,14 @@
     {
         if (this.worldInfo.getSpawnY() <= 0)
         {
-            this.worldInfo.setSpawnY(this.func_181545_F() + 1);
+            this.worldInfo.setSpawnY(this.getSpawnLevel() + 1);
         }
 
         int i = this.worldInfo.getSpawnX();
         int j = this.worldInfo.getSpawnZ();
         int k = 0;
 
-        while (this.func_184141_c(new BlockPos(i, 0, j)).func_185904_a() == Material.air)
+        while (this.getBlockSomething(new BlockPos(i, 0, j)).getMaterial() == Material.air)
         {
             i += this.rand.nextInt(8) - this.rand.nextInt(8);
             j += this.rand.nextInt(8) - this.rand.nextInt(8);
@@ -412,13 +412,13 @@
                             entityhorse.func_184778_a(HorseType.SKELETON);
                             entityhorse.func_184784_x(true);
                             entityhorse.setGrowingAge(0);
-                            entityhorse.setPosition((double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p());
+                            entityhorse.setPosition((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ());
                             this.spawnEntityInWorld(entityhorse);
-                            this.addWeatherEffect(new EntityLightningBolt(this, (double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p(), true));
+                            this.addWeatherEffect(new EntityLightningBolt(this, (double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), true));
                         }
                         else
                         {
-                            this.addWeatherEffect(new EntityLightningBolt(this, (double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p(), false));
+                            this.addWeatherEffect(new EntityLightningBolt(this, (double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), false));
                         }
                     }
                 }
@@ -430,21 +430,21 @@
                     this.updateLCG = this.updateLCG * 3 + 1013904223;
                     int j2 = this.updateLCG >> 2;
                     BlockPos blockpos1 = this.func_175725_q(new BlockPos(j + (j2 & 15), 0, k + (j2 >> 8 & 15)));
-                    BlockPos blockpos2 = blockpos1.func_177977_b();
+                    BlockPos blockpos2 = blockpos1.getBlockPosBelow();
 
                     if (this.func_175662_w(blockpos2))
                     {
-                        this.func_175656_a(blockpos2, Blocks.ice.func_176223_P());
+                        this.func_175656_a(blockpos2, Blocks.ice.getBlockState());
                     }
 
                     if (flag && this.func_175708_f(blockpos1, true))
                     {
-                        this.func_175656_a(blockpos1, Blocks.snow_layer.func_176223_P());
+                        this.func_175656_a(blockpos1, Blocks.snow_layer.getBlockState());
                     }
 
                     if (flag && this.func_180494_b(blockpos2).canSpawnLightningBolt())
                     {
-                        this.func_180495_p(blockpos2).func_177230_c().func_176224_k(this, blockpos2);
+                        this.getBlockState(blockpos2).getBlock().func_176224_k(this, blockpos2);
                     }
                 }
 
@@ -464,7 +464,7 @@
                                 int l1 = j1 >> 8 & 15;
                                 int i2 = j1 >> 16 & 15;
                                 IBlockState iblockstate = extendedblockstorage.func_177485_a(k1, i2, l1);
-                                Block block = iblockstate.func_177230_c();
+                                Block block = iblockstate.getBlock();
                                 this.theProfiler.startSection("randomTick");
 
                                 if (block.getTickRandomly())
@@ -486,7 +486,7 @@
     protected BlockPos func_175736_a(BlockPos p_175736_1_)
     {
         BlockPos blockpos = this.func_175725_q(p_175736_1_);
-        AxisAlignedBB axisalignedbb = (new AxisAlignedBB(blockpos, new BlockPos(blockpos.func_177958_n(), this.getHeight(), blockpos.func_177952_p()))).func_186662_g(3.0D);
+        AxisAlignedBB axisalignedbb = (new AxisAlignedBB(blockpos, new BlockPos(blockpos.getX(), this.getHeight(), blockpos.getZ()))).expand(3.0D);
         List<EntityLivingBase> list = this.func_175647_a(EntityLivingBase.class, axisalignedbb, new Predicate<EntityLivingBase>()
         {
             public boolean apply(@Nullable EntityLivingBase p_apply_1_)
@@ -501,9 +501,9 @@
         }
         else
         {
-            if (blockpos.func_177956_o() == -1)
+            if (blockpos.getY() == -1)
             {
-                blockpos = blockpos.func_177981_b(2);
+                blockpos = blockpos.getBlockPosAbove(2);
             }
 
             return blockpos;
@@ -535,7 +535,7 @@
             LogManager.getLogger().warn((String)"Tried to assign a mutable BlockPos to tick data...", (Throwable)(new Error(p_175654_1_.getClass().toString())));
         }
 
-        Material material = p_175654_2_.func_176223_P().func_185904_a();
+        Material material = p_175654_2_.getBlockState().getMaterial();
 
         if (this.scheduledUpdatesAreImmediate && material != Material.air)
         {
@@ -543,11 +543,11 @@
             {
                 if (this.func_175707_a(p_175654_1_.func_177982_a(-8, -8, -8), p_175654_1_.func_177982_a(8, 8, 8)))
                 {
-                    IBlockState iblockstate = this.func_180495_p(p_175654_1_);
+                    IBlockState iblockstate = this.getBlockState(p_175654_1_);
 
-                    if (iblockstate.func_185904_a() != Material.air && iblockstate.func_177230_c() == p_175654_2_)
+                    if (iblockstate.getMaterial() != Material.air && iblockstate.getBlock() == p_175654_2_)
                     {
-                        iblockstate.func_177230_c().func_180650_b(this, p_175654_1_, iblockstate, this.rand);
+                        iblockstate.getBlock().func_180650_b(this, p_175654_1_, iblockstate, this.rand);
                     }
                 }
 
@@ -585,7 +585,7 @@
 
         NextTickListEntry nextticklistentry = new NextTickListEntry(p_180497_1_, p_180497_2_);
         nextticklistentry.setPriority(p_180497_4_);
-        Material material = p_180497_2_.func_176223_P().func_185904_a();
+        Material material = p_180497_2_.getBlockState().getMaterial();
 
         if (material != Material.air)
         {
@@ -628,7 +628,7 @@
         for (int i = 0; i < this.playerEntities.size(); ++i)
         {
             Entity entity = (Entity)this.playerEntities.get(i);
-            Entity entity1 = entity.func_184187_bx();
+            Entity entity1 = entity.getRidingEntity();
 
             if (entity1 != null)
             {
@@ -637,7 +637,7 @@
                     continue;
                 }
 
-                entity.func_184210_p();
+                entity.dismount();
             }
 
             this.theProfiler.startSection("tick");
@@ -738,13 +738,13 @@
 
                     if (this.func_175707_a(nextticklistentry1.field_180282_a.func_177982_a(0, 0, 0), nextticklistentry1.field_180282_a.func_177982_a(0, 0, 0)))
                     {
-                        IBlockState iblockstate = this.func_180495_p(nextticklistentry1.field_180282_a);
+                        IBlockState iblockstate = this.getBlockState(nextticklistentry1.field_180282_a);
 
-                        if (iblockstate.func_185904_a() != Material.air && Block.isEqualTo(iblockstate.func_177230_c(), nextticklistentry1.func_151351_a()))
+                        if (iblockstate.getMaterial() != Material.air && Block.isEqualTo(iblockstate.getBlock(), nextticklistentry1.func_151351_a()))
                         {
                             try
                             {
-                                iblockstate.func_177230_c().func_180650_b(this, nextticklistentry1.field_180282_a, iblockstate, this.rand);
+                                iblockstate.getBlock().func_180650_b(this, nextticklistentry1.field_180282_a, iblockstate, this.rand);
                             }
                             catch (Throwable throwable)
                             {
@@ -802,7 +802,7 @@
                 NextTickListEntry nextticklistentry = (NextTickListEntry)iterator.next();
                 BlockPos blockpos = nextticklistentry.field_180282_a;
 
-                if (blockpos.func_177958_n() >= p_175712_1_.minX && blockpos.func_177958_n() < p_175712_1_.maxX && blockpos.func_177952_p() >= p_175712_1_.minZ && blockpos.func_177952_p() < p_175712_1_.maxZ)
+                if (blockpos.getX() >= p_175712_1_.minX && blockpos.getX() < p_175712_1_.maxX && blockpos.getZ() >= p_175712_1_.minZ && blockpos.getZ() < p_175712_1_.maxZ)
                 {
                     if (p_175712_2_)
                     {
@@ -927,11 +927,11 @@
     {
         if (!this.provider.canRespawnHere())
         {
-            this.worldInfo.func_176143_a(BlockPos.field_177992_a.func_177981_b(this.provider.getAverageGroundLevel()));
+            this.worldInfo.func_176143_a(BlockPos.createDefault.getBlockPosAbove(this.provider.getAverageGroundLevel()));
         }
         else if (this.worldInfo.getTerrainType() == WorldType.field_180272_g)
         {
-            this.worldInfo.func_176143_a(BlockPos.field_177992_a.func_177984_a());
+            this.worldInfo.func_176143_a(BlockPos.createDefault.getBlockPosAbove());
         }
         else
         {
@@ -946,8 +946,8 @@
 
             if (blockpos != null)
             {
-                i = blockpos.func_177958_n();
-                k = blockpos.func_177952_p();
+                i = blockpos.getX();
+                k = blockpos.getZ();
             }
             else
             {
@@ -989,7 +989,7 @@
         {
             int j = this.worldInfo.getSpawnX() + this.rand.nextInt(6) - this.rand.nextInt(6);
             int k = this.worldInfo.getSpawnZ() + this.rand.nextInt(6) - this.rand.nextInt(6);
-            BlockPos blockpos = this.func_175672_r(new BlockPos(j, 0, k)).func_177984_a();
+            BlockPos blockpos = this.func_175672_r(new BlockPos(j, 0, k)).getBlockPosAbove();
 
             if (worldgeneratorbonuschest.func_180709_b(this, this.rand, blockpos))
             {
@@ -1249,7 +1249,7 @@
             {
                 if (this.func_147485_a(blockeventdata))
                 {
-                    this.mcServer.func_184103_al().sendToAllNearExcept((EntityPlayer)null, (double)blockeventdata.func_180328_a().func_177958_n(), (double)blockeventdata.func_180328_a().func_177956_o(), (double)blockeventdata.func_180328_a().func_177952_p(), 64.0D, this.provider.func_186058_p().func_186068_a(), new SPacketBlockAction(blockeventdata.func_180328_a(), blockeventdata.getBlock(), blockeventdata.getEventID(), blockeventdata.getEventParameter()));
+                    this.mcServer.func_184103_al().sendToAllNearExcept((EntityPlayer)null, (double)blockeventdata.func_180328_a().getX(), (double)blockeventdata.func_180328_a().getY(), (double)blockeventdata.func_180328_a().getZ(), 64.0D, this.provider.func_186058_p().func_186068_a(), new SPacketBlockAction(blockeventdata.func_180328_a(), blockeventdata.getBlock(), blockeventdata.getEventID(), blockeventdata.getEventParameter()));
                 }
             }
 
@@ -1259,8 +1259,8 @@
 
     private boolean func_147485_a(BlockEventData p_147485_1_)
     {
-        IBlockState iblockstate = this.func_180495_p(p_147485_1_.func_180328_a());
-        return iblockstate.func_177230_c() == p_147485_1_.getBlock() ? iblockstate.func_189547_a(this, p_147485_1_.func_180328_a(), p_147485_1_.getEventID(), p_147485_1_.getEventParameter()) : false;
+        IBlockState iblockstate = this.getBlockState(p_147485_1_.func_180328_a());
+        return iblockstate.getBlock() == p_147485_1_.getBlock() ? iblockstate.func_189547_a(this, p_147485_1_.func_180328_a(), p_147485_1_.getEventID(), p_147485_1_.getEventParameter()) : false;
     }
 
     /**
