--- a/net/minecraft/entity/boss/dragon/phase/PhaseLanding.java
+++ b/net/minecraft/entity/boss/dragon/phase/PhaseLanding.java
@@ -19,7 +19,7 @@
     public void func_188657_b()
     {
         Vec3d vec3d = this.field_188661_a.func_184665_a(1.0F).normalize();
-        vec3d.func_178785_b(-((float)Math.PI / 4F));
+        vec3d.rotateY(-((float)Math.PI / 4F));
         double d0 = this.field_188661_a.dragonPartHead.posX;
         double d1 = this.field_188661_a.dragonPartHead.posY + (double)(this.field_188661_a.dragonPartHead.height / 2.0F);
         double d2 = this.field_188661_a.dragonPartHead.posZ;
@@ -29,8 +29,8 @@
             double d3 = d0 + this.field_188661_a.getRNG().nextGaussian() / 2.0D;
             double d4 = d1 + this.field_188661_a.getRNG().nextGaussian() / 2.0D;
             double d5 = d2 + this.field_188661_a.getRNG().nextGaussian() / 2.0D;
-            this.field_188661_a.worldObj.func_175688_a(EnumParticleTypes.DRAGON_BREATH, d3, d4, d5, -vec3d.xCoord * 0.07999999821186066D + this.field_188661_a.motionX, -vec3d.yCoord * 0.30000001192092896D + this.field_188661_a.motionY, -vec3d.zCoord * 0.07999999821186066D + this.field_188661_a.motionZ, new int[0]);
-            vec3d.func_178785_b(0.19634955F);
+            this.field_188661_a.worldObj.spawnParticle(EnumParticleTypes.DRAGON_BREATH, d3, d4, d5, -vec3d.xCoord * 0.07999999821186066D + this.field_188661_a.motionX, -vec3d.yCoord * 0.30000001192092896D + this.field_188661_a.motionY, -vec3d.zCoord * 0.07999999821186066D + this.field_188661_a.motionZ, new int[0]);
+            vec3d.rotateY(0.19634955F);
         }
     }
 
@@ -41,7 +41,7 @@
             this.field_188685_b = new Vec3d(this.field_188661_a.worldObj.func_175672_r(WorldGenEndPodium.field_186139_a));
         }
 
-        if (this.field_188685_b.func_186679_c(this.field_188661_a.posX, this.field_188661_a.posY, this.field_188661_a.posZ) < 1.0D)
+        if (this.field_188685_b.squareDistanceTo(this.field_188661_a.posX, this.field_188661_a.posY, this.field_188661_a.posZ) < 1.0D)
         {
             ((PhaseSittingFlaming)this.field_188661_a.func_184670_cT().func_188757_b(PhaseList.field_188746_f)).func_188663_j();
             this.field_188661_a.func_184670_cT().func_188758_a(PhaseList.field_188747_g);
