--- a/net/minecraft/entity/boss/dragon/phase/PhaseSittingFlaming.java
+++ b/net/minecraft/entity/boss/dragon/phase/PhaseSittingFlaming.java
@@ -27,7 +27,7 @@
         if (this.field_188664_b % 2 == 0 && this.field_188664_b < 10)
         {
             Vec3d vec3d = this.field_188661_a.func_184665_a(1.0F).normalize();
-            vec3d.func_178785_b(-((float)Math.PI / 4F));
+            vec3d.rotateY(-((float)Math.PI / 4F));
             double d0 = this.field_188661_a.dragonPartHead.posX;
             double d1 = this.field_188661_a.dragonPartHead.posY + (double)(this.field_188661_a.dragonPartHead.height / 2.0F);
             double d2 = this.field_188661_a.dragonPartHead.posZ;
@@ -40,10 +40,10 @@
 
                 for (int j = 0; j < 6; ++j)
                 {
-                    this.field_188661_a.worldObj.func_175688_a(EnumParticleTypes.DRAGON_BREATH, d3, d4, d5, -vec3d.xCoord * 0.07999999821186066D * (double)j, -vec3d.yCoord * 0.6000000238418579D, -vec3d.zCoord * 0.07999999821186066D * (double)j, new int[0]);
+                    this.field_188661_a.worldObj.spawnParticle(EnumParticleTypes.DRAGON_BREATH, d3, d4, d5, -vec3d.xCoord * 0.07999999821186066D * (double)j, -vec3d.yCoord * 0.6000000238418579D, -vec3d.zCoord * 0.07999999821186066D * (double)j, new int[0]);
                 }
 
-                vec3d.func_178785_b(0.19634955F);
+                vec3d.rotateY(0.19634955F);
             }
         }
     }
@@ -72,7 +72,7 @@
             double d2 = this.field_188661_a.dragonPartHead.posY + (double)(this.field_188661_a.dragonPartHead.height / 2.0F);
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(MathHelper.floor_double(d0), MathHelper.floor_double(d2), MathHelper.floor_double(d1));
 
-            while (this.field_188661_a.worldObj.func_175623_d(blockpos$mutableblockpos))
+            while (this.field_188661_a.worldObj.isEmpty(blockpos$mutableblockpos))
             {
                 --d2;
                 blockpos$mutableblockpos.func_181079_c(MathHelper.floor_double(d0), MathHelper.floor_double(d2), MathHelper.floor_double(d1));
