--- a/net/minecraft/entity/boss/dragon/phase/PhaseDying.java
+++ b/net/minecraft/entity/boss/dragon/phase/PhaseDying.java
@@ -24,7 +24,7 @@
             float f = (this.field_188661_a.getRNG().nextFloat() - 0.5F) * 8.0F;
             float f1 = (this.field_188661_a.getRNG().nextFloat() - 0.5F) * 4.0F;
             float f2 = (this.field_188661_a.getRNG().nextFloat() - 0.5F) * 8.0F;
-            this.field_188661_a.worldObj.func_175688_a(EnumParticleTypes.EXPLOSION_HUGE, this.field_188661_a.posX + (double)f, this.field_188661_a.posY + 2.0D + (double)f1, this.field_188661_a.posZ + (double)f2, 0.0D, 0.0D, 0.0D, new int[0]);
+            this.field_188661_a.worldObj.spawnParticle(EnumParticleTypes.EXPLOSION_HUGE, this.field_188661_a.posX + (double)f, this.field_188661_a.posY + 2.0D + (double)f1, this.field_188661_a.posZ + (double)f2, 0.0D, 0.0D, 0.0D, new int[0]);
         }
     }
 
@@ -35,10 +35,10 @@
         if (this.field_188672_b == null)
         {
             BlockPos blockpos = this.field_188661_a.worldObj.func_175645_m(WorldGenEndPodium.field_186139_a);
-            this.field_188672_b = new Vec3d((double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p());
+            this.field_188672_b = new Vec3d((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ());
         }
 
-        double d0 = this.field_188672_b.func_186679_c(this.field_188661_a.posX, this.field_188661_a.posY, this.field_188661_a.posZ);
+        double d0 = this.field_188672_b.squareDistanceTo(this.field_188661_a.posX, this.field_188661_a.posY, this.field_188661_a.posZ);
 
         if (d0 >= 100.0D && d0 <= 22500.0D && !this.field_188661_a.isCollidedHorizontally && !this.field_188661_a.isCollidedVertically)
         {
