--- a/net/minecraft/entity/boss/dragon/phase/PhaseHoldingPattern.java
+++ b/net/minecraft/entity/boss/dragon/phase/PhaseHoldingPattern.java
@@ -30,7 +30,7 @@
 
     public void func_188659_c()
     {
-        double d0 = this.field_188678_c == null ? 0.0D : this.field_188678_c.func_186679_c(this.field_188661_a.posX, this.field_188661_a.posY, this.field_188661_a.posZ);
+        double d0 = this.field_188678_c == null ? 0.0D : this.field_188678_c.squareDistanceTo(this.field_188661_a.posX, this.field_188661_a.posY, this.field_188661_a.posZ);
 
         if (d0 < 100.0D || d0 > 22500.0D || this.field_188661_a.isCollidedHorizontally || this.field_188661_a.isCollidedVertically)
         {
