--- a/net/minecraft/entity/boss/EntityDragon.java
+++ b/net/minecraft/entity/boss/EntityDragon.java
@@ -193,7 +193,7 @@
             float f13 = (this.rand.nextFloat() - 0.5F) * 8.0F;
             float f15 = (this.rand.nextFloat() - 0.5F) * 4.0F;
             float f17 = (this.rand.nextFloat() - 0.5F) * 8.0F;
-            this.worldObj.func_175688_a(EnumParticleTypes.EXPLOSION_LARGE, this.posX + (double)f13, this.posY + 2.0D + (double)f15, this.posZ + (double)f17, 0.0D, 0.0D, 0.0D, new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.EXPLOSION_LARGE, this.posX + (double)f13, this.posY + 2.0D + (double)f15, this.posZ + (double)f17, 0.0D, 0.0D, 0.0D, new int[0]);
         }
         else
         {
@@ -342,8 +342,8 @@
                 {
                     this.collideWithEntities(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.dragonPartWing1.func_174813_aQ().expand(4.0D, 2.0D, 4.0D).offset(0.0D, -2.0D, 0.0D)));
                     this.collideWithEntities(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.dragonPartWing2.func_174813_aQ().expand(4.0D, 2.0D, 4.0D).offset(0.0D, -2.0D, 0.0D)));
-                    this.attackEntitiesInList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.dragonPartHead.func_174813_aQ().func_186662_g(1.0D)));
-                    this.attackEntitiesInList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.field_184673_bv.func_174813_aQ().func_186662_g(1.0D)));
+                    this.attackEntitiesInList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.dragonPartHead.func_174813_aQ().expand(1.0D)));
+                    this.attackEntitiesInList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.field_184673_bv.func_174813_aQ().expand(1.0D)));
                 }
 
                 double[] adouble = this.getMovementOffsets(5, 1.0F);
@@ -434,7 +434,7 @@
 
         if (this.rand.nextInt(10) == 0)
         {
-            List<EntityEnderCrystal> list = this.worldObj.<EntityEnderCrystal>getEntitiesWithinAABB(EntityEnderCrystal.class, this.func_174813_aQ().func_186662_g(32.0D));
+            List<EntityEnderCrystal> list = this.worldObj.<EntityEnderCrystal>getEntitiesWithinAABB(EntityEnderCrystal.class, this.func_174813_aQ().expand(32.0D));
             EntityEnderCrystal entityendercrystal = null;
             double d0 = Double.MAX_VALUE;
 
@@ -525,10 +525,10 @@
                 for (int i2 = k; i2 <= j1; ++i2)
                 {
                     BlockPos blockpos = new BlockPos(k1, l1, i2);
-                    IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
-                    Block block = iblockstate.func_177230_c();
+                    IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
+                    Block block = iblockstate.getBlock();
 
-                    if (iblockstate.func_185904_a() != Material.air && iblockstate.func_185904_a() != Material.fire)
+                    if (iblockstate.getMaterial() != Material.air && iblockstate.getMaterial() != Material.fire)
                     {
                         if (!this.worldObj.getGameRules().getGameRuleBooleanValue("mobGriefing"))
                         {
@@ -559,7 +559,7 @@
             double d0 = p_70972_1_.minX + (p_70972_1_.maxX - p_70972_1_.minX) * (double)this.rand.nextFloat();
             double d1 = p_70972_1_.minY + (p_70972_1_.maxY - p_70972_1_.minY) * (double)this.rand.nextFloat();
             double d2 = p_70972_1_.minZ + (p_70972_1_.maxZ - p_70972_1_.minZ) * (double)this.rand.nextFloat();
-            this.worldObj.func_175688_a(EnumParticleTypes.EXPLOSION_LARGE, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.EXPLOSION_LARGE, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
         }
 
         return flag;
@@ -653,7 +653,7 @@
             float f = (this.rand.nextFloat() - 0.5F) * 8.0F;
             float f1 = (this.rand.nextFloat() - 0.5F) * 4.0F;
             float f2 = (this.rand.nextFloat() - 0.5F) * 8.0F;
-            this.worldObj.func_175688_a(EnumParticleTypes.EXPLOSION_HUGE, this.posX + (double)f, this.posY + 2.0D + (double)f1, this.posZ + (double)f2, 0.0D, 0.0D, 0.0D, new int[0]);
+            this.worldObj.spawnParticle(EnumParticleTypes.EXPLOSION_HUGE, this.posX + (double)f, this.posY + 2.0D + (double)f1, this.posZ + (double)f2, 0.0D, 0.0D, 0.0D, new int[0]);
         }
 
         boolean flag = this.worldObj.getGameRules().getGameRuleBooleanValue("doMobLoot");
@@ -736,7 +736,7 @@
                     i1 = (int)(20.0F * MathHelper.sin(2.0F * (-(float)Math.PI + ((float)Math.PI / 4F) * (float)k1)));
                 }
 
-                int j1 = Math.max(this.worldObj.func_181545_F() + 10, this.worldObj.func_175672_r(new BlockPos(l, 0, i1)).func_177956_o() + j);
+                int j1 = Math.max(this.worldObj.getSpawnLevel() + 10, this.worldObj.func_175672_r(new BlockPos(l, 0, i1)).getY() + j);
                 this.field_184680_bM[i] = new PathPoint(l, j1, i1);
             }
 
