--- a/net/minecraft/client/audio/MovingSoundMinecartRiding.java
+++ b/net/minecraft/client/audio/MovingSoundMinecartRiding.java
@@ -26,7 +26,7 @@
      */
     public void update()
     {
-        if (!this.field_147671_l.isDead && this.field_147672_k.func_184218_aH() && this.field_147672_k.func_184187_bx() == this.field_147671_l)
+        if (!this.field_147671_l.isDead && this.field_147672_k.isRiding() && this.field_147672_k.getRidingEntity() == this.field_147671_l)
         {
             float f = MathHelper.sqrt_double(this.field_147671_l.motionX * this.field_147671_l.motionX + this.field_147671_l.motionZ * this.field_147671_l.motionZ);
 
