--- a/net/minecraft/client/audio/SoundManager.java
+++ b/net/minecraft/client/audio/SoundManager.java
@@ -52,7 +52,7 @@
     private final GameSettings options;
 
     /** A reference to the sound system. */
-    private SoundManager.SoundSystemStarterThread sndSystem;
+    public SoundManager.SoundSystemStarterThread sndSystem;
 
     /** Set to true when the SoundManager has been initialised. */
     private boolean loaded;
