--- a/net/minecraft/client/audio/SoundHandler.java
+++ b/net/minecraft/client/audio/SoundHandler.java
@@ -50,7 +50,9 @@
         }
     };
     private final SoundRegistry soundRegistry = new SoundRegistry();
-    private final SoundManager sndManager;
+    //VIVE make public
+    public final SoundManager sndManager;
+    ///
     private final IResourceManager mcResourceManager;
 
     public SoundHandler(IResourceManager manager, GameSettings gameSettingsIn)
