--- a/net/minecraft/client/renderer/entity/RenderIronGolem.java
+++ b/net/minecraft/client/renderer/entity/RenderIronGolem.java
@@ -33,7 +33,7 @@
             float f = 13.0F;
             float f1 = p_77043_1_.field_184619_aG - p_77043_1_.limbSwingAmount * (1.0F - p_77043_4_) + 6.0F;
             float f2 = (Math.abs(f1 % 13.0F - 6.5F) - 3.25F) / 3.25F;
-            GlStateManager.func_179114_b(6.5F * f2, 0.0F, 0.0F, 1.0F);
+            GlStateManager.rotate(6.5F * f2, 0.0F, 0.0F, 1.0F);
         }
     }
 }
