--- a/net/minecraft/client/renderer/entity/RenderDragon.java
+++ b/net/minecraft/client/renderer/entity/RenderDragon.java
@@ -33,8 +33,8 @@
     {
         float f = (float)p_77043_1_.getMovementOffsets(7, p_77043_4_)[0];
         float f1 = (float)(p_77043_1_.getMovementOffsets(5, p_77043_4_)[1] - p_77043_1_.getMovementOffsets(10, p_77043_4_)[1]);
-        GlStateManager.func_179114_b(-f, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b(f1 * 10.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(-f, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(f1 * 10.0F, 1.0F, 0.0F, 0.0F);
         GlStateManager.func_179109_b(0.0F, 0.0F, 1.0F);
 
         if (p_77043_1_.deathTime > 0)
@@ -47,7 +47,7 @@
                 f2 = 1.0F;
             }
 
-            GlStateManager.func_179114_b(f2 * this.getDeathMaxRotation(p_77043_1_), 0.0F, 0.0F, 1.0F);
+            GlStateManager.rotate(f2 * this.getDeathMaxRotation(p_77043_1_), 0.0F, 0.0F, 1.0F);
         }
     }
 
@@ -111,10 +111,10 @@
         float f2 = (float)(p_188325_18_ - p_188325_11_);
         float f3 = MathHelper.sqrt_float(f * f + f2 * f2);
         float f4 = MathHelper.sqrt_float(f * f + f1 * f1 + f2 * f2);
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         GlStateManager.func_179109_b((float)p_188325_0_, (float)p_188325_2_ + 2.0F, (float)p_188325_4_);
-        GlStateManager.func_179114_b((float)(-Math.atan2((double)f2, (double)f)) * (180F / (float)Math.PI) - 90.0F, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b((float)(-Math.atan2((double)f3, (double)f1)) * (180F / (float)Math.PI) - 90.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate((float)(-Math.atan2((double)f2, (double)f)) * (180F / (float)Math.PI) - 90.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate((float)(-Math.atan2((double)f3, (double)f1)) * (180F / (float)Math.PI) - 90.0F, 1.0F, 0.0F, 0.0F);
         Tessellator tessellator = Tessellator.func_178181_a();
         VertexBuffer vertexbuffer = tessellator.func_178180_c();
         RenderHelper.disableStandardItemLighting();
@@ -138,7 +138,7 @@
         GlStateManager.func_179089_o();
         GlStateManager.func_179103_j(7424);
         RenderHelper.enableStandardItemLighting();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 
     /**
