--- a/net/minecraft/client/renderer/entity/RenderPolarBear.java
+++ b/net/minecraft/client/renderer/entity/RenderPolarBear.java
@@ -39,7 +39,7 @@
      */
     protected void preRenderCallback(EntityPolarBear p_77041_1_, float p_77041_2_)
     {
-        GlStateManager.func_179152_a(1.2F, 1.2F, 1.2F);
+        GlStateManager.scale(1.2F, 1.2F, 1.2F);
         super.preRenderCallback(p_77041_1_, p_77041_2_);
     }
 }
