--- a/net/minecraft/client/renderer/entity/RenderGiantZombie.java
+++ b/net/minecraft/client/renderer/entity/RenderGiantZombie.java
@@ -41,7 +41,7 @@
      */
     protected void preRenderCallback(EntityGiantZombie p_77041_1_, float p_77041_2_)
     {
-        GlStateManager.func_179152_a(this.scale, this.scale, this.scale);
+        GlStateManager.scale(this.scale, this.scale, this.scale);
     }
 
     /**
