--- a/net/minecraft/client/renderer/entity/RenderManager.java
+++ b/net/minecraft/client/renderer/entity/RenderManager.java
@@ -1,6 +1,8 @@
 package net.minecraft.client.renderer.entity;
 
 import com.google.common.collect.Maps;
+import com.mtbs3d.minecrift.render.PlayerModelController;
+
 import java.util.Collections;
 import java.util.Map;
 import javax.annotation.Nullable;
@@ -115,10 +117,12 @@
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.World;
 import net.optifine.entity.model.CustomEntityModels;
+import net.minecraft.client.renderer.entity.RenderPlayerVR;
 
 public class RenderManager
 {
-    private final Map <Class, Render> entityRenderMap = Maps. <Class, Render> newHashMap();
+	//Forge make public
+	public  final Map <Class, Render> entityRenderMap = Maps. <Class, Render> newHashMap();
     private final Map<String, RenderPlayer> skinMap = Maps.<String, RenderPlayer>newHashMap();
     private final RenderPlayer playerRenderer;
 
@@ -224,10 +228,12 @@
         this.skinMap.put("slim", new RenderPlayer(this, true));
         PlayerItemsLayer.register(this.skinMap);
 
+        //Forge
         if (Reflector.RenderingRegistry_loadEntityRenderers.exists())
         {
             Reflector.call(Reflector.RenderingRegistry_loadEntityRenderers, new Object[] {this, this.entityRenderMap});
         }
+        ///
     }
 
     public void setRenderPosition(double renderPosXIn, double renderPosYIn, double renderPosZIn)
@@ -250,19 +256,27 @@
         return (Render<T>) render;
     }
 
+    RenderPlayerVR n;
+    
     @Nullable
     public <T extends Entity> Render<T> getEntityRenderObject(T entityIn)
     {
-        if (entityIn instanceof AbstractClientPlayer)
-        {
-            String s = ((AbstractClientPlayer)entityIn).getSkinType();
-            RenderPlayer renderplayer = (RenderPlayer)this.skinMap.get(s);
-            return (Render<T>) (renderplayer != null ? renderplayer : this.playerRenderer);
-        }
-        else
-        {
-            return this.<T>getEntityClassRenderObject(entityIn.getClass());
-        }
+    	if (entityIn instanceof AbstractClientPlayer)
+    	{
+    		String s = ((AbstractClientPlayer)entityIn).getSkinType();
+    		RenderPlayer renderplayer = (RenderPlayer)this.skinMap.get(s);
+
+    		if(PlayerModelController.getInstance().isTracked(((AbstractClientPlayer) entityIn).getGameProfile().getId()))
+    		{
+    			if(n==null) n = new RenderPlayerVR(this);
+    			return (Render<T>) n;            	
+    		}  else 
+    			return (Render<T>) (renderplayer != null ? renderplayer : this.playerRenderer);
+    	}      
+    	else
+    	{
+    		return this.<T>getEntityClassRenderObject(entityIn.getClass());
+    	}
     }
 
     public void cacheActiveRenderInfo(World worldIn, FontRenderer textRendererIn, Entity livingPlayerIn, Entity pointedEntityIn, GameSettings optionsIn, float partialTicks)
@@ -542,8 +556,10 @@
         return this.entityRenderMap;
     }
 
+    //Forge
     public Map<String, RenderPlayer> getSkinMap()
     {
         return Collections.<String, RenderPlayer>unmodifiableMap(this.skinMap);
     }
+    ///
 }
