--- a/net/minecraft/client/renderer/entity/RenderBat.java
+++ b/net/minecraft/client/renderer/entity/RenderBat.java
@@ -29,7 +29,7 @@
      */
     protected void preRenderCallback(EntityBat p_77041_1_, float p_77041_2_)
     {
-        GlStateManager.func_179152_a(0.35F, 0.35F, 0.35F);
+        GlStateManager.scale(0.35F, 0.35F, 0.35F);
     }
 
     protected void rotateCorpse(EntityBat p_77043_1_, float p_77043_2_, float p_77043_3_, float p_77043_4_)
