--- a/net/minecraft/client/renderer/entity/RenderZombie.java
+++ b/net/minecraft/client/renderer/entity/RenderZombie.java
@@ -68,7 +68,7 @@
         if (p_77041_1_.func_189777_di() == ZombieType.HUSK)
         {
             float f = 1.0625F;
-            GlStateManager.func_179152_a(1.0625F, 1.0625F, 1.0625F);
+            GlStateManager.scale(1.0625F, 1.0625F, 1.0625F);
         }
 
         super.preRenderCallback(p_77041_1_, p_77041_2_);
