--- a/net/minecraft/client/renderer/entity/RenderShulkerBullet.java
+++ b/net/minecraft/client/renderer/entity/RenderShulkerBullet.java
@@ -41,25 +41,25 @@
      */
     public void doRender(EntityShulkerBullet p_76986_1_, double p_76986_2_, double p_76986_4_, double p_76986_6_, float p_76986_8_, float p_76986_9_)
     {
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         float f = this.func_188347_a(p_76986_1_.prevRotationYaw, p_76986_1_.rotationYaw, p_76986_9_);
         float f1 = p_76986_1_.prevRotationPitch + (p_76986_1_.rotationPitch - p_76986_1_.prevRotationPitch) * p_76986_9_;
         float f2 = (float)p_76986_1_.ticksExisted + p_76986_9_;
         GlStateManager.func_179109_b((float)p_76986_2_, (float)p_76986_4_ + 0.15F, (float)p_76986_6_);
-        GlStateManager.func_179114_b(MathHelper.sin(f2 * 0.1F) * 180.0F, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b(MathHelper.cos(f2 * 0.1F) * 180.0F, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b(MathHelper.sin(f2 * 0.15F) * 360.0F, 0.0F, 0.0F, 1.0F);
+        GlStateManager.rotate(MathHelper.sin(f2 * 0.1F) * 180.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(MathHelper.cos(f2 * 0.1F) * 180.0F, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(MathHelper.sin(f2 * 0.15F) * 360.0F, 0.0F, 0.0F, 1.0F);
         float f3 = 0.03125F;
         GlStateManager.func_179091_B();
-        GlStateManager.func_179152_a(-1.0F, -1.0F, 1.0F);
+        GlStateManager.scale(-1.0F, -1.0F, 1.0F);
         this.func_180548_c(p_76986_1_);
         this.field_188349_b.render(p_76986_1_, 0.0F, 0.0F, 0.0F, f, f1, 0.03125F);
         GlStateManager.func_179147_l();
         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 0.5F);
-        GlStateManager.func_179152_a(1.5F, 1.5F, 1.5F);
+        GlStateManager.scale(1.5F, 1.5F, 1.5F);
         this.field_188349_b.render(p_76986_1_, 0.0F, 0.0F, 0.0F, f, f1, 0.03125F);
         GlStateManager.func_179084_k();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         super.doRender(p_76986_1_, p_76986_2_, p_76986_4_, p_76986_6_, p_76986_8_, p_76986_9_);
     }
 
