--- a/net/minecraft/client/renderer/entity/RenderSnowball.java
+++ b/net/minecraft/client/renderer/entity/RenderSnowball.java
@@ -29,12 +29,12 @@
      */
     public void doRender(T p_76986_1_, double p_76986_2_, double p_76986_4_, double p_76986_6_, float p_76986_8_, float p_76986_9_)
     {
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         GlStateManager.func_179109_b((float)p_76986_2_, (float)p_76986_4_, (float)p_76986_6_);
         GlStateManager.func_179091_B();
-        GlStateManager.func_179114_b(-this.renderManager.playerViewY, 0.0F, 1.0F, 0.0F);
-        GlStateManager.func_179114_b((float)(this.renderManager.options.thirdPersonView == 2 ? -1 : 1) * this.renderManager.playerViewX, 1.0F, 0.0F, 0.0F);
-        GlStateManager.func_179114_b(180.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(-this.renderManager.playerViewY, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate((float)(this.renderManager.options.thirdPersonView == 2 ? -1 : 1) * this.renderManager.playerViewX, 1.0F, 0.0F, 0.0F);
+        GlStateManager.rotate(180.0F, 0.0F, 1.0F, 0.0F);
         this.bindTexture(TextureMap.locationBlocksTexture);
 
         if (this.field_188301_f)
@@ -52,7 +52,7 @@
         }
 
         GlStateManager.func_179101_C();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         super.doRender(p_76986_1_, p_76986_2_, p_76986_4_, p_76986_6_, p_76986_8_, p_76986_9_);
     }
 
