--- a/net/minecraft/client/renderer/entity/RenderEntityItem.java
+++ b/net/minecraft/client/renderer/entity/RenderEntityItem.java
@@ -48,7 +48,7 @@
             if (flag || this.renderManager.options != null)
             {
                 float f3 = (((float)p_177077_1_.func_174872_o() + p_177077_8_) / 20.0F + p_177077_1_.hoverStart) * (180F / (float)Math.PI);
-                GlStateManager.func_179114_b(f3, 0.0F, 1.0F, 0.0F);
+                GlStateManager.rotate(f3, 0.0F, 1.0F, 0.0F);
             }
 
             GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
@@ -114,7 +114,7 @@
         GlStateManager.func_179147_l();
         RenderHelper.enableStandardItemLighting();
         GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         IBakedModel ibakedmodel = this.field_177080_a.func_184393_a(itemstack, p_76986_1_.worldObj, (EntityLivingBase)null);
         int j = this.func_177077_a(p_76986_1_, p_76986_2_, p_76986_4_, p_76986_6_, p_76986_9_, ibakedmodel);
         float f = ibakedmodel.func_177552_f().field_181699_o.field_178363_d.x;
@@ -140,7 +140,7 @@
         {
             if (flag1)
             {
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
 
                 if (k > 0)
                 {
@@ -152,11 +152,11 @@
 
                 ibakedmodel.func_177552_f().func_181689_a(ItemCameraTransforms.TransformType.GROUND);
                 this.field_177080_a.func_180454_a(itemstack, ibakedmodel);
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
             else
             {
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
 
                 if (k > 0)
                 {
@@ -167,7 +167,7 @@
 
                 ibakedmodel.func_177552_f().func_181689_a(ItemCameraTransforms.TransformType.GROUND);
                 this.field_177080_a.func_180454_a(itemstack, ibakedmodel);
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
                 GlStateManager.func_179109_b(0.0F * f, 0.0F * f1, 0.09375F * f2);
             }
         }
@@ -178,7 +178,7 @@
             GlStateManager.func_179119_h();
         }
 
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         GlStateManager.func_179101_C();
         GlStateManager.func_179084_k();
         this.func_180548_c(p_76986_1_);
