--- a/net/minecraft/client/renderer/entity/RenderXPOrb.java
+++ b/net/minecraft/client/renderer/entity/RenderXPOrb.java
@@ -31,7 +31,7 @@
     {
         if (!this.field_188301_f)
         {
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             GlStateManager.func_179109_b((float)p_76986_2_, (float)p_76986_4_, (float)p_76986_6_);
             this.func_180548_c(p_76986_1_);
             RenderHelper.enableStandardItemLighting();
@@ -54,10 +54,10 @@
             int i1 = 255;
             int j1 = (int)((MathHelper.sin(f9 + 4.1887903F) + 1.0F) * 0.1F * 255.0F);
             GlStateManager.func_179109_b(0.0F, 0.1F, 0.0F);
-            GlStateManager.func_179114_b(180.0F - this.renderManager.playerViewY, 0.0F, 1.0F, 0.0F);
-            GlStateManager.func_179114_b((float)(this.renderManager.options.thirdPersonView == 2 ? -1 : 1) * -this.renderManager.playerViewX, 1.0F, 0.0F, 0.0F);
+            GlStateManager.rotate(180.0F - this.renderManager.playerViewY, 0.0F, 1.0F, 0.0F);
+            GlStateManager.rotate((float)(this.renderManager.options.thirdPersonView == 2 ? -1 : 1) * -this.renderManager.playerViewX, 1.0F, 0.0F, 0.0F);
             float f7 = 0.3F;
-            GlStateManager.func_179152_a(0.3F, 0.3F, 0.3F);
+            GlStateManager.scale(0.3F, 0.3F, 0.3F);
             Tessellator tessellator = Tessellator.func_178181_a();
             VertexBuffer vertexbuffer = tessellator.func_178180_c();
             vertexbuffer.func_181668_a(7, DefaultVertexFormats.field_181712_l);
@@ -68,7 +68,7 @@
             tessellator.draw();
             GlStateManager.func_179084_k();
             GlStateManager.func_179101_C();
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
             super.doRender(p_76986_1_, p_76986_2_, p_76986_4_, p_76986_6_, p_76986_8_, p_76986_9_);
         }
     }
