--- a/net/minecraft/client/renderer/entity/RenderShulker.java
+++ b/net/minecraft/client/renderer/entity/RenderShulker.java
@@ -46,9 +46,9 @@
             BlockPos blockpos1 = p_76986_1_.func_184692_dd();
             double d0 = (double)((float)i - p_76986_9_) / 6.0D;
             d0 = d0 * d0;
-            double d1 = (double)(blockpos.func_177958_n() - blockpos1.func_177958_n()) * d0;
-            double d2 = (double)(blockpos.func_177956_o() - blockpos1.func_177956_o()) * d0;
-            double d3 = (double)(blockpos.func_177952_p() - blockpos1.func_177952_p()) * d0;
+            double d1 = (double)(blockpos.getX() - blockpos1.getX()) * d0;
+            double d2 = (double)(blockpos.getY() - blockpos1.getY()) * d0;
+            double d3 = (double)(blockpos.getZ() - blockpos1.getZ()) * d0;
             super.doRender(p_76986_1_, p_76986_2_ - d1, p_76986_4_ - d2, p_76986_6_ - d3, p_76986_8_, p_76986_9_);
         }
         else
@@ -69,8 +69,8 @@
             {
                 BlockPos blockpos = p_177071_1_.func_184692_dd();
                 BlockPos blockpos1 = p_177071_1_.func_184699_da();
-                Vec3d vec3d = new Vec3d((double)blockpos1.func_177958_n(), (double)blockpos1.func_177956_o(), (double)blockpos1.func_177952_p());
-                Vec3d vec3d1 = new Vec3d((double)blockpos.func_177958_n(), (double)blockpos.func_177956_o(), (double)blockpos.func_177952_p());
+                Vec3d vec3d = new Vec3d((double)blockpos1.getX(), (double)blockpos1.getY(), (double)blockpos1.getZ());
+                Vec3d vec3d1 = new Vec3d((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ());
 
                 if (p_177071_2_.isBoundingBoxInFrustum(new AxisAlignedBB(vec3d1.xCoord, vec3d1.yCoord, vec3d1.zCoord, vec3d.xCoord, vec3d.yCoord, vec3d.zCoord)))
                 {
@@ -102,30 +102,30 @@
 
             case EAST:
                 GlStateManager.func_179109_b(0.5F, 0.5F, 0.0F);
-                GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
-                GlStateManager.func_179114_b(90.0F, 0.0F, 0.0F, 1.0F);
+                GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(90.0F, 0.0F, 0.0F, 1.0F);
                 break;
 
             case WEST:
                 GlStateManager.func_179109_b(-0.5F, 0.5F, 0.0F);
-                GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
-                GlStateManager.func_179114_b(-90.0F, 0.0F, 0.0F, 1.0F);
+                GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(-90.0F, 0.0F, 0.0F, 1.0F);
                 break;
 
             case NORTH:
                 GlStateManager.func_179109_b(0.0F, 0.5F, -0.5F);
-                GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
                 break;
 
             case SOUTH:
                 GlStateManager.func_179109_b(0.0F, 0.5F, 0.5F);
-                GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
-                GlStateManager.func_179114_b(180.0F, 0.0F, 0.0F, 1.0F);
+                GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(180.0F, 0.0F, 0.0F, 1.0F);
                 break;
 
             case UP:
                 GlStateManager.func_179109_b(0.0F, 1.0F, 0.0F);
-                GlStateManager.func_179114_b(180.0F, 1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(180.0F, 1.0F, 0.0F, 0.0F);
         }
     }
 
@@ -136,7 +136,7 @@
     protected void preRenderCallback(EntityShulker p_77041_1_, float p_77041_2_)
     {
         float f = 0.999F;
-        GlStateManager.func_179152_a(0.999F, 0.999F, 0.999F);
+        GlStateManager.scale(0.999F, 0.999F, 0.999F);
     }
 
     class HeadLayer implements LayerRenderer<EntityShulker>
@@ -147,7 +147,7 @@
 
         public void func_177141_a(EntityShulker p_177141_1_, float p_177141_2_, float p_177141_3_, float p_177141_4_, float p_177141_5_, float p_177141_6_, float p_177141_7_, float p_177141_8_)
         {
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
 
             switch (p_177141_1_.func_184696_cZ())
             {
@@ -156,32 +156,32 @@
                     break;
 
                 case EAST:
-                    GlStateManager.func_179114_b(90.0F, 0.0F, 0.0F, 1.0F);
-                    GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
+                    GlStateManager.rotate(90.0F, 0.0F, 0.0F, 1.0F);
+                    GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
                     GlStateManager.func_179109_b(1.0F, -1.0F, 0.0F);
-                    GlStateManager.func_179114_b(180.0F, 0.0F, 1.0F, 0.0F);
+                    GlStateManager.rotate(180.0F, 0.0F, 1.0F, 0.0F);
                     break;
 
                 case WEST:
-                    GlStateManager.func_179114_b(-90.0F, 0.0F, 0.0F, 1.0F);
-                    GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
+                    GlStateManager.rotate(-90.0F, 0.0F, 0.0F, 1.0F);
+                    GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
                     GlStateManager.func_179109_b(-1.0F, -1.0F, 0.0F);
-                    GlStateManager.func_179114_b(180.0F, 0.0F, 1.0F, 0.0F);
+                    GlStateManager.rotate(180.0F, 0.0F, 1.0F, 0.0F);
                     break;
 
                 case NORTH:
-                    GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
+                    GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
                     GlStateManager.func_179109_b(0.0F, -1.0F, -1.0F);
                     break;
 
                 case SOUTH:
-                    GlStateManager.func_179114_b(180.0F, 0.0F, 0.0F, 1.0F);
-                    GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
+                    GlStateManager.rotate(180.0F, 0.0F, 0.0F, 1.0F);
+                    GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
                     GlStateManager.func_179109_b(0.0F, -1.0F, 1.0F);
                     break;
 
                 case UP:
-                    GlStateManager.func_179114_b(180.0F, 1.0F, 0.0F, 0.0F);
+                    GlStateManager.rotate(180.0F, 1.0F, 0.0F, 0.0F);
                     GlStateManager.func_179109_b(0.0F, -2.0F, 0.0F);
             }
 
@@ -190,7 +190,7 @@
             modelrenderer.rotateAngleX = p_177141_7_ * 0.017453292F;
             RenderShulker.this.bindTexture(RenderShulker.field_188342_a);
             modelrenderer.render(p_177141_8_);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
 
         public boolean func_177142_b()
