--- a/net/minecraft/entity/monster/EntityEndermite.java
+++ b/net/minecraft/entity/monster/EntityEndermite.java
@@ -85,7 +85,7 @@
 
     protected void func_180429_a(BlockPos p_180429_1_, Block p_180429_2_)
     {
-        this.func_184185_a(SoundEvents.field_187592_bb, 0.15F, 1.0F);
+        this.playSound(SoundEvents.field_187592_bb, 0.15F, 1.0F);
     }
 
     @Nullable
@@ -158,7 +158,7 @@
         {
             for (int i = 0; i < 2; ++i)
             {
-                this.worldObj.func_175688_a(EnumParticleTypes.PORTAL, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, (this.rand.nextDouble() - 0.5D) * 2.0D, -this.rand.nextDouble(), (this.rand.nextDouble() - 0.5D) * 2.0D, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.PORTAL, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, (this.rand.nextDouble() - 0.5D) * 2.0D, -this.rand.nextDouble(), (this.rand.nextDouble() - 0.5D) * 2.0D, new int[0]);
             }
         }
         else
