--- a/net/minecraft/entity/monster/EntityWitch.java
+++ b/net/minecraft/entity/monster/EntityWitch.java
@@ -128,8 +128,8 @@
                 if (this.witchAttackTimer-- <= 0)
                 {
                     this.setAggressive(false);
-                    ItemStack itemstack = this.func_184614_ca();
-                    this.func_184201_a(EntityEquipmentSlot.MAINHAND, (ItemStack)null);
+                    ItemStack itemstack = this.getMainHandItem();
+                    this.equipItemInSlot(EntityEquipmentSlot.MAINHAND, (ItemStack)null);
 
                     if (itemstack != null && itemstack.getItem() == Items.potionitem)
                     {
@@ -170,10 +170,10 @@
 
                 if (potiontype != null)
                 {
-                    this.func_184201_a(EntityEquipmentSlot.MAINHAND, PotionUtils.func_185188_a(new ItemStack(Items.potionitem), potiontype));
-                    this.witchAttackTimer = this.func_184614_ca().getMaxItemUseDuration();
+                    this.equipItemInSlot(EntityEquipmentSlot.MAINHAND, PotionUtils.func_185188_a(new ItemStack(Items.potionitem), potiontype));
+                    this.witchAttackTimer = this.getMainHandItem().getMaxItemUseDuration();
                     this.setAggressive(true);
-                    this.worldObj.func_184148_a((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187922_gv, this.func_184176_by(), 1.0F, 0.8F + this.rand.nextFloat() * 0.4F);
+                    this.worldObj.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187922_gv, this.func_184176_by(), 1.0F, 0.8F + this.rand.nextFloat() * 0.4F);
                     IAttributeInstance iattributeinstance = this.getEntityAttribute(SharedMonsterAttributes.movementSpeed);
                     iattributeinstance.removeModifier(field_110185_bq);
                     iattributeinstance.applyModifier(field_110185_bq);
@@ -195,7 +195,7 @@
         {
             for (int i = 0; i < this.rand.nextInt(35) + 10; ++i)
             {
-                this.worldObj.func_175688_a(EnumParticleTypes.SPELL_WITCH, this.posX + this.rand.nextGaussian() * 0.12999999523162842D, this.func_174813_aQ().maxY + 0.5D + this.rand.nextGaussian() * 0.12999999523162842D, this.posZ + this.rand.nextGaussian() * 0.12999999523162842D, 0.0D, 0.0D, 0.0D, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.SPELL_WITCH, this.posX + this.rand.nextGaussian() * 0.12999999523162842D, this.func_174813_aQ().maxY + 0.5D + this.rand.nextGaussian() * 0.12999999523162842D, this.posZ + this.rand.nextGaussian() * 0.12999999523162842D, 0.0D, 0.0D, 0.0D, new int[0]);
             }
         }
         else
@@ -260,7 +260,7 @@
             EntityPotion entitypotion = new EntityPotion(this.worldObj, this, PotionUtils.func_185188_a(new ItemStack(Items.field_185155_bH), potiontype));
             entitypotion.rotationPitch -= -20.0F;
             entitypotion.setThrowableHeading(d1, d2 + (double)(f * 0.2F), d3, 0.75F, 8.0F);
-            this.worldObj.func_184148_a((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187924_gx, this.func_184176_by(), 1.0F, 0.8F + this.rand.nextFloat() * 0.4F);
+            this.worldObj.playSound((EntityPlayer)null, this.posX, this.posY, this.posZ, SoundEvents.field_187924_gx, this.func_184176_by(), 1.0F, 0.8F + this.rand.nextFloat() * 0.4F);
             this.worldObj.spawnEntityInWorld(entitypotion);
         }
     }
