--- a/net/minecraft/entity/monster/EntityShulker.java
+++ b/net/minecraft/entity/monster/EntityShulker.java
@@ -188,9 +188,9 @@
 
         if (blockpos != null)
         {
-            tagCompound.setInteger("APX", blockpos.func_177958_n());
-            tagCompound.setInteger("APY", blockpos.func_177956_o());
-            tagCompound.setInteger("APZ", blockpos.func_177952_p());
+            tagCompound.setInteger("APX", blockpos.getX());
+            tagCompound.setInteger("APY", blockpos.getY());
+            tagCompound.setInteger("APZ", blockpos.getZ());
         }
     }
 
@@ -208,10 +208,10 @@
             this.dataWatcher.func_187227_b(field_184701_b, Optional.of(blockpos));
         }
 
-        if (this.func_184218_aH())
+        if (this.isRiding())
         {
             blockpos = null;
-            float f = this.func_184187_bx().rotationYaw;
+            float f = this.getRidingEntity().rotationYaw;
             this.rotationYaw = f;
             this.renderYawOffset = f;
             this.prevRenderYawOffset = f;
@@ -219,20 +219,20 @@
         }
         else if (!this.worldObj.isRemote)
         {
-            IBlockState iblockstate = this.worldObj.func_180495_p(blockpos);
+            IBlockState iblockstate = this.worldObj.getBlockState(blockpos);
 
-            if (iblockstate.func_185904_a() != Material.air)
+            if (iblockstate.getMaterial() != Material.air)
             {
-                if (iblockstate.func_177230_c() == Blocks.field_180384_M)
+                if (iblockstate.getBlock() == Blocks.field_180384_M)
                 {
                     EnumFacing enumfacing = (EnumFacing)iblockstate.func_177229_b(BlockPistonBase.field_176387_N);
-                    blockpos = blockpos.func_177972_a(enumfacing);
+                    blockpos = blockpos.getBlockPosRelative(enumfacing);
                     this.dataWatcher.func_187227_b(field_184701_b, Optional.of(blockpos));
                 }
-                else if (iblockstate.func_177230_c() == Blocks.piston_head)
+                else if (iblockstate.getBlock() == Blocks.piston_head)
                 {
                     EnumFacing enumfacing3 = (EnumFacing)iblockstate.func_177229_b(BlockPistonExtension.field_176387_N);
-                    blockpos = blockpos.func_177972_a(enumfacing3);
+                    blockpos = blockpos.getBlockPosRelative(enumfacing3);
                     this.dataWatcher.func_187227_b(field_184701_b, Optional.of(blockpos));
                 }
                 else
@@ -241,7 +241,7 @@
                 }
             }
 
-            BlockPos blockpos1 = blockpos.func_177972_a(this.func_184696_cZ());
+            BlockPos blockpos1 = blockpos.getBlockPosRelative(this.func_184696_cZ());
 
             if (!this.worldObj.func_175677_d(blockpos1, false))
             {
@@ -249,7 +249,7 @@
 
                 for (EnumFacing enumfacing1 : EnumFacing.values())
                 {
-                    blockpos1 = blockpos.func_177972_a(enumfacing1);
+                    blockpos1 = blockpos.getBlockPosRelative(enumfacing1);
 
                     if (this.worldObj.func_175677_d(blockpos1, false))
                     {
@@ -265,7 +265,7 @@
                 }
             }
 
-            BlockPos blockpos2 = blockpos.func_177972_a(this.func_184696_cZ().func_176734_d());
+            BlockPos blockpos2 = blockpos.getBlockPosRelative(this.func_184696_cZ().func_176734_d());
 
             if (this.worldObj.func_175677_d(blockpos2, false))
             {
@@ -299,9 +299,9 @@
                 }
             }
 
-            this.posX = (double)blockpos.func_177958_n() + 0.5D;
-            this.posY = (double)blockpos.func_177956_o();
-            this.posZ = (double)blockpos.func_177952_p() + 0.5D;
+            this.posX = (double)blockpos.getX() + 0.5D;
+            this.posY = (double)blockpos.getY();
+            this.posZ = (double)blockpos.getZ() + 0.5D;
             this.prevPosX = this.posX;
             this.prevPosY = this.posY;
             this.prevPosZ = this.posZ;
@@ -319,32 +319,32 @@
             switch (enumfacing2)
             {
                 case DOWN:
-                    this.func_174826_a(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D + d3, this.posZ + 0.5D));
+                    this.setBoundingBox(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D + d3, this.posZ + 0.5D));
                     d1 = d5;
                     break;
 
                 case UP:
-                    this.func_174826_a(new AxisAlignedBB(this.posX - 0.5D, this.posY - d3, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D));
+                    this.setBoundingBox(new AxisAlignedBB(this.posX - 0.5D, this.posY - d3, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D));
                     d1 = -d5;
                     break;
 
                 case NORTH:
-                    this.func_174826_a(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D + d3));
+                    this.setBoundingBox(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D + d3));
                     d2 = d5;
                     break;
 
                 case SOUTH:
-                    this.func_174826_a(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D - d3, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D));
+                    this.setBoundingBox(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D - d3, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D));
                     d2 = -d5;
                     break;
 
                 case WEST:
-                    this.func_174826_a(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D, this.posX + 0.5D + d3, this.posY + 1.0D, this.posZ + 0.5D));
+                    this.setBoundingBox(new AxisAlignedBB(this.posX - 0.5D, this.posY, this.posZ - 0.5D, this.posX + 0.5D + d3, this.posY + 1.0D, this.posZ + 0.5D));
                     d0 = d5;
                     break;
 
                 case EAST:
-                    this.func_174826_a(new AxisAlignedBB(this.posX - 0.5D - d3, this.posY, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D));
+                    this.setBoundingBox(new AxisAlignedBB(this.posX - 0.5D - d3, this.posY, this.posZ - 0.5D, this.posX + 0.5D, this.posY + 1.0D, this.posZ + 0.5D));
                     d0 = -d5;
             }
 
@@ -397,13 +397,13 @@
             {
                 BlockPos blockpos1 = blockpos.func_177982_a(8 - this.rand.nextInt(17), 8 - this.rand.nextInt(17), 8 - this.rand.nextInt(17));
 
-                if (blockpos1.func_177956_o() > 0 && this.worldObj.func_175623_d(blockpos1) && this.worldObj.func_175673_a(this.worldObj.func_175723_af(), this) && this.worldObj.func_184144_a(this, new AxisAlignedBB(blockpos1)).isEmpty())
+                if (blockpos1.getY() > 0 && this.worldObj.isEmpty(blockpos1) && this.worldObj.func_175673_a(this.worldObj.func_175723_af(), this) && this.worldObj.getCollisionBoxes(this, new AxisAlignedBB(blockpos1)).isEmpty())
                 {
                     boolean flag = false;
 
                     for (EnumFacing enumfacing : EnumFacing.values())
                     {
-                        if (this.worldObj.func_175677_d(blockpos1.func_177972_a(enumfacing), false))
+                        if (this.worldObj.func_175677_d(blockpos1.getBlockPosRelative(enumfacing), false))
                         {
                             this.dataWatcher.func_187227_b(field_184700_a, enumfacing);
                             flag = true;
@@ -413,7 +413,7 @@
 
                     if (flag)
                     {
-                        this.func_184185_a(SoundEvents.field_187791_eX, 1.0F, 1.0F);
+                        this.playSound(SoundEvents.field_187791_eX, 1.0F, 1.0F);
                         this.dataWatcher.func_187227_b(field_184701_b, Optional.of(blockpos1));
                         this.dataWatcher.func_187227_b(field_184702_c, Byte.valueOf((byte)0));
                         this.setAttackTarget((EntityLivingBase)null);
@@ -447,7 +447,7 @@
 
     public void func_184206_a(DataParameter<?> p_184206_1_)
     {
-        if (field_184701_b.equals(p_184206_1_) && this.worldObj.isRemote && !this.func_184218_aH())
+        if (field_184701_b.equals(p_184206_1_) && this.worldObj.isRemote && !this.isRiding())
         {
             BlockPos blockpos = this.func_184699_da();
 
@@ -462,9 +462,9 @@
                     this.field_184708_bA = 6;
                 }
 
-                this.posX = (double)blockpos.func_177958_n() + 0.5D;
-                this.posY = (double)blockpos.func_177956_o();
-                this.posZ = (double)blockpos.func_177952_p() + 0.5D;
+                this.posX = (double)blockpos.getX() + 0.5D;
+                this.posY = (double)blockpos.getY();
+                this.posZ = (double)blockpos.getZ() + 0.5D;
                 this.prevPosX = this.posX;
                 this.prevPosY = this.posY;
                 this.prevPosZ = this.posZ;
@@ -557,11 +557,11 @@
             if (p_184691_1_ == 0)
             {
                 this.getEntityAttribute(SharedMonsterAttributes.field_188791_g).applyModifier(field_184704_bw);
-                this.func_184185_a(SoundEvents.field_187779_eR, 1.0F, 1.0F);
+                this.playSound(SoundEvents.field_187779_eR, 1.0F, 1.0F);
             }
             else
             {
-                this.func_184185_a(SoundEvents.field_187787_eV, 1.0F, 1.0F);
+                this.playSound(SoundEvents.field_187787_eV, 1.0F, 1.0F);
             }
         }
 
@@ -667,7 +667,7 @@
                         this.field_188520_b = 20 + EntityShulker.this.rand.nextInt(10) * 20 / 2;
                         EntityShulkerBullet entityshulkerbullet = new EntityShulkerBullet(EntityShulker.this.worldObj, EntityShulker.this, entitylivingbase, EntityShulker.this.func_184696_cZ().func_176740_k());
                         EntityShulker.this.worldObj.spawnEntityInWorld(entityshulkerbullet);
-                        EntityShulker.this.func_184185_a(SoundEvents.field_187789_eW, 2.0F, (EntityShulker.this.rand.nextFloat() - EntityShulker.this.rand.nextFloat()) * 0.2F + 1.0F);
+                        EntityShulker.this.playSound(SoundEvents.field_187789_eW, 2.0F, (EntityShulker.this.rand.nextFloat() - EntityShulker.this.rand.nextFloat()) * 0.2F + 1.0F);
                     }
                 }
                 else
