--- a/net/minecraft/entity/monster/EntityZombie.java
+++ b/net/minecraft/entity/monster/EntityZombie.java
@@ -237,12 +237,12 @@
         if (this.worldObj.isDaytime() && !this.worldObj.isRemote && !this.isChild() && this.func_189777_di().func_190155_e())
         {
             float f = this.getBrightness(1.0F);
-            BlockPos blockpos = this.func_184187_bx() instanceof EntityBoat ? (new BlockPos(this.posX, (double)Math.round(this.posY), this.posZ)).func_177984_a() : new BlockPos(this.posX, (double)Math.round(this.posY), this.posZ);
+            BlockPos blockpos = this.getRidingEntity() instanceof EntityBoat ? (new BlockPos(this.posX, (double)Math.round(this.posY), this.posZ)).getBlockPosAbove() : new BlockPos(this.posX, (double)Math.round(this.posY), this.posZ);
 
             if (f > 0.5F && this.rand.nextFloat() * 30.0F < (f - 0.4F) * 2.0F && this.worldObj.func_175678_i(blockpos))
             {
                 boolean flag = true;
-                ItemStack itemstack = this.func_184582_a(EntityEquipmentSlot.HEAD);
+                ItemStack itemstack = this.getEquippedItemStack(EntityEquipmentSlot.HEAD);
 
                 if (itemstack != null)
                 {
@@ -253,7 +253,7 @@
                         if (itemstack.getCurrentDurability() >= itemstack.getMaxDurability())
                         {
                             this.renderBrokenItemStack(itemstack);
-                            this.func_184201_a(EntityEquipmentSlot.HEAD, (ItemStack)null);
+                            this.equipItemInSlot(EntityEquipmentSlot.HEAD, (ItemStack)null);
                         }
                     }
 
@@ -297,11 +297,11 @@
                     int j1 = j + MathHelper.getRandomIntegerInRange(this.rand, 7, 40) * MathHelper.getRandomIntegerInRange(this.rand, -1, 1);
                     int k1 = k + MathHelper.getRandomIntegerInRange(this.rand, 7, 40) * MathHelper.getRandomIntegerInRange(this.rand, -1, 1);
 
-                    if (this.worldObj.func_180495_p(new BlockPos(i1, j1 - 1, k1)).func_185896_q() && this.worldObj.func_175671_l(new BlockPos(i1, j1, k1)) < 10)
+                    if (this.worldObj.getBlockState(new BlockPos(i1, j1 - 1, k1)).func_185896_q() && this.worldObj.func_175671_l(new BlockPos(i1, j1, k1)) < 10)
                     {
                         entityzombie.setPosition((double)i1, (double)j1, (double)k1);
 
-                        if (!this.worldObj.func_175636_b((double)i1, (double)j1, (double)k1, 7.0D) && this.worldObj.checkNoEntityCollision(entityzombie.func_174813_aQ(), entityzombie) && this.worldObj.func_184144_a(entityzombie, entityzombie.func_174813_aQ()).isEmpty() && !this.worldObj.isAnyLiquid(entityzombie.func_174813_aQ()))
+                        if (!this.worldObj.func_175636_b((double)i1, (double)j1, (double)k1, 7.0D) && this.worldObj.checkNoEntityCollision(entityzombie.func_174813_aQ(), entityzombie) && this.worldObj.getCollisionBoxes(entityzombie, entityzombie.func_174813_aQ()).isEmpty() && !this.worldObj.isAnyLiquid(entityzombie.func_174813_aQ()))
                         {
                             this.worldObj.spawnEntityInWorld(entityzombie);
                             entityzombie.setAttackTarget(entitylivingbase);
@@ -349,7 +349,7 @@
         {
             float f = this.worldObj.func_175649_E(new BlockPos(this)).func_180168_b();
 
-            if (this.func_184614_ca() == null)
+            if (this.getMainHandItem() == null)
             {
                 if (this.isBurning() && this.rand.nextFloat() < f * 0.3F)
                 {
@@ -384,7 +384,7 @@
     protected void func_180429_a(BlockPos p_180429_1_, Block p_180429_2_)
     {
         SoundEvent soundevent = this.func_189777_di().func_190149_i();
-        this.func_184185_a(soundevent, 0.15F, 1.0F);
+        this.playSound(soundevent, 0.15F, 1.0F);
     }
 
     /**
@@ -411,11 +411,11 @@
 
             if (i == 0)
             {
-                this.func_184201_a(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.iron_sword));
+                this.equipItemInSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.iron_sword));
             }
             else
             {
-                this.func_184201_a(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.iron_shovel));
+                this.equipItemInSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.iron_shovel));
             }
         }
     }
@@ -527,7 +527,7 @@
 
     protected boolean func_175448_a(ItemStack p_175448_1_)
     {
-        return p_175448_1_.getItem() == Items.egg && this.isChild() && this.func_184218_aH() ? false : super.func_175448_a(p_175448_1_);
+        return p_175448_1_.getItem() == Items.egg && this.isChild() && this.isRiding() ? false : super.func_175448_a(p_175448_1_);
     }
 
     @Nullable
@@ -590,13 +590,13 @@
         this.func_180481_a(p_180482_1_);
         this.func_180483_b(p_180482_1_);
 
-        if (this.func_184582_a(EntityEquipmentSlot.HEAD) == null)
+        if (this.getEquippedItemStack(EntityEquipmentSlot.HEAD) == null)
         {
             Calendar calendar = this.worldObj.getCurrentDate();
 
             if (calendar.get(2) + 1 == 10 && calendar.get(5) == 31 && this.rand.nextFloat() < 0.25F)
             {
-                this.func_184201_a(EntityEquipmentSlot.HEAD, new ItemStack(this.rand.nextFloat() < 0.1F ? Blocks.lit_pumpkin : Blocks.pumpkin));
+                this.equipItemInSlot(EntityEquipmentSlot.HEAD, new ItemStack(this.rand.nextFloat() < 0.1F ? Blocks.lit_pumpkin : Blocks.pumpkin));
                 this.field_184655_bs[EntityEquipmentSlot.HEAD.func_188454_b()] = 0.0F;
             }
         }
@@ -733,7 +733,7 @@
                 {
                     for (int i1 = (int)this.posZ - 4; i1 < (int)this.posZ + 4 && j < 14; ++i1)
                     {
-                        Block block = this.worldObj.func_180495_p(blockpos$mutableblockpos.func_181079_c(k, l, i1)).func_177230_c();
+                        Block block = this.worldObj.getBlockState(blockpos$mutableblockpos.func_181079_c(k, l, i1)).getBlock();
 
                         if (block == Blocks.iron_bars || block == Blocks.bed)
                         {
