--- a/net/minecraft/entity/monster/EntityPolarBear.java
+++ b/net/minecraft/entity/monster/EntityPolarBear.java
@@ -103,14 +103,14 @@
 
     protected void func_180429_a(BlockPos p_180429_1_, Block p_180429_2_)
     {
-        this.func_184185_a(SoundEvents.field_190030_ev, 0.15F, 1.0F);
+        this.playSound(SoundEvents.field_190030_ev, 0.15F, 1.0F);
     }
 
     protected void func_189796_de()
     {
         if (this.field_189797_bB <= 0)
         {
-            this.func_184185_a(SoundEvents.field_190031_ew, 1.0F, 1.0F);
+            this.playSound(SoundEvents.field_190031_ew, 1.0F, 1.0F);
             this.field_189797_bB = 40;
         }
     }
