--- a/net/minecraft/entity/monster/EntityEnderman.java
+++ b/net/minecraft/entity/monster/EntityEnderman.java
@@ -166,8 +166,8 @@
 
         if (iblockstate != null)
         {
-            tagCompound.setShort("carried", (short)Block.getIdFromBlock(iblockstate.func_177230_c()));
-            tagCompound.setShort("carriedData", (short)iblockstate.func_177230_c().func_176201_c(iblockstate));
+            tagCompound.setShort("carried", (short)Block.getIdFromBlock(iblockstate.getBlock()));
+            tagCompound.setShort("carriedData", (short)iblockstate.getBlock().func_176201_c(iblockstate));
         }
     }
 
@@ -188,7 +188,7 @@
             iblockstate = Block.getBlockById(tagCompund.getShort("carried")).func_176203_a(tagCompund.getShort("carriedData") & 65535);
         }
 
-        if (iblockstate == null || iblockstate.func_177230_c() == null || iblockstate.func_185904_a() == Material.air)
+        if (iblockstate == null || iblockstate.getBlock() == null || iblockstate.getMaterial() == Material.air)
         {
             iblockstate = null;
         }
@@ -233,7 +233,7 @@
         {
             for (int i = 0; i < 2; ++i)
             {
-                this.worldObj.func_175688_a(EnumParticleTypes.PORTAL, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height - 0.25D, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, (this.rand.nextDouble() - 0.5D) * 2.0D, -this.rand.nextDouble(), (this.rand.nextDouble() - 0.5D) * 2.0D, new int[0]);
+                this.worldObj.spawnParticle(EnumParticleTypes.PORTAL, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width, this.posY + this.rand.nextDouble() * (double)this.height - 0.25D, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width, (this.rand.nextDouble() - 0.5D) * 2.0D, -this.rand.nextDouble(), (this.rand.nextDouble() - 0.5D) * 2.0D, new int[0]);
             }
         }
 
@@ -296,8 +296,8 @@
 
         if (flag)
         {
-            this.worldObj.func_184148_a((EntityPlayer)null, this.prevPosX, this.prevPosY, this.prevPosZ, SoundEvents.field_187534_aX, this.func_184176_by(), 1.0F, 1.0F);
-            this.func_184185_a(SoundEvents.field_187534_aX, 1.0F, 1.0F);
+            this.worldObj.playSound((EntityPlayer)null, this.prevPosX, this.prevPosY, this.prevPosZ, SoundEvents.field_187534_aX, this.func_184176_by(), 1.0F, 1.0F);
+            this.playSound(SoundEvents.field_187534_aX, 1.0F, 1.0F);
         }
 
         return flag;
@@ -328,11 +328,11 @@
 
         if (iblockstate != null)
         {
-            Item item = Item.getItemFromBlock(iblockstate.func_177230_c());
+            Item item = Item.getItemFromBlock(iblockstate.getBlock());
 
             if (item != null)
             {
-                int i = item.getHasSubtypes() ? iblockstate.func_177230_c().func_176201_c(iblockstate) : 0;
+                int i = item.getHasSubtypes() ? iblockstate.getBlock().func_176201_c(iblockstate) : 0;
                 this.entityDropItem(new ItemStack(item, 1, i), 0.0F);
             }
         }
@@ -528,11 +528,11 @@
             int j = MathHelper.floor_double(this.field_179475_a.posY + random.nextDouble() * 2.0D);
             int k = MathHelper.floor_double(this.field_179475_a.posZ - 1.0D + random.nextDouble() * 2.0D);
             BlockPos blockpos = new BlockPos(i, j, k);
-            IBlockState iblockstate = world.func_180495_p(blockpos);
-            IBlockState iblockstate1 = world.func_180495_p(blockpos.func_177977_b());
+            IBlockState iblockstate = world.getBlockState(blockpos);
+            IBlockState iblockstate1 = world.getBlockState(blockpos.getBlockPosBelow());
             IBlockState iblockstate2 = this.field_179475_a.func_175489_ck();
 
-            if (iblockstate2 != null && this.func_188518_a(world, blockpos, iblockstate2.func_177230_c(), iblockstate, iblockstate1))
+            if (iblockstate2 != null && this.func_188518_a(world, blockpos, iblockstate2.getBlock(), iblockstate, iblockstate1))
             {
                 world.func_180501_a(blockpos, iblockstate2, 3);
                 this.field_179475_a.func_175490_a((IBlockState)null);
@@ -541,7 +541,7 @@
 
         private boolean func_188518_a(World p_188518_1_, BlockPos p_188518_2_, Block p_188518_3_, IBlockState p_188518_4_, IBlockState p_188518_5_)
         {
-            return !p_188518_3_.func_176196_c(p_188518_1_, p_188518_2_) ? false : (p_188518_4_.func_185904_a() != Material.air ? false : (p_188518_5_.func_185904_a() == Material.air ? false : p_188518_5_.func_185917_h()));
+            return !p_188518_3_.func_176196_c(p_188518_1_, p_188518_2_) ? false : (p_188518_4_.getMaterial() != Material.air ? false : (p_188518_5_.getMaterial() == Material.air ? false : p_188518_5_.func_185917_h()));
         }
     }
 
@@ -567,10 +567,10 @@
             int j = MathHelper.floor_double(this.field_179473_a.posY + random.nextDouble() * 3.0D);
             int k = MathHelper.floor_double(this.field_179473_a.posZ - 2.0D + random.nextDouble() * 4.0D);
             BlockPos blockpos = new BlockPos(i, j, k);
-            IBlockState iblockstate = world.func_180495_p(blockpos);
-            Block block = iblockstate.func_177230_c();
+            IBlockState iblockstate = world.getBlockState(blockpos);
+            Block block = iblockstate.getBlock();
             RayTraceResult raytraceresult = world.rayTraceBlocks(new Vec3d((double)((float)MathHelper.floor_double(this.field_179473_a.posX) + 0.5F), (double)((float)j + 0.5F), (double)((float)MathHelper.floor_double(this.field_179473_a.posZ) + 0.5F)), new Vec3d((double)((float)i + 0.5F), (double)((float)j + 0.5F), (double)((float)k + 0.5F)), false, true, false);
-            boolean flag = raytraceresult != null && raytraceresult.func_178782_a().equals(blockpos);
+            boolean flag = raytraceresult != null && raytraceresult.getBlock().equals(blockpos);
 
             if (EntityEnderman.carriableBlocks.contains(block) && flag)
             {
