--- a/net/minecraft/entity/monster/EntityGuardian.java
+++ b/net/minecraft/entity/monster/EntityGuardian.java
@@ -289,7 +289,7 @@
 
     public float func_180484_a(BlockPos p_180484_1_)
     {
-        return this.worldObj.func_180495_p(p_180484_1_).func_185904_a() == Material.water ? 10.0F + this.worldObj.func_175724_o(p_180484_1_) - 0.5F : super.func_180484_a(p_180484_1_);
+        return this.worldObj.getBlockState(p_180484_1_).getMaterial() == Material.water ? 10.0F + this.worldObj.func_175724_o(p_180484_1_) - 0.5F : super.func_180484_a(p_180484_1_);
     }
 
     /**
@@ -311,7 +311,7 @@
                     this.worldObj.func_184134_a(this.posX, this.posY, this.posZ, SoundEvents.field_187684_cg, this.func_184176_by(), 1.0F, 1.0F, false);
                 }
 
-                this.field_175480_bp = this.motionY < 0.0D && this.worldObj.func_175677_d((new BlockPos(this)).func_177977_b(), false);
+                this.field_175480_bp = this.motionY < 0.0D && this.worldObj.func_175677_d((new BlockPos(this)).getBlockPosBelow(), false);
             }
             else if (this.func_175472_n())
             {
@@ -351,7 +351,7 @@
 
                 for (int i = 0; i < 2; ++i)
                 {
-                    this.worldObj.func_175688_a(EnumParticleTypes.WATER_BUBBLE, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width - vec3d.xCoord * 1.5D, this.posY + this.rand.nextDouble() * (double)this.height - vec3d.yCoord * 1.5D, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width - vec3d.zCoord * 1.5D, 0.0D, 0.0D, 0.0D, new int[0]);
+                    this.worldObj.spawnParticle(EnumParticleTypes.WATER_BUBBLE, this.posX + (this.rand.nextDouble() - 0.5D) * (double)this.width - vec3d.xCoord * 1.5D, this.posY + this.rand.nextDouble() * (double)this.height - vec3d.yCoord * 1.5D, this.posZ + (this.rand.nextDouble() - 0.5D) * (double)this.width - vec3d.zCoord * 1.5D, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
             }
 
@@ -381,7 +381,7 @@
                     while (d4 < d3)
                     {
                         d4 += 1.8D - d5 + this.rand.nextDouble() * (1.7D - d5);
-                        this.worldObj.func_175688_a(EnumParticleTypes.WATER_BUBBLE, this.posX + d0 * d4, this.posY + d1 * d4 + (double)this.getEyeHeight(), this.posZ + d2 * d4, 0.0D, 0.0D, 0.0D, new int[0]);
+                        this.worldObj.spawnParticle(EnumParticleTypes.WATER_BUBBLE, this.posX + d0 * d4, this.posY + d1 * d4 + (double)this.getEyeHeight(), this.posZ + d2 * d4, 0.0D, 0.0D, 0.0D, new int[0]);
                     }
                 }
             }
@@ -481,7 +481,7 @@
      */
     public boolean handleLavaMovement()
     {
-        return this.worldObj.checkNoEntityCollision(this.func_174813_aQ(), this) && this.worldObj.func_184144_a(this, this.func_174813_aQ()).isEmpty();
+        return this.worldObj.checkNoEntityCollision(this.func_174813_aQ(), this) && this.worldObj.getCollisionBoxes(this, this.func_174813_aQ()).isEmpty();
     }
 
     /**
