--- a/net/minecraft/entity/monster/EntityMagmaCube.java
+++ b/net/minecraft/entity/monster/EntityMagmaCube.java
@@ -44,7 +44,7 @@
      */
     public boolean handleLavaMovement()
     {
-        return this.worldObj.checkNoEntityCollision(this.func_174813_aQ(), this) && this.worldObj.func_184144_a(this, this.func_174813_aQ()).isEmpty() && !this.worldObj.isAnyLiquid(this.func_174813_aQ());
+        return this.worldObj.checkNoEntityCollision(this.func_174813_aQ(), this) && this.worldObj.getCollisionBoxes(this, this.func_174813_aQ()).isEmpty() && !this.worldObj.isAnyLiquid(this.func_174813_aQ());
     }
 
     protected void setSlimeSize(int p_70799_1_)
