--- a/net/minecraft/entity/monster/EntitySlime.java
+++ b/net/minecraft/entity/monster/EntitySlime.java
@@ -155,10 +155,10 @@
                 EnumParticleTypes enumparticletypes = this.func_180487_n();
                 double d0 = this.posX + (double)f2;
                 double d1 = this.posZ + (double)f3;
-                world.func_175688_a(enumparticletypes, d0, this.func_174813_aQ().minY, d1, 0.0D, 0.0D, 0.0D, new int[0]);
+                world.spawnParticle(enumparticletypes, d0, this.func_174813_aQ().minY, d1, 0.0D, 0.0D, 0.0D, new int[0]);
             }
 
-            this.func_184185_a(this.func_184709_cY(), this.getSoundVolume(), ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F) / 0.8F);
+            this.playSound(this.func_184709_cY(), this.getSoundVolume(), ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F) / 0.8F);
             this.squishAmount = -0.5F;
         }
         else if (!this.onGround && this.field_175452_bi)
@@ -272,7 +272,7 @@
 
         if (this.canEntityBeSeen(p_175451_1_) && this.getDistanceSqToEntity(p_175451_1_) < 0.6D * (double)i * 0.6D * (double)i && p_175451_1_.attackEntityFrom(DamageSource.causeMobDamage(this), (float)this.getAttackStrength()))
         {
-            this.func_184185_a(SoundEvents.field_187870_fk, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
+            this.playSound(SoundEvents.field_187870_fk, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
             this.func_174815_a(this, p_175451_1_);
         }
     }
@@ -330,7 +330,7 @@
     public boolean getCanSpawnHere()
     {
         BlockPos blockpos = new BlockPos(MathHelper.floor_double(this.posX), 0, MathHelper.floor_double(this.posZ));
-        Chunk chunk = this.worldObj.func_175726_f(blockpos);
+        Chunk chunk = this.worldObj.getChunk(blockpos);
 
         if (this.worldObj.getWorldInfo().getTerrainType() == WorldType.FLAT && this.rand.nextInt(4) != 1)
         {
@@ -581,7 +581,7 @@
 
                         if (this.field_179925_i.makesSoundOnJump())
                         {
-                            this.field_179925_i.func_184185_a(this.field_179925_i.func_184710_cZ(), this.field_179925_i.getSoundVolume(), ((this.field_179925_i.getRNG().nextFloat() - this.field_179925_i.getRNG().nextFloat()) * 0.2F + 1.0F) * 0.8F);
+                            this.field_179925_i.playSound(this.field_179925_i.func_184710_cZ(), this.field_179925_i.getSoundVolume(), ((this.field_179925_i.getRNG().nextFloat() - this.field_179925_i.getRNG().nextFloat()) * 0.2F + 1.0F) * 0.8F);
                         }
                     }
                     else
