--- a/net/minecraft/enchantment/Enchantment.java
+++ b/net/minecraft/enchantment/Enchantment.java
@@ -56,7 +56,7 @@
 
         for (EntityEquipmentSlot entityequipmentslot : this.field_185263_a)
         {
-            ItemStack itemstack = p_185260_1_.func_184582_a(entityequipmentslot);
+            ItemStack itemstack = p_185260_1_.getEquippedItemStack(entityequipmentslot);
 
             if (itemstack != null)
             {
