--- a/net/minecraft/item/ItemBow.java
+++ b/net/minecraft/item/ItemBow.java
@@ -48,6 +48,20 @@
         });
     }
 
+    public ItemStack findAmmoItemStack(EntityPlayer player){
+        boolean flag = player.capabilities.isCreativeMode || EnchantmentHelper.getEnchantmentLevel(Enchantments.INFINITY, player.getHeldItemMainhand()) > 0;
+        ItemStack itemstack = this.findAmmo(player);
+
+        if (itemstack != null || flag)
+        {
+            if (itemstack == null)
+            {
+                return new ItemStack(Items.ARROW);
+            }
+        }
+        return itemstack;
+    }
+    
     private ItemStack findAmmo(EntityPlayer player)
     {
         if (this.isArrow(player.getHeldItem(EnumHand.OFF_HAND)))
