--- a/shadersmod/client/Shaders.java
+++ b/shadersmod/client/Shaders.java
@@ -29,6 +29,7 @@
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.Minecraft.renderPass;
 import net.minecraft.client.model.ModelBase;
 import net.minecraft.client.model.ModelRenderer;
 import net.minecraft.client.renderer.EntityRenderer;
@@ -191,7 +192,7 @@
     static int usedShadowDepthBuffers = 0;
     static int usedColorAttachs = 0;
     static int usedDrawBuffers = 0;
-    static int dfb = 0;
+    public static int dfb = 0;
     static int sfb = 0;
     private static int[] gbuffersFormat = new int[8];
     public static int activeProgram = 0;
@@ -349,10 +350,10 @@
     static final FloatBuffer tempMatrixDirectBuffer = nextFloatBuffer(16);
     static final FloatBuffer tempDirectFloatBuffer = nextFloatBuffer(16);
     static final IntBuffer dfbColorTextures = nextIntBuffer(16);
-    static final IntBuffer dfbDepthTextures = nextIntBuffer(3);
+    public static final IntBuffer dfbDepthTextures = nextIntBuffer(3);
     static final IntBuffer sfbColorTextures = nextIntBuffer(8);
     static final IntBuffer sfbDepthTextures = nextIntBuffer(2);
-    static final IntBuffer dfbDrawBuffers = nextIntBuffer(8);
+        static final IntBuffer dfbDrawBuffers = nextIntBuffer(8);
     static final IntBuffer sfbDrawBuffers = nextIntBuffer(8);
     static final IntBuffer drawBuffersNone = nextIntBuffer(8);
     static final IntBuffer drawBuffersAll = nextIntBuffer(8);
@@ -687,11 +688,11 @@
             flag2 = true;
         }
 
-        if (Config.isFastRender())
-        {
-            SMCLog.info("Shaders can not be loaded, Fast Render is enabled.");
-            flag2 = true;
-        }
+//        if (Config.isFastRender())
+//        {
+//            SMCLog.info("Shaders can not be loaded, Fast Render is enabled.");
+//            flag2 = true;
+//        }
 
         String s = shadersConfig.getProperty(EnumShaderOption.SHADER_PACK.getPropertyKey(), packNameDefault);
 
@@ -1231,11 +1232,10 @@
         SMCLog.info(stringbuilder.toString());
     }
 
-    public static void startup(Minecraft mc)
+    public static void startup(Minecraft mcin)
     {
         checkShadersModInstalled();
-        mc = mc;
-        mc = Minecraft.getMinecraft();
+        mc = mcin;
         capabilities = GLContext.getCapabilities();
         glVersionString = GL11.glGetString(GL11.GL_VERSION);
         glVendorString = GL11.glGetString(GL11.GL_VENDOR);
@@ -2822,14 +2822,14 @@
                 programsDrawBuffers[i] = null;
                 programsCompositeMipmapSetting[i] = 0;
             }
-
+           
             if (dfb != 0)
             {
                 EXTFramebufferObject.glDeleteFramebuffersEXT(dfb);
                 dfb = 0;
-                checkGLError("del dfb");
+                checkGLError("del dfbR");
             }
-
+            
             if (sfb != 0)
             {
                 EXTFramebufferObject.glDeleteFramebuffersEXT(sfb);
@@ -2896,11 +2896,11 @@
 
     private static void resize()
     {
-        renderDisplayWidth = mc.displayWidth;
-        renderDisplayHeight = mc.displayHeight;
+        renderDisplayWidth = mc.displayFBWidth;
+        renderDisplayHeight = mc.displayFBHeight;
         renderWidth = Math.round((float)renderDisplayWidth * configRenderResMul);
         renderHeight = Math.round((float)renderDisplayHeight * configRenderResMul);
-        setupFrameBuffer();
+        dfb = setupFrameBuffer(dfb, renderWidth, renderHeight);
     }
 
     private static void resizeShadow()
@@ -2908,25 +2908,25 @@
         needResizeShadow = false;
         shadowMapWidth = Math.round((float)spShadowMapWidth * configShadowResMul);
         shadowMapHeight = Math.round((float)spShadowMapHeight * configShadowResMul);
-        setupShadowFrameBuffer();
+        sfb = setupShadowFrameBuffer(sfb, shadowMapHeight, shadowMapWidth);
     }
 
-    private static void setupFrameBuffer()
+    private static int setupFrameBuffer(int id, int w, int h)
     {
-        if (dfb != 0)
+        if (id != 0)
         {
-            EXTFramebufferObject.glDeleteFramebuffersEXT(dfb);
+            EXTFramebufferObject.glDeleteFramebuffersEXT(id);
             GlStateManager.deleteTextures(dfbDepthTextures);
             GlStateManager.deleteTextures(dfbColorTextures);
         }
 
-        dfb = EXTFramebufferObject.glGenFramebuffersEXT();
+        id = EXTFramebufferObject.glGenFramebuffersEXT();
         GL11.glGenTextures((IntBuffer)dfbDepthTextures.clear().limit(usedDepthBuffers));
         GL11.glGenTextures((IntBuffer)dfbColorTextures.clear().limit(16));
         dfbDepthTextures.position(0);
         dfbColorTextures.position(0);
         dfbColorTextures.get(dfbColorTexturesA).position(0);
-        EXTFramebufferObject.glBindFramebufferEXT(36160, dfb);
+        EXTFramebufferObject.glBindFramebufferEXT(36160, id);
         GL20.glDrawBuffers(0);
         GL11.glReadBuffer(0);
 
@@ -2938,10 +2938,18 @@
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MIN_FILTER, GL11.GL_NEAREST);
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, GL11.GL_NEAREST);
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL14.GL_DEPTH_TEXTURE_MODE, GL11.GL_LUMINANCE);
-            GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_DEPTH_COMPONENT, renderWidth, renderHeight, 0, GL11.GL_DEPTH_COMPONENT, GL11.GL_FLOAT, (FloatBuffer)((FloatBuffer)null));
+            GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL30.GL_DEPTH24_STENCIL8, w, h, 0, GL30.GL_DEPTH_STENCIL, GL30.GL_UNSIGNED_INT_24_8, (FloatBuffer)(null));
         }
+        
+        dfbDepthTextures.position(0);
+        GlStateManager.bindTexture(0);
+        EXTFramebufferObject.glFramebufferTexture2DEXT(GL30.GL_FRAMEBUFFER, GL30.GL_DEPTH_STENCIL_ATTACHMENT, GL11.GL_TEXTURE_2D, dfbDepthTextures.get(0), 0);
+
+		int ok = GL30.glCheckFramebufferStatus(GL30.GL_FRAMEBUFFER);
+		System.out.println("hack stencil into SMC " + id  + " " +ok);
+		mc.checkGLError("hack smc stencil");
+        //VIVECRAFT ADD STENCIL
 
-        EXTFramebufferObject.glFramebufferTexture2DEXT(36160, 36096, 3553, dfbDepthTextures.get(0), 0);
         GL20.glDrawBuffers(dfbDrawBuffers);
         GL11.glReadBuffer(0);
         checkGLError("FT d");
@@ -2953,7 +2961,7 @@
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_T, GL11.GL_CLAMP);
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MIN_FILTER, GL11.GL_LINEAR);
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, GL11.GL_LINEAR);
-            GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, gbuffersFormat[k], renderWidth, renderHeight, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
+            GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, gbuffersFormat[k], w, h, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
             EXTFramebufferObject.glFramebufferTexture2DEXT(36160, 36064 + k, 3553, dfbColorTexturesA[k], 0);
             checkGLError("FT c");
         }
@@ -2965,10 +2973,10 @@
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_T, GL11.GL_CLAMP);
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MIN_FILTER, GL11.GL_LINEAR);
             GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, GL11.GL_LINEAR);
-            GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, gbuffersFormat[l], renderWidth, renderHeight, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
+            GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, gbuffersFormat[l], w, h, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
             checkGLError("FT ca");
         }
-
+        
         int i1 = EXTFramebufferObject.glCheckFramebufferStatusEXT(36160);
 
         if (i1 == 36058)
@@ -2978,7 +2986,7 @@
             for (int j = 0; j < usedColorBuffers; ++j)
             {
                 GlStateManager.bindTexture(dfbColorTextures.get(j));
-                GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_RGBA, renderWidth, renderHeight, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
+                GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_RGBA, w, h, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
                 EXTFramebufferObject.glFramebufferTexture2DEXT(36160, 36064 + j, 3553, dfbColorTextures.get(j), 0);
                 checkGLError("FT c");
             }
@@ -2999,23 +3007,25 @@
         }
         else
         {
-            SMCLog.info("Framebuffer created.");
+            SMCLog.info("Framebuffer created." + id + ": " + w + " " + h + " " + i1);
         }
+        return id;
+        
     }
 
-    private static void setupShadowFrameBuffer()
+    private static int setupShadowFrameBuffer(int id, int w, int h)
     {
         if (usedShadowDepthBuffers != 0)
         {
-            if (sfb != 0)
+            if (id != 0)
             {
-                EXTFramebufferObject.glDeleteFramebuffersEXT(sfb);
+                EXTFramebufferObject.glDeleteFramebuffersEXT(id);
                 GlStateManager.deleteTextures(sfbDepthTextures);
                 GlStateManager.deleteTextures(sfbColorTextures);
             }
 
-            sfb = EXTFramebufferObject.glGenFramebuffersEXT();
-            EXTFramebufferObject.glBindFramebufferEXT(36160, sfb);
+            id = EXTFramebufferObject.glGenFramebuffersEXT();
+            EXTFramebufferObject.glBindFramebufferEXT(36160, id);
             GL11.glDrawBuffer(0);
             GL11.glReadBuffer(0);
             GL11.glGenTextures((IntBuffer)sfbDepthTextures.clear().limit(usedShadowDepthBuffers));
@@ -3037,7 +3047,7 @@
                     GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL14.GL_TEXTURE_COMPARE_MODE, GL14.GL_COMPARE_R_TO_TEXTURE);
                 }
 
-                GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_DEPTH_COMPONENT, shadowMapWidth, shadowMapHeight, 0, GL11.GL_DEPTH_COMPONENT, GL11.GL_FLOAT, (FloatBuffer)((FloatBuffer)null));
+                GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_DEPTH_COMPONENT, w, h, 0, GL11.GL_DEPTH_COMPONENT, GL11.GL_FLOAT, (FloatBuffer)((FloatBuffer)null));
             }
 
             EXTFramebufferObject.glFramebufferTexture2DEXT(36160, 36096, 3553, sfbDepthTextures.get(0), 0);
@@ -3051,7 +3061,7 @@
                 int i1 = shadowColorFilterNearest[k] ? 9728 : 9729;
                 GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MIN_FILTER, i1);
                 GL11.glTexParameteri(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, i1);
-                GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_RGBA, shadowMapWidth, shadowMapHeight, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
+                GL11.glTexImage2D(GL11.GL_TEXTURE_2D, 0, GL11.GL_RGBA, w, h, 0, GL12.GL_BGRA, GL12.GL_UNSIGNED_INT_8_8_8_8_REV, (ByteBuffer)((ByteBuffer)null));
                 EXTFramebufferObject.glFramebufferTexture2DEXT(36160, 36064 + k, 3553, sfbColorTextures.get(k), 0);
                 checkGLError("FT sc");
             }
@@ -3074,6 +3084,8 @@
                 SMCLog.info("Shadow framebuffer created.");
             }
         }
+        
+        return id;
     }
 
     public static void beginRender(Minecraft minecraft, float partialTicks, long finishTimeNano)
@@ -3102,10 +3114,23 @@
             }
         }
 
-        if (mc.displayWidth != renderDisplayWidth || mc.displayHeight != renderDisplayHeight)
-        {
-            resize();
-        }
+        if(mc.reinitflag || (mc.displayFBWidth != renderDisplayWidth || mc.displayFBHeight != renderDisplayHeight)){
+        	resize();
+        	mc.reinitflag = false;
+        }
+        
+        //Vive use right sizes and buffers
+//        if(mc.currentPass == renderPass.Center){
+//        	dfb = dfbU;
+//        	renderWidth = Math.round(mc.mirrorFBWidth * configRenderResMul);
+//        	renderHeight = Math.round(mc.mirrorFBHeight * configRenderResMul);
+//        }else{
+        //NOPE!
+//        	dfb = dfbR;
+//        	renderWidth = Math.round(mc.displayFBWidth * configRenderResMul);
+//        	renderHeight = Math.round(mc.displayFBHeight * configRenderResMul);
+//        }
+        
 
         if (needResizeShadow)
         {
@@ -3145,8 +3170,9 @@
         float f2 = (float)Math.exp(Math.log(0.5D) * (double)f1 / (double)eyeBrightnessHalflife);
         eyeBrightnessFadeX = eyeBrightnessFadeX * f2 + (float)(eyeBrightness & 65535) * (1.0F - f2);
         eyeBrightnessFadeY = eyeBrightnessFadeY * f2 + (float)(eyeBrightness >> 16) * (1.0F - f2);
-        isEyeInWater = mc.gameSettings.thirdPersonView == 0 && !isSleeping && mc.player.isInsideOfMaterial(Material.WATER) ? 1 : 0;
-        Vec3d vec3d = mc.world.getSkyColor(mc.getRenderViewEntity(), partialTicks);
+        //VIVE use my method its better.
+        isEyeInWater = mc.gameSettings.thirdPersonView == 0 && !isSleeping && mc.getItemRenderer().isInsideOfMaterial(mc.entityRenderer.getEyeRenderPos(mc.currentPass),Material.WATER) ? 1 : 0;
+        Vec3d vec3d = mc.world.getSkyColor(entity, partialTicks);
         skyColorR = (float)vec3d.xCoord;
         skyColorG = (float)vec3d.yCoord;
         skyColorB = (float)vec3d.zCoord;
@@ -3154,6 +3180,7 @@
         isCompositeRendered = false;
         isHandRendered = false;
 
+                
         if (usedShadowDepthBuffers >= 1)
         {
             GlStateManager.setActiveTexture(33988);
@@ -3245,16 +3272,15 @@
         checkGLError("end beginRender");
     }
 
-    private static void checkWorldChanged(World worldin)
+    private static void checkWorldChanged(World world)
     {
-        if (currentWorld != worldin)
+        if (currentWorld != world)
         {
-            currentWorld = worldin;
-
-            if (currentWorld != null && worldin != null)
+ 
+            if (world != null && currentWorld != null)
             {
-                int i = worldin.provider.getDimensionType().getId();
-                int j = worldin.provider.getDimensionType().getId();
+                int i = world.provider.getDimensionType().getId();
+                int j = currentWorld.provider.getDimensionType().getId();
                 boolean flag = shaderPackDimensions.contains(Integer.valueOf(i));
                 boolean flag1 = shaderPackDimensions.contains(Integer.valueOf(j));
 
@@ -3263,6 +3289,7 @@
                     uninit();
                 }
             }
+            currentWorld = world;
         }
     }
 
@@ -3271,7 +3298,7 @@
         if (!isShadowPass)
         {
             EXTFramebufferObject.glBindFramebufferEXT(36160, dfb);
-            GL11.glViewport(0, 0, renderWidth, renderHeight);
+            GlStateManager.viewport(0, 0, renderWidth, renderHeight);
             activeDrawBuffers = null;
             ShadersTex.bindNSTextures(defaultTexture.getMultiTexID());
             useProgram(2);
@@ -3285,12 +3312,12 @@
 
         if (isShadowPass)
         {
-            GL11.glViewport(0, 0, shadowMapWidth, shadowMapHeight);
+        	GlStateManager.viewport(0, 0, shadowMapWidth, shadowMapHeight);
         }
         else
         {
-            GL11.glViewport(0, 0, renderWidth, renderHeight);
             EXTFramebufferObject.glBindFramebufferEXT(36160, dfb);
+        	GL11.glViewport(vx, vy, vw, vh);
             isRenderingDfb = true;
             GlStateManager.enableCull();
             GlStateManager.enableDepth();
@@ -3327,7 +3354,7 @@
         {
             checkGLError("shadow clear pre");
             EXTFramebufferObject.glFramebufferTexture2DEXT(36160, 36096, 3553, sfbDepthTextures.get(0), 0);
-            GL11.glClearColor(1.0F, 1.0F, 1.0F, 1.0F);
+            GlStateManager.clearColor(1.0F, 1.0F, 1.0F, 1.0F);
             GL20.glDrawBuffers(programsDrawBuffers[30]);
             checkFramebufferStatus("shadow clear");
             GL11.glClear(GL11.GL_COLOR_BUFFER_BIT | GL11.GL_DEPTH_BUFFER_BIT);
@@ -3337,16 +3364,16 @@
         {
             checkGLError("clear pre");
             GL20.glDrawBuffers(36064);
-            GL11.glClear(16384);
+            GlStateManager.clear(16384);
             GL20.glDrawBuffers(36065);
-            GL11.glClearColor(1.0F, 1.0F, 1.0F, 1.0F);
-            GL11.glClear(16384);
+            GlStateManager.clearColor(1.0F, 1.0F, 1.0F, 1.0F);
+            GlStateManager.clear(16384);
 
             for (int i = 2; i < usedColorBuffers; ++i)
             {
                 GL20.glDrawBuffers(36064 + i);
-                GL11.glClearColor(0.0F, 0.0F, 0.0F, 0.0F);
-                GL11.glClear(16384);
+                GlStateManager.clearColor(0.0F, 0.0F, 0.0F, 0.0F);
+                GlStateManager.clear(16384);
             }
 
             setDrawBuffers(dfbDrawBuffers);
@@ -3523,7 +3550,7 @@
 
     public static void drawComposite()
     {
-        GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
+        GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
         GL11.glBegin(GL11.GL_QUADS);
         GL11.glTexCoord2f(0.0F, 0.0F);
         GL11.glVertex3f(0.0F, 0.0F, 0.0F);
@@ -3547,7 +3574,7 @@
             GL11.glPushMatrix();
             GL11.glLoadIdentity();
             GL11.glOrtho(0.0D, 1.0D, 0.0D, 1.0D, 0.0D, 1.0D);
-            GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
+            GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
             GlStateManager.enableTexture2D();
             GlStateManager.disableAlpha();
             GlStateManager.disableBlend();
@@ -3651,7 +3678,7 @@
             isRenderingDfb = false;
             mc.getFramebuffer().bindFramebuffer(true);
             OpenGlHelper.glFramebufferTexture2D(OpenGlHelper.GL_FRAMEBUFFER, OpenGlHelper.GL_COLOR_ATTACHMENT0, 3553, mc.getFramebuffer().framebufferTexture, 0);
-            GL11.glViewport(0, 0, mc.displayWidth, mc.displayHeight);
+            // the bind does this for you.  GlStateManager.viewport(0, 0, renderWidth, renderHeight);
 
             if (EntityRenderer.anaglyphEnable)
             {
@@ -3660,9 +3687,9 @@
             }
 
             GlStateManager.depthMask(true);
-            GL11.glClearColor(clearColorR, clearColorG, clearColorB, 1.0F);
-            GL11.glClear(GL11.GL_COLOR_BUFFER_BIT | GL11.GL_DEPTH_BUFFER_BIT);
-            GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
+            GlStateManager.clearColor(clearColorR, clearColorG, clearColorB, 1.0F);
+            GlStateManager.clear(GL11.GL_COLOR_BUFFER_BIT | GL11.GL_DEPTH_BUFFER_BIT);
+            GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
             GlStateManager.enableTexture2D();
             GlStateManager.disableAlpha();
             GlStateManager.disableBlend();
@@ -3780,9 +3807,9 @@
 
     public static void preSkyList()
     {
-        GL11.glColor3f(fogColorR, fogColorG, fogColorB);
+        GlStateManager.color(fogColorR, fogColorG, fogColorB);
         drawHorizon();
-        GL11.glColor3f(skyColorR, skyColorG, skyColorB);
+        GlStateManager.color(skyColorR, skyColorG, skyColorB);
     }
 
     public static void endSky()
@@ -4116,7 +4143,8 @@
     public static void endFPOverlay()
     {
     }
-
+    
+//unused
     public static void glEnableWrapper(int cap)
     {
         GL11.glEnable(cap);
@@ -4131,6 +4159,7 @@
         }
     }
 
+    //unused
     public static void glDisableWrapper(int cap)
     {
         GL11.glDisable(cap);
@@ -4384,7 +4413,7 @@
 
                         for (Object s7 : properties.keySet())
                         {
-                        	String s4 = (String) s7;
+                        	String s4 = (String)s7;
                             String s5 = properties.getProperty(s4);
                             shaderPackResources.put(s4, s5);
                         }
