--- a/net/minecraft/block/BlockEnderChest.java
+++ b/net/minecraft/block/BlockEnderChest.java
@@ -84,7 +84,7 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        return this.func_176223_P().func_177226_a(field_176437_a, p_180642_8_.func_174811_aO().func_176734_d());
+        return this.getBlockState().func_177226_a(field_176437_a, p_180642_8_.func_174811_aO().func_176734_d());
     }
 
     public void func_180633_a(World p_180633_1_, BlockPos p_180633_2_, IBlockState p_180633_3_, EntityLivingBase p_180633_4_, ItemStack p_180633_5_)
@@ -99,7 +99,7 @@
 
         if (inventoryenderchest != null && tileentity instanceof TileEntityEnderChest)
         {
-            if (p_180639_1_.func_180495_p(p_180639_2_.func_177984_a()).func_185915_l())
+            if (p_180639_1_.getBlockState(p_180639_2_.getBlockPosAbove()).func_185915_l())
             {
                 return true;
             }
@@ -135,13 +135,13 @@
         {
             int j = p_180655_4_.nextInt(2) * 2 - 1;
             int k = p_180655_4_.nextInt(2) * 2 - 1;
-            double d0 = (double)p_180655_3_.func_177958_n() + 0.5D + 0.25D * (double)j;
-            double d1 = (double)((float)p_180655_3_.func_177956_o() + p_180655_4_.nextFloat());
-            double d2 = (double)p_180655_3_.func_177952_p() + 0.5D + 0.25D * (double)k;
+            double d0 = (double)p_180655_3_.getX() + 0.5D + 0.25D * (double)j;
+            double d1 = (double)((float)p_180655_3_.getY() + p_180655_4_.nextFloat());
+            double d2 = (double)p_180655_3_.getZ() + 0.5D + 0.25D * (double)k;
             double d3 = (double)(p_180655_4_.nextFloat() * (float)j);
             double d4 = ((double)p_180655_4_.nextFloat() - 0.5D) * 0.125D;
             double d5 = (double)(p_180655_4_.nextFloat() * (float)k);
-            p_180655_2_.func_175688_a(EnumParticleTypes.PORTAL, d0, d1, d2, d3, d4, d5, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.PORTAL, d0, d1, d2, d3, d4, d5, new int[0]);
         }
     }
 
@@ -154,7 +154,7 @@
             enumfacing = EnumFacing.NORTH;
         }
 
-        return this.func_176223_P().func_177226_a(field_176437_a, enumfacing);
+        return this.getBlockState().func_177226_a(field_176437_a, enumfacing);
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
