--- a/net/minecraft/block/BlockPurpurSlab.java
+++ b/net/minecraft/block/BlockPurpurSlab.java
@@ -46,7 +46,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        IBlockState iblockstate = this.func_176223_P().func_177226_a(field_185678_d, BlockPurpurSlab.Variant.DEFAULT);
+        IBlockState iblockstate = this.getBlockState().func_177226_a(field_185678_d, BlockPurpurSlab.Variant.DEFAULT);
 
         if (!this.func_176552_j())
         {
