--- a/net/minecraft/block/BlockRedstoneOre.java
+++ b/net/minecraft/block/BlockRedstoneOre.java
@@ -64,7 +64,7 @@
 
         if (this == Blocks.redstone_ore)
         {
-            p_176352_1_.func_175656_a(p_176352_2_, Blocks.lit_redstone_ore.func_176223_P());
+            p_176352_1_.func_175656_a(p_176352_2_, Blocks.lit_redstone_ore.getBlockState());
         }
     }
 
@@ -72,7 +72,7 @@
     {
         if (this == Blocks.lit_redstone_ore)
         {
-            p_180650_1_.func_175656_a(p_180650_2_, Blocks.redstone_ore.func_176223_P());
+            p_180650_1_.func_175656_a(p_180650_2_, Blocks.redstone_ore.getBlockState());
         }
     }
 
@@ -124,43 +124,43 @@
 
         for (int i = 0; i < 6; ++i)
         {
-            double d1 = (double)((float)p_180691_2_.func_177958_n() + random.nextFloat());
-            double d2 = (double)((float)p_180691_2_.func_177956_o() + random.nextFloat());
-            double d3 = (double)((float)p_180691_2_.func_177952_p() + random.nextFloat());
+            double d1 = (double)((float)p_180691_2_.getX() + random.nextFloat());
+            double d2 = (double)((float)p_180691_2_.getY() + random.nextFloat());
+            double d3 = (double)((float)p_180691_2_.getZ() + random.nextFloat());
 
-            if (i == 0 && !p_180691_1_.func_180495_p(p_180691_2_.func_177984_a()).func_185914_p())
+            if (i == 0 && !p_180691_1_.getBlockState(p_180691_2_.getBlockPosAbove()).func_185914_p())
             {
-                d2 = (double)p_180691_2_.func_177956_o() + 0.0625D + 1.0D;
+                d2 = (double)p_180691_2_.getY() + 0.0625D + 1.0D;
             }
 
-            if (i == 1 && !p_180691_1_.func_180495_p(p_180691_2_.func_177977_b()).func_185914_p())
+            if (i == 1 && !p_180691_1_.getBlockState(p_180691_2_.getBlockPosBelow()).func_185914_p())
             {
-                d2 = (double)p_180691_2_.func_177956_o() - 0.0625D;
+                d2 = (double)p_180691_2_.getY() - 0.0625D;
             }
 
-            if (i == 2 && !p_180691_1_.func_180495_p(p_180691_2_.func_177968_d()).func_185914_p())
+            if (i == 2 && !p_180691_1_.getBlockState(p_180691_2_.getBlockPosSouth()).func_185914_p())
             {
-                d3 = (double)p_180691_2_.func_177952_p() + 0.0625D + 1.0D;
+                d3 = (double)p_180691_2_.getZ() + 0.0625D + 1.0D;
             }
 
-            if (i == 3 && !p_180691_1_.func_180495_p(p_180691_2_.func_177978_c()).func_185914_p())
+            if (i == 3 && !p_180691_1_.getBlockState(p_180691_2_.getBlockPosNorth()).func_185914_p())
             {
-                d3 = (double)p_180691_2_.func_177952_p() - 0.0625D;
+                d3 = (double)p_180691_2_.getZ() - 0.0625D;
             }
 
-            if (i == 4 && !p_180691_1_.func_180495_p(p_180691_2_.func_177974_f()).func_185914_p())
+            if (i == 4 && !p_180691_1_.getBlockState(p_180691_2_.getBlockPosEast()).func_185914_p())
             {
-                d1 = (double)p_180691_2_.func_177958_n() + 0.0625D + 1.0D;
+                d1 = (double)p_180691_2_.getX() + 0.0625D + 1.0D;
             }
 
-            if (i == 5 && !p_180691_1_.func_180495_p(p_180691_2_.func_177976_e()).func_185914_p())
+            if (i == 5 && !p_180691_1_.getBlockState(p_180691_2_.getBlockPosWest()).func_185914_p())
             {
-                d1 = (double)p_180691_2_.func_177958_n() - 0.0625D;
+                d1 = (double)p_180691_2_.getX() - 0.0625D;
             }
 
-            if (d1 < (double)p_180691_2_.func_177958_n() || d1 > (double)(p_180691_2_.func_177958_n() + 1) || d2 < 0.0D || d2 > (double)(p_180691_2_.func_177956_o() + 1) || d3 < (double)p_180691_2_.func_177952_p() || d3 > (double)(p_180691_2_.func_177952_p() + 1))
+            if (d1 < (double)p_180691_2_.getX() || d1 > (double)(p_180691_2_.getX() + 1) || d2 < 0.0D || d2 > (double)(p_180691_2_.getY() + 1) || d3 < (double)p_180691_2_.getZ() || d3 > (double)(p_180691_2_.getZ() + 1))
             {
-                p_180691_1_.func_175688_a(EnumParticleTypes.REDSTONE, d1, d2, d3, 0.0D, 0.0D, 0.0D, new int[0]);
+                p_180691_1_.spawnParticle(EnumParticleTypes.REDSTONE, d1, d2, d3, 0.0D, 0.0D, 0.0D, new int[0]);
             }
         }
     }
