--- a/net/minecraft/block/BlockStandingSign.java
+++ b/net/minecraft/block/BlockStandingSign.java
@@ -20,7 +20,7 @@
 
     public void func_189540_a(IBlockState p_189540_1_, World p_189540_2_, BlockPos p_189540_3_, Block p_189540_4_)
     {
-        if (!p_189540_2_.func_180495_p(p_189540_3_.func_177977_b()).func_185904_a().isSolid())
+        if (!p_189540_2_.getBlockState(p_189540_3_.getBlockPosBelow()).getMaterial().isSolid())
         {
             this.func_176226_b(p_189540_2_, p_189540_3_, p_189540_1_, 0);
             p_189540_2_.func_175698_g(p_189540_3_);
@@ -31,7 +31,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176413_a, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176413_a, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
