--- a/net/minecraft/block/BlockBush.java
+++ b/net/minecraft/block/BlockBush.java
@@ -36,12 +36,12 @@
 
     public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
     {
-        return super.func_176196_c(p_176196_1_, p_176196_2_) && this.func_185514_i(p_176196_1_.func_180495_p(p_176196_2_.func_177977_b()));
+        return super.func_176196_c(p_176196_1_, p_176196_2_) && this.func_185514_i(p_176196_1_.getBlockState(p_176196_2_.getBlockPosBelow()));
     }
 
     protected boolean func_185514_i(IBlockState p_185514_1_)
     {
-        return p_185514_1_.func_177230_c() == Blocks.grass || p_185514_1_.func_177230_c() == Blocks.dirt || p_185514_1_.func_177230_c() == Blocks.farmland;
+        return p_185514_1_.getBlock() == Blocks.grass || p_185514_1_.getBlock() == Blocks.dirt || p_185514_1_.getBlock() == Blocks.farmland;
     }
 
     public void func_189540_a(IBlockState p_189540_1_, World p_189540_2_, BlockPos p_189540_3_, Block p_189540_4_)
@@ -60,13 +60,13 @@
         if (!this.func_180671_f(p_176475_1_, p_176475_2_, p_176475_3_))
         {
             this.func_176226_b(p_176475_1_, p_176475_2_, p_176475_3_, 0);
-            p_176475_1_.func_180501_a(p_176475_2_, Blocks.air.func_176223_P(), 3);
+            p_176475_1_.func_180501_a(p_176475_2_, Blocks.air.getBlockState(), 3);
         }
     }
 
     public boolean func_180671_f(World p_180671_1_, BlockPos p_180671_2_, IBlockState p_180671_3_)
     {
-        return this.func_185514_i(p_180671_1_.func_180495_p(p_180671_2_.func_177977_b()));
+        return this.func_185514_i(p_180671_1_.getBlockState(p_180671_2_.getBlockPosBelow()));
     }
 
     public AxisAlignedBB func_185496_a(IBlockState p_185496_1_, IBlockAccess p_185496_2_, BlockPos p_185496_3_)
