--- a/net/minecraft/block/BlockRedstoneTorch.java
+++ b/net/minecraft/block/BlockRedstoneTorch.java
@@ -80,7 +80,7 @@
         {
             for (EnumFacing enumfacing : EnumFacing.values())
             {
-                p_176213_1_.func_175685_c(p_176213_2_.func_177972_a(enumfacing), this);
+                p_176213_1_.func_175685_c(p_176213_2_.getBlockPosRelative(enumfacing), this);
             }
         }
     }
@@ -91,7 +91,7 @@
         {
             for (EnumFacing enumfacing : EnumFacing.values())
             {
-                p_180663_1_.func_175685_c(p_180663_2_.func_177972_a(enumfacing), this);
+                p_180663_1_.func_175685_c(p_180663_2_.getBlockPosRelative(enumfacing), this);
             }
         }
     }
@@ -104,7 +104,7 @@
     private boolean func_176597_g(World p_176597_1_, BlockPos p_176597_2_, IBlockState p_176597_3_)
     {
         EnumFacing enumfacing = ((EnumFacing)p_176597_3_.func_177229_b(field_176596_a)).func_176734_d();
-        return p_176597_1_.func_175709_b(p_176597_2_.func_177972_a(enumfacing), enumfacing);
+        return p_176597_1_.func_175709_b(p_176597_2_.getBlockPosRelative(enumfacing), enumfacing);
     }
 
     public void func_180645_a(World p_180645_1_, BlockPos p_180645_2_, IBlockState p_180645_3_, Random p_180645_4_)
@@ -125,7 +125,7 @@
         {
             if (flag)
             {
-                p_180650_1_.func_180501_a(p_180650_2_, Blocks.unlit_redstone_torch.func_176223_P().func_177226_a(field_176596_a, p_180650_3_.func_177229_b(field_176596_a)), 3);
+                p_180650_1_.func_180501_a(p_180650_2_, Blocks.unlit_redstone_torch.getBlockState().func_177226_a(field_176596_a, p_180650_3_.func_177229_b(field_176596_a)), 3);
 
                 if (this.func_176598_a(p_180650_1_, p_180650_2_, true))
                 {
@@ -133,19 +133,19 @@
 
                     for (int i = 0; i < 5; ++i)
                     {
-                        double d0 = (double)p_180650_2_.func_177958_n() + p_180650_4_.nextDouble() * 0.6D + 0.2D;
-                        double d1 = (double)p_180650_2_.func_177956_o() + p_180650_4_.nextDouble() * 0.6D + 0.2D;
-                        double d2 = (double)p_180650_2_.func_177952_p() + p_180650_4_.nextDouble() * 0.6D + 0.2D;
-                        p_180650_1_.func_175688_a(EnumParticleTypes.SMOKE_NORMAL, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
+                        double d0 = (double)p_180650_2_.getX() + p_180650_4_.nextDouble() * 0.6D + 0.2D;
+                        double d1 = (double)p_180650_2_.getY() + p_180650_4_.nextDouble() * 0.6D + 0.2D;
+                        double d2 = (double)p_180650_2_.getZ() + p_180650_4_.nextDouble() * 0.6D + 0.2D;
+                        p_180650_1_.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
                     }
 
-                    p_180650_1_.func_175684_a(p_180650_2_, p_180650_1_.func_180495_p(p_180650_2_).func_177230_c(), 160);
+                    p_180650_1_.func_175684_a(p_180650_2_, p_180650_1_.getBlockState(p_180650_2_).getBlock(), 160);
                 }
             }
         }
         else if (!flag && !this.func_176598_a(p_180650_1_, p_180650_2_, false))
         {
-            p_180650_1_.func_180501_a(p_180650_2_, Blocks.redstone_torch.func_176223_P().func_177226_a(field_176596_a, p_180650_3_.func_177229_b(field_176596_a)), 3);
+            p_180650_1_.func_180501_a(p_180650_2_, Blocks.redstone_torch.getBlockState().func_177226_a(field_176596_a, p_180650_3_.func_177229_b(field_176596_a)), 3);
         }
     }
 
@@ -183,9 +183,9 @@
     {
         if (this.field_150113_a)
         {
-            double d0 = (double)p_180655_3_.func_177958_n() + 0.5D + (p_180655_4_.nextDouble() - 0.5D) * 0.2D;
-            double d1 = (double)p_180655_3_.func_177956_o() + 0.7D + (p_180655_4_.nextDouble() - 0.5D) * 0.2D;
-            double d2 = (double)p_180655_3_.func_177952_p() + 0.5D + (p_180655_4_.nextDouble() - 0.5D) * 0.2D;
+            double d0 = (double)p_180655_3_.getX() + 0.5D + (p_180655_4_.nextDouble() - 0.5D) * 0.2D;
+            double d1 = (double)p_180655_3_.getY() + 0.7D + (p_180655_4_.nextDouble() - 0.5D) * 0.2D;
+            double d2 = (double)p_180655_3_.getZ() + 0.5D + (p_180655_4_.nextDouble() - 0.5D) * 0.2D;
             EnumFacing enumfacing = (EnumFacing)p_180655_1_.func_177229_b(field_176596_a);
 
             if (enumfacing.func_176740_k().func_176722_c())
@@ -197,7 +197,7 @@
                 d2 += 0.27D * (double)enumfacing1.getFrontOffsetZ();
             }
 
-            p_180655_2_.func_175688_a(EnumParticleTypes.REDSTONE, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.REDSTONE, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
         }
     }
 
