--- a/net/minecraft/block/BlockDoublePlant.java
+++ b/net/minecraft/block/BlockDoublePlant.java
@@ -46,7 +46,7 @@
 
     private BlockDoublePlant.EnumPlantType func_185517_a(IBlockAccess p_185517_1_, BlockPos p_185517_2_, IBlockState p_185517_3_)
     {
-        if (p_185517_3_.func_177230_c() == this)
+        if (p_185517_3_.getBlock() == this)
         {
             p_185517_3_ = p_185517_3_.func_185899_b(p_185517_1_, p_185517_2_);
             return (BlockDoublePlant.EnumPlantType)p_185517_3_.func_177229_b(field_176493_a);
@@ -59,14 +59,14 @@
 
     public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
     {
-        return super.func_176196_c(p_176196_1_, p_176196_2_) && p_176196_1_.func_175623_d(p_176196_2_.func_177984_a());
+        return super.func_176196_c(p_176196_1_, p_176196_2_) && p_176196_1_.isEmpty(p_176196_2_.getBlockPosAbove());
     }
 
     public boolean func_176200_f(IBlockAccess p_176200_1_, BlockPos p_176200_2_)
     {
-        IBlockState iblockstate = p_176200_1_.func_180495_p(p_176200_2_);
+        IBlockState iblockstate = p_176200_1_.getBlockState(p_176200_2_);
 
-        if (iblockstate.func_177230_c() != this)
+        if (iblockstate.getBlock() != this)
         {
             return true;
         }
@@ -82,19 +82,19 @@
         if (!this.func_180671_f(p_176475_1_, p_176475_2_, p_176475_3_))
         {
             boolean flag = p_176475_3_.func_177229_b(field_176492_b) == BlockDoublePlant.EnumBlockHalf.UPPER;
-            BlockPos blockpos = flag ? p_176475_2_ : p_176475_2_.func_177984_a();
-            BlockPos blockpos1 = flag ? p_176475_2_.func_177977_b() : p_176475_2_;
-            Block block = (Block)(flag ? this : p_176475_1_.func_180495_p(blockpos).func_177230_c());
-            Block block1 = (Block)(flag ? p_176475_1_.func_180495_p(blockpos1).func_177230_c() : this);
+            BlockPos blockpos = flag ? p_176475_2_ : p_176475_2_.getBlockPosAbove();
+            BlockPos blockpos1 = flag ? p_176475_2_.getBlockPosBelow() : p_176475_2_;
+            Block block = (Block)(flag ? this : p_176475_1_.getBlockState(blockpos).getBlock());
+            Block block1 = (Block)(flag ? p_176475_1_.getBlockState(blockpos1).getBlock() : this);
 
             if (block == this)
             {
-                p_176475_1_.func_180501_a(blockpos, Blocks.air.func_176223_P(), 2);
+                p_176475_1_.func_180501_a(blockpos, Blocks.air.getBlockState(), 2);
             }
 
             if (block1 == this)
             {
-                p_176475_1_.func_180501_a(blockpos1, Blocks.air.func_176223_P(), 3);
+                p_176475_1_.func_180501_a(blockpos1, Blocks.air.getBlockState(), 3);
 
                 if (!flag)
                 {
@@ -108,12 +108,12 @@
     {
         if (p_180671_3_.func_177229_b(field_176492_b) == BlockDoublePlant.EnumBlockHalf.UPPER)
         {
-            return p_180671_1_.func_180495_p(p_180671_2_.func_177977_b()).func_177230_c() == this;
+            return p_180671_1_.getBlockState(p_180671_2_.getBlockPosBelow()).getBlock() == this;
         }
         else
         {
-            IBlockState iblockstate = p_180671_1_.func_180495_p(p_180671_2_.func_177984_a());
-            return iblockstate.func_177230_c() == this && super.func_180671_f(p_180671_1_, p_180671_2_, iblockstate);
+            IBlockState iblockstate = p_180671_1_.getBlockState(p_180671_2_.getBlockPosAbove());
+            return iblockstate.getBlock() == this && super.func_180671_f(p_180671_1_, p_180671_2_, iblockstate);
         }
     }
 
@@ -138,13 +138,13 @@
 
     public void func_176491_a(World p_176491_1_, BlockPos p_176491_2_, BlockDoublePlant.EnumPlantType p_176491_3_, int p_176491_4_)
     {
-        p_176491_1_.func_180501_a(p_176491_2_, this.func_176223_P().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.LOWER).func_177226_a(field_176493_a, p_176491_3_), p_176491_4_);
-        p_176491_1_.func_180501_a(p_176491_2_.func_177984_a(), this.func_176223_P().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.UPPER), p_176491_4_);
+        p_176491_1_.func_180501_a(p_176491_2_, this.getBlockState().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.LOWER).func_177226_a(field_176493_a, p_176491_3_), p_176491_4_);
+        p_176491_1_.func_180501_a(p_176491_2_.getBlockPosAbove(), this.getBlockState().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.UPPER), p_176491_4_);
     }
 
     public void func_180633_a(World p_180633_1_, BlockPos p_180633_2_, IBlockState p_180633_3_, EntityLivingBase p_180633_4_, ItemStack p_180633_5_)
     {
-        p_180633_1_.func_180501_a(p_180633_2_.func_177984_a(), this.func_176223_P().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.UPPER), 2);
+        p_180633_1_.func_180501_a(p_180633_2_.getBlockPosAbove(), this.getBlockState().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.UPPER), 2);
     }
 
     public void func_180657_a(World p_180657_1_, EntityPlayer p_180657_2_, BlockPos p_180657_3_, IBlockState p_180657_4_, @Nullable TileEntity p_180657_5_, @Nullable ItemStack p_180657_6_)
@@ -159,40 +159,40 @@
     {
         if (p_176208_3_.func_177229_b(field_176492_b) == BlockDoublePlant.EnumBlockHalf.UPPER)
         {
-            if (p_176208_1_.func_180495_p(p_176208_2_.func_177977_b()).func_177230_c() == this)
+            if (p_176208_1_.getBlockState(p_176208_2_.getBlockPosBelow()).getBlock() == this)
             {
                 if (p_176208_4_.capabilities.isCreativeMode)
                 {
-                    p_176208_1_.func_175698_g(p_176208_2_.func_177977_b());
+                    p_176208_1_.func_175698_g(p_176208_2_.getBlockPosBelow());
                 }
                 else
                 {
-                    IBlockState iblockstate = p_176208_1_.func_180495_p(p_176208_2_.func_177977_b());
+                    IBlockState iblockstate = p_176208_1_.getBlockState(p_176208_2_.getBlockPosBelow());
                     BlockDoublePlant.EnumPlantType blockdoubleplant$enumplanttype = (BlockDoublePlant.EnumPlantType)iblockstate.func_177229_b(field_176493_a);
 
                     if (blockdoubleplant$enumplanttype != BlockDoublePlant.EnumPlantType.FERN && blockdoubleplant$enumplanttype != BlockDoublePlant.EnumPlantType.GRASS)
                     {
-                        p_176208_1_.func_175655_b(p_176208_2_.func_177977_b(), true);
+                        p_176208_1_.func_175655_b(p_176208_2_.getBlockPosBelow(), true);
                     }
                     else if (p_176208_1_.isRemote)
                     {
-                        p_176208_1_.func_175698_g(p_176208_2_.func_177977_b());
+                        p_176208_1_.func_175698_g(p_176208_2_.getBlockPosBelow());
                     }
-                    else if (p_176208_4_.func_184614_ca() != null && p_176208_4_.func_184614_ca().getItem() == Items.shears)
+                    else if (p_176208_4_.getMainHandItem() != null && p_176208_4_.getMainHandItem().getItem() == Items.shears)
                     {
                         this.func_176489_b(p_176208_1_, p_176208_2_, iblockstate, p_176208_4_);
-                        p_176208_1_.func_175698_g(p_176208_2_.func_177977_b());
+                        p_176208_1_.func_175698_g(p_176208_2_.getBlockPosBelow());
                     }
                     else
                     {
-                        p_176208_1_.func_175655_b(p_176208_2_.func_177977_b(), true);
+                        p_176208_1_.func_175655_b(p_176208_2_.getBlockPosBelow(), true);
                     }
                 }
             }
         }
-        else if (p_176208_1_.func_180495_p(p_176208_2_.func_177984_a()).func_177230_c() == this)
+        else if (p_176208_1_.getBlockState(p_176208_2_.getBlockPosAbove()).getBlock() == this)
         {
-            p_176208_1_.func_180501_a(p_176208_2_.func_177984_a(), Blocks.air.func_176223_P(), 2);
+            p_176208_1_.func_180501_a(p_176208_2_.getBlockPosAbove(), Blocks.air.getBlockState(), 2);
         }
 
         super.func_176208_a(p_176208_1_, p_176208_2_, p_176208_3_, p_176208_4_);
@@ -249,16 +249,16 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return (p_176203_1_ & 8) > 0 ? this.func_176223_P().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.UPPER) : this.func_176223_P().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.LOWER).func_177226_a(field_176493_a, BlockDoublePlant.EnumPlantType.func_176938_a(p_176203_1_ & 7));
+        return (p_176203_1_ & 8) > 0 ? this.getBlockState().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.UPPER) : this.getBlockState().func_177226_a(field_176492_b, BlockDoublePlant.EnumBlockHalf.LOWER).func_177226_a(field_176493_a, BlockDoublePlant.EnumPlantType.func_176938_a(p_176203_1_ & 7));
     }
 
     public IBlockState func_176221_a(IBlockState p_176221_1_, IBlockAccess p_176221_2_, BlockPos p_176221_3_)
     {
         if (p_176221_1_.func_177229_b(field_176492_b) == BlockDoublePlant.EnumBlockHalf.UPPER)
         {
-            IBlockState iblockstate = p_176221_2_.func_180495_p(p_176221_3_.func_177977_b());
+            IBlockState iblockstate = p_176221_2_.getBlockState(p_176221_3_.getBlockPosBelow());
 
-            if (iblockstate.func_177230_c() == this)
+            if (iblockstate.getBlock() == this)
             {
                 p_176221_1_ = p_176221_1_.func_177226_a(field_176493_a, iblockstate.func_177229_b(field_176493_a));
             }
