--- a/net/minecraft/block/BlockQuartz.java
+++ b/net/minecraft/block/BlockQuartz.java
@@ -35,17 +35,17 @@
             switch (p_180642_3_.func_176740_k())
             {
                 case Z:
-                    return this.func_176223_P().func_177226_a(field_176335_a, BlockQuartz.EnumType.LINES_Z);
+                    return this.getBlockState().func_177226_a(field_176335_a, BlockQuartz.EnumType.LINES_Z);
 
                 case X:
-                    return this.func_176223_P().func_177226_a(field_176335_a, BlockQuartz.EnumType.LINES_X);
+                    return this.getBlockState().func_177226_a(field_176335_a, BlockQuartz.EnumType.LINES_X);
 
                 case Y:
-                    return this.func_176223_P().func_177226_a(field_176335_a, BlockQuartz.EnumType.LINES_Y);
+                    return this.getBlockState().func_177226_a(field_176335_a, BlockQuartz.EnumType.LINES_Y);
             }
         }
 
-        return p_180642_7_ == BlockQuartz.EnumType.CHISELED.func_176796_a() ? this.func_176223_P().func_177226_a(field_176335_a, BlockQuartz.EnumType.CHISELED) : this.func_176223_P().func_177226_a(field_176335_a, BlockQuartz.EnumType.DEFAULT);
+        return p_180642_7_ == BlockQuartz.EnumType.CHISELED.func_176796_a() ? this.getBlockState().func_177226_a(field_176335_a, BlockQuartz.EnumType.CHISELED) : this.getBlockState().func_177226_a(field_176335_a, BlockQuartz.EnumType.DEFAULT);
     }
 
     public int func_180651_a(IBlockState p_180651_1_)
@@ -77,7 +77,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176335_a, BlockQuartz.EnumType.func_176794_a(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176335_a, BlockQuartz.EnumType.func_176794_a(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
