--- a/net/minecraft/block/BlockAnvil.java
+++ b/net/minecraft/block/BlockAnvil.java
@@ -129,7 +129,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176506_a, EnumFacing.func_176731_b(p_176203_1_ & 3)).func_177226_a(field_176505_b, Integer.valueOf((p_176203_1_ & 15) >> 2));
+        return this.getBlockState().func_177226_a(field_176506_a, EnumFacing.func_176731_b(p_176203_1_ & 3)).func_177226_a(field_176505_b, Integer.valueOf((p_176203_1_ & 15) >> 2));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
@@ -142,7 +142,7 @@
 
     public IBlockState func_185499_a(IBlockState p_185499_1_, Rotation p_185499_2_)
     {
-        return p_185499_1_.func_177230_c() != this ? p_185499_1_ : p_185499_1_.func_177226_a(field_176506_a, p_185499_2_.func_185831_a((EnumFacing)p_185499_1_.func_177229_b(field_176506_a)));
+        return p_185499_1_.getBlock() != this ? p_185499_1_ : p_185499_1_.func_177226_a(field_176506_a, p_185499_2_.func_185831_a((EnumFacing)p_185499_1_.func_177229_b(field_176506_a)));
     }
 
     protected BlockStateContainer func_180661_e()
