--- a/net/minecraft/block/BlockTallGrass.java
+++ b/net/minecraft/block/BlockTallGrass.java
@@ -40,7 +40,7 @@
 
     public boolean func_180671_f(World p_180671_1_, BlockPos p_180671_2_, IBlockState p_180671_3_)
     {
-        return this.func_185514_i(p_180671_1_.func_180495_p(p_180671_2_.func_177977_b()));
+        return this.func_185514_i(p_180671_1_.getBlockState(p_180671_2_.getBlockPosBelow()));
     }
 
     public boolean func_176200_f(IBlockAccess p_176200_1_, BlockPos p_176200_2_)
@@ -77,7 +77,7 @@
 
     public ItemStack func_185473_a(World p_185473_1_, BlockPos p_185473_2_, IBlockState p_185473_3_)
     {
-        return new ItemStack(this, 1, p_185473_3_.func_177230_c().func_176201_c(p_185473_3_));
+        return new ItemStack(this, 1, p_185473_3_.getBlock().func_176201_c(p_185473_3_));
     }
 
     /**
@@ -118,7 +118,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176497_a, BlockTallGrass.EnumType.func_177045_a(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176497_a, BlockTallGrass.EnumType.func_177045_a(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
