--- a/net/minecraft/block/BlockGrassPath.java
+++ b/net/minecraft/block/BlockGrassPath.java
@@ -34,8 +34,8 @@
             case SOUTH:
             case WEST:
             case EAST:
-                IBlockState iblockstate = p_176225_2_.func_180495_p(p_176225_3_.func_177972_a(p_176225_4_));
-                Block block = iblockstate.func_177230_c();
+                IBlockState iblockstate = p_176225_2_.getBlockState(p_176225_3_.getBlockPosRelative(p_176225_4_));
+                Block block = iblockstate.getBlock();
                 return !iblockstate.func_185914_p() && block != Blocks.farmland && block != Blocks.field_185774_da;
 
             default:
@@ -61,7 +61,7 @@
     @Nullable
     public Item func_180660_a(IBlockState p_180660_1_, Random p_180660_2_, int p_180660_3_)
     {
-        return Blocks.dirt.func_180660_a(Blocks.dirt.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT), p_180660_2_, p_180660_3_);
+        return Blocks.dirt.func_180660_a(Blocks.dirt.getBlockState().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT), p_180660_2_, p_180660_3_);
     }
 
     public ItemStack func_185473_a(World p_185473_1_, BlockPos p_185473_2_, IBlockState p_185473_3_)
@@ -73,9 +73,9 @@
     {
         super.func_189540_a(p_189540_1_, p_189540_2_, p_189540_3_, p_189540_4_);
 
-        if (p_189540_2_.func_180495_p(p_189540_3_.func_177984_a()).func_185904_a().isSolid())
+        if (p_189540_2_.getBlockState(p_189540_3_.getBlockPosAbove()).getMaterial().isSolid())
         {
-            p_189540_2_.func_175656_a(p_189540_3_, Blocks.dirt.func_176223_P());
+            p_189540_2_.func_175656_a(p_189540_3_, Blocks.dirt.getBlockState());
         }
     }
 }
