--- a/net/minecraft/block/BlockButton.java
+++ b/net/minecraft/block/BlockButton.java
@@ -93,13 +93,13 @@
 
     protected static boolean func_181088_a(World p_181088_0_, BlockPos p_181088_1_, EnumFacing p_181088_2_)
     {
-        BlockPos blockpos = p_181088_1_.func_177972_a(p_181088_2_);
-        return p_181088_2_ == EnumFacing.DOWN ? p_181088_0_.func_180495_p(blockpos).func_185896_q() : p_181088_0_.func_180495_p(blockpos).func_185915_l();
+        BlockPos blockpos = p_181088_1_.getBlockPosRelative(p_181088_2_);
+        return p_181088_2_ == EnumFacing.DOWN ? p_181088_0_.getBlockState(blockpos).func_185896_q() : p_181088_0_.getBlockState(blockpos).func_185915_l();
     }
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        return func_181088_a(p_180642_1_, p_180642_2_, p_180642_3_.func_176734_d()) ? this.func_176223_P().func_177226_a(field_176387_N, p_180642_3_).func_177226_a(field_176584_b, Boolean.valueOf(false)) : this.func_176223_P().func_177226_a(field_176387_N, EnumFacing.DOWN).func_177226_a(field_176584_b, Boolean.valueOf(false));
+        return func_181088_a(p_180642_1_, p_180642_2_, p_180642_3_.func_176734_d()) ? this.getBlockState().func_177226_a(field_176387_N, p_180642_3_).func_177226_a(field_176584_b, Boolean.valueOf(false)) : this.getBlockState().func_177226_a(field_176387_N, EnumFacing.DOWN).func_177226_a(field_176584_b, Boolean.valueOf(false));
     }
 
     public void func_189540_a(IBlockState p_189540_1_, World p_189540_2_, BlockPos p_189540_3_, Block p_189540_4_)
@@ -243,7 +243,7 @@
 
     private void func_185616_e(IBlockState p_185616_1_, World p_185616_2_, BlockPos p_185616_3_)
     {
-        List <? extends Entity > list = p_185616_2_.<Entity>getEntitiesWithinAABB(EntityArrow.class, p_185616_1_.func_185900_c(p_185616_2_, p_185616_3_).func_186670_a(p_185616_3_));
+        List <? extends Entity > list = p_185616_2_.<Entity>getEntitiesWithinAABB(EntityArrow.class, p_185616_1_.func_185900_c(p_185616_2_, p_185616_3_).offset(p_185616_3_));
         boolean flag = !list.isEmpty();
         boolean flag1 = ((Boolean)p_185616_1_.func_177229_b(field_176584_b)).booleanValue();
 
@@ -272,7 +272,7 @@
     private void func_176582_b(World p_176582_1_, BlockPos p_176582_2_, EnumFacing p_176582_3_)
     {
         p_176582_1_.func_175685_c(p_176582_2_, this);
-        p_176582_1_.func_175685_c(p_176582_2_.func_177972_a(p_176582_3_.func_176734_d()), this);
+        p_176582_1_.func_175685_c(p_176582_2_.getBlockPosRelative(p_176582_3_.func_176734_d()), this);
     }
 
     public IBlockState func_176203_a(int p_176203_1_)
@@ -306,7 +306,7 @@
                 enumfacing = EnumFacing.UP;
         }
 
-        return this.func_176223_P().func_177226_a(field_176387_N, enumfacing).func_177226_a(field_176584_b, Boolean.valueOf((p_176203_1_ & 8) > 0));
+        return this.getBlockState().func_177226_a(field_176387_N, enumfacing).func_177226_a(field_176584_b, Boolean.valueOf((p_176203_1_ & 8) > 0));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
