--- a/net/minecraft/block/BlockMushroom.java
+++ b/net/minecraft/block/BlockMushroom.java
@@ -33,7 +33,7 @@
 
             for (BlockPos blockpos : BlockPos.func_177975_b(p_180650_2_.func_177982_a(-4, -1, -4), p_180650_2_.func_177982_a(4, 1, 4)))
             {
-                if (p_180650_1_.func_180495_p(blockpos).func_177230_c() == this)
+                if (p_180650_1_.getBlockState(blockpos).getBlock() == this)
                 {
                     --i;
 
@@ -48,7 +48,7 @@
 
             for (int k = 0; k < 4; ++k)
             {
-                if (p_180650_1_.func_175623_d(blockpos1) && this.func_180671_f(p_180650_1_, blockpos1, this.func_176223_P()))
+                if (p_180650_1_.isEmpty(blockpos1) && this.func_180671_f(p_180650_1_, blockpos1, this.getBlockState()))
                 {
                     p_180650_2_ = blockpos1;
                 }
@@ -56,16 +56,16 @@
                 blockpos1 = p_180650_2_.func_177982_a(p_180650_4_.nextInt(3) - 1, p_180650_4_.nextInt(2) - p_180650_4_.nextInt(2), p_180650_4_.nextInt(3) - 1);
             }
 
-            if (p_180650_1_.func_175623_d(blockpos1) && this.func_180671_f(p_180650_1_, blockpos1, this.func_176223_P()))
+            if (p_180650_1_.isEmpty(blockpos1) && this.func_180671_f(p_180650_1_, blockpos1, this.getBlockState()))
             {
-                p_180650_1_.func_180501_a(blockpos1, this.func_176223_P(), 2);
+                p_180650_1_.func_180501_a(blockpos1, this.getBlockState(), 2);
             }
         }
     }
 
     public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
     {
-        return super.func_176196_c(p_176196_1_, p_176196_2_) && this.func_180671_f(p_176196_1_, p_176196_2_, this.func_176223_P());
+        return super.func_176196_c(p_176196_1_, p_176196_2_) && this.func_180671_f(p_176196_1_, p_176196_2_, this.getBlockState());
     }
 
     protected boolean func_185514_i(IBlockState p_185514_1_)
@@ -75,10 +75,10 @@
 
     public boolean func_180671_f(World p_180671_1_, BlockPos p_180671_2_, IBlockState p_180671_3_)
     {
-        if (p_180671_2_.func_177956_o() >= 0 && p_180671_2_.func_177956_o() < 256)
+        if (p_180671_2_.getY() >= 0 && p_180671_2_.getY() < 256)
         {
-            IBlockState iblockstate = p_180671_1_.func_180495_p(p_180671_2_.func_177977_b());
-            return iblockstate.func_177230_c() == Blocks.mycelium ? true : (iblockstate.func_177230_c() == Blocks.dirt && iblockstate.func_177229_b(BlockDirt.field_176386_a) == BlockDirt.DirtType.PODZOL ? true : p_180671_1_.func_175699_k(p_180671_2_) < 13 && this.func_185514_i(iblockstate));
+            IBlockState iblockstate = p_180671_1_.getBlockState(p_180671_2_.getBlockPosBelow());
+            return iblockstate.getBlock() == Blocks.mycelium ? true : (iblockstate.getBlock() == Blocks.dirt && iblockstate.func_177229_b(BlockDirt.field_176386_a) == BlockDirt.DirtType.PODZOL ? true : p_180671_1_.func_175699_k(p_180671_2_) < 13 && this.func_185514_i(iblockstate));
         }
         else
         {
