--- a/net/minecraft/block/BlockPane.java
+++ b/net/minecraft/block/BlockPane.java
@@ -104,7 +104,7 @@
 
     public IBlockState func_176221_a(IBlockState p_176221_1_, IBlockAccess p_176221_2_, BlockPos p_176221_3_)
     {
-        return p_176221_1_.func_177226_a(field_176241_b, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.func_180495_p(p_176221_3_.func_177978_c()).func_177230_c()))).func_177226_a(field_176243_N, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.func_180495_p(p_176221_3_.func_177968_d()).func_177230_c()))).func_177226_a(field_176244_O, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.func_180495_p(p_176221_3_.func_177976_e()).func_177230_c()))).func_177226_a(field_176242_M, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.func_180495_p(p_176221_3_.func_177974_f()).func_177230_c())));
+        return p_176221_1_.func_177226_a(field_176241_b, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.getBlockState(p_176221_3_.getBlockPosNorth()).getBlock()))).func_177226_a(field_176243_N, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.getBlockState(p_176221_3_.getBlockPosSouth()).getBlock()))).func_177226_a(field_176244_O, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.getBlockState(p_176221_3_.getBlockPosWest()).getBlock()))).func_177226_a(field_176242_M, Boolean.valueOf(this.canPaneConnectToBlock(p_176221_2_.getBlockState(p_176221_3_.getBlockPosEast()).getBlock())));
     }
 
     @Nullable
@@ -125,12 +125,12 @@
 
     public boolean func_176225_a(IBlockState p_176225_1_, IBlockAccess p_176225_2_, BlockPos p_176225_3_, EnumFacing p_176225_4_)
     {
-        return p_176225_2_.func_180495_p(p_176225_3_.func_177972_a(p_176225_4_)).func_177230_c() == this ? false : super.func_176225_a(p_176225_1_, p_176225_2_, p_176225_3_, p_176225_4_);
+        return p_176225_2_.getBlockState(p_176225_3_.getBlockPosRelative(p_176225_4_)).getBlock() == this ? false : super.func_176225_a(p_176225_1_, p_176225_2_, p_176225_3_, p_176225_4_);
     }
 
     public final boolean canPaneConnectToBlock(Block p_150098_1_)
     {
-        return p_150098_1_.func_176223_P().func_185917_h() || p_150098_1_ == this || p_150098_1_ == Blocks.glass || p_150098_1_ == Blocks.stained_glass || p_150098_1_ == Blocks.stained_glass_pane || p_150098_1_ instanceof BlockPane;
+        return p_150098_1_.getBlockState().func_185917_h() || p_150098_1_ == this || p_150098_1_ == Blocks.glass || p_150098_1_ == Blocks.stained_glass || p_150098_1_ == Blocks.stained_glass_pane || p_150098_1_ instanceof BlockPane;
     }
 
     protected boolean canSilkHarvest()
