--- a/net/minecraft/block/BlockLiquid.java
+++ b/net/minecraft/block/BlockLiquid.java
@@ -66,7 +66,7 @@
 
     protected int func_189542_i(IBlockState p_189542_1_)
     {
-        return p_189542_1_.func_185904_a() == this.blockMaterial ? ((Integer)p_189542_1_.func_177229_b(field_176367_b)).intValue() : -1;
+        return p_189542_1_.getMaterial() == this.blockMaterial ? ((Integer)p_189542_1_.func_177229_b(field_176367_b)).intValue() : -1;
     }
 
     protected int func_189545_x(IBlockState p_189545_1_)
@@ -92,13 +92,13 @@
 
     public boolean func_176212_b(IBlockAccess p_176212_1_, BlockPos p_176212_2_, EnumFacing p_176212_3_)
     {
-        Material material = p_176212_1_.func_180495_p(p_176212_2_).func_185904_a();
+        Material material = p_176212_1_.getBlockState(p_176212_2_).getMaterial();
         return material == this.blockMaterial ? false : (p_176212_3_ == EnumFacing.UP ? true : (material == Material.ice ? false : super.func_176212_b(p_176212_1_, p_176212_2_, p_176212_3_)));
     }
 
     public boolean func_176225_a(IBlockState p_176225_1_, IBlockAccess p_176225_2_, BlockPos p_176225_3_, EnumFacing p_176225_4_)
     {
-        return p_176225_2_.func_180495_p(p_176225_3_.func_177972_a(p_176225_4_)).func_185904_a() == this.blockMaterial ? false : (p_176225_4_ == EnumFacing.UP ? true : super.func_176225_a(p_176225_1_, p_176225_2_, p_176225_3_, p_176225_4_));
+        return p_176225_2_.getBlockState(p_176225_3_.getBlockPosRelative(p_176225_4_)).getMaterial() == this.blockMaterial ? false : (p_176225_4_ == EnumFacing.UP ? true : super.func_176225_a(p_176225_1_, p_176225_2_, p_176225_3_, p_176225_4_));
     }
 
     public boolean func_176364_g(IBlockAccess p_176364_1_, BlockPos p_176364_2_)
@@ -107,9 +107,9 @@
         {
             for (int j = -1; j <= 1; ++j)
             {
-                IBlockState iblockstate = p_176364_1_.func_180495_p(p_176364_2_.func_177982_a(i, 0, j));
+                IBlockState iblockstate = p_176364_1_.getBlockState(p_176364_2_.func_177982_a(i, 0, j));
 
-                if (iblockstate.func_185904_a() != this.blockMaterial && !iblockstate.func_185913_b())
+                if (iblockstate.getMaterial() != this.blockMaterial && !iblockstate.func_185913_b())
                 {
                     return true;
                 }
@@ -152,13 +152,13 @@
         for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
         {
             blockpos$pooledmutableblockpos.func_189533_g(p_189543_2_).func_189536_c(enumfacing);
-            int j = this.func_189545_x(p_189543_1_.func_180495_p(blockpos$pooledmutableblockpos));
+            int j = this.func_189545_x(p_189543_1_.getBlockState(blockpos$pooledmutableblockpos));
 
             if (j < 0)
             {
-                if (!p_189543_1_.func_180495_p(blockpos$pooledmutableblockpos).func_185904_a().blocksMovement())
+                if (!p_189543_1_.getBlockState(blockpos$pooledmutableblockpos).getMaterial().blocksMovement())
                 {
-                    j = this.func_189545_x(p_189543_1_.func_180495_p(blockpos$pooledmutableblockpos.func_177977_b()));
+                    j = this.func_189545_x(p_189543_1_.getBlockState(blockpos$pooledmutableblockpos.getBlockPosBelow()));
 
                     if (j >= 0)
                     {
@@ -186,7 +186,7 @@
             {
                 blockpos$pooledmutableblockpos.func_189533_g(p_189543_2_).func_189536_c(enumfacing1);
 
-                if (this.func_176212_b(p_189543_1_, blockpos$pooledmutableblockpos, enumfacing1) || this.func_176212_b(p_189543_1_, blockpos$pooledmutableblockpos.func_177984_a(), enumfacing1))
+                if (this.func_176212_b(p_189543_1_, blockpos$pooledmutableblockpos, enumfacing1) || this.func_176212_b(p_189543_1_, blockpos$pooledmutableblockpos.getBlockPosAbove(), enumfacing1))
                 {
                     vec3d = vec3d.normalize().addVector(0.0D, -6.0D, 0.0D);
                     break;
@@ -200,7 +200,7 @@
 
     public Vec3d func_176197_a(World p_176197_1_, BlockPos p_176197_2_, Entity p_176197_3_, Vec3d p_176197_4_)
     {
-        return p_176197_4_.func_178787_e(this.func_189543_a(p_176197_1_, p_176197_2_, p_176197_1_.func_180495_p(p_176197_2_)));
+        return p_176197_4_.addVector(this.func_189543_a(p_176197_1_, p_176197_2_, p_176197_1_.getBlockState(p_176197_2_)));
     }
 
     /**
@@ -214,7 +214,7 @@
     public int func_185484_c(IBlockState p_185484_1_, IBlockAccess p_185484_2_, BlockPos p_185484_3_)
     {
         int i = p_185484_2_.func_175626_b(p_185484_3_, 0);
-        int j = p_185484_2_.func_175626_b(p_185484_3_.func_177984_a(), 0);
+        int j = p_185484_2_.func_175626_b(p_185484_3_.getBlockPosAbove(), 0);
         int k = i & 255;
         int l = j & 255;
         int i1 = i >> 16 & 255;
@@ -229,9 +229,9 @@
 
     public void func_180655_c(IBlockState p_180655_1_, World p_180655_2_, BlockPos p_180655_3_, Random p_180655_4_)
     {
-        double d0 = (double)p_180655_3_.func_177958_n();
-        double d1 = (double)p_180655_3_.func_177956_o();
-        double d2 = (double)p_180655_3_.func_177952_p();
+        double d0 = (double)p_180655_3_.getX();
+        double d1 = (double)p_180655_3_.getY();
+        double d2 = (double)p_180655_3_.getZ();
 
         if (this.blockMaterial == Material.water)
         {
@@ -246,18 +246,18 @@
             }
             else if (p_180655_4_.nextInt(10) == 0)
             {
-                p_180655_2_.func_175688_a(EnumParticleTypes.SUSPENDED, d0 + (double)p_180655_4_.nextFloat(), d1 + (double)p_180655_4_.nextFloat(), d2 + (double)p_180655_4_.nextFloat(), 0.0D, 0.0D, 0.0D, new int[0]);
+                p_180655_2_.spawnParticle(EnumParticleTypes.SUSPENDED, d0 + (double)p_180655_4_.nextFloat(), d1 + (double)p_180655_4_.nextFloat(), d2 + (double)p_180655_4_.nextFloat(), 0.0D, 0.0D, 0.0D, new int[0]);
             }
         }
 
-        if (this.blockMaterial == Material.lava && p_180655_2_.func_180495_p(p_180655_3_.func_177984_a()).func_185904_a() == Material.air && !p_180655_2_.func_180495_p(p_180655_3_.func_177984_a()).func_185914_p())
+        if (this.blockMaterial == Material.lava && p_180655_2_.getBlockState(p_180655_3_.getBlockPosAbove()).getMaterial() == Material.air && !p_180655_2_.getBlockState(p_180655_3_.getBlockPosAbove()).func_185914_p())
         {
             if (p_180655_4_.nextInt(100) == 0)
             {
                 double d8 = d0 + (double)p_180655_4_.nextFloat();
                 double d4 = d1 + p_180655_1_.func_185900_c(p_180655_2_, p_180655_3_).maxY;
                 double d6 = d2 + (double)p_180655_4_.nextFloat();
-                p_180655_2_.func_175688_a(EnumParticleTypes.LAVA, d8, d4, d6, 0.0D, 0.0D, 0.0D, new int[0]);
+                p_180655_2_.spawnParticle(EnumParticleTypes.LAVA, d8, d4, d6, 0.0D, 0.0D, 0.0D, new int[0]);
                 p_180655_2_.func_184134_a(d8, d4, d6, SoundEvents.field_187662_cZ, SoundCategory.BLOCKS, 0.2F + p_180655_4_.nextFloat() * 0.2F, 0.9F + p_180655_4_.nextFloat() * 0.15F, false);
             }
 
@@ -267,9 +267,9 @@
             }
         }
 
-        if (p_180655_4_.nextInt(10) == 0 && p_180655_2_.func_180495_p(p_180655_3_.func_177977_b()).func_185896_q())
+        if (p_180655_4_.nextInt(10) == 0 && p_180655_2_.getBlockState(p_180655_3_.getBlockPosBelow()).func_185896_q())
         {
-            Material material = p_180655_2_.func_180495_p(p_180655_3_.func_177979_c(2)).func_185904_a();
+            Material material = p_180655_2_.getBlockState(p_180655_3_.getBlockPosBelow(2)).getMaterial();
 
             if (!material.blocksMovement() && !material.isLiquid())
             {
@@ -279,11 +279,11 @@
 
                 if (this.blockMaterial == Material.water)
                 {
-                    p_180655_2_.func_175688_a(EnumParticleTypes.DRIP_WATER, d3, d5, d7, 0.0D, 0.0D, 0.0D, new int[0]);
+                    p_180655_2_.spawnParticle(EnumParticleTypes.DRIP_WATER, d3, d5, d7, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
                 else
                 {
-                    p_180655_2_.func_175688_a(EnumParticleTypes.DRIP_LAVA, d3, d5, d7, 0.0D, 0.0D, 0.0D, new int[0]);
+                    p_180655_2_.spawnParticle(EnumParticleTypes.DRIP_LAVA, d3, d5, d7, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
             }
         }
@@ -313,7 +313,7 @@
 
             for (EnumFacing enumfacing : EnumFacing.values())
             {
-                if (enumfacing != EnumFacing.DOWN && p_176365_1_.func_180495_p(p_176365_2_.func_177972_a(enumfacing)).func_185904_a() == Material.water)
+                if (enumfacing != EnumFacing.DOWN && p_176365_1_.getBlockState(p_176365_2_.getBlockPosRelative(enumfacing)).getMaterial() == Material.water)
                 {
                     flag = true;
                     break;
@@ -326,14 +326,14 @@
 
                 if (integer.intValue() == 0)
                 {
-                    p_176365_1_.func_175656_a(p_176365_2_, Blocks.obsidian.func_176223_P());
+                    p_176365_1_.func_175656_a(p_176365_2_, Blocks.obsidian.getBlockState());
                     this.func_180688_d(p_176365_1_, p_176365_2_);
                     return true;
                 }
 
                 if (integer.intValue() <= 4)
                 {
-                    p_176365_1_.func_175656_a(p_176365_2_, Blocks.cobblestone.func_176223_P());
+                    p_176365_1_.func_175656_a(p_176365_2_, Blocks.cobblestone.getBlockState());
                     this.func_180688_d(p_176365_1_, p_176365_2_);
                     return true;
                 }
@@ -345,20 +345,20 @@
 
     protected void func_180688_d(World p_180688_1_, BlockPos p_180688_2_)
     {
-        double d0 = (double)p_180688_2_.func_177958_n();
-        double d1 = (double)p_180688_2_.func_177956_o();
-        double d2 = (double)p_180688_2_.func_177952_p();
+        double d0 = (double)p_180688_2_.getX();
+        double d1 = (double)p_180688_2_.getY();
+        double d2 = (double)p_180688_2_.getZ();
         p_180688_1_.func_184133_a((EntityPlayer)null, p_180688_2_, SoundEvents.field_187659_cY, SoundCategory.BLOCKS, 0.5F, 2.6F + (p_180688_1_.rand.nextFloat() - p_180688_1_.rand.nextFloat()) * 0.8F);
 
         for (int i = 0; i < 8; ++i)
         {
-            p_180688_1_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, d0 + Math.random(), d1 + 1.2D, d2 + Math.random(), 0.0D, 0.0D, 0.0D, new int[0]);
+            p_180688_1_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, d0 + Math.random(), d1 + 1.2D, d2 + Math.random(), 0.0D, 0.0D, 0.0D, new int[0]);
         }
     }
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176367_b, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176367_b, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
