--- a/net/minecraft/block/BlockFlower.java
+++ b/net/minecraft/block/BlockFlower.java
@@ -43,7 +43,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(this.func_176494_l(), BlockFlower.EnumFlowerType.func_176967_a(this.func_176495_j(), p_176203_1_));
+        return this.getBlockState().func_177226_a(this.func_176494_l(), BlockFlower.EnumFlowerType.func_176967_a(this.func_176495_j(), p_176203_1_));
     }
 
     public abstract BlockFlower.EnumFlowerColor func_176495_j();
