--- a/net/minecraft/block/BlockEndPortalFrame.java
+++ b/net/minecraft/block/BlockEndPortalFrame.java
@@ -55,7 +55,7 @@
     {
         func_185492_a(p_185477_3_, p_185477_4_, p_185477_5_, field_185662_c);
 
-        if (((Boolean)p_185477_2_.func_180495_p(p_185477_3_).func_177229_b(field_176507_b)).booleanValue())
+        if (((Boolean)p_185477_2_.getBlockState(p_185477_3_).func_177229_b(field_176507_b)).booleanValue())
         {
             func_185492_a(p_185477_3_, p_185477_4_, p_185477_5_, field_185663_d);
         }
@@ -69,7 +69,7 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        return this.func_176223_P().func_177226_a(field_176508_a, p_180642_8_.func_174811_aO().func_176734_d()).func_177226_a(field_176507_b, Boolean.valueOf(false));
+        return this.getBlockState().func_177226_a(field_176508_a, p_180642_8_.func_174811_aO().func_176734_d()).func_177226_a(field_176507_b, Boolean.valueOf(false));
     }
 
     public boolean hasComparatorInputOverride(IBlockState p_149740_1_)
@@ -84,7 +84,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176507_b, Boolean.valueOf((p_176203_1_ & 4) != 0)).func_177226_a(field_176508_a, EnumFacing.func_176731_b(p_176203_1_ & 3));
+        return this.getBlockState().func_177226_a(field_176507_b, Boolean.valueOf((p_176203_1_ & 4) != 0)).func_177226_a(field_176508_a, EnumFacing.func_176731_b(p_176203_1_ & 3));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
