--- a/net/minecraft/block/BlockRotatedPillar.java
+++ b/net/minecraft/block/BlockRotatedPillar.java
@@ -65,7 +65,7 @@
             enumfacing$axis = EnumFacing.Axis.Z;
         }
 
-        return this.func_176223_P().func_177226_a(field_176298_M, enumfacing$axis);
+        return this.getBlockState().func_177226_a(field_176298_M, enumfacing$axis);
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
