--- a/net/minecraft/block/BlockFarmland.java
+++ b/net/minecraft/block/BlockFarmland.java
@@ -57,7 +57,7 @@
     {
         int i = ((Integer)p_180650_3_.func_177229_b(field_176531_a)).intValue();
 
-        if (!this.func_176530_e(p_180650_1_, p_180650_2_) && !p_180650_1_.func_175727_C(p_180650_2_.func_177984_a()))
+        if (!this.func_176530_e(p_180650_1_, p_180650_2_) && !p_180650_1_.func_175727_C(p_180650_2_.getBlockPosAbove()))
         {
             if (i > 0)
             {
@@ -65,7 +65,7 @@
             }
             else if (!this.func_176529_d(p_180650_1_, p_180650_2_))
             {
-                p_180650_1_.func_175656_a(p_180650_2_, Blocks.dirt.func_176223_P());
+                p_180650_1_.func_175656_a(p_180650_2_, Blocks.dirt.getBlockState());
             }
         }
         else if (i < 7)
@@ -78,7 +78,7 @@
     {
         if (!p_180658_1_.isRemote && p_180658_1_.rand.nextFloat() < p_180658_4_ - 0.5F && p_180658_3_ instanceof EntityLivingBase && (p_180658_3_ instanceof EntityPlayer || p_180658_1_.getGameRules().getGameRuleBooleanValue("mobGriefing")) && p_180658_3_.width * p_180658_3_.width * p_180658_3_.height > 0.512F)
         {
-            p_180658_1_.func_175656_a(p_180658_2_, Blocks.dirt.func_176223_P());
+            p_180658_1_.func_175656_a(p_180658_2_, Blocks.dirt.getBlockState());
         }
 
         super.func_180658_a(p_180658_1_, p_180658_2_, p_180658_3_, p_180658_4_);
@@ -86,7 +86,7 @@
 
     private boolean func_176529_d(World p_176529_1_, BlockPos p_176529_2_)
     {
-        Block block = p_176529_1_.func_180495_p(p_176529_2_.func_177984_a()).func_177230_c();
+        Block block = p_176529_1_.getBlockState(p_176529_2_.getBlockPosAbove()).getBlock();
         return block instanceof BlockCrops || block instanceof BlockStem;
     }
 
@@ -94,7 +94,7 @@
     {
         for (BlockPos.MutableBlockPos blockpos$mutableblockpos : BlockPos.func_177975_b(p_176530_2_.func_177982_a(-4, 0, -4), p_176530_2_.func_177982_a(4, 1, 4)))
         {
-            if (p_176530_1_.func_180495_p(blockpos$mutableblockpos).func_185904_a() == Material.water)
+            if (p_176530_1_.getBlockState(blockpos$mutableblockpos).getMaterial() == Material.water)
             {
                 return true;
             }
@@ -107,9 +107,9 @@
     {
         super.func_189540_a(p_189540_1_, p_189540_2_, p_189540_3_, p_189540_4_);
 
-        if (p_189540_2_.func_180495_p(p_189540_3_.func_177984_a()).func_185904_a().isSolid())
+        if (p_189540_2_.getBlockState(p_189540_3_.getBlockPosAbove()).getMaterial().isSolid())
         {
-            p_189540_2_.func_175656_a(p_189540_3_, Blocks.dirt.func_176223_P());
+            p_189540_2_.func_175656_a(p_189540_3_, Blocks.dirt.getBlockState());
         }
     }
 
@@ -124,8 +124,8 @@
             case SOUTH:
             case WEST:
             case EAST:
-                IBlockState iblockstate = p_176225_2_.func_180495_p(p_176225_3_.func_177972_a(p_176225_4_));
-                Block block = iblockstate.func_177230_c();
+                IBlockState iblockstate = p_176225_2_.getBlockState(p_176225_3_.getBlockPosRelative(p_176225_4_));
+                Block block = iblockstate.getBlock();
                 return !iblockstate.func_185914_p() && block != Blocks.farmland && block != Blocks.field_185774_da;
 
             default:
@@ -136,7 +136,7 @@
     @Nullable
     public Item func_180660_a(IBlockState p_180660_1_, Random p_180660_2_, int p_180660_3_)
     {
-        return Blocks.dirt.func_180660_a(Blocks.dirt.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT), p_180660_2_, p_180660_3_);
+        return Blocks.dirt.func_180660_a(Blocks.dirt.getBlockState().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT), p_180660_2_, p_180660_3_);
     }
 
     public ItemStack func_185473_a(World p_185473_1_, BlockPos p_185473_2_, IBlockState p_185473_3_)
@@ -146,7 +146,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176531_a, Integer.valueOf(p_176203_1_ & 7));
+        return this.getBlockState().func_177226_a(field_176531_a, Integer.valueOf(p_176203_1_ & 7));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
