--- a/net/minecraft/block/BlockSourceImpl.java
+++ b/net/minecraft/block/BlockSourceImpl.java
@@ -24,17 +24,17 @@
 
     public double getX()
     {
-        return (double)this.field_179317_b.func_177958_n() + 0.5D;
+        return (double)this.field_179317_b.getX() + 0.5D;
     }
 
     public double getY()
     {
-        return (double)this.field_179317_b.func_177956_o() + 0.5D;
+        return (double)this.field_179317_b.getY() + 0.5D;
     }
 
     public double getZ()
     {
-        return (double)this.field_179317_b.func_177952_p() + 0.5D;
+        return (double)this.field_179317_b.getZ() + 0.5D;
     }
 
     public BlockPos func_180699_d()
@@ -44,7 +44,7 @@
 
     public IBlockState func_189992_e()
     {
-        return this.worldObj.func_180495_p(this.field_179317_b);
+        return this.worldObj.getBlockState(this.field_179317_b);
     }
 
     public <T extends TileEntity> T getBlockTileEntity()
