--- a/net/minecraft/block/BlockLog.java
+++ b/net/minecraft/block/BlockLog.java
@@ -30,11 +30,11 @@
 
         if (p_180663_1_.func_175707_a(p_180663_2_.func_177982_a(-5, -5, -5), p_180663_2_.func_177982_a(5, 5, 5)))
         {
-            for (BlockPos blockpos : BlockPos.func_177980_a(p_180663_2_.func_177982_a(-4, -4, -4), p_180663_2_.func_177982_a(4, 4, 4)))
+            for (BlockPos blockpos : BlockPos.getBlocksBetween(p_180663_2_.func_177982_a(-4, -4, -4), p_180663_2_.func_177982_a(4, 4, 4)))
             {
-                IBlockState iblockstate = p_180663_1_.func_180495_p(blockpos);
+                IBlockState iblockstate = p_180663_1_.getBlockState(blockpos);
 
-                if (iblockstate.func_185904_a() == Material.leaves && !((Boolean)iblockstate.func_177229_b(BlockLeaves.field_176236_b)).booleanValue())
+                if (iblockstate.getMaterial() == Material.leaves && !((Boolean)iblockstate.func_177229_b(BlockLeaves.field_176236_b)).booleanValue())
                 {
                     p_180663_1_.func_180501_a(blockpos, iblockstate.func_177226_a(BlockLeaves.field_176236_b, Boolean.valueOf(true)), 4);
                 }
