--- a/net/minecraft/block/BlockBeacon.java
+++ b/net/minecraft/block/BlockBeacon.java
@@ -114,20 +114,20 @@
         {
             public void run()
             {
-                Chunk chunk = p_176450_0_.func_175726_f(p_176450_1_);
+                Chunk chunk = p_176450_0_.getChunk(p_176450_1_);
 
-                for (int i = p_176450_1_.func_177956_o() - 1; i >= 0; --i)
+                for (int i = p_176450_1_.getY() - 1; i >= 0; --i)
                 {
-                    final BlockPos blockpos = new BlockPos(p_176450_1_.func_177958_n(), i, p_176450_1_.func_177952_p());
+                    final BlockPos blockpos = new BlockPos(p_176450_1_.getX(), i, p_176450_1_.getZ());
 
                     if (!chunk.func_177444_d(blockpos))
                     {
                         break;
                     }
 
-                    IBlockState iblockstate = p_176450_0_.func_180495_p(blockpos);
+                    IBlockState iblockstate = p_176450_0_.getBlockState(blockpos);
 
-                    if (iblockstate.func_177230_c() == Blocks.beacon)
+                    if (iblockstate.getBlock() == Blocks.beacon)
                     {
                         ((WorldServer)p_176450_0_).addScheduledTask(new Runnable()
                         {
