--- a/net/minecraft/block/BlockCactus.java
+++ b/net/minecraft/block/BlockCactus.java
@@ -32,13 +32,13 @@
 
     public void func_180650_b(World p_180650_1_, BlockPos p_180650_2_, IBlockState p_180650_3_, Random p_180650_4_)
     {
-        BlockPos blockpos = p_180650_2_.func_177984_a();
+        BlockPos blockpos = p_180650_2_.getBlockPosAbove();
 
-        if (p_180650_1_.func_175623_d(blockpos))
+        if (p_180650_1_.isEmpty(blockpos))
         {
             int i;
 
-            for (i = 1; p_180650_1_.func_180495_p(p_180650_2_.func_177979_c(i)).func_177230_c() == this; ++i)
+            for (i = 1; p_180650_1_.getBlockState(p_180650_2_.getBlockPosBelow(i)).getBlock() == this; ++i)
             {
                 ;
             }
@@ -49,7 +49,7 @@
 
                 if (j == 15)
                 {
-                    p_180650_1_.func_175656_a(blockpos, this.func_176223_P());
+                    p_180650_1_.func_175656_a(blockpos, this.getBlockState());
                     IBlockState iblockstate = p_180650_3_.func_177226_a(field_176587_a, Integer.valueOf(0));
                     p_180650_1_.func_180501_a(p_180650_2_, iblockstate, 4);
                     iblockstate.func_189546_a(p_180650_1_, blockpos, this);
@@ -69,7 +69,7 @@
 
     public AxisAlignedBB func_180640_a(IBlockState p_180640_1_, World p_180640_2_, BlockPos p_180640_3_)
     {
-        return field_185594_c.func_186670_a(p_180640_3_);
+        return field_185594_c.offset(p_180640_3_);
     }
 
     public boolean renderAsNormalBlock(IBlockState p_149686_1_)
@@ -99,7 +99,7 @@
     {
         for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
         {
-            Material material = p_176586_1_.func_180495_p(p_176586_2_.func_177972_a(enumfacing)).func_185904_a();
+            Material material = p_176586_1_.getBlockState(p_176586_2_.getBlockPosRelative(enumfacing)).getMaterial();
 
             if (material.isSolid() || material == Material.lava)
             {
@@ -107,8 +107,8 @@
             }
         }
 
-        Block block = p_176586_1_.func_180495_p(p_176586_2_.func_177977_b()).func_177230_c();
-        return block == Blocks.cactus || block == Blocks.sand && !p_176586_1_.func_180495_p(p_176586_2_.func_177984_a()).func_185904_a().isLiquid();
+        Block block = p_176586_1_.getBlockState(p_176586_2_.getBlockPosBelow()).getBlock();
+        return block == Blocks.cactus || block == Blocks.sand && !p_176586_1_.getBlockState(p_176586_2_.getBlockPosAbove()).getMaterial().isLiquid();
     }
 
     public void func_180634_a(World p_180634_1_, BlockPos p_180634_2_, IBlockState p_180634_3_, Entity p_180634_4_)
@@ -123,7 +123,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176587_a, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176587_a, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
