--- a/net/minecraft/block/BlockCauldron.java
+++ b/net/minecraft/block/BlockCauldron.java
@@ -71,7 +71,7 @@
     public void func_180634_a(World p_180634_1_, BlockPos p_180634_2_, IBlockState p_180634_3_, Entity p_180634_4_)
     {
         int i = ((Integer)p_180634_3_.func_177229_b(field_176591_a)).intValue();
-        float f = (float)p_180634_2_.func_177956_o() + (6.0F + (float)(3 * i)) / 16.0F;
+        float f = (float)p_180634_2_.getY() + (6.0F + (float)(3 * i)) / 16.0F;
 
         if (!p_180634_1_.isRemote && p_180634_4_.isBurning() && i > 0 && p_180634_4_.func_174813_aQ().minY <= (double)f)
         {
@@ -97,7 +97,7 @@
                 {
                     if (!p_180639_4_.capabilities.isCreativeMode)
                     {
-                        p_180639_4_.func_184611_a(p_180639_5_, new ItemStack(Items.bucket));
+                        p_180639_4_.equipItemInHand(p_180639_5_, new ItemStack(Items.bucket));
                     }
 
                     p_180639_4_.triggerAchievement(StatList.field_188077_K);
@@ -116,7 +116,7 @@
 
                         if (p_180639_6_.stackSize == 0)
                         {
-                            p_180639_4_.func_184611_a(p_180639_5_, new ItemStack(Items.water_bucket));
+                            p_180639_4_.equipItemInHand(p_180639_5_, new ItemStack(Items.water_bucket));
                         }
                         else if (!p_180639_4_.inventory.addItemStackToInventory(new ItemStack(Items.water_bucket)))
                         {
@@ -141,7 +141,7 @@
 
                         if (--p_180639_6_.stackSize == 0)
                         {
-                            p_180639_4_.func_184611_a(p_180639_5_, itemstack1);
+                            p_180639_4_.equipItemInHand(p_180639_5_, itemstack1);
                         }
                         else if (!p_180639_4_.inventory.addItemStackToInventory(itemstack1))
                         {
@@ -189,7 +189,7 @@
 
                         if (p_180639_6_.stackSize == 0)
                         {
-                            p_180639_4_.func_184611_a(p_180639_5_, itemstack);
+                            p_180639_4_.equipItemInHand(p_180639_5_, itemstack);
                         }
                         else if (!p_180639_4_.inventory.addItemStackToInventory(itemstack))
                         {
@@ -228,9 +228,9 @@
         {
             float f = p_176224_1_.func_180494_b(p_176224_2_).func_180626_a(p_176224_2_);
 
-            if (p_176224_1_.getWorldChunkManager().getTemperatureAtHeight(f, p_176224_2_.func_177956_o()) >= 0.15F)
+            if (p_176224_1_.getWorldChunkManager().getTemperatureAtHeight(f, p_176224_2_.getY()) >= 0.15F)
             {
-                IBlockState iblockstate = p_176224_1_.func_180495_p(p_176224_2_);
+                IBlockState iblockstate = p_176224_1_.getBlockState(p_176224_2_);
 
                 if (((Integer)iblockstate.func_177229_b(field_176591_a)).intValue() < 3)
                 {
@@ -263,7 +263,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176591_a, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176591_a, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
