--- a/net/minecraft/block/BlockTorch.java
+++ b/net/minecraft/block/BlockTorch.java
@@ -83,13 +83,13 @@
 
     private boolean func_176594_d(World p_176594_1_, BlockPos p_176594_2_)
     {
-        if (p_176594_1_.func_180495_p(p_176594_2_).func_185896_q())
+        if (p_176594_1_.getBlockState(p_176594_2_).func_185896_q())
         {
             return true;
         }
         else
         {
-            Block block = p_176594_1_.func_180495_p(p_176594_2_).func_177230_c();
+            Block block = p_176594_1_.getBlockState(p_176594_2_).getBlock();
             return block instanceof BlockFence || block == Blocks.glass || block == Blocks.cobblestone_wall || block == Blocks.stained_glass;
         }
     }
@@ -109,7 +109,7 @@
 
     private boolean func_176595_b(World p_176595_1_, BlockPos p_176595_2_, EnumFacing p_176595_3_)
     {
-        BlockPos blockpos = p_176595_2_.func_177972_a(p_176595_3_.func_176734_d());
+        BlockPos blockpos = p_176595_2_.getBlockPosRelative(p_176595_3_.func_176734_d());
         boolean flag = p_176595_3_.func_176740_k().func_176722_c();
         return flag && p_176595_1_.func_175677_d(blockpos, true) || p_176595_3_.equals(EnumFacing.UP) && this.func_176594_d(p_176595_1_, blockpos);
     }
@@ -118,19 +118,19 @@
     {
         if (this.func_176595_b(p_180642_1_, p_180642_2_, p_180642_3_))
         {
-            return this.func_176223_P().func_177226_a(field_176596_a, p_180642_3_);
+            return this.getBlockState().func_177226_a(field_176596_a, p_180642_3_);
         }
         else
         {
             for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
             {
-                if (p_180642_1_.func_175677_d(p_180642_2_.func_177972_a(enumfacing.func_176734_d()), true))
+                if (p_180642_1_.func_175677_d(p_180642_2_.getBlockPosRelative(enumfacing.func_176734_d()), true))
                 {
-                    return this.func_176223_P().func_177226_a(field_176596_a, enumfacing);
+                    return this.getBlockState().func_177226_a(field_176596_a, enumfacing);
                 }
             }
 
-            return this.func_176223_P();
+            return this.getBlockState();
         }
     }
 
@@ -157,11 +157,11 @@
             EnumFacing enumfacing1 = enumfacing.func_176734_d();
             boolean flag = false;
 
-            if (enumfacing$axis.func_176722_c() && !p_176592_1_.func_175677_d(p_176592_2_.func_177972_a(enumfacing1), true))
+            if (enumfacing$axis.func_176722_c() && !p_176592_1_.func_175677_d(p_176592_2_.getBlockPosRelative(enumfacing1), true))
             {
                 flag = true;
             }
-            else if (enumfacing$axis.func_176720_b() && !this.func_176594_d(p_176592_1_, p_176592_2_.func_177972_a(enumfacing1)))
+            else if (enumfacing$axis.func_176720_b() && !this.func_176594_d(p_176592_1_, p_176592_2_.getBlockPosRelative(enumfacing1)))
             {
                 flag = true;
             }
@@ -181,13 +181,13 @@
 
     protected boolean func_176593_f(World p_176593_1_, BlockPos p_176593_2_, IBlockState p_176593_3_)
     {
-        if (p_176593_3_.func_177230_c() == this && this.func_176595_b(p_176593_1_, p_176593_2_, (EnumFacing)p_176593_3_.func_177229_b(field_176596_a)))
+        if (p_176593_3_.getBlock() == this && this.func_176595_b(p_176593_1_, p_176593_2_, (EnumFacing)p_176593_3_.func_177229_b(field_176596_a)))
         {
             return true;
         }
         else
         {
-            if (p_176593_1_.func_180495_p(p_176593_2_).func_177230_c() == this)
+            if (p_176593_1_.getBlockState(p_176593_2_).getBlock() == this)
             {
                 this.func_176226_b(p_176593_1_, p_176593_2_, p_176593_3_, 0);
                 p_176593_1_.func_175698_g(p_176593_2_);
@@ -200,22 +200,22 @@
     public void func_180655_c(IBlockState p_180655_1_, World p_180655_2_, BlockPos p_180655_3_, Random p_180655_4_)
     {
         EnumFacing enumfacing = (EnumFacing)p_180655_1_.func_177229_b(field_176596_a);
-        double d0 = (double)p_180655_3_.func_177958_n() + 0.5D;
-        double d1 = (double)p_180655_3_.func_177956_o() + 0.7D;
-        double d2 = (double)p_180655_3_.func_177952_p() + 0.5D;
+        double d0 = (double)p_180655_3_.getX() + 0.5D;
+        double d1 = (double)p_180655_3_.getY() + 0.7D;
+        double d2 = (double)p_180655_3_.getZ() + 0.5D;
         double d3 = 0.22D;
         double d4 = 0.27D;
 
         if (enumfacing.func_176740_k().func_176722_c())
         {
             EnumFacing enumfacing1 = enumfacing.func_176734_d();
-            p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_NORMAL, d0 + 0.27D * (double)enumfacing1.getFrontOffsetX(), d1 + 0.22D, d2 + 0.27D * (double)enumfacing1.getFrontOffsetZ(), 0.0D, 0.0D, 0.0D, new int[0]);
-            p_180655_2_.func_175688_a(EnumParticleTypes.FLAME, d0 + 0.27D * (double)enumfacing1.getFrontOffsetX(), d1 + 0.22D, d2 + 0.27D * (double)enumfacing1.getFrontOffsetZ(), 0.0D, 0.0D, 0.0D, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, d0 + 0.27D * (double)enumfacing1.getFrontOffsetX(), d1 + 0.22D, d2 + 0.27D * (double)enumfacing1.getFrontOffsetZ(), 0.0D, 0.0D, 0.0D, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.FLAME, d0 + 0.27D * (double)enumfacing1.getFrontOffsetX(), d1 + 0.22D, d2 + 0.27D * (double)enumfacing1.getFrontOffsetZ(), 0.0D, 0.0D, 0.0D, new int[0]);
         }
         else
         {
-            p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_NORMAL, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
-            p_180655_2_.func_175688_a(EnumParticleTypes.FLAME, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.FLAME, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
         }
     }
 
@@ -226,7 +226,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        IBlockState iblockstate = this.func_176223_P();
+        IBlockState iblockstate = this.getBlockState();
 
         switch (p_176203_1_)
         {
