--- a/net/minecraft/block/BlockCommandBlock.java
+++ b/net/minecraft/block/BlockCommandBlock.java
@@ -139,7 +139,7 @@
     public boolean func_185562_e(World p_185562_1_, BlockPos p_185562_2_, IBlockState p_185562_3_)
     {
         EnumFacing enumfacing = (EnumFacing)p_185562_3_.func_177229_b(field_185564_a);
-        TileEntity tileentity = p_185562_1_.func_175625_s(p_185562_2_.func_177972_a(enumfacing.func_176734_d()));
+        TileEntity tileentity = p_185562_1_.func_175625_s(p_185562_2_.getBlockPosRelative(enumfacing.func_176734_d()));
         return tileentity instanceof TileEntityCommandBlock && ((TileEntityCommandBlock)tileentity).func_145993_a().getSuccessCount() > 0;
     }
 
@@ -228,7 +228,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_185564_a, EnumFacing.getFront(p_176203_1_ & 7)).func_177226_a(field_185565_b, Boolean.valueOf((p_176203_1_ & 8) != 0));
+        return this.getBlockState().func_177226_a(field_185564_a, EnumFacing.getFront(p_176203_1_ & 7)).func_177226_a(field_185565_b, Boolean.valueOf((p_176203_1_ & 8) != 0));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
@@ -253,14 +253,14 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        return this.func_176223_P().func_177226_a(field_185564_a, BlockPistonBase.func_185647_a(p_180642_2_, p_180642_8_)).func_177226_a(field_185565_b, Boolean.valueOf(false));
+        return this.getBlockState().func_177226_a(field_185564_a, BlockPistonBase.func_185647_a(p_180642_2_, p_180642_8_)).func_177226_a(field_185565_b, Boolean.valueOf(false));
     }
 
     public void func_185563_c(World p_185563_1_, BlockPos p_185563_2_)
     {
-        IBlockState iblockstate = p_185563_1_.func_180495_p(p_185563_2_);
+        IBlockState iblockstate = p_185563_1_.getBlockState(p_185563_2_);
 
-        if (iblockstate.func_177230_c() == Blocks.command_block || iblockstate.func_177230_c() == Blocks.field_185776_dc)
+        if (iblockstate.getBlock() == Blocks.command_block || iblockstate.getBlock() == Blocks.field_185776_dc)
         {
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(p_185563_2_);
             blockpos$mutableblockpos.func_189536_c((EnumFacing)iblockstate.func_177229_b(field_185564_a));
@@ -274,8 +274,8 @@
                     break;
                 }
 
-                IBlockState iblockstate1 = p_185563_1_.func_180495_p(blockpos$mutableblockpos);
-                Block block = iblockstate1.func_177230_c();
+                IBlockState iblockstate1 = p_185563_1_.getBlockState(blockpos$mutableblockpos);
+                Block block = iblockstate1.getBlock();
 
                 if (block != Blocks.field_185777_dd || p_185563_1_.func_184145_b(blockpos$mutableblockpos, block))
                 {
