--- a/net/minecraft/block/BlockHugeMushroom.java
+++ b/net/minecraft/block/BlockHugeMushroom.java
@@ -69,12 +69,12 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        return this.func_176223_P();
+        return this.getBlockState();
     }
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176380_a, BlockHugeMushroom.EnumType.func_176895_a(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176380_a, BlockHugeMushroom.EnumType.func_176895_a(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
