--- a/net/minecraft/block/BlockPumpkin.java
+++ b/net/minecraft/block/BlockPumpkin.java
@@ -34,7 +34,7 @@
     {
         public boolean apply(@Nullable IBlockState p_apply_1_)
         {
-            return p_apply_1_ != null && (p_apply_1_.func_177230_c() == Blocks.pumpkin || p_apply_1_.func_177230_c() == Blocks.lit_pumpkin);
+            return p_apply_1_ != null && (p_apply_1_.getBlock() == Blocks.pumpkin || p_apply_1_.getBlock() == Blocks.lit_pumpkin);
         }
     };
 
@@ -66,17 +66,17 @@
             for (int i = 0; i < this.func_176391_l().func_177685_b(); ++i)
             {
                 BlockWorldState blockworldstate = blockpattern$patternhelper.func_177670_a(0, i, 0);
-                p_180673_1_.func_180501_a(blockworldstate.func_177508_d(), Blocks.air.func_176223_P(), 2);
+                p_180673_1_.func_180501_a(blockworldstate.func_177508_d(), Blocks.air.getBlockState(), 2);
             }
 
             EntitySnowman entitysnowman = new EntitySnowman(p_180673_1_);
             BlockPos blockpos1 = blockpattern$patternhelper.func_177670_a(0, 2, 0).func_177508_d();
-            entitysnowman.setLocationAndAngles((double)blockpos1.func_177958_n() + 0.5D, (double)blockpos1.func_177956_o() + 0.05D, (double)blockpos1.func_177952_p() + 0.5D, 0.0F, 0.0F);
+            entitysnowman.setLocationAndAngles((double)blockpos1.getX() + 0.5D, (double)blockpos1.getY() + 0.05D, (double)blockpos1.getZ() + 0.5D, 0.0F, 0.0F);
             p_180673_1_.spawnEntityInWorld(entitysnowman);
 
             for (int j = 0; j < 120; ++j)
             {
-                p_180673_1_.func_175688_a(EnumParticleTypes.SNOW_SHOVEL, (double)blockpos1.func_177958_n() + p_180673_1_.rand.nextDouble(), (double)blockpos1.func_177956_o() + p_180673_1_.rand.nextDouble() * 2.5D, (double)blockpos1.func_177952_p() + p_180673_1_.rand.nextDouble(), 0.0D, 0.0D, 0.0D, new int[0]);
+                p_180673_1_.spawnParticle(EnumParticleTypes.SNOW_SHOVEL, (double)blockpos1.getX() + p_180673_1_.rand.nextDouble(), (double)blockpos1.getY() + p_180673_1_.rand.nextDouble() * 2.5D, (double)blockpos1.getZ() + p_180673_1_.rand.nextDouble(), 0.0D, 0.0D, 0.0D, new int[0]);
             }
 
             for (int i1 = 0; i1 < this.func_176391_l().func_177685_b(); ++i1)
@@ -95,19 +95,19 @@
                 {
                     for (int l = 0; l < this.func_176388_T().func_177685_b(); ++l)
                     {
-                        p_180673_1_.func_180501_a(blockpattern$patternhelper.func_177670_a(k, l, 0).func_177508_d(), Blocks.air.func_176223_P(), 2);
+                        p_180673_1_.func_180501_a(blockpattern$patternhelper.func_177670_a(k, l, 0).func_177508_d(), Blocks.air.getBlockState(), 2);
                     }
                 }
 
                 BlockPos blockpos = blockpattern$patternhelper.func_177670_a(1, 2, 0).func_177508_d();
                 EntityIronGolem entityirongolem = new EntityIronGolem(p_180673_1_);
                 entityirongolem.setPlayerCreated(true);
-                entityirongolem.setLocationAndAngles((double)blockpos.func_177958_n() + 0.5D, (double)blockpos.func_177956_o() + 0.05D, (double)blockpos.func_177952_p() + 0.5D, 0.0F, 0.0F);
+                entityirongolem.setLocationAndAngles((double)blockpos.getX() + 0.5D, (double)blockpos.getY() + 0.05D, (double)blockpos.getZ() + 0.5D, 0.0F, 0.0F);
                 p_180673_1_.spawnEntityInWorld(entityirongolem);
 
                 for (int j1 = 0; j1 < 120; ++j1)
                 {
-                    p_180673_1_.func_175688_a(EnumParticleTypes.SNOWBALL, (double)blockpos.func_177958_n() + p_180673_1_.rand.nextDouble(), (double)blockpos.func_177956_o() + p_180673_1_.rand.nextDouble() * 3.9D, (double)blockpos.func_177952_p() + p_180673_1_.rand.nextDouble(), 0.0D, 0.0D, 0.0D, new int[0]);
+                    p_180673_1_.spawnParticle(EnumParticleTypes.SNOWBALL, (double)blockpos.getX() + p_180673_1_.rand.nextDouble(), (double)blockpos.getY() + p_180673_1_.rand.nextDouble() * 3.9D, (double)blockpos.getZ() + p_180673_1_.rand.nextDouble(), 0.0D, 0.0D, 0.0D, new int[0]);
                 }
 
                 for (int k1 = 0; k1 < this.func_176388_T().func_177684_c(); ++k1)
@@ -124,7 +124,7 @@
 
     public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
     {
-        return p_176196_1_.func_180495_p(p_176196_2_).func_177230_c().blockMaterial.isReplaceable() && p_176196_1_.func_180495_p(p_176196_2_.func_177977_b()).func_185896_q();
+        return p_176196_1_.getBlockState(p_176196_2_).getBlock().blockMaterial.isReplaceable() && p_176196_1_.getBlockState(p_176196_2_.getBlockPosBelow()).func_185896_q();
     }
 
     public IBlockState func_185499_a(IBlockState p_185499_1_, Rotation p_185499_2_)
@@ -139,12 +139,12 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        return this.func_176223_P().func_177226_a(field_185512_D, p_180642_8_.func_174811_aO().func_176734_d());
+        return this.getBlockState().func_177226_a(field_185512_D, p_180642_8_.func_174811_aO().func_176734_d());
     }
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_185512_D, EnumFacing.func_176731_b(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_185512_D, EnumFacing.func_176731_b(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
