--- a/net/minecraft/block/BlockCarpet.java
+++ b/net/minecraft/block/BlockCarpet.java
@@ -76,12 +76,12 @@
 
     private boolean func_176329_d(World p_176329_1_, BlockPos p_176329_2_)
     {
-        return !p_176329_1_.func_175623_d(p_176329_2_.func_177977_b());
+        return !p_176329_1_.isEmpty(p_176329_2_.getBlockPosBelow());
     }
 
     public boolean func_176225_a(IBlockState p_176225_1_, IBlockAccess p_176225_2_, BlockPos p_176225_3_, EnumFacing p_176225_4_)
     {
-        return p_176225_4_ == EnumFacing.UP ? true : (p_176225_2_.func_180495_p(p_176225_3_.func_177972_a(p_176225_4_)).func_177230_c() == this ? true : super.func_176225_a(p_176225_1_, p_176225_2_, p_176225_3_, p_176225_4_));
+        return p_176225_4_ == EnumFacing.UP ? true : (p_176225_2_.getBlockState(p_176225_3_.getBlockPosRelative(p_176225_4_)).getBlock() == this ? true : super.func_176225_a(p_176225_1_, p_176225_2_, p_176225_3_, p_176225_4_));
     }
 
     public int func_180651_a(IBlockState p_180651_1_)
@@ -102,7 +102,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176330_a, EnumDyeColor.func_176764_b(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176330_a, EnumDyeColor.func_176764_b(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
