--- a/net/minecraft/block/BlockLilyPad.java
+++ b/net/minecraft/block/BlockLilyPad.java
@@ -47,15 +47,15 @@
 
     protected boolean func_185514_i(IBlockState p_185514_1_)
     {
-        return p_185514_1_.func_177230_c() == Blocks.water || p_185514_1_.func_185904_a() == Material.ice;
+        return p_185514_1_.getBlock() == Blocks.water || p_185514_1_.getMaterial() == Material.ice;
     }
 
     public boolean func_180671_f(World p_180671_1_, BlockPos p_180671_2_, IBlockState p_180671_3_)
     {
-        if (p_180671_2_.func_177956_o() >= 0 && p_180671_2_.func_177956_o() < 256)
+        if (p_180671_2_.getY() >= 0 && p_180671_2_.getY() < 256)
         {
-            IBlockState iblockstate = p_180671_1_.func_180495_p(p_180671_2_.func_177977_b());
-            Material material = iblockstate.func_185904_a();
+            IBlockState iblockstate = p_180671_1_.getBlockState(p_180671_2_.getBlockPosBelow());
+            Material material = iblockstate.getMaterial();
             return material == Material.water && ((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue() == 0 || material == Material.ice;
         }
         else
