--- a/net/minecraft/block/BlockNetherWart.java
+++ b/net/minecraft/block/BlockNetherWart.java
@@ -38,12 +38,12 @@
 
     protected boolean func_185514_i(IBlockState p_185514_1_)
     {
-        return p_185514_1_.func_177230_c() == Blocks.soul_sand;
+        return p_185514_1_.getBlock() == Blocks.soul_sand;
     }
 
     public boolean func_180671_f(World p_180671_1_, BlockPos p_180671_2_, IBlockState p_180671_3_)
     {
-        return this.func_185514_i(p_180671_1_.func_180495_p(p_180671_2_.func_177977_b()));
+        return this.func_185514_i(p_180671_1_.getBlockState(p_180671_2_.getBlockPosBelow()));
     }
 
     public void func_180650_b(World p_180650_1_, BlockPos p_180650_2_, IBlockState p_180650_3_, Random p_180650_4_)
@@ -103,7 +103,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176486_a, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176486_a, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
