--- a/net/minecraft/block/BlockContainer.java
+++ b/net/minecraft/block/BlockContainer.java
@@ -24,7 +24,7 @@
 
     protected boolean func_181086_a(World p_181086_1_, BlockPos p_181086_2_, EnumFacing p_181086_3_)
     {
-        return p_181086_1_.func_180495_p(p_181086_2_.func_177972_a(p_181086_3_)).func_185904_a() == Material.cactus;
+        return p_181086_1_.getBlockState(p_181086_2_.getBlockPosRelative(p_181086_3_)).getMaterial() == Material.cactus;
     }
 
     protected boolean func_181087_e(World p_181087_1_, BlockPos p_181087_2_)
