--- a/net/minecraft/block/BlockCake.java
+++ b/net/minecraft/block/BlockCake.java
@@ -92,7 +92,7 @@
 
     private boolean func_176588_d(World p_176588_1_, BlockPos p_176588_2_)
     {
-        return p_176588_1_.func_180495_p(p_176588_2_.func_177977_b()).func_185904_a().isSolid();
+        return p_176588_1_.getBlockState(p_176588_2_.getBlockPosBelow()).getMaterial().isSolid();
     }
 
     /**
@@ -121,7 +121,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176589_a, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176589_a, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
