--- a/net/minecraft/block/BlockPistonBase.java
+++ b/net/minecraft/block/BlockPistonBase.java
@@ -127,7 +127,7 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        return this.func_176223_P().func_177226_a(field_176387_N, func_185647_a(p_180642_2_, p_180642_8_)).func_177226_a(field_176320_b, Boolean.valueOf(false));
+        return this.getBlockState().func_177226_a(field_176387_N, func_185647_a(p_180642_2_, p_180642_8_)).func_177226_a(field_176320_b, Boolean.valueOf(false));
     }
 
     private void func_176316_e(World p_176316_1_, BlockPos p_176316_2_, IBlockState p_176316_3_)
@@ -152,7 +152,7 @@
     {
         for (EnumFacing enumfacing : EnumFacing.values())
         {
-            if (enumfacing != p_176318_3_ && p_176318_1_.func_175709_b(p_176318_2_.func_177972_a(enumfacing), enumfacing))
+            if (enumfacing != p_176318_3_ && p_176318_1_.func_175709_b(p_176318_2_.getBlockPosRelative(enumfacing), enumfacing))
             {
                 return true;
             }
@@ -164,11 +164,11 @@
         }
         else
         {
-            BlockPos blockpos = p_176318_2_.func_177984_a();
+            BlockPos blockpos = p_176318_2_.getBlockPosAbove();
 
             for (EnumFacing enumfacing1 : EnumFacing.values())
             {
-                if (enumfacing1 != EnumFacing.DOWN && p_176318_1_.func_175709_b(blockpos.func_177972_a(enumfacing1), enumfacing1))
+                if (enumfacing1 != EnumFacing.DOWN && p_176318_1_.func_175709_b(blockpos.getBlockPosRelative(enumfacing1), enumfacing1))
                 {
                     return true;
                 }
@@ -210,21 +210,21 @@
         }
         else if (p_189539_4_ == 1)
         {
-            TileEntity tileentity1 = p_189539_2_.func_175625_s(p_189539_3_.func_177972_a(enumfacing));
+            TileEntity tileentity1 = p_189539_2_.func_175625_s(p_189539_3_.getBlockPosRelative(enumfacing));
 
             if (tileentity1 instanceof TileEntityPiston)
             {
                 ((TileEntityPiston)tileentity1).clearPistonTileEntity();
             }
 
-            p_189539_2_.func_180501_a(p_189539_3_, Blocks.field_180384_M.func_176223_P().func_177226_a(BlockPistonMoving.field_176426_a, enumfacing).func_177226_a(BlockPistonMoving.field_176425_b, this.isSticky ? BlockPistonExtension.EnumPistonType.STICKY : BlockPistonExtension.EnumPistonType.DEFAULT), 3);
+            p_189539_2_.func_180501_a(p_189539_3_, Blocks.field_180384_M.getBlockState().func_177226_a(BlockPistonMoving.field_176426_a, enumfacing).func_177226_a(BlockPistonMoving.field_176425_b, this.isSticky ? BlockPistonExtension.EnumPistonType.STICKY : BlockPistonExtension.EnumPistonType.DEFAULT), 3);
             p_189539_2_.func_175690_a(p_189539_3_, BlockPistonMoving.func_185588_a(this.func_176203_a(p_189539_5_), enumfacing, false, true));
 
             if (this.isSticky)
             {
                 BlockPos blockpos = p_189539_3_.func_177982_a(enumfacing.getFrontOffsetX() * 2, enumfacing.getFrontOffsetY() * 2, enumfacing.getFrontOffsetZ() * 2);
-                IBlockState iblockstate = p_189539_2_.func_180495_p(blockpos);
-                Block block = iblockstate.func_177230_c();
+                IBlockState iblockstate = p_189539_2_.getBlockState(blockpos);
+                Block block = iblockstate.getBlock();
                 boolean flag1 = false;
 
                 if (block == Blocks.field_180384_M)
@@ -243,14 +243,14 @@
                     }
                 }
 
-                if (!flag1 && iblockstate.func_185904_a() != Material.air && func_185646_a(iblockstate, p_189539_2_, blockpos, enumfacing.func_176734_d(), false) && (iblockstate.func_185905_o() == EnumPushReaction.NORMAL || block == Blocks.piston || block == Blocks.sticky_piston))
+                if (!flag1 && iblockstate.getMaterial() != Material.air && func_185646_a(iblockstate, p_189539_2_, blockpos, enumfacing.func_176734_d(), false) && (iblockstate.func_185905_o() == EnumPushReaction.NORMAL || block == Blocks.piston || block == Blocks.sticky_piston))
                 {
                     this.func_176319_a(p_189539_2_, p_189539_3_, enumfacing, false);
                 }
             }
             else
             {
-                p_189539_2_.func_175698_g(p_189539_3_.func_177972_a(enumfacing));
+                p_189539_2_.func_175698_g(p_189539_3_.getBlockPosRelative(enumfacing));
             }
 
             p_189539_2_.func_184133_a((EntityPlayer)null, p_189539_3_, SoundEvents.field_187712_dQ, SoundCategory.BLOCKS, 0.5F, p_189539_2_.rand.nextFloat() * 0.15F + 0.6F);
@@ -273,16 +273,16 @@
 
     public static EnumFacing func_185647_a(BlockPos p_185647_0_, EntityLivingBase p_185647_1_)
     {
-        if (MathHelper.abs((float)p_185647_1_.posX - (float)p_185647_0_.func_177958_n()) < 2.0F && MathHelper.abs((float)p_185647_1_.posZ - (float)p_185647_0_.func_177952_p()) < 2.0F)
+        if (MathHelper.abs((float)p_185647_1_.posX - (float)p_185647_0_.getX()) < 2.0F && MathHelper.abs((float)p_185647_1_.posZ - (float)p_185647_0_.getZ()) < 2.0F)
         {
             double d0 = p_185647_1_.posY + (double)p_185647_1_.getEyeHeight();
 
-            if (d0 - (double)p_185647_0_.func_177956_o() > 2.0D)
+            if (d0 - (double)p_185647_0_.getY() > 2.0D)
             {
                 return EnumFacing.UP;
             }
 
-            if ((double)p_185647_0_.func_177956_o() - d0 > 0.0D)
+            if ((double)p_185647_0_.getY() - d0 > 0.0D)
             {
                 return EnumFacing.DOWN;
             }
@@ -293,7 +293,7 @@
 
     public static boolean func_185646_a(IBlockState p_185646_0_, World p_185646_1_, BlockPos p_185646_2_, EnumFacing p_185646_3_, boolean p_185646_4_)
     {
-        Block block = p_185646_0_.func_177230_c();
+        Block block = p_185646_0_.getBlock();
 
         if (block == Blocks.obsidian)
         {
@@ -303,9 +303,9 @@
         {
             return false;
         }
-        else if (p_185646_2_.func_177956_o() >= 0 && (p_185646_3_ != EnumFacing.DOWN || p_185646_2_.func_177956_o() != 0))
+        else if (p_185646_2_.getY() >= 0 && (p_185646_3_ != EnumFacing.DOWN || p_185646_2_.getY() != 0))
         {
-            if (p_185646_2_.func_177956_o() <= p_185646_1_.getHeight() - 1 && (p_185646_3_ != EnumFacing.UP || p_185646_2_.func_177956_o() != p_185646_1_.getHeight() - 1))
+            if (p_185646_2_.getY() <= p_185646_1_.getHeight() - 1 && (p_185646_3_ != EnumFacing.UP || p_185646_2_.getY() != p_185646_1_.getHeight() - 1))
             {
                 if (block != Blocks.piston && block != Blocks.sticky_piston)
                 {
@@ -346,7 +346,7 @@
     {
         if (!p_176319_4_)
         {
-            p_176319_1_.func_175698_g(p_176319_2_.func_177972_a(p_176319_3_));
+            p_176319_1_.func_175698_g(p_176319_2_.getBlockPosRelative(p_176319_3_));
         }
 
         BlockPistonStructureHelper blockpistonstructurehelper = new BlockPistonStructureHelper(p_176319_1_, p_176319_2_, p_176319_3_, p_176319_4_);
@@ -363,7 +363,7 @@
             for (int i = 0; i < list.size(); ++i)
             {
                 BlockPos blockpos = (BlockPos)list.get(i);
-                list1.add(p_176319_1_.func_180495_p(blockpos).func_185899_b(p_176319_1_, blockpos));
+                list1.add(p_176319_1_.getBlockState(blockpos).func_185899_b(p_176319_1_, blockpos));
             }
 
             List<BlockPos> list2 = blockpistonstructurehelper.func_177252_d();
@@ -374,8 +374,8 @@
             for (int j = list2.size() - 1; j >= 0; --j)
             {
                 BlockPos blockpos1 = (BlockPos)list2.get(j);
-                IBlockState iblockstate = p_176319_1_.func_180495_p(blockpos1);
-                iblockstate.func_177230_c().func_176226_b(p_176319_1_, blockpos1, iblockstate, 0);
+                IBlockState iblockstate = p_176319_1_.getBlockState(blockpos1);
+                iblockstate.getBlock().func_176226_b(p_176319_1_, blockpos1, iblockstate, 0);
                 p_176319_1_.func_175698_g(blockpos1);
                 --k;
                 aiblockstate[k] = iblockstate;
@@ -384,34 +384,34 @@
             for (int l = list.size() - 1; l >= 0; --l)
             {
                 BlockPos blockpos3 = (BlockPos)list.get(l);
-                IBlockState iblockstate2 = p_176319_1_.func_180495_p(blockpos3);
-                p_176319_1_.func_180501_a(blockpos3, Blocks.air.func_176223_P(), 2);
-                blockpos3 = blockpos3.func_177972_a(enumfacing);
-                p_176319_1_.func_180501_a(blockpos3, Blocks.field_180384_M.func_176223_P().func_177226_a(field_176387_N, p_176319_3_), 4);
+                IBlockState iblockstate2 = p_176319_1_.getBlockState(blockpos3);
+                p_176319_1_.func_180501_a(blockpos3, Blocks.air.getBlockState(), 2);
+                blockpos3 = blockpos3.getBlockPosRelative(enumfacing);
+                p_176319_1_.func_180501_a(blockpos3, Blocks.field_180384_M.getBlockState().func_177226_a(field_176387_N, p_176319_3_), 4);
                 p_176319_1_.func_175690_a(blockpos3, BlockPistonMoving.func_185588_a((IBlockState)list1.get(l), p_176319_3_, p_176319_4_, false));
                 --k;
                 aiblockstate[k] = iblockstate2;
             }
 
-            BlockPos blockpos2 = p_176319_2_.func_177972_a(p_176319_3_);
+            BlockPos blockpos2 = p_176319_2_.getBlockPosRelative(p_176319_3_);
 
             if (p_176319_4_)
             {
                 BlockPistonExtension.EnumPistonType blockpistonextension$enumpistontype = this.isSticky ? BlockPistonExtension.EnumPistonType.STICKY : BlockPistonExtension.EnumPistonType.DEFAULT;
-                IBlockState iblockstate3 = Blocks.piston_head.func_176223_P().func_177226_a(BlockPistonExtension.field_176387_N, p_176319_3_).func_177226_a(BlockPistonExtension.field_176325_b, blockpistonextension$enumpistontype);
-                IBlockState iblockstate1 = Blocks.field_180384_M.func_176223_P().func_177226_a(BlockPistonMoving.field_176426_a, p_176319_3_).func_177226_a(BlockPistonMoving.field_176425_b, this.isSticky ? BlockPistonExtension.EnumPistonType.STICKY : BlockPistonExtension.EnumPistonType.DEFAULT);
+                IBlockState iblockstate3 = Blocks.piston_head.getBlockState().func_177226_a(BlockPistonExtension.field_176387_N, p_176319_3_).func_177226_a(BlockPistonExtension.field_176325_b, blockpistonextension$enumpistontype);
+                IBlockState iblockstate1 = Blocks.field_180384_M.getBlockState().func_177226_a(BlockPistonMoving.field_176426_a, p_176319_3_).func_177226_a(BlockPistonMoving.field_176425_b, this.isSticky ? BlockPistonExtension.EnumPistonType.STICKY : BlockPistonExtension.EnumPistonType.DEFAULT);
                 p_176319_1_.func_180501_a(blockpos2, iblockstate1, 4);
                 p_176319_1_.func_175690_a(blockpos2, BlockPistonMoving.func_185588_a(iblockstate3, p_176319_3_, true, false));
             }
 
             for (int i1 = list2.size() - 1; i1 >= 0; --i1)
             {
-                p_176319_1_.func_175685_c((BlockPos)list2.get(i1), aiblockstate[k++].func_177230_c());
+                p_176319_1_.func_175685_c((BlockPos)list2.get(i1), aiblockstate[k++].getBlock());
             }
 
             for (int j1 = list.size() - 1; j1 >= 0; --j1)
             {
-                p_176319_1_.func_175685_c((BlockPos)list.get(j1), aiblockstate[k++].func_177230_c());
+                p_176319_1_.func_175685_c((BlockPos)list.get(j1), aiblockstate[k++].getBlock());
             }
 
             if (p_176319_4_)
@@ -426,7 +426,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176387_N, func_176317_b(p_176203_1_)).func_177226_a(field_176320_b, Boolean.valueOf((p_176203_1_ & 8) > 0));
+        return this.getBlockState().func_177226_a(field_176387_N, func_176317_b(p_176203_1_)).func_177226_a(field_176320_b, Boolean.valueOf((p_176203_1_ & 8) > 0));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
