--- a/net/minecraft/block/BlockReed.java
+++ b/net/minecraft/block/BlockReed.java
@@ -37,13 +37,13 @@
 
     public void func_180650_b(World p_180650_1_, BlockPos p_180650_2_, IBlockState p_180650_3_, Random p_180650_4_)
     {
-        if (p_180650_1_.func_180495_p(p_180650_2_.func_177977_b()).func_177230_c() == Blocks.reeds || this.func_176353_e(p_180650_1_, p_180650_2_, p_180650_3_))
+        if (p_180650_1_.getBlockState(p_180650_2_.getBlockPosBelow()).getBlock() == Blocks.reeds || this.func_176353_e(p_180650_1_, p_180650_2_, p_180650_3_))
         {
-            if (p_180650_1_.func_175623_d(p_180650_2_.func_177984_a()))
+            if (p_180650_1_.isEmpty(p_180650_2_.getBlockPosAbove()))
             {
                 int i;
 
-                for (i = 1; p_180650_1_.func_180495_p(p_180650_2_.func_177979_c(i)).func_177230_c() == this; ++i)
+                for (i = 1; p_180650_1_.getBlockState(p_180650_2_.getBlockPosBelow(i)).getBlock() == this; ++i)
                 {
                     ;
                 }
@@ -54,7 +54,7 @@
 
                     if (j == 15)
                     {
-                        p_180650_1_.func_175656_a(p_180650_2_.func_177984_a(), this.func_176223_P());
+                        p_180650_1_.func_175656_a(p_180650_2_.getBlockPosAbove(), this.getBlockState());
                         p_180650_1_.func_180501_a(p_180650_2_, p_180650_3_.func_177226_a(field_176355_a, Integer.valueOf(0)), 4);
                     }
                     else
@@ -68,7 +68,7 @@
 
     public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
     {
-        Block block = p_176196_1_.func_180495_p(p_176196_2_.func_177977_b()).func_177230_c();
+        Block block = p_176196_1_.getBlockState(p_176196_2_.getBlockPosBelow()).getBlock();
 
         if (block == this)
         {
@@ -80,13 +80,13 @@
         }
         else
         {
-            BlockPos blockpos = p_176196_2_.func_177977_b();
+            BlockPos blockpos = p_176196_2_.getBlockPosBelow();
 
             for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
             {
-                IBlockState iblockstate = p_176196_1_.func_180495_p(blockpos.func_177972_a(enumfacing));
+                IBlockState iblockstate = p_176196_1_.getBlockState(blockpos.getBlockPosRelative(enumfacing));
 
-                if (iblockstate.func_185904_a() == Material.water || iblockstate.func_177230_c() == Blocks.field_185778_de)
+                if (iblockstate.getMaterial() == Material.water || iblockstate.getBlock() == Blocks.field_185778_de)
                 {
                     return true;
                 }
@@ -154,7 +154,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176355_a, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176355_a, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
