--- a/net/minecraft/block/BlockPlanks.java
+++ b/net/minecraft/block/BlockPlanks.java
@@ -41,7 +41,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176383_a, BlockPlanks.EnumType.func_176837_a(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176383_a, BlockPlanks.EnumType.func_176837_a(p_176203_1_));
     }
 
     public MapColor func_180659_g(IBlockState p_180659_1_)
