--- a/net/minecraft/block/BlockFlowerPot.java
+++ b/net/minecraft/block/BlockFlowerPot.java
@@ -139,12 +139,12 @@
 
     public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
     {
-        return super.func_176196_c(p_176196_1_, p_176196_2_) && p_176196_1_.func_180495_p(p_176196_2_.func_177977_b()).func_185896_q();
+        return super.func_176196_c(p_176196_1_, p_176196_2_) && p_176196_1_.getBlockState(p_176196_2_.getBlockPosBelow()).func_185896_q();
     }
 
     public void func_189540_a(IBlockState p_189540_1_, World p_189540_2_, BlockPos p_189540_3_, Block p_189540_4_)
     {
-        if (!p_189540_2_.func_180495_p(p_189540_3_.func_177977_b()).func_185896_q())
+        if (!p_189540_2_.getBlockState(p_189540_3_.getBlockPosBelow()).func_185896_q())
         {
             this.func_176226_b(p_189540_2_, p_189540_3_, p_189540_1_, 0);
             p_189540_2_.func_175698_g(p_189540_3_);
