--- a/net/minecraft/block/BlockSand.java
+++ b/net/minecraft/block/BlockSand.java
@@ -43,7 +43,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176504_a, BlockSand.EnumType.func_176686_a(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176504_a, BlockSand.EnumType.func_176686_a(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
