--- a/net/minecraft/block/BlockCocoa.java
+++ b/net/minecraft/block/BlockCocoa.java
@@ -54,9 +54,9 @@
 
     public boolean func_176499_e(World p_176499_1_, BlockPos p_176499_2_, IBlockState p_176499_3_)
     {
-        p_176499_2_ = p_176499_2_.func_177972_a((EnumFacing)p_176499_3_.func_177229_b(field_185512_D));
-        IBlockState iblockstate = p_176499_1_.func_180495_p(p_176499_2_);
-        return iblockstate.func_177230_c() == Blocks.log && iblockstate.func_177229_b(BlockOldLog.field_176301_b) == BlockPlanks.EnumType.JUNGLE;
+        p_176499_2_ = p_176499_2_.getBlockPosRelative((EnumFacing)p_176499_3_.func_177229_b(field_185512_D));
+        IBlockState iblockstate = p_176499_1_.getBlockState(p_176499_2_);
+        return iblockstate.getBlock() == Blocks.log && iblockstate.func_177229_b(BlockOldLog.field_176301_b) == BlockPlanks.EnumType.JUNGLE;
     }
 
     public boolean renderAsNormalBlock(IBlockState p_149686_1_)
@@ -113,7 +113,7 @@
             p_180642_3_ = EnumFacing.NORTH;
         }
 
-        return this.func_176223_P().func_177226_a(field_185512_D, p_180642_3_.func_176734_d()).func_177226_a(field_176501_a, Integer.valueOf(0));
+        return this.getBlockState().func_177226_a(field_185512_D, p_180642_3_.func_176734_d()).func_177226_a(field_176501_a, Integer.valueOf(0));
     }
 
     public void func_189540_a(IBlockState p_189540_1_, World p_189540_2_, BlockPos p_189540_3_, Block p_189540_4_)
@@ -126,7 +126,7 @@
 
     private void func_176500_f(World p_176500_1_, BlockPos p_176500_2_, IBlockState p_176500_3_)
     {
-        p_176500_1_.func_180501_a(p_176500_2_, Blocks.air.func_176223_P(), 3);
+        p_176500_1_.func_180501_a(p_176500_2_, Blocks.air.getBlockState(), 3);
         this.func_176226_b(p_176500_1_, p_176500_2_, p_176500_3_, 0);
     }
 
@@ -173,7 +173,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_185512_D, EnumFacing.func_176731_b(p_176203_1_)).func_177226_a(field_176501_a, Integer.valueOf((p_176203_1_ & 15) >> 2));
+        return this.getBlockState().func_177226_a(field_185512_D, EnumFacing.func_176731_b(p_176203_1_)).func_177226_a(field_176501_a, Integer.valueOf((p_176203_1_ & 15) >> 2));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
