--- a/net/minecraft/block/BlockFire.java
+++ b/net/minecraft/block/BlockFire.java
@@ -36,7 +36,7 @@
 
     public IBlockState func_176221_a(IBlockState p_176221_1_, IBlockAccess p_176221_2_, BlockPos p_176221_3_)
     {
-        return !p_176221_2_.func_180495_p(p_176221_3_.func_177977_b()).func_185896_q() && !Blocks.fire.func_176535_e(p_176221_2_, p_176221_3_.func_177977_b()) ? p_176221_1_.func_177226_a(field_176545_N, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.func_177978_c()))).func_177226_a(field_176546_O, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.func_177974_f()))).func_177226_a(field_176541_P, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.func_177968_d()))).func_177226_a(field_176539_Q, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.func_177976_e()))).func_177226_a(field_176542_R, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.func_177984_a()))) : this.func_176223_P();
+        return !p_176221_2_.getBlockState(p_176221_3_.getBlockPosBelow()).func_185896_q() && !Blocks.fire.func_176535_e(p_176221_2_, p_176221_3_.getBlockPosBelow()) ? p_176221_1_.func_177226_a(field_176545_N, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.getBlockPosNorth()))).func_177226_a(field_176546_O, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.getBlockPosEast()))).func_177226_a(field_176541_P, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.getBlockPosSouth()))).func_177226_a(field_176539_Q, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.getBlockPosWest()))).func_177226_a(field_176542_R, Boolean.valueOf(this.func_176535_e(p_176221_2_, p_176221_3_.getBlockPosAbove()))) : this.getBlockState();
     }
 
     protected BlockFire()
@@ -134,7 +134,7 @@
                 p_180650_1_.func_175698_g(p_180650_2_);
             }
 
-            Block block = p_180650_1_.func_180495_p(p_180650_2_.func_177977_b()).func_177230_c();
+            Block block = p_180650_1_.getBlockState(p_180650_2_.getBlockPosBelow()).getBlock();
             boolean flag = block == Blocks.netherrack;
 
             if (p_180650_1_.provider instanceof WorldProviderEnd && block == Blocks.bedrock)
@@ -162,7 +162,7 @@
                 {
                     if (!this.func_176533_e(p_180650_1_, p_180650_2_))
                     {
-                        if (!p_180650_1_.func_180495_p(p_180650_2_.func_177977_b()).func_185896_q() || i > 3)
+                        if (!p_180650_1_.getBlockState(p_180650_2_.getBlockPosBelow()).func_185896_q() || i > 3)
                         {
                             p_180650_1_.func_175698_g(p_180650_2_);
                         }
@@ -170,7 +170,7 @@
                         return;
                     }
 
-                    if (!this.func_176535_e(p_180650_1_, p_180650_2_.func_177977_b()) && i == 15 && p_180650_4_.nextInt(4) == 0)
+                    if (!this.func_176535_e(p_180650_1_, p_180650_2_.getBlockPosBelow()) && i == 15 && p_180650_4_.nextInt(4) == 0)
                     {
                         p_180650_1_.func_175698_g(p_180650_2_);
                         return;
@@ -185,12 +185,12 @@
                     j = -50;
                 }
 
-                this.func_176536_a(p_180650_1_, p_180650_2_.func_177974_f(), 300 + j, p_180650_4_, i);
-                this.func_176536_a(p_180650_1_, p_180650_2_.func_177976_e(), 300 + j, p_180650_4_, i);
-                this.func_176536_a(p_180650_1_, p_180650_2_.func_177977_b(), 250 + j, p_180650_4_, i);
-                this.func_176536_a(p_180650_1_, p_180650_2_.func_177984_a(), 250 + j, p_180650_4_, i);
-                this.func_176536_a(p_180650_1_, p_180650_2_.func_177978_c(), 300 + j, p_180650_4_, i);
-                this.func_176536_a(p_180650_1_, p_180650_2_.func_177968_d(), 300 + j, p_180650_4_, i);
+                this.func_176536_a(p_180650_1_, p_180650_2_.getBlockPosEast(), 300 + j, p_180650_4_, i);
+                this.func_176536_a(p_180650_1_, p_180650_2_.getBlockPosWest(), 300 + j, p_180650_4_, i);
+                this.func_176536_a(p_180650_1_, p_180650_2_.getBlockPosBelow(), 250 + j, p_180650_4_, i);
+                this.func_176536_a(p_180650_1_, p_180650_2_.getBlockPosAbove(), 250 + j, p_180650_4_, i);
+                this.func_176536_a(p_180650_1_, p_180650_2_.getBlockPosNorth(), 300 + j, p_180650_4_, i);
+                this.func_176536_a(p_180650_1_, p_180650_2_.getBlockPosSouth(), 300 + j, p_180650_4_, i);
 
                 for (int k = -1; k <= 1; ++k)
                 {
@@ -241,7 +241,7 @@
 
     protected boolean func_176537_d(World p_176537_1_, BlockPos p_176537_2_)
     {
-        return p_176537_1_.func_175727_C(p_176537_2_) || p_176537_1_.func_175727_C(p_176537_2_.func_177976_e()) || p_176537_1_.func_175727_C(p_176537_2_.func_177974_f()) || p_176537_1_.func_175727_C(p_176537_2_.func_177978_c()) || p_176537_1_.func_175727_C(p_176537_2_.func_177968_d());
+        return p_176537_1_.func_175727_C(p_176537_2_) || p_176537_1_.func_175727_C(p_176537_2_.getBlockPosWest()) || p_176537_1_.func_175727_C(p_176537_2_.getBlockPosEast()) || p_176537_1_.func_175727_C(p_176537_2_.getBlockPosNorth()) || p_176537_1_.func_175727_C(p_176537_2_.getBlockPosSouth());
     }
 
     public boolean requiresUpdates()
@@ -263,11 +263,11 @@
 
     private void func_176536_a(World p_176536_1_, BlockPos p_176536_2_, int p_176536_3_, Random p_176536_4_, int p_176536_5_)
     {
-        int i = this.func_176532_c(p_176536_1_.func_180495_p(p_176536_2_).func_177230_c());
+        int i = this.func_176532_c(p_176536_1_.getBlockState(p_176536_2_).getBlock());
 
         if (p_176536_4_.nextInt(p_176536_3_) < i)
         {
-            IBlockState iblockstate = p_176536_1_.func_180495_p(p_176536_2_);
+            IBlockState iblockstate = p_176536_1_.getBlockState(p_176536_2_);
 
             if (p_176536_4_.nextInt(p_176536_5_ + 10) < 5 && !p_176536_1_.func_175727_C(p_176536_2_))
             {
@@ -278,14 +278,14 @@
                     j = 15;
                 }
 
-                p_176536_1_.func_180501_a(p_176536_2_, this.func_176223_P().func_177226_a(field_176543_a, Integer.valueOf(j)), 3);
+                p_176536_1_.func_180501_a(p_176536_2_, this.getBlockState().func_177226_a(field_176543_a, Integer.valueOf(j)), 3);
             }
             else
             {
                 p_176536_1_.func_175698_g(p_176536_2_);
             }
 
-            if (iblockstate.func_177230_c() == Blocks.tnt)
+            if (iblockstate.getBlock() == Blocks.tnt)
             {
                 Blocks.tnt.func_176206_d(p_176536_1_, p_176536_2_, iblockstate.func_177226_a(BlockTNT.field_176246_a, Boolean.valueOf(true)));
             }
@@ -296,7 +296,7 @@
     {
         for (EnumFacing enumfacing : EnumFacing.values())
         {
-            if (this.func_176535_e(p_176533_1_, p_176533_2_.func_177972_a(enumfacing)))
+            if (this.func_176535_e(p_176533_1_, p_176533_2_.getBlockPosRelative(enumfacing)))
             {
                 return true;
             }
@@ -307,7 +307,7 @@
 
     private int func_176538_m(World p_176538_1_, BlockPos p_176538_2_)
     {
-        if (!p_176538_1_.func_175623_d(p_176538_2_))
+        if (!p_176538_1_.isEmpty(p_176538_2_))
         {
             return 0;
         }
@@ -317,7 +317,7 @@
 
             for (EnumFacing enumfacing : EnumFacing.values())
             {
-                i = Math.max(this.func_176534_d(p_176538_1_.func_180495_p(p_176538_2_.func_177972_a(enumfacing)).func_177230_c()), i);
+                i = Math.max(this.func_176534_d(p_176538_1_.getBlockState(p_176538_2_.getBlockPosRelative(enumfacing)).getBlock()), i);
             }
 
             return i;
@@ -334,17 +334,17 @@
 
     public boolean func_176535_e(IBlockAccess p_176535_1_, BlockPos p_176535_2_)
     {
-        return this.func_176534_d(p_176535_1_.func_180495_p(p_176535_2_).func_177230_c()) > 0;
+        return this.func_176534_d(p_176535_1_.getBlockState(p_176535_2_).getBlock()) > 0;
     }
 
     public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
     {
-        return p_176196_1_.func_180495_p(p_176196_2_.func_177977_b()).func_185896_q() || this.func_176533_e(p_176196_1_, p_176196_2_);
+        return p_176196_1_.getBlockState(p_176196_2_.getBlockPosBelow()).func_185896_q() || this.func_176533_e(p_176196_1_, p_176196_2_);
     }
 
     public void func_189540_a(IBlockState p_189540_1_, World p_189540_2_, BlockPos p_189540_3_, Block p_189540_4_)
     {
-        if (!p_189540_2_.func_180495_p(p_189540_3_.func_177977_b()).func_185896_q() && !this.func_176533_e(p_189540_2_, p_189540_3_))
+        if (!p_189540_2_.getBlockState(p_189540_3_.getBlockPosBelow()).func_185896_q() && !this.func_176533_e(p_189540_2_, p_189540_3_))
         {
             p_189540_2_.func_175698_g(p_189540_3_);
         }
@@ -354,7 +354,7 @@
     {
         if (p_176213_1_.provider.func_186058_p().func_186068_a() > 0 || !Blocks.portal.func_176548_d(p_176213_1_, p_176213_2_))
         {
-            if (!p_176213_1_.func_180495_p(p_176213_2_.func_177977_b()).func_185896_q() && !this.func_176533_e(p_176213_1_, p_176213_2_))
+            if (!p_176213_1_.getBlockState(p_176213_2_.getBlockPosBelow()).func_185896_q() && !this.func_176533_e(p_176213_1_, p_176213_2_))
             {
                 p_176213_1_.func_175698_g(p_176213_2_);
             }
@@ -369,63 +369,63 @@
     {
         if (p_180655_4_.nextInt(24) == 0)
         {
-            p_180655_2_.func_184134_a((double)((float)p_180655_3_.func_177958_n() + 0.5F), (double)((float)p_180655_3_.func_177956_o() + 0.5F), (double)((float)p_180655_3_.func_177952_p() + 0.5F), SoundEvents.field_187643_bs, SoundCategory.BLOCKS, 1.0F + p_180655_4_.nextFloat(), p_180655_4_.nextFloat() * 0.7F + 0.3F, false);
+            p_180655_2_.func_184134_a((double)((float)p_180655_3_.getX() + 0.5F), (double)((float)p_180655_3_.getY() + 0.5F), (double)((float)p_180655_3_.getZ() + 0.5F), SoundEvents.field_187643_bs, SoundCategory.BLOCKS, 1.0F + p_180655_4_.nextFloat(), p_180655_4_.nextFloat() * 0.7F + 0.3F, false);
         }
 
-        if (!p_180655_2_.func_180495_p(p_180655_3_.func_177977_b()).func_185896_q() && !Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.func_177977_b()))
+        if (!p_180655_2_.getBlockState(p_180655_3_.getBlockPosBelow()).func_185896_q() && !Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.getBlockPosBelow()))
         {
-            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.func_177976_e()))
+            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.getBlockPosWest()))
             {
                 for (int j = 0; j < 2; ++j)
                 {
-                    double d3 = (double)p_180655_3_.func_177958_n() + p_180655_4_.nextDouble() * 0.10000000149011612D;
-                    double d8 = (double)p_180655_3_.func_177956_o() + p_180655_4_.nextDouble();
-                    double d13 = (double)p_180655_3_.func_177952_p() + p_180655_4_.nextDouble();
-                    p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, d3, d8, d13, 0.0D, 0.0D, 0.0D, new int[0]);
+                    double d3 = (double)p_180655_3_.getX() + p_180655_4_.nextDouble() * 0.10000000149011612D;
+                    double d8 = (double)p_180655_3_.getY() + p_180655_4_.nextDouble();
+                    double d13 = (double)p_180655_3_.getZ() + p_180655_4_.nextDouble();
+                    p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, d3, d8, d13, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
             }
 
-            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.func_177974_f()))
+            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.getBlockPosEast()))
             {
                 for (int k = 0; k < 2; ++k)
                 {
-                    double d4 = (double)(p_180655_3_.func_177958_n() + 1) - p_180655_4_.nextDouble() * 0.10000000149011612D;
-                    double d9 = (double)p_180655_3_.func_177956_o() + p_180655_4_.nextDouble();
-                    double d14 = (double)p_180655_3_.func_177952_p() + p_180655_4_.nextDouble();
-                    p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, d4, d9, d14, 0.0D, 0.0D, 0.0D, new int[0]);
+                    double d4 = (double)(p_180655_3_.getX() + 1) - p_180655_4_.nextDouble() * 0.10000000149011612D;
+                    double d9 = (double)p_180655_3_.getY() + p_180655_4_.nextDouble();
+                    double d14 = (double)p_180655_3_.getZ() + p_180655_4_.nextDouble();
+                    p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, d4, d9, d14, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
             }
 
-            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.func_177978_c()))
+            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.getBlockPosNorth()))
             {
                 for (int l = 0; l < 2; ++l)
                 {
-                    double d5 = (double)p_180655_3_.func_177958_n() + p_180655_4_.nextDouble();
-                    double d10 = (double)p_180655_3_.func_177956_o() + p_180655_4_.nextDouble();
-                    double d15 = (double)p_180655_3_.func_177952_p() + p_180655_4_.nextDouble() * 0.10000000149011612D;
-                    p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, d5, d10, d15, 0.0D, 0.0D, 0.0D, new int[0]);
+                    double d5 = (double)p_180655_3_.getX() + p_180655_4_.nextDouble();
+                    double d10 = (double)p_180655_3_.getY() + p_180655_4_.nextDouble();
+                    double d15 = (double)p_180655_3_.getZ() + p_180655_4_.nextDouble() * 0.10000000149011612D;
+                    p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, d5, d10, d15, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
             }
 
-            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.func_177968_d()))
+            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.getBlockPosSouth()))
             {
                 for (int i1 = 0; i1 < 2; ++i1)
                 {
-                    double d6 = (double)p_180655_3_.func_177958_n() + p_180655_4_.nextDouble();
-                    double d11 = (double)p_180655_3_.func_177956_o() + p_180655_4_.nextDouble();
-                    double d16 = (double)(p_180655_3_.func_177952_p() + 1) - p_180655_4_.nextDouble() * 0.10000000149011612D;
-                    p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, d6, d11, d16, 0.0D, 0.0D, 0.0D, new int[0]);
+                    double d6 = (double)p_180655_3_.getX() + p_180655_4_.nextDouble();
+                    double d11 = (double)p_180655_3_.getY() + p_180655_4_.nextDouble();
+                    double d16 = (double)(p_180655_3_.getZ() + 1) - p_180655_4_.nextDouble() * 0.10000000149011612D;
+                    p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, d6, d11, d16, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
             }
 
-            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.func_177984_a()))
+            if (Blocks.fire.func_176535_e(p_180655_2_, p_180655_3_.getBlockPosAbove()))
             {
                 for (int j1 = 0; j1 < 2; ++j1)
                 {
-                    double d7 = (double)p_180655_3_.func_177958_n() + p_180655_4_.nextDouble();
-                    double d12 = (double)(p_180655_3_.func_177956_o() + 1) - p_180655_4_.nextDouble() * 0.10000000149011612D;
-                    double d17 = (double)p_180655_3_.func_177952_p() + p_180655_4_.nextDouble();
-                    p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, d7, d12, d17, 0.0D, 0.0D, 0.0D, new int[0]);
+                    double d7 = (double)p_180655_3_.getX() + p_180655_4_.nextDouble();
+                    double d12 = (double)(p_180655_3_.getY() + 1) - p_180655_4_.nextDouble() * 0.10000000149011612D;
+                    double d17 = (double)p_180655_3_.getZ() + p_180655_4_.nextDouble();
+                    p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, d7, d12, d17, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
             }
         }
@@ -433,10 +433,10 @@
         {
             for (int i = 0; i < 3; ++i)
             {
-                double d0 = (double)p_180655_3_.func_177958_n() + p_180655_4_.nextDouble();
-                double d1 = (double)p_180655_3_.func_177956_o() + p_180655_4_.nextDouble() * 0.5D + 0.5D;
-                double d2 = (double)p_180655_3_.func_177952_p() + p_180655_4_.nextDouble();
-                p_180655_2_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
+                double d0 = (double)p_180655_3_.getX() + p_180655_4_.nextDouble();
+                double d1 = (double)p_180655_3_.getY() + p_180655_4_.nextDouble() * 0.5D + 0.5D;
+                double d2 = (double)p_180655_3_.getZ() + p_180655_4_.nextDouble();
+                p_180655_2_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
             }
         }
     }
@@ -453,7 +453,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176543_a, Integer.valueOf(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176543_a, Integer.valueOf(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
