--- a/net/minecraft/block/BlockDeadBush.java
+++ b/net/minecraft/block/BlockDeadBush.java
@@ -38,7 +38,7 @@
 
     protected boolean func_185514_i(IBlockState p_185514_1_)
     {
-        return p_185514_1_.func_177230_c() == Blocks.sand || p_185514_1_.func_177230_c() == Blocks.hardened_clay || p_185514_1_.func_177230_c() == Blocks.stained_hardened_clay || p_185514_1_.func_177230_c() == Blocks.dirt;
+        return p_185514_1_.getBlock() == Blocks.sand || p_185514_1_.getBlock() == Blocks.hardened_clay || p_185514_1_.getBlock() == Blocks.stained_hardened_clay || p_185514_1_.getBlock() == Blocks.dirt;
     }
 
     public boolean func_176200_f(IBlockAccess p_176200_1_, BlockPos p_176200_2_)
