--- a/net/minecraft/block/BlockDragonEgg.java
+++ b/net/minecraft/block/BlockDragonEgg.java
@@ -47,27 +47,27 @@
 
     private void func_180683_d(World p_180683_1_, BlockPos p_180683_2_)
     {
-        if (BlockFalling.func_185759_i(p_180683_1_.func_180495_p(p_180683_2_.func_177977_b())) && p_180683_2_.func_177956_o() >= 0)
+        if (BlockFalling.func_185759_i(p_180683_1_.getBlockState(p_180683_2_.getBlockPosBelow())) && p_180683_2_.getY() >= 0)
         {
             int i = 32;
 
             if (!BlockFalling.fallInstantly && p_180683_1_.func_175707_a(p_180683_2_.func_177982_a(-32, -32, -32), p_180683_2_.func_177982_a(32, 32, 32)))
             {
-                p_180683_1_.spawnEntityInWorld(new EntityFallingBlock(p_180683_1_, (double)((float)p_180683_2_.func_177958_n() + 0.5F), (double)p_180683_2_.func_177956_o(), (double)((float)p_180683_2_.func_177952_p() + 0.5F), this.func_176223_P()));
+                p_180683_1_.spawnEntityInWorld(new EntityFallingBlock(p_180683_1_, (double)((float)p_180683_2_.getX() + 0.5F), (double)p_180683_2_.getY(), (double)((float)p_180683_2_.getZ() + 0.5F), this.getBlockState()));
             }
             else
             {
                 p_180683_1_.func_175698_g(p_180683_2_);
                 BlockPos blockpos;
 
-                for (blockpos = p_180683_2_; BlockFalling.func_185759_i(p_180683_1_.func_180495_p(blockpos)) && blockpos.func_177956_o() > 0; blockpos = blockpos.func_177977_b())
+                for (blockpos = p_180683_2_; BlockFalling.func_185759_i(p_180683_1_.getBlockState(blockpos)) && blockpos.getY() > 0; blockpos = blockpos.getBlockPosBelow())
                 {
                     ;
                 }
 
-                if (blockpos.func_177956_o() > 0)
+                if (blockpos.getY() > 0)
                 {
-                    p_180683_1_.func_180501_a(blockpos, this.func_176223_P(), 2);
+                    p_180683_1_.func_180501_a(blockpos, this.getBlockState(), 2);
                 }
             }
         }
@@ -86,15 +86,15 @@
 
     private void func_180684_e(World p_180684_1_, BlockPos p_180684_2_)
     {
-        IBlockState iblockstate = p_180684_1_.func_180495_p(p_180684_2_);
+        IBlockState iblockstate = p_180684_1_.getBlockState(p_180684_2_);
 
-        if (iblockstate.func_177230_c() == this)
+        if (iblockstate.getBlock() == this)
         {
             for (int i = 0; i < 1000; ++i)
             {
                 BlockPos blockpos = p_180684_2_.func_177982_a(p_180684_1_.rand.nextInt(16) - p_180684_1_.rand.nextInt(16), p_180684_1_.rand.nextInt(8) - p_180684_1_.rand.nextInt(8), p_180684_1_.rand.nextInt(16) - p_180684_1_.rand.nextInt(16));
 
-                if (p_180684_1_.func_180495_p(blockpos).func_177230_c().blockMaterial == Material.air)
+                if (p_180684_1_.getBlockState(blockpos).getBlock().blockMaterial == Material.air)
                 {
                     if (p_180684_1_.isRemote)
                     {
@@ -104,10 +104,10 @@
                             float f = (p_180684_1_.rand.nextFloat() - 0.5F) * 0.2F;
                             float f1 = (p_180684_1_.rand.nextFloat() - 0.5F) * 0.2F;
                             float f2 = (p_180684_1_.rand.nextFloat() - 0.5F) * 0.2F;
-                            double d1 = (double)blockpos.func_177958_n() + (double)(p_180684_2_.func_177958_n() - blockpos.func_177958_n()) * d0 + (p_180684_1_.rand.nextDouble() - 0.5D) + 0.5D;
-                            double d2 = (double)blockpos.func_177956_o() + (double)(p_180684_2_.func_177956_o() - blockpos.func_177956_o()) * d0 + p_180684_1_.rand.nextDouble() - 0.5D;
-                            double d3 = (double)blockpos.func_177952_p() + (double)(p_180684_2_.func_177952_p() - blockpos.func_177952_p()) * d0 + (p_180684_1_.rand.nextDouble() - 0.5D) + 0.5D;
-                            p_180684_1_.func_175688_a(EnumParticleTypes.PORTAL, d1, d2, d3, (double)f, (double)f1, (double)f2, new int[0]);
+                            double d1 = (double)blockpos.getX() + (double)(p_180684_2_.getX() - blockpos.getX()) * d0 + (p_180684_1_.rand.nextDouble() - 0.5D) + 0.5D;
+                            double d2 = (double)blockpos.getY() + (double)(p_180684_2_.getY() - blockpos.getY()) * d0 + p_180684_1_.rand.nextDouble() - 0.5D;
+                            double d3 = (double)blockpos.getZ() + (double)(p_180684_2_.getZ() - blockpos.getZ()) * d0 + (p_180684_1_.rand.nextDouble() - 0.5D) + 0.5D;
+                            p_180684_1_.spawnParticle(EnumParticleTypes.PORTAL, d1, d2, d3, (double)f, (double)f1, (double)f2, new int[0]);
                         }
                     }
                     else
