--- a/net/minecraft/block/BlockMycelium.java
+++ b/net/minecraft/block/BlockMycelium.java
@@ -30,7 +30,7 @@
 
     public IBlockState func_176221_a(IBlockState p_176221_1_, IBlockAccess p_176221_2_, BlockPos p_176221_3_)
     {
-        Block block = p_176221_2_.func_180495_p(p_176221_3_.func_177984_a()).func_177230_c();
+        Block block = p_176221_2_.getBlockState(p_176221_3_.getBlockPosAbove()).getBlock();
         return p_176221_1_.func_177226_a(field_176384_a, Boolean.valueOf(block == Blocks.snow || block == Blocks.snow_layer));
     }
 
@@ -38,23 +38,23 @@
     {
         if (!p_180650_1_.isRemote)
         {
-            if (p_180650_1_.func_175671_l(p_180650_2_.func_177984_a()) < 4 && p_180650_1_.func_180495_p(p_180650_2_.func_177984_a()).func_185891_c() > 2)
+            if (p_180650_1_.func_175671_l(p_180650_2_.getBlockPosAbove()) < 4 && p_180650_1_.getBlockState(p_180650_2_.getBlockPosAbove()).func_185891_c() > 2)
             {
-                p_180650_1_.func_175656_a(p_180650_2_, Blocks.dirt.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT));
+                p_180650_1_.func_175656_a(p_180650_2_, Blocks.dirt.getBlockState().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT));
             }
             else
             {
-                if (p_180650_1_.func_175671_l(p_180650_2_.func_177984_a()) >= 9)
+                if (p_180650_1_.func_175671_l(p_180650_2_.getBlockPosAbove()) >= 9)
                 {
                     for (int i = 0; i < 4; ++i)
                     {
                         BlockPos blockpos = p_180650_2_.func_177982_a(p_180650_4_.nextInt(3) - 1, p_180650_4_.nextInt(5) - 3, p_180650_4_.nextInt(3) - 1);
-                        IBlockState iblockstate = p_180650_1_.func_180495_p(blockpos);
-                        IBlockState iblockstate1 = p_180650_1_.func_180495_p(blockpos.func_177984_a());
+                        IBlockState iblockstate = p_180650_1_.getBlockState(blockpos);
+                        IBlockState iblockstate1 = p_180650_1_.getBlockState(blockpos.getBlockPosAbove());
 
-                        if (iblockstate.func_177230_c() == Blocks.dirt && iblockstate.func_177229_b(BlockDirt.field_176386_a) == BlockDirt.DirtType.DIRT && p_180650_1_.func_175671_l(blockpos.func_177984_a()) >= 4 && iblockstate1.func_185891_c() <= 2)
+                        if (iblockstate.getBlock() == Blocks.dirt && iblockstate.func_177229_b(BlockDirt.field_176386_a) == BlockDirt.DirtType.DIRT && p_180650_1_.func_175671_l(blockpos.getBlockPosAbove()) >= 4 && iblockstate1.func_185891_c() <= 2)
                         {
-                            p_180650_1_.func_175656_a(blockpos, this.func_176223_P());
+                            p_180650_1_.func_175656_a(blockpos, this.getBlockState());
                         }
                     }
                 }
@@ -68,14 +68,14 @@
 
         if (p_180655_4_.nextInt(10) == 0)
         {
-            p_180655_2_.func_175688_a(EnumParticleTypes.TOWN_AURA, (double)((float)p_180655_3_.func_177958_n() + p_180655_4_.nextFloat()), (double)((float)p_180655_3_.func_177956_o() + 1.1F), (double)((float)p_180655_3_.func_177952_p() + p_180655_4_.nextFloat()), 0.0D, 0.0D, 0.0D, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.TOWN_AURA, (double)((float)p_180655_3_.getX() + p_180655_4_.nextFloat()), (double)((float)p_180655_3_.getY() + 1.1F), (double)((float)p_180655_3_.getZ() + p_180655_4_.nextFloat()), 0.0D, 0.0D, 0.0D, new int[0]);
         }
     }
 
     @Nullable
     public Item func_180660_a(IBlockState p_180660_1_, Random p_180660_2_, int p_180660_3_)
     {
-        return Blocks.dirt.func_180660_a(Blocks.dirt.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT), p_180660_2_, p_180660_3_);
+        return Blocks.dirt.func_180660_a(Blocks.dirt.getBlockState().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT), p_180660_2_, p_180660_3_);
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
