--- a/net/minecraft/block/BlockRedstoneLight.java
+++ b/net/minecraft/block/BlockRedstoneLight.java
@@ -31,11 +31,11 @@
         {
             if (this.field_150171_a && !p_176213_1_.func_175640_z(p_176213_2_))
             {
-                p_176213_1_.func_180501_a(p_176213_2_, Blocks.redstone_lamp.func_176223_P(), 2);
+                p_176213_1_.func_180501_a(p_176213_2_, Blocks.redstone_lamp.getBlockState(), 2);
             }
             else if (!this.field_150171_a && p_176213_1_.func_175640_z(p_176213_2_))
             {
-                p_176213_1_.func_180501_a(p_176213_2_, Blocks.lit_redstone_lamp.func_176223_P(), 2);
+                p_176213_1_.func_180501_a(p_176213_2_, Blocks.lit_redstone_lamp.getBlockState(), 2);
             }
         }
     }
@@ -50,7 +50,7 @@
             }
             else if (!this.field_150171_a && p_189540_2_.func_175640_z(p_189540_3_))
             {
-                p_189540_2_.func_180501_a(p_189540_3_, Blocks.lit_redstone_lamp.func_176223_P(), 2);
+                p_189540_2_.func_180501_a(p_189540_3_, Blocks.lit_redstone_lamp.getBlockState(), 2);
             }
         }
     }
@@ -61,7 +61,7 @@
         {
             if (this.field_150171_a && !p_180650_1_.func_175640_z(p_180650_2_))
             {
-                p_180650_1_.func_180501_a(p_180650_2_, Blocks.redstone_lamp.func_176223_P(), 2);
+                p_180650_1_.func_180501_a(p_180650_2_, Blocks.redstone_lamp.getBlockState(), 2);
             }
         }
     }
