--- a/net/minecraft/block/BlockFrostedIce.java
+++ b/net/minecraft/block/BlockFrostedIce.java
@@ -28,7 +28,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_185682_a, Integer.valueOf(MathHelper.clamp_int(p_176203_1_, 0, 3)));
+        return this.getBlockState().func_177226_a(field_185682_a, Integer.valueOf(MathHelper.clamp_int(p_176203_1_, 0, 3)));
     }
 
     public void func_180650_b(World p_180650_1_, BlockPos p_180650_2_, IBlockState p_180650_3_, Random p_180650_4_)
@@ -62,7 +62,7 @@
 
         for (EnumFacing enumfacing : EnumFacing.values())
         {
-            if (p_185680_1_.func_180495_p(p_185680_2_.func_177972_a(enumfacing)).func_177230_c() == this)
+            if (p_185680_1_.getBlockState(p_185680_2_.getBlockPosRelative(enumfacing)).getBlock() == this)
             {
                 ++i;
 
@@ -93,10 +93,10 @@
             {
                 for (EnumFacing enumfacing : EnumFacing.values())
                 {
-                    BlockPos blockpos = p_185681_2_.func_177972_a(enumfacing);
-                    IBlockState iblockstate = p_185681_1_.func_180495_p(blockpos);
+                    BlockPos blockpos = p_185681_2_.getBlockPosRelative(enumfacing);
+                    IBlockState iblockstate = p_185681_1_.getBlockState(blockpos);
 
-                    if (iblockstate.func_177230_c() == this)
+                    if (iblockstate.getBlock() == this)
                     {
                         this.func_185681_a(p_185681_1_, blockpos, iblockstate, p_185681_4_, false);
                     }
