--- a/net/minecraft/block/BlockColored.java
+++ b/net/minecraft/block/BlockColored.java
@@ -46,7 +46,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176581_a, EnumDyeColor.func_176764_b(p_176203_1_));
+        return this.getBlockState().func_177226_a(field_176581_a, EnumDyeColor.func_176764_b(p_176203_1_));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
