--- a/net/minecraft/block/BlockEndRod.java
+++ b/net/minecraft/block/BlockEndRod.java
@@ -75,19 +75,19 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        IBlockState iblockstate = p_180642_1_.func_180495_p(p_180642_2_.func_177972_a(p_180642_3_.func_176734_d()));
+        IBlockState iblockstate = p_180642_1_.getBlockState(p_180642_2_.getBlockPosRelative(p_180642_3_.func_176734_d()));
 
-        if (iblockstate.func_177230_c() == Blocks.field_185764_cQ)
+        if (iblockstate.getBlock() == Blocks.field_185764_cQ)
         {
             EnumFacing enumfacing = (EnumFacing)iblockstate.func_177229_b(field_176387_N);
 
             if (enumfacing == p_180642_3_)
             {
-                return this.func_176223_P().func_177226_a(field_176387_N, p_180642_3_.func_176734_d());
+                return this.getBlockState().func_177226_a(field_176387_N, p_180642_3_.func_176734_d());
             }
         }
 
-        return this.func_176223_P().func_177226_a(field_176387_N, p_180642_3_);
+        return this.getBlockState().func_177226_a(field_176387_N, p_180642_3_);
     }
 
     public void func_176213_c(World p_176213_1_, BlockPos p_176213_2_, IBlockState p_176213_3_)
@@ -101,14 +101,14 @@
     public void func_180655_c(IBlockState p_180655_1_, World p_180655_2_, BlockPos p_180655_3_, Random p_180655_4_)
     {
         EnumFacing enumfacing = (EnumFacing)p_180655_1_.func_177229_b(field_176387_N);
-        double d0 = (double)p_180655_3_.func_177958_n() + 0.55D - (double)(p_180655_4_.nextFloat() * 0.1F);
-        double d1 = (double)p_180655_3_.func_177956_o() + 0.55D - (double)(p_180655_4_.nextFloat() * 0.1F);
-        double d2 = (double)p_180655_3_.func_177952_p() + 0.55D - (double)(p_180655_4_.nextFloat() * 0.1F);
+        double d0 = (double)p_180655_3_.getX() + 0.55D - (double)(p_180655_4_.nextFloat() * 0.1F);
+        double d1 = (double)p_180655_3_.getY() + 0.55D - (double)(p_180655_4_.nextFloat() * 0.1F);
+        double d2 = (double)p_180655_3_.getZ() + 0.55D - (double)(p_180655_4_.nextFloat() * 0.1F);
         double d3 = (double)(0.4F - (p_180655_4_.nextFloat() + p_180655_4_.nextFloat()) * 0.4F);
 
         if (p_180655_4_.nextInt(5) == 0)
         {
-            p_180655_2_.func_175688_a(EnumParticleTypes.END_ROD, d0 + (double)enumfacing.getFrontOffsetX() * d3, d1 + (double)enumfacing.getFrontOffsetY() * d3, d2 + (double)enumfacing.getFrontOffsetZ() * d3, p_180655_4_.nextGaussian() * 0.005D, p_180655_4_.nextGaussian() * 0.005D, p_180655_4_.nextGaussian() * 0.005D, new int[0]);
+            p_180655_2_.spawnParticle(EnumParticleTypes.END_ROD, d0 + (double)enumfacing.getFrontOffsetX() * d3, d1 + (double)enumfacing.getFrontOffsetY() * d3, d2 + (double)enumfacing.getFrontOffsetZ() * d3, p_180655_4_.nextGaussian() * 0.005D, p_180655_4_.nextGaussian() * 0.005D, p_180655_4_.nextGaussian() * 0.005D, new int[0]);
         }
     }
 
@@ -119,7 +119,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        IBlockState iblockstate = this.func_176223_P();
+        IBlockState iblockstate = this.getBlockState();
         iblockstate = iblockstate.func_177226_a(field_176387_N, EnumFacing.getFront(p_176203_1_));
         return iblockstate;
     }
