--- a/net/minecraft/block/BlockLever.java
+++ b/net/minecraft/block/BlockLever.java
@@ -82,7 +82,7 @@
 
     public IBlockState func_180642_a(World p_180642_1_, BlockPos p_180642_2_, EnumFacing p_180642_3_, float p_180642_4_, float p_180642_5_, float p_180642_6_, int p_180642_7_, EntityLivingBase p_180642_8_)
     {
-        IBlockState iblockstate = this.func_176223_P().func_177226_a(field_176359_b, Boolean.valueOf(false));
+        IBlockState iblockstate = this.getBlockState().func_177226_a(field_176359_b, Boolean.valueOf(false));
 
         if (func_181090_a(p_180642_1_, p_180642_2_, p_180642_3_.func_176734_d()))
         {
@@ -98,7 +98,7 @@
                 }
             }
 
-            if (p_180642_1_.func_180495_p(p_180642_2_.func_177977_b()).func_185896_q())
+            if (p_180642_1_.getBlockState(p_180642_2_.getBlockPosBelow()).func_185896_q())
             {
                 return iblockstate.func_177226_a(field_176360_a, BlockLever.EnumOrientation.func_176856_a(EnumFacing.UP, p_180642_8_.func_174811_aO()));
             }
@@ -173,7 +173,7 @@
             p_180639_1_.func_184133_a((EntityPlayer)null, p_180639_2_, SoundEvents.field_187750_dc, SoundCategory.BLOCKS, 0.3F, f);
             p_180639_1_.func_175685_c(p_180639_2_, this);
             EnumFacing enumfacing = ((BlockLever.EnumOrientation)p_180639_3_.func_177229_b(field_176360_a)).func_176852_c();
-            p_180639_1_.func_175685_c(p_180639_2_.func_177972_a(enumfacing.func_176734_d()), this);
+            p_180639_1_.func_175685_c(p_180639_2_.getBlockPosRelative(enumfacing.func_176734_d()), this);
             return true;
         }
     }
@@ -184,7 +184,7 @@
         {
             p_180663_1_.func_175685_c(p_180663_2_, this);
             EnumFacing enumfacing = ((BlockLever.EnumOrientation)p_180663_3_.func_177229_b(field_176360_a)).func_176852_c();
-            p_180663_1_.func_175685_c(p_180663_2_.func_177972_a(enumfacing.func_176734_d()), this);
+            p_180663_1_.func_175685_c(p_180663_2_.getBlockPosRelative(enumfacing.func_176734_d()), this);
         }
 
         super.func_180663_b(p_180663_1_, p_180663_2_, p_180663_3_);
@@ -210,7 +210,7 @@
 
     public IBlockState func_176203_a(int p_176203_1_)
     {
-        return this.func_176223_P().func_177226_a(field_176360_a, BlockLever.EnumOrientation.func_176853_a(p_176203_1_ & 7)).func_177226_a(field_176359_b, Boolean.valueOf((p_176203_1_ & 8) > 0));
+        return this.getBlockState().func_177226_a(field_176360_a, BlockLever.EnumOrientation.func_176853_a(p_176203_1_ & 7)).func_177226_a(field_176359_b, Boolean.valueOf((p_176203_1_ & 8) > 0));
     }
 
     public int func_176201_c(IBlockState p_176201_1_)
