--- a/net/minecraft/block/BlockChorusPlant.java
+++ b/net/minecraft/block/BlockChorusPlant.java
@@ -38,12 +38,12 @@
 
     public IBlockState func_176221_a(IBlockState p_176221_1_, IBlockAccess p_176221_2_, BlockPos p_176221_3_)
     {
-        Block block = p_176221_2_.func_180495_p(p_176221_3_.func_177977_b()).func_177230_c();
-        Block block1 = p_176221_2_.func_180495_p(p_176221_3_.func_177984_a()).func_177230_c();
-        Block block2 = p_176221_2_.func_180495_p(p_176221_3_.func_177978_c()).func_177230_c();
-        Block block3 = p_176221_2_.func_180495_p(p_176221_3_.func_177974_f()).func_177230_c();
-        Block block4 = p_176221_2_.func_180495_p(p_176221_3_.func_177968_d()).func_177230_c();
-        Block block5 = p_176221_2_.func_180495_p(p_176221_3_.func_177976_e()).func_177230_c();
+        Block block = p_176221_2_.getBlockState(p_176221_3_.getBlockPosBelow()).getBlock();
+        Block block1 = p_176221_2_.getBlockState(p_176221_3_.getBlockPosAbove()).getBlock();
+        Block block2 = p_176221_2_.getBlockState(p_176221_3_.getBlockPosNorth()).getBlock();
+        Block block3 = p_176221_2_.getBlockState(p_176221_3_.getBlockPosEast()).getBlock();
+        Block block4 = p_176221_2_.getBlockState(p_176221_3_.getBlockPosSouth()).getBlock();
+        Block block5 = p_176221_2_.getBlockState(p_176221_3_.getBlockPosWest()).getBlock();
         return p_176221_1_.func_177226_a(field_185614_f, Boolean.valueOf(block == this || block == Blocks.field_185766_cS || block == Blocks.end_stone)).func_177226_a(field_185613_e, Boolean.valueOf(block1 == this || block1 == Blocks.field_185766_cS)).func_177226_a(field_185609_a, Boolean.valueOf(block2 == this || block2 == Blocks.field_185766_cS)).func_177226_a(field_185610_b, Boolean.valueOf(block3 == this || block3 == Blocks.field_185766_cS)).func_177226_a(field_185611_c, Boolean.valueOf(block4 == this || block4 == Blocks.field_185766_cS)).func_177226_a(field_185612_d, Boolean.valueOf(block5 == this || block5 == Blocks.field_185766_cS));
     }
 
@@ -150,13 +150,13 @@
 
     public boolean func_185608_b(World p_185608_1_, BlockPos p_185608_2_)
     {
-        boolean flag = p_185608_1_.func_175623_d(p_185608_2_.func_177984_a());
-        boolean flag1 = p_185608_1_.func_175623_d(p_185608_2_.func_177977_b());
+        boolean flag = p_185608_1_.isEmpty(p_185608_2_.getBlockPosAbove());
+        boolean flag1 = p_185608_1_.isEmpty(p_185608_2_.getBlockPosBelow());
 
         for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
         {
-            BlockPos blockpos = p_185608_2_.func_177972_a(enumfacing);
-            Block block = p_185608_1_.func_180495_p(blockpos).func_177230_c();
+            BlockPos blockpos = p_185608_2_.getBlockPosRelative(enumfacing);
+            Block block = p_185608_1_.getBlockState(blockpos).getBlock();
 
             if (block == this)
             {
@@ -165,7 +165,7 @@
                     return false;
                 }
 
-                Block block1 = p_185608_1_.func_180495_p(blockpos.func_177977_b()).func_177230_c();
+                Block block1 = p_185608_1_.getBlockState(blockpos.getBlockPosBelow()).getBlock();
 
                 if (block1 == this || block1 == Blocks.end_stone)
                 {
@@ -174,7 +174,7 @@
             }
         }
 
-        Block block2 = p_185608_1_.func_180495_p(p_185608_2_.func_177977_b()).func_177230_c();
+        Block block2 = p_185608_1_.getBlockState(p_185608_2_.getBlockPosBelow()).getBlock();
         return block2 == this || block2 == Blocks.end_stone;
     }
 
@@ -185,7 +185,7 @@
 
     public boolean func_176225_a(IBlockState p_176225_1_, IBlockAccess p_176225_2_, BlockPos p_176225_3_, EnumFacing p_176225_4_)
     {
-        Block block = p_176225_2_.func_180495_p(p_176225_3_.func_177972_a(p_176225_4_)).func_177230_c();
+        Block block = p_176225_2_.getBlockState(p_176225_3_.getBlockPosRelative(p_176225_4_)).getBlock();
         return block != this && block != Blocks.field_185766_cS && (p_176225_4_ != EnumFacing.DOWN || block != Blocks.end_stone);
     }
 
