--- a/net/minecraft/command/server/CommandTeleport.java
+++ b/net/minecraft/command/server/CommandTeleport.java
@@ -93,7 +93,7 @@
                 f1 = MathHelper.wrapAngleTo180_float(f1);
             }
 
-            p_189862_0_.func_184210_p();
+            p_189862_0_.dismount();
             ((EntityPlayerMP)p_189862_0_).playerNetServerHandler.func_175089_a(p_189862_1_.func_179628_a(), p_189862_2_.func_179628_a(), p_189862_3_.func_179628_a(), f, f1, set);
             p_189862_0_.setRotationYawHead(f);
         }
