--- a/net/minecraft/command/server/CommandTestForBlock.java
+++ b/net/minecraft/command/server/CommandTestForBlock.java
@@ -91,22 +91,22 @@
                         }
                     }
 
-                    IBlockState iblockstate = world.func_180495_p(blockpos);
-                    Block block1 = iblockstate.func_177230_c();
+                    IBlockState iblockstate = world.getBlockState(blockpos);
+                    Block block1 = iblockstate.getBlock();
 
                     if (block1 != block)
                     {
-                        throw new CommandException("commands.testforblock.failed.tile", new Object[] {Integer.valueOf(blockpos.func_177958_n()), Integer.valueOf(blockpos.func_177956_o()), Integer.valueOf(blockpos.func_177952_p()), block1.getLocalizedName(), block.getLocalizedName()});
+                        throw new CommandException("commands.testforblock.failed.tile", new Object[] {Integer.valueOf(blockpos.getX()), Integer.valueOf(blockpos.getY()), Integer.valueOf(blockpos.getZ()), block1.getLocalizedName(), block.getLocalizedName()});
                     }
                     else
                     {
                         if (i > -1)
                         {
-                            int j = iblockstate.func_177230_c().func_176201_c(iblockstate);
+                            int j = iblockstate.getBlock().func_176201_c(iblockstate);
 
                             if (j != i)
                             {
-                                throw new CommandException("commands.testforblock.failed.data", new Object[] {Integer.valueOf(blockpos.func_177958_n()), Integer.valueOf(blockpos.func_177956_o()), Integer.valueOf(blockpos.func_177952_p()), Integer.valueOf(j), Integer.valueOf(i)});
+                                throw new CommandException("commands.testforblock.failed.data", new Object[] {Integer.valueOf(blockpos.getX()), Integer.valueOf(blockpos.getY()), Integer.valueOf(blockpos.getZ()), Integer.valueOf(j), Integer.valueOf(i)});
                             }
                         }
 
@@ -116,19 +116,19 @@
 
                             if (tileentity == null)
                             {
-                                throw new CommandException("commands.testforblock.failed.tileEntity", new Object[] {Integer.valueOf(blockpos.func_177958_n()), Integer.valueOf(blockpos.func_177956_o()), Integer.valueOf(blockpos.func_177952_p())});
+                                throw new CommandException("commands.testforblock.failed.tileEntity", new Object[] {Integer.valueOf(blockpos.getX()), Integer.valueOf(blockpos.getY()), Integer.valueOf(blockpos.getZ())});
                             }
 
                             NBTTagCompound nbttagcompound1 = tileentity.func_189515_b(new NBTTagCompound());
 
                             if (!NBTUtil.func_181123_a(nbttagcompound, nbttagcompound1, true))
                             {
-                                throw new CommandException("commands.testforblock.failed.nbt", new Object[] {Integer.valueOf(blockpos.func_177958_n()), Integer.valueOf(blockpos.func_177956_o()), Integer.valueOf(blockpos.func_177952_p())});
+                                throw new CommandException("commands.testforblock.failed.nbt", new Object[] {Integer.valueOf(blockpos.getX()), Integer.valueOf(blockpos.getY()), Integer.valueOf(blockpos.getZ())});
                             }
                         }
 
                         p_184881_2_.func_174794_a(CommandResultStats.Type.AFFECTED_BLOCKS, 1);
-                        notifyOperators(p_184881_2_, this, "commands.testforblock.success", new Object[] {Integer.valueOf(blockpos.func_177958_n()), Integer.valueOf(blockpos.func_177956_o()), Integer.valueOf(blockpos.func_177952_p())});
+                        notifyOperators(p_184881_2_, this, "commands.testforblock.success", new Object[] {Integer.valueOf(blockpos.getX()), Integer.valueOf(blockpos.getY()), Integer.valueOf(blockpos.getZ())});
                     }
                 }
             }
