--- a/net/minecraft/client/settings/KeyBinding.java
+++ b/net/minecraft/client/settings/KeyBinding.java
@@ -22,6 +22,23 @@
     private boolean pressed;
     private int pressTime;
 
+
+    //MINECRIFT NON STATIC METHODS
+    public static void setKeyBindState(KeyBinding kb, boolean pressed)
+    {
+            if (kb != null)
+            {
+                kb.pressed = pressed;
+                ++kb.pressTime;
+            }       
+    }
+    
+    public void pressKey(){
+    	setKeyBindState(this, true);
+    }
+
+    //END MINECRIFT
+    
     public static void onTick(int keyCode)
     {
         if (keyCode != 0)
@@ -127,7 +144,8 @@
         }
     }
 
-    private void unpressKey()
+    //	VIVE make public
+    public void unpressKey()
     {
         this.pressTime = 0;
         this.pressed = false;
