--- a/net/minecraft/util/CombatTracker.java
+++ b/net/minecraft/util/CombatTracker.java
@@ -37,7 +37,7 @@
 
         if (this.fighter.isOnLadder())
         {
-            Block block = this.fighter.worldObj.func_180495_p(new BlockPos(this.fighter.posX, this.fighter.func_174813_aQ().minY, this.fighter.posZ)).func_177230_c();
+            Block block = this.fighter.worldObj.getBlockState(new BlockPos(this.fighter.posX, this.fighter.func_174813_aQ().minY, this.fighter.posZ)).getBlock();
 
             if (block == Blocks.ladder)
             {
@@ -95,7 +95,7 @@
                     if (itextcomponent2 != null && (itextcomponent1 == null || !itextcomponent2.equals(itextcomponent1)))
                     {
                         Entity entity1 = combatentry.getDamageSrc().getEntity();
-                        ItemStack itemstack1 = entity1 instanceof EntityLivingBase ? ((EntityLivingBase)entity1).func_184614_ca() : null;
+                        ItemStack itemstack1 = entity1 instanceof EntityLivingBase ? ((EntityLivingBase)entity1).getMainHandItem() : null;
 
                         if (itemstack1 != null && itemstack1.hasDisplayName())
                         {
@@ -108,7 +108,7 @@
                     }
                     else if (itextcomponent1 != null)
                     {
-                        ItemStack itemstack = entity instanceof EntityLivingBase ? ((EntityLivingBase)entity).func_184614_ca() : null;
+                        ItemStack itemstack = entity instanceof EntityLivingBase ? ((EntityLivingBase)entity).getMainHandItem() : null;
 
                         if (itemstack != null && itemstack.hasDisplayName())
                         {
