--- a/net/minecraft/util/EntityDamageSourceIndirect.java
+++ b/net/minecraft/util/EntityDamageSourceIndirect.java
@@ -36,7 +36,7 @@
     public ITextComponent getDeathMessage(EntityLivingBase p_151519_1_)
     {
         ITextComponent itextcomponent = this.indirectEntity == null ? this.damageSourceEntity.getFormattedCommandSenderName() : this.indirectEntity.getFormattedCommandSenderName();
-        ItemStack itemstack = this.indirectEntity instanceof EntityLivingBase ? ((EntityLivingBase)this.indirectEntity).func_184614_ca() : null;
+        ItemStack itemstack = this.indirectEntity instanceof EntityLivingBase ? ((EntityLivingBase)this.indirectEntity).getMainHandItem() : null;
         String s = "death.attack." + this.damageType;
         String s1 = s + ".item";
         return itemstack != null && itemstack.hasDisplayName() && I18n.canTranslate(s1) ? new TextComponentTranslation(s1, new Object[] {p_151519_1_.getFormattedCommandSenderName(), itextcomponent, itemstack.func_151000_E()}): new TextComponentTranslation(s, new Object[] {p_151519_1_.getFormattedCommandSenderName(), itextcomponent});
