--- a/net/minecraft/item/ItemBlockSpecial.java
+++ b/net/minecraft/item/ItemBlockSpecial.java
@@ -25,8 +25,8 @@
 
     public EnumActionResult func_180614_a(ItemStack p_180614_1_, EntityPlayer p_180614_2_, World p_180614_3_, BlockPos p_180614_4_, EnumHand p_180614_5_, EnumFacing p_180614_6_, float p_180614_7_, float p_180614_8_, float p_180614_9_)
     {
-        IBlockState iblockstate = p_180614_3_.func_180495_p(p_180614_4_);
-        Block block = iblockstate.func_177230_c();
+        IBlockState iblockstate = p_180614_3_.getBlockState(p_180614_4_);
+        Block block = iblockstate.getBlock();
 
         if (block == Blocks.snow_layer && ((Integer)iblockstate.func_177229_b(BlockSnow.field_176315_a)).intValue() < 1)
         {
@@ -34,7 +34,7 @@
         }
         else if (!block.func_176200_f(p_180614_3_, p_180614_4_))
         {
-            p_180614_4_ = p_180614_4_.func_177972_a(p_180614_6_);
+            p_180614_4_ = p_180614_4_.getBlockPosRelative(p_180614_6_);
         }
 
         if (p_180614_2_.func_175151_a(p_180614_4_, p_180614_6_, p_180614_1_) && p_180614_1_.stackSize != 0 && p_180614_3_.func_175716_a(this.field_150935_a, p_180614_4_, false, p_180614_6_, (Entity)null, p_180614_1_))
@@ -47,12 +47,12 @@
             }
             else
             {
-                iblockstate1 = p_180614_3_.func_180495_p(p_180614_4_);
+                iblockstate1 = p_180614_3_.getBlockState(p_180614_4_);
 
-                if (iblockstate1.func_177230_c() == this.field_150935_a)
+                if (iblockstate1.getBlock() == this.field_150935_a)
                 {
                     ItemBlock.func_179224_a(p_180614_3_, p_180614_2_, p_180614_4_, p_180614_1_);
-                    iblockstate1.func_177230_c().func_180633_a(p_180614_3_, p_180614_4_, iblockstate1, p_180614_2_, p_180614_1_);
+                    iblockstate1.getBlock().func_180633_a(p_180614_3_, p_180614_4_, iblockstate1, p_180614_2_, p_180614_1_);
                 }
 
                 SoundType soundtype = this.field_150935_a.func_185467_w();
