--- a/net/minecraft/item/ItemPickaxe.java
+++ b/net/minecraft/item/ItemPickaxe.java
@@ -21,7 +21,7 @@
      */
     public boolean canItemHarvestBlock(IBlockState p_150897_1_)
     {
-        Block block = p_150897_1_.func_177230_c();
+        Block block = p_150897_1_.getBlock();
 
         if (block == Blocks.obsidian)
         {
@@ -39,7 +39,7 @@
                         {
                             if (block != Blocks.redstone_ore && block != Blocks.lit_redstone_ore)
                             {
-                                Material material = p_150897_1_.func_185904_a();
+                                Material material = p_150897_1_.getMaterial();
                                 return material == Material.rock ? true : (material == Material.iron ? true : material == Material.anvil);
                             }
                             else
@@ -75,7 +75,7 @@
 
     public float getStrVsBlock(ItemStack p_150893_1_, IBlockState p_150893_2_)
     {
-        Material material = p_150893_2_.func_185904_a();
+        Material material = p_150893_2_.getMaterial();
         return material != Material.iron && material != Material.anvil && material != Material.rock ? super.getStrVsBlock(p_150893_1_, p_150893_2_) : this.efficiencyOnProperMaterial;
     }
 }
