--- a/net/minecraft/item/ItemRecord.java
+++ b/net/minecraft/item/ItemRecord.java
@@ -35,9 +35,9 @@
 
     public EnumActionResult func_180614_a(ItemStack p_180614_1_, EntityPlayer p_180614_2_, World p_180614_3_, BlockPos p_180614_4_, EnumHand p_180614_5_, EnumFacing p_180614_6_, float p_180614_7_, float p_180614_8_, float p_180614_9_)
     {
-        IBlockState iblockstate = p_180614_3_.func_180495_p(p_180614_4_);
+        IBlockState iblockstate = p_180614_3_.getBlockState(p_180614_4_);
 
-        if (iblockstate.func_177230_c() == Blocks.jukebox && !((Boolean)iblockstate.func_177229_b(BlockJukebox.field_176432_a)).booleanValue())
+        if (iblockstate.getBlock() == Blocks.jukebox && !((Boolean)iblockstate.func_177229_b(BlockJukebox.field_176432_a)).booleanValue())
         {
             if (!p_180614_3_.isRemote)
             {
