--- a/net/minecraft/item/ItemMonsterPlacer.java
+++ b/net/minecraft/item/ItemMonsterPlacer.java
@@ -56,15 +56,15 @@
         {
             return EnumActionResult.SUCCESS;
         }
-        else if (!p_180614_2_.func_175151_a(p_180614_4_.func_177972_a(p_180614_6_), p_180614_6_, p_180614_1_))
+        else if (!p_180614_2_.func_175151_a(p_180614_4_.getBlockPosRelative(p_180614_6_), p_180614_6_, p_180614_1_))
         {
             return EnumActionResult.FAIL;
         }
         else
         {
-            IBlockState iblockstate = p_180614_3_.func_180495_p(p_180614_4_);
+            IBlockState iblockstate = p_180614_3_.getBlockState(p_180614_4_);
 
-            if (iblockstate.func_177230_c() == Blocks.mob_spawner)
+            if (iblockstate.getBlock() == Blocks.mob_spawner)
             {
                 TileEntity tileentity = p_180614_3_.func_175625_s(p_180614_4_);
 
@@ -84,7 +84,7 @@
                 }
             }
 
-            p_180614_4_ = p_180614_4_.func_177972_a(p_180614_6_);
+            p_180614_4_ = p_180614_4_.getBlockPosRelative(p_180614_6_);
             double d0 = 0.0D;
 
             if (p_180614_6_ == EnumFacing.UP && iblockstate instanceof BlockFence)
@@ -92,7 +92,7 @@
                 d0 = 0.5D;
             }
 
-            Entity entity = spawnCreature(p_180614_3_, func_185080_h(p_180614_1_), (double)p_180614_4_.func_177958_n() + 0.5D, (double)p_180614_4_.func_177956_o() + d0, (double)p_180614_4_.func_177952_p() + 0.5D);
+            Entity entity = spawnCreature(p_180614_3_, func_185080_h(p_180614_1_), (double)p_180614_4_.getX() + 0.5D, (double)p_180614_4_.getY() + d0, (double)p_180614_4_.getZ() + 0.5D);
 
             if (entity != null)
             {
@@ -149,15 +149,15 @@
 
             if (raytraceresult != null && raytraceresult.typeOfHit == RayTraceResult.Type.BLOCK)
             {
-                BlockPos blockpos = raytraceresult.func_178782_a();
+                BlockPos blockpos = raytraceresult.getBlock();
 
-                if (!(worldIn.func_180495_p(blockpos).func_177230_c() instanceof BlockLiquid))
+                if (!(worldIn.getBlockState(blockpos).getBlock() instanceof BlockLiquid))
                 {
                     return new ActionResult(EnumActionResult.PASS, itemStackIn);
                 }
-                else if (worldIn.func_175660_a(player, blockpos) && player.func_175151_a(blockpos, raytraceresult.field_178784_b, itemStackIn))
+                else if (worldIn.func_175660_a(player, blockpos) && player.func_175151_a(blockpos, raytraceresult.sideHit, itemStackIn))
                 {
-                    Entity entity = spawnCreature(worldIn, func_185080_h(itemStackIn), (double)blockpos.func_177958_n() + 0.5D, (double)blockpos.func_177956_o() + 0.5D, (double)blockpos.func_177952_p() + 0.5D);
+                    Entity entity = spawnCreature(worldIn, func_185080_h(itemStackIn), (double)blockpos.getX() + 0.5D, (double)blockpos.getY() + 0.5D, (double)blockpos.getZ() + 0.5D);
 
                     if (entity == null)
                     {
