--- a/net/minecraft/item/ItemBucket.java
+++ b/net/minecraft/item/ItemBucket.java
@@ -49,7 +49,7 @@
         }
         else
         {
-            BlockPos blockpos = raytraceresult.func_178782_a();
+            BlockPos blockpos = raytraceresult.getBlock();
 
             if (!worldIn.func_175660_a(player, blockpos))
             {
@@ -57,26 +57,26 @@
             }
             else if (flag)
             {
-                if (!player.func_175151_a(blockpos.func_177972_a(raytraceresult.field_178784_b), raytraceresult.field_178784_b, itemStackIn))
+                if (!player.func_175151_a(blockpos.getBlockPosRelative(raytraceresult.sideHit), raytraceresult.sideHit, itemStackIn))
                 {
                     return new ActionResult(EnumActionResult.FAIL, itemStackIn);
                 }
                 else
                 {
-                    IBlockState iblockstate = worldIn.func_180495_p(blockpos);
-                    Material material = iblockstate.func_185904_a();
+                    IBlockState iblockstate = worldIn.getBlockState(blockpos);
+                    Material material = iblockstate.getMaterial();
 
                     if (material == Material.water && ((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue() == 0)
                     {
-                        worldIn.func_180501_a(blockpos, Blocks.air.func_176223_P(), 11);
+                        worldIn.func_180501_a(blockpos, Blocks.air.getBlockState(), 11);
                         player.triggerAchievement(StatList.func_188057_b(this));
-                        player.func_184185_a(SoundEvents.field_187630_M, 1.0F, 1.0F);
+                        player.playSound(SoundEvents.field_187630_M, 1.0F, 1.0F);
                         return new ActionResult(EnumActionResult.SUCCESS, this.func_150910_a(itemStackIn, player, Items.water_bucket));
                     }
                     else if (material == Material.lava && ((Integer)iblockstate.func_177229_b(BlockLiquid.field_176367_b)).intValue() == 0)
                     {
-                        player.func_184185_a(SoundEvents.field_187633_N, 1.0F, 1.0F);
-                        worldIn.func_180501_a(blockpos, Blocks.air.func_176223_P(), 11);
+                        player.playSound(SoundEvents.field_187633_N, 1.0F, 1.0F);
+                        worldIn.func_180501_a(blockpos, Blocks.air.getBlockState(), 11);
                         player.triggerAchievement(StatList.func_188057_b(this));
                         return new ActionResult(EnumActionResult.SUCCESS, this.func_150910_a(itemStackIn, player, Items.lava_bucket));
                     }
@@ -88,10 +88,10 @@
             }
             else
             {
-                boolean flag1 = worldIn.func_180495_p(blockpos).func_177230_c().func_176200_f(worldIn, blockpos);
-                BlockPos blockpos1 = flag1 && raytraceresult.field_178784_b == EnumFacing.UP ? blockpos : blockpos.func_177972_a(raytraceresult.field_178784_b);
+                boolean flag1 = worldIn.getBlockState(blockpos).getBlock().func_176200_f(worldIn, blockpos);
+                BlockPos blockpos1 = flag1 && raytraceresult.sideHit == EnumFacing.UP ? blockpos : blockpos.getBlockPosRelative(raytraceresult.sideHit);
 
-                if (!player.func_175151_a(blockpos1, raytraceresult.field_178784_b, itemStackIn))
+                if (!player.func_175151_a(blockpos1, raytraceresult.sideHit, itemStackIn))
                 {
                     return new ActionResult(EnumActionResult.FAIL, itemStackIn);
                 }
@@ -137,12 +137,12 @@
         }
         else
         {
-            IBlockState iblockstate = p_180616_2_.func_180495_p(p_180616_3_);
-            Material material = iblockstate.func_185904_a();
+            IBlockState iblockstate = p_180616_2_.getBlockState(p_180616_3_);
+            Material material = iblockstate.getMaterial();
             boolean flag = !material.isSolid();
-            boolean flag1 = iblockstate.func_177230_c().func_176200_f(p_180616_2_, p_180616_3_);
+            boolean flag1 = iblockstate.getBlock().func_176200_f(p_180616_2_, p_180616_3_);
 
-            if (!p_180616_2_.func_175623_d(p_180616_3_) && !flag && !flag1)
+            if (!p_180616_2_.isEmpty(p_180616_3_) && !flag && !flag1)
             {
                 return false;
             }
@@ -150,14 +150,14 @@
             {
                 if (p_180616_2_.provider.func_177500_n() && this.isFull == Blocks.flowing_water)
                 {
-                    int l = p_180616_3_.func_177958_n();
-                    int i = p_180616_3_.func_177956_o();
-                    int j = p_180616_3_.func_177952_p();
+                    int l = p_180616_3_.getX();
+                    int i = p_180616_3_.getY();
+                    int j = p_180616_3_.getZ();
                     p_180616_2_.func_184133_a(p_180616_1_, p_180616_3_, SoundEvents.field_187646_bt, SoundCategory.BLOCKS, 0.5F, 2.6F + (p_180616_2_.rand.nextFloat() - p_180616_2_.rand.nextFloat()) * 0.8F);
 
                     for (int k = 0; k < 8; ++k)
                     {
-                        p_180616_2_.func_175688_a(EnumParticleTypes.SMOKE_LARGE, (double)l + Math.random(), (double)i + Math.random(), (double)j + Math.random(), 0.0D, 0.0D, 0.0D, new int[0]);
+                        p_180616_2_.spawnParticle(EnumParticleTypes.SMOKE_LARGE, (double)l + Math.random(), (double)i + Math.random(), (double)j + Math.random(), 0.0D, 0.0D, 0.0D, new int[0]);
                     }
                 }
                 else
@@ -169,7 +169,7 @@
 
                     SoundEvent soundevent = this.isFull == Blocks.flowing_lava ? SoundEvents.field_187627_L : SoundEvents.field_187624_K;
                     p_180616_2_.func_184133_a(p_180616_1_, p_180616_3_, soundevent, SoundCategory.BLOCKS, 1.0F, 1.0F);
-                    p_180616_2_.func_180501_a(p_180616_3_, this.isFull.func_176223_P(), 11);
+                    p_180616_2_.func_180501_a(p_180616_3_, this.isFull.getBlockState(), 11);
                 }
 
                 return true;
