--- a/net/minecraft/item/ItemSeeds.java
+++ b/net/minecraft/item/ItemSeeds.java
@@ -25,9 +25,9 @@
 
     public EnumActionResult func_180614_a(ItemStack p_180614_1_, EntityPlayer p_180614_2_, World p_180614_3_, BlockPos p_180614_4_, EnumHand p_180614_5_, EnumFacing p_180614_6_, float p_180614_7_, float p_180614_8_, float p_180614_9_)
     {
-        if (p_180614_6_ == EnumFacing.UP && p_180614_2_.func_175151_a(p_180614_4_.func_177972_a(p_180614_6_), p_180614_6_, p_180614_1_) && p_180614_3_.func_180495_p(p_180614_4_).func_177230_c() == this.soilBlockID && p_180614_3_.func_175623_d(p_180614_4_.func_177984_a()))
+        if (p_180614_6_ == EnumFacing.UP && p_180614_2_.func_175151_a(p_180614_4_.getBlockPosRelative(p_180614_6_), p_180614_6_, p_180614_1_) && p_180614_3_.getBlockState(p_180614_4_).getBlock() == this.soilBlockID && p_180614_3_.isEmpty(p_180614_4_.getBlockPosAbove()))
         {
-            p_180614_3_.func_175656_a(p_180614_4_.func_177984_a(), this.field_150925_a.func_176223_P());
+            p_180614_3_.func_175656_a(p_180614_4_.getBlockPosAbove(), this.field_150925_a.getBlockState());
             --p_180614_1_.stackSize;
             return EnumActionResult.SUCCESS;
         }
