--- a/net/minecraft/item/ItemArmorStand.java
+++ b/net/minecraft/item/ItemArmorStand.java
@@ -32,8 +32,8 @@
         }
         else
         {
-            boolean flag = p_180614_3_.func_180495_p(p_180614_4_).func_177230_c().func_176200_f(p_180614_3_, p_180614_4_);
-            BlockPos blockpos = flag ? p_180614_4_ : p_180614_4_.func_177972_a(p_180614_6_);
+            boolean flag = p_180614_3_.getBlockState(p_180614_4_).getBlock().func_176200_f(p_180614_3_, p_180614_4_);
+            BlockPos blockpos = flag ? p_180614_4_ : p_180614_4_.getBlockPosRelative(p_180614_6_);
 
             if (!p_180614_2_.func_175151_a(blockpos, p_180614_6_, p_180614_1_))
             {
@@ -41,9 +41,9 @@
             }
             else
             {
-                BlockPos blockpos1 = blockpos.func_177984_a();
-                boolean flag1 = !p_180614_3_.func_175623_d(blockpos) && !p_180614_3_.func_180495_p(blockpos).func_177230_c().func_176200_f(p_180614_3_, blockpos);
-                flag1 = flag1 | (!p_180614_3_.func_175623_d(blockpos1) && !p_180614_3_.func_180495_p(blockpos1).func_177230_c().func_176200_f(p_180614_3_, blockpos1));
+                BlockPos blockpos1 = blockpos.getBlockPosAbove();
+                boolean flag1 = !p_180614_3_.isEmpty(blockpos) && !p_180614_3_.getBlockState(blockpos).getBlock().func_176200_f(p_180614_3_, blockpos);
+                flag1 = flag1 | (!p_180614_3_.isEmpty(blockpos1) && !p_180614_3_.getBlockState(blockpos1).getBlock().func_176200_f(p_180614_3_, blockpos1));
 
                 if (flag1)
                 {
@@ -51,9 +51,9 @@
                 }
                 else
                 {
-                    double d0 = (double)blockpos.func_177958_n();
-                    double d1 = (double)blockpos.func_177956_o();
-                    double d2 = (double)blockpos.func_177952_p();
+                    double d0 = (double)blockpos.getX();
+                    double d1 = (double)blockpos.getY();
+                    double d2 = (double)blockpos.getZ();
                     List<Entity> list = p_180614_3_.getEntitiesWithinAABBExcludingEntity((Entity)null, new AxisAlignedBB(d0, d1, d2, d0 + 1.0D, d1 + 2.0D, d2 + 1.0D));
 
                     if (!list.isEmpty())
@@ -72,7 +72,7 @@
                             this.func_179221_a(entityarmorstand, p_180614_3_.rand);
                             ItemMonsterPlacer.func_185079_a(p_180614_3_, p_180614_2_, p_180614_1_, entityarmorstand);
                             p_180614_3_.spawnEntityInWorld(entityarmorstand);
-                            p_180614_3_.func_184148_a((EntityPlayer)null, entityarmorstand.posX, entityarmorstand.posY, entityarmorstand.posZ, SoundEvents.field_187710_m, SoundCategory.BLOCKS, 0.75F, 0.8F);
+                            p_180614_3_.playSound((EntityPlayer)null, entityarmorstand.posX, entityarmorstand.posY, entityarmorstand.posZ, SoundEvents.field_187710_m, SoundCategory.BLOCKS, 0.75F, 0.8F);
                         }
 
                         --p_180614_1_.stackSize;
