--- a/net/minecraft/item/ItemLingeringPotion.java
+++ b/net/minecraft/item/ItemLingeringPotion.java
@@ -35,7 +35,7 @@
             --itemStackIn.stackSize;
         }
 
-        worldIn.func_184148_a((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187756_df, SoundCategory.NEUTRAL, 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
+        worldIn.playSound((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187756_df, SoundCategory.NEUTRAL, 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
 
         if (!worldIn.isRemote)
         {
