--- a/net/minecraft/item/ItemCompass.java
+++ b/net/minecraft/item/ItemCompass.java
@@ -79,7 +79,7 @@
             private double func_185092_a(World p_185092_1_, Entity p_185092_2_)
             {
                 BlockPos blockpos = p_185092_1_.func_175694_M();
-                return Math.atan2((double)blockpos.func_177952_p() - p_185092_2_.posZ, (double)blockpos.func_177958_n() - p_185092_2_.posX);
+                return Math.atan2((double)blockpos.getZ() - p_185092_2_.posZ, (double)blockpos.getX() - p_185092_2_.posX);
             }
         });
     }
