--- a/net/minecraft/item/ItemFishingRod.java
+++ b/net/minecraft/item/ItemFishingRod.java
@@ -25,7 +25,7 @@
         {
             public float func_185085_a(ItemStack p_185085_1_, @Nullable World p_185085_2_, @Nullable EntityLivingBase p_185085_3_)
             {
-                return p_185085_3_ == null ? 0.0F : (p_185085_3_.func_184614_ca() == p_185085_1_ && p_185085_3_ instanceof EntityPlayer && ((EntityPlayer)p_185085_3_).fishEntity != null ? 1.0F : 0.0F);
+                return p_185085_3_ == null ? 0.0F : (p_185085_3_.getMainHandItem() == p_185085_1_ && p_185085_3_ instanceof EntityPlayer && ((EntityPlayer)p_185085_3_).fishEntity != null ? 1.0F : 0.0F);
             }
         });
     }
@@ -57,7 +57,7 @@
         }
         else
         {
-            worldIn.func_184148_a((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187612_G, SoundCategory.NEUTRAL, 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
+            worldIn.playSound((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187612_G, SoundCategory.NEUTRAL, 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
 
             if (!worldIn.isRemote)
             {
