--- a/net/minecraft/item/ItemCarrotOnAStick.java
+++ b/net/minecraft/item/ItemCarrotOnAStick.java
@@ -38,9 +38,9 @@
 
     public ActionResult<ItemStack> onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player, EnumHand p_77659_4_)
     {
-        if (player.func_184218_aH() && player.func_184187_bx() instanceof EntityPig)
+        if (player.isRiding() && player.getRidingEntity() instanceof EntityPig)
         {
-            EntityPig entitypig = (EntityPig)player.func_184187_bx();
+            EntityPig entitypig = (EntityPig)player.getRidingEntity();
 
             if (itemStackIn.getMaxDurability() - itemStackIn.getMetadata() >= 7 && entitypig.func_184762_da())
             {
