--- a/net/minecraft/item/ItemElytra.java
+++ b/net/minecraft/item/ItemElytra.java
@@ -47,11 +47,11 @@
     public ActionResult<ItemStack> onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player, EnumHand p_77659_4_)
     {
         EntityEquipmentSlot entityequipmentslot = EntityLiving.func_184640_d(itemStackIn);
-        ItemStack itemstack = player.func_184582_a(entityequipmentslot);
+        ItemStack itemstack = player.getEquippedItemStack(entityequipmentslot);
 
         if (itemstack == null)
         {
-            player.func_184201_a(entityequipmentslot, itemStackIn.copy());
+            player.equipItemInSlot(entityequipmentslot, itemStackIn.copy());
             itemStackIn.stackSize = 0;
             return new ActionResult(EnumActionResult.SUCCESS, itemStackIn);
         }
