--- a/net/minecraft/item/ItemGlassBottle.java
+++ b/net/minecraft/item/ItemGlassBottle.java
@@ -28,7 +28,7 @@
 
     public ActionResult<ItemStack> onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player, EnumHand p_77659_4_)
     {
-        List<EntityAreaEffectCloud> list = worldIn.<EntityAreaEffectCloud>func_175647_a(EntityAreaEffectCloud.class, player.func_174813_aQ().func_186662_g(2.0D), new Predicate<EntityAreaEffectCloud>()
+        List<EntityAreaEffectCloud> list = worldIn.<EntityAreaEffectCloud>func_175647_a(EntityAreaEffectCloud.class, player.func_174813_aQ().expand(2.0D), new Predicate<EntityAreaEffectCloud>()
         {
             public boolean apply(@Nullable EntityAreaEffectCloud p_apply_1_)
             {
@@ -40,7 +40,7 @@
         {
             EntityAreaEffectCloud entityareaeffectcloud = (EntityAreaEffectCloud)list.get(0);
             entityareaeffectcloud.func_184483_a(entityareaeffectcloud.func_184490_j() - 0.5F);
-            worldIn.func_184148_a((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187618_I, SoundCategory.NEUTRAL, 1.0F, 1.0F);
+            worldIn.playSound((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187618_I, SoundCategory.NEUTRAL, 1.0F, 1.0F);
             return new ActionResult(EnumActionResult.SUCCESS, this.func_185061_a(itemStackIn, player, new ItemStack(Items.field_185157_bK)));
         }
         else
@@ -55,16 +55,16 @@
             {
                 if (raytraceresult.typeOfHit == RayTraceResult.Type.BLOCK)
                 {
-                    BlockPos blockpos = raytraceresult.func_178782_a();
+                    BlockPos blockpos = raytraceresult.getBlock();
 
-                    if (!worldIn.func_175660_a(player, blockpos) || !player.func_175151_a(blockpos.func_177972_a(raytraceresult.field_178784_b), raytraceresult.field_178784_b, itemStackIn))
+                    if (!worldIn.func_175660_a(player, blockpos) || !player.func_175151_a(blockpos.getBlockPosRelative(raytraceresult.sideHit), raytraceresult.sideHit, itemStackIn))
                     {
                         return new ActionResult(EnumActionResult.PASS, itemStackIn);
                     }
 
-                    if (worldIn.func_180495_p(blockpos).func_185904_a() == Material.water)
+                    if (worldIn.getBlockState(blockpos).getMaterial() == Material.water)
                     {
-                        worldIn.func_184148_a(player, player.posX, player.posY, player.posZ, SoundEvents.field_187615_H, SoundCategory.NEUTRAL, 1.0F, 1.0F);
+                        worldIn.playSound(player, player.posX, player.posY, player.posZ, SoundEvents.field_187615_H, SoundCategory.NEUTRAL, 1.0F, 1.0F);
                         return new ActionResult(EnumActionResult.SUCCESS, this.func_185061_a(itemStackIn, player, new ItemStack(Items.potionitem)));
                     }
                 }
