--- a/net/minecraft/item/ItemArmor.java
+++ b/net/minecraft/item/ItemArmor.java
@@ -63,7 +63,7 @@
 
     public static ItemStack func_185082_a(IBlockSource p_185082_0_, ItemStack p_185082_1_)
     {
-        BlockPos blockpos = p_185082_0_.func_180699_d().func_177972_a((EnumFacing)p_185082_0_.func_189992_e().func_177229_b(BlockDispenser.field_176441_a));
+        BlockPos blockpos = p_185082_0_.func_180699_d().getBlockPosRelative((EnumFacing)p_185082_0_.func_189992_e().func_177229_b(BlockDispenser.field_176441_a));
         List<EntityLivingBase> list = p_185082_0_.getWorld().<EntityLivingBase>func_175647_a(EntityLivingBase.class, new AxisAlignedBB(blockpos), Predicates.<EntityLivingBase> and (EntitySelectors.field_180132_d, new EntitySelectors.ArmoredMob(p_185082_1_)));
 
         if (list.isEmpty())
@@ -76,7 +76,7 @@
             EntityEquipmentSlot entityequipmentslot = EntityLiving.func_184640_d(p_185082_1_);
             ItemStack itemstack = p_185082_1_.copy();
             itemstack.stackSize = 1;
-            entitylivingbase.func_184201_a(entityequipmentslot, itemstack);
+            entitylivingbase.equipItemInSlot(entityequipmentslot, itemstack);
 
             if (entitylivingbase instanceof EntityLiving)
             {
@@ -224,11 +224,11 @@
     public ActionResult<ItemStack> onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player, EnumHand p_77659_4_)
     {
         EntityEquipmentSlot entityequipmentslot = EntityLiving.func_184640_d(itemStackIn);
-        ItemStack itemstack = player.func_184582_a(entityequipmentslot);
+        ItemStack itemstack = player.getEquippedItemStack(entityequipmentslot);
 
         if (itemstack == null)
         {
-            player.func_184201_a(entityequipmentslot, itemStackIn.copy());
+            player.equipItemInSlot(entityequipmentslot, itemStackIn.copy());
             itemStackIn.stackSize = 0;
             return new ActionResult(EnumActionResult.SUCCESS, itemStackIn);
         }
