--- a/net/minecraft/item/ItemMap.java
+++ b/net/minecraft/item/ItemMap.java
@@ -100,7 +100,7 @@
                             int k2 = (j / i + k1 - 64) * i;
                             int l2 = (k / i + l1 - 64) * i;
                             Multiset<MapColor> multiset = HashMultiset.<MapColor>create();
-                            Chunk chunk = p_77872_1_.func_175726_f(new BlockPos(k2, 0, l2));
+                            Chunk chunk = p_77872_1_.getChunk(new BlockPos(k2, 0, l2));
 
                             if (!chunk.isEmpty())
                             {
@@ -116,11 +116,11 @@
 
                                     if ((l3 >> 20 & 1) == 0)
                                     {
-                                        multiset.add(Blocks.dirt.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT).func_185909_g(), 10);
+                                        multiset.add(Blocks.dirt.getBlockState().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT).func_185909_g(), 10);
                                     }
                                     else
                                     {
-                                        multiset.add(Blocks.stone.func_176223_P().func_177226_a(BlockStone.field_176247_a, BlockStone.EnumType.STONE).func_185909_g(), 100);
+                                        multiset.add(Blocks.stone.getBlockState().func_177226_a(BlockStone.field_176247_a, BlockStone.EnumType.STONE).func_185909_g(), 100);
                                     }
 
                                     d1 = 100.0D;
@@ -134,7 +134,7 @@
                                         for (int j4 = 0; j4 < i; ++j4)
                                         {
                                             int k4 = chunk.getHeightValue(i4 + i3, j4 + j3) + 1;
-                                            IBlockState iblockstate = Blocks.air.func_176223_P();
+                                            IBlockState iblockstate = Blocks.air.getBlockState();
 
                                             if (k4 > 1)
                                             {
@@ -143,7 +143,7 @@
                                                     while (true)
                                                     {
                                                         --k4;
-                                                        iblockstate = chunk.func_177435_g(blockpos$mutableblockpos.func_181079_c(i4 + i3, k4, j4 + j3));
+                                                        iblockstate = chunk.getBlockState(blockpos$mutableblockpos.func_181079_c(i4 + i3, k4, j4 + j3));
 
                                                         if (iblockstate.func_185909_g() != MapColor.airColor || k4 <= 0)
                                                         {
@@ -151,7 +151,7 @@
                                                         }
                                                     }
 
-                                                    if (k4 > 0 && iblockstate.func_185904_a().isLiquid())
+                                                    if (k4 > 0 && iblockstate.getMaterial().isLiquid())
                                                     {
                                                         int l4 = k4 - 1;
 
@@ -160,7 +160,7 @@
                                                             IBlockState iblockstate1 = chunk.func_186032_a(i4 + i3, l4--, j4 + j3);
                                                             ++k3;
 
-                                                            if (l4 <= 0 || !iblockstate1.func_185904_a().isLiquid())
+                                                            if (l4 <= 0 || !iblockstate1.getMaterial().isLiquid())
                                                             {
                                                                 break label542;
                                                             }
@@ -245,7 +245,7 @@
                 mapdata.updateVisiblePlayers(entityplayer, stack);
             }
 
-            if (p_77663_5_ || entityIn instanceof EntityPlayer && ((EntityPlayer)entityIn).func_184592_cb() == stack)
+            if (p_77663_5_ || entityIn instanceof EntityPlayer && ((EntityPlayer)entityIn).getOffhandItem() == stack)
             {
                 this.updateMapData(worldIn, entityIn, mapdata);
             }
