--- a/net/minecraft/item/ItemFireball.java
+++ b/net/minecraft/item/ItemFireball.java
@@ -27,7 +27,7 @@
         }
         else
         {
-            p_180614_4_ = p_180614_4_.func_177972_a(p_180614_6_);
+            p_180614_4_ = p_180614_4_.getBlockPosRelative(p_180614_6_);
 
             if (!p_180614_2_.func_175151_a(p_180614_4_, p_180614_6_, p_180614_1_))
             {
@@ -35,10 +35,10 @@
             }
             else
             {
-                if (p_180614_3_.func_180495_p(p_180614_4_).func_185904_a() == Material.air)
+                if (p_180614_3_.getBlockState(p_180614_4_).getMaterial() == Material.air)
                 {
                     p_180614_3_.func_184133_a((EntityPlayer)null, p_180614_4_, SoundEvents.field_187616_bj, SoundCategory.BLOCKS, 1.0F, (itemRand.nextFloat() - itemRand.nextFloat()) * 0.2F + 1.0F);
-                    p_180614_3_.func_175656_a(p_180614_4_, Blocks.fire.func_176223_P());
+                    p_180614_3_.func_175656_a(p_180614_4_, Blocks.fire.getBlockState());
                 }
 
                 if (!p_180614_2_.capabilities.isCreativeMode)
