--- a/net/minecraft/item/ItemMinecart.java
+++ b/net/minecraft/item/ItemMinecart.java
@@ -28,9 +28,9 @@
             double d0 = source.getX() + (double)enumfacing.getFrontOffsetX() * 1.125D;
             double d1 = Math.floor(source.getY()) + (double)enumfacing.getFrontOffsetY();
             double d2 = source.getZ() + (double)enumfacing.getFrontOffsetZ() * 1.125D;
-            BlockPos blockpos = source.func_180699_d().func_177972_a(enumfacing);
-            IBlockState iblockstate = world.func_180495_p(blockpos);
-            BlockRailBase.EnumRailDirection blockrailbase$enumraildirection = iblockstate.func_177230_c() instanceof BlockRailBase ? (BlockRailBase.EnumRailDirection)iblockstate.func_177229_b(((BlockRailBase)iblockstate.func_177230_c()).func_176560_l()) : BlockRailBase.EnumRailDirection.NORTH_SOUTH;
+            BlockPos blockpos = source.func_180699_d().getBlockPosRelative(enumfacing);
+            IBlockState iblockstate = world.getBlockState(blockpos);
+            BlockRailBase.EnumRailDirection blockrailbase$enumraildirection = iblockstate.getBlock() instanceof BlockRailBase ? (BlockRailBase.EnumRailDirection)iblockstate.func_177229_b(((BlockRailBase)iblockstate.getBlock()).func_176560_l()) : BlockRailBase.EnumRailDirection.NORTH_SOUTH;
             double d3;
 
             if (BlockRailBase.func_176563_d(iblockstate))
@@ -46,13 +46,13 @@
             }
             else
             {
-                if (iblockstate.func_185904_a() != Material.air || !BlockRailBase.func_176563_d(world.func_180495_p(blockpos.func_177977_b())))
+                if (iblockstate.getMaterial() != Material.air || !BlockRailBase.func_176563_d(world.getBlockState(blockpos.getBlockPosBelow())))
                 {
                     return this.behaviourDefaultDispenseItem.dispense(source, stack);
                 }
 
-                IBlockState iblockstate1 = world.func_180495_p(blockpos.func_177977_b());
-                BlockRailBase.EnumRailDirection blockrailbase$enumraildirection1 = iblockstate1.func_177230_c() instanceof BlockRailBase ? (BlockRailBase.EnumRailDirection)iblockstate1.func_177229_b(((BlockRailBase)iblockstate1.func_177230_c()).func_176560_l()) : BlockRailBase.EnumRailDirection.NORTH_SOUTH;
+                IBlockState iblockstate1 = world.getBlockState(blockpos.getBlockPosBelow());
+                BlockRailBase.EnumRailDirection blockrailbase$enumraildirection1 = iblockstate1.getBlock() instanceof BlockRailBase ? (BlockRailBase.EnumRailDirection)iblockstate1.func_177229_b(((BlockRailBase)iblockstate1.getBlock()).func_176560_l()) : BlockRailBase.EnumRailDirection.NORTH_SOUTH;
 
                 if (enumfacing != EnumFacing.DOWN && blockrailbase$enumraildirection1.func_177018_c())
                 {
@@ -92,7 +92,7 @@
 
     public EnumActionResult func_180614_a(ItemStack p_180614_1_, EntityPlayer p_180614_2_, World p_180614_3_, BlockPos p_180614_4_, EnumHand p_180614_5_, EnumFacing p_180614_6_, float p_180614_7_, float p_180614_8_, float p_180614_9_)
     {
-        IBlockState iblockstate = p_180614_3_.func_180495_p(p_180614_4_);
+        IBlockState iblockstate = p_180614_3_.getBlockState(p_180614_4_);
 
         if (!BlockRailBase.func_176563_d(iblockstate))
         {
@@ -102,7 +102,7 @@
         {
             if (!p_180614_3_.isRemote)
             {
-                BlockRailBase.EnumRailDirection blockrailbase$enumraildirection = iblockstate.func_177230_c() instanceof BlockRailBase ? (BlockRailBase.EnumRailDirection)iblockstate.func_177229_b(((BlockRailBase)iblockstate.func_177230_c()).func_176560_l()) : BlockRailBase.EnumRailDirection.NORTH_SOUTH;
+                BlockRailBase.EnumRailDirection blockrailbase$enumraildirection = iblockstate.getBlock() instanceof BlockRailBase ? (BlockRailBase.EnumRailDirection)iblockstate.func_177229_b(((BlockRailBase)iblockstate.getBlock()).func_176560_l()) : BlockRailBase.EnumRailDirection.NORTH_SOUTH;
                 double d0 = 0.0D;
 
                 if (blockrailbase$enumraildirection.func_177018_c())
@@ -110,7 +110,7 @@
                     d0 = 0.5D;
                 }
 
-                EntityMinecart entityminecart = EntityMinecart.func_184263_a(p_180614_3_, (double)p_180614_4_.func_177958_n() + 0.5D, (double)p_180614_4_.func_177956_o() + 0.0625D + d0, (double)p_180614_4_.func_177952_p() + 0.5D, this.minecartType);
+                EntityMinecart entityminecart = EntityMinecart.func_184263_a(p_180614_3_, (double)p_180614_4_.getX() + 0.5D, (double)p_180614_4_.getY() + 0.0625D + d0, (double)p_180614_4_.getZ() + 0.5D, this.minecartType);
 
                 if (p_180614_1_.hasDisplayName())
                 {
