--- a/net/minecraft/item/ItemBow.java
+++ b/net/minecraft/item/ItemBow.java
@@ -48,23 +48,23 @@
         });
     }
 
-    private ItemStack func_185060_a(EntityPlayer p_185060_1_)
+    private ItemStack getArrowItemStack(EntityPlayer player_)
     {
-        if (this.func_185058_h_(p_185060_1_.func_184586_b(EnumHand.OFF_HAND)))
+        if (this.isArrow(player_.getItemStackInHand(EnumHand.OFF_HAND)))
         {
-            return p_185060_1_.func_184586_b(EnumHand.OFF_HAND);
+            return player_.getItemStackInHand(EnumHand.OFF_HAND);
         }
-        else if (this.func_185058_h_(p_185060_1_.func_184586_b(EnumHand.MAIN_HAND)))
+        else if (this.isArrow(player_.getItemStackInHand(EnumHand.MAIN_HAND)))
         {
-            return p_185060_1_.func_184586_b(EnumHand.MAIN_HAND);
+            return player_.getItemStackInHand(EnumHand.MAIN_HAND);
         }
         else
         {
-            for (int i = 0; i < p_185060_1_.inventory.getSizeInventory(); ++i)
+            for (int i = 0; i < player_.inventory.getSizeInventory(); ++i)
             {
-                ItemStack itemstack = p_185060_1_.inventory.getStackInSlot(i);
+                ItemStack itemstack = player_.inventory.getStackInSlot(i);
 
-                if (this.func_185058_h_(itemstack))
+                if (this.isArrow(itemstack))
                 {
                     return itemstack;
                 }
@@ -74,7 +74,7 @@
         }
     }
 
-    protected boolean func_185058_h_(@Nullable ItemStack p_185058_1_)
+    protected boolean isArrow(@Nullable ItemStack p_185058_1_)
     {
         return p_185058_1_ != null && p_185058_1_.getItem() instanceof ItemArrow;
     }
@@ -87,8 +87,8 @@
         if (p_77615_3_ instanceof EntityPlayer)
         {
             EntityPlayer entityplayer = (EntityPlayer)p_77615_3_;
-            boolean flag = entityplayer.capabilities.isCreativeMode || EnchantmentHelper.getEnchantmentLevel(Enchantments.field_185312_x, p_77615_1_) > 0;
-            ItemStack itemstack = this.func_185060_a(entityplayer);
+            boolean flag = entityplayer.capabilities.isCreativeMode || EnchantmentHelper.getEnchantmentLevel(Enchantments.infinity, p_77615_1_) > 0;
+            ItemStack itemstack = this.getArrowItemStack(entityplayer);
 
             if (itemstack != null || flag)
             {
@@ -144,7 +144,7 @@
                         p_77615_2_.spawnEntityInWorld(entityarrow);
                     }
 
-                    p_77615_2_.func_184148_a((EntityPlayer)null, entityplayer.posX, entityplayer.posY, entityplayer.posZ, SoundEvents.field_187737_v, SoundCategory.NEUTRAL, 1.0F, 1.0F / (itemRand.nextFloat() * 0.4F + 1.2F) + f * 0.5F);
+                    p_77615_2_.playSound((EntityPlayer)null, entityplayer.posX, entityplayer.posY, entityplayer.posZ, SoundEvents.field_187737_v, SoundCategory.NEUTRAL, 1.0F, 1.0F / (itemRand.nextFloat() * 0.4F + 1.2F) + f * 0.5F);
 
                     if (!flag1)
                     {
@@ -193,7 +193,7 @@
 
     public ActionResult<ItemStack> onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player, EnumHand p_77659_4_)
     {
-        boolean flag = this.func_185060_a(player) != null;
+        boolean flag = this.getArrowItemStack(player) != null;
 
         if (!player.capabilities.isCreativeMode && !flag)
         {
