--- a/net/minecraft/item/ItemFood.java
+++ b/net/minecraft/item/ItemFood.java
@@ -66,7 +66,7 @@
         {
             EntityPlayer entityplayer = (EntityPlayer)p_77654_3_;
             entityplayer.getFoodStats().addStats(this, p_77654_1_);
-            p_77654_2_.func_184148_a((EntityPlayer)null, entityplayer.posX, entityplayer.posY, entityplayer.posZ, SoundEvents.field_187739_dZ, SoundCategory.PLAYERS, 0.5F, p_77654_2_.rand.nextFloat() * 0.1F + 0.9F);
+            p_77654_2_.playSound((EntityPlayer)null, entityplayer.posX, entityplayer.posY, entityplayer.posZ, SoundEvents.field_187739_dZ, SoundCategory.PLAYERS, 0.5F, p_77654_2_.rand.nextFloat() * 0.1F + 0.9F);
             this.onFoodEaten(p_77654_1_, p_77654_2_, entityplayer);
             entityplayer.triggerAchievement(StatList.func_188057_b(this));
         }
