--- a/net/minecraft/item/ItemSign.java
+++ b/net/minecraft/item/ItemSign.java
@@ -25,12 +25,12 @@
 
     public EnumActionResult func_180614_a(ItemStack p_180614_1_, EntityPlayer p_180614_2_, World p_180614_3_, BlockPos p_180614_4_, EnumHand p_180614_5_, EnumFacing p_180614_6_, float p_180614_7_, float p_180614_8_, float p_180614_9_)
     {
-        IBlockState iblockstate = p_180614_3_.func_180495_p(p_180614_4_);
-        boolean flag = iblockstate.func_177230_c().func_176200_f(p_180614_3_, p_180614_4_);
+        IBlockState iblockstate = p_180614_3_.getBlockState(p_180614_4_);
+        boolean flag = iblockstate.getBlock().func_176200_f(p_180614_3_, p_180614_4_);
 
-        if (p_180614_6_ != EnumFacing.DOWN && (iblockstate.func_185904_a().isSolid() || flag) && (!flag || p_180614_6_ == EnumFacing.UP))
+        if (p_180614_6_ != EnumFacing.DOWN && (iblockstate.getMaterial().isSolid() || flag) && (!flag || p_180614_6_ == EnumFacing.UP))
         {
-            p_180614_4_ = p_180614_4_.func_177972_a(p_180614_6_);
+            p_180614_4_ = p_180614_4_.getBlockPosRelative(p_180614_6_);
 
             if (p_180614_2_.func_175151_a(p_180614_4_, p_180614_6_, p_180614_1_) && Blocks.standing_sign.func_176196_c(p_180614_3_, p_180614_4_))
             {
@@ -40,16 +40,16 @@
                 }
                 else
                 {
-                    p_180614_4_ = flag ? p_180614_4_.func_177977_b() : p_180614_4_;
+                    p_180614_4_ = flag ? p_180614_4_.getBlockPosBelow() : p_180614_4_;
 
                     if (p_180614_6_ == EnumFacing.UP)
                     {
                         int i = MathHelper.floor_double((double)((p_180614_2_.rotationYaw + 180.0F) * 16.0F / 360.0F) + 0.5D) & 15;
-                        p_180614_3_.func_180501_a(p_180614_4_, Blocks.standing_sign.func_176223_P().func_177226_a(BlockStandingSign.field_176413_a, Integer.valueOf(i)), 11);
+                        p_180614_3_.func_180501_a(p_180614_4_, Blocks.standing_sign.getBlockState().func_177226_a(BlockStandingSign.field_176413_a, Integer.valueOf(i)), 11);
                     }
                     else
                     {
-                        p_180614_3_.func_180501_a(p_180614_4_, Blocks.wall_sign.func_176223_P().func_177226_a(BlockWallSign.field_176412_a, p_180614_6_), 11);
+                        p_180614_3_.func_180501_a(p_180614_4_, Blocks.wall_sign.getBlockState().func_177226_a(BlockWallSign.field_176412_a, p_180614_6_), 11);
                     }
 
                     --p_180614_1_.stackSize;
