--- a/net/minecraft/item/ItemEnderEye.java
+++ b/net/minecraft/item/ItemEnderEye.java
@@ -29,9 +29,9 @@
 
     public EnumActionResult func_180614_a(ItemStack p_180614_1_, EntityPlayer p_180614_2_, World p_180614_3_, BlockPos p_180614_4_, EnumHand p_180614_5_, EnumFacing p_180614_6_, float p_180614_7_, float p_180614_8_, float p_180614_9_)
     {
-        IBlockState iblockstate = p_180614_3_.func_180495_p(p_180614_4_);
+        IBlockState iblockstate = p_180614_3_.getBlockState(p_180614_4_);
 
-        if (p_180614_2_.func_175151_a(p_180614_4_.func_177972_a(p_180614_6_), p_180614_6_, p_180614_1_) && iblockstate.func_177230_c() == Blocks.end_portal_frame && !((Boolean)iblockstate.func_177229_b(BlockEndPortalFrame.field_176507_b)).booleanValue())
+        if (p_180614_2_.func_175151_a(p_180614_4_.getBlockPosRelative(p_180614_6_), p_180614_6_, p_180614_1_) && iblockstate.getBlock() == Blocks.end_portal_frame && !((Boolean)iblockstate.func_177229_b(BlockEndPortalFrame.field_176507_b)).booleanValue())
         {
             if (p_180614_3_.isRemote)
             {
@@ -45,13 +45,13 @@
 
                 for (int i = 0; i < 16; ++i)
                 {
-                    double d0 = (double)((float)p_180614_4_.func_177958_n() + (5.0F + itemRand.nextFloat() * 6.0F) / 16.0F);
-                    double d1 = (double)((float)p_180614_4_.func_177956_o() + 0.8125F);
-                    double d2 = (double)((float)p_180614_4_.func_177952_p() + (5.0F + itemRand.nextFloat() * 6.0F) / 16.0F);
+                    double d0 = (double)((float)p_180614_4_.getX() + (5.0F + itemRand.nextFloat() * 6.0F) / 16.0F);
+                    double d1 = (double)((float)p_180614_4_.getY() + 0.8125F);
+                    double d2 = (double)((float)p_180614_4_.getZ() + (5.0F + itemRand.nextFloat() * 6.0F) / 16.0F);
                     double d3 = 0.0D;
                     double d4 = 0.0D;
                     double d5 = 0.0D;
-                    p_180614_3_.func_175688_a(EnumParticleTypes.SMOKE_NORMAL, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
+                    p_180614_3_.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, d0, d1, d2, 0.0D, 0.0D, 0.0D, new int[0]);
                 }
 
                 BlockPattern.PatternHelper blockpattern$patternhelper = BlockEndPortalFrame.func_185661_e().func_177681_a(p_180614_3_, p_180614_4_);
@@ -64,7 +64,7 @@
                     {
                         for (int k = 0; k < 3; ++k)
                         {
-                            p_180614_3_.func_180501_a(blockpos.func_177982_a(j, 0, k), Blocks.end_portal.func_176223_P(), 2);
+                            p_180614_3_.func_180501_a(blockpos.func_177982_a(j, 0, k), Blocks.end_portal.getBlockState(), 2);
                         }
                     }
                 }
@@ -82,7 +82,7 @@
     {
         RayTraceResult raytraceresult = this.getMovingObjectPositionFromPlayer(worldIn, player, false);
 
-        if (raytraceresult != null && raytraceresult.typeOfHit == RayTraceResult.Type.BLOCK && worldIn.func_180495_p(raytraceresult.func_178782_a()).func_177230_c() == Blocks.end_portal_frame)
+        if (raytraceresult != null && raytraceresult.typeOfHit == RayTraceResult.Type.BLOCK && worldIn.getBlockState(raytraceresult.getBlock()).getBlock() == Blocks.end_portal_frame)
         {
             return new ActionResult(EnumActionResult.PASS, itemStackIn);
         }
@@ -97,7 +97,7 @@
                     EntityEnderEye entityendereye = new EntityEnderEye(worldIn, player.posX, player.posY + (double)(player.height / 2.0F), player.posZ);
                     entityendereye.func_180465_a(blockpos);
                     worldIn.spawnEntityInWorld(entityendereye);
-                    worldIn.func_184148_a((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187528_aR, SoundCategory.NEUTRAL, 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
+                    worldIn.playSound((EntityPlayer)null, player.posX, player.posY, player.posZ, SoundEvents.field_187528_aR, SoundCategory.NEUTRAL, 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
                     worldIn.func_180498_a((EntityPlayer)null, 1003, new BlockPos(player), 0);
 
                     if (!player.capabilities.isCreativeMode)
