--- a/net/minecraft/util/math/MathHelper.java
+++ b/net/minecraft/util/math/MathHelper.java
@@ -63,7 +63,7 @@
     {
         return (float)Math.sqrt(value);
     }
-
+    
     /**
      * Returns the greatest integer less than or equal to the float argument
      */
@@ -89,7 +89,7 @@
         int i = (int)value;
         return value < (double)i ? i - 1 : i;
     }
-
+   
     /**
      * Long version of floor()
      */
