--- a/net/minecraft/util/math/Vec2f.java
+++ b/net/minecraft/util/math/Vec2f.java
@@ -10,12 +10,12 @@
     public static final Vec2f field_189979_f = new Vec2f(0.0F, -1.0F);
     public static final Vec2f field_189980_g = new Vec2f(Float.MAX_VALUE, Float.MAX_VALUE);
     public static final Vec2f field_189981_h = new Vec2f(Float.MIN_VALUE, Float.MIN_VALUE);
-    public final float field_189982_i;
-    public final float field_189983_j;
+    public final float x;
+    public final float y;
 
-    public Vec2f(float p_i47143_1_, float p_i47143_2_)
+    public Vec2f(float x, float y)
     {
-        this.field_189982_i = p_i47143_1_;
-        this.field_189983_j = p_i47143_2_;
+        this.x = x;
+        this.y = y;
     }
 }
