--- a/net/minecraft/client/model/ModelArmorStand.java
+++ b/net/minecraft/client/model/ModelArmorStand.java
@@ -94,12 +94,12 @@
     public void render(Entity p_78088_1_, float p_78088_2_, float p_78088_3_, float p_78088_4_, float p_78088_5_, float p_78088_6_, float p_78088_7_)
     {
         super.render(p_78088_1_, p_78088_2_, p_78088_3_, p_78088_4_, p_78088_5_, p_78088_6_, p_78088_7_);
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
 
         if (this.isChild)
         {
             float f = 2.0F;
-            GlStateManager.func_179152_a(0.5F, 0.5F, 0.5F);
+            GlStateManager.scale(0.5F, 0.5F, 0.5F);
             GlStateManager.func_179109_b(0.0F, 24.0F * p_78088_7_, 0.0F);
             this.field_178740_a.render(p_78088_7_);
             this.field_178738_b.render(p_78088_7_);
@@ -119,7 +119,7 @@
             this.field_178737_d.render(p_78088_7_);
         }
 
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 
     public void func_187073_a(float p_187073_1_, EnumHandSide p_187073_2_)
