--- a/net/minecraft/client/model/ModelWolf.java
+++ b/net/minecraft/client/model/ModelWolf.java
@@ -76,12 +76,12 @@
         if (this.isChild)
         {
             float f = 2.0F;
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             GlStateManager.func_179109_b(0.0F, 5.0F * p_78088_7_, 2.0F * p_78088_7_);
             this.wolfHeadMain.renderWithRotation(p_78088_7_);
-            GlStateManager.func_179121_F();
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179152_a(0.5F, 0.5F, 0.5F);
+            GlStateManager.pop();
+            GlStateManager.push();
+            GlStateManager.scale(0.5F, 0.5F, 0.5F);
             GlStateManager.func_179109_b(0.0F, 24.0F * p_78088_7_, 0.0F);
             this.wolfBody.render(p_78088_7_);
             this.wolfLeg1.render(p_78088_7_);
@@ -90,7 +90,7 @@
             this.wolfLeg4.render(p_78088_7_);
             this.wolfTail.renderWithRotation(p_78088_7_);
             this.wolfMane.render(p_78088_7_);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
         else
         {
