--- a/net/minecraft/client/model/ModelEnderCrystal.java
+++ b/net/minecraft/client/model/ModelEnderCrystal.java
@@ -32,8 +32,8 @@
      */
     public void render(Entity p_78088_1_, float p_78088_2_, float p_78088_3_, float p_78088_4_, float p_78088_5_, float p_78088_6_, float p_78088_7_)
     {
-        GlStateManager.func_179094_E();
-        GlStateManager.func_179152_a(2.0F, 2.0F, 2.0F);
+        GlStateManager.push();
+        GlStateManager.scale(2.0F, 2.0F, 2.0F);
         GlStateManager.func_179109_b(0.0F, -0.5F, 0.0F);
 
         if (this.base != null)
@@ -41,19 +41,19 @@
             this.base.render(p_78088_7_);
         }
 
-        GlStateManager.func_179114_b(p_78088_3_, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(p_78088_3_, 0.0F, 1.0F, 0.0F);
         GlStateManager.func_179109_b(0.0F, 0.8F + p_78088_4_, 0.0F);
-        GlStateManager.func_179114_b(60.0F, 0.7071F, 0.0F, 0.7071F);
+        GlStateManager.rotate(60.0F, 0.7071F, 0.0F, 0.7071F);
         this.glass.render(p_78088_7_);
         float f = 0.875F;
-        GlStateManager.func_179152_a(0.875F, 0.875F, 0.875F);
-        GlStateManager.func_179114_b(60.0F, 0.7071F, 0.0F, 0.7071F);
-        GlStateManager.func_179114_b(p_78088_3_, 0.0F, 1.0F, 0.0F);
+        GlStateManager.scale(0.875F, 0.875F, 0.875F);
+        GlStateManager.rotate(60.0F, 0.7071F, 0.0F, 0.7071F);
+        GlStateManager.rotate(p_78088_3_, 0.0F, 1.0F, 0.0F);
         this.glass.render(p_78088_7_);
-        GlStateManager.func_179152_a(0.875F, 0.875F, 0.875F);
-        GlStateManager.func_179114_b(60.0F, 0.7071F, 0.0F, 0.7071F);
-        GlStateManager.func_179114_b(p_78088_3_, 0.0F, 1.0F, 0.0F);
+        GlStateManager.scale(0.875F, 0.875F, 0.875F);
+        GlStateManager.rotate(60.0F, 0.7071F, 0.0F, 0.7071F);
+        GlStateManager.rotate(p_78088_3_, 0.0F, 1.0F, 0.0F);
         this.cube.render(p_78088_7_);
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
     }
 }
