--- a/net/minecraft/client/model/ModelBoat.java
+++ b/net/minecraft/client/model/ModelBoat.java
@@ -57,7 +57,7 @@
      */
     public void render(Entity p_78088_1_, float p_78088_2_, float p_78088_3_, float p_78088_4_, float p_78088_5_, float p_78088_6_, float p_78088_7_)
     {
-        GlStateManager.func_179114_b(90.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(90.0F, 0.0F, 1.0F, 0.0F);
         EntityBoat entityboat = (EntityBoat)p_78088_1_;
         this.setRotationAngles(p_78088_2_, p_78088_3_, p_78088_4_, p_78088_5_, p_78088_6_, p_78088_7_, p_78088_1_);
 
@@ -72,7 +72,7 @@
 
     public void func_187054_b(Entity p_187054_1_, float p_187054_2_, float p_187054_3_, float p_187054_4_, float p_187054_5_, float p_187054_6_, float p_187054_7_)
     {
-        GlStateManager.func_179114_b(90.0F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(90.0F, 0.0F, 1.0F, 0.0F);
         GlStateManager.func_179135_a(false, false, false, false);
         this.field_187058_c.render(p_187054_7_);
         GlStateManager.func_179135_a(true, true, true, true);
