--- a/net/minecraft/client/model/ModelBiped.java
+++ b/net/minecraft/client/model/ModelBiped.java
@@ -1,5 +1,8 @@
 package net.minecraft.client.model;
 
+import java.util.UUID;
+
+import com.mtbs3d.minecrift.render.PlayerModelController;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
@@ -29,6 +32,12 @@
     public ModelBiped.ArmPose rightArmPose;
     public boolean isSneak;
 
+    //VIVE START
+    public boolean isVrPlayer;
+    public UUID playerUuid;
+
+    //VIVE END
+
     public ModelBiped()
     {
         this(0.0F);
@@ -177,39 +186,50 @@
             this.bipedLeftLeg.rotateAngleZ = -0.07853982F;
         }
 
-        this.bipedRightArm.rotateAngleY = 0.0F;
-        this.bipedRightArm.rotateAngleZ = 0.0F;
 
-        switch (this.leftArmPose)
-        {
-            case EMPTY:
-                this.bipedLeftArm.rotateAngleY = 0.0F;
-                break;
+        if(isVrPlayer){
+            PlayerModelController.RotInfo rotInfo=PlayerModelController.getInstance().getRotationsForPlayer(playerUuid);
 
-            case BLOCK:
-                this.bipedLeftArm.rotateAngleX = this.bipedLeftArm.rotateAngleX * 0.5F - 0.9424779F;
-                this.bipedLeftArm.rotateAngleY = 0.5235988F;
-                break;
+            this.bipedHead.rotateAngleX = (float)  -rotInfo.head.x;
+            this.bipedHead.rotateAngleY = (float) rotInfo.head.y  - entityIn.rotationYaw;
 
-            case ITEM:
-                this.bipedLeftArm.rotateAngleX = this.bipedLeftArm.rotateAngleX * 0.5F - ((float)Math.PI / 10F);
-                this.bipedLeftArm.rotateAngleY = 0.0F;
-        }
+            this.bipedLeftArm.rotateAngleX=(float) rotInfo.leftArm.x;
+            this.bipedLeftArm.rotateAngleY=(float) rotInfo.leftArm.y - entityIn.rotationYaw;
 
-        switch (this.rightArmPose)
-        {
-            case EMPTY:
-                this.bipedRightArm.rotateAngleY = 0.0F;
-                break;
+            this.bipedRightArm.rotateAngleX=(float) rotInfo.rightArm.x;
+            this.bipedRightArm.rotateAngleY=(float) rotInfo.rightArm.y - entityIn.rotationYaw;
+
+        }else {
+
+            switch (this.leftArmPose) {
+                case EMPTY:
+                    this.bipedLeftArm.rotateAngleY = 0.0F;
+                    break;
 
-            case BLOCK:
-                this.bipedRightArm.rotateAngleX = this.bipedRightArm.rotateAngleX * 0.5F - 0.9424779F;
-                this.bipedRightArm.rotateAngleY = -0.5235988F;
-                break;
+                case BLOCK:
+                    this.bipedLeftArm.rotateAngleX = this.bipedLeftArm.rotateAngleX * 0.5F - 0.9424779F;
+                    this.bipedLeftArm.rotateAngleY = 0.5235988F;
+                    break;
 
-            case ITEM:
-                this.bipedRightArm.rotateAngleX = this.bipedRightArm.rotateAngleX * 0.5F - ((float)Math.PI / 10F);
-                this.bipedRightArm.rotateAngleY = 0.0F;
+                case ITEM:
+                    this.bipedLeftArm.rotateAngleX = this.bipedLeftArm.rotateAngleX * 0.5F - ((float) Math.PI / 10F);
+                    this.bipedLeftArm.rotateAngleY = 0.0F;
+            }
+
+            switch (this.rightArmPose) {
+                case EMPTY:
+                    this.bipedRightArm.rotateAngleY = 0.0F;
+                    break;
+
+                case BLOCK:
+                    this.bipedRightArm.rotateAngleX = this.bipedRightArm.rotateAngleX * 0.5F - 0.9424779F;
+                    this.bipedRightArm.rotateAngleY = -0.5235988F;
+                    break;
+
+                case ITEM:
+                    this.bipedRightArm.rotateAngleX = this.bipedRightArm.rotateAngleX * 0.5F - ((float) Math.PI / 10F);
+                    this.bipedRightArm.rotateAngleY = 0.0F;
+            }
         }
 
         if (this.swingProgress > 0.0F)
@@ -296,6 +316,8 @@
             this.leftArmPose = modelbiped.leftArmPose;
             this.rightArmPose = modelbiped.rightArmPose;
             this.isSneak = modelbiped.isSneak;
+            this.isVrPlayer = modelbiped.isVrPlayer;
+            this.playerUuid = modelbiped.playerUuid;
         }
     }
 
