--- a/net/minecraft/client/model/ModelRabbit.java
+++ b/net/minecraft/client/model/ModelRabbit.java
@@ -107,16 +107,16 @@
         if (this.isChild)
         {
             float f = 1.5F;
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179152_a(0.56666666F, 0.56666666F, 0.56666666F);
+            GlStateManager.push();
+            GlStateManager.scale(0.56666666F, 0.56666666F, 0.56666666F);
             GlStateManager.func_179109_b(0.0F, 22.0F * p_78088_7_, 2.0F * p_78088_7_);
             this.field_178704_h.render(p_78088_7_);
             this.field_178702_j.render(p_78088_7_);
             this.field_178705_i.render(p_78088_7_);
             this.field_178700_l.render(p_78088_7_);
-            GlStateManager.func_179121_F();
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179152_a(0.4F, 0.4F, 0.4F);
+            GlStateManager.pop();
+            GlStateManager.push();
+            GlStateManager.scale(0.4F, 0.4F, 0.4F);
             GlStateManager.func_179109_b(0.0F, 36.0F * p_78088_7_, 0.0F);
             this.field_178698_a.render(p_78088_7_);
             this.field_178696_b.render(p_78088_7_);
@@ -126,12 +126,12 @@
             this.field_178692_f.render(p_78088_7_);
             this.field_178693_g.render(p_78088_7_);
             this.field_178703_k.render(p_78088_7_);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
         else
         {
-            GlStateManager.func_179094_E();
-            GlStateManager.func_179152_a(0.6F, 0.6F, 0.6F);
+            GlStateManager.push();
+            GlStateManager.scale(0.6F, 0.6F, 0.6F);
             GlStateManager.func_179109_b(0.0F, 16.0F * p_78088_7_, 0.0F);
             this.field_178698_a.render(p_78088_7_);
             this.field_178696_b.render(p_78088_7_);
@@ -145,7 +145,7 @@
             this.field_178702_j.render(p_78088_7_);
             this.field_178703_k.render(p_78088_7_);
             this.field_178700_l.render(p_78088_7_);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
         }
     }
 
