--- a/net/minecraft/command/CommandTP.java
+++ b/net/minecraft/command/CommandTP.java
@@ -83,7 +83,7 @@
                 }
                 else
                 {
-                    entity.func_184210_p();
+                    entity.dismount();
 
                     if (entity instanceof EntityPlayerMP)
                     {
@@ -145,7 +145,7 @@
                 f1 = MathHelper.wrapAngleTo180_float(f1);
             }
 
-            p_189863_0_.func_184210_p();
+            p_189863_0_.dismount();
             ((EntityPlayerMP)p_189863_0_).playerNetServerHandler.func_175089_a(p_189863_1_.func_179629_b(), p_189863_2_.func_179629_b(), p_189863_3_.func_179629_b(), f, f1, set);
             p_189863_0_.setRotationYawHead(f);
         }
