--- a/net/minecraft/command/CommandExecuteAt.java
+++ b/net/minecraft/command/CommandExecuteAt.java
@@ -62,9 +62,9 @@
                     throw new CommandException("commands.execute.failed", new Object[] {"detect", entity.getCommandSenderName()});
                 }
 
-                IBlockState iblockstate = world.func_180495_p(blockpos1);
+                IBlockState iblockstate = world.getBlockState(blockpos1);
 
-                if (iblockstate.func_177230_c() != block || j >= 0 && iblockstate.func_177230_c().func_176201_c(iblockstate) != j)
+                if (iblockstate.getBlock() != block || j >= 0 && iblockstate.getBlock().func_176201_c(iblockstate) != j)
                 {
                     throw new CommandException("commands.execute.failed", new Object[] {"detect", entity.getCommandSenderName()});
                 }
