--- a/net/minecraft/command/CommandBase.java
+++ b/net/minecraft/command/CommandBase.java
@@ -155,7 +155,7 @@
     public static BlockPos func_175757_a(ICommandSender p_175757_0_, String[] p_175757_1_, int p_175757_2_, boolean p_175757_3_) throws NumberInvalidException
     {
         BlockPos blockpos = p_175757_0_.func_180425_c();
-        return new BlockPos(func_175769_b((double)blockpos.func_177958_n(), p_175757_1_[p_175757_2_], -30000000, 30000000, p_175757_3_), func_175769_b((double)blockpos.func_177956_o(), p_175757_1_[p_175757_2_ + 1], 0, 256, false), func_175769_b((double)blockpos.func_177952_p(), p_175757_1_[p_175757_2_ + 2], -30000000, 30000000, p_175757_3_));
+        return new BlockPos(func_175769_b((double)blockpos.getX(), p_175757_1_[p_175757_2_], -30000000, 30000000, p_175757_3_), func_175769_b((double)blockpos.getY(), p_175757_1_[p_175757_2_ + 1], 0, 256, false), func_175769_b((double)blockpos.getZ(), p_175757_1_[p_175757_2_ + 2], -30000000, 30000000, p_175757_3_));
     }
 
     public static double func_175765_c(String p_175765_0_) throws NumberInvalidException
@@ -647,11 +647,11 @@
 
             if (i == p_175771_1_)
             {
-                s = Integer.toString(p_175771_2_.func_177958_n());
+                s = Integer.toString(p_175771_2_.getX());
             }
             else if (i == p_175771_1_ + 1)
             {
-                s = Integer.toString(p_175771_2_.func_177956_o());
+                s = Integer.toString(p_175771_2_.getY());
             }
             else
             {
@@ -660,7 +660,7 @@
                     return Collections.<String>emptyList();
                 }
 
-                s = Integer.toString(p_175771_2_.func_177952_p());
+                s = Integer.toString(p_175771_2_.getZ());
             }
 
             return Lists.newArrayList(new String[] {s});
@@ -681,7 +681,7 @@
 
             if (i == p_181043_1_)
             {
-                s = Integer.toString(p_181043_2_.func_177958_n());
+                s = Integer.toString(p_181043_2_.getX());
             }
             else
             {
@@ -690,7 +690,7 @@
                     return null;
                 }
 
-                s = Integer.toString(p_181043_2_.func_177952_p());
+                s = Integer.toString(p_181043_2_.getZ());
             }
 
             return Lists.newArrayList(new String[] {s});
