--- a/net/minecraft/command/CommandCompare.java
+++ b/net/minecraft/command/CommandCompare.java
@@ -77,13 +77,13 @@
                             for (int l = structureboundingbox.minX; l <= structureboundingbox.maxX; ++l)
                             {
                                 blockpos$mutableblockpos.func_181079_c(l, k, j);
-                                blockpos$mutableblockpos1.func_181079_c(l + blockpos3.func_177958_n(), k + blockpos3.func_177956_o(), j + blockpos3.func_177952_p());
+                                blockpos$mutableblockpos1.func_181079_c(l + blockpos3.getX(), k + blockpos3.getY(), j + blockpos3.getZ());
                                 boolean flag1 = false;
-                                IBlockState iblockstate = world.func_180495_p(blockpos$mutableblockpos);
+                                IBlockState iblockstate = world.getBlockState(blockpos$mutableblockpos);
 
-                                if (!flag || iblockstate.func_177230_c() != Blocks.air)
+                                if (!flag || iblockstate.getBlock() != Blocks.air)
                                 {
-                                    if (iblockstate == world.func_180495_p(blockpos$mutableblockpos1))
+                                    if (iblockstate == world.getBlockState(blockpos$mutableblockpos1))
                                     {
                                         TileEntity tileentity = world.func_175625_s(blockpos$mutableblockpos);
                                         TileEntity tileentity1 = world.func_175625_s(blockpos$mutableblockpos1);
