--- a/net/minecraft/command/CommandBlockData.java
+++ b/net/minecraft/command/CommandBlockData.java
@@ -50,7 +50,7 @@
             }
             else
             {
-                IBlockState iblockstate = world.func_180495_p(blockpos);
+                IBlockState iblockstate = world.getBlockState(blockpos);
                 TileEntity tileentity = world.func_175625_s(blockpos);
 
                 if (tileentity == null)
@@ -73,9 +73,9 @@
                     }
 
                     nbttagcompound.func_179237_a(nbttagcompound2);
-                    nbttagcompound.setInteger("x", blockpos.func_177958_n());
-                    nbttagcompound.setInteger("y", blockpos.func_177956_o());
-                    nbttagcompound.setInteger("z", blockpos.func_177952_p());
+                    nbttagcompound.setInteger("x", blockpos.getX());
+                    nbttagcompound.setInteger("y", blockpos.getY());
+                    nbttagcompound.setInteger("z", blockpos.getZ());
 
                     if (nbttagcompound.equals(nbttagcompound1))
                     {
