--- a/net/minecraft/command/CommandFill.java
+++ b/net/minecraft/command/CommandFill.java
@@ -55,23 +55,23 @@
                 i = func_175764_a(p_184881_3_[7], 0, 15);
             }
 
-            BlockPos blockpos2 = new BlockPos(Math.min(blockpos.func_177958_n(), blockpos1.func_177958_n()), Math.min(blockpos.func_177956_o(), blockpos1.func_177956_o()), Math.min(blockpos.func_177952_p(), blockpos1.func_177952_p()));
-            BlockPos blockpos3 = new BlockPos(Math.max(blockpos.func_177958_n(), blockpos1.func_177958_n()), Math.max(blockpos.func_177956_o(), blockpos1.func_177956_o()), Math.max(blockpos.func_177952_p(), blockpos1.func_177952_p()));
-            int j = (blockpos3.func_177958_n() - blockpos2.func_177958_n() + 1) * (blockpos3.func_177956_o() - blockpos2.func_177956_o() + 1) * (blockpos3.func_177952_p() - blockpos2.func_177952_p() + 1);
+            BlockPos blockpos2 = new BlockPos(Math.min(blockpos.getX(), blockpos1.getX()), Math.min(blockpos.getY(), blockpos1.getY()), Math.min(blockpos.getZ(), blockpos1.getZ()));
+            BlockPos blockpos3 = new BlockPos(Math.max(blockpos.getX(), blockpos1.getX()), Math.max(blockpos.getY(), blockpos1.getY()), Math.max(blockpos.getZ(), blockpos1.getZ()));
+            int j = (blockpos3.getX() - blockpos2.getX() + 1) * (blockpos3.getY() - blockpos2.getY() + 1) * (blockpos3.getZ() - blockpos2.getZ() + 1);
 
             if (j > 32768)
             {
                 throw new CommandException("commands.fill.tooManyBlocks", new Object[] {Integer.valueOf(j), Integer.valueOf(32768)});
             }
-            else if (blockpos2.func_177956_o() >= 0 && blockpos3.func_177956_o() < 256)
+            else if (blockpos2.getY() >= 0 && blockpos3.getY() < 256)
             {
                 World world = p_184881_2_.getEntityWorld();
 
-                for (int k = blockpos2.func_177952_p(); k <= blockpos3.func_177952_p(); k += 16)
+                for (int k = blockpos2.getZ(); k <= blockpos3.getZ(); k += 16)
                 {
-                    for (int l = blockpos2.func_177958_n(); l <= blockpos3.func_177958_n(); l += 16)
+                    for (int l = blockpos2.getX(); l <= blockpos3.getX(); l += 16)
                     {
-                        if (!world.func_175667_e(new BlockPos(l, blockpos3.func_177956_o() - blockpos2.func_177956_o(), k)))
+                        if (!world.func_175667_e(new BlockPos(l, blockpos3.getY() - blockpos2.getY(), k)))
                         {
                             throw new CommandException("commands.fill.outOfWorld", new Object[0]);
                         }
@@ -99,11 +99,11 @@
                 List<BlockPos> list = Lists.<BlockPos>newArrayList();
                 j = 0;
 
-                for (int i1 = blockpos2.func_177952_p(); i1 <= blockpos3.func_177952_p(); ++i1)
+                for (int i1 = blockpos2.getZ(); i1 <= blockpos3.getZ(); ++i1)
                 {
-                    for (int j1 = blockpos2.func_177956_o(); j1 <= blockpos3.func_177956_o(); ++j1)
+                    for (int j1 = blockpos2.getY(); j1 <= blockpos3.getY(); ++j1)
                     {
-                        for (int k1 = blockpos2.func_177958_n(); k1 <= blockpos3.func_177958_n(); ++k1)
+                        for (int k1 = blockpos2.getX(); k1 <= blockpos3.getX(); ++k1)
                         {
                             BlockPos blockpos4 = new BlockPos(k1, j1, i1);
 
@@ -117,7 +117,7 @@
                                     }
                                     else if ("keep".equals(p_184881_3_[8]))
                                     {
-                                        if (!world.func_175623_d(blockpos4))
+                                        if (!world.isEmpty(blockpos4))
                                         {
                                             continue;
                                         }
@@ -128,7 +128,7 @@
                                         {
                                             Block block1 = CommandBase.getBlockByText(p_184881_2_, p_184881_3_[9]);
 
-                                            if (world.func_180495_p(blockpos4).func_177230_c() != block1)
+                                            if (world.getBlockState(blockpos4).getBlock() != block1)
                                             {
                                                 continue;
                                             }
@@ -137,20 +137,20 @@
                                         if (p_184881_3_.length > 10)
                                         {
                                             int l1 = CommandBase.func_175755_a(p_184881_3_[10]);
-                                            IBlockState iblockstate = world.func_180495_p(blockpos4);
+                                            IBlockState iblockstate = world.getBlockState(blockpos4);
 
-                                            if (iblockstate.func_177230_c().func_176201_c(iblockstate) != l1)
+                                            if (iblockstate.getBlock().func_176201_c(iblockstate) != l1)
                                             {
                                                 continue;
                                             }
                                         }
                                     }
                                 }
-                                else if (k1 != blockpos2.func_177958_n() && k1 != blockpos3.func_177958_n() && j1 != blockpos2.func_177956_o() && j1 != blockpos3.func_177956_o() && i1 != blockpos2.func_177952_p() && i1 != blockpos3.func_177952_p())
+                                else if (k1 != blockpos2.getX() && k1 != blockpos3.getX() && j1 != blockpos2.getY() && j1 != blockpos3.getY() && i1 != blockpos2.getZ() && i1 != blockpos3.getZ())
                                 {
                                     if ("hollow".equals(p_184881_3_[8]))
                                     {
-                                        world.func_180501_a(blockpos4, Blocks.air.func_176223_P(), 2);
+                                        world.func_180501_a(blockpos4, Blocks.air.getBlockState(), 2);
                                         list.add(blockpos4);
                                     }
 
@@ -167,7 +167,7 @@
                                     ((IInventory)tileentity1).func_174888_l();
                                 }
 
-                                world.func_180501_a(blockpos4, Blocks.field_180401_cv.func_176223_P(), block == Blocks.field_180401_cv ? 2 : 4);
+                                world.func_180501_a(blockpos4, Blocks.field_180401_cv.getBlockState(), block == Blocks.field_180401_cv ? 2 : 4);
                             }
 
                             IBlockState iblockstate1 = block.func_176203_a(i);
@@ -183,9 +183,9 @@
 
                                     if (tileentity != null)
                                     {
-                                        nbttagcompound.setInteger("x", blockpos4.func_177958_n());
-                                        nbttagcompound.setInteger("y", blockpos4.func_177956_o());
-                                        nbttagcompound.setInteger("z", blockpos4.func_177952_p());
+                                        nbttagcompound.setInteger("x", blockpos4.getX());
+                                        nbttagcompound.setInteger("y", blockpos4.getY());
+                                        nbttagcompound.setInteger("z", blockpos4.getZ());
                                         tileentity.readFromNBT(nbttagcompound);
                                     }
                                 }
@@ -196,7 +196,7 @@
 
                 for (BlockPos blockpos5 : list)
                 {
-                    Block block2 = world.func_180495_p(blockpos5).func_177230_c();
+                    Block block2 = world.getBlockState(blockpos5).getBlock();
                     world.func_175722_b(blockpos5, block2);
                 }
 
