--- a/net/minecraft/command/CommandWorldBorder.java
+++ b/net/minecraft/command/CommandWorldBorder.java
@@ -107,8 +107,8 @@
                 }
 
                 BlockPos blockpos = p_184881_2_.func_180425_c();
-                double d1 = func_175761_b((double)blockpos.func_177958_n() + 0.5D, p_184881_3_[1], true);
-                double d3 = func_175761_b((double)blockpos.func_177952_p() + 0.5D, p_184881_3_[2], true);
+                double d1 = func_175761_b((double)blockpos.getX() + 0.5D, p_184881_3_[1], true);
+                double d3 = func_175761_b((double)blockpos.getZ() + 0.5D, p_184881_3_[2], true);
                 worldborder.func_177739_c(d1, d3);
                 notifyOperators(p_184881_2_, this, "commands.worldborder.center.success", new Object[] {Double.valueOf(d1), Double.valueOf(d3)});
             }
