--- a/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
+++ b/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
@@ -282,10 +282,10 @@
     private void func_178510_a(VertexBuffer p_178510_1_, int p_178510_2_, RenderChunk p_178510_3_)
     {
         GlStateManager.func_187423_f(p_178510_2_, 4864);
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         p_178510_3_.func_178572_f();
         this.field_178517_f.func_181679_a(p_178510_1_);
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         GlStateManager.func_187415_K();
     }
 
