--- a/net/minecraft/client/renderer/chunk/ChunkRenderWorker.java
+++ b/net/minecraft/client/renderer/chunk/ChunkRenderWorker.java
@@ -100,7 +100,7 @@
             p_178474_1_.func_178540_f().unlock();
         }
 
-        Entity lvt_2_2_ = Minecraft.getMinecraft().func_175606_aa();
+        Entity lvt_2_2_ = Minecraft.getMinecraft().getRenderViewEntity();
 
         if (lvt_2_2_ == null)
         {
@@ -216,7 +216,7 @@
 
     private boolean func_188263_a(BlockPos p_188263_1_, World p_188263_2_)
     {
-        return !p_188263_2_.getChunkFromChunkCoords(p_188263_1_.func_177958_n() >> 4, p_188263_1_.func_177952_p() >> 4).isEmpty();
+        return !p_188263_2_.getChunkFromChunkCoords(p_188263_1_.getX() >> 4, p_188263_1_.getZ() >> 4).isEmpty();
     }
 
     private RegionRenderCacheBuilder func_178475_b() throws InterruptedException
