--- a/net/minecraft/client/renderer/chunk/RenderChunk.java
+++ b/net/minecraft/client/renderer/chunk/RenderChunk.java
@@ -125,6 +125,7 @@
 
             this.chunk = null;
             this.initModelviewMatrix();
+            this.needsUpdate = true;
         }
     }
 
@@ -140,7 +141,7 @@
             this.postRenderBlocks(BlockRenderLayer.TRANSLUCENT, x, y, z, vertexbuffer, compiledchunk);
         }
     }
-
+    BlockPosM blockposm;
     public void rebuildChunk(float x, float y, float z, ChunkCompileTaskGenerator generator)
     {
         CompiledChunk compiledchunk = new CompiledChunk();
@@ -181,7 +182,7 @@
 
                 for (Object b : BlockPosM.getAllInBoxMutable(blockpos, blockpos1))
                 {
-                    BlockPosM blockposm = (BlockPosM)b;
+                    blockposm = (BlockPosM)b;
                     IBlockState iblockstate = chunkcacheof.getBlockState(blockposm);
                     Block block = iblockstate.getBlock();
 
