--- a/net/minecraft/client/renderer/chunk/VisGraph.java
+++ b/net/minecraft/client/renderer/chunk/VisGraph.java
@@ -25,7 +25,7 @@
 
     private static int func_178608_c(BlockPos p_178608_0_)
     {
-        return func_178605_a(p_178608_0_.func_177958_n() & 15, p_178608_0_.func_177956_o() & 15, p_178608_0_.func_177952_p() & 15);
+        return func_178605_a(p_178608_0_.getX() & 15, p_178608_0_.getY() & 15, p_178608_0_.getZ() & 15);
     }
 
     private static int func_178605_a(int p_178605_0_, int p_178605_1_, int p_178605_2_)
