--- a/net/minecraft/client/renderer/block/statemap/DefaultStateMapper.java
+++ b/net/minecraft/client/renderer/block/statemap/DefaultStateMapper.java
@@ -9,6 +9,6 @@
 {
     protected ModelResourceLocation func_178132_a(IBlockState p_178132_1_)
     {
-        return new ModelResourceLocation((ResourceLocation)Block.blockRegistry.func_177774_c(p_178132_1_.func_177230_c()), this.func_178131_a(p_178132_1_.func_177228_b()));
+        return new ModelResourceLocation((ResourceLocation)Block.blockRegistry.func_177774_c(p_178132_1_.getBlock()), this.func_178131_a(p_178132_1_.func_177228_b()));
     }
 }
