--- a/net/minecraft/network/play/server/SPacketMultiBlockChange.java
+++ b/net/minecraft/network/play/server/SPacketMultiBlockChange.java
@@ -87,7 +87,7 @@
         public BlockUpdateData(short p_i46545_2_, Chunk p_i46545_3_)
         {
             this.field_180091_b = p_i46545_2_;
-            this.field_180092_c = p_i46545_3_.func_177435_g(this.func_180090_a());
+            this.field_180092_c = p_i46545_3_.getBlockState(this.func_180090_a());
         }
 
         public BlockPos func_180090_a()
