--- a/net/minecraft/network/play/server/SPacketChunkData.java
+++ b/net/minecraft/network/play/server/SPacketChunkData.java
@@ -42,7 +42,7 @@
         {
             BlockPos blockpos = (BlockPos)entry.getKey();
             TileEntity tileentity = (TileEntity)entry.getValue();
-            int i = blockpos.func_177956_o() >> 4;
+            int i = blockpos.getY() >> 4;
 
             if (this.func_149274_i() || (p_i47124_2_ & 1 << i) != 0)
             {
