--- a/net/minecraft/network/play/server/SPacketExplosion.java
+++ b/net/minecraft/network/play/server/SPacketExplosion.java
@@ -84,9 +84,9 @@
 
         for (BlockPos blockpos : this.field_149155_e)
         {
-            int l = blockpos.func_177958_n() - i;
-            int i1 = blockpos.func_177956_o() - j;
-            int j1 = blockpos.func_177952_p() - k;
+            int l = blockpos.getX() - i;
+            int i1 = blockpos.getY() - j;
+            int j1 = blockpos.getZ() - k;
             data.writeByte(l);
             data.writeByte(i1);
             data.writeByte(j1);
