--- a/net/minecraft/client/entity/EntityPlayerSP.java
+++ b/net/minecraft/client/entity/EntityPlayerSP.java
@@ -1,7 +1,15 @@
 package net.minecraft.client.entity;
 
+import java.lang.reflect.Field;
 import java.util.List;
 import javax.annotation.Nullable;
+
+import com.mtbs3d.minecrift.api.IRoomscaleAdapter;
+import com.mtbs3d.minecrift.provider.OpenVRPlayer;
+import com.mtbs3d.minecrift.settings.VRSettings;
+import com.mtbs3d.minecrift.utils.Utils;
+
+import jopenvr.VR_IVRSystem_FnTable.GetEyeToHeadTransform_callback;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.audio.ElytraSound;
@@ -134,6 +142,21 @@
      */
     protected int sprintToggleTimer;
 
+    
+    // VIVE START - teleport movement
+    public int movementTeleportTimer;
+	public boolean initFromServer;
+	public float spEyeHeight;
+    // VIVE END - teleport movement
+    /** MINECRIFT **/
+    public float headPitch = 0;
+    public double additionX = 0;
+    public double additionZ = 0;
+    public double additionY = 0;
+    public final float PIOVER180 = (float)Math.PI / 180.0F;
+    /** END MINECRIFT **/
+    
+    
     /** Ticks left before sprinting is disabled. */
     public int sprintingTicksLeft;
     public float renderArmYaw;
@@ -151,9 +174,9 @@
     private boolean handActive;
     private EnumHand activeHand;
     private boolean rowingBoat;
-    private boolean field_189811_cr = true;
+    private boolean autoJump = true;
     private int field_189812_cs;
-    private boolean field_189813_ct;
+    private boolean wasElytraFlying;
 
     public EntityPlayerSP(Minecraft mcIn, World worldIn, NetHandlerPlayClient netHandler, StatisticsManager statFile)
     {
@@ -162,8 +185,23 @@
         this.statWriter = statFile;
         this.mc = mcIn;
         this.dimension = 0;
+       
+        
     }
 
+    //VIVE
+    public void setItemInUseClient(ItemStack item){
+    	this.activeItemStack = item;
+    }
+    
+    
+    public void setItemInUseCountClient(int count){
+    	this.activeItemStackUseCount = count;
+    }
+    
+    //END VIVECRAFT
+    
+    
     /**
      * Called when the entity is attacked.
      */
@@ -322,7 +360,12 @@
             }
 
             this.prevOnGround = this.onGround;
-            this.field_189811_cr = this.mc.gameSettings.autoJump;
+            this.autoJump = this.mc.gameSettings.autoJump;
+            
+            
+            mc.vrPlayer.IAmLookingAtMyHand[0] = mc.vrPlayer.shouldIlookatMyHand[0];
+            mc.vrPlayer.IAmLookingAtMyHand[1] = mc.vrPlayer.shouldIlookatMyHand[1];
+            
         }
     }
 
@@ -698,7 +741,7 @@
             }
         }
 
-        if (FLAGS.equals(key) && this.isElytraFlying() && !this.field_189813_ct)
+        if (FLAGS.equals(key) && this.isElytraFlying() && !this.wasElytraFlying)
         {
             this.mc.getSoundHandler().playSound(new ElytraSound(this));
         }
@@ -977,7 +1020,7 @@
                     this.sendPlayerAbilities();
                 }
             }
-            else if (!flag && this.movementInput.jump && !flag3)
+            else if (!flag &&   this.movementInput.jump && !flag3)
             {
                 if (this.flyToggleTimer == 0)
                 {
@@ -1002,7 +1045,7 @@
             }
         }
 
-        this.field_189813_ct = this.isElytraFlying();
+        this.wasElytraFlying = this.isElytraFlying();
 
         if (this.capabilities.isFlying && this.isCurrentViewEntity())
         {
@@ -1065,6 +1108,11 @@
 
         super.onLivingUpdate();
 
+        // VIVE START - teleport movement
+        mc.vrPlayer.onLivingUpdate(this, this.mc, this.rand);
+        // VIVE END - teleport movement
+
+        
         if (this.onGround && this.capabilities.isFlying && !this.mc.playerController.isSpectatorMode())
         {
             this.capabilities.isFlying = false;
@@ -1115,20 +1163,53 @@
      */
     public void moveEntity(double x, double y, double z)
     {
-        double d0 = this.posX;
-        double d1 = this.posZ;
-        super.moveEntity(x, y, z);
-        this.func_189810_i((float)(this.posX - d0), (float)(this.posZ - d1));
+    	boolean client = this instanceof EntityPlayerSP;
+    	boolean freemove = OpenVRPlayer.get().getFreeMoveMode();
+    	boolean doY = freemove || (Minecraft.getMinecraft().vrSettings.simulateFalling && !this.isOnLadder()) && !this.isSneaking();
+    	Vec3d roomOrigin = OpenVRPlayer.get().getRoomOriginPos_World();
+ //   	Vec3 camloc = Minecraft.getMinecraft().roomScale.getHMDPos_World();
+
+    	if (this.isRiding() ||  freemove && (this.moveForward != 0 || this.isElytraFlying() || Math.abs(this.motionX) > 0.01 || Math.abs(this.motionZ) > 0.01))
+    	{
+    		double ox = roomOrigin.xCoord - posX;
+    		double oz = roomOrigin.zCoord - posZ;
+            double d0 = this.posX;
+            double d1 = this.posZ;
+    		super.moveEntity(x,y,z);
+    		this.doAutoJump((float)(this.posX - d0), (float)(this.posZ - d1));
+    		if (client){	
+    			double oy = this.posY;
+    			if (this.isRiding()) oy = this.getRidingEntity().posY;
+    			OpenVRPlayer.get().setRoomOrigin(
+    					posX + ox, 
+    					oy, 
+    					posZ  + oz, false, false);
+    					}
+    	} else {
+    		if(doY) {      	 
+    			super.moveEntity(0,y,0);
+    			if (client){
+    				OpenVRPlayer.get().setRoomOrigin(
+    					OpenVRPlayer.get().getRoomOriginPos_World().xCoord, 
+    					this.posY,
+    					OpenVRPlayer.get().getRoomOriginPos_World().zCoord, false, false);
+    			}
+
+    		} else {
+    			this.onGround = true; //
+    			//do not move player, VRPlayer.moveplayerinroom will move him around.
+    		}
+    	}
     }
 
-    public boolean func_189809_N()
+    public boolean autoJumpEnabled()
     {
-        return this.field_189811_cr;
+        return this.autoJump;
     }
 
-    protected void func_189810_i(float p_189810_1_, float p_189810_2_)
+    protected void doAutoJump(float p_189810_1_, float p_189810_2_)
     {
-        if (this.func_189809_N())
+        if (this.autoJumpEnabled())
         {
             if (this.field_189812_cs <= 0 && this.onGround && !this.isSneaking() && !this.isRiding())
             {
@@ -1144,12 +1225,16 @@
                     float f = this.getAIMoveSpeed();
                     float f1 = (float)vec3d2.lengthSquared();
 
+                    //VIVE
+                    float yaw = Minecraft.getMinecraft().roomScale.getControllerYaw_World(0);
+                    //END VIVE
+                    
                     if (f1 <= 0.001F)
                     {
                         float f2 = f * vec2f.x;
                         float f3 = f * vec2f.y;
-                        float f4 = MathHelper.sin(this.rotationYaw * 0.017453292F);
-                        float f5 = MathHelper.cos(this.rotationYaw * 0.017453292F);
+                        float f4 = MathHelper.sin(yaw * 0.017453292F);
+                        float f5 = MathHelper.cos(yaw * 0.017453292F);
                         vec3d2 = new Vec3d((double)(f2 * f5 - f3 * f4), vec3d2.yCoord, (double)(f3 * f5 + f2 * f4));
                         f1 = (float)vec3d2.lengthSquared();
 
@@ -1269,4 +1354,210 @@
             }
         }
     }
+    
+    
+    // VIVECREAFT ADDITIONS
+    
+    
+    /**
+     * Sets the location and Yaw/Pitch of an entity in the world
+     */
+    @Override
+    public void setLocationAndAngles(double x, double y, double z, float yaw, float pitch)
+    {
+    	 	
+//    	Vec3 roomOrigin = OpenVRPlayer.get().getRoomOriginPos_World();
+//    	double ox = roomOrigin.xCoord - posX;
+//    	double oz = roomOrigin.zCoord - posZ;
+//    	
+    	super.setLocationAndAngles(x, y, z, yaw, pitch);
+    	
+    	if(!initFromServer) return;
+    	else { //still init.
+            mc.vrPlayer.snapRoomOriginToPlayerEntity(this, true, false);
+            
+//    		OpenVRPlayer.get().setRoomOrigin(
+//    				posX + ox, 
+//    				this.boundingBox.minY, 
+//    				posZ  + oz, false);
+    	}
+    }
+    
+    @Override
+    public void setPositionAndRotation(double x, double y, double z, float yaw, float pitch)
+    {
+        super.setPositionAndRotation(x, y, z, yaw, pitch);
+		
+        OpenVRPlayer.get().snapRoomOriginToPlayerEntity(this, true, false);
+      
+		if(initFromServer == false){
+            initFromServer = true;    
+        }
+    }
+    
+    // VIVE START - update room origin when player entity is moved
+    @Override
+    public void setPosition(double x, double y, double z)
+    { //this is also called when riding to move this around.
+    	if(this.isRiding()){
+    		Vec3d roomOrigin = OpenVRPlayer.get().getRoomOriginPos_World();
+    		double ox = roomOrigin.xCoord - posX;
+    		double oz = roomOrigin.zCoord - posZ;
+    		super.setPosition(x, y, z);
+    		OpenVRPlayer.get().setRoomOrigin(
+    				posX + ox, 
+    				this.getRidingEntity().posY,
+    				posZ  + oz, false, false);
+
+    	} else
+    		super.setPosition(x, y, z);
+
+    }
+        
+    
+    /** MINECRIFT **/
+    public void doDrag()
+    {
+
+   	
+    	float var3 = 0.91F;
+
+    	if (this.onGround)
+    	{
+    		var3 = this.worldObj.getBlockState(new BlockPos(this.posX, this.getEntityBoundingBox().minY - 1.0D, this.posZ)).getBlock().slipperiness * 0.91F;
+    	}
+
+    	double xFactor = var3;
+    	double zFactor = var3;
+
+    	//VIVECRAFT account for stock drag code we can't change in EntityLivingBase
+    	this.motionX /= xFactor;
+    	this.motionZ /= zFactor;
+    	//
+
+    	int inertiaFactor = Minecraft.getMinecraft().vrSettings.inertiaFactor;
+    	double addFactor = VRSettings.getInertiaAddFactor(inertiaFactor);
+
+    	double boundedAdditionX = getBoundedAddition(additionX);
+    	double targetLimitX = (var3 * boundedAdditionX) / (1f - var3);
+    	double multiFactorX = targetLimitX / (var3 * (targetLimitX + (boundedAdditionX * addFactor)));
+    	xFactor *= multiFactorX;
+
+    	double boundedAdditionZ = getBoundedAddition(additionZ);
+    	double targetLimitZ = (var3 * boundedAdditionZ) / (1f - var3);
+    	double multiFactorZ = targetLimitZ / (var3 * (targetLimitZ + (boundedAdditionZ * addFactor)));
+    	zFactor *= multiFactorZ;
+
+
+    	this.motionX *= xFactor;
+    	this.motionZ *= zFactor;
+    }
+    
+    public double getBoundedAddition(double orig)
+    {
+        if (orig >= -1.0E-6D && orig <= 1.0E-6D) {
+            return 1.0E-6D;
+        }
+        else {
+            return orig;
+        }
+    }
+    
+    
+    /**
+     * Used in both water and by flying objects
+     */
+    @Override
+    public void moveRelative(float strafe, float forward, float friction)
+    {
+//   	super.moveFlying(strafe, forward, friction);;
+    	
+    	OpenVRPlayer vr = mc.vrPlayer;		
+    	if (!vr.getFreeMoveMode()) {return;}
+		
+        int inertiaFactor = Minecraft.getMinecraft().vrSettings.inertiaFactor;
+        float speed = strafe * strafe + forward * forward;
+
+        double mX = 0d;
+        double mZ = 0d;
+        double mY = 0d;
+        double addFactor = 1f;
+
+        if (speed >= 1.0E-4F)
+        {
+            speed = MathHelper.sqrt_float(speed);
+
+            if (speed < 1.0F)
+            {
+                speed = 1.0F;
+            }
+
+            speed = friction / speed;
+            strafe *= speed;
+            forward *= speed;
+
+            
+            Vec3d directionVec = new Vec3d(strafe, 0,forward);
+	  						
+         	//VIVE DO PITCH FLYING		
+
+    		IRoomscaleAdapter con = mc.roomScale;
+    		
+            // Account for pitch if required
+    		if(this.capabilities.isFlying || this.inWater)
+            if (Minecraft.getMinecraft().vrSettings.allowPitchAffectsHeightWhileFlying) {
+            	if (vr.useLControllerForRestricedMovement) {//these are negative cause directionvec is backwards in the z direction.
+            		directionVec= directionVec.rotatePitch(con.getControllerPitch_World(1) * PIOVER180);		   
+            	} else {
+            		directionVec=directionVec.rotatePitch(con.getHMDPitch_World()* PIOVER180);	
+            	}		
+            }
+    		
+    		// Account for yaw
+            if (vr.useLControllerForRestricedMovement) {//these are negative cause directionvec is backwards in the z direction.
+            	directionVec=directionVec.rotateYaw(-con.getControllerYaw_World(1) * PIOVER180);	
+            } else {
+            	directionVec=directionVec.rotateYaw(-con.getHMDYaw_World()* PIOVER180);
+    		}
+    	
+            mX = directionVec.xCoord;
+            mY = directionVec.yCoord;
+            mZ = directionVec.zCoord;
+
+     	   // Modify acceleration sequence (each tick)
+            if(!this.capabilities.isFlying && !this.inWater) addFactor = VRSettings.getInertiaAddFactor(inertiaFactor);
+            
+            float yAdd = 1f;
+            if(this.capabilities.isFlying) yAdd = 3f; //HACK
+            
+            this.motionX = this.motionX + (mX * addFactor);
+            this.motionZ = this.motionZ + (mZ * addFactor);
+            this.motionY = this.motionY + (mY * yAdd);
+      
+            this.additionX = mX;
+            this.additionZ = mZ;
+        }
+        
+        //if (this instanceof EntityPlayerSP) {
+        //    System.out.println(String.format("FLYING: %B, forward: %.4f, strafe: %.4f, pitch: %.4f, yaw: %.4f, mx: %.4f, mz: %.4f, my: %.4f", allowYAdjust, forward, strafe, this.headPitch, this.rotationYaw, mX, mZ, mY));
+        //}
+
+        
+     if(!this.capabilities.isFlying && !this.inWater) doDrag();       //TODO put back intertia while flying.. doesnt work for some reason. 
+        
+/** END MINECRIFT **/
+   
+    }
+    
+    public float eyeHeightOverride = 0;
+    public boolean overrideEyeHeight;
+    
+    @Override
+    public float getEyeHeight(){
+    	if(overrideEyeHeight) return eyeHeightOverride;
+    	return super.getEyeHeight();
+    	
+    }
+    
+    
 }
