--- a/net/minecraft/client/gui/GuiChat.java
+++ b/net/minecraft/client/gui/GuiChat.java
@@ -292,7 +292,7 @@
 
             if (this.field_186853_g.objectMouseOver != null && this.field_186853_g.objectMouseOver.typeOfHit == RayTraceResult.Type.BLOCK)
             {
-                blockpos = this.field_186853_g.objectMouseOver.func_178782_a();
+                blockpos = this.field_186853_g.objectMouseOver.getBlock();
             }
 
             return blockpos;
