--- a/net/minecraft/client/gui/GuiCreateFlatWorld.java
+++ b/net/minecraft/client/gui/GuiCreateFlatWorld.java
@@ -201,7 +201,7 @@
         {
             FlatLayerInfo flatlayerinfo = (FlatLayerInfo)GuiCreateFlatWorld.this.theFlatGeneratorInfo.getFlatLayers().get(GuiCreateFlatWorld.this.theFlatGeneratorInfo.getFlatLayers().size() - p_180791_1_ - 1);
             IBlockState iblockstate = flatlayerinfo.func_175900_c();
-            Block block = iblockstate.func_177230_c();
+            Block block = iblockstate.getBlock();
             Item item = Item.getItemFromBlock(block);
             ItemStack itemstack = block != Blocks.air && item != null ? new ItemStack(item, 1, block.func_176201_c(iblockstate)) : null;
             String s = itemstack == null ? I18n.format("createWorld.customize.flat.air", new Object[0]) : item.getItemStackDisplayName(itemstack);
