--- a/net/minecraft/client/gui/GuiGameOver.java
+++ b/net/minecraft/client/gui/GuiGameOver.java
@@ -105,10 +105,10 @@
     {
         boolean flag = this.mc.theWorld.getWorldInfo().isHardcoreModeEnabled();
         this.drawGradientRect(0, 0, this.width, this.height, 1615855616, -1602211792);
-        GlStateManager.func_179094_E();
-        GlStateManager.func_179152_a(2.0F, 2.0F, 2.0F);
+        GlStateManager.push();
+        GlStateManager.scale(2.0F, 2.0F, 2.0F);
         this.drawCenteredString(this.fontRendererObj, I18n.format(flag ? "deathScreen.title.hardcore" : "deathScreen.title", new Object[0]), this.width / 2 / 2, 30, 16777215);
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
 
         if (this.field_184871_f != null)
         {
