--- a/net/minecraft/client/gui/GuiIngameMenu.java
+++ b/net/minecraft/client/gui/GuiIngameMenu.java
@@ -1,11 +1,18 @@
 package net.minecraft.client.gui;
 
 import java.io.IOException;
+
+import com.mtbs3d.minecrift.gui.GuiQuickCommandsInGame;
+import com.mtbs3d.minecrift.provider.MCOpenVR;
+import com.mtbs3d.minecrift.settings.AutoCalibration;
+import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.achievement.GuiAchievements;
 import net.minecraft.client.gui.achievement.GuiStats;
 import net.minecraft.client.multiplayer.WorldClient;
 import net.minecraft.client.resources.I18n;
+import net.minecraft.client.settings.KeyBinding;
 import net.minecraft.realms.RealmsBridge;
+import net.minecraft.src.Reflector;
 
 public class GuiIngameMenu extends GuiScreen
 {
@@ -71,27 +78,64 @@
 
             case 2:
             case 3:
-            default:
-                break;
-
             case 4:
                 this.mc.displayGuiScreen((GuiScreen)null);
                 this.mc.setIngameFocus();
                 break;
 
             case 5:
+            	if (this.mc.player != null)
                 this.mc.displayGuiScreen(new GuiAchievements(this, this.mc.player.getStatFileWriter()));
                 break;
 
             case 6:
+            	if (this.mc.player != null)
                 this.mc.displayGuiScreen(new GuiStats(this, this.mc.player.getStatFileWriter()));
-                break;
+            	break;
 
             case 7:
-                this.mc.displayGuiScreen(new GuiShareToLan(this));
+            	this.mc.displayGuiScreen(new GuiShareToLan(this));
+            	break;
+            case 12:
+            	Reflector.call(Reflector.call(Reflector.FMLClientHandler_instance), Reflector.FMLClientHandler_showInGameModOptions,new Object[]{this});
+            	break;
+            case 100:
+            	this.mc.displayGuiScreen(new GuiChat());
+            	break;
+            case 110:
+            	this.mc.displayGuiScreen(new GuiChat());
+            	break;
+            case 101:
+            	this.mc.displayGuiScreen(new GuiQuickCommandsInGame());
+            	this.initGui();
+            	break;
+            case 102:
+            	mc.gameSettings.showDebugInfo = !mc.gameSettings.showDebugInfo;
+            	this.mc.displayGuiScreen((GuiScreen)null);
+            	break;
+            case 103:
+            	mc.gameSettings.showDebugProfilerChart = !mc.gameSettings.showDebugProfilerChart;
+            	mc.gameSettings.showDebugInfo = mc.gameSettings.showDebugProfilerChart;
+            	break;
+            case 104:
+            	this.mc.displayGuiScreen((GuiScreen)null);
+            	this.mc.grabScreenShot = 10;
+            	break;
+	        case 105:
+		        AutoCalibration.calibrateManual();
+		        Minecraft.getMinecraft().vrSettings.saveOptions();
+		        this.mc.displayGuiScreen((GuiScreen)null);
+		        this.mc.setIngameFocus();
+		        break;
+	        case 999:
+            	this.mc.displayGuiScreen((GuiScreen)null);
+            	break;
+            default:
+
         }
     }
 
+
     /**
      * Called from the main game loop to update the screen.
      */
