--- a/net/minecraft/client/gui/GuiIngame.java
+++ b/net/minecraft/client/gui/GuiIngame.java
@@ -243,7 +243,7 @@
 
             if (l1 > 8)
             {
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
                 GlStateManager.func_179109_b((float)(i / 2), (float)(j - 68), 0.0F);
                 GlStateManager.func_179147_l();
                 GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
@@ -256,7 +256,7 @@
 
                 fontrenderer.drawString(this.recordPlaying, -fontrenderer.getStringWidth(this.recordPlaying) / 2, -4, l + (l1 << 24 & -16777216));
                 GlStateManager.func_179084_k();
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
 
             this.mc.mcProfiler.endSection();
@@ -285,21 +285,21 @@
 
             if (i2 > 8)
             {
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
                 GlStateManager.func_179109_b((float)(i / 2), (float)(j / 2), 0.0F);
                 GlStateManager.func_179147_l();
                 GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
-                GlStateManager.func_179094_E();
-                GlStateManager.func_179152_a(4.0F, 4.0F, 4.0F);
+                GlStateManager.push();
+                GlStateManager.scale(4.0F, 4.0F, 4.0F);
                 int j2 = i2 << 24 & -16777216;
                 fontrenderer.func_175065_a(this.field_175201_x, (float)(-fontrenderer.getStringWidth(this.field_175201_x) / 2), -10.0F, 16777215 | j2, true);
-                GlStateManager.func_179121_F();
-                GlStateManager.func_179094_E();
-                GlStateManager.func_179152_a(2.0F, 2.0F, 2.0F);
+                GlStateManager.pop();
+                GlStateManager.push();
+                GlStateManager.scale(2.0F, 2.0F, 2.0F);
                 fontrenderer.func_175065_a(this.field_175200_y, (float)(-fontrenderer.getStringWidth(this.field_175200_y) / 2), 5.0F, 16777215 | j2, true);
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
                 GlStateManager.func_179084_k();
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
 
             this.mc.mcProfiler.endSection();
@@ -329,12 +329,12 @@
         GlStateManager.func_179147_l();
         GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
         GlStateManager.func_179118_c();
-        GlStateManager.func_179094_E();
+        GlStateManager.push();
         GlStateManager.func_179109_b(0.0F, (float)(j - 48), 0.0F);
         this.mc.mcProfiler.startSection("chat");
         this.persistantChatGUI.drawChat(this.updateCounter);
         this.mc.mcProfiler.endSection();
-        GlStateManager.func_179121_F();
+        GlStateManager.pop();
         scoreobjective1 = scoreboard.getObjectiveInDisplaySlot(0);
 
         if (!this.mc.gameSettings.keyBindPlayerList.getIsKeyPressed() || this.mc.isIntegratedServerRunning() && this.mc.thePlayer.sendQueue.func_175106_d().size() <= 1 && scoreobjective1 == null)
@@ -367,9 +367,9 @@
                     return;
                 }
 
-                BlockPos blockpos = raytraceresult.func_178782_a();
+                BlockPos blockpos = raytraceresult.getBlock();
 
-                if (!this.mc.theWorld.func_180495_p(blockpos).func_177230_c().hasTileEntity() || !(this.mc.theWorld.func_175625_s(blockpos) instanceof IInventory))
+                if (!this.mc.theWorld.getBlockState(blockpos).getBlock().hasTileEntity() || !(this.mc.theWorld.func_175625_s(blockpos) instanceof IInventory))
                 {
                     return;
                 }
@@ -380,14 +380,14 @@
 
             if (gamesettings.showDebugInfo && !gamesettings.hideGUI && !this.mc.thePlayer.func_175140_cp() && !gamesettings.field_178879_v)
             {
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
                 GlStateManager.func_179109_b((float)(l / 2), (float)(i1 / 2), this.zLevel);
-                Entity entity = this.mc.func_175606_aa();
-                GlStateManager.func_179114_b(entity.prevRotationPitch + (entity.rotationPitch - entity.prevRotationPitch) * p_184045_1_, -1.0F, 0.0F, 0.0F);
-                GlStateManager.func_179114_b(entity.prevRotationYaw + (entity.rotationYaw - entity.prevRotationYaw) * p_184045_1_, 0.0F, 1.0F, 0.0F);
-                GlStateManager.func_179152_a(-1.0F, -1.0F, -1.0F);
+                Entity entity = this.mc.getRenderViewEntity();
+                GlStateManager.rotate(entity.prevRotationPitch + (entity.rotationPitch - entity.prevRotationPitch) * p_184045_1_, -1.0F, 0.0F, 0.0F);
+                GlStateManager.rotate(entity.prevRotationYaw + (entity.rotationYaw - entity.prevRotationYaw) * p_184045_1_, 0.0F, 1.0F, 0.0F);
+                GlStateManager.scale(-1.0F, -1.0F, -1.0F);
                 OpenGlHelper.func_188785_m(10);
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
             else
             {
@@ -472,12 +472,12 @@
 
     protected void func_180479_a(ScaledResolution p_180479_1_, float p_180479_2_)
     {
-        if (this.mc.func_175606_aa() instanceof EntityPlayer)
+        if (this.mc.getRenderViewEntity() instanceof EntityPlayer)
         {
             GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
             this.mc.getTextureManager().bindTexture(widgetsTexPath);
-            EntityPlayer entityplayer = (EntityPlayer)this.mc.func_175606_aa();
-            ItemStack itemstack = entityplayer.func_184592_cb();
+            EntityPlayer entityplayer = (EntityPlayer)this.mc.getRenderViewEntity();
+            ItemStack itemstack = entityplayer.getOffhandItem();
             EnumHandSide enumhandside = entityplayer.func_184591_cq().func_188468_a();
             int i = p_180479_1_.getScaledWidth() / 2;
             float f = this.zLevel;
@@ -638,12 +638,12 @@
 
             if (k > 0)
             {
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
                 GlStateManager.func_179147_l();
                 GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
                 this.func_175179_f().func_175063_a(s, (float)i, (float)j, 16777215 + (k << 24));
                 GlStateManager.func_179084_k();
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
         }
 
@@ -729,9 +729,9 @@
 
     private void func_180477_d(ScaledResolution p_180477_1_)
     {
-        if (this.mc.func_175606_aa() instanceof EntityPlayer)
+        if (this.mc.getRenderViewEntity() instanceof EntityPlayer)
         {
-            EntityPlayer entityplayer = (EntityPlayer)this.mc.func_175606_aa();
+            EntityPlayer entityplayer = (EntityPlayer)this.mc.getRenderViewEntity();
             int i = MathHelper.ceiling_float_int(entityplayer.getHealth());
             boolean flag = this.field_175191_F > (long)this.updateCounter && (this.field_175191_F - (long)this.updateCounter) / 3L % 2L == 1L;
 
@@ -887,7 +887,7 @@
                 }
             }
 
-            Entity entity = entityplayer.func_184187_bx();
+            Entity entity = entityplayer.getRidingEntity();
 
             if (entity == null)
             {
@@ -952,10 +952,10 @@
 
     private void func_184047_e(ScaledResolution p_184047_1_)
     {
-        if (this.mc.func_175606_aa() instanceof EntityPlayer)
+        if (this.mc.getRenderViewEntity() instanceof EntityPlayer)
         {
-            EntityPlayer entityplayer = (EntityPlayer)this.mc.func_175606_aa();
-            Entity entity = entityplayer.func_184187_bx();
+            EntityPlayer entityplayer = (EntityPlayer)this.mc.getRenderViewEntity();
+            Entity entity = entityplayer.getRidingEntity();
 
             if (entity instanceof EntityLivingBase)
             {
@@ -1088,7 +1088,7 @@
         GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, p_180474_1_);
         this.mc.getTextureManager().bindTexture(TextureMap.locationBlocksTexture);
-        TextureAtlasSprite textureatlassprite = this.mc.func_175602_ab().func_175023_a().func_178122_a(Blocks.portal.func_176223_P());
+        TextureAtlasSprite textureatlassprite = this.mc.func_175602_ab().func_175023_a().func_178122_a(Blocks.portal.getBlockState());
         float f = textureatlassprite.getMinU();
         float f1 = textureatlassprite.getMinV();
         float f2 = textureatlassprite.getMaxU();
@@ -1115,10 +1115,10 @@
 
             if (f > 0.0F)
             {
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
                 float f1 = 1.0F + f / 5.0F;
                 GlStateManager.func_179109_b((float)(p_184044_1_ + 8), (float)(p_184044_2_ + 12), 0.0F);
-                GlStateManager.func_179152_a(1.0F / f1, (f1 + 1.0F) / 2.0F, 1.0F);
+                GlStateManager.scale(1.0F / f1, (f1 + 1.0F) / 2.0F, 1.0F);
                 GlStateManager.func_179109_b((float)(-(p_184044_1_ + 8)), (float)(-(p_184044_2_ + 12)), 0.0F);
             }
 
@@ -1126,7 +1126,7 @@
 
             if (f > 0.0F)
             {
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
 
             this.itemRenderer.func_175030_a(this.mc.fontRendererObj, p_184044_5_, p_184044_1_, p_184044_2_);
