--- a/net/minecraft/client/gui/GuiLanguage.java
+++ b/net/minecraft/client/gui/GuiLanguage.java
@@ -129,15 +129,19 @@
 
         protected void elementClicked(int slotIndex, boolean isDoubleClick, int mouseX, int mouseY)
         {
-            Language language = (Language)this.languageMap.get(this.langCodeList.get(slotIndex));
-            GuiLanguage.this.languageManager.setCurrentLanguage(language);
-            GuiLanguage.this.game_settings_3.language = language.getLanguageCode();
-            this.mc.refreshResources();
-            GuiLanguage.this.fontRendererObj.setUnicodeFlag(GuiLanguage.this.languageManager.isCurrentLocaleUnicode() || GuiLanguage.this.game_settings_3.forceUnicodeFont);
-            GuiLanguage.this.fontRendererObj.setBidiFlag(GuiLanguage.this.languageManager.isCurrentLanguageBidirectional());
-            GuiLanguage.this.confirmSettingsBtn.displayString = I18n.format("gui.done", new Object[0]);
-            GuiLanguage.this.forceUnicodeFontBtn.displayString = GuiLanguage.this.game_settings_3.getKeyBinding(GameSettings.Options.FORCE_UNICODE_FONT);
-            GuiLanguage.this.game_settings_3.saveOptions();
+
+        	Language language = (Language)this.languageMap.get(this.langCodeList.get(slotIndex));
+        	System.out.println(slotIndex + " " + language +  GuiLanguage.this.languageManager.getCurrentLanguage());
+        	if(!language.getLanguageCode().equalsIgnoreCase(GuiLanguage.this.languageManager.getCurrentLanguage().getLanguageCode())){ //VIVECRAFT FIX POSSIBLE LOOP
+        		GuiLanguage.this.languageManager.setCurrentLanguage(language);
+        		GuiLanguage.this.game_settings_3.language = language.getLanguageCode();
+        		this.mc.refreshResources();
+        		GuiLanguage.this.fontRendererObj.setUnicodeFlag(GuiLanguage.this.languageManager.isCurrentLocaleUnicode() || GuiLanguage.this.game_settings_3.forceUnicodeFont);
+        		GuiLanguage.this.fontRendererObj.setBidiFlag(GuiLanguage.this.languageManager.isCurrentLanguageBidirectional());
+        		GuiLanguage.this.confirmSettingsBtn.displayString = I18n.format("gui.done", new Object[0]);
+        		GuiLanguage.this.forceUnicodeFontBtn.displayString = GuiLanguage.this.game_settings_3.getKeyBinding(GameSettings.Options.FORCE_UNICODE_FONT);
+        		GuiLanguage.this.game_settings_3.saveOptions();
+        	}
         }
 
         protected boolean isSelected(int slotIndex)
