--- a/net/minecraft/client/gui/GuiMerchant.java
+++ b/net/minecraft/client/gui/GuiMerchant.java
@@ -165,7 +165,7 @@
             ItemStack itemstack = merchantrecipe.getItemToBuy();
             ItemStack itemstack1 = merchantrecipe.getSecondItemToBuy();
             ItemStack itemstack2 = merchantrecipe.getItemToSell();
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             RenderHelper.enableGUIStandardItemLighting();
             GlStateManager.func_179140_f();
             GlStateManager.func_179091_B();
@@ -203,7 +203,7 @@
                 this.drawCreativeTabHoveringText(I18n.format("merchant.deprecated", new Object[0]), mouseX, mouseY);
             }
 
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
             GlStateManager.func_179145_e();
             GlStateManager.func_179126_j();
             RenderHelper.enableStandardItemLighting();
