--- a/net/minecraft/client/gui/GuiMainMenu.java
+++ b/net/minecraft/client/gui/GuiMainMenu.java
@@ -277,6 +277,7 @@
         {
             this.modUpdateNotification = (GuiScreen)Reflector.call(Reflector.NotificationModUpdateScreen_init, new Object[] {this, this.modButton});
         }
+
     }
 
     /**
@@ -709,6 +710,7 @@
         {
             this.modUpdateNotification.drawScreen(mouseX, mouseY, partialTicks);
         }
+
     }
 
     /**
