--- a/net/minecraft/client/gui/GuiMainMenu.java
+++ b/net/minecraft/client/gui/GuiMainMenu.java
@@ -2,6 +2,7 @@
 
 import com.google.common.base.Strings;
 import com.google.common.collect.Lists;
+
 import java.io.BufferedReader;
 import java.io.Closeable;
 import java.io.IOException;
@@ -12,6 +13,7 @@
 import java.util.List;
 import java.util.Random;
 import java.util.concurrent.atomic.AtomicInteger;
+
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.OpenGlHelper;
@@ -23,6 +25,7 @@
 import net.minecraft.client.resources.IResource;
 import net.minecraft.client.settings.GameSettings;
 import net.minecraft.realms.RealmsBridge;
+import net.minecraft.src.Config;
 import net.minecraft.src.CustomPanorama;
 import net.minecraft.src.CustomPanoramaProperties;
 import net.minecraft.src.Reflector;
@@ -32,6 +35,7 @@
 import net.minecraft.world.demo.DemoWorldServer;
 import net.minecraft.world.storage.ISaveFormat;
 import net.minecraft.world.storage.WorldInfo;
+
 import org.apache.commons.io.Charsets;
 import org.apache.commons.io.IOUtils;
 import org.apache.logging.log4j.LogManager;
@@ -168,6 +172,11 @@
             this.openGLWarning2 = I18n.format("title.oldgl2", new Object[0]);
             this.openGLWarningLink = "https://help.mojang.com/customer/portal/articles/325948?ref=game";
         }
+        else if (Reflector.forgeExists() && Config.isShaders()) // We'll just piggyback off this, not like the OpenGL message should ever show up anyways
+        {
+        	this.openGLWarning1 = "WARNING: You have shaders enabled under Forge! Some mods are";
+        	this.openGLWarning2 = "incompatible, so please set them to OFF if you experience issues.";
+        }
     }
 
     /**
@@ -277,6 +286,7 @@
         {
             this.modUpdateNotification = (GuiScreen)Reflector.call(Reflector.NotificationModUpdateScreen_init, new Object[] {this, this.modButton});
         }
+
     }
 
     /**
@@ -709,6 +719,7 @@
         {
             this.modUpdateNotification.drawScreen(mouseX, mouseY, partialTicks);
         }
+
     }
 
     /**
