--- a/net/minecraft/client/gui/GuiNewChat.java
+++ b/net/minecraft/client/gui/GuiNewChat.java
@@ -51,9 +51,9 @@
 
                 float f1 = this.getChatScale();
                 int k = MathHelper.ceiling_float_int((float)this.getChatWidth() / f1);
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
                 GlStateManager.func_179109_b(2.0F, 8.0F, 0.0F);
-                GlStateManager.func_179152_a(f1, f1, 1.0F);
+                GlStateManager.scale(f1, f1, 1.0F);
                 int l = 0;
 
                 for (int i1 = 0; i1 + this.scrollPos < this.field_146253_i.size() && i1 < i; ++i1)
@@ -114,7 +114,7 @@
                     }
                 }
 
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
         }
     }
