--- a/net/minecraft/client/gui/GuiSpectator.java
+++ b/net/minecraft/client/gui/GuiSpectator.java
@@ -100,12 +100,12 @@
         if (p_175266_5_ != SpectatorMenu.field_178657_a)
         {
             int i = (int)(p_175266_4_ * 255.0F);
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             GlStateManager.func_179109_b((float)p_175266_2_, p_175266_3_, 0.0F);
             float f = p_175266_5_.func_178662_A_() ? 1.0F : 0.25F;
             GlStateManager.func_179131_c(f, f, f, p_175266_4_);
             p_175266_5_.func_178663_a(f, i);
-            GlStateManager.func_179121_F();
+            GlStateManager.pop();
             String s = String.valueOf((Object)GameSettings.getKeyDisplayString(this.field_175268_g.gameSettings.keyBindsHotbar[p_175266_1_].getKeyCode()));
 
             if (i > 3 && p_175266_5_.func_178662_A_())
@@ -128,12 +128,12 @@
             {
                 int j = (p_175263_1_.getScaledWidth() - this.field_175268_g.fontRendererObj.getStringWidth(s)) / 2;
                 int k = p_175263_1_.getScaledHeight() - 35;
-                GlStateManager.func_179094_E();
+                GlStateManager.push();
                 GlStateManager.func_179147_l();
                 GlStateManager.func_187428_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
                 this.field_175268_g.fontRendererObj.func_175063_a(s, (float)j, (float)k, 16777215 + (i << 24));
                 GlStateManager.func_179084_k();
-                GlStateManager.func_179121_F();
+                GlStateManager.pop();
             }
         }
     }
