--- a/net/minecraft/client/gui/GuiTextField.java
+++ b/net/minecraft/client/gui/GuiTextField.java
@@ -2,6 +2,8 @@
 
 import com.google.common.base.Predicate;
 import com.google.common.base.Predicates;
+import com.mtbs3d.minecrift.provider.MCOpenVR;
+
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.client.renderer.VertexBuffer;
@@ -16,9 +18,10 @@
     public int xPosition;
     public int yPosition;
 
+    //Forge make public
     /** The width of this text field. */
-    private final int width;
-    private final int height;
+    public int width;
+    public int height;
 
     /** Has the current text being edited on the textbox. */
     private String text = "";
@@ -511,7 +514,13 @@
         if (this.canLoseFocus)
         {
             this.setFocused(flag);
-        }
+       }  else {
+           	
+            // VIVE START - show keyboard
+             MCOpenVR.setKeyboardOverlayShowing(true, this);
+           // VIVE END - show keyboard
+ 	
+         }
 
         if (this.isFocused && flag && mouseButton == 0)
         {
@@ -717,6 +726,15 @@
         }
 
         this.isFocused = isFocusedIn;
+        
+        // VIVE START - show keyboard
+        for (StackTraceElement ste : Thread.currentThread().getStackTrace()) {
+            if(ste.getClassName().contains("GuiChattableCamera")) return;
+        }
+       if(isFocusedIn) MCOpenVR.setKeyboardOverlayShowing(isFocusedIn, this);
+        // VIVE END - show keyboard
+        
+        
     }
 
     /**
