--- a/net/minecraft/client/gui/GuiTextField.java
+++ b/net/minecraft/client/gui/GuiTextField.java
@@ -2,6 +2,9 @@
 
 import com.google.common.base.Predicate;
 import com.google.common.base.Predicates;
+import com.mtbs3d.minecrift.provider.MCOpenVR;
+
+import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.client.renderer.VertexBuffer;
@@ -16,9 +19,10 @@
     public int xPosition;
     public int yPosition;
 
+    //Forge make public
     /** The width of this text field. */
-    private final int width;
-    private final int height;
+    public int width;
+    public int height;
 
     /** Has the current text being edited on the textbox. */
     private String text = "";
@@ -511,8 +515,13 @@
         if (this.canLoseFocus)
         {
             this.setFocused(flag);
-        }
+       }  else {
+    	   
+       }
+        
+        if(flag && this.canLoseFocus == true) MCOpenVR.setKeyboardOverlayShowing(true, this);
 
+        
         if (this.isFocused && flag && mouseButton == 0)
         {
             int i = mouseX - this.xPosition;
@@ -559,6 +568,9 @@
             {
                 String s1 = flag ? s.substring(0, j) : s;
                 j1 = this.fontRendererInstance.drawStringWithShadow(s1, (float)l, (float)i1, i);
+            } else {
+            	if(!Minecraft.getMinecraft().vrSettings.seated && !MCOpenVR.keyboardShowing && this.canLoseFocus)
+            		this.fontRendererInstance.drawStringWithShadow("Click to open keyboard", (float)l, (float)i1, this.disabledColor);
             }
 
             boolean flag2 = this.cursorPosition < this.text.length() || this.text.length() >= this.getMaxStringLength();
@@ -717,6 +729,16 @@
         }
 
         this.isFocused = isFocusedIn;
+        
+       //  VIVE START - show keyboard
+        //for (StackTraceElement ste : Thread.currentThread().getStackTrace()) {
+        //    if(ste.getClassName().contains("GuiChattableCamera")) return;
+        //   if(ste.toString().contains("journeymap.client.ui.fullscreen.Fullscreen")) return;
+        //}
+        
+       if(isFocusedIn && this.visible && (this.canLoseFocus == false || Minecraft.getMinecraft().vrSettings.autoOpenKeyboard))
+    	   MCOpenVR.setKeyboardOverlayShowing(isFocusedIn, this);
+        //VIVE END - show keyboard            
     }
 
     /**
