--- a/net/minecraft/client/gui/MapItemRenderer.java
+++ b/net/minecraft/client/gui/MapItemRenderer.java
@@ -122,10 +122,10 @@
             {
                 if (!p_148237_1_ || vec4b.func_176110_a() == 1)
                 {
-                    GlStateManager.func_179094_E();
+                    GlStateManager.push();
                     GlStateManager.func_179109_b(0.0F + (float)vec4b.func_176112_b() / 2.0F + 64.0F, 0.0F + (float)vec4b.func_176113_c() / 2.0F + 64.0F, -0.02F);
-                    GlStateManager.func_179114_b((float)(vec4b.func_176111_d() * 360) / 16.0F, 0.0F, 0.0F, 1.0F);
-                    GlStateManager.func_179152_a(4.0F, 4.0F, 3.0F);
+                    GlStateManager.rotate((float)(vec4b.func_176111_d() * 360) / 16.0F, 0.0F, 0.0F, 1.0F);
+                    GlStateManager.scale(4.0F, 4.0F, 3.0F);
                     GlStateManager.func_179109_b(-0.125F, 0.125F, 0.0F);
                     byte b0 = vec4b.func_176110_a();
                     float f1 = (float)(b0 % 4 + 0) / 4.0F;
@@ -139,15 +139,15 @@
                     vertexbuffer.func_181662_b(1.0D, -1.0D, (double)((float)k * -0.001F)).func_187315_a((double)f3, (double)f4).func_181675_d();
                     vertexbuffer.func_181662_b(-1.0D, -1.0D, (double)((float)k * -0.001F)).func_187315_a((double)f1, (double)f4).func_181675_d();
                     tessellator.draw();
-                    GlStateManager.func_179121_F();
+                    GlStateManager.pop();
                     ++k;
                 }
             }
 
-            GlStateManager.func_179094_E();
+            GlStateManager.push();
             GlStateManager.func_179109_b(0.0F, 0.0F, -0.04F);
-            GlStateManager.func_179152_a(1.0F, 1.0F, 1.0F);
-            GlStateManager.func_179121_F();
+            GlStateManager.scale(1.0F, 1.0F, 1.0F);
+            GlStateManager.pop();
         }
     }
 }
