--- a/net/minecraft/client/renderer/block/model/ItemCameraTransforms.java
+++ b/net/minecraft/client/renderer/block/model/ItemCameraTransforms.java
@@ -82,7 +82,7 @@
             }
 
             GlStateManager.func_187444_a(func_188035_a(f, f1, f2));
-            GlStateManager.func_179152_a(field_181696_h + p_188034_0_.field_178363_d.x, field_181697_i + p_188034_0_.field_178363_d.y, field_181698_j + p_188034_0_.field_178363_d.z);
+            GlStateManager.scale(field_181696_h + p_188034_0_.field_178363_d.x, field_181697_i + p_188034_0_.field_178363_d.y, field_181698_j + p_188034_0_.field_178363_d.z);
         }
     }
 
