--- a/net/minecraft/world/gen/feature/WorldGenIceSpike.java
+++ b/net/minecraft/world/gen/feature/WorldGenIceSpike.java
@@ -13,24 +13,24 @@
 {
     public boolean func_180709_b(World p_180709_1_, Random p_180709_2_, BlockPos p_180709_3_)
     {
-        while (p_180709_1_.func_175623_d(p_180709_3_) && p_180709_3_.func_177956_o() > 2)
+        while (p_180709_1_.isEmpty(p_180709_3_) && p_180709_3_.getY() > 2)
         {
-            p_180709_3_ = p_180709_3_.func_177977_b();
+            p_180709_3_ = p_180709_3_.getBlockPosBelow();
         }
 
-        if (p_180709_1_.func_180495_p(p_180709_3_).func_177230_c() != Blocks.snow)
+        if (p_180709_1_.getBlockState(p_180709_3_).getBlock() != Blocks.snow)
         {
             return false;
         }
         else
         {
-            p_180709_3_ = p_180709_3_.func_177981_b(p_180709_2_.nextInt(4));
+            p_180709_3_ = p_180709_3_.getBlockPosAbove(p_180709_2_.nextInt(4));
             int i = p_180709_2_.nextInt(4) + 7;
             int j = i / 4 + p_180709_2_.nextInt(2);
 
             if (j > 1 && p_180709_2_.nextInt(60) == 0)
             {
-                p_180709_3_ = p_180709_3_.func_177981_b(10 + p_180709_2_.nextInt(30));
+                p_180709_3_ = p_180709_3_.getBlockPosAbove(10 + p_180709_2_.nextInt(30));
             }
 
             for (int k = 0; k < i; ++k)
@@ -48,22 +48,22 @@
 
                         if ((i1 == 0 && j1 == 0 || f1 * f1 + f2 * f2 <= f * f) && (i1 != -l && i1 != l && j1 != -l && j1 != l || p_180709_2_.nextFloat() <= 0.75F))
                         {
-                            IBlockState iblockstate = p_180709_1_.func_180495_p(p_180709_3_.func_177982_a(i1, k, j1));
-                            Block block = iblockstate.func_177230_c();
+                            IBlockState iblockstate = p_180709_1_.getBlockState(p_180709_3_.func_177982_a(i1, k, j1));
+                            Block block = iblockstate.getBlock();
 
-                            if (iblockstate.func_185904_a() == Material.air || block == Blocks.dirt || block == Blocks.snow || block == Blocks.ice)
+                            if (iblockstate.getMaterial() == Material.air || block == Blocks.dirt || block == Blocks.snow || block == Blocks.ice)
                             {
-                                this.func_175903_a(p_180709_1_, p_180709_3_.func_177982_a(i1, k, j1), Blocks.packed_ice.func_176223_P());
+                                this.func_175903_a(p_180709_1_, p_180709_3_.func_177982_a(i1, k, j1), Blocks.packed_ice.getBlockState());
                             }
 
                             if (k != 0 && l > 1)
                             {
-                                iblockstate = p_180709_1_.func_180495_p(p_180709_3_.func_177982_a(i1, -k, j1));
-                                block = iblockstate.func_177230_c();
+                                iblockstate = p_180709_1_.getBlockState(p_180709_3_.func_177982_a(i1, -k, j1));
+                                block = iblockstate.getBlock();
 
-                                if (iblockstate.func_185904_a() == Material.air || block == Blocks.dirt || block == Blocks.snow || block == Blocks.ice)
+                                if (iblockstate.getMaterial() == Material.air || block == Blocks.dirt || block == Blocks.snow || block == Blocks.ice)
                                 {
-                                    this.func_175903_a(p_180709_1_, p_180709_3_.func_177982_a(i1, -k, j1), Blocks.packed_ice.func_176223_P());
+                                    this.func_175903_a(p_180709_1_, p_180709_3_.func_177982_a(i1, -k, j1), Blocks.packed_ice.getBlockState());
                                 }
                             }
                         }
@@ -94,23 +94,23 @@
                         j2 = p_180709_2_.nextInt(5);
                     }
 
-                    while (blockpos.func_177956_o() > 50)
+                    while (blockpos.getY() > 50)
                     {
-                        IBlockState iblockstate1 = p_180709_1_.func_180495_p(blockpos);
-                        Block block1 = iblockstate1.func_177230_c();
+                        IBlockState iblockstate1 = p_180709_1_.getBlockState(blockpos);
+                        Block block1 = iblockstate1.getBlock();
 
-                        if (iblockstate1.func_185904_a() != Material.air && block1 != Blocks.dirt && block1 != Blocks.snow && block1 != Blocks.ice && block1 != Blocks.packed_ice)
+                        if (iblockstate1.getMaterial() != Material.air && block1 != Blocks.dirt && block1 != Blocks.snow && block1 != Blocks.ice && block1 != Blocks.packed_ice)
                         {
                             break;
                         }
 
-                        this.func_175903_a(p_180709_1_, blockpos, Blocks.packed_ice.func_176223_P());
-                        blockpos = blockpos.func_177977_b();
+                        this.func_175903_a(p_180709_1_, blockpos, Blocks.packed_ice.getBlockState());
+                        blockpos = blockpos.getBlockPosBelow();
                         --j2;
 
                         if (j2 <= 0)
                         {
-                            blockpos = blockpos.func_177979_c(p_180709_2_.nextInt(5) + 1);
+                            blockpos = blockpos.getBlockPosBelow(p_180709_2_.nextInt(5) + 1);
                             j2 = p_180709_2_.nextInt(5);
                         }
                     }
