--- a/net/minecraft/world/gen/feature/WorldGenEndIsland.java
+++ b/net/minecraft/world/gen/feature/WorldGenEndIsland.java
@@ -20,7 +20,7 @@
                 {
                     if ((float)(j * j + k * k) <= (f + 1.0F) * (f + 1.0F))
                     {
-                        this.func_175903_a(p_180709_1_, p_180709_3_.func_177982_a(j, i, k), Blocks.end_stone.func_176223_P());
+                        this.func_175903_a(p_180709_1_, p_180709_3_.func_177982_a(j, i, k), Blocks.end_stone.getBlockState());
                     }
                 }
             }
