--- a/net/minecraft/world/gen/feature/WorldGenCactus.java
+++ b/net/minecraft/world/gen/feature/WorldGenCactus.java
@@ -13,7 +13,7 @@
         {
             BlockPos blockpos = p_180709_3_.func_177982_a(p_180709_2_.nextInt(8) - p_180709_2_.nextInt(8), p_180709_2_.nextInt(4) - p_180709_2_.nextInt(4), p_180709_2_.nextInt(8) - p_180709_2_.nextInt(8));
 
-            if (p_180709_1_.func_175623_d(blockpos))
+            if (p_180709_1_.isEmpty(blockpos))
             {
                 int j = 1 + p_180709_2_.nextInt(p_180709_2_.nextInt(3) + 1);
 
@@ -21,7 +21,7 @@
                 {
                     if (Blocks.cactus.func_176586_d(p_180709_1_, blockpos))
                     {
-                        p_180709_1_.func_180501_a(blockpos.func_177981_b(k), Blocks.cactus.func_176223_P(), 2);
+                        p_180709_1_.func_180501_a(blockpos.getBlockPosAbove(k), Blocks.cactus.getBlockState(), 2);
                     }
                 }
             }
