--- a/net/minecraft/world/gen/feature/WorldGenBigTree.java
+++ b/net/minecraft/world/gen/feature/WorldGenBigTree.java
@@ -17,7 +17,7 @@
 {
     private Random field_175949_k;
     private World field_175946_l;
-    private BlockPos field_175947_m = BlockPos.field_177992_a;
+    private BlockPos field_175947_m = BlockPos.createDefault;
     int heightLimit;
     int height;
     double heightAttenuation = 0.618D;
@@ -57,10 +57,10 @@
             i = 1;
         }
 
-        int j = this.field_175947_m.func_177956_o() + this.height;
+        int j = this.field_175947_m.getY() + this.height;
         int k = this.heightLimit - this.leafDistanceLimit;
         this.field_175948_j = Lists.<WorldGenBigTree.FoliageCoordinates>newArrayList();
-        this.field_175948_j.add(new WorldGenBigTree.FoliageCoordinates(this.field_175947_m.func_177981_b(k), j));
+        this.field_175948_j.add(new WorldGenBigTree.FoliageCoordinates(this.field_175947_m.getBlockPosAbove(k), j));
 
         for (; k >= 0; --k)
         {
@@ -75,19 +75,19 @@
                     double d2 = d0 * Math.sin(d1) + 0.5D;
                     double d3 = d0 * Math.cos(d1) + 0.5D;
                     BlockPos blockpos = this.field_175947_m.func_177963_a(d2, (double)(k - 1), d3);
-                    BlockPos blockpos1 = blockpos.func_177981_b(this.leafDistanceLimit);
+                    BlockPos blockpos1 = blockpos.getBlockPosAbove(this.leafDistanceLimit);
 
                     if (this.func_175936_a(blockpos, blockpos1) == -1)
                     {
-                        int i1 = this.field_175947_m.func_177958_n() - blockpos.func_177958_n();
-                        int j1 = this.field_175947_m.func_177952_p() - blockpos.func_177952_p();
-                        double d4 = (double)blockpos.func_177956_o() - Math.sqrt((double)(i1 * i1 + j1 * j1)) * this.field_175944_d;
+                        int i1 = this.field_175947_m.getX() - blockpos.getX();
+                        int j1 = this.field_175947_m.getZ() - blockpos.getZ();
+                        double d4 = (double)blockpos.getY() - Math.sqrt((double)(i1 * i1 + j1 * j1)) * this.field_175944_d;
                         int k1 = d4 > (double)j ? j : (int)d4;
-                        BlockPos blockpos2 = new BlockPos(this.field_175947_m.func_177958_n(), k1, this.field_175947_m.func_177952_p());
+                        BlockPos blockpos2 = new BlockPos(this.field_175947_m.getX(), k1, this.field_175947_m.getZ());
 
                         if (this.func_175936_a(blockpos2, blockpos) == -1)
                         {
-                            this.field_175948_j.add(new WorldGenBigTree.FoliageCoordinates(blockpos, blockpos2.func_177956_o()));
+                            this.field_175948_j.add(new WorldGenBigTree.FoliageCoordinates(blockpos, blockpos2.getY()));
                         }
                     }
                 }
@@ -106,7 +106,7 @@
                 if (Math.pow((double)Math.abs(j) + 0.5D, 2.0D) + Math.pow((double)Math.abs(k) + 0.5D, 2.0D) <= (double)(p_181631_2_ * p_181631_2_))
                 {
                     BlockPos blockpos = p_181631_1_.func_177982_a(j, 0, k);
-                    Material material = this.field_175946_l.func_180495_p(blockpos).func_185904_a();
+                    Material material = this.field_175946_l.getBlockState(blockpos).getMaterial();
 
                     if (material == Material.air || material == Material.leaves)
                     {
@@ -154,39 +154,39 @@
     {
         for (int i = 0; i < this.leafDistanceLimit; ++i)
         {
-            this.func_181631_a(p_175940_1_.func_177981_b(i), this.leafSize(i), Blocks.leaves.func_176223_P().func_177226_a(BlockLeaves.field_176236_b, Boolean.valueOf(false)));
+            this.func_181631_a(p_175940_1_.getBlockPosAbove(i), this.leafSize(i), Blocks.leaves.getBlockState().func_177226_a(BlockLeaves.field_176236_b, Boolean.valueOf(false)));
         }
     }
 
     void func_175937_a(BlockPos p_175937_1_, BlockPos p_175937_2_, Block p_175937_3_)
     {
-        BlockPos blockpos = p_175937_2_.func_177982_a(-p_175937_1_.func_177958_n(), -p_175937_1_.func_177956_o(), -p_175937_1_.func_177952_p());
+        BlockPos blockpos = p_175937_2_.func_177982_a(-p_175937_1_.getX(), -p_175937_1_.getY(), -p_175937_1_.getZ());
         int i = this.func_175935_b(blockpos);
-        float f = (float)blockpos.func_177958_n() / (float)i;
-        float f1 = (float)blockpos.func_177956_o() / (float)i;
-        float f2 = (float)blockpos.func_177952_p() / (float)i;
+        float f = (float)blockpos.getX() / (float)i;
+        float f1 = (float)blockpos.getY() / (float)i;
+        float f2 = (float)blockpos.getZ() / (float)i;
 
         for (int j = 0; j <= i; ++j)
         {
             BlockPos blockpos1 = p_175937_1_.func_177963_a((double)(0.5F + (float)j * f), (double)(0.5F + (float)j * f1), (double)(0.5F + (float)j * f2));
             BlockLog.EnumAxis blocklog$enumaxis = this.func_175938_b(p_175937_1_, blockpos1);
-            this.func_175903_a(this.field_175946_l, blockpos1, p_175937_3_.func_176223_P().func_177226_a(BlockLog.field_176299_a, blocklog$enumaxis));
+            this.func_175903_a(this.field_175946_l, blockpos1, p_175937_3_.getBlockState().func_177226_a(BlockLog.field_176299_a, blocklog$enumaxis));
         }
     }
 
     private int func_175935_b(BlockPos p_175935_1_)
     {
-        int i = MathHelper.abs_int(p_175935_1_.func_177958_n());
-        int j = MathHelper.abs_int(p_175935_1_.func_177956_o());
-        int k = MathHelper.abs_int(p_175935_1_.func_177952_p());
+        int i = MathHelper.abs_int(p_175935_1_.getX());
+        int j = MathHelper.abs_int(p_175935_1_.getY());
+        int k = MathHelper.abs_int(p_175935_1_.getZ());
         return k > i && k > j ? k : (j > i ? j : i);
     }
 
     private BlockLog.EnumAxis func_175938_b(BlockPos p_175938_1_, BlockPos p_175938_2_)
     {
         BlockLog.EnumAxis blocklog$enumaxis = BlockLog.EnumAxis.Y;
-        int i = Math.abs(p_175938_2_.func_177958_n() - p_175938_1_.func_177958_n());
-        int j = Math.abs(p_175938_2_.func_177952_p() - p_175938_1_.func_177952_p());
+        int i = Math.abs(p_175938_2_.getX() - p_175938_1_.getX());
+        int j = Math.abs(p_175938_2_.getZ() - p_175938_1_.getZ());
         int k = Math.max(i, j);
 
         if (k > 0)
@@ -223,15 +223,15 @@
     void func_175942_c()
     {
         BlockPos blockpos = this.field_175947_m;
-        BlockPos blockpos1 = this.field_175947_m.func_177981_b(this.height);
+        BlockPos blockpos1 = this.field_175947_m.getBlockPosAbove(this.height);
         Block block = Blocks.log;
         this.func_175937_a(blockpos, blockpos1, block);
 
         if (this.field_175943_g == 2)
         {
-            this.func_175937_a(blockpos.func_177974_f(), blockpos1.func_177974_f(), block);
-            this.func_175937_a(blockpos.func_177974_f().func_177968_d(), blockpos1.func_177974_f().func_177968_d(), block);
-            this.func_175937_a(blockpos.func_177968_d(), blockpos1.func_177968_d(), block);
+            this.func_175937_a(blockpos.getBlockPosEast(), blockpos1.getBlockPosEast(), block);
+            this.func_175937_a(blockpos.getBlockPosEast().getBlockPosSouth(), blockpos1.getBlockPosEast().getBlockPosSouth(), block);
+            this.func_175937_a(blockpos.getBlockPosSouth(), blockpos1.getBlockPosSouth(), block);
         }
     }
 
@@ -240,9 +240,9 @@
         for (WorldGenBigTree.FoliageCoordinates worldgenbigtree$foliagecoordinates : this.field_175948_j)
         {
             int i = worldgenbigtree$foliagecoordinates.func_177999_q();
-            BlockPos blockpos = new BlockPos(this.field_175947_m.func_177958_n(), i, this.field_175947_m.func_177952_p());
+            BlockPos blockpos = new BlockPos(this.field_175947_m.getX(), i, this.field_175947_m.getZ());
 
-            if (!blockpos.equals(worldgenbigtree$foliagecoordinates) && this.leafNodeNeedsBase(i - this.field_175947_m.func_177956_o()))
+            if (!blockpos.equals(worldgenbigtree$foliagecoordinates) && this.leafNodeNeedsBase(i - this.field_175947_m.getY()))
             {
                 this.func_175937_a(blockpos, worldgenbigtree$foliagecoordinates, Blocks.log);
             }
@@ -251,11 +251,11 @@
 
     int func_175936_a(BlockPos p_175936_1_, BlockPos p_175936_2_)
     {
-        BlockPos blockpos = p_175936_2_.func_177982_a(-p_175936_1_.func_177958_n(), -p_175936_1_.func_177956_o(), -p_175936_1_.func_177952_p());
+        BlockPos blockpos = p_175936_2_.func_177982_a(-p_175936_1_.getX(), -p_175936_1_.getY(), -p_175936_1_.getZ());
         int i = this.func_175935_b(blockpos);
-        float f = (float)blockpos.func_177958_n() / (float)i;
-        float f1 = (float)blockpos.func_177956_o() / (float)i;
-        float f2 = (float)blockpos.func_177952_p() / (float)i;
+        float f = (float)blockpos.getX() / (float)i;
+        float f1 = (float)blockpos.getY() / (float)i;
+        float f2 = (float)blockpos.getZ() / (float)i;
 
         if (i == 0)
         {
@@ -267,7 +267,7 @@
             {
                 BlockPos blockpos1 = p_175936_1_.func_177963_a((double)(0.5F + (float)j * f), (double)(0.5F + (float)j * f1), (double)(0.5F + (float)j * f2));
 
-                if (!this.func_150523_a(this.field_175946_l.func_180495_p(blockpos1).func_177230_c()))
+                if (!this.func_150523_a(this.field_175946_l.getBlockState(blockpos1).getBlock()))
                 {
                     return j;
                 }
@@ -313,7 +313,7 @@
      */
     private boolean validTreeLocation()
     {
-        Block block = this.field_175946_l.func_180495_p(this.field_175947_m.func_177977_b()).func_177230_c();
+        Block block = this.field_175946_l.getBlockState(this.field_175947_m.getBlockPosBelow()).getBlock();
 
         if (block != Blocks.dirt && block != Blocks.grass && block != Blocks.farmland)
         {
@@ -321,7 +321,7 @@
         }
         else
         {
-            int i = this.func_175936_a(this.field_175947_m, this.field_175947_m.func_177981_b(this.heightLimit - 1));
+            int i = this.func_175936_a(this.field_175947_m, this.field_175947_m.getBlockPosAbove(this.heightLimit - 1));
 
             if (i == -1)
             {
@@ -345,7 +345,7 @@
 
         public FoliageCoordinates(BlockPos p_i45635_1_, int p_i45635_2_)
         {
-            super(p_i45635_1_.func_177958_n(), p_i45635_1_.func_177956_o(), p_i45635_1_.func_177952_p());
+            super(p_i45635_1_.getX(), p_i45635_1_.getY(), p_i45635_1_.getZ());
             this.field_178000_b = p_i45635_2_;
         }
 
