--- a/net/minecraft/world/gen/feature/WorldGenMinable.java
+++ b/net/minecraft/world/gen/feature/WorldGenMinable.java
@@ -32,12 +32,12 @@
     public boolean func_180709_b(World p_180709_1_, Random p_180709_2_, BlockPos p_180709_3_)
     {
         float f = p_180709_2_.nextFloat() * (float)Math.PI;
-        double d0 = (double)((float)(p_180709_3_.func_177958_n() + 8) + MathHelper.sin(f) * (float)this.numberOfBlocks / 8.0F);
-        double d1 = (double)((float)(p_180709_3_.func_177958_n() + 8) - MathHelper.sin(f) * (float)this.numberOfBlocks / 8.0F);
-        double d2 = (double)((float)(p_180709_3_.func_177952_p() + 8) + MathHelper.cos(f) * (float)this.numberOfBlocks / 8.0F);
-        double d3 = (double)((float)(p_180709_3_.func_177952_p() + 8) - MathHelper.cos(f) * (float)this.numberOfBlocks / 8.0F);
-        double d4 = (double)(p_180709_3_.func_177956_o() + p_180709_2_.nextInt(3) - 2);
-        double d5 = (double)(p_180709_3_.func_177956_o() + p_180709_2_.nextInt(3) - 2);
+        double d0 = (double)((float)(p_180709_3_.getX() + 8) + MathHelper.sin(f) * (float)this.numberOfBlocks / 8.0F);
+        double d1 = (double)((float)(p_180709_3_.getX() + 8) - MathHelper.sin(f) * (float)this.numberOfBlocks / 8.0F);
+        double d2 = (double)((float)(p_180709_3_.getZ() + 8) + MathHelper.cos(f) * (float)this.numberOfBlocks / 8.0F);
+        double d3 = (double)((float)(p_180709_3_.getZ() + 8) - MathHelper.cos(f) * (float)this.numberOfBlocks / 8.0F);
+        double d4 = (double)(p_180709_3_.getY() + p_180709_2_.nextInt(3) - 2);
+        double d5 = (double)(p_180709_3_.getY() + p_180709_2_.nextInt(3) - 2);
 
         for (int i = 0; i < this.numberOfBlocks; ++i)
         {
@@ -75,7 +75,7 @@
                                 {
                                     BlockPos blockpos = new BlockPos(l1, i2, j2);
 
-                                    if (this.field_175919_c.apply(p_180709_1_.func_180495_p(blockpos)))
+                                    if (this.field_175919_c.apply(p_180709_1_.getBlockState(blockpos)))
                                     {
                                         p_180709_1_.func_180501_a(blockpos, this.field_175920_a, 2);
                                     }
