--- a/net/minecraft/world/gen/feature/WorldGenDungeons.java
+++ b/net/minecraft/world/gen/feature/WorldGenDungeons.java
@@ -38,7 +38,7 @@
                 for (int i3 = l1; i3 <= i2; ++i3)
                 {
                     BlockPos blockpos = p_180709_3_.func_177982_a(k2, l2, i3);
-                    Material material = p_180709_1_.func_180495_p(blockpos).func_185904_a();
+                    Material material = p_180709_1_.getBlockState(blockpos).getMaterial();
                     boolean flag = material.isSolid();
 
                     if (l2 == -1 && !flag)
@@ -51,7 +51,7 @@
                         return false;
                     }
 
-                    if ((k2 == k || k2 == l || i3 == l1 || i3 == i2) && l2 == 0 && p_180709_1_.func_175623_d(blockpos) && p_180709_1_.func_175623_d(blockpos.func_177984_a()))
+                    if ((k2 == k || k2 == l || i3 == l1 || i3 == i2) && l2 == 0 && p_180709_1_.isEmpty(blockpos) && p_180709_1_.isEmpty(blockpos.getBlockPosAbove()))
                     {
                         ++j2;
                     }
@@ -71,24 +71,24 @@
 
                         if (k3 != k && i4 != -1 && k4 != l1 && k3 != l && i4 != 4 && k4 != i2)
                         {
-                            if (p_180709_1_.func_180495_p(blockpos1).func_177230_c() != Blocks.chest)
+                            if (p_180709_1_.getBlockState(blockpos1).getBlock() != Blocks.chest)
                             {
                                 p_180709_1_.func_175698_g(blockpos1);
                             }
                         }
-                        else if (blockpos1.func_177956_o() >= 0 && !p_180709_1_.func_180495_p(blockpos1.func_177977_b()).func_185904_a().isSolid())
+                        else if (blockpos1.getY() >= 0 && !p_180709_1_.getBlockState(blockpos1.getBlockPosBelow()).getMaterial().isSolid())
                         {
                             p_180709_1_.func_175698_g(blockpos1);
                         }
-                        else if (p_180709_1_.func_180495_p(blockpos1).func_185904_a().isSolid() && p_180709_1_.func_180495_p(blockpos1).func_177230_c() != Blocks.chest)
+                        else if (p_180709_1_.getBlockState(blockpos1).getMaterial().isSolid() && p_180709_1_.getBlockState(blockpos1).getBlock() != Blocks.chest)
                         {
                             if (i4 == -1 && p_180709_2_.nextInt(4) != 0)
                             {
-                                p_180709_1_.func_180501_a(blockpos1, Blocks.mossy_cobblestone.func_176223_P(), 2);
+                                p_180709_1_.func_180501_a(blockpos1, Blocks.mossy_cobblestone.getBlockState(), 2);
                             }
                             else
                             {
-                                p_180709_1_.func_180501_a(blockpos1, Blocks.cobblestone.func_176223_P(), 2);
+                                p_180709_1_.func_180501_a(blockpos1, Blocks.cobblestone.getBlockState(), 2);
                             }
                         }
                     }
@@ -99,18 +99,18 @@
             {
                 for (int j4 = 0; j4 < 3; ++j4)
                 {
-                    int l4 = p_180709_3_.func_177958_n() + p_180709_2_.nextInt(j * 2 + 1) - j;
-                    int i5 = p_180709_3_.func_177956_o();
-                    int j5 = p_180709_3_.func_177952_p() + p_180709_2_.nextInt(k1 * 2 + 1) - k1;
+                    int l4 = p_180709_3_.getX() + p_180709_2_.nextInt(j * 2 + 1) - j;
+                    int i5 = p_180709_3_.getY();
+                    int j5 = p_180709_3_.getZ() + p_180709_2_.nextInt(k1 * 2 + 1) - k1;
                     BlockPos blockpos2 = new BlockPos(l4, i5, j5);
 
-                    if (p_180709_1_.func_175623_d(blockpos2))
+                    if (p_180709_1_.isEmpty(blockpos2))
                     {
                         int j3 = 0;
 
                         for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
                         {
-                            if (p_180709_1_.func_180495_p(blockpos2.func_177972_a(enumfacing)).func_185904_a().isSolid())
+                            if (p_180709_1_.getBlockState(blockpos2.getBlockPosRelative(enumfacing)).getMaterial().isSolid())
                             {
                                 ++j3;
                             }
@@ -118,7 +118,7 @@
 
                         if (j3 == 1)
                         {
-                            p_180709_1_.func_180501_a(blockpos2, Blocks.chest.func_176458_f(p_180709_1_, blockpos2, Blocks.chest.func_176223_P()), 2);
+                            p_180709_1_.func_180501_a(blockpos2, Blocks.chest.func_176458_f(p_180709_1_, blockpos2, Blocks.chest.getBlockState()), 2);
                             TileEntity tileentity1 = p_180709_1_.func_175625_s(blockpos2);
 
                             if (tileentity1 instanceof TileEntityChest)
@@ -132,7 +132,7 @@
                 }
             }
 
-            p_180709_1_.func_180501_a(p_180709_3_, Blocks.mob_spawner.func_176223_P(), 2);
+            p_180709_1_.func_180501_a(p_180709_3_, Blocks.mob_spawner.getBlockState(), 2);
             TileEntity tileentity = p_180709_1_.func_175625_s(p_180709_3_);
 
             if (tileentity instanceof TileEntityMobSpawner)
@@ -141,7 +141,7 @@
             }
             else
             {
-                field_175918_a.error("Failed to fetch mob spawner entity at ({}, {}, {})", new Object[] {Integer.valueOf(p_180709_3_.func_177958_n()), Integer.valueOf(p_180709_3_.func_177956_o()), Integer.valueOf(p_180709_3_.func_177952_p())});
+                field_175918_a.error("Failed to fetch mob spawner entity at ({}, {}, {})", new Object[] {Integer.valueOf(p_180709_3_.getX()), Integer.valueOf(p_180709_3_.getY()), Integer.valueOf(p_180709_3_.getZ())});
             }
 
             return true;
