--- a/net/minecraft/world/gen/feature/WorldGenIcePath.java
+++ b/net/minecraft/world/gen/feature/WorldGenIcePath.java
@@ -18,12 +18,12 @@
 
     public boolean func_180709_b(World p_180709_1_, Random p_180709_2_, BlockPos p_180709_3_)
     {
-        while (p_180709_1_.func_175623_d(p_180709_3_) && p_180709_3_.func_177956_o() > 2)
+        while (p_180709_1_.isEmpty(p_180709_3_) && p_180709_3_.getY() > 2)
         {
-            p_180709_3_ = p_180709_3_.func_177977_b();
+            p_180709_3_ = p_180709_3_.getBlockPosBelow();
         }
 
-        if (p_180709_1_.func_180495_p(p_180709_3_).func_177230_c() != Blocks.snow)
+        if (p_180709_1_.getBlockState(p_180709_3_).getBlock() != Blocks.snow)
         {
             return false;
         }
@@ -32,23 +32,23 @@
             int i = p_180709_2_.nextInt(this.field_150554_b - 2) + 2;
             int j = 1;
 
-            for (int k = p_180709_3_.func_177958_n() - i; k <= p_180709_3_.func_177958_n() + i; ++k)
+            for (int k = p_180709_3_.getX() - i; k <= p_180709_3_.getX() + i; ++k)
             {
-                for (int l = p_180709_3_.func_177952_p() - i; l <= p_180709_3_.func_177952_p() + i; ++l)
+                for (int l = p_180709_3_.getZ() - i; l <= p_180709_3_.getZ() + i; ++l)
                 {
-                    int i1 = k - p_180709_3_.func_177958_n();
-                    int j1 = l - p_180709_3_.func_177952_p();
+                    int i1 = k - p_180709_3_.getX();
+                    int j1 = l - p_180709_3_.getZ();
 
                     if (i1 * i1 + j1 * j1 <= i * i)
                     {
-                        for (int k1 = p_180709_3_.func_177956_o() - 1; k1 <= p_180709_3_.func_177956_o() + 1; ++k1)
+                        for (int k1 = p_180709_3_.getY() - 1; k1 <= p_180709_3_.getY() + 1; ++k1)
                         {
                             BlockPos blockpos = new BlockPos(k, k1, l);
-                            Block block = p_180709_1_.func_180495_p(blockpos).func_177230_c();
+                            Block block = p_180709_1_.getBlockState(blockpos).getBlock();
 
                             if (block == Blocks.dirt || block == Blocks.snow || block == Blocks.ice)
                             {
-                                p_180709_1_.func_180501_a(blockpos, this.field_150555_a.func_176223_P(), 2);
+                                p_180709_1_.func_180501_a(blockpos, this.field_150555_a.getBlockState(), 2);
                             }
                         }
                     }
