--- a/net/minecraft/world/gen/feature/WorldGenEndPodium.java
+++ b/net/minecraft/world/gen/feature/WorldGenEndPodium.java
@@ -9,8 +9,8 @@
 
 public class WorldGenEndPodium extends WorldGenerator
 {
-    public static final BlockPos field_186139_a = BlockPos.field_177992_a;
-    public static final BlockPos field_186140_b = new BlockPos(field_186139_a.func_177958_n() - 4 & -16, 0, field_186139_a.func_177952_p() - 4 & -16);
+    public static final BlockPos field_186139_a = BlockPos.createDefault;
+    public static final BlockPos field_186140_b = new BlockPos(field_186139_a.getX() - 4 & -16, 0, field_186139_a.getZ() - 4 & -16);
     private final boolean field_186141_c;
 
     public WorldGenEndPodium(boolean p_i46666_1_)
@@ -20,52 +20,52 @@
 
     public boolean func_180709_b(World p_180709_1_, Random p_180709_2_, BlockPos p_180709_3_)
     {
-        for (BlockPos.MutableBlockPos blockpos$mutableblockpos : BlockPos.func_177975_b(new BlockPos(p_180709_3_.func_177958_n() - 4, p_180709_3_.func_177956_o() - 1, p_180709_3_.func_177952_p() - 4), new BlockPos(p_180709_3_.func_177958_n() + 4, p_180709_3_.func_177956_o() + 32, p_180709_3_.func_177952_p() + 4)))
+        for (BlockPos.MutableBlockPos blockpos$mutableblockpos : BlockPos.func_177975_b(new BlockPos(p_180709_3_.getX() - 4, p_180709_3_.getY() - 1, p_180709_3_.getZ() - 4), new BlockPos(p_180709_3_.getX() + 4, p_180709_3_.getY() + 32, p_180709_3_.getZ() + 4)))
         {
-            double d0 = blockpos$mutableblockpos.func_185332_f(p_180709_3_.func_177958_n(), blockpos$mutableblockpos.func_177956_o(), p_180709_3_.func_177952_p());
+            double d0 = blockpos$mutableblockpos.func_185332_f(p_180709_3_.getX(), blockpos$mutableblockpos.getY(), p_180709_3_.getZ());
 
             if (d0 <= 3.5D)
             {
-                if (blockpos$mutableblockpos.func_177956_o() < p_180709_3_.func_177956_o())
+                if (blockpos$mutableblockpos.getY() < p_180709_3_.getY())
                 {
                     if (d0 <= 2.5D)
                     {
-                        this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.bedrock.func_176223_P());
+                        this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.bedrock.getBlockState());
                     }
-                    else if (blockpos$mutableblockpos.func_177956_o() < p_180709_3_.func_177956_o())
+                    else if (blockpos$mutableblockpos.getY() < p_180709_3_.getY())
                     {
-                        this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.end_stone.func_176223_P());
+                        this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.end_stone.getBlockState());
                     }
                 }
-                else if (blockpos$mutableblockpos.func_177956_o() > p_180709_3_.func_177956_o())
+                else if (blockpos$mutableblockpos.getY() > p_180709_3_.getY())
                 {
-                    this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.air.func_176223_P());
+                    this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.air.getBlockState());
                 }
                 else if (d0 > 2.5D)
                 {
-                    this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.bedrock.func_176223_P());
+                    this.func_175903_a(p_180709_1_, blockpos$mutableblockpos, Blocks.bedrock.getBlockState());
                 }
                 else if (this.field_186141_c)
                 {
-                    this.func_175903_a(p_180709_1_, new BlockPos(blockpos$mutableblockpos), Blocks.end_portal.func_176223_P());
+                    this.func_175903_a(p_180709_1_, new BlockPos(blockpos$mutableblockpos), Blocks.end_portal.getBlockState());
                 }
                 else
                 {
-                    this.func_175903_a(p_180709_1_, new BlockPos(blockpos$mutableblockpos), Blocks.air.func_176223_P());
+                    this.func_175903_a(p_180709_1_, new BlockPos(blockpos$mutableblockpos), Blocks.air.getBlockState());
                 }
             }
         }
 
         for (int i = 0; i < 4; ++i)
         {
-            this.func_175903_a(p_180709_1_, p_180709_3_.func_177981_b(i), Blocks.bedrock.func_176223_P());
+            this.func_175903_a(p_180709_1_, p_180709_3_.getBlockPosAbove(i), Blocks.bedrock.getBlockState());
         }
 
-        BlockPos blockpos = p_180709_3_.func_177981_b(2);
+        BlockPos blockpos = p_180709_3_.getBlockPosAbove(2);
 
         for (EnumFacing enumfacing : EnumFacing.Plane.HORIZONTAL)
         {
-            this.func_175903_a(p_180709_1_, blockpos.func_177972_a(enumfacing), Blocks.torch.func_176223_P().func_177226_a(BlockTorch.field_176596_a, enumfacing));
+            this.func_175903_a(p_180709_1_, blockpos.getBlockPosRelative(enumfacing), Blocks.torch.getBlockState().func_177226_a(BlockTorch.field_176596_a, enumfacing));
         }
 
         return true;
