--- a/net/minecraft/world/gen/feature/WorldGenMelon.java
+++ b/net/minecraft/world/gen/feature/WorldGenMelon.java
@@ -13,9 +13,9 @@
         {
             BlockPos blockpos = p_180709_3_.func_177982_a(p_180709_2_.nextInt(8) - p_180709_2_.nextInt(8), p_180709_2_.nextInt(4) - p_180709_2_.nextInt(4), p_180709_2_.nextInt(8) - p_180709_2_.nextInt(8));
 
-            if (Blocks.melon_block.func_176196_c(p_180709_1_, blockpos) && p_180709_1_.func_180495_p(blockpos.func_177977_b()).func_177230_c() == Blocks.grass)
+            if (Blocks.melon_block.func_176196_c(p_180709_1_, blockpos) && p_180709_1_.getBlockState(blockpos.getBlockPosBelow()).getBlock() == Blocks.grass)
             {
-                p_180709_1_.func_180501_a(blockpos, Blocks.melon_block.func_176223_P(), 2);
+                p_180709_1_.func_180501_a(blockpos, Blocks.melon_block.getBlockState(), 2);
             }
         }
 
