--- a/net/minecraft/world/gen/feature/WorldGenSavannaTree.java
+++ b/net/minecraft/world/gen/feature/WorldGenSavannaTree.java
@@ -15,8 +15,8 @@
 
 public class WorldGenSavannaTree extends WorldGenAbstractTree
 {
-    private static final IBlockState field_181643_a = Blocks.log2.func_176223_P().func_177226_a(BlockNewLog.field_176300_b, BlockPlanks.EnumType.ACACIA);
-    private static final IBlockState field_181644_b = Blocks.leaves2.func_176223_P().func_177226_a(BlockNewLeaf.field_176240_P, BlockPlanks.EnumType.ACACIA).func_177226_a(BlockLeaves.field_176236_b, Boolean.valueOf(false));
+    private static final IBlockState field_181643_a = Blocks.log2.getBlockState().func_177226_a(BlockNewLog.field_176300_b, BlockPlanks.EnumType.ACACIA);
+    private static final IBlockState field_181644_b = Blocks.leaves2.getBlockState().func_177226_a(BlockNewLeaf.field_176240_P, BlockPlanks.EnumType.ACACIA).func_177226_a(BlockLeaves.field_176236_b, Boolean.valueOf(false));
 
     public WorldGenSavannaTree(boolean p_i45463_1_)
     {
@@ -28,31 +28,31 @@
         int i = p_180709_2_.nextInt(3) + p_180709_2_.nextInt(3) + 5;
         boolean flag = true;
 
-        if (p_180709_3_.func_177956_o() >= 1 && p_180709_3_.func_177956_o() + i + 1 <= 256)
+        if (p_180709_3_.getY() >= 1 && p_180709_3_.getY() + i + 1 <= 256)
         {
-            for (int j = p_180709_3_.func_177956_o(); j <= p_180709_3_.func_177956_o() + 1 + i; ++j)
+            for (int j = p_180709_3_.getY(); j <= p_180709_3_.getY() + 1 + i; ++j)
             {
                 int k = 1;
 
-                if (j == p_180709_3_.func_177956_o())
+                if (j == p_180709_3_.getY())
                 {
                     k = 0;
                 }
 
-                if (j >= p_180709_3_.func_177956_o() + 1 + i - 2)
+                if (j >= p_180709_3_.getY() + 1 + i - 2)
                 {
                     k = 2;
                 }
 
                 BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-                for (int l = p_180709_3_.func_177958_n() - k; l <= p_180709_3_.func_177958_n() + k && flag; ++l)
+                for (int l = p_180709_3_.getX() - k; l <= p_180709_3_.getX() + k && flag; ++l)
                 {
-                    for (int i1 = p_180709_3_.func_177952_p() - k; i1 <= p_180709_3_.func_177952_p() + k && flag; ++i1)
+                    for (int i1 = p_180709_3_.getZ() - k; i1 <= p_180709_3_.getZ() + k && flag; ++i1)
                     {
                         if (j >= 0 && j < 256)
                         {
-                            if (!this.func_150523_a(p_180709_1_.func_180495_p(blockpos$mutableblockpos.func_181079_c(l, j, i1)).func_177230_c()))
+                            if (!this.func_150523_a(p_180709_1_.getBlockState(blockpos$mutableblockpos.func_181079_c(l, j, i1)).getBlock()))
                             {
                                 flag = false;
                             }
@@ -71,21 +71,21 @@
             }
             else
             {
-                Block block = p_180709_1_.func_180495_p(p_180709_3_.func_177977_b()).func_177230_c();
+                Block block = p_180709_1_.getBlockState(p_180709_3_.getBlockPosBelow()).getBlock();
 
-                if ((block == Blocks.grass || block == Blocks.dirt) && p_180709_3_.func_177956_o() < 256 - i - 1)
+                if ((block == Blocks.grass || block == Blocks.dirt) && p_180709_3_.getY() < 256 - i - 1)
                 {
-                    this.func_175921_a(p_180709_1_, p_180709_3_.func_177977_b());
+                    this.func_175921_a(p_180709_1_, p_180709_3_.getBlockPosBelow());
                     EnumFacing enumfacing = EnumFacing.Plane.HORIZONTAL.func_179518_a(p_180709_2_);
                     int k2 = i - p_180709_2_.nextInt(4) - 1;
                     int l2 = 3 - p_180709_2_.nextInt(3);
-                    int i3 = p_180709_3_.func_177958_n();
-                    int j1 = p_180709_3_.func_177952_p();
+                    int i3 = p_180709_3_.getX();
+                    int j1 = p_180709_3_.getZ();
                     int k1 = 0;
 
                     for (int l1 = 0; l1 < i; ++l1)
                     {
-                        int i2 = p_180709_3_.func_177956_o() + l1;
+                        int i2 = p_180709_3_.getY() + l1;
 
                         if (l1 >= k2 && l2 > 0)
                         {
@@ -95,7 +95,7 @@
                         }
 
                         BlockPos blockpos = new BlockPos(i3, i2, j1);
-                        Material material = p_180709_1_.func_180495_p(blockpos).func_185904_a();
+                        Material material = p_180709_1_.getBlockState(blockpos).getMaterial();
 
                         if (material == Material.air || material == Material.leaves)
                         {
@@ -117,7 +117,7 @@
                         }
                     }
 
-                    blockpos2 = blockpos2.func_177984_a();
+                    blockpos2 = blockpos2.getBlockPosAbove();
 
                     for (int k3 = -1; k3 <= 1; ++k3)
                     {
@@ -127,12 +127,12 @@
                         }
                     }
 
-                    this.func_175924_b(p_180709_1_, blockpos2.func_177965_g(2));
-                    this.func_175924_b(p_180709_1_, blockpos2.func_177985_f(2));
-                    this.func_175924_b(p_180709_1_, blockpos2.func_177970_e(2));
-                    this.func_175924_b(p_180709_1_, blockpos2.func_177964_d(2));
-                    i3 = p_180709_3_.func_177958_n();
-                    j1 = p_180709_3_.func_177952_p();
+                    this.func_175924_b(p_180709_1_, blockpos2.getBlockPosEast(2));
+                    this.func_175924_b(p_180709_1_, blockpos2.getBlockPosWest(2));
+                    this.func_175924_b(p_180709_1_, blockpos2.getBlockPosSouth(2));
+                    this.func_175924_b(p_180709_1_, blockpos2.getBlockPosNorth(2));
+                    i3 = p_180709_3_.getX();
+                    j1 = p_180709_3_.getZ();
                     EnumFacing enumfacing1 = EnumFacing.Plane.HORIZONTAL.func_179518_a(p_180709_2_);
 
                     if (enumfacing1 != enumfacing)
@@ -145,11 +145,11 @@
                         {
                             if (l4 >= 1)
                             {
-                                int j2 = p_180709_3_.func_177956_o() + l4;
+                                int j2 = p_180709_3_.getY() + l4;
                                 i3 += enumfacing1.getFrontOffsetX();
                                 j1 += enumfacing1.getFrontOffsetZ();
                                 BlockPos blockpos1 = new BlockPos(i3, j2, j1);
-                                Material material1 = p_180709_1_.func_180495_p(blockpos1).func_185904_a();
+                                Material material1 = p_180709_1_.getBlockState(blockpos1).getMaterial();
 
                                 if (material1 == Material.air || material1 == Material.leaves)
                                 {
@@ -176,7 +176,7 @@
                                 }
                             }
 
-                            blockpos3 = blockpos3.func_177984_a();
+                            blockpos3 = blockpos3.getBlockPosAbove();
 
                             for (int j5 = -1; j5 <= 1; ++j5)
                             {
@@ -209,7 +209,7 @@
 
     private void func_175924_b(World p_175924_1_, BlockPos p_175924_2_)
     {
-        Material material = p_175924_1_.func_180495_p(p_175924_2_).func_185904_a();
+        Material material = p_175924_1_.getBlockState(p_175924_2_).getMaterial();
 
         if (material == Material.air || material == Material.leaves)
         {
