--- a/net/minecraft/world/gen/feature/WorldGenAbstractTree.java
+++ b/net/minecraft/world/gen/feature/WorldGenAbstractTree.java
@@ -16,7 +16,7 @@
 
     protected boolean func_150523_a(Block p_150523_1_)
     {
-        Material material = p_150523_1_.func_176223_P().func_185904_a();
+        Material material = p_150523_1_.getBlockState().getMaterial();
         return material == Material.air || material == Material.leaves || p_150523_1_ == Blocks.grass || p_150523_1_ == Blocks.dirt || p_150523_1_ == Blocks.log || p_150523_1_ == Blocks.log2 || p_150523_1_ == Blocks.sapling || p_150523_1_ == Blocks.vine;
     }
 
@@ -26,9 +26,9 @@
 
     protected void func_175921_a(World p_175921_1_, BlockPos p_175921_2_)
     {
-        if (p_175921_1_.func_180495_p(p_175921_2_).func_177230_c() != Blocks.dirt)
+        if (p_175921_1_.getBlockState(p_175921_2_).getBlock() != Blocks.dirt)
         {
-            this.func_175903_a(p_175921_1_, p_175921_2_, Blocks.dirt.func_176223_P());
+            this.func_175903_a(p_175921_1_, p_175921_2_, Blocks.dirt.getBlockState());
         }
     }
 }
